From 4fd32aee8a9fc0d58e615a9de445d31038915f9d Mon Sep 17 00:00:00 2001
From: wlf <wulf@rock-chips.com>
Date: Wed, 11 Jun 2014 10:45:44 +0800
Subject: [PATCH] USB: fix usb gadget ffs string id bug.

---
 drivers/usb/gadget/android.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index aade8773c8a2..58c2b3c1c4d1 100755
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -347,6 +347,9 @@ static void functionfs_closed_callback(struct ffs_data *ffs)
 	config->opened = false;
 	config->data = NULL;
 
+	if (!WARN_ON(!ffs->gadget)) {
+		dev->cdev->next_string_id -= ffs->strings_count;
+	}
 	functionfs_unbind(ffs);
 
 	mutex_unlock(&dev->mutex);
-- 
2.35.3

