From 7add70c62f722dd205ec62c94995599428b16e1f Mon Sep 17 00:00:00 2001
From: zyc <zyc@rock-chips.com>
Date: Mon, 24 Dec 2012 13:03:21 +0800
Subject: [PATCH] camera:query max camera resolution failed ,fix it

---
 drivers/media/video/rk30_camera_oneframe.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/video/rk30_camera_oneframe.c b/drivers/media/video/rk30_camera_oneframe.c
index eed218f6bbb7..912dc7fe9b15 100755
--- a/drivers/media/video/rk30_camera_oneframe.c
+++ b/drivers/media/video/rk30_camera_oneframe.c
@@ -2219,6 +2219,12 @@ static int rk_camera_try_fmt(struct soc_camera_device *icd,
 	if (ret < 0)
 		goto RK_CAMERA_TRY_FMT_END;
     RKCAMERA_DG("%s mf.width:%d  mf.height:%d\n",__FUNCTION__,mf.width,mf.height);
+	//query resolution.
+	if((usr_w == 10000) && (usr_h == 10000)) {
+		pix->width = mf.width;
+        pix->height = mf.height;
+		goto RK_CAMERA_TRY_FMT_END;
+	}
 	#ifdef CONFIG_VIDEO_RK29_WORK_IPP       
 	if ((mf.width != usr_w) || (mf.height != usr_h)) {
         bytes_per_line_host = soc_mbus_bytes_per_line(mf.width,icd->current_fmt->host_fmt); 
-- 
2.35.3

