From 5893c39ffaca26b503ced7af8ec3f066379bdb01 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Feb 2012 14:34:30 +0800
Subject: [PATCH] rk30: clock: add clk_set_rate

---
 arch/arm/mach-rk30/clock.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/mach-rk30/clock.c b/arch/arm/mach-rk30/clock.c
index 8982ed56f9bd..06dce19f3018 100644
--- a/arch/arm/mach-rk30/clock.c
+++ b/arch/arm/mach-rk30/clock.c
@@ -98,3 +98,14 @@ unsigned long clk_get_rate(struct clk *clk)
 	return 24000000;
 }
 EXPORT_SYMBOL(clk_get_rate);
+
+int clk_set_rate(struct clk *clk, unsigned long rate)
+{
+	int ret = -EINVAL;
+
+	if (clk == NULL || IS_ERR(clk))
+		return ret;
+
+	return ret;
+}
+EXPORT_SYMBOL(clk_set_rate);
-- 
2.35.3

