From 0b96f0d2db3a859ab37cfe5feb87181904c02eff Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 11 Apr 2011 11:12:36 +0800
Subject: [PATCH] update sdmmc driver: add some debug info

---
 drivers/mmc/host/rk29_sdmmc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/rk29_sdmmc.c b/drivers/mmc/host/rk29_sdmmc.c
index 6a1b7c929d28..d45b952f79ec 100755
--- a/drivers/mmc/host/rk29_sdmmc.c
+++ b/drivers/mmc/host/rk29_sdmmc.c
@@ -651,6 +651,7 @@ static void rk29_sdmmc_queue_request(struct rk29_sdmmc *host,struct mmc_request
 		host->state = STATE_SENDING_CMD;
 		rk29_sdmmc_start_request(host);
 	} else {
+		dev_info(&host->pdev->dev, "list add tail\n");
 		list_add_tail(&host->queue_node, &host->queue);
 	}
 	spin_unlock(&host->lock);
@@ -782,6 +783,7 @@ static void rk29_sdmmc_request_end(struct rk29_sdmmc *host, struct mmc_request *
 		rk29_sdmmc_reset_fifo(host);
 	}
 	if (!list_empty(&host->queue)) {
+		dev_info(&host->pdev->dev, "queque list is not empty\n");
 		host = list_entry(host->queue.next,
 				struct rk29_sdmmc, queue_node);
 		list_del(&host->queue_node);
@@ -1367,6 +1369,7 @@ static void rk29_sdmmc_detect_change(unsigned long data)
 
 				rk29_sdmmc_request_end(host, mrq);
 			} else {
+				dev_info(&host->pdev->dev, "mrq != host->curr_mrq")
 				if (host->queue_node.next && host->queue_node.prev)
 					list_del(&host->queue_node);
 				mrq->cmd->error = -ENOMEDIUM;
-- 
2.35.3

