From 8eaf8928886744ba971036e0682a5c79696da8cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Mon, 21 Apr 2014 14:36:47 +0800
Subject: [PATCH] USB_AVDD_1V8: optimizing the usb1v8 power consumption

---
 arch/arm/mach-rockchip/pm-rk3288.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-rockchip/pm-rk3288.c b/arch/arm/mach-rockchip/pm-rk3288.c
index 6b45b8d7663b..304a907b67ca 100755
--- a/arch/arm/mach-rockchip/pm-rk3288.c
+++ b/arch/arm/mach-rockchip/pm-rk3288.c
@@ -25,6 +25,8 @@
 //#include "sram.h"
 #include "pm-pie.c"
 
+__weak void rk_usb_power_down(void);
+__weak void rk_usb_power_up(void);
 
 static void ddr_pin_set_pull(u8 port,u8 bank,u8 b_gpio,u8 fun);
 static void ddr_gpio_set_in_output(u8 port,u8 bank,u8 b_gpio,u8 type);
@@ -1133,6 +1135,8 @@ static inline void  rkpm_peri_resume_first(u32 power_mode)
 
 static void rkpm_slp_setting(void)
 {
+	rk_usb_power_down();
+
    //rkpm_gic_disable(130);
   //  rkpm_gic_disable(132);
 
@@ -1143,6 +1147,7 @@ static void rkpm_slp_setting(void)
 
 static void rkpm_save_setting_resume_first(void)
 {
+	rk_usb_power_down();
         rkpm_peri_resume_first(rk3288_powermode);
         
         // rkpm_ddr_printhex(cru_readl(RK3288_CRU_MODE_CON));
-- 
2.35.3

