From c48baf6be57e5391484f9c4b05efacb09f4ecc45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 20 Feb 2012 14:28:26 +0800
Subject: [PATCH] arm: pl330: revert REVISION change

---
 arch/arm/common/pl330.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/common/pl330.c b/arch/arm/common/pl330.c
index e510a75dd222..955b72760a34 100644
--- a/arch/arm/common/pl330.c
+++ b/arch/arm/common/pl330.c
@@ -145,7 +145,7 @@
 
 #define	PART		0x330
 #define DESIGNER	0x41
-#define REVISION	0x1
+#define REVISION	0x0
 #define INTEG_CFG	0x0
 #define PERIPH_ID_VAL	((PART << 0) | (DESIGNER << 12))
 
-- 
2.35.3

