From 71bdafdf309bf5b29dc3debbb9170ef56787b2d7 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Wed, 17 Apr 2013 20:02:30 +0800
Subject: [PATCH] NL80211_TESTMODE depends on CFG80211 && (MT5931 ||
 MT5931_MT6622)

---
 net/wireless/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/Kconfig b/net/wireless/Kconfig
index 8e2a668c9230..9898263ac94f 100644
--- a/net/wireless/Kconfig
+++ b/net/wireless/Kconfig
@@ -31,7 +31,7 @@ config CFG80211
 
 config NL80211_TESTMODE
 	bool "nl80211 testmode command"
-	depends on CFG80211
+	depends on CFG80211 && (MT5931 || MT5931_MT6622)
 	help
 	  The nl80211 testmode command helps implementing things like
 	  factory calibration or validation tools for wireless chips.
-- 
2.35.3

