From 537290edac96e1e92db21789dc7d955ab6e93814 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 6 Feb 2013 13:35:00 +0800
Subject: [PATCH] rk fb:set xvir for exteranl display when set main fb xvir
 change

---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index b7aeb7fa4255..1f17becf8111 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -552,6 +552,7 @@ static int rk_fb_set_par(struct fb_info *info)
 					par2->xact = par->xact;
 					par2->yact = par->yact;
 					par2->format = par->format;
+					par2->xvir = par->xvir;
 					info2->var.nonstd &= 0xffffff00;
 					info2->var.nonstd |= data_format;
 					dev_drv1->set_par(dev_drv1,layer_id);
-- 
2.35.3

