From b392850b9d541d868419d3645d5d2c5589b429ae Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 4 Jul 2014 10:23:16 +0800
Subject: [PATCH] 3036: add CRU_MODE_CON offset

---
 include/linux/rockchip/cru.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/rockchip/cru.h b/include/linux/rockchip/cru.h
index 74d03bb36dcb..46cdb6fca8b5 100755
--- a/include/linux/rockchip/cru.h
+++ b/include/linux/rockchip/cru.h
@@ -331,6 +331,8 @@ static inline void rk3288_cru_set_soft_reset(enum rk3288_cru_soft_reset idx, boo
 	dsb();
 }
 
+#define RK3036_CRU_MODE_CON 0x0040
+
 #define RK3036_CRU_GLB_SRST_FST_VALUE 0x00100
 #define RK3036_CRU_GLB_SRST_SND_VALUE 0x00104
 #define RK3036_CRU_SOFTRST0_CON 0x00110
-- 
2.35.3

