From e43caac69de701ee33f3dd51ecd93a9d89993d5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 27 Jan 2011 11:17:05 +0800
Subject: [PATCH] input: eeti_egalax_i2c: do not wake up system by default

---
 drivers/input/touchscreen/eeti_egalax_i2c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/eeti_egalax_i2c.c b/drivers/input/touchscreen/eeti_egalax_i2c.c
index 0124c767b69a..f16067d690c5 100644
--- a/drivers/input/touchscreen/eeti_egalax_i2c.c
+++ b/drivers/input/touchscreen/eeti_egalax_i2c.c
@@ -606,7 +606,7 @@ if (pdata->init_platform_hw)
 	TS_DEBUG("egalax_i2c request irq(%d) gpio(%d) with result:%d\n", p_egalax_i2c_dev->irq, gpio, ret);
 
 #ifdef CONFIG_PM
-	device_init_wakeup(&client->dev, 1);
+	device_init_wakeup(&client->dev, 0);
 #endif
 
 	printk(KERN_DEBUG "[egalax_i2c]: probe done\n");
-- 
2.35.3

