From 490802f8f8bf5e4da8c1a984e31cc9defac47c0c Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Sat, 1 Dec 2012 14:47:40 +0800
Subject: [PATCH] rk fb: only load screeen when needed

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 533a57e29256..1f0068a678e3 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1088,7 +1088,6 @@ int rk_fb_register(struct rk_lcdc_device_driver *dev_drv,
 	}
 		
 	dev_drv->init_lcdc(dev_drv);
-	dev_drv->load_screen(dev_drv,1);
 	/************fb set,one layer one fb ***********/
 	dev_drv->fb_index_base = fb_inf->num_fb;
 	for(i=0;i<dev_drv->num_layer;i++)
@@ -1147,6 +1146,7 @@ int rk_fb_register(struct rk_lcdc_device_driver *dev_drv,
     if(id == 0)
     {
 	    fb_inf->fb[0]->fbops->fb_open(fb_inf->fb[0],1);
+	    dev_drv->load_screen(dev_drv,1);
 	    fb_inf->fb[0]->fbops->fb_set_par(fb_inf->fb[0]);
 
 #if  defined(CONFIG_LOGO_LINUX_BMP)
-- 
2.35.3

