From 086e6833b840a4ce20aeb75886e013b75d427315 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 24 Sep 2012 11:48:00 +0800
Subject: [PATCH] rk2928: add ddr_freq param

---
 arch/arm/mach-rk2928/common.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/common.c b/arch/arm/mach-rk2928/common.c
index 5968a7557eb5..a5f8c904e6c6 100755
--- a/arch/arm/mach-rk2928/common.c
+++ b/arch/arm/mach-rk2928/common.c
@@ -114,6 +114,14 @@ void __init rk2928_init_irq(void)
 extern void __init rk2928_map_common_io(void);
 extern int __init clk_disable_unused(void);
 
+static unsigned int __initdata ddr_freq = DDR_FREQ;
+static int __init ddr_freq_setup(char *str)
+{
+	get_option(&str, &ddr_freq);
+	return 0;
+}
+early_param("ddr_freq", ddr_freq_setup);
+
 void __init rk2928_map_io(void)
 {
 	rk2928_map_common_io();
@@ -135,7 +143,7 @@ void __init rk2928_map_io(void)
 	rk29_sram_init();
 	board_clock_init();
 	rk2928_l2_cache_init();
-	ddr_init(DDR_TYPE, DDR_FREQ);
+	ddr_init(DDR_TYPE, ddr_freq);
 //	clk_disable_unused();
 	rk2928_iomux_init();
 	rk2928_boot_mode_init();
-- 
2.35.3

