From 2a6d243d715984f3a2df6344fffa157eaf696bcf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 24 May 2013 16:28:19 +0800
Subject: [PATCH] rk30: pm: fix link error when build without KEYS_RK29

---
 arch/arm/mach-rk30/pm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index aa6009cfb637..6ff5a64d901e 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -1076,12 +1076,13 @@ static int rk30_pm_prepare(void)
 static void rk30_pm_finish(void)
 {
 	enable_hlt();
+#ifdef CONFIG_KEYS_RK29
 	if(rk_soc_pm_ctr_bits_check(1<<RK_PM_CTR_WAKE_UP_KEY))
 	{
 		rk28_send_wakeup_key();
 		printk("rk30_pm_finish rk28_send_wakeup_key\n");
 	}
-
+#endif
 }
 
 static struct platform_suspend_ops rk30_pm_ops = {
-- 
2.35.3

