From fc5c15eaa31d009b7708716e1937270a2aca64d5 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Sat, 29 Jun 2013 17:01:28 +0800
Subject: [PATCH] rk3188: uboot: open pd_lcdc0/1 for uboot display

---
 arch/arm/mach-rk3188/clock_data.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk3188/clock_data.c b/arch/arm/mach-rk3188/clock_data.c
index d51fe8e58439..4d0f734e608b 100755
--- a/arch/arm/mach-rk3188/clock_data.c
+++ b/arch/arm/mach-rk3188/clock_data.c
@@ -3050,6 +3050,8 @@ static void __init rk30_init_enable_clocks(void)
 		clk_enable_nolock(&clk_aclk_lcdc1);
 		clk_enable_nolock(&aclk_lcdc0_pre);
 		clk_enable_nolock(&aclk_lcdc1_pre);
+		clk_enable_nolock(&pd_lcdc0);
+		clk_enable_nolock(&pd_lcdc1);
 	}
 	#if 0
 	 clk_enable_nolock(&clk_gpu);
-- 
2.35.3

