From a8c373dc42b52dc87e54ab04191201505e8ee335 Mon Sep 17 00:00:00 2001
From: ywj <ywj@rock-chips.com>
Date: Sat, 2 Feb 2013 10:21:21 +0800
Subject: [PATCH] bmp Full resolution support for factoryTool 4.0

---
 drivers/video/rockchip/rk_fb.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index d632d8b5cefe..ba3a2643806a 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1096,15 +1096,11 @@ static void fb_show_bmp_logo(struct fb_info *info, int rotate)
 	unsigned char *src=bmp_logo->data;
 	unsigned char *dst=info->screen_base;
 	int i;
-
-	if(bmp_logo->width>info->var.xres)
-		bmp_logo->width=info->var.xres;
-
-	if(bmp_logo->height>info->var.yres)
-		bmp_logo->height=info->var.yres;
-
-	for(i=0;i<bmp_logo->height;i++)
-		memcpy(dst+info->var.xres*i*4, src+bmp_logo->width*i*4, bmp_logo->width*4);
+	unsigned int Needwidth=(*(src-24)<<8)|(*(src-23));
+	unsigned int Needheight=(*(src-22)<<8)|(*(src-21));
+		
+	for(i=0;i<Needheight;i++)
+		memcpy(dst+info->var.xres*i*4, src+bmp_logo->width*i*4, Needwidth*4);
 	
 }
 #endif
-- 
2.35.3

