From 5cc24fbb15ec84ee064d67216d62b10fb4760790 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Sat, 19 Oct 2013 16:32:02 +0800
Subject: [PATCH] fiq debugger: change 66 to IRQ_UART0

---
 kernel/irq/chip.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/kernel/irq/chip.c b/kernel/irq/chip.c
index b1c616e81b8e..923aeca01fa7 100644
--- a/kernel/irq/chip.c
+++ b/kernel/irq/chip.c
@@ -18,6 +18,10 @@
 
 #include "internals.h"
 
+#ifdef CONFIG_FIQ_DEBUGGER
+#include <mach/irqs.h>
+#endif
+
 /**
  *	irq_set_chip - set the irq chip for an irq
  *	@irq:	irq number
@@ -410,7 +414,7 @@ handle_fasteoi_irq(unsigned int irq, struct irq_desc *desc)
 	if (unlikely(!desc->action || irqd_irq_disabled(&desc->irq_data))) {
 		desc->istate |= IRQS_PENDING;
 #ifdef CONFIG_FIQ_DEBUGGER
-		if(irq != (CONFIG_RK_DEBUG_UART + 66))
+		if(irq != (CONFIG_RK_DEBUG_UART + IRQ_UART0))
 #endif
 			mask_irq(desc);
 		goto out;
-- 
2.35.3

