From 0b843f131e34f543fa6fb6755b65a3485973e9b1 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 7 Aug 2012 07:09:57 -0700
Subject: [PATCH] rk2928: sdmmc: f_max = 24M(normal sd)

---
 drivers/mmc/host/rk29_sdmmc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mmc/host/rk29_sdmmc.c b/drivers/mmc/host/rk29_sdmmc.c
index 952b51c1dde1..09eb4058d038 100755
--- a/drivers/mmc/host/rk29_sdmmc.c
+++ b/drivers/mmc/host/rk29_sdmmc.c
@@ -3326,7 +3326,11 @@ static int rk29_sdmmc_probe(struct platform_device *pdev)
 #else
     if(RK29_CTRL_SDMMC_ID== host->pdev->id)
     {
+#ifdef CONFIG_ARCH_RK2928
+	mmc->f_max = SD_FPP_FREQ;
+#else
         mmc->f_max = SDHC_FPP_FREQ;
+#endif
     }
     else
     {
-- 
2.35.3

