From 4777c18c246372f4cf03a92ab6c00f83b21b6ca3 Mon Sep 17 00:00:00 2001
From: eddie <cf@rock-chips.com>
Date: Wed, 24 Aug 2011 17:43:35 +0800
Subject: [PATCH] remove newton default

---
 drivers/input/keyboard/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/input/keyboard/Kconfig b/drivers/input/keyboard/Kconfig
index 3dee66eebc5a..e453b1be9937 100644
--- a/drivers/input/keyboard/Kconfig
+++ b/drivers/input/keyboard/Kconfig
@@ -21,7 +21,6 @@ config KEYS_RK29
 config KEYS_RK29_NEWTON
 	tristate "rk29 newton keyboard"
 	depends on ARCH_RK29
-	default y
 	help
 		rk29 newton keyboard drivers(gpio and adc)
 
-- 
2.35.3

