From fcd7a57c9a6b723d7d2c2b8cb16e0ff8fbb36c14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 4 Jul 2013 10:17:40 +0800
Subject: [PATCH] serial: rk3188: Kconfig disable Serial port 2 CTS/RTS support

---
 drivers/tty/serial/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/Kconfig b/drivers/tty/serial/Kconfig
index 814f55a56583..1a2f6e535689 100644
--- a/drivers/tty/serial/Kconfig
+++ b/drivers/tty/serial/Kconfig
@@ -1555,7 +1555,7 @@ config UART2_RK29
 
 config UART2_CTS_RTS_RK29
 	bool "Serial port 2 CTS/RTS support"
-	depends on UART2_RK29 && !ARCH_RK30
+	depends on UART2_RK29 && !ARCH_RK30 && !ARCH_RK3188
 
 config UART2_DMA_RK29
 	int "Serial port 2 DMA support (EXPERIMENTAL)"
-- 
2.35.3

