From fa273d24511c0f4e4f4e14eb4cdf748653f34690 Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Wed, 20 Aug 2014 11:46:57 +0800
Subject: [PATCH] rga fix scale down ratio over 2 error

---
 drivers/video/rockchip/rga/rga_drv.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index acad02776311..7664cafa1e6d 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -534,6 +534,8 @@ static struct rga_reg * rga_reg_init_2(rga_session *session, struct rga_req *req
         INIT_LIST_HEAD(&reg1->session_link);
     	INIT_LIST_HEAD(&reg1->status_link);
 
+        req0->mmu_info.mmu_flag &= (~(1 << 10));
+
         if(req0->mmu_info.mmu_en)
         {
             ret = rga_set_mmu_info(reg0, req0);
@@ -545,6 +547,8 @@ static struct rga_reg * rga_reg_init_2(rga_session *session, struct rga_req *req
 
         RGA_gen_reg_info(req0, (uint8_t *)reg0->cmd_reg);
 
+        req1->mmu_info.mmu_flag &= (~(1 << 8));
+
         if(req1->mmu_info.mmu_en)
         {
             ret = rga_set_mmu_info(reg1, req1);
-- 
2.35.3

