From 6c018993f71a82b4e6b074434723aae2774adcc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 21 Aug 2012 18:06:42 +0800
Subject: [PATCH] rk: add RK_USB_UART config

---
 arch/arm/plat-rk/Kconfig | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 4d74efde0397..5aa49d777a93 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -159,10 +159,15 @@ config RK_EARLY_PRINTK
 config RK_DEBUG_UART
 	int "Debug UART"
 	default 1 if ARCH_RK29
-	default 2 if ARCH_RK30
+	default 2 if ARCH_RK30 || ARCH_RK2928
 	help
 	  Select a UART for debugging. -1 disable.
 
+config RK_USB_UART
+	bool "Support USB UART Bypass Function"
+	depends on ARCH_RK2928 && (RK_DEBUG_UART = 2)
+	default y
+
 config RK_CONSOLE_THREAD
 	bool "Console write by thread"
 	depends on FIQ_DEBUGGER_CONSOLE
-- 
2.35.3

