From 4be091f4ef1e67163a622ca1be7d0e041a9f601f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 18 Jan 2011 16:30:50 +0800
Subject: [PATCH] rk29: clock: enable ddr clock

---
 arch/arm/mach-rk29/clock.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 25eb8135cfbb..ad98a7808c98 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -281,7 +281,7 @@ static void delay_500ns(void)
 {
 	int delay = 2000;
 	while (delay--)
-           barrier();
+		barrier();
 }
 
 static void delay_300us(void)
@@ -2266,6 +2266,7 @@ static void __init clk_enable_init_clocks(void)
 	clk_enable_nolock(&clk_dma1);
 	clk_enable_nolock(&clk_emem);
 	clk_enable_nolock(&clk_intmem);
+	clk_enable_nolock(&clk_ddr);
 #ifdef CONFIG_DEBUG_LL
 	clk_enable_nolock(&clk_uart1);
 #endif
-- 
2.35.3

