From 7aebc796272601bab888ae4822e0b871012a5cea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Mon, 18 Apr 2011 21:38:50 -0700
Subject: [PATCH] if select bq27510, select I2C_DEV_RK29 at the same time

---
 drivers/power/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/power/Kconfig b/drivers/power/Kconfig
index d5185b06efb9..a5d37157e344 100755
--- a/drivers/power/Kconfig
+++ b/drivers/power/Kconfig
@@ -118,6 +118,7 @@ config CHARGER_PCF50633
 
 config BATTERY_STC3100
 	tristate "STC3100 battery driver"
+	select I2C_DEV_RK29
 	depends on I2C && ARCH_RK29
 	help
 	  Say Y here to enable support for batteries with STC3100(I2C) chip.
-- 
2.35.3

