From acc6ec5afe7c2eac9aca24cb9470bf0493fcfcec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 29 Mar 2012 15:42:27 +0800
Subject: [PATCH] rk: camera: fix Makefile for pack kernel

---
 drivers/media/video/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/video/Makefile b/drivers/media/video/Makefile
index 85e7797e4dac..f2d11b6d83a2 100755
--- a/drivers/media/video/Makefile
+++ b/drivers/media/video/Makefile
@@ -189,10 +189,14 @@ obj-$(CONFIG_VIDEO_MX1)			+= mx1_camera.o
 obj-$(CONFIG_VIDEO_MX2)			+= mx2_camera.o
 obj-$(CONFIG_VIDEO_MX3)			+= mx3_camera.o
 obj-$(CONFIG_VIDEO_PXA27x)		+= pxa_camera.o
+ifeq ($(CONFIG_ARCH_RK30),y)
 obj-$(CONFIG_VIDEO_RK29_WORK_ONEFRAME)	+= rk30_camera_oneframe.o
 obj-$(CONFIG_VIDEO_RK29_WORK_PINGPONG)	+= rk30_camera_pingpong.o
+endif
+ifeq ($(CONFIG_ARCH_RK29),y)
 obj-$(CONFIG_VIDEO_RK29_WORK_ONEFRAME)    += rk29_camera_oneframe.o
 obj-$(CONFIG_VIDEO_RK29_WORK_PINGPONG)    += rk29_camera_pingpong.o
+endif
 
 obj-$(CONFIG_VIDEO_RK29XX_VOUT)		+= rk29xx/
 obj-$(CONFIG_VIDEO_SH_MOBILE_CSI2)	+= sh_mobile_csi2.o
-- 
2.35.3

