From 29e8202db5dee30b4a39cccc8711106b6f1fe9c4 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Thu, 13 Sep 2012 10:55:15 +0800
Subject: [PATCH] rk3066b: fix 3066b rk610 reset err

---
 drivers/mfd/rk610-core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/rk610-core.c b/drivers/mfd/rk610-core.c
index daad513a74c8..8ce5c2b5f73b 100755
--- a/drivers/mfd/rk610-core.c
+++ b/drivers/mfd/rk610-core.c
@@ -255,7 +255,10 @@ static int rk610_control_probe(struct i2c_client *client,
 	    else {
 	    	DBG("rk610_control_probe request gpio ok\n");
 	    	gpio_direction_output(RK610_RESET_PIN, GPIO_HIGH);
-		    gpio_direction_output(RK610_RESET_PIN, GPIO_LOW);
+#if defined(CONFIG_ARCH_RK3066B)
+			msleep(100);
+#endif
+			gpio_direction_output(RK610_RESET_PIN, GPIO_LOW);
 			msleep(100);
 		    gpio_set_value(RK610_RESET_PIN, GPIO_HIGH);
 		}
-- 
2.35.3

