From d38e801e1df97bda3fb56539401bd1b9d608ba44 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Mon, 28 Jul 2014 16:26:04 +0800
Subject: [PATCH] rk: ion: iommu: fix build error

---
 drivers/staging/android/ion/ion.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/staging/android/ion/ion.c b/drivers/staging/android/ion/ion.c
index 556a3e80cdc6..c26e3cee6dd4 100755
--- a/drivers/staging/android/ion/ion.c
+++ b/drivers/staging/android/ion/ion.c
@@ -928,6 +928,16 @@ static int ion_debug_client_show_buffer_map(struct seq_file *s, struct ion_buffe
 
 	return 0;
 }
+#else
+int ion_map_iommu(struct device *iommu_dev, struct ion_client *client,
+                struct ion_handle *handle, unsigned long *iova, unsigned long *size)
+{
+       return 0;
+}
+void ion_unmap_iommu(struct device *iommu_dev, struct ion_client *client,
+                        struct ion_handle *handle)
+{
+}
 #endif
 
 static int ion_debug_client_show_buffer(struct seq_file *s, void *unused)
-- 
2.35.3

