From 8dea5b71dbe4b8c66ac63f798b2ed8e8dbc0bfa2 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 16 May 2014 15:05:22 +0800
Subject: [PATCH] rk fb: delete acq wait message

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9d0219f04204..45c685990ac3 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1842,7 +1842,7 @@ static int rk_fb_set_win_config(struct fb_info *info,
 		reg_win_data = &regs->reg_win_data[i];
 		for(j=0;j<RK_WIN_MAX_AREA;j++){
 			if(reg_win_data->reg_area_data[j].acq_fence){
-				printk("acq_fence wait!!!!!\n");
+				//printk("acq_fence wait!!!!!\n");
 				rk_fd_fence_wait(dev_drv,
 				reg_win_data->reg_area_data[j].acq_fence);
 			}
-- 
2.35.3

