From 30b9a5e323a42808fb9301c0e3a31cf8318f8df5 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 24 Oct 2017 19:39:16 +0800
Subject: [PATCH] rk: .gitignore: ignore kernel.img and resource.img

Change-Id: Icfd5518a10fef49c76164f4943f6451877c32f99
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index fd3a35592543..54e1e7445bed 100644
--- a/.gitignore
+++ b/.gitignore
@@ -41,6 +41,8 @@ Module.symvers
 #
 # Top-level generic files
 #
+/kernel.img
+/resource.img
 /tags
 /TAGS
 /linux
-- 
2.35.3

