From 5161a4faf2a48c329f56c36e5112cfecb6ddefb4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 1 Aug 2013 09:27:14 +0800
Subject: [PATCH] rk_serial: fix compile error when printascii no define

---
 drivers/tty/serial/rk_serial.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/rk_serial.c b/drivers/tty/serial/rk_serial.c
index 1818c45e02ba..5272a00100af 100644
--- a/drivers/tty/serial/rk_serial.c
+++ b/drivers/tty/serial/rk_serial.c
@@ -161,7 +161,9 @@ static void dbg(const char *fmt, ...)
 	vsprintf(buff, fmt, va);
 	va_end(va);
 
+#if defined(CONFIG_DEBUG_LL) || defined(CONFIG_RK_EARLY_PRINTK)
 	printascii(buff);
+#endif
 }
 
 //enable log output
-- 
2.35.3

