From 58d7488f6cd0088fdf27a8967dfaad0804fc4433 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 6 Mar 2012 14:19:27 +0800
Subject: [PATCH] rk30: cpu hotplug: better cpu die

---
 arch/arm/mach-rk30/hotplug.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/hotplug.c b/arch/arm/mach-rk30/hotplug.c
index 5ea1ae3d48fe..159269149cb4 100644
--- a/arch/arm/mach-rk30/hotplug.c
+++ b/arch/arm/mach-rk30/hotplug.c
@@ -60,8 +60,10 @@ void platform_cpu_die(unsigned int cpu)
 	/* wait for SoC code in platform_cpu_kill() to shut off CPU core
 	 * power. CPU bring up starts from the reset vector.
 	 */
-	while (1)
-		cpu_do_idle();
+	while (1) {
+		dsb();
+		wfi();
+	}
 }
 
 int platform_cpu_disable(unsigned int cpu)
-- 
2.35.3

