From de9ae09c6425db8fb839ce75ef8d11ae4ede8764 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Tue, 7 Jul 2015 10:05:45 +0800
Subject: [PATCH] mmc: rk_sdmmc: fix wrong skip condiction of post_tmo

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 7f8012ead0cd..f8af868f2075 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -1969,7 +1969,7 @@ static void dw_mci_post_tmo(struct mmc_host *mmc)
 
 	data = host->data;
 
-	if ((opcode == MMC_SEND_TUNING_BLOCK_HS200) &&
+	if ((opcode == MMC_SEND_TUNING_BLOCK_HS200) ||
 	    (opcode == MMC_SEND_TUNING_BLOCK))
 	    return;
 
-- 
2.35.3

