From 4ce7352414b8c6e81e3fcc9cf8ca760b95965650 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Tue, 5 Mar 2013 10:59:30 +0800
Subject: [PATCH] rk610 codec: add poweroff codec diable clk

---
 sound/soc/codecs/rk610_codec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rk610_codec.c b/sound/soc/codecs/rk610_codec.c
index ef3993082b2a..2f5ee435457c 100755
--- a/sound/soc/codecs/rk610_codec.c
+++ b/sound/soc/codecs/rk610_codec.c
@@ -366,7 +366,7 @@ static int rk610_codec_set_bias_level(struct snd_soc_codec *codec,
 #endif
 		printk("rk610 standby\n");
 		spk_ctrl_fun(GPIO_LOW);
-		 rk610_codec_write(codec,ACCELCODEC_R0A, ASC_CLK_DISABLE);
+		rk610_codec_write(codec,ACCELCODEC_R0A, ASC_CLK_DISABLE);
 		rk610_codec_write(codec, ACCELCODEC_R1D, 0xFE);
 		rk610_codec_write(codec, ACCELCODEC_R1E, 0xFF);
 		rk610_codec_write(codec, ACCELCODEC_R1F, 0xFF);
@@ -375,6 +375,7 @@ static int rk610_codec_set_bias_level(struct snd_soc_codec *codec,
 	case SND_SOC_BIAS_OFF:
 		printk("rk610 power off\n");
 		spk_ctrl_fun(GPIO_LOW);
+		rk610_codec_write(codec,ACCELCODEC_R0A, ASC_CLK_DISABLE);		
 		rk610_codec_write(codec, ACCELCODEC_R1D, 0xFF);
 		rk610_codec_write(codec, ACCELCODEC_R1E, 0xFF);
 		rk610_codec_write(codec, ACCELCODEC_R1F, 0xFF);
-- 
2.35.3

