From 2fcd8b788cd6f7b2c04a0a22779ee602f17f7142 Mon Sep 17 00:00:00 2001
From: cmc <cmc@rock-chips.com>
Date: Mon, 29 Nov 2010 15:25:37 +0800
Subject: [PATCH] modify comment for spi

---
 drivers/spi/rk29xx_spim.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/rk29xx_spim.c b/drivers/spi/rk29xx_spim.c
index 09e3527b5823..17654f349b6a 100755
--- a/drivers/spi/rk29xx_spim.c
+++ b/drivers/spi/rk29xx_spim.c
@@ -1947,7 +1947,7 @@ static void __exit rk29xx_spim_exit(void)
 subsys_initcall(rk29xx_spim_init);
 module_exit(rk29xx_spim_exit);
 
-MODULE_AUTHOR("cmc cmc@rock-chips.com");
+MODULE_AUTHOR("www.rock-chips.com");
 MODULE_DESCRIPTION("Rockchip RK29xx spim port driver");
 MODULE_LICENSE("GPL");;
 
-- 
2.35.3

