From fe2f080be82212dd015b5ed225cc9ce65f329d28 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 26 Mar 2012 11:56:57 +0800
Subject: [PATCH] rk30: pm: enable scu standby mode

---
 arch/arm/mach-rk30/platsmp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-rk30/platsmp.c b/arch/arm/mach-rk30/platsmp.c
index 144f04d3c9d3..8e26efa25b00 100644
--- a/arch/arm/mach-rk30/platsmp.c
+++ b/arch/arm/mach-rk30/platsmp.c
@@ -17,6 +17,9 @@
 
 #include <mach/pmu.h>
 
+#define SCU_CTRL             0x00
+#define   SCU_STANDBY_EN     (1 << 5)
+
 #ifdef CONFIG_FIQ
 static void gic_raise_softirq_non_secure(const struct cpumask *mask, unsigned int irq)
 {
@@ -126,5 +129,7 @@ void __init platform_smp_prepare_cpus(unsigned int max_cpus)
 		set_cpu_present(i, true);
 #endif
 
+	writel_relaxed(readl_relaxed(RK30_SCU_BASE + SCU_CTRL) | SCU_STANDBY_EN, RK30_SCU_BASE + SCU_CTRL);
+
 	scu_enable(RK30_SCU_BASE);
 }
-- 
2.35.3

