From b9195ee4f9c4a6f101a298c79d2515d3cd286027 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Wed, 9 Apr 2014 09:53:08 +0800
Subject: [PATCH] rk: ion: resolve build err

---
 drivers/staging/android/ion/ion.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/android/ion/ion.c b/drivers/staging/android/ion/ion.c
index d53172eecf1d..0fe30de66169 100755
--- a/drivers/staging/android/ion/ion.c
+++ b/drivers/staging/android/ion/ion.c
@@ -116,7 +116,7 @@ struct ion_handle {
 	int id;
 };
 
-void ion_iommu_force_unmap(struct ion_buffer *buffer);
+static void ion_iommu_force_unmap(struct ion_buffer *buffer);
 
 bool ion_buffer_fault_user_mappings(struct ion_buffer *buffer)
 {
-- 
2.35.3

