From 146fa2842deb70092aefe2fff14e38eb644045ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 30 Jan 2013 12:02:40 +0800
Subject: [PATCH] rk: ddr_freq: remove BUG_ON

---
 arch/arm/plat-rk/ddr_freq.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/plat-rk/ddr_freq.c b/arch/arm/plat-rk/ddr_freq.c
index 4f919c741525..f46d8d412b62 100644
--- a/arch/arm/plat-rk/ddr_freq.c
+++ b/arch/arm/plat-rk/ddr_freq.c
@@ -168,8 +168,6 @@ static void _ddr_change_freq(uint32_t nMHz)
 	unsigned int this_cpu = get_cpu();
 	cpu_maps_update_begin();
 
-	BUG_ON(this_cpu != 0);
-
 	set_other_cpus_pause(false);
 
 	smp_call_function((smp_call_func_t)pause_cpu, NULL, 0);
-- 
2.35.3

