From 8516dcfc23e28f96b940c1e26099baff715dc7a9 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Fri, 28 Sep 2012 15:48:27 +0800
Subject: [PATCH] rk3066b: adjust backlight min bright for M701

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index f3f8b83a6053..cfdc6fc68ec3 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -285,6 +285,7 @@ static struct rk29_bl_info rk29_bl_info = {
 	.bl_ref = PWM_EFFECT_VALUE,
 	.io_init = rk29_backlight_io_init,
 	.io_deinit = rk29_backlight_io_deinit,
+	.min_brightness = 48,
 	.pwm_suspend = rk29_backlight_pwm_suspend,
 	.pwm_resume = rk29_backlight_pwm_resume,
 };
-- 
2.35.3

