From 5dbd2a80f28c47be3d9b0f10939567c55f5a3534 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 28 Feb 2012 19:47:54 +0800
Subject: [PATCH] rk30: fb: fix compile error

---
 drivers/video/rockchip/rk_fb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index b79bee5cf325..56ec9a1f574a 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -441,13 +441,13 @@ struct fb_info * rk_get_fb(int fb_id)
     struct fb_info *fb = inf->fb[fb_id];
     return fb;
 }
-EXPORT_SYMBOL(get_fb);
+EXPORT_SYMBOL(rk_get_fb);
 
 void rk_direct_fb_show(struct fb_info * fbi)
 {
     rk_fb_set_par(fbi);
 }
-EXPORT_SYMBOL(direct_fb_show);
+EXPORT_SYMBOL(rk_direct_fb_show);
 
 static int request_fb_buffer(struct fb_info *fbi,int fb_id)
 {
-- 
2.35.3

