From ebaf5512a628b63aaa39f39984f131bdcc073be9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Dec 2012 18:20:28 +0800
Subject: [PATCH] rk: printk: add pm_emit_log_char

---
 kernel/printk.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kernel/printk.c b/kernel/printk.c
index c6b2f794a210..d99f1fca19a6 100644
--- a/kernel/printk.c
+++ b/kernel/printk.c
@@ -1023,6 +1023,13 @@ asmlinkage int vprintk(const char *fmt, va_list args)
 EXPORT_SYMBOL(printk);
 EXPORT_SYMBOL(vprintk);
 
+#ifdef CONFIG_PLAT_RK
+void pm_emit_log_char(char c)
+{
+	emit_log_char(c);
+}
+#endif
+
 #ifdef CONFIG_RK29_LAST_LOG
 void switch_log_buf(char *new_log_buf, int size)
 {
-- 
2.35.3

