From a44982e45922096726ed9376ef9bc40d8e2bb954 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 4 Jul 2012 17:23:30 +0800
Subject: [PATCH] rk30:sdk: add GPU frequency limit between 133MHz to 400MHz

---
 arch/arm/mach-rk30/cpufreq.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index b352b1f9f7a5..f7a1c87c21b7 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -256,6 +256,9 @@ static int rk30_cpu_init(struct cpufreq_policy *policy)
 		}
 		clk_enable_dvfs(cpu_clk);
 
+		/* Limit gpu frequency between 133M to 400M */
+		dvfs_clk_enable_limit(gpu_clk, 133000000, 400000000);
+
 		freq_wq = create_singlethread_workqueue("rk30_cpufreqd");
 #ifdef CONFIG_RK30_CPU_FREQ_LIMIT_BY_TEMP
 		if (rk30_cpufreq_is_ondemand_policy(policy)) {
-- 
2.35.3

