From 8be996a98c6e85d69e7ab175126b409b262c2f97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Jan 2013 15:15:03 +0800
Subject: [PATCH] rk3188: default debug uart 2

---
 arch/arm/plat-rk/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 68960de88248..47ceeade6135 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -159,7 +159,7 @@ config RK_EARLY_PRINTK
 config RK_DEBUG_UART
 	int "Debug UART"
 	default 1 if ARCH_RK29
-	default 2 if ARCH_RK30 || ARCH_RK2928
+	default 2 if ARCH_RK30 || ARCH_RK2928 || ARCH_RK3188
 	help
 	  Select a UART for debugging. -1 disable.
 
-- 
2.35.3

