From 1d6f135e0db722b21fd519ce0a83ed52ac4d4aa1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 6 Sep 2012 12:09:47 +0800
Subject: [PATCH] rk3066b: arm-pmu: support dual core

---
 arch/arm/mach-rk30/devices.c | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index 9cf7314d12a4..fcd0d64c0c70 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -1296,17 +1296,24 @@ static struct platform_device device_wdt = {
 };
 #endif
 
-static struct resource resource_arm_pmu = {
-	.start		= IRQ_ARM_PMU,
-	.end		= IRQ_ARM_PMU,
-	.flags		= IORESOURCE_IRQ,
+static struct resource resource_arm_pmu[] = {
+	{
+		.start	= IRQ_ARM_PMU,
+		.end	= IRQ_ARM_PMU,
+		.flags	= IORESOURCE_IRQ,
+	},
+	{
+		.start	= IRQ_ARM_PMU + 1,
+		.end	= IRQ_ARM_PMU + 1,
+		.flags	= IORESOURCE_IRQ,
+	},
 };
 
 struct platform_device device_arm_pmu = {
 	.name		= "arm-pmu",
 	.id		= ARM_PMU_DEVICE_CPU,
-	.num_resources	= 1,
-	.resource	= &resource_arm_pmu,
+	.num_resources	= ARRAY_SIZE(resource_arm_pmu),
+	.resource	= resource_arm_pmu,
 };
 
 static int __init rk30_init_devices(void)
-- 
2.35.3

