From b517a4ebbfb9302cb687f8981d6cf26b34ebcdb1 Mon Sep 17 00:00:00 2001
From: David Wu <wdc@rock-chips.com>
Date: Fri, 26 Jun 2015 16:57:36 +0800
Subject: [PATCH] rk3368: tsadc: support print temp from console

Signed-off-by: David Wu <wdc@rock-chips.com>
---
 drivers/thermal/rockchip_thermal.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/thermal/rockchip_thermal.c b/drivers/thermal/rockchip_thermal.c
index 7767d880470c..007b0b29558e 100755
--- a/drivers/thermal/rockchip_thermal.c
+++ b/drivers/thermal/rockchip_thermal.c
@@ -750,6 +750,7 @@ int rockchip_tsadc_get_temp(int chn, int voltage)
 		code_temp = (tsadc_data * voltage + 500000) / 1000000;
 		temp = rk_tsadcv3_code_to_temp(code_temp) / 1000;
 		temp = temp - thermal->cpu_temp_adjust;
+		thermal->cpu_temp = temp;
 		if(thermal->logout)
 			printk("cpu temp:[%d], voltage: %d\n"
 				, temp, voltage);
-- 
2.35.3

