From 9128b0233031bad28fe81335208b622664c3028f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Tue, 11 May 2010 14:02:35 +0000
Subject: [PATCH] add i2c

---
 arch/arm/mach-rk2818/board-midsdk.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2818/board-midsdk.c b/arch/arm/mach-rk2818/board-midsdk.c
index f2707735db3d..27edba489a3d 100644
--- a/arch/arm/mach-rk2818/board-midsdk.c
+++ b/arch/arm/mach-rk2818/board-midsdk.c
@@ -32,6 +32,7 @@
 #include <mach/rk2818_iomap.h>
 #include <mach/iomux.h>
 #include <mach/gpio.h>
+#include <mach/i2c.h>
 
 #include <linux/mtd/nand.h>
 #include <linux/mtd/partitions.h>
-- 
2.35.3

