From c69849fe1aea5286fdf55031a06eb9fb2c0e8276 Mon Sep 17 00:00:00 2001
From: qjb <qjb@rock-chips.com>
Date: Tue, 3 Jun 2014 11:05:21 +0800
Subject: [PATCH] i2s : fix i2s suspend resume set error

---
 sound/soc/rockchip/rk30_i2s.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rockchip/rk30_i2s.c b/sound/soc/rockchip/rk30_i2s.c
index dd4a6fa89f5f..72784ab2058a 100755
--- a/sound/soc/rockchip/rk30_i2s.c
+++ b/sound/soc/rockchip/rk30_i2s.c
@@ -682,8 +682,8 @@ MODULE_DEVICE_TABLE(of, exynos_i2s_match);
 #endif
 
 static const struct dev_pm_ops rockchip_i2s_pm_ops = {
-        SET_RUNTIME_PM_OPS(rockchip_i2s_suspend_noirq,
-                                rockchip_i2s_resume_noirq, NULL)
+	.suspend_noirq = rockchip_i2s_suspend_noirq,
+	.resume_noirq  = rockchip_i2s_resume_noirq,
 };
 
 static struct platform_driver rockchip_i2s_driver = {
-- 
2.35.3

