From 14bed994d7b005494890335c0e88d67a9cea6d99 Mon Sep 17 00:00:00 2001
From: smj <smj@rock-chips.com>
Date: Tue, 28 Oct 2014 16:47:54 +0800
Subject: [PATCH] audio : fix the problem of usb-audio code

---
 sound/usb/card.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/usb/card.c b/sound/usb/card.c
index bfac3568ae2c..21a4e03e80dd 100644
--- a/sound/usb/card.c
+++ b/sound/usb/card.c
@@ -923,7 +923,7 @@ static void __exit snd_usb_audio_cleanup(void)
 	switch_dev_unregister(&sUsbaudio_Switch.sUsbaudio_Playback);
 }
 
-module_init(snd_usb_audio_init);
+/*module_init(snd_usb_audio_init);*/
 /* use late initcall_sync instead of module_init,
  * make sure that usbaudio probe after board codec.
  * added by zxg@rock-chips.com
-- 
2.35.3

