From 10ed7070954575122bb97896fc27e088b517eea8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=98=8A?= <zhanghao@rock-chips.com>
Date: Mon, 28 Nov 2011 15:10:41 +0800
Subject: [PATCH] fix building error mtd23d

---
 drivers/misc/mtk23d.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/mtk23d.c b/drivers/misc/mtk23d.c
index 8f8dae623395..51a2198f5461 100755
--- a/drivers/misc/mtk23d.c
+++ b/drivers/misc/mtk23d.c
@@ -291,9 +291,9 @@ int modem_poweron_off(int on_off)
 static int power_on =1;
 static int mtk23d_open(struct inode *inode, struct file *file)
 {
-	//struct rk2818_23d_data *pdata = gpdata;
+	struct rk2818_23d_data *pdata = gpdata;
 	//struct rk2818_23d_data *pdata = gpdata = pdev->dev.platform_data;
-	//struct platform_data *pdev = container_of(pdata, struct device, platform_data);
+	struct platform_data *pdev = container_of(pdata, struct device, platform_data);
 
 	MODEMDBG("modem_open\n");
 
@@ -303,7 +303,7 @@ static int mtk23d_open(struct inode *inode, struct file *file)
 		power_on = 0;
 		modem_poweron_off(1);
 	}
-	device_init_wakeup(&pdev->dev, 1);
+	device_init_wakeup(&pdev, 1);
 
 	return 0;
 }
-- 
2.35.3

