From 41211c67278ddc58d1a612086a596fc05ede2352 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 14 Feb 2012 18:35:39 +0800
Subject: [PATCH] rk30: timer: set clockevent irq, fix for smp

---
 arch/arm/mach-rk30/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/timer.c b/arch/arm/mach-rk30/timer.c
index 071fd1447c64..972f801c4b2e 100644
--- a/arch/arm/mach-rk30/timer.c
+++ b/arch/arm/mach-rk30/timer.c
@@ -146,7 +146,7 @@ static __init int rk30_timer_init_clockevent(void)
 	clockevents_calc_mult_shift(ce, 24000000, 4);
 	ce->max_delta_ns = clockevent_delta2ns(0xFFFFFFFFUL, ce);
 	ce->min_delta_ns = clockevent_delta2ns(1, ce) + 1;
-
+	ce->irq = rk30_timer_clockevent_irq.irq;
 	ce->cpumask = cpu_all_mask;
 
 	clockevents_register_device(ce);
-- 
2.35.3

