From b8e5d19274433639cb79784d70c0db599344e480 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 20:19:22 +0800
Subject: [PATCH] rockchip: iommu: fix compilation error on arm64

---
 drivers/iommu/rockchip-iommu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 4386d0e88bc8..b5a815cb1d88 100755
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -487,8 +487,12 @@ static bool iommu_reset(void __iomem *base, const char *dbgname)
 
 static inline void pgtable_flush(void *vastart, void *vaend)
 {
+#ifdef CONFIG_ARM
 	dmac_flush_range(vastart, vaend);
 	outer_flush_range(virt_to_phys(vastart), virt_to_phys(vaend));
+#elif defined(CONFIG_ARM64)
+	__dma_flush_range(vastart, vaend);
+#endif
 }
 
 static void set_fault_handler(struct iommu_drvdata *data,
-- 
2.35.3

