From aa7ff457c433a9cb78f55c84be5a58a78da887db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 13 Jul 2011 14:23:05 +0800
Subject: [PATCH] rk29: enable HIGHMEM and ZONE_DMA

---
 arch/arm/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index df6eafbf69b8..9fb734a45351 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -731,6 +731,8 @@ config ARCH_RK29
 	select ARCH_REQUIRE_GPIOLIB
 	select ARM_GIC
 	select PL330
+	select HIGHMEM
+	select ZONE_DMA
 	help
 	  Support for Rockchip RK29 soc.
 
-- 
2.35.3

