From 300588e5c875f93e6c43709849dc1bed56e869b8 Mon Sep 17 00:00:00 2001
From: smj <smj@rock-chips.com>
Date: Fri, 31 Oct 2014 16:10:05 +0800
Subject: [PATCH] spdif : disable the spdif debug print

---
 sound/soc/rockchip/rk_spdif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk_spdif.c b/sound/soc/rockchip/rk_spdif.c
index da0cf7bbbd4c..0113e7452942 100755
--- a/sound/soc/rockchip/rk_spdif.c
+++ b/sound/soc/rockchip/rk_spdif.c
@@ -43,7 +43,7 @@
 #include "rk_pcm.h"
 
 #undef  DEBUG_SPDIF
-#define DEBUG_SPDIF 1
+#define DEBUG_SPDIF 0
 
 #if DEBUG_SPDIF
 #define RK_SPDIF_DBG(x...) pr_info("rk_spdif:"x)
-- 
2.35.3

