From cf0b398a468b367d2a8b54ec12d3c2fd2e4f68eb Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Mon, 29 Aug 2011 02:50:38 -0700
Subject: [PATCH] newton: remove redefined of bu92747_lock && bu92747_unlock
 function

---
 drivers/cir/bu92747guw_cir.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/cir/bu92747guw_cir.c b/drivers/cir/bu92747guw_cir.c
index a4022801a2d0..6963654aedea 100755
--- a/drivers/cir/bu92747guw_cir.c
+++ b/drivers/cir/bu92747guw_cir.c
@@ -58,20 +58,6 @@ static struct miscdevice bu92747guw_device;
 
 int repeat_flag=-1;
 int start_flag = 0;
-//mutex lock between remote and irda
-static DEFINE_MUTEX(bu92747_mutex);
-void bu92747_lock(void)
-{
-	mutex_lock(&bu92747_mutex);
-}
-void bu92747_unlock(void)
-{
-	mutex_unlock(&bu92747_mutex);
-}
-
-
-
-
 
 static int bu92747_cir_i2c_read_regs(struct i2c_client *client, u8 reg, u8 *buf, int len)
 {
-- 
2.35.3

