From c3f1f1de4e84fddb21a930de9b623c1be1655929 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Thu, 18 Oct 2012 12:25:57 +0800
Subject: [PATCH] rk3066b m701: add tps65910 shutdown support

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index 7735e9e08860..688598f2a3ba 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -1631,6 +1631,12 @@ static void rk30_pm_power_off(void)
 #if defined(CONFIG_MFD_WM831X)
 	wm831x_set_bits(Wm831x,WM831X_GPIO_LEVEL,0x0001,0x0000);  //set sys_pwr 0
 	wm831x_device_shutdown(Wm831x);//wm8326 shutdown
+#endif
+#if defined(CONFIG_MFD_TPS65910)
+	if(pmic_is_tps65910())
+	{
+	    tps65910_device_shutdown();//tps65910 shutdown
+	}
 #endif
 	while (1);
 }
-- 
2.35.3

