From 12bd489d4d3cc304d17233298716246f6040a9b3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Fri, 17 Oct 2014 10:36:52 +0800
Subject: [PATCH] chargedisplay: add for adc-battery-driver
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 许盛飞 <xsf@rock-chips.com>
---
 drivers/power/rk29_charger_display.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/rk29_charger_display.c b/drivers/power/rk29_charger_display.c
index af377b17e97c..407486f68bf0 100755
--- a/drivers/power/rk29_charger_display.c
+++ b/drivers/power/rk29_charger_display.c
@@ -127,7 +127,7 @@ static int  __init start_charge_logo_display(void)
 
 	if(val_status.intval == POWER_SUPPLY_STATUS_CHARGING)
 	{
-		if (((rockchip_boot_mode() == BOOT_MODE_NORMAL) ||(rockchip_boot_mode() == BOOT_MODE_CHARGE)) && (val_capacity.intval <= pwr_on_thrsd))  
+		if (((rockchip_boot_mode() == BOOT_MODE_NORMAL) || (rockchip_boot_mode() == BOOT_MODE_CHARGE)) || (val_capacity.intval <= pwr_on_thrsd))
 	    {			
 			add_bootmode_charger_to_cmdline();
 			printk("power in charge mode %d %d  %d\n\n",rockchip_boot_mode(),val_capacity.intval,pwr_on_thrsd);
@@ -137,5 +137,5 @@ static int  __init start_charge_logo_display(void)
 	return 0;
 } 
 
-module_init(start_charge_logo_display);
+late_initcall(start_charge_logo_display);
 #endif
-- 
2.35.3

