From 039233cc0a7343a6da9436bc08a724f1cee15fac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Jan 2013 16:25:15 +0800
Subject: [PATCH] rk_screen.h: fix when DISPLAY_SUPPORT no selected

---
 include/linux/rk_screen.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index a04c826899d5..977a0891fd0e 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -129,7 +129,11 @@ typedef struct rk29fb_screen {
 	int (*sscreen_set)(struct rk29fb_screen *screen, bool type);// 1: use scaler 0:bypass
 } rk_screen;
 
+#ifndef CONFIG_DISPLAY_SUPPORT
+static inline void set_lcd_info(struct rk29fb_screen *screen, struct rk29lcd_info *lcd_info) {}
+#else
 extern void set_lcd_info(struct rk29fb_screen *screen, struct rk29lcd_info *lcd_info);
+#endif
 extern void set_tv_info(struct rk29fb_screen *screen);
 extern void set_hdmi_info(struct rk29fb_screen *screen);
 
-- 
2.35.3

