From 6c2c81d2c85d8fe173c21ee7f0bce2bb47ff9010 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Thu, 7 Jun 2012 20:12:01 +0800
Subject: [PATCH] rk30 phone: modify codec as master when connect to BT and BP

---
 sound/soc/rk29/rk29_aic3262.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rk29/rk29_aic3262.c b/sound/soc/rk29/rk29_aic3262.c
index 88f04ce9fcf0..89f30b5f5e21 100644
--- a/sound/soc/rk29/rk29_aic3262.c
+++ b/sound/soc/rk29/rk29_aic3262.c
@@ -295,7 +295,7 @@ static int rk29_aif2_hw_params(struct snd_pcm_substream *substream,
 
 	/* set codec DAI configuration */
 	ret = snd_soc_dai_set_fmt(codec_dai, SND_SOC_DAIFMT_LEFT_J |
-			SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBS_CFS);
+			SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBM_CFM);
 	if (ret < 0)
 	{
 		printk("%s: snd_soc_dai_set_fmt err =%d\n",__FUNCTION__,ret);
@@ -367,7 +367,7 @@ static int rk29_aif3_hw_params(struct snd_pcm_substream *substream,
 
 	/* set codec DAI configuration */
 	ret = snd_soc_dai_set_fmt(codec_dai, SND_SOC_DAIFMT_LEFT_J |
-			SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBS_CFS);
+			SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBM_CFM);
 	if (ret < 0)
 	{
 		printk("%s: snd_soc_dai_set_fmt err =%d\n",__FUNCTION__,ret);
-- 
2.35.3

