From 2b412f066b25a67fb4b5705f49b65c4318a97eda Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 18 Apr 2011 16:40:18 +0800
Subject: [PATCH] update i2c-dev-rk29.c

---
 drivers/i2c/busses/i2c-dev-rk29.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-dev-rk29.c b/drivers/i2c/busses/i2c-dev-rk29.c
index c56114bea5ed..1d9d922d1c60 100755
--- a/drivers/i2c/busses/i2c-dev-rk29.c
+++ b/drivers/i2c/busses/i2c-dev-rk29.c
@@ -20,7 +20,7 @@
 #define I2C_DEV_SCL_RATE	100 * 1000
 
 struct completion		i2c_dev_complete = {
-	.done = -1;
+	.done = -1,
 };
 struct i2c_dump_info	g_dump;
 
-- 
2.35.3

