From b086d4e66c410bb0c74c31c24bf7349112223179 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 30 Jun 2014 20:27:52 +0800
Subject: [PATCH] ARM: rockchip: enable ARM_ERRATA_818325 for RK3288

---
 arch/arm/mach-rockchip/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rockchip/Kconfig b/arch/arm/mach-rockchip/Kconfig
index 827f48ccd8f6..9d7c3acd8704 100755
--- a/arch/arm/mach-rockchip/Kconfig
+++ b/arch/arm/mach-rockchip/Kconfig
@@ -20,6 +20,7 @@ config ARCH_ROCKCHIP
 	select DMADEVICES
 	select PL330_DMA
 	select FIQ_GLUE
+	select ARM_ERRATA_818325
 
 if ARCH_ROCKCHIP
 
-- 
2.35.3

