From 8e031bb6474bd4674bf1356bb0c93d94b9e3856a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 25 Dec 2012 09:55:43 +0800
Subject: [PATCH] pack-kernel.sh: open source every file

---
 pack-kernel.sh | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index c56a82b94c73..a891021b7d75 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -26,11 +26,6 @@ EXCLUDES=()
 
 if [ "$arch" == "rk30" ]; then
 
-FILES=(
-arch/arm/mach-rk30/clock_data.c
-arch/arm/mach-rk30/ddr.c
-)
-
 EXCLUDES=(
 arch/arm/mach-rk30/*rk3066b*
 arch/arm/mach-rk30/include/mach/*rk3066b*
@@ -52,11 +47,6 @@ defconfig=rk3066_sdk_defconfig
 
 elif [ "$arch" == "rk2928" ]; then
 
-FILES=(
-arch/arm/mach-rk2928/clock_data.c
-arch/arm/mach-rk2928/ddr.c
-)
-
 EXCLUDES=(
 arch/arm/mach-rk30/*.c
 arch/arm/mach-rk30/*.h
@@ -84,11 +74,6 @@ defconfig=rk2928_sdk_defconfig
 
 elif [ "$arch" == "rk3066b" ]; then
 
-FILES=(
-arch/arm/mach-rk30/clock_data-rk3066b.c
-arch/arm/mach-rk30/ddr.c
-)
-
 EXCLUDES=(
 arch/arm/mach-rk30/clock_data.c
 arch/arm/mach-rk30/board-rk30-*
-- 
2.35.3

