From 9e03362d2de8eb31d60992266e56fe91c89589c9 Mon Sep 17 00:00:00 2001
From: yangyun <yangyun@rock-chips.com>
Date: Thu, 22 Mar 2012 09:46:16 +0800
Subject: [PATCH] add verifyID to pach uu

---
 pack-kernel.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 36f0e9d8c8cf..e34829076ddb 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -6,6 +6,7 @@ FILES=(
 arch/arm/mach-rk29/clock.c
 arch/arm/mach-rk29/ddr.c
 arch/arm/mach-rk29/vpu*.c
+arch/arm/mach-rk29/verifyID.c
 
 drivers/staging/rk29/ipp/rk29-ipp.c
 )
-- 
2.35.3

