From 5fd886fa3067e281254b508062e43df3af2f5fc0 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 10 Dec 2014 19:30:04 +0800
Subject: [PATCH] rk_serial: covert dsb() to dsb(sy)

---
 drivers/tty/serial/rk_serial.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/rk_serial.c b/drivers/tty/serial/rk_serial.c
index 1f1f61d43a16..ea915c6a41b4 100644
--- a/drivers/tty/serial/rk_serial.c
+++ b/drivers/tty/serial/rk_serial.c
@@ -327,7 +327,7 @@ static inline void serial_out(struct uart_rk_port *up, int offset, unsigned char
 	dwapb_save_out_value(up, offset, value);
 	__raw_writel(value, up->port.membase + (offset << 2));
 	if (offset != UART_TX)
-		dsb();
+		dsb(sy);
 	dwapb_check_clear_ier(up, offset);
 }
 
-- 
2.35.3

