From 889a3650c5a8ce45e751dac3cb5317a3fd77e088 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=92=9F=E5=8B=87=E6=B1=AA?= <zyw@rock-chips.com>
Date: Wed, 19 May 2010 03:50:17 +0000
Subject: [PATCH] update fb

---
 drivers/video/rk2818_fb.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/drivers/video/rk2818_fb.c b/drivers/video/rk2818_fb.c
index 56763e432958..93ac65e92560 100644
--- a/drivers/video/rk2818_fb.c
+++ b/drivers/video/rk2818_fb.c
@@ -60,7 +60,7 @@
 #define WIN1_USE_DOUBLE_BUF     1       //win1 use double buf to accelerate display
 #define LANDSCAPE_USE_ROTATE    1       //rotate win1 in landscape with mcu panel
 
-#if 0
+#if 1
 	#define fbprintk(msg...)	printk(msg);
 #else
 	#define fbprintk(msg...)
@@ -1871,20 +1871,6 @@ static int __init rk2818fb_probe (struct platform_device *pdev)
 	int irq = 0;
     int ret = 0;
 
-//------------------back light-----------------------
-    rk2818_mux_api_set(GPIOF2_APWM0_SEL_NAME, 0);
-    ret = gpio_request(RK2818_PIN_PF2, NULL); 
-    if(ret != 0)
-    {
-        gpio_free(RK2818_PIN_PF2);
-        printk(KERN_ERR ">>>>>> back light gpio_request err \n ");
-        return 1;
-    }
-    gpio_direction_output(RK2818_PIN_PF2, 0);
-	gpio_set_value(RK2818_PIN_PF2, 0);
-
-//-----------------------------------------
-
     fbprintk(">>>>>> %s : %s\n", __FILE__, __FUNCTION__);
 
     /* Malloc rk2818fb_inf and set it to pdev for drvdata */
-- 
2.35.3

