From bd0229c230e7ce936612e36e1cb8915faef90018 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 9 Apr 2015 17:37:15 +0800
Subject: [PATCH] rk fb: hdmi 24hz frame time need 41ms

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index bcfe1925ec8f..dc225b198e4a 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1770,7 +1770,7 @@ static void rk_fb_update_reg(struct rk_lcdc_driver *dev_drv,
 		timestamp = dev_drv->vsync_info.timestamp;
 		timeout = wait_event_interruptible_timeout(dev_drv->vsync_info.wait,
 				ktime_compare(dev_drv->vsync_info.timestamp, timestamp) > 0,
-				msecs_to_jiffies(25));
+				msecs_to_jiffies(50));
 
 		wait_for_vsync = rk_fb_reg_effect(dev_drv, regs, count);
 	} while (wait_for_vsync && count--);
-- 
2.35.3

