From 8a4dc647f1326bac1cde57fb44ccdc3d49c76e80 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 4 Mar 2015 17:58:53 +0800
Subject: [PATCH] rk3368: rk818 battery: allow capacity to be negative value

Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/power/rk818_battery.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index 1aa7c5a645a3..26285db4e507 100644
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -2446,8 +2446,6 @@ static void update_battery_info(struct battery_info *di)
 	di->remain_capacity = _get_realtime_capacity(di);
 	if (di->remain_capacity > di->fcc)
 		_capacity_init(di, di->fcc);
-	else if (di->remain_capacity < 0)
-		_capacity_init(di, 0);
 
 	if (di->real_soc > 100)
 		di->real_soc = 100;
-- 
2.35.3

