From 49a2dd374c3881db681d432ba5cadb2ff400ed26 Mon Sep 17 00:00:00 2001
From: "dalon.zhang" <dalon.zhang@rock-chips.com>
Date: Wed, 18 Mar 2015 21:22:06 +0800
Subject: [PATCH] camsys_head: v0.0xb.0

---
 include/media/camsys_head.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/media/camsys_head.h b/include/media/camsys_head.h
index 31a828f7bebf..3eba31f9c430 100755
--- a/include/media/camsys_head.h
+++ b/include/media/camsys_head.h
@@ -27,8 +27,10 @@
          1) add dev_name in struct camsys_devio_name_s;
 *v0.a.0:
          1) support external flash IC
+*v0.b.0:
+		 1) add CamSys_SensorBit0_CifBit4 in enum camsys_cifio_e. 
 */
-#define CAMSYS_HEAD_VERSION           KERNEL_VERSION(0,0xa,0)
+#define CAMSYS_HEAD_VERSION           KERNEL_VERSION(0,0xb,0)
 
 #define CAMSYS_MARVIN_DEVNAME         "camsys_marvin"           
 #define CAMSYS_CIF0_DEVNAME           "camsys_cif0"
@@ -184,6 +186,7 @@ typedef enum camsys_fmt_e {
 typedef enum camsys_cifio_e {
     CamSys_SensorBit0_CifBit0 = 0x00,
     CamSys_SensorBit0_CifBit2 = 0x01,
+    CamSys_SensorBit0_CifBit4 = 0x02,
 } camsys_cifio_t;
 
 typedef struct camsys_cifphy_s {
-- 
2.35.3

