From 2be870c46403ef1a8f9e275e77b91dd132129d75 Mon Sep 17 00:00:00 2001
From: oyyf <oyyf@rock-chips.com>
Date: Wed, 14 Aug 2013 15:00:59 +0800
Subject: [PATCH] camera: when camera close, mclk can close

---
 drivers/media/video/rk30_camera_oneframe.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/video/rk30_camera_oneframe.c b/drivers/media/video/rk30_camera_oneframe.c
index c5597ec7c9a6..7571cb4eecbb 100755
--- a/drivers/media/video/rk30_camera_oneframe.c
+++ b/drivers/media/video/rk30_camera_oneframe.c
@@ -1473,6 +1473,11 @@ static int rk_camera_mclk_ctrl(int cif_idx, int on, int clk_rate)
     	clk_disable(clk->cif_clk_out);
     	clk_disable(clk->pd_cif);
         clk->on = false;
+        if(cif){
+           err = clk_set_parent(clk->cif_clk_out, clk_get(NULL, "cif1_out_div"));
+        }else{
+           err = clk_set_parent(clk->cif_clk_out, clk_get(NULL, "cif0_out_div"));
+        }
     }
     spin_unlock(&clk->lock);
 rk_camera_clk_ctrl_end:
-- 
2.35.3

