From b96a1096ffac6bcf981c4ab7b5ec16ae7dd2d14e Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 23 Mar 2015 09:29:40 +0800
Subject: [PATCH] rk fb: update check var screen info

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e6adcbbaf202..a498859a221d 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1837,8 +1837,9 @@ static int rk_fb_set_win_buffer(struct fb_info *info,
 	struct fb_fix_screeninfo *fix = &info->fix;
 	struct rk_fb_par *fb_par = (struct rk_fb_par *)info->par;
 	struct rk_lcdc_driver *dev_drv = fb_par->lcdc_drv;
-	/*if hdmi size move to hwc,screen should point to cur_screen*/
-	struct rk_screen *screen = dev_drv->screen0;/*cur_screen;*/
+	/*if hdmi size move to hwc,screen should point to cur_screen
+	  otherwise point to screen0[main screen]*/
+	struct rk_screen *screen = dev_drv->cur_screen;/*screen0;*/
 	struct fb_info *fbi;
 	int i, ion_fd, acq_fence_fd;
 	u32 xvir, yvir;
-- 
2.35.3

