From 8f6f13848602790b7d0b5961f79319693d91c323 Mon Sep 17 00:00:00 2001
From: Cody Xie <Cody.Xie@rock-chips.com>
Date: Thu, 3 Apr 2014 14:52:01 +0800
Subject: [PATCH] rk3288 p977 : gpu dvfs table add 500M 1.1V

---
 arch/arm/boot/dts/rk3288-p977.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index a4d9444e0958..eb9a2627e781 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -498,6 +498,7 @@
 		200000 850000
 		300000 900000
 		400000 1000000
+		500000 1100000
 	//	600000 1250000
 		>;
 	status="okay";
-- 
2.35.3

