From a3933acd3547fcc21802f87e57c5fafec479eee5 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Thu, 21 May 2015 18:23:29 +0800
Subject: [PATCH] clk: rockchip: allow no rockchip,grf in dts on arm

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/clk/rockchip/clk.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/rockchip/clk.c b/drivers/clk/rockchip/clk.c
index dcc80e542a4b..d198c73001a7 100755
--- a/drivers/clk/rockchip/clk.c
+++ b/drivers/clk/rockchip/clk.c
@@ -1885,7 +1885,12 @@ static void __init rk_clk_tree_init(struct device_node *np)
 	}
 
 	node = of_parse_phandle(np, "rockchip,grf", 0);
-	rk_grf_base = of_iomap(node, 0);
+	if (node)
+		rk_grf_base = of_iomap(node, 0);
+#ifdef CONFIG_ARM
+	if (!rk_grf_base)
+		rk_grf_base = RK_GRF_VIRT;
+#endif
 
 	for_each_available_child_of_node(np, node) {
 		clk_debug("\n");
-- 
2.35.3

