From c7a375ec6eef6cc1c3334d6f4864d71c8388801d Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Thu, 26 Jun 2014 14:29:01 +0800
Subject: [PATCH] rk: p977: adjust lowmem/vmalloc area to 496/504 MB

---
 arch/arm/boot/dts/rk3288-p977.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index 1f9d623214db..fa69e5cef412 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -152,6 +152,25 @@
 		rockchip,remote_wakeup;
 		rockchip,usb_irq_wakeup;
 	};
+
+        chosen {
+                bootargs = "vmalloc=512M";
+        };
+
+        ion {
+                compatible = "rockchip,ion";
+                #address-cells = <1>;
+                #size-cells = <0>;
+
+                rockchip,ion-heap@1 { /* CMA HEAP */
+                        compatible = "rockchip,ion-reserve";
+                        rockchip,ion_heap = <1>;
+                        reg = <0x40000000 0x20000000>; /* 512MB */
+                };
+                rockchip,ion-heap@3 { /* VMALLOC HEAP */
+                        rockchip,ion_heap = <3>;
+                };
+        };
 };
 &pinctrl {
 	gpio0_gpio {
-- 
2.35.3

