From 23ca7791ab8294fc9918c52489edd030b5bf7449 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 23 Mar 2012 10:34:06 +0800
Subject: [PATCH] rk30: enable errata 764369, 754322

---
 arch/arm/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 6d78f3af4ee6..b7635e71087d 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -879,6 +879,8 @@ config ARCH_RK30
 	select PL330
 	select HAVE_SMP
 	select MIGHT_HAVE_CACHE_L2X0
+	select ARM_ERRATA_764369
+	select ARM_ERRATA_754322
 	help
 	  Support for Rockchip's RK30xx SoCs.
 
-- 
2.35.3

