From 7163070e3f57a239ee807670140e534070c80ec9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 2 Apr 2012 20:05:05 +0800
Subject: [PATCH] rk30: pm: enable fiq later

---
 arch/arm/mach-rk30/pm.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 80c0cf81a12f..c284eaf6d978 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -396,6 +396,7 @@ static int rk30_pm_enter(suspend_state_t state)
 	rk30_pm_set_power_domain(pmu_pwrdn_st, false);
 
 	sram_printch('1');
+	local_fiq_disable();
 
 	for (i = 0; i < CRU_CLKGATES_CON_CNT; i++) {
 		clkgt_regs[i] = cru_readl(CRU_CLKGATES_CON(i));
@@ -482,8 +483,6 @@ static int rk30_pm_enter(suspend_state_t state)
 
 	board_gpio_suspend();
 
-	local_fiq_disable();
-
 	flush_tlb_all();
 	interface_ctr_reg_pread();
 
@@ -491,8 +490,6 @@ static int rk30_pm_enter(suspend_state_t state)
 	rk30_suspend();
 	sram_printch('4');
 
-	local_fiq_enable();
-
 	board_gpio_resume();
 
 	sram_printch('3');
@@ -522,6 +519,7 @@ static int rk30_pm_enter(suspend_state_t state)
 		cru_writel(clkgt_regs[i] | 0xffff0000, CRU_CLKGATES_CON(i));
 	}
 
+	local_fiq_enable();
 	sram_printch('1');
 
 	rk30_pm_set_power_domain(pmu_pwrdn_st, true);
-- 
2.35.3

