From 8723f665dc16d6ad12292fd987a5b48c589f482f Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Thu, 26 Sep 2013 12:35:11 +0800
Subject: [PATCH] rk3026: cpufreq: limit cpu max frequency

---
 arch/arm/mach-rk3188/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk3188/cpufreq.c b/arch/arm/mach-rk3188/cpufreq.c
index aa129cb96e70..50c45040ae19 100644
--- a/arch/arm/mach-rk3188/cpufreq.c
+++ b/arch/arm/mach-rk3188/cpufreq.c
@@ -160,7 +160,7 @@ static struct cpufreq_frequency_table temp_limits_gpu_perf[] = {
 #elif defined(CONFIG_ARCH_RK3026)
 static struct cpufreq_frequency_table temp_limits[2][1] = {
 	{	// 1 CPU busy
-		{.frequency =          -1, .index = 0},
+		{.frequency =  912 * 1000, .index = 0},
 	}, {	// 2 CPUs busy
 		{.frequency =  816 * 1000, .index = 0},
 	}
-- 
2.35.3

