From c705010c4a4a5d4b13586afaf1772b4983c2e848 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 22 Jan 2013 14:11:05 +0800
Subject: [PATCH] rk30: fix a dvfs error judgement

---
 arch/arm/mach-rk30/dvfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 10ba749267a8..1ab3a49f4fc1 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -365,7 +365,8 @@ int dvfs_target_core(struct clk *clk, unsigned long rate_hz)
 			goto out;
 		ret = dvfs_scale_volt_bystep(dvfs_clk->vd, dvfs_clk_cpu->vd, volt_new, volt_dep_new,
 				g_logic_high_arm, g_arm_high_logic, g_logic_high_arm, g_arm_high_logic);
-		goto out;
+		if (ret < 0)
+			goto out;
 	}
 
 	return ret;
-- 
2.35.3

