From ecde737368818393cecea695ac1376fa498f7c65 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Tue, 25 Mar 2014 12:16:29 +0800
Subject: [PATCH] p977: codec add phone detect level control

---
 arch/arm/boot/dts/rk3288-p977.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index 4704deb6afde..272accfb7b98 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -241,6 +241,7 @@
 	rt5631: rt5631@1a {
 		compatible = "rt5631";
 		reg = <0x1a>;
+		phone_det_level = <1>;
 	};
 	
 };
-- 
2.35.3

