From 8e6e5648c66dbbe6db14c41b0ab7df12cf9081d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 7 Apr 2012 14:16:38 +0800
Subject: [PATCH] rk30: cpufreq: remove one line debug print

---
 arch/arm/mach-rk30/cpufreq.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index d3bac9c1cbd1..e2d8eb3a2319 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -311,7 +311,6 @@ static unsigned int cpufreq_scale_limt(unsigned int target_freq,struct cpufreq_p
 #ifdef CONFIG_RK30_CPU_FREQ_LIMIT_BY_TEMP
 	if(rk30_cpufreq_is_ondemand_policy(policy)&&temp_limt_freq)
 	{
-		printk("temp_limt_freq=%u\n",temp_limt_freq);
 		target_freq=min(target_freq,temp_limt_freq);
 	}
 #endif
-- 
2.35.3

