From 79794223ea6b8e1f6d99e54aeb8074c3fb816e05 Mon Sep 17 00:00:00 2001
From: chm <chm@rock-chips.com>
Date: Thu, 9 Aug 2012 14:25:13 +0800
Subject: [PATCH] rk2928: fix RK29_VCODEC_PHYS definition compile error

---
 arch/arm/plat-rk/vpu_service.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/plat-rk/vpu_service.c b/arch/arm/plat-rk/vpu_service.c
index e254a9b71657..4d3ffc16a524 100644
--- a/arch/arm/plat-rk/vpu_service.c
+++ b/arch/arm/plat-rk/vpu_service.c
@@ -76,6 +76,9 @@ static const u16 dec_hw_ids[] = { 0x8190, 0x8170, 0x9170, 0x9190, 0x6731 };
 #if defined(CONFIG_ARCH_RK29) || defined(CONFIG_ARCH_RK2928)
 static const u16 enc_hw_ids[] = { 0x6280, 0x7280, 0x8270 };
 #define DEC_PHY_OFFSET 				0x200
+#if defined(CONFIG_ARCH_RK2928)
+#define RK29_VCODEC_PHYS			RK2928_VCODEC_PHYS
+#endif
 #elif defined(CONFIG_ARCH_RK30)
 static const u16 enc_hw_ids[] = { 0x6280, 0x7280, 0x8270, 0x8290, 0x4831 };
 #define DEC_PHY_OFFSET 				0x400
-- 
2.35.3

