From 48ef6f66b80e02d6c78bd9a039e40f5d8d380793 Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Thu, 26 Mar 2015 10:38:56 +0800
Subject: [PATCH] rk3368: dvfs: avoid null-pointer dereference in dvfs_get_temp

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm/mach-rockchip/dvfs.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/dvfs.c b/arch/arm/mach-rockchip/dvfs.c
index 382e37fdc0f4..f800200f481a 100644
--- a/arch/arm/mach-rockchip/dvfs.c
+++ b/arch/arm/mach-rockchip/dvfs.c
@@ -44,11 +44,15 @@ struct regulator *vdd_gpu_regulator;
 
 static int dvfs_get_temp(int chn)
 {
-	int temp = 0;
+	int temp = INVALID_TEMP;
 
 #if IS_ENABLED(CONFIG_ROCKCHIP_THERMAL)
 	int read_back = 0;
 
+	if (clk_cpu_bl_dvfs_node == NULL ||
+	    IS_ERR_OR_NULL(clk_cpu_bl_dvfs_node->vd->regulator))
+		return temp;
+
 	mutex_lock(&clk_cpu_bl_dvfs_node->vd->mutex);
 	read_back = dvfs_regulator_get_voltage(
 		clk_cpu_bl_dvfs_node->vd->regulator);
-- 
2.35.3

