From c05a124c5755a0d65298a4ca1325678b4f890f22 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=93=9D=E6=96=8C=E5=85=83?= <lby@rock-chips.com>
Date: Thu, 7 Mar 2013 10:18:31 +0800
Subject: [PATCH] add board path for /sys/module/board/parameters/

---
 arch/arm/mach-rk30/Makefile | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/arm/mach-rk30/Makefile b/arch/arm/mach-rk30/Makefile
index 00296ee26a20..6f9db8b94c40 100755
--- a/arch/arm/mach-rk30/Makefile
+++ b/arch/arm/mach-rk30/Makefile
@@ -38,14 +38,14 @@ obj-$(CONFIG_MACH_RK30_PHONE_LOQUAT) += board-rk30-phone-loquat.o board-rk30-pho
 obj-$(CONFIG_MACH_RK30_DS1001B) += board-rk30-ds1001b.o board-rk30-ds1001b-key.o board-rk30-ds1001b-rfkill.o
 obj-$(CONFIG_MACH_RK30_PHONE_A22) += board-rk30-phone-a22.o board-rk30-phone-a22-key.o 
 
-obj-$(CONFIG_MACH_RK3066B_FPGA) += board-rk3066b-fpga.o
-obj-$(CONFIG_MACH_RK3066B_SDK) += board-rk3066b-sdk.o
-obj-$(CONFIG_MACH_RK3066B_M701) += board-rk3066b-m701.o
-obj-$(CONFIG_MACH_RK3168M_TB) += board-rk3168m-tb.o
-obj-$(CONFIG_MACH_RK3108_TB) += board-rk3108-tb.o
-obj-$(CONFIG_MACH_RK3168_TB) += board-rk3168-tb.o
-obj-$(CONFIG_MACH_RK3168_LR097) += board-rk3168-LR097.o
-obj-$(CONFIG_MACH_RK3168_DS1006H) += board-rk3168-ds1006h.o
-obj-$(CONFIG_MACH_RK3168_86V) += board-rk3168-86v.o
-obj-$(CONFIG_MACH_RK3168_86V_OLD) += board-rk3168-86v-old.o
+board-$(CONFIG_MACH_RK3066B_FPGA) += board-rk3066b-fpga.o
+board-$(CONFIG_MACH_RK3066B_SDK) += board-rk3066b-sdk.o
+board-$(CONFIG_MACH_RK3066B_M701) += board-rk3066b-m701.o
+board-$(CONFIG_MACH_RK3168M_TB) += board-rk3168m-tb.o
+board-$(CONFIG_MACH_RK3108_TB) += board-rk3108-tb.o
+board-$(CONFIG_MACH_RK3168_TB) += board-rk3168-tb.o
+board-$(CONFIG_MACH_RK3168_LR097) += board-rk3168-LR097.o
+board-$(CONFIG_MACH_RK3168_DS1006H) += board-rk3168-ds1006h.o
+board-$(CONFIG_MACH_RK3168_86V) += board-rk3168-86v.o
+board-$(CONFIG_MACH_RK3168_86V_OLD) += board-rk3168-86v-old.o
 
-- 
2.35.3

