From 2c6dd7f8291a60c064bb194858cb28b2a50056b1 Mon Sep 17 00:00:00 2001
From: JacobChen <cc@rock-chips.com>
Date: Mon, 5 Dec 2011 15:00:31 +0800
Subject: [PATCH] Mtp func: prevent dev state change frome cancel to error.

---
 drivers/usb/gadget/f_mtp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/f_mtp.c b/drivers/usb/gadget/f_mtp.c
index 2829231327d4..d1c259e3eb2d 100644
--- a/drivers/usb/gadget/f_mtp.c
+++ b/drivers/usb/gadget/f_mtp.c
@@ -362,7 +362,7 @@ static void mtp_complete_out(struct usb_ep *ep, struct usb_request *req)
 	struct mtp_dev *dev = _mtp_dev;
 
 	dev->rx_done = 1;
-	if (req->status != 0)
+	if (req->status != 0 && dev->state != STATE_CANCELED)
 		dev->state = STATE_ERROR;
 
 	wake_up(&dev->read_wq);
-- 
2.35.3

