From d65f5701d47e2d2420f1f7e4adac4959b05d7de2 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 11 Sep 2012 17:03:32 +0800
Subject: [PATCH] rk2928: limit gpu freq [133,400]MHz

---
 arch/arm/mach-rk2928/cpufreq.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rk2928/cpufreq.c b/arch/arm/mach-rk2928/cpufreq.c
index 4f796715415d..0f9c842d231e 100644
--- a/arch/arm/mach-rk2928/cpufreq.c
+++ b/arch/arm/mach-rk2928/cpufreq.c
@@ -257,9 +257,7 @@ static int rk30_cpu_init(struct cpufreq_policy *policy)
 		clk_enable_dvfs(cpu_clk);
 
 		/* Limit gpu frequency between 133M to 400M */
-#if 0
 		dvfs_clk_enable_limit(gpu_clk, 133000000, 400000000);
-#endif
 
 		freq_wq = create_singlethread_workqueue("rk30_cpufreqd");
 #ifdef CONFIG_RK30_CPU_FREQ_LIMIT_BY_TEMP
-- 
2.35.3

