From 1c8302e7ba956630dc3fab1c945829e0ebad65eb Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Tue, 9 Oct 2012 10:23:20 +0800
Subject: [PATCH] rk3066b m701: add lcd_stb_pin ctrl

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index cfdc6fc68ec3..1a27d78d4cdb 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -661,6 +661,10 @@ static int rk_fb_io_disable(void)
 	{
 		gpio_set_value(LCD_EN_PIN, !LCD_EN_VALUE);
 	}
+	if(LCD_STB_PIN !=INVALID_GPIO)
+	{
+		gpio_set_value(LCD_STB_PIN, !LCD_STB_VALUE);
+	}
 	return 0;
 }
 static int rk_fb_io_enable(void)
@@ -673,6 +677,10 @@ static int rk_fb_io_enable(void)
 	{
 		gpio_set_value(LCD_EN_PIN, LCD_EN_VALUE);
 	}
+	if(LCD_STB_PIN !=INVALID_GPIO)
+	{
+		gpio_set_value(LCD_STB_PIN, LCD_STB_VALUE);
+	}
 	return 0;
 }
 
-- 
2.35.3

