From 0c690c89470d2563590e1da927f4cbb339009aec Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Sat, 12 Jul 2014 16:31:59 +0800
Subject: [PATCH] 3036: enable usb bypass

---
 arch/arm/mach-rockchip/rk3036.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 4ff1ecefda72..68405cc54e9a 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -82,7 +82,6 @@ static void __init rk3036_boot_mode_init(void)
     rockchip_boot_mode_init(flag, mode);
 }
 
-#if 0
 static void usb_uart_init(void)
 {
 #ifdef CONFIG_RK_USB_UART
@@ -96,17 +95,14 @@ static void usb_uart_init(void)
             writel_relaxed(0x34003000, RK_GRF_VIRT + RK3036_GRF_UOC1_CON4);
     }
 #endif
-#ifdef RK_DEBUG_UART_VIRT
-    writel_relaxed(0x07, RK_DEBUG_UART_VIRT + 0x88);
+
     writel_relaxed(0x07, RK_DEBUG_UART_VIRT + 0x88);
     writel_relaxed(0x00, RK_DEBUG_UART_VIRT + 0x04);
     writel_relaxed(0x83, RK_DEBUG_UART_VIRT + 0x0c);
     writel_relaxed(0x0d, RK_DEBUG_UART_VIRT + 0x00);
     writel_relaxed(0x00, RK_DEBUG_UART_VIRT + 0x04);
     writel_relaxed(0x03, RK_DEBUG_UART_VIRT + 0x0c);
-#endif //end of DEBUG_UART_BASE
 }
-#endif
 
 static void __init rk3036_dt_map_io(void)
 {
-- 
2.35.3

