From 7bbcfdf7f1edc130e6e00d211f15a61b1d8e0006 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Thu, 9 Aug 2012 18:42:58 +0800
Subject: [PATCH] rk30:phonepad:Remove debug print

---
 arch/arm/mach-rk30/i2c_sram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/i2c_sram.c b/arch/arm/mach-rk30/i2c_sram.c
index 184ab9a63c90..e871a5b0218c 100755
--- a/arch/arm/mach-rk30/i2c_sram.c
+++ b/arch/arm/mach-rk30/i2c_sram.c
@@ -147,7 +147,7 @@ void __sramfunc sram_i2c_init()
 	divh = divl = rk30_ceil(div, 2);	
 	writel_relaxed(I2C_CLKDIV_VAL(divl, divh), SRAM_I2C_ADDRBASE + I2C_CLKDIV);
 	data[3]  = readl_relaxed(SRAM_I2C_ADDRBASE + I2C_CLKDIV);
-	sram_printhex(data[3]);
+	
 }
 
 
-- 
2.35.3

