From f118410e82756b99e6af2f0847d37f3b46d52337 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Dec 2010 14:27:53 +0800
Subject: [PATCH] rk29: clock: more v0.3

---
 arch/arm/mach-rk29/clock.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index fc49f16319d1..383781399f8e 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2208,8 +2208,12 @@ static void __init rk29_clock_common_init(void)
 	clk_set_parent_nolock(&dclk_lcdc_div, &general_pll_clk);
 	clk_set_parent_nolock(&aclk_lcdc, &general_pll_clk);
 	clk_set_parent_nolock(&aclk_vepu, &general_pll_clk);
+	clk_set_rate_nolock(&aclk_vepu, 288 * MHZ);
+	clk_set_rate_nolock(&clk_aclk_ddr_vepu, 288 * MHZ);
+	clk_set_rate_nolock(&hclk_vepu, 144 * MHZ);
 	clk_set_parent_nolock(&aclk_vdpu, &general_pll_clk);
 	clk_set_parent_nolock(&aclk_gpu, &general_pll_clk);
+	clk_set_rate_nolock(&aclk_gpu, 288 * MHZ);
 	clk_set_parent_nolock(&clk_mac_ref_div, &general_pll_clk);
 	clk_set_parent_nolock(&clk_hsadc_div, &general_pll_clk);
 
-- 
2.35.3

