From 525fdeba3fbe14b0617f0a86891323fd61bbdf70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Thu, 16 Feb 2012 18:10:08 +0800
Subject: [PATCH] wm8994 add 50ms delay between wm_en and i2c read register in
 device setup

---
 drivers/regulator/wm8994-regulator.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/regulator/wm8994-regulator.c b/drivers/regulator/wm8994-regulator.c
index 91c8043fc576..348bfa844439 100755
--- a/drivers/regulator/wm8994-regulator.c
+++ b/drivers/regulator/wm8994-regulator.c
@@ -21,6 +21,7 @@
 #include <linux/gpio.h>
 #include <linux/slab.h>
 
+#include <linux/delay.h>
 #include <linux/mfd/wm8994/core.h>
 #include <linux/mfd/wm8994/registers.h>
 #include <linux/mfd/wm8994/pdata.h>
@@ -258,6 +259,7 @@ static __devinit int wm8994_ldo_probe(struct platform_device *pdev)
 				ret);
 			goto err_gpio;
 		}
+		msleep(50);
 	} else
 		ldo->is_enabled = true;
 
-- 
2.35.3

