From b283ec8cc448d5cfec727ce96eb633a23e267e0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 15:41:14 +0800
Subject: [PATCH] temp revert alarm change

---
 drivers/rtc/alarm.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/rtc/alarm.c b/drivers/rtc/alarm.c
index a6c07075a709..7232a538d399 100644
--- a/drivers/rtc/alarm.c
+++ b/drivers/rtc/alarm.c
@@ -419,7 +419,6 @@ static int alarm_suspend(struct platform_device *pdev, pm_message_t state)
 		if (rtc_current_time + 1 >= rtc_alarm_time) {
 			pr_alarm(SUSPEND, "alarm about to go off\n");
 			memset(&rtc_alarm, 0, sizeof(rtc_alarm));
-			rtc_time_to_tm(0, &rtc_alarm.time);
 			rtc_alarm.enabled = 0;
 			rtc_set_alarm(alarm_rtc_dev, &rtc_alarm);
 
@@ -445,7 +444,6 @@ static int alarm_resume(struct platform_device *pdev)
 	pr_alarm(SUSPEND, "alarm_resume(%p)\n", pdev);
 
 	memset(&alarm, 0, sizeof(alarm));
-	rtc_time_to_tm(0, &alarm.time);
 	alarm.enabled = 0;
 	rtc_set_alarm(alarm_rtc_dev, &alarm);
 
@@ -459,18 +457,6 @@ static int alarm_resume(struct platform_device *pdev)
 	return 0;
 }
 
-static void alarm_shutdown(struct platform_device *pdev)
-{
-	struct rtc_wkalrm alarm;
-
-	pr_alarm(FLOW, "alarm_shutdown(%p)\n", pdev);
-
-	memset(&alarm, 0, sizeof(alarm));
-	rtc_time_to_tm(0, &alarm.time);
-	alarm.enabled = 0;
-	rtc_set_alarm(alarm_rtc_dev, &alarm);
-}
-
 static struct rtc_task alarm_rtc_task = {
 	.func = alarm_triggered_func
 };
@@ -530,7 +516,6 @@ static struct class_interface rtc_alarm_interface = {
 static struct platform_driver alarm_driver = {
 	.suspend = alarm_suspend,
 	.resume = alarm_resume,
-	.shutdown = alarm_shutdown,
 	.driver = {
 		.name = "alarm"
 	}
-- 
2.35.3

