From 9796e3348921fd21c79b69cc3d680bcf910272d6 Mon Sep 17 00:00:00 2001
From: yangyun <yangyun@rock-chips.com>
Date: Tue, 31 Jan 2012 11:48:49 +0800
Subject: [PATCH] alter path of toolchain in pack-kernel.sh

---
 pack-kernel.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 7e2761b17634..9df543470c8b 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -152,7 +152,7 @@ for file in ${EXCLUDES[@]}; do
 done
 echo TAR $(pwd)/$package
 tar cf $package --numeric-owner --exclude-from $ex --exclude=.git --exclude=`basename $0` $(basename $kerndir)
-tar rf $package --numeric-owner --exclude=.git toolchain/arm-eabi-4.4.0
+tar rf $package --numeric-owner --exclude=.git prebuilt/linux-x86/toolchain/arm-eabi-4.4.0
 echo GZIP $(pwd)/$package.gz
 gzip -9 -c $package > $package.gz
 rm $ex
-- 
2.35.3

