From 523c764f960f32c3d108ac63b7553964558e1ced Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 27 Sep 2012 09:43:22 +0800
Subject: [PATCH] rk2928 codec: enable adc 6db digital gain.

---
 sound/soc/codecs/rk2928_codec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/rk2928_codec.c b/sound/soc/codecs/rk2928_codec.c
index 8fe8fedb9f0f..b3de543b0ee6 100755
--- a/sound/soc/codecs/rk2928_codec.c
+++ b/sound/soc/codecs/rk2928_codec.c
@@ -155,8 +155,8 @@ static int rk2928_audio_trigger(struct snd_pcm_substream *substream, int cmd,
 			}
 			else {
 				rk2928_write(codec, CODEC_REG_ADC_PGA_GAIN, 0xFF);
-//				rk2928_write(codec, 0x08, 0xff);
-//				rk2928_write(codec, 0x09, 0x07);
+				rk2928_write(codec, 0x08, 0xff);
+				rk2928_write(codec, 0x09, 0x07);
 			}
 			break;
 		case SNDRV_PCM_TRIGGER_STOP:
-- 
2.35.3

