From e587a1ca4814162cdd868bffe1e75e152d723b5a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 28 Jun 2011 16:27:10 +0800
Subject: [PATCH] Staging: dream: Kconfig fix for non-HTC hardware

---
 drivers/staging/dream/Kconfig | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/dream/Kconfig b/drivers/staging/dream/Kconfig
index 52bd187de5a7..263b17d2dc6d 100644
--- a/drivers/staging/dream/Kconfig
+++ b/drivers/staging/dream/Kconfig
@@ -1,12 +1,15 @@
+config DREAM
+	tristate "HTC Dream support"
+	depends on MACH_TROUT
+
+if DREAM
+
 source "drivers/staging/dream/smd/Kconfig"
 
 source "drivers/staging/dream/camera/Kconfig"
 
-
 config INPUT_GPIO
 	tristate "GPIO driver support"
 	help
 	  Say Y here if you want to support gpio based keys, wheels etc...
-
-
-
+endif
-- 
2.35.3

