From de5d667375f7f08cc9f49f7b57d36275c1b7cc9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 20 Apr 2012 12:37:53 +0800
Subject: [PATCH] rk30: cpufreq: flags add CPUFREQ_CONST_LOOPS for non-SMP
 systems

---
 arch/arm/mach-rk30/cpufreq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index cad6459dd238..704323390e0e 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -433,6 +433,7 @@ static struct notifier_block rk30_cpufreq_reboot_notifier = {
 };
 
 static struct cpufreq_driver rk30_cpufreq_driver = {
+	.flags		= CPUFREQ_CONST_LOOPS,
 	.verify		= rk30_verify_speed,
 	.target		= rk30_target,
 	.get		= rk30_getspeed,
-- 
2.35.3

