From 868e165d4f41f7b15b3883c5eacb1066f5909e8e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E6=81=92=E6=98=8E?= <chm@rock-chips.com>
Date: Tue, 10 Apr 2012 09:48:58 +0800
Subject: [PATCH] rk30: vpu_service: limit vpu_reset() only on RK30

---
 arch/arm/plat-rk/vpu_service.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/plat-rk/vpu_service.c b/arch/arm/plat-rk/vpu_service.c
index ce6ed0648106..e9aa4937ea3d 100644
--- a/arch/arm/plat-rk/vpu_service.c
+++ b/arch/arm/plat-rk/vpu_service.c
@@ -450,7 +450,9 @@ void reg_copy_to_hw(vpu_reg *reg)
 		u32 *dst = (u32 *)enc_dev.hwregs;
 		service.reg_codec = reg;
 
+#if defined(CONFIG_ARCH_RK30)
 		vpu_reset();
+#endif
 
 		dst[VPU_REG_EN_ENC] = src[VPU_REG_EN_ENC] & 0x6;
 
-- 
2.35.3

