From 9118760938369525c36817e04424230f1ee7e77a Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 13 Apr 2015 11:13:32 +0800
Subject: [PATCH] rk fb: do something when user set invalid config

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index b663a6ea5a96..1ff46f27d4ba 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -2218,6 +2218,9 @@ static int rk_fb_set_win_config(struct fb_info *info,
 		}
 	}
 
+	if (regs->win_num <= 0)
+		goto err_null_frame;
+
 	mutex_lock(&dev_drv->output_lock);
 
 	dev_drv->timeline_max++;
@@ -2288,6 +2291,13 @@ static int rk_fb_set_win_config(struct fb_info *info,
 err:
 	mutex_unlock(&dev_drv->output_lock);
 	return ret;
+err_null_frame:
+	kfree(regs);
+	for (j = 0; j < RK_MAX_BUF_NUM; j++)
+		win_data->rel_fence_fd[j] = -1;
+	win_data->ret_fence_fd = -1;
+	pr_info("win num = %d,null frame\n", regs->win_num);
+	return 0;
 }
 
 #if 1
-- 
2.35.3

