From e2911b98ca0295f968be1a4f1dbe10ce8e1073ea Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 7 Mar 2014 19:33:41 +0800
Subject: [PATCH] mmc: rk_sdmmc: fix dw_mci_resume

---
 drivers/mmc/host/rk_sdmmc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index ebd0edbb09d9..c3ce486ed7ac 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -2807,9 +2807,9 @@ int dw_mci_resume(struct dw_mci *host)
 			dw_mci_setup_bus(slot, true);
 		}
 
-		ret = mmc_resume_host(host->slot[i]->mmc);
-		if (ret < 0)
-			return ret;
+	//	ret = mmc_resume_host(host->slot[i]->mmc);
+	//	if (ret < 0)
+	//		return ret;
 	}
 	return 0;
 }
-- 
2.35.3

