From 028a0082b7dcc5ab850e191053d9fa06f5e57ebf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 19 Mar 2011 16:16:12 +0800
Subject: [PATCH] rk29: timer: clock source do not need irq

---
 arch/arm/mach-rk29/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/timer.c b/arch/arm/mach-rk29/timer.c
index b00430d78533..89f24857346b 100644
--- a/arch/arm/mach-rk29/timer.c
+++ b/arch/arm/mach-rk29/timer.c
@@ -33,7 +33,7 @@
 
 #define TIMER_DISABLE			6
 #define TIMER_ENABLE			3
-#define TIMER_ENABLE_FREE_RUNNING	1
+#define TIMER_ENABLE_FREE_RUNNING	5
 
 #if 1	/* by default, use periph sync timer */
 
-- 
2.35.3

