From 6e0861f97ce2894f5dc81cd79d03b04f51566065 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Mon, 27 May 2013 10:24:40 +0800
Subject: [PATCH] rk30 i2s : modify CONFIG_HDMI_RK30 to CONFIG_RK_HDMI

---
 sound/soc/rk29/rk30_i2s.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index 230fd1c226a1..d22d3173d516 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -89,7 +89,7 @@ static struct rk29_i2s_info rk29_i2s[MAX_I2S];
 
 struct snd_soc_dai_driver rk29_i2s_dai[MAX_I2S];
 EXPORT_SYMBOL_GPL(rk29_i2s_dai);
-#ifdef CONFIG_HDMI_RK30		
+#ifdef CONFIG_RK_HDMI		
 extern int hdmi_get_hotplug(void);
 #endif
 /* 
@@ -125,7 +125,7 @@ static void rockchip_snd_txctrl(struct rk29_i2s_info *i2s, int on)
 		opr  &= ~I2S_TRAN_DMA_ENABLE;        
 		writel(opr, &(pheadi2s->I2S_DMACR));  
 		if(!i2s->i2s_tx_status && !i2s->i2s_rx_status//sync stop i2s rx tx lcrk
-#ifdef CONFIG_HDMI_RK30	
+#ifdef CONFIG_RK_HDMI	
 			&& 	hdmi_get_hotplug() == 0	//HDMI_HPD_REMOVED
 #endif			
 		)
@@ -175,7 +175,7 @@ static void rockchip_snd_rxctrl(struct rk29_i2s_info *i2s, int on)
 		opr  &= ~I2S_RECE_DMA_ENABLE;
 		writel(opr, &(pheadi2s->I2S_DMACR));		
 		if(!i2s->i2s_tx_status && !i2s->i2s_rx_status	//sync stop i2s rx tx lcrk
-#ifdef CONFIG_HDMI_RK30	
+#ifdef CONFIG_RK_HDMI	
 			&& 	hdmi_get_hotplug() == 0	//HDMI_HPD_REMOVED
 #endif			
 		)
-- 
2.35.3

