From 1a163a44869160cdb4b2cf1c1a74ef02e2c75a14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 15 May 2012 14:37:58 +0800
Subject: [PATCH] pack-kernel: rk30 exclude more phone config

---
 pack-kernel-rk30.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pack-kernel-rk30.sh b/pack-kernel-rk30.sh
index e3ef9cad1327..6c411e852222 100755
--- a/pack-kernel-rk30.sh
+++ b/pack-kernel-rk30.sh
@@ -29,7 +29,7 @@ drivers/staging/rk29/vivante
 drivers/staging/rk29/ipp/rk29-ipp.c
 
 arch/arm/mach-rk30/board-rk30-phone*
-arch/arm/configs/rk30_phone_defconfig
+arch/arm/configs/rk30_phone_*
 )
 
 defconfig=${2-rk30_sdk_defconfig}
-- 
2.35.3

