From e09b4cd5a81c7364f973534d61432da3fd4f0a43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 26 Mar 2012 14:39:56 +0800
Subject: [PATCH] rk: module.c: enable debug locks for rk plat

---
 kernel/module.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/module.c b/kernel/module.c
index 4b52684a013e..242a688757d4 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1982,6 +1982,10 @@ static void layout_sections(struct module *mod, struct load_info *info)
 
 static void set_license(struct module *mod, const char *license)
 {
+#ifdef CONFIG_PLAT_RK
+	return;
+#endif
+
 	if (!license)
 		license = "unspecified";
 
-- 
2.35.3

