From c2370f88727cc0472f458b578ce758d3547e1226 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 17 Sep 2014 16:41:04 +0800
Subject: [PATCH] ARM: rockchip: printk.c: fix compilation warning

---
 kernel/printk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/printk.c b/kernel/printk.c
index 0f7eb6c133e0..cbf71b406e33 100644
--- a/kernel/printk.c
+++ b/kernel/printk.c
@@ -1717,7 +1717,7 @@ asmlinkage int printk(const char *fmt, ...)
 }
 EXPORT_SYMBOL(printk);
 
-#if CONFIG_RK_DEBUG_UART >= 0
+#if defined(CONFIG_RK_DEBUG_UART) && (CONFIG_RK_DEBUG_UART >= 0)
 void console_disable_suspend(void)
 {
 	console_suspended = 0;
-- 
2.35.3

