From 48b21a2c31d042d55b21d7af94b8002301684b65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 16 Jan 2013 15:14:13 +0800
Subject: [PATCH] rk610-core: support new iomux api on rk3066b

---
 drivers/mfd/rk610-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/rk610-core.c b/drivers/mfd/rk610-core.c
index 88054431828a..68184a72511b 100755
--- a/drivers/mfd/rk610-core.c
+++ b/drivers/mfd/rk610-core.c
@@ -240,7 +240,7 @@ static int rk610_control_probe(struct i2c_client *client,
 		#if defined(CONFIG_ARCH_RK29)
 		rk29_mux_api_set(GPIO2D0_I2S0CLK_MIIRXCLKIN_NAME, GPIO2H_I2S0_CLK);
 		#elif defined(CONFIG_ARCH_RK3066B)
-		rk30_mux_api_set(GPIO1C0_I2SCLK_NAME, GPIO1C_I2SCLK);
+		iomux_set(I2S0_CLK);
 		#elif defined(CONFIG_ARCH_RK30)
                 rk30_mux_api_set(GPIO0B0_I2S8CHCLK_NAME, GPIO0B_I2S_8CH_CLK);
 		#endif
-- 
2.35.3

