From 87b23fc6f370eea5268f996239988df52e922330 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sun, 9 Oct 2011 14:12:14 +0800
Subject: [PATCH] regulator: act8891.h: fix 'struct act8891' forward
 declaration

---
 include/linux/regulator/act8891.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/regulator/act8891.h b/include/linux/regulator/act8891.h
index bab564f7dc48..8e8a14726f12 100755
--- a/include/linux/regulator/act8891.h
+++ b/include/linux/regulator/act8891.h
@@ -31,7 +31,7 @@
 
 
 #define act8891_NUM_REGULATORS 7
-static struct act8891 *act8891;
+struct act8891;
 
 struct act8891_regulator_subdev {
 	int id;
-- 
2.35.3

