From 1b9ec24e0a6d274fd68033748a1f4c14764161d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 31 Dec 2010 09:26:29 +0800
Subject: [PATCH] rk29_backlight: init eraly, show logo early

---
 drivers/video/backlight/rk29_backlight.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/video/backlight/rk29_backlight.c b/drivers/video/backlight/rk29_backlight.c
index f6fff00e50e7..ecb362ee57b4 100644
--- a/drivers/video/backlight/rk29_backlight.c
+++ b/drivers/video/backlight/rk29_backlight.c
@@ -304,6 +304,7 @@ static int rk29_backlight_probe(struct platform_device *pdev)
     register_early_suspend(&bl_early_suspend);
 #endif
 
+    printk("RK29 Backlight Driver Initialized.\n");
     return ret;
 }
 
@@ -387,7 +388,4 @@ static int __init rk29_backlight_init(void)
 	platform_driver_register(&rk29_backlight_driver);
 	return 0;
 }
-//rootfs_initcall(rk29_backlight_init);
-
-late_initcall(rk29_backlight_init);
-//module_init(rk29_backlight_init);
+postcore_initcall_sync(rk29_backlight_init);
-- 
2.35.3

