From 075d8e0b992db44c4b573f7288b50bc636798f6b Mon Sep 17 00:00:00 2001
From: zyc <zyc@rock-chips.com>
Date: Sat, 21 Aug 2010 14:52:34 +0800
Subject: [PATCH] add cmmb object.

---
 drivers/Kconfig  | 3 +++
 drivers/Makefile | 1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/Kconfig b/drivers/Kconfig
index 24aee0fcf34a..ea153397d1fc 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -119,4 +119,7 @@ source "drivers/xen/Kconfig"
 source "drivers/staging/Kconfig"
 
 source "drivers/platform/Kconfig"
+
+source "drivers/cmmb/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 92e2365041fc..fe5282739cba 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -113,3 +113,4 @@ obj-$(CONFIG_VLYNQ)		+= vlynq/
 obj-$(CONFIG_STAGING)		+= staging/
 obj-y				+= platform/
 obj-y				+= ieee802154/
+obj-$(CONFIG_CMMB)		+= cmmb/	
-- 
2.35.3

