From e4c098b438276c2fad89e8bcc04918a9e407cd6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sun, 25 Mar 2012 11:53:20 +0800
Subject: [PATCH] rk29: Kconfig: reformat MACH_RK29_2906

---
 arch/arm/mach-rk29/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rk29/Kconfig b/arch/arm/mach-rk29/Kconfig
index 7af906271e2f..e9596a60f19a 100755
--- a/arch/arm/mach-rk29/Kconfig
+++ b/arch/arm/mach-rk29/Kconfig
@@ -1,5 +1,8 @@
 if ARCH_RK29
 
+config MACH_RK29_2906
+	bool "ROCKCHIP RK2906 Feature"
+
 choice
 	prompt "Select Board Type"
 	depends on ARCH_RK29
@@ -163,9 +166,6 @@ config DDR_SDRAM_FREQ
 	int "DDR SDRAM frequence (in MHz)"
 	default 400
 
-config MACH_RK29_2906
-	bool "ROCKCHIP RK2906 Feature"
-
 config DDR_FREQ
 	bool "Enable DDR frequency scaling"
 
-- 
2.35.3

