From d2afa052215b8fab8c1a772c3f0b7835b12d7864 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Wed, 28 Sep 2011 16:35:11 +0800
Subject: [PATCH] rk29fb:fix the cast bug in WIN1 DYNAMIC CLOSE

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 464331509405..918197cbfdc6 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1238,7 +1238,7 @@ static void win1_check_work_func(struct work_struct *work)
             }
             break;
     }
-    p = (int*)fb0_inf->screen_base + offset; 
+    p = (u32)fb0_inf->screen_base + offset; 
     blank_data = (inf->fb0_color_deepth==32) ? 0xff000000 : 0;
     total_data = fb0_inf->var.xres * fb0_inf->var.yres / (format+1);
     
-- 
2.35.3

