From 1870cd71532db20594051aca419d4939c3128bdd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 7 Dec 2010 17:17:29 +0800
Subject: [PATCH] add kernel.img target

---
 arch/arm/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 798442a24c1b..fde93af92ad3 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -271,8 +271,13 @@ archclean:
 bp:;	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $(boot)/bootpImage
 i zi:;	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $@
 
+PHONY += kernel.img
+kernel.img: Image FORCE
+	$(Q)$(srctree)/mkkrnlimg $(obj)/arch/arm/boot/Image $(obj)/kernel.img
+	@echo '  Image:  $@ is ready'
 
 define archhelp
+  echo  '  kernel.img    - Rockchip kernel image'
   echo  '* zImage        - Compressed kernel image (arch/$(ARCH)/boot/zImage)'
   echo  '  Image         - Uncompressed kernel image (arch/$(ARCH)/boot/Image)'
   echo  '* xipImage      - XIP kernel image, if configured (arch/$(ARCH)/boot/xipImage)'
-- 
2.35.3

