From 88fe0132cc3beeb7692c29b54fc9d92086d8308d Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 25 May 2015 15:26:54 +0800
Subject: [PATCH] arm64: rockchip: rk3368: dts: add rockchip,rk3368-efuse-256

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index eb2a3e526189..f65097e21a70 100644
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -288,6 +288,11 @@
 		};
 	};
 
+	efuse_256@ffb00000 {
+		compatible = "rockchip,rk3368-efuse-256";
+		reg = <0x0 0xffb00000 0x0 0x8>;
+	};
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_HIGH)>,
-- 
2.35.3

