From 49c796842f02e1795752c5df562732b382264532 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Sat, 5 Jan 2013 15:14:31 +0800
Subject: [PATCH] rk30:modify pwm clk when in sleep,set clk 660k

---
 drivers/regulator/rk30-pwm-regulator.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/regulator/rk30-pwm-regulator.c b/drivers/regulator/rk30-pwm-regulator.c
index 34dae77e77b7..bead02ecb183 100755
--- a/drivers/regulator/rk30-pwm-regulator.c
+++ b/drivers/regulator/rk30-pwm-regulator.c
@@ -269,21 +269,21 @@ static int __devinit pwm_regulator_probe(struct platform_device *pdev)
 	}
 	clk_enable(dcdc->pwm_clk);
 
-	dcdc->suspend_lrc = 0x25;
+	dcdc->suspend_lrc = 0x12;
 	switch (pdata->suspend_voltage)
 	{
 	case 1000000:
 	default:
-		dcdc->suspend_hrc = 0x20;
+		dcdc->suspend_hrc = 0x10;
 		break;
 	case 1050000:
-		dcdc->suspend_hrc = 0x1c;
+		dcdc->suspend_hrc = 0x0e;
 		break;
 	case 1100000:
-		dcdc->suspend_hrc = 0x18;
+		dcdc->suspend_hrc = 0x0c;
 		break;
 	case 1150000:
-		dcdc->suspend_hrc = 0x13;
+		dcdc->suspend_hrc = 0x0a;
 		break;
 	}
 
-- 
2.35.3

