From d8a4cd863b29f7481d6c71deed7334838107e264 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 9 Aug 2011 10:03:30 +0800
Subject: [PATCH] input: touchscreen: Makefile: indent only

---
 drivers/input/touchscreen/Makefile | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/input/touchscreen/Makefile b/drivers/input/touchscreen/Makefile
index 1bacb58bed3a..4c05a40954ae 100644
--- a/drivers/input/touchscreen/Makefile
+++ b/drivers/input/touchscreen/Makefile
@@ -43,20 +43,20 @@ obj-$(CONFIG_TOUCHSCREEN_WM97XX_MAINSTONE)	+= mainstone-wm97xx.o
 obj-$(CONFIG_TOUCHSCREEN_WM97XX_ZYLONITE)	+= zylonite-wm97xx.o
 obj-$(CONFIG_TOUCHSCREEN_W90X900)	+= w90p910_ts.o
 obj-$(CONFIG_TOUCHSCREEN_PCAP)		+= pcap_ts.o
-obj-$(CONFIG_TOUCHSCREEN_XPT2046_NORMAL_SPI)		+= xpt2046_ts.o
-obj-$(CONFIG_TOUCHSCREEN_XPT2046_TSLIB_SPI)		+= xpt2046_tslib_ts.o ts_lib/
-obj-$(CONFIG_TOUCHSCREEN_XPT2046_CBN_SPI)		+= xpt2046_cbn_ts.o calibration_ts.o largenum_ts.o calib_iface_ts.o
-obj-$(CONFIG_TOUCHSCREEN_IT7250) += ctp_it7250.o
+obj-$(CONFIG_TOUCHSCREEN_XPT2046_NORMAL_SPI)	+= xpt2046_ts.o
+obj-$(CONFIG_TOUCHSCREEN_XPT2046_TSLIB_SPI)	+= xpt2046_tslib_ts.o ts_lib/
+obj-$(CONFIG_TOUCHSCREEN_XPT2046_CBN_SPI)	+= xpt2046_cbn_ts.o calibration_ts.o largenum_ts.o calib_iface_ts.o
+obj-$(CONFIG_TOUCHSCREEN_IT7250)	+= ctp_it7250.o
 obj-$(CONFIG_RK28_I2C_TS_NTP070)	+= ntp070.o
 obj-$(CONFIG_HANNSTAR_P1003)		+= hannstar_p1003.o
-obj-$(CONFIG_TOUCHSCREEN_IT7260)		+= it7260_ts.o
-obj-$(CONFIG_SINTEK_3FA16)	+=  sintek_3FA16.o
+obj-$(CONFIG_TOUCHSCREEN_IT7260)	+= it7260_ts.o
+obj-$(CONFIG_SINTEK_3FA16)		+= sintek_3FA16.o
 obj-$(CONFIG_EETI_EGALAX)		+= eeti_egalax_i2c.o
-obj-$(CONFIG_ATMEL_MXT224)               += atmel_maxtouch.o
-obj-$(CONFIG_TOUCHSCREEN_GT801_IIC)      += gt801_ts.o
-obj-$(CONFIG_TOUCHSCREEN_GT818_IIC)      += gt818_ts.o
-obj-$(CONFIG_TOUCHSCREEN_ILI2102_IIC)      += ili2102_ts.o
-obj-$(CONFIG_D70_L3188A)      += goodix_touch.o
-obj-$(CONFIG_TOUCHSCREEN_FT5406) += ft5406_ts.o  
-obj-$(CONFIG_TOUCHSCREEN_GT819) += gt819.o
-obj-$(CONFIG_TOUCHSCREEN_NAS)      += nas_ts.o
+obj-$(CONFIG_ATMEL_MXT224)		+= atmel_maxtouch.o
+obj-$(CONFIG_TOUCHSCREEN_GT801_IIC)	+= gt801_ts.o
+obj-$(CONFIG_TOUCHSCREEN_GT818_IIC)	+= gt818_ts.o
+obj-$(CONFIG_TOUCHSCREEN_ILI2102_IIC)	+= ili2102_ts.o
+obj-$(CONFIG_D70_L3188A)		+= goodix_touch.o
+obj-$(CONFIG_TOUCHSCREEN_FT5406)	+= ft5406_ts.o
+obj-$(CONFIG_TOUCHSCREEN_GT819)		+= gt819.o
+obj-$(CONFIG_TOUCHSCREEN_NAS)		+= nas_ts.o
-- 
2.35.3

