From e00d7ceccb7d2f48865f1cb92ed049cfd4d44623 Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Tue, 16 Apr 2013 17:45:12 +0800
Subject: [PATCH] fix sysc wait queue error

---
 drivers/video/rockchip/rga/rga_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index 9c61d838cd13..fe5eec29d0f9 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -880,6 +880,8 @@ static int rga_blit_sync(rga_session *session, struct rga_req *req)
     printk("*** rga_blit_sync proc ***\n");
     print_info(req);
     #endif
+
+    atomic_set(&session->done, 0);
     
     ret = rga_blit(session, req);
     if(ret < 0)
-- 
2.35.3

