From 64fdea83d9575773f8015c5f8b708d30c04057da Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 21 Aug 2012 11:57:29 +0800
Subject: [PATCH] logo: bmp: default n

---
 drivers/video/logo/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/logo/Kconfig b/drivers/video/logo/Kconfig
index a783e87086c2..b53c9627852e 100755
--- a/drivers/video/logo/Kconfig
+++ b/drivers/video/logo/Kconfig
@@ -29,7 +29,6 @@ config LOGO_LINUX_CLUT224
 
 config LOGO_LINUX_BMP
        bool "bmp logo"
-       default y
 
 config LOGO_BLACKFIN_VGA16
 	bool "16-colour Blackfin Processor Linux logo"
-- 
2.35.3

