From ce551b679900328cd0d5cac93c273b292adf4523 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 21 Oct 2010 20:30:46 +0800
Subject: [PATCH] rk29 fpga use uncompressed uImage

---
 arch/arm/boot/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/Makefile b/arch/arm/boot/Makefile
index d4ed498aefce..8d4387a9d885 100644
--- a/arch/arm/boot/Makefile
+++ b/arch/arm/boot/Makefile
@@ -61,7 +61,7 @@ endif
 
 quiet_cmd_uimage = UIMAGE  $@
       cmd_uimage = $(CONFIG_SHELL) $(MKIMAGE) -A arm -O linux -T kernel \
-		   -C gzip -a $(LOADADDR) -e $(STARTADDR) \
+		   -C none -a $(LOADADDR) -e $(STARTADDR) \
 		   -n 'Linux-$(KERNELRELEASE)' -d $< $@
 
 ifeq ($(CONFIG_ZBOOT_ROM),y)
@@ -81,7 +81,7 @@ $(obj)/Image.gz: $(obj)/Image FORCE
 	$(call if_changed,gzip)
 	@echo '  Image:  $@ is ready'
 
-$(obj)/uImage: $(obj)/Image.gz FORCE
+$(obj)/uImage: $(obj)/Image FORCE
 	$(call if_changed,uimage)
 	@echo '  Image:  $@ is ready'
 
-- 
2.35.3

