From 16fe2d5531dd6f8b12f14e987e7e3f5ce456553f Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 25 Feb 2013 09:11:24 +0800
Subject: [PATCH] rk3168_86v: add maxbrightness restriction

---
 arch/arm/mach-rk30/board-rk3168-86v.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3168-86v.c b/arch/arm/mach-rk30/board-rk3168-86v.c
index dfe682afe9f0..02946cb6e9c4 100755
--- a/arch/arm/mach-rk30/board-rk3168-86v.c
+++ b/arch/arm/mach-rk30/board-rk3168-86v.c
@@ -278,6 +278,8 @@ static int rk29_backlight_pwm_resume(void)
 
 static struct rk29_bl_info rk29_bl_info = {
         .min_brightness = 30,
+	 .max_brightness=255,
+	 .brightness_mode =1,
 	.pre_div = 20 * 1000,  // pwm output clk: 20k;
 	.pwm_id = PWM_ID,
 	.bl_ref = PWM_EFFECT_VALUE,
-- 
2.35.3

