From d1b81ed90483c0d6161910fe2ed3acdc3da1d9cf Mon Sep 17 00:00:00 2001
From: liuy <liuy@rock-chips.com>
Date: Wed, 10 Aug 2011 18:43:56 +0800
Subject: [PATCH] dd macro compile control for phonepad project in anx7150.c

---
 drivers/video/hdmi/chips/anx7150.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/hdmi/chips/anx7150.c b/drivers/video/hdmi/chips/anx7150.c
index 51b33b0fec0c..74949e968b10 100755
--- a/drivers/video/hdmi/chips/anx7150.c
+++ b/drivers/video/hdmi/chips/anx7150.c
@@ -159,10 +159,12 @@ static int anx7150_i2c_probe(struct i2c_client *client,const struct i2c_device_i
 	struct hdmi *hdmi = NULL;
 	struct anx7150_pdata *anx = NULL;
 
+#if defined(CONFIG_MACH_RK29_PHONEPADSDK)
 	gpio_request(HDMI_VDD_CTL, "hdmi pwr ctl");
 	gpio_direction_output(HDMI_VDD_CTL, GPIO_HIGH);
 	//gpio_set_value(HDMI_VDD_CTL, GPIO_HIGH); 
 	mdelay(10);
+#endif
 
 	hdmi = hdmi_register(sizeof(struct anx7150_pdata), &client->dev);
     if (!hdmi)
-- 
2.35.3

