From 623036385b9cbcda4dd325156e2350e4b29ed9e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 21 Sep 2012 19:03:39 +0800
Subject: [PATCH] rk2928:a720:cancel the voltage limit of act8931

---
 drivers/regulator/act8931.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/act8931.c b/drivers/regulator/act8931.c
index 72ac5386f699..ba4e6ed30de5 100755
--- a/drivers/regulator/act8931.c
+++ b/drivers/regulator/act8931.c
@@ -307,7 +307,7 @@ static int act8931_dcdc_set_voltage(struct regulator_dev *dev,
 			break;}
 
 	if (vol_map[val] > max_vol)
-		return -EINVAL;
+		printk("WARNING:this voltage is not support!voltage set is %d mv\n",vol_map[val]);
 	ret = act8931_set_bits(act8931, act8931_BUCK_SET_VOL_REG(buck),
 	       	BUCK_VOL_MASK, val);
 	ret = act8931_set_bits(act8931, act8931_BUCK_SET_VOL_REG(buck) + 0x01,
-- 
2.35.3

