From 2c04c6058c6adec59fa023d1c6dd0ed70e0f3e83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 30 Jun 2011 11:53:11 +0800
Subject: [PATCH] rk: serial: remove printk from suspend/resume

---
 drivers/serial/rk_serial.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/serial/rk_serial.c b/drivers/serial/rk_serial.c
index d7fc1ed25126..bef5b8ffeeb2 100644
--- a/drivers/serial/rk_serial.c
+++ b/drivers/serial/rk_serial.c
@@ -1768,7 +1768,6 @@ static int serial_rk_suspend(struct platform_device *dev, pm_message_t state)
 
 	if (up)
 		uart_suspend_port(&serial_rk_reg, &up->port);
-	printk("%s:%s\n", __func__, dev->name);
 	return 0;
 }
 
@@ -1778,7 +1777,6 @@ static int serial_rk_resume(struct platform_device *dev)
 
 	if (up)
 		uart_resume_port(&serial_rk_reg, &up->port);
-	printk("%s:%s\n", __func__, dev->name);
 	return 0;
 }
 
-- 
2.35.3

