From 0a4e4997536597ba22820bcd5f39b5c0cba340c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Mar 2014 19:22:11 +0800
Subject: [PATCH] ARM: rockchip: rk3288 do disable address remap temporarily

---
 arch/arm/mach-rockchip/rk3288.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index 9496432d8f9d..1b4afaf981b3 100755
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -124,6 +124,9 @@ static void __init rk3288_dt_map_io(void)
 	/* rkpwm is used instead of old pwm */
 	writel_relaxed(0x00010001, RK_GRF_VIRT + RK3288_GRF_SOC_CON2);
 
+	/* disable address remap */
+	writel_relaxed(0x08000000, RK_SGRF_VIRT + RK3288_SGRF_SOC_CON0);
+
 	rk3288_boot_mode_init();
 }
 
-- 
2.35.3

