From f34e8d047dbd4d60483cad8da7a423ae3eb0bbab Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Wed, 4 Sep 2013 09:20:38 +0800
Subject: [PATCH] sensors:update sensor driver version

---
 drivers/input/sensors/sensor-dev.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/input/sensors/sensor-dev.c b/drivers/input/sensors/sensor-dev.c
index 037f9e9ee5e0..81852bde8377 100755
--- a/drivers/input/sensors/sensor-dev.c
+++ b/drivers/input/sensors/sensor-dev.c
@@ -40,9 +40,10 @@
 sensor-dev.c v1.1 add pressure and temperature support 2013-2-27
 sensor-dev.c v1.2 add akm8963 support 2013-3-10
 sensor-dev.c v1.3 add sensor debug support 2013-3-15
+sensor-dev.c v1.4 add angle calculation support between two gsensors 2013-09-01
 */
 
-#define SENSOR_VERSION_AND_TIME  "sensor-dev.c v1.3 add sensor debug support 2013-3-15"
+#define SENSOR_VERSION_AND_TIME  "sensor-dev.c v1.4 add angle calculation support between two gsensors 2013-09-01"
 
 
 struct sensor_private_data *g_sensor[SENSOR_NUM_TYPES];
-- 
2.35.3

