From 58786046a31e05d24f78d371516b4eb5642d5d31 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Sun, 7 Apr 2013 14:29:52 +0800
Subject: [PATCH] rk30 lcdc:set parameters for rk610 if needed

---
 drivers/video/rockchip/lcdc/rk30_lcdc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk30_lcdc.c b/drivers/video/rockchip/lcdc/rk30_lcdc.c
index f9c4ce8f3bc3..1d594077e484 100644
--- a/drivers/video/rockchip/lcdc/rk30_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk30_lcdc.c
@@ -280,6 +280,10 @@ static int rk30_load_screen(struct rk_lcdc_device_driver *dev_drv, bool initscre
     	{
     		screen->init();
     	}
+	if(screen->sscreen_set)
+        {
+                screen->sscreen_set(screen,!initscreen);
+        }
 	
 	printk("%s for lcdc%d ok!\n",__func__,lcdc_dev->id);
 	return 0;
-- 
2.35.3

