From e49620dc6d4b13db2c1822808b2cd51238342e8f Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 29 Aug 2012 14:39:53 +0800
Subject: [PATCH] rkfb: stanby the screen ,then power off

---
 drivers/video/rockchip/rk_fb.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e11b731aa26f..f25d6a115422 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1131,11 +1131,10 @@ static void rkfb_early_suspend(struct early_suspend *h)
 	{
 		if (!inf->lcdc_dev_drv[i])
 			continue;
-		if(inf->lcdc_dev_drv[i]->screen_ctr_info->io_disable)
-			inf->lcdc_dev_drv[i]->screen_ctr_info->io_disable();
 		if(inf->lcdc_dev_drv[i]->screen0->standby)
 			inf->lcdc_dev_drv[i]->screen0->standby(1);
-		
+		if(inf->lcdc_dev_drv[i]->screen_ctr_info->io_disable)
+			inf->lcdc_dev_drv[i]->screen_ctr_info->io_disable();
 		inf->lcdc_dev_drv[i]->suspend(inf->lcdc_dev_drv[i]);
 	}
 }
-- 
2.35.3

