From 76631fc74e3f61cd7c1a33d4a106dcbb92808d18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 7 Feb 2012 18:47:47 +0800
Subject: [PATCH] rk30: clock: add clk_get_rate

---
 arch/arm/mach-rk30/clock.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-rk30/clock.c b/arch/arm/mach-rk30/clock.c
index 1017eaf3965c..6510885f295d 100644
--- a/arch/arm/mach-rk30/clock.c
+++ b/arch/arm/mach-rk30/clock.c
@@ -20,3 +20,9 @@ void clk_disable(struct clk *clk)
 		return;
 }
 EXPORT_SYMBOL(clk_disable);
+
+unsigned long clk_get_rate(struct clk *clk)
+{
+	return 24000000;
+}
+EXPORT_SYMBOL(clk_get_rate);
-- 
2.35.3

