From ab9534f4be16152ace08e9f469f13722c4693bfa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 29 Jul 2011 09:50:43 +0800
Subject: [PATCH] rtc: HYM8563: include slab.h, prepare for 2.6.36

---
 drivers/rtc/rtc-HYM8563.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/rtc/rtc-HYM8563.c b/drivers/rtc/rtc-HYM8563.c
index 9c6f9b72f68c..341f20d550d9 100644
--- a/drivers/rtc/rtc-HYM8563.c
+++ b/drivers/rtc/rtc-HYM8563.c
@@ -21,6 +21,7 @@
 #include <linux/rtc.h>
 #include <linux/delay.h>
 #include <linux/wakelock.h>
+#include <linux/slab.h>
 #include <mach/gpio.h>
 #include <mach/iomux.h>
 #include "rtc-HYM8563.h"
-- 
2.35.3

