From 707c7ce8cbf16ebf15fe9a6c688d69329edf1fa1 Mon Sep 17 00:00:00 2001
From: typ <typ@rock-chips.com>
Date: Mon, 16 Sep 2013 16:18:51 +0800
Subject: [PATCH] RK3026 DDchange DDR DTT Check mode

---
 arch/arm/mach-rk2928/ddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/ddr.c b/arch/arm/mach-rk2928/ddr.c
index 25418c10cb06..929e051e301b 100755
--- a/arch/arm/mach-rk2928/ddr.c
+++ b/arch/arm/mach-rk2928/ddr.c
@@ -2282,9 +2282,9 @@ uint32_t __sramfunc ddr_change_freq_sram(uint32_t nMHz, struct ddr_freq_t ddr_fr
     deidle_port();
 #endif
 
-    ddr_dtt_check();
     /** 5. Issues a Mode Exit command   */
     DDR_RESTORE_SP(save_sp);
+    ddr_dtt_check();
     local_fiq_enable();
     local_irq_restore(flags);
 //    clk_set_rate(clk_get(NULL, "ddr_pll"), 0);    
-- 
2.35.3

