From fef082090f9a1f80661c58177484610a1ab34ec3 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Mon, 8 Jun 2015 17:52:06 +0800
Subject: [PATCH] mmc: core: limit tuning block timeout

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/core/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 21233c158987..33bdf60c8096 100755
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -474,6 +474,9 @@ static void mmc_get_req_timeout(struct mmc_request *mrq, u32 *timeout)
 	if ((mrq->cmd->opcode == SD_IO_RW_DIRECT) ||
 	    (mrq->cmd->opcode == SD_IO_RW_EXTENDED))
 		*timeout = 8000;
+	else if ((mrq->cmd->opcode == MMC_SEND_TUNING_BLOCK_HS200) ||
+		 (mrq->cmd->opcode == MMC_SEND_TUNING_BLOCK))
+		*timeout = 100;
 }
 
 static void mmc_wait_for_req_done(struct mmc_host *host,
-- 
2.35.3

