From 68a53d4656fdbeef5bbf07bd1adf740a8bca85ea Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 May 2015 17:50:33 +0800
Subject: [PATCH] spi: rockchip: todos spi-rockchip-test.c

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/spi/spi-rockchip-test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-rockchip-test.c b/drivers/spi/spi-rockchip-test.c
index 0dfca6766764..a0ca22868397 100755
--- a/drivers/spi/spi-rockchip-test.c
+++ b/drivers/spi/spi-rockchip-test.c
@@ -177,7 +177,7 @@ static int rockchip_spi_test_probe(struct spi_device *spi)
 	spi_test_data = (struct spi_test_data *)kzalloc(sizeof(struct spi_test_data), GFP_KERNEL);
 	if(!spi_test_data){
 		dev_err(&spi->dev, "ERR: no memory for spi_test_data\n");
-		return -ENOMEM;
+		return -ENOMEM;
 	}
 
 	spi->bits_per_word = 8;	
-- 
2.35.3

