From 62c67dcbaf457c50663a690231b07b56e805d008 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Mon, 4 Aug 2014 02:56:07 -0700
Subject: [PATCH] rk312x: reserve memory for ION, call rockchip_ion_reserve()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 陈亮 <cl@rock-chips.com>
---
 arch/arm/mach-rockchip/rk312x.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk312x.c b/arch/arm/mach-rockchip/rk312x.c
index 328053282e45..e7e8c5482afe 100755
--- a/arch/arm/mach-rockchip/rk312x.c
+++ b/arch/arm/mach-rockchip/rk312x.c
@@ -181,6 +181,8 @@ static void __init rk312x_dt_init_timer(void)
 
 static void __init rk312x_reserve(void)
 {
+	/* reserve memory for ION */
+	rockchip_ion_reserve();
 }
 
 static void __init rk312x_init_late(void)
-- 
2.35.3

