From a7302dbbdc121bd6d61f809585d6e337bcf9cd22 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Thu, 3 May 2012 10:16:01 +0800
Subject: [PATCH] add reset_pin in struct rk29lcd_info

---
 include/linux/rk_screen.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index efa47e53cde0..ef404c609752 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -85,6 +85,7 @@ struct rk29lcd_info {
 	u32 txd_pin;
 	u32 clk_pin;
 	u32 cs_pin;
+	u32	reset_pin;
 	int (*io_init)(void);
 	int (*io_deinit)(void);
 };
-- 
2.35.3

