From e8ca677ea2819455d6c52ca64893356bbd0d66c1 Mon Sep 17 00:00:00 2001
From: hcy <hcy@rock-chips.com>
Date: Fri, 28 Sep 2012 16:41:23 +0800
Subject: [PATCH] update DPLL rate after change DDR frequency

---
 arch/arm/mach-rk2928/ddr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/ddr.c b/arch/arm/mach-rk2928/ddr.c
index 9acb76b5c726..05dbd8f7ce13 100755
--- a/arch/arm/mach-rk2928/ddr.c
+++ b/arch/arm/mach-rk2928/ddr.c
@@ -2242,6 +2242,7 @@ int ddr_init(uint32_t dram_speed_bin, uint32_t freq)
                                                                     (ddr_get_cap()>>20));
     ddr_adjust_config(mem_type);
     value=ddr_change_freq(freq);
+    clk_set_rate(clk_get(NULL, "ddr_pll"), 0);
     ddr_print("init success!!! freq=%dMHz\n", value);
 
     calStatusLeft = pPHY_Reg->PHY_REG60;
-- 
2.35.3

