From e0476af1e75769c4f7b33eda0d7e1335bf83c107 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 1 Jul 2011 18:46:56 +0800
Subject: [PATCH] rk29: reset: support reboot to recovery

---
 arch/arm/mach-rk29/reset.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/reset.c b/arch/arm/mach-rk29/reset.c
index d586d474ab62..93e9a52861d2 100755
--- a/arch/arm/mach-rk29/reset.c
+++ b/arch/arm/mach-rk29/reset.c
@@ -118,8 +118,10 @@ void rk29_arch_reset(int mode, const char *cmd)
 	void (*rb2)(void);
 
 	if (cmd) {
-		if (!strcmp(cmd, "loader"))
+		if (!strcmp(cmd, "loader") || !strcmp(cmd, "bootloader"))
 			reboot_reason = 0x1888AAFF;
+		else if (!strcmp(cmd, "recovery"))
+			reboot_reason = 0x5242C303;
 	}
 
 	rb2 = (void(*)(void))((u32)rk29_rb_with_softreset - SRAM_CODE_OFFSET + 0x10130000);
-- 
2.35.3

