From 2a3379e0a60dec96c904e788b15c599cc4027d3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 11 Apr 2012 17:32:47 +0800
Subject: [PATCH] pack-kernel-rk30.sh: remove rkwifi config

---
 pack-kernel-rk30.sh | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/pack-kernel-rk30.sh b/pack-kernel-rk30.sh
index 736e46304ea3..4bbf858dc977 100755
--- a/pack-kernel-rk30.sh
+++ b/pack-kernel-rk30.sh
@@ -7,9 +7,6 @@ arch/arm/mach-rk30/clock_data.c
 arch/arm/mach-rk30/ddr*.c
 
 arch/arm/plat-rk/vpu*.c
-
-# exclude drivers/net/wireless/rkwifi/fw_path.c
-drivers/net/wireless/rkwifi/[abcdeghijklmnopqrstuvwxyz]*.c
 )
 
 DIRS=(
@@ -28,9 +25,6 @@ arch/arm/configs/rk29*
 drivers/*rk28*.c
 include/*rk28*
 
-drivers/net/wireless/rkwifi/*.h
-drivers/net/wireless/rkwifi/include
-
 drivers/staging/rk29/vivante
 
 drivers/staging/rk29/ipp/rk29-ipp.c
-- 
2.35.3

