From d976e7cdcc9806c7f98f09e3ad7d4f025dd67c08 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Tue, 6 Aug 2013 14:53:50 +0800
Subject: [PATCH] wdt: use two clock name that is wdt and pclk_wdt for all soc

---
 drivers/watchdog/rk29_wdt.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/watchdog/rk29_wdt.c b/drivers/watchdog/rk29_wdt.c
index 1f450a5e4161..ff9c51690ddc 100644
--- a/drivers/watchdog/rk29_wdt.c
+++ b/drivers/watchdog/rk29_wdt.c
@@ -374,6 +374,9 @@ static int __devinit rk29_wdt_probe(struct platform_device *pdev)
 #endif
 
 	wdt_clock = clk_get(&pdev->dev, "wdt");
+	if (IS_ERR(wdt_clock)) {
+		wdt_clock = clk_get(&pdev->dev, "pclk_wdt");
+	}
 	if (IS_ERR(wdt_clock)) {
 		dev_err(dev, "failed to find watchdog clock source\n");
 		ret = PTR_ERR(wdt_clock);
-- 
2.35.3

