From e80a72573af25a68cc39ee8f475b4915abade048 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 8 Jun 2012 16:10:30 +0800
Subject: [PATCH] rk: RK_SRAM_DMA only available on ARCH_RK30

---
 arch/arm/plat-rk/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 04f708017a24..4313698fc5b1 100644
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -88,7 +88,7 @@ config DDR_SDRAM_FREQ
 
 config DDR_FREQ
 	bool "Enable DDR frequency scaling"
-	select RK_SRAM_DMA
+	select RK_SRAM_DMA if ARCH_RK30
 
 config DDR_TEST
 	bool "DDR Test"
-- 
2.35.3

