From c7a230d5968c38813ad97f2301609a49f9eb0e3d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Mon, 5 Dec 2011 18:09:40 +0800
Subject: [PATCH] z5: add automatic I2S1/I2S0 for wm8994

---
 sound/soc/rk29/rk29_wm8994.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/sound/soc/rk29/rk29_wm8994.c b/sound/soc/rk29/rk29_wm8994.c
index 821ee7d19069..ab1a718f6612 100755
--- a/sound/soc/rk29/rk29_wm8994.c
+++ b/sound/soc/rk29/rk29_wm8994.c
@@ -271,7 +271,11 @@ static struct snd_soc_dai_link rk29_dai[] = {
 		.stream_name = "WM8994 PCM",
 		.codec_name = "wm8994-codec",
 		.platform_name = "rockchip-audio",
-        	.cpu_dai_name = "rk29_i2s.0",
+#ifdef CONFIG_SND_RK29_SOC_I2S_2CH		
+        	.cpu_dai_name = "rk29_i2s.1",
+#else if  CONFIG_SND_RK29_SOC_I2S_8CH
+		.cpu_dai_name = "rk29_i2s.0",
+#endif
 		.codec_dai_name = "wm8994-aif1",
 		.ops = &rk29_aif1_ops,
 	},
@@ -280,7 +284,11 @@ static struct snd_soc_dai_link rk29_dai[] = {
 		.stream_name = "WM8994 PCM",
 		.codec_name = "wm8994-codec",
 		.platform_name = "rockchip-audio",
-        	.cpu_dai_name = "rk29_i2s.0",
+#ifdef CONFIG_SND_RK29_SOC_I2S_2CH		
+        	.cpu_dai_name = "rk29_i2s.1",
+#else if  CONFIG_SND_RK29_SOC_I2S_8CH
+		.cpu_dai_name = "rk29_i2s.0",
+#endif
 		.codec_dai_name = "wm8994-aif2",
 		.ops = &rk29_aif2_ops,
 	},
@@ -289,7 +297,11 @@ static struct snd_soc_dai_link rk29_dai[] = {
 		.stream_name = "WM8994 PCM",
 		.codec_name = "wm8994-codec",
 		.platform_name = "rockchip-audio",
-        	.cpu_dai_name = "rk29_i2s.0",
+#ifdef CONFIG_SND_RK29_SOC_I2S_2CH		
+        	.cpu_dai_name = "rk29_i2s.1",
+#else if  CONFIG_SND_RK29_SOC_I2S_8CH
+		.cpu_dai_name = "rk29_i2s.0",
+#endif
 		.codec_dai_name = "wm8994-aif3",
 		.ops = &rk29_aif3_ops,
 	},
-- 
2.35.3

