From f0db0fef3bf5caac542281f13c1d449d9b0f975d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Thu, 31 Jul 2014 06:06:33 -0700
Subject: [PATCH] rk312x: set clk_ignore_unused = true
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 陈亮 <cl@rock-chips.com>
---
 drivers/clk/clk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 251c10b63c41..f18956ec5394 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -498,7 +498,7 @@ static void clk_disable_unused_subtree(struct clk *clk)
 	return;
 }
 
-static bool clk_ignore_unused;
+static bool clk_ignore_unused = true;
 static int __init clk_ignore_unused_setup(char *__unused)
 {
 	clk_ignore_unused = true;
-- 
2.35.3

