From ca6be0fe2207cb125db911d9823ee75897a81a86 Mon Sep 17 00:00:00 2001
From: lw <lw@rock-chips.com>
Date: Tue, 21 Aug 2012 11:51:35 +0800
Subject: [PATCH] phonepad:add touch screen Kconfig again

---
 drivers/input/touchscreen/Kconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/touchscreen/Kconfig b/drivers/input/touchscreen/Kconfig
index 50d915718bcc..c4b92a3d671d 100755
--- a/drivers/input/touchscreen/Kconfig
+++ b/drivers/input/touchscreen/Kconfig
@@ -969,6 +969,10 @@ config LAIBAO_TS
 config TOUCHSCREEN_GT801_IIC
 	tristate "GT801_IIC based touchscreens"
 	depends on I2C2_RK29
+	
+config TOUCHSCREEN_GT82X_IIC
+	tristate "GT82x_IIC based touchscreens"
+	depends on I2C2_RK30
 
 config TOUCHSCREEN_GT818_IIC
 	tristate "GT818_IIC based touchscreens"
-- 
2.35.3

