From 1184b8c22aa624681f55b0165a93ae8aca7faad6 Mon Sep 17 00:00:00 2001
From: wuhao <wuhao@wuhao@rock-chips.com>
Date: Thu, 6 Jun 2013 10:16:58 +0800
Subject: [PATCH] rk616 codec: playback do not power up while hdmi is in

---
 sound/soc/codecs/rk616_codec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/codecs/rk616_codec.c b/sound/soc/codecs/rk616_codec.c
index 78a93f3c5611..4d1617e782f9 100755
--- a/sound/soc/codecs/rk616_codec.c
+++ b/sound/soc/codecs/rk616_codec.c
@@ -1727,6 +1727,7 @@ static int rk616_codec_power_up(int type)
 		type == RK616_CODEC_CAPTURE ? "capture" : "");
 
 	if (type == RK616_CODEC_PALYBACK) {
+		if(! get_hdmi_state())
 		for (i = 0; i < RK616_CODEC_PALYBACK_POWER_UP_LIST_LEN; i++) {
 			snd_soc_write(codec, palyback_power_up_list[i].reg,
 				palyback_power_up_list[i].value);
-- 
2.35.3

