From 182c55e79b95d34334e950b843553f382e86aae0 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 14 Mar 2013 15:30:29 +0800
Subject: [PATCH] lcdc Kconfig:RK30 LCDC depend on SOC_RK3066

---
 drivers/video/rockchip/lcdc/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/lcdc/Kconfig b/drivers/video/rockchip/lcdc/Kconfig
index a714142eabde..44d7182b6080 100644
--- a/drivers/video/rockchip/lcdc/Kconfig
+++ b/drivers/video/rockchip/lcdc/Kconfig
@@ -6,7 +6,7 @@ config LCDC_RK2928
 
 config LCDC_RK30
         tristate "rk30 lcdc support"
-        depends on FB_ROCKCHIP && ARCH_RK30
+        depends on FB_ROCKCHIP && SOC_RK3066
         help
           Driver for rk30 lcdc .There are two lcd controllers on rk30
 
-- 
2.35.3

