From 01e8d2e7cbb8d72e2f8249e354e7345aa0392997 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 20 Mar 2014 21:33:42 +0800
Subject: [PATCH] ARM: rockchip: fix debug/rockchip.S out of vmalloc space bug

---
 arch/arm/include/debug/rockchip.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/debug/rockchip.S b/arch/arm/include/debug/rockchip.S
index 91a662f1d49b..ebb6e2a8a31e 100644
--- a/arch/arm/include/debug/rockchip.S
+++ b/arch/arm/include/debug/rockchip.S
@@ -21,7 +21,7 @@
 		cmp	\tmp, \rv
 		ldreq	\rp, = RK3188_UART2_PHYS
 		ldrne	\rp, = RK3288_UART_DBG_PHYS
-		ldr	\rv, = 0xff000000
+		ldr	\rv, = 0xfec00000
 		mov	\tmp, \rp
 		bic	\tmp, \tmp, #0xff000000
 		bic	\tmp, \tmp, #0x00f00000
-- 
2.35.3

