From 48d8ad46cf0339f15f3c0ba767a0f4a549fb5abd Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Thu, 21 Apr 2011 19:11:04 -0700
Subject: [PATCH] backlight: update bl_brightness

---
 drivers/video/backlight/wm831x_bl.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/video/backlight/wm831x_bl.c b/drivers/video/backlight/wm831x_bl.c
index d0c1d6f38461..0f52691b04fa 100755
--- a/drivers/video/backlight/wm831x_bl.c
+++ b/drivers/video/backlight/wm831x_bl.c
@@ -24,6 +24,7 @@
 #include <linux/delay.h>
 #include <linux/ktime.h>
 #define BL_SET   255
+#define BL_MISC_VALUE 20
 #define BL_INIT_VALUE 102
 struct wm831x_backlight_data {
 	struct wm831x *wm831x;
@@ -117,8 +118,11 @@ static int wm831x_backlight_set(struct backlight_device *bl, int brightness)
 static int wm831x_backlight_update_status(struct backlight_device *bl)
 {
 	int brightness = bl->props.brightness;
-	if (brightness<=BL_INIT_VALUE) {
-		brightness = 111*brightness/51;
+	if (brightness<=BL_MISC_VALUE) {
+		brightness = 8*brightness;
+	}
+	else if (brightness<=BL_INIT_VALUE) {
+		brightness = 31*brightness/41 + 145;
 	}
 	else {
 		brightness = 33*brightness/153 + 200;
-- 
2.35.3

