From aa71396e5fea620e583e3ed9b59bb9978e73f347 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=83=AD=E6=AF=85?= <gy@rock-chips.com>
Date: Mon, 24 Mar 2014 16:33:31 +0800
Subject: [PATCH] tp: ct36x touchscreen compatible sdk and 977

---
 drivers/input/touchscreen/ct36x/ct363.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/ct36x/ct363.c b/drivers/input/touchscreen/ct36x/ct363.c
index 0a29b737b939..b2e293d814dc 100755
--- a/drivers/input/touchscreen/ct36x/ct363.c
+++ b/drivers/input/touchscreen/ct36x/ct363.c
@@ -204,7 +204,9 @@ static void ct363_report(struct ct36x_data *ts)
                    if( (ct363->x > ts->x_max) || (ct363->y > ts->y_max) || (ct363->x < 0) || (ct363->y < 0) ){
                           continue ;
                     }
-            ct363->y = ts->y_max - ct363->y;       
+                   if(flag_ct36x_model==363){
+			ct363->y = ts->y_max - ct363->y;      //add for p977 
+		   }       
 			input_mt_slot(ts->input, ct363->pts[i].id - 1);
 			input_mt_report_slot_state(ts->input, MT_TOOL_FINGER, true);
 			input_report_abs(ts->input, ABS_MT_TOUCH_MAJOR, 1);
-- 
2.35.3

