From e88c7b949d9d7dfcebe3cdbb215f0d1f9b2d8f09 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 20 Jan 2014 18:49:30 +0800
Subject: [PATCH] ARM: rockchip: rk3188 use timer-based delay

---
 arch/arm/mach-rockchip/rk3188.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3188.c b/arch/arm/mach-rockchip/rk3188.c
index 23b5db53b980..edffe72909e9 100644
--- a/arch/arm/mach-rockchip/rk3188.c
+++ b/arch/arm/mach-rockchip/rk3188.c
@@ -108,7 +108,6 @@ static void __init rk3188_boot_mode_init(void)
 
 static void __init rk3188_dt_map_io(void)
 {
-	preset_lpj = 11996091ULL / 2;
 	iotable_init(rk3188_io_desc, ARRAY_SIZE(rk3188_io_desc));
 	debug_ll_io_init();
 
-- 
2.35.3

