From 467d586ba77d82edf7ffe84b2e2f0c44e9bda425 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 12 Apr 2012 17:04:04 +0800
Subject: [PATCH] rk30: l2: setup prefetch ctrl

---
 arch/arm/mach-rk30/common.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 2e03a11e2a22..b78ea3195bae 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -57,6 +57,9 @@ static void __init rk30_l2_cache_init(void)
 				|L2_LY_SET(6,L2_LY_RD_OFF)
 				|L2_LY_SET(1,L2_LY_WR_OFF), RK30_L2C_BASE + L2X0_DATA_LATENCY_CTRL);
 
+	/* L2X0 Prefetch Control */
+	writel_relaxed(0x70000003, RK30_L2C_BASE + L2X0_PREFETCH_CTRL);
+
 	/* L2X0 Power Control */
 	writel_relaxed(L2X0_DYNAMIC_CLK_GATING_EN | L2X0_STNDBY_MODE_EN, RK30_L2C_BASE + L2X0_POWER_CTRL);
 
-- 
2.35.3

