From 237c8cfdc9d2064dd51599814e25d35817c2a5ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 18 Jul 2011 19:16:33 +0800
Subject: [PATCH] rk29: clock: fix aclk_lcdc from ddr pll

---
 arch/arm/mach-rk29/clock.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index f15715cdaecf..9fa3180e36c9 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2545,7 +2545,8 @@ static void __init rk29_clock_common_init(unsigned long ppll_rate, unsigned long
 	clk_set_rate_nolock(&codec_pll_clk, cpll_rate);
 	clk_set_parent_nolock(&clk_gpu, &codec_pll_clk);
 
-	clk_set_parent_nolock(&aclk_lcdc, cpll_rate > ppll_rate ? &codec_pll_clk : &general_pll_clk);
+	/* ddr pll */
+	clk_set_parent_nolock(&aclk_lcdc, &ddr_pll_clk);
 
 	/* arm pll */
 	clk_set_rate_nolock(&arm_pll_clk, armclk);
-- 
2.35.3

