From 808c5ddbae9cb7df9359a88d0ee68fd374279278 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Mon, 4 Aug 2014 09:14:23 +0800
Subject: [PATCH] spi:fix the bug while read one byte

---
 drivers/spi/spi-rockchip-core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-rockchip-core.c b/drivers/spi/spi-rockchip-core.c
index af4ac86d6d60..63aacfd2f8be 100755
--- a/drivers/spi/spi-rockchip-core.c
+++ b/drivers/spi/spi-rockchip-core.c
@@ -731,10 +731,10 @@ static void pump_transfers(unsigned long data)
 
         dw_writew(dws, SPIM_CTRLR1, dws->len-1);
 		
-		if (txint_level)
+		if (txint_level != dw_readl(dws, SPIM_TXFTLR))
 			dw_writew(dws, SPIM_TXFTLR, txint_level);
 			
-		if (rxint_level)
+		if (rxint_level != dw_readl(dws, SPIM_RXFTLR))
 		{
 			dw_writew(dws, SPIM_RXFTLR, rxint_level);
 			DBG_SPI("%s:rxint_level=%d\n",__func__,rxint_level);
-- 
2.35.3

