From 10c9f5b470c582e009cf134c058d0c64412eab1c Mon Sep 17 00:00:00 2001
From: Cody Xie <Cody.Xie@rock-chips.com>
Date: Mon, 21 Apr 2014 09:39:53 +0800
Subject: [PATCH] rk3288-p977 : cpu max freq 1.8G

---
 arch/arm/boot/dts/rk3288-p977.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index 1e8871522aa8..6d1d9db785da 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -535,6 +535,10 @@
 		1008000 1000000
 		1200000 1050000
 		1416000 1150000
+		1512000 1200000
+		1608000 1250000
+		1704000 1300000
+		1800000 1350000
 		>;
 	status="okay";
 };
-- 
2.35.3

