From 125ecc7284b3c041386637ed2cd0245a3900132b Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 16 Jun 2015 18:02:49 +0800
Subject: [PATCH] rk fb: vsync thread distinction two vop

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index a9ee1f0c3853..acee4b0924da 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3307,7 +3307,7 @@ static int rk_fb_wait_for_vsync_thread(void *data)
 {
 	struct rk_lcdc_driver *dev_drv = data;
 	struct rk_fb *rk_fb = platform_get_drvdata(fb_pdev);
-	struct fb_info *fbi = rk_fb->fb[0];
+	struct fb_info *fbi = rk_fb->fb[dev_drv->fb_index_base];
 
 	while (!kthread_should_stop()) {
 		ktime_t timestamp = dev_drv->vsync_info.timestamp;
-- 
2.35.3

