From b3f163f9b122a1820ac269f293a51e0164d01749 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 5 May 2015 17:56:57 +0800
Subject: [PATCH] display-sys: list supported 3d format of all resolution.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/display-sys.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/display-sys.c b/drivers/video/rockchip/display-sys.c
index b85d5db7e680..4ee561ef60d5 100755
--- a/drivers/video/rockchip/display-sys.c
+++ b/drivers/video/rockchip/display-sys.c
@@ -239,6 +239,8 @@ static ssize_t display_show_3dmode(struct device *dev,
 	struct display_modelist *display_modelist;
 	struct fb_videomode mode;
 	int i = 0, cur_3d_mode = -1;
+	char mode_str[128];
+	int mode_strlen, format_3d;
 
 	if (dsp->ops && dsp->ops->getmodelist) {
 		if (dsp->ops->getmodelist(dsp, &modelist))
@@ -271,7 +273,19 @@ static ssize_t display_show_3dmode(struct device *dev,
 
 	if (dsp->ops && dsp->ops->get3dmode)
 		cur_3d_mode = dsp->ops->get3dmode(dsp);
-	i += snprintf(buf + i, PAGE_SIZE - i, "cur3dmode=%d", cur_3d_mode);
+	i += snprintf(buf + i, PAGE_SIZE - i, "cur3dmode=%d\n", cur_3d_mode);
+
+	list_for_each(pos, modelist) {
+		display_modelist = list_entry(pos,
+					      struct display_modelist,
+					      list);
+		mode_strlen = mode_string(mode_str, 0,
+					  &(display_modelist->mode));
+		mode_str[mode_strlen-1] = 0;
+		format_3d = display_modelist->format_3d;
+		i += snprintf(buf+i, PAGE_SIZE, "%s,%d\n",
+			      mode_str, format_3d);
+	}
 	return i;
 }
 
-- 
2.35.3

