From a4904893b201e1338a252f924d25cb269c693f3b Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 4 Jul 2014 14:32:27 +0800
Subject: [PATCH] 3036: fix iomap error

---
 arch/arm/mach-rockchip/rk3036.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index bba0701c6379..9ebc09de083f 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -53,7 +53,6 @@
 
 
 static struct map_desc rk3036_io_desc[] __initdata = {
-	RK3036_DEVICE(TIMER),
 	RK3036_DEVICE(CRU),
 	RK3036_DEVICE(GRF),
 	RK3036_DEVICE(ROM),
@@ -66,7 +65,6 @@ static struct map_desc rk3036_io_desc[] __initdata = {
 	RK_DEVICE(RK_DEBUG_UART_VIRT, RK3036_UART2_PHYS, RK3036_UART_SIZE),
 	RK_DEVICE(RK_GIC_VIRT, RK3036_GIC_DIST_PHYS, RK3036_GIC_DIST_SIZE),
 	RK_DEVICE(RK_GIC_VIRT + RK3036_GIC_DIST_SIZE,RK3036_GIC_CPU_PHYS, RK3036_GIC_CPU_SIZE),
-	RK_DEVICE(RK_BOOTRAM_VIRT, RK3036_IMEM_PHYS, RK3036_IMEM_SIZE),
 	RK_DEVICE(RK3036_IMEM_VIRT, RK3036_IMEM_PHYS, SZ_4K),
 	RK_DEVICE(RK_TIMER_VIRT, RK3036_TIMER_PHYS, RK3036_TIMER_SIZE),
 };
-- 
2.35.3

