From ae87f9de3ad2690bd46af1bd49e45afeee1c4c9c Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Wed, 11 Feb 2015 14:02:37 +0800
Subject: [PATCH] rk fb: enable mmu when uboot-logo is 0

---
 drivers/video/rockchip/rk_fb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 30d78948f70b..5c86a71a0cdc 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3747,6 +3747,10 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 				freed_index = 0;
 			}
 			return 0;
+		} else {
+			if (dev_drv->ops->mmu_en)
+				dev_drv->ops->mmu_en(dev_drv);
+			freed_index = 0;
 		}
 #if defined(CONFIG_LOGO)
 		main_fbi->fbops->fb_set_par(main_fbi);
-- 
2.35.3

