From b93c87153a60b24e13873ada44464348922ef980 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Thu, 11 Dec 2014 12:44:50 +0800
Subject: [PATCH] rk: arm64: support build kernel.img and resource.img

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/Makefile | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index c5090fed0679..67a6d3dead4a 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -95,3 +95,12 @@ define archhelp
   echo  '                  (distribution) /sbin/installkernel or'
   echo  '                  install to $$(INSTALL_PATH) and run lilo'
 endef
+
+kernel.img: Image
+	$(Q)$(srctree)/mkkrnlimg $(objtree)/arch/arm64/boot/Image $(objtree)/kernel.img >/dev/null
+	@echo '  Image:  kernel.img is ready'
+
+LOGO := $(notdir $(wildcard $(srctree)/logo.bmp))
+%.img: %.dtb kernel.img $(LOGO)
+	$(Q)$(srctree)/resource_tool $(objtree)/arch/arm64/boot/dts/$*.dtb $(LOGO)
+	@echo '  Image:  resource.img (with $*.dtb $(LOGO)) is ready'
-- 
2.35.3

