From 6dd0c49a3eb5aaabf4db6d0d47db36c1ee872a7e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 7 Nov 2011 11:45:09 +0800
Subject: [PATCH] usb: serial: option: fix warning: "CINTERION_VENDOR_ID"
 redefined

---
 drivers/usb/serial/option.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 8cf88aa5c8f4..3f50b0fef1c3 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -400,8 +400,6 @@ static void option_instat_callback(struct urb *urb);
 #define THINKWILL_PRODUCT_ID	0x9909
 #define	THINKWILL_MI900_PRODUCT_ID	0x9013
 
-#define CINTERION_VENDOR_ID			0x0681
-
 /* Olivetti products */
 #define OLIVETTI_VENDOR_ID			0x0b3c
 #define OLIVETTI_PRODUCT_OLICARD100		0xc000
@@ -1042,7 +1040,6 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x5700) },
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x6341) },
 
-	{ USB_DEVICE(CINTERION_VENDOR_ID, 0x0047) },
 	{ USB_DEVICE(OLIVETTI_VENDOR_ID, OLIVETTI_PRODUCT_OLICARD100) },
 	{ USB_DEVICE(CELOT_VENDOR_ID, CELOT_PRODUCT_CT680M) }, /* CT-650 CDMA 450 1xEVDO modem */
 	{ USB_DEVICE(ONDA_VENDOR_ID, ONDA_MT825UP) }, /* ONDA MT825UP modem */
-- 
2.35.3

