From 92b2ba2625186cee9dbd4a379075748f4217dfb0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Thu, 13 Jun 2013 14:35:26 +0800
Subject: [PATCH] i2s:Set the default drive current 8ma

---
 sound/soc/rk29/rk30_i2s.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index 0e66efffff72..0f8f219b8a4d 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -557,16 +557,15 @@ static int __devinit rockchip_i2s_probe(struct platform_device *pdev)
 	struct rk29_i2s_info *i2s;
 	struct snd_soc_dai_driver *dai;
 	int    ret;
-	int v;
 
 #if defined(CONFIG_SND_I2S_USE_18V)	
-	v = 0x2000200;//bit9: 1,1.8v;0,3.3v
-	writel_relaxed(v,RK30_GRF_BASE + GRF_IO_CON4);
+	writel_relaxed(0x2000200,RK30_GRF_BASE + GRF_IO_CON4);//bit9: 1,1.8v;0,3.3v
 #elif defined(CONFIG_SND_I2S_USE_33V)
-	v = 0x2000000;
-	writel_relaxed(v,RK30_GRF_BASE + GRF_IO_CON4);
+	writel_relaxed(0x2000000,RK30_GRF_BASE + GRF_IO_CON4);
 #endif
-
+	//default 8ma  0xF000F = 12ma 0xF0005=4ma 0xF0000=2ma
+	writel_relaxed(0xF000A,RK30_GRF_BASE + GRF_IO_CON1);
+	
 	I2S_DBG("Enter %s, %d pdev->id = %d >>>>>>>>>>>\n", __func__, __LINE__, pdev->id);
 	
 	if(pdev->id >= MAX_I2S) {
-- 
2.35.3

