From 6140ec6529898ddbf9889f165609f661896ca084 Mon Sep 17 00:00:00 2001
From: cl <cl@rock-chips.com>
Date: Tue, 11 Nov 2014 18:11:50 +0800
Subject: [PATCH] efuse: clear compile warning

Signed-off-by: cl <cl@rock-chips.com>
---
 arch/arm/mach-rockchip/efuse.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/efuse.c b/arch/arm/mach-rockchip/efuse.c
index 4a0da82cd76c..235e124e9be6 100644
--- a/arch/arm/mach-rockchip/efuse.c
+++ b/arch/arm/mach-rockchip/efuse.c
@@ -99,7 +99,6 @@ static void __init rk3288_set_system_serial(void)
 
 int rk312x_efuse_readregs(u32 addr, u32 length, u8 *buf)
 {
-	unsigned long flags;
 	int ret = length;
 
 	if (!length)
-- 
2.35.3

