From fa13b39aad1788af6170f882fd3087a5811fd988 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:25:20 +0800
Subject: [PATCH] vmac: fix for v2.6.36

---
 drivers/net/rk29_vmac.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/rk29_vmac.c b/drivers/net/rk29_vmac.c
index 5f3c6e6c4e25..ea6f723bb7a2 100755
--- a/drivers/net/rk29_vmac.c
+++ b/drivers/net/rk29_vmac.c
@@ -45,6 +45,7 @@
 #include <linux/slab.h>
 #include <linux/types.h>
 #include <linux/wakelock.h>
+#include <linux/version.h>
 
 #include <mach/iomux.h>
 #include <mach/gpio.h>
@@ -1275,7 +1276,7 @@ void vmac_tx_timeout(struct net_device *dev)
 static void create_multicast_filter(struct net_device *dev,
 	unsigned long *bitmask)
 {
-#if 1
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 34))
 	struct netdev_hw_addr *ha;
 	unsigned long crc;
 	char *addrs;
@@ -1323,7 +1324,7 @@ static void create_multicast_filter(struct net_device *dev,
 }
 static void vmac_set_multicast_list(struct net_device *dev)
 {
-#if 1
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 34))
 	struct vmac_priv *ap = netdev_priv(dev);
 	unsigned long flags, bitmask[2];
 	int promisc, reg;
-- 
2.35.3

