From 55c6e001c34a3d55ef6eab06917d5455f2271332 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Tue, 24 Jun 2014 10:51:07 +0800
Subject: [PATCH] rockchip:iommu:fix compiling warning when iommu enabled

---
 drivers/iommu/rockchip-iommu.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 07fd3bb4b045..fee03f2d9945 100755
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -17,6 +17,7 @@
 #include <linux/errno.h>
 #include <linux/memblock.h>
 #include <linux/export.h>
+#include <linux/module.h>
 
 #include <asm/cacheflush.h>
 #include <asm/pgtable.h>
@@ -1022,10 +1023,11 @@ static DEVICE_ATTR(dump_mmu_pgtable, 0644, NULL, dump_mmu_pagetbl);
 
 void dump_iommu_sysfs_init(void )
 {
+	u32 ret;
 	dump_mmu_object = kobject_create_and_add("rk_iommu", NULL);
 	if (dump_mmu_object == NULL)
 		return;
-	sysfs_create_file(dump_mmu_object, &dev_attr_dump_mmu_pgtable.attr);
+	ret = sysfs_create_file(dump_mmu_object, &dev_attr_dump_mmu_pgtable.attr);
 	return;
 }
 	
-- 
2.35.3

