From d00b3197688e89b12ec3b8f601f8bae8974854f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Jan 2013 11:30:13 +0800
Subject: [PATCH] rk_timer: support sched_clock

---
 arch/arm/plat-rk/rk_timer.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/plat-rk/rk_timer.c b/arch/arm/plat-rk/rk_timer.c
index 24dc1b508ce8..51e5e5934b8d 100644
--- a/arch/arm/plat-rk/rk_timer.c
+++ b/arch/arm/plat-rk/rk_timer.c
@@ -282,3 +282,18 @@ int local_timer_ack(void)
 	return 0;
 }
 #endif
+
+/*
+ * Scheduler clock - returns current time in nanosec units.
+ */
+unsigned long long notrace sched_clock(void)
+{
+	const struct clocksource *cs = &rk_timer_clocksource;
+	cycle_t cyc;
+
+	if (!timer.cs_base)
+		return 0;
+
+	cyc = ~rk_timer_read_current_value(timer.cs_base);
+	return clocksource_cyc2ns(cyc, cs->mult, cs->shift);
+}
-- 
2.35.3

