From 7ec39878a634bc6e0aacccfda2d90ec1ff21314f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 2 Dec 2013 10:43:47 +0800
Subject: [PATCH] ARM: PIE: pie.lds.S fix PIE_OVERLAY_END

---
 arch/arm/kernel/pie.lds.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/kernel/pie.lds.S b/arch/arm/kernel/pie.lds.S
index 7950758f36fb..25f2e9ee4c3f 100644
--- a/arch/arm/kernel/pie.lds.S
+++ b/arch/arm/kernel/pie.lds.S
@@ -25,7 +25,7 @@ SECTIONS
 	OVERLAY : NOCROSSREFS {
 		PIE_OVERLAY_SECTION(rockchip)
 	}
-	PIE_OVERLAY_SEND
+	PIE_OVERLAY_END
 
 	__pie_rel_dyn_start : {
 		VMLINUX_SYMBOL(__pie_rel_dyn_start) = .;
-- 
2.35.3

