From 646e27e0f57619acf2328d05f30638c247d4231f Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 24 Oct 2017 19:37:24 +0800
Subject: [PATCH] rk: Makefile: rm kernel.img and resource.img when make clean

Change-Id: I79d53d683cdff51c2077d77e049bc533c79b76f1
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index b3681de415da..f0536128ec8e 100644
--- a/Makefile
+++ b/Makefile
@@ -1468,6 +1468,7 @@ scripts: ;
 endif # KBUILD_EXTMOD
 
 clean: $(clean-dirs)
+	$(Q)rm -f kernel.img resource.img
 	$(call cmd,rmdirs)
 	$(call cmd,rmfiles)
 	@find $(if $(KBUILD_EXTMOD), $(KBUILD_EXTMOD), .) $(RCS_FIND_IGNORE) \
-- 
2.35.3

