From e73f53aacca0cb1bd6959718399443350e922f38 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 20 Mar 2014 20:54:59 +0800
Subject: [PATCH] ARM: rockchip: default ARCH_NR_GPIO 512

---
 arch/arm/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index fe8d330568e5..ddb8c28ab213 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1715,6 +1715,7 @@ config ARCH_NR_GPIO
 	int
 	default 1024 if ARCH_SHMOBILE || ARCH_TEGRA
 	default 512 if SOC_OMAP5
+	default 512 if ARCH_ROCKCHIP
 	default 392 if ARCH_U8500
 	default 352 if ARCH_VT8500
 	default 288 if ARCH_SUNXI
-- 
2.35.3

