From 217b72e6fdac509b7a7b92843e9a4764527ba161 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 26 Feb 2013 17:09:29 +0800
Subject: [PATCH] rk fb:add ioctl RK_FBIOPUT_COLOR_KEY_CFG for color key config

---
 include/linux/rk_fb.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/linux/rk_fb.h b/include/linux/rk_fb.h
index 1a5a2fe2244e..6ed2e8103735 100644
--- a/include/linux/rk_fb.h
+++ b/include/linux/rk_fb.h
@@ -58,7 +58,8 @@
 #define RK_FBIOGET_ENABLE		0x5020
 #define RK_FBIOSET_CONFIG_DONE		0x4628
 #define RK_FBIOSET_VSYNC_ENABLE		0x4629
-#define RK_FBIOPUT_NUM_BUFFERS 	0x4625
+#define RK_FBIOPUT_NUM_BUFFERS 		0x4625
+#define RK_FBIOPUT_COLOR_KEY_CFG	0x4626
 
 
 /********************************************************************
@@ -180,6 +181,12 @@ struct rk_fb_vsync {
 	struct task_struct	*thread;
 };
 
+struct color_key_cfg {
+	u32 win0_color_key_cfg;
+	u32 win1_color_key_cfg;
+	u32 win2_color_key_cfg;
+};
+
 typedef enum _TRSP_MODE
 {
     TRSP_CLOSE = 0,
@@ -224,6 +231,7 @@ struct layer_par {
 	u8 swap_rb;
 };
 
+
 struct rk_lcdc_device_driver{
 	char name[6];
 	int id;
-- 
2.35.3

