From d928cf097f4ef64d08f9ecd627a8a833009bcb8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Aug 2012 19:40:37 +0800
Subject: [PATCH] rk3066b: disable tsadc device register

---
 arch/arm/mach-rk30/devices.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index b528a9498b9a..f24f6e7e137a 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -51,7 +51,7 @@ struct platform_device device_adc = {
 };
 #endif
 
-#ifdef CONFIG_ARCH_RK30
+#if !defined(CONFIG_ARCH_RK3066B)
 static struct resource rk30_tsadc_resource[] = {
 	{
 		.start	= IRQ_TSADC,
@@ -1360,7 +1360,7 @@ static int __init rk30_init_devices(void)
 #ifdef CONFIG_ADC_RK30
 	platform_device_register(&device_adc);
 #endif
-#ifdef CONFIG_ARCH_RK30
+#if !defined(CONFIG_ARCH_RK3066B)
 	platform_device_register(&device_tsadc);
 #endif
 	rk30_init_sdmmc();
-- 
2.35.3

