From b7fae6b9a2659fdf7957ac8cca4609c0a1025eaa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 2 Aug 2010 10:50:28 +0800
Subject: [PATCH] Updating the .gitignore to ignore
 include/asm-arm/mach-types.h

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 28d040bd9366..072174fd1833 100644
--- a/.gitignore
+++ b/.gitignore
@@ -55,6 +55,7 @@ include/linux/version.h
 include/linux/utsrelease.h
 include/linux/bounds.h
 include/generated
+include/asm-arm/mach-types.h
 
 # stgit generated dirs
 patches-*
-- 
2.35.3

