From dc1e1bf064a7aeba41324d1bd79484188cfa183f Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 12 Sep 2014 09:03:55 +0800
Subject: [PATCH] rk312x dtsi: lvds status depend on dts file

---
 arch/arm/boot/dts/rk312x.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 3a6102bdac27..3d2d4c825917 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -699,6 +699,7 @@
 		reg = <0x20038000 0x4000>;
 		clocks = <&clk_gates5 0>;
 		clock-names = "pclk_lvds";
+		status = "disabled";
 	};
 
 	lcdc: lcdc@1010e000 {
-- 
2.35.3

