From e418a64850a21cc8300b5fe42b4d1ca9ce7eaeaf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 2 Apr 2011 15:31:46 +0800
Subject: [PATCH] wm831x-on: default use rk29 keypad driver to send wake up key

---
 drivers/input/misc/wm831x-on.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/misc/wm831x-on.c b/drivers/input/misc/wm831x-on.c
index 8705da4d30b9..3cc51bb26a15 100755
--- a/drivers/input/misc/wm831x-on.c
+++ b/drivers/input/misc/wm831x-on.c
@@ -46,6 +46,7 @@ struct wm831x_on {
 
 struct wm831x_on *g_wm831x_on;
 
+#ifndef CONFIG_KEYS_RK29
 void rk28_send_wakeup_key(void)
 {
         printk("%s\n", __FUNCTION__);
@@ -60,6 +61,7 @@ void rk28_send_wakeup_key(void)
         input_sync(g_wm831x_on->dev);
         printk("%s end\n", __FUNCTION__);
 }
+#endif
 
 #if 1
 
-- 
2.35.3

