From 031e71fa6404b32f22b5cbb76f3db1e3295bf422 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 4 Jun 2015 16:10:44 +0800
Subject: [PATCH] rk1000: shorter reset delay time.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/mfd/rk1000-core.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/mfd/rk1000-core.c b/drivers/mfd/rk1000-core.c
index 1aa3d2ce5b9e..36ad89fc6280 100755
--- a/drivers/mfd/rk1000-core.c
+++ b/drivers/mfd/rk1000-core.c
@@ -227,11 +227,9 @@ static int rk1000_probe(struct i2c_client *client,
 			goto err;
 		}
 		rk1000->io_reset.active = !(flags & OF_GPIO_ACTIVE_LOW);
-
-
 		gpio_direction_output(rk1000->io_reset.gpio,
 				      !(rk1000->io_reset.active));
-		msleep(20);
+		usleep_range(500, 1000);
 		/********Get power pin***********/
 		rk1000->io_power.gpio = of_get_named_gpio_flags(rk1000_np,
 								"gpio-power",
@@ -252,7 +250,7 @@ static int rk1000_probe(struct i2c_client *client,
 		/********rk1000 reset***********/
 		gpio_set_value(rk1000->io_reset.gpio,
 			       rk1000->io_reset.active);
-		msleep(100);
+		usleep_range(5000, 10000);
 		gpio_set_value(rk1000->io_reset.gpio,
 			       !(rk1000->io_reset.active));
 	}
-- 
2.35.3

