From 899a4f18b982ce5408ee862b7d51030bcf20b617 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 31 Aug 2012 19:30:32 +0800
Subject: [PATCH] rk3066b: init debug uart

---
 arch/arm/mach-rk30/common.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 21775fb0d53b..02ef0baf48a1 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -120,6 +120,16 @@ int __init clk_disable_unused(void);
 void __init rk30_map_io(void)
 {
 	rk30_map_common_io();
+#ifdef DEBUG_UART_BASE
+	writel_relaxed(0x000f0005, RK30_GRF_BASE + 0x74);
+	writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
+	writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
+	writel_relaxed(0x00, DEBUG_UART_BASE + 0x04);
+	writel_relaxed(0x83, DEBUG_UART_BASE + 0x0c);
+	writel_relaxed(0x0d, DEBUG_UART_BASE + 0x00);
+	writel_relaxed(0x00, DEBUG_UART_BASE + 0x04);
+	writel_relaxed(0x03, DEBUG_UART_BASE + 0x0c);
+#endif
 	rk29_setup_early_printk();
 	rk30_cpu_axi_init();
 	rk29_sram_init();
-- 
2.35.3

