From 098b2f4058ad57e30b0264bedbd8347d89b24ca6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 31 Aug 2012 20:41:10 +0800
Subject: [PATCH] rk3066b: disable ddr_init

---
 arch/arm/mach-rk30/common.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 02ef0baf48a1..217ff4ae8b2e 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -120,7 +120,7 @@ int __init clk_disable_unused(void);
 void __init rk30_map_io(void)
 {
 	rk30_map_common_io();
-#ifdef DEBUG_UART_BASE
+#if defined(CONFIG_ARCH_RK3066B) && defined(DEBUG_UART_BASE)
 	writel_relaxed(0x000f0005, RK30_GRF_BASE + 0x74);
 	writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
 	writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
@@ -135,8 +135,10 @@ void __init rk30_map_io(void)
 	rk29_sram_init();
 	board_clock_init();
 	rk30_l2_cache_init();
+#if !defined(CONFIG_ARCH_RK3066B)
 	ddr_init(DDR_TYPE, DDR_FREQ);
 	clk_disable_unused();
+#endif
 	rk30_iomux_init();
 	rk30_boot_mode_init();
 }
-- 
2.35.3

