From 63ce107bd4ec202df7f274cc27fb03a31debd1ee Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 30 Mar 2012 10:41:03 +0800
Subject: [PATCH] ion: modify memory leak

---
 drivers/gpu/ion/ion.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/ion/ion.c b/drivers/gpu/ion/ion.c
index abf2dbe884fa..954a765a8db4 100755
--- a/drivers/gpu/ion/ion.c
+++ b/drivers/gpu/ion/ion.c
@@ -831,6 +831,7 @@ static void ion_vma_close(struct vm_area_struct *vma)
 	list_for_each_entry(map, &buffer->map_addr, list)
 		if(map->vaddr == vma->vm_start){
 			list_del(&map->list);
+                        kfree(map);
 			break;
 		}
 }
-- 
2.35.3

