From 5e69085097804f80bca754c85f6a7025d815fc8e Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 9 Mar 2015 18:32:01 +0800
Subject: [PATCH] rk3368 lcdc: FBDC_ABGR_888 data format support alpha

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index 0766541fc9c6..ba13a770bbb5 100755
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -648,6 +648,7 @@ static int rk3368_lcdc_alpha_cfg(struct rk_lcdc_driver *dev_drv, int win_id)
 	for (i = 0; i < win->area_num; i++) {
 		ppixel_alpha |= ((win->area[i].format == ARGB888) ||
 				 (win->area[i].format == FBDC_ARGB_888) ||
+				 (win->area[i].format == FBDC_ABGR_888) ||
 				 (win->area[i].format == ABGR888)) ? 1 : 0;
 	}
 	global_alpha = (win->g_alpha_val == 0) ? 0 : 1;
-- 
2.35.3

