From 57298805aac6499795c99bbcc68788155d79a7aa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 15 Jan 2013 13:12:07 +0800
Subject: [PATCH] rk30: pm: fix core periph setup when suspend

---
 arch/arm/mach-rk30/pm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 218f1af8dd1a..4fde4e51c79e 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -644,7 +644,7 @@ static int rk30_pm_enter(suspend_state_t state)
 	clk_sel1 = cru_readl(CRU_CLKSELS_CON(1));
 
 	cru_writel(PLL_MODE_SLOW(APLL_ID), CRU_MODE_CON);
-	cru_writel(CORE_PERIPH_MSK | CORE_PERIPH_2
+	cru_writel(CORE_PERIPH_W_MSK | CORE_PERIPH_2
 		   | CORE_CLK_DIV_W_MSK | CORE_CLK_DIV(1)
 		   | CPU_CLK_DIV_W_MSK | CPU_CLK_DIV(1)
 		   | CORE_SEL_PLL_W_MSK | CORE_SEL_APLL
-- 
2.35.3

