From 4e84efeb44022ad6662cd8edbd979dcac4ff2247 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 18 Sep 2012 21:26:56 +0800
Subject: [PATCH] rk2928: set suspend freq 600MHz

---
 arch/arm/mach-rk2928/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/cpufreq.c b/arch/arm/mach-rk2928/cpufreq.c
index 0f9c842d231e..c379a36e06d7 100644
--- a/arch/arm/mach-rk2928/cpufreq.c
+++ b/arch/arm/mach-rk2928/cpufreq.c
@@ -57,7 +57,7 @@ static unsigned int max_freq = -1;
 #define MASK_FURTHER_CPUFREQ            0x30
 /* With 0x00(NOCHANGE), it depends on the previous "further" status */
 static int no_cpufreq_access;
-static unsigned int suspend_freq = 816 * 1000;
+static unsigned int suspend_freq = 600 * 1000;
 
 static struct workqueue_struct *freq_wq;
 static struct clk *cpu_clk;
-- 
2.35.3

