From 15ebb6845231bfb780b9b8c1265a8f3864a3b8ac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 29 Jul 2011 09:48:10 +0800
Subject: [PATCH] rk29: cpufreq: fix section mismatch warning

---
 arch/arm/mach-rk29/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/cpufreq.c b/arch/arm/mach-rk29/cpufreq.c
index 46d8551404e8..9952961eb7c1 100755
--- a/arch/arm/mach-rk29/cpufreq.c
+++ b/arch/arm/mach-rk29/cpufreq.c
@@ -330,7 +330,7 @@ static void rk29_cpufreq_work_func(struct work_struct *work)
 }
 #endif
 
-static int __init rk29_cpufreq_init(struct cpufreq_policy *policy)
+static int rk29_cpufreq_init(struct cpufreq_policy *policy)
 {
 	arm_clk = clk_get(NULL, "arm_pll");
 	if (IS_ERR(arm_clk))
-- 
2.35.3

