From d2fbcf6518652021b7e3263faaf4fea820eaed3b Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 26 Jul 2013 18:35:24 +0800
Subject: [PATCH] Wi-Fi Group:update Kconfig

---
 drivers/net/wireless/esp8089/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/esp8089/Kconfig b/drivers/net/wireless/esp8089/Kconfig
index 5ee8bc63381b..c5430c536e2b 100755
--- a/drivers/net/wireless/esp8089/Kconfig
+++ b/drivers/net/wireless/esp8089/Kconfig
@@ -3,7 +3,8 @@
       select WIRELESS_EXT
       select WEXT_PRIV
       select CFG80211
-      select NL80211_TESTMODE
+      select MAC80211
+	  select NL80211_TESTMODE
       bool "Espressif 8089 sdio Wi-Fi"
       ---help---
         Espressif 8089 sdio Wi-Fi support
-- 
2.35.3

