From 676e6f9da7162dc49ce688b78ae8957563a4906c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 30 Jun 2011 11:29:37 +0800
Subject: [PATCH] rk29: backlight: default delay set to 100ms

---
 drivers/video/backlight/rk29_backlight.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/backlight/rk29_backlight.c b/drivers/video/backlight/rk29_backlight.c
index 9e7bd67fc5aa..21d0d8782bf9 100755
--- a/drivers/video/backlight/rk29_backlight.c
+++ b/drivers/video/backlight/rk29_backlight.c
@@ -172,7 +172,7 @@ static int rk29_backlight_probe(struct platform_device *pdev)
 	}
 
 	if (!rk29_bl_info->delay_ms)
-		rk29_bl_info->delay_ms = 30;
+		rk29_bl_info->delay_ms = 100;
 
 	if (rk29_bl_info->min_brightness < 0 || rk29_bl_info->min_brightness > BL_STEP)
 		rk29_bl_info->min_brightness = 52;
-- 
2.35.3

