From feedd180c85f485e8db56c4907abf3b4408149ba Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Wed, 16 Jul 2014 14:50:41 +0800
Subject: [PATCH] 3036: add ion reverse

---
 arch/arm/mach-rockchip/rk3036.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index a992fe03f6ca..aee51a47bec9 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -180,7 +180,7 @@ static void __init rk3036_init_late(void)
 static void __init rk3036_reserve(void)
 {
 	/* reserve memory for ION */
-	//rockchip_ion_reserve();
+	rockchip_ion_reserve();
 }
 
 static void rk3036_restart(char mode, const char *cmd)
-- 
2.35.3

