From 8381af2ba996fa08b23b6fbd9f147f2aad8ed39f Mon Sep 17 00:00:00 2001
From: lhh <lhh@rock-chips.com>
Date: Fri, 24 Dec 2010 11:33:00 +0800
Subject: [PATCH] no battery temp is 25.8

---
 drivers/power/bq27510_battery.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/power/bq27510_battery.c b/drivers/power/bq27510_battery.c
index 6ab0df6e8bc0..02abbb5d7b8f 100644
--- a/drivers/power/bq27510_battery.c
+++ b/drivers/power/bq27510_battery.c
@@ -114,6 +114,9 @@ static int bq27510_battery_temperature(struct bq27510_device_info *di)
 	}
 	temp = get_unaligned_le16(buf);
 	temp = temp - 2731;
+	#if CONFIG_NO_BATTERY_IC
+	temp = 258;
+	#endif
 	DBG("Enter:%s %d--temp = %d\n",__FUNCTION__,__LINE__,temp);
 	return temp;
 }
-- 
2.35.3

