From b1618719241bbfed9d27e91405db5cc900e471e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Sat, 2 Feb 2013 12:10:45 +0800
Subject: [PATCH] rk29_pcm : close debug

---
 sound/soc/rk29/rk29_pcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk29_pcm.c b/sound/soc/rk29/rk29_pcm.c
index c28bfc69cff5..d936b2653d09 100755
--- a/sound/soc/rk29/rk29_pcm.c
+++ b/sound/soc/rk29/rk29_pcm.c
@@ -106,7 +106,7 @@ static void rockchip_pcm_enqueue(struct snd_pcm_substream *substream)
 	unsigned int limit;
 	int ret;
 
-	printk("Enter::%s----%d prtd->dma_period = %d prtd->dma_limit = %d\n",__FUNCTION__,__LINE__,prtd->dma_period,prtd->dma_limit);
+	DBG("Enter::%s----%d prtd->dma_period = %d prtd->dma_limit = %d\n",__FUNCTION__,__LINE__,prtd->dma_period,prtd->dma_limit);
 
 	if (rk29_dma_has_circular())
 		limit = (prtd->dma_end - prtd->dma_start) / prtd->dma_period;
-- 
2.35.3

