From db95a55ffa214b0bab8643454875a42d655a8ad7 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Fri, 27 Feb 2015 17:59:00 +0800
Subject: [PATCH] rk: ion: allow the cacheable buffer to map iommu

Signed-off-by: CMY <cmy@rock-chips.com>
---
 drivers/staging/android/ion/ion.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/staging/android/ion/ion.c b/drivers/staging/android/ion/ion.c
index 3a97ad0367bb..0271c19ced2c 100755
--- a/drivers/staging/android/ion/ion.c
+++ b/drivers/staging/android/ion/ion.c
@@ -811,12 +811,6 @@ int ion_map_iommu(struct device *iommu_dev, struct ion_client *client,
 
 	mutex_lock(&buffer->lock);
 
-	if (ion_buffer_cached(buffer)) {
-		pr_err("%s: Cannot map iommu as cached.\n", __func__);
-		ret = -EINVAL;
-		goto out;
-	}
-
 	if (!handle->buffer->heap->ops->map_iommu) {
 		pr_err("%s: map_iommu is not implemented by this heap.\n",
 		       __func__);
-- 
2.35.3

