From 4e12a2b1a9705e75f8ae99bfadf1b50b05060c77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:29:37 +0800
Subject: [PATCH] partitions: msdos: fix for v2.6.36

---
 fs/partitions/msdos.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/partitions/msdos.c b/fs/partitions/msdos.c
index 3c47d5db12d7..251b425dfecf 100644
--- a/fs/partitions/msdos.c
+++ b/fs/partitions/msdos.c
@@ -431,6 +431,9 @@ static struct {
  
 int msdos_partition(struct parsed_partitions *state)
 {
+#if defined(CONFIG_SDMMC_RK29) && !defined(CONFIG_SDMMC_RK29_OLD)
+	const struct block_device *bdev = state->bdev;
+#endif
 	sector_t sector_size = bdev_logical_block_size(state->bdev) / 512;
 	Sector sect;
 	unsigned char *data;
-- 
2.35.3

