From 686c2212a69cadabe0bff2cd0f1a65ab7143b845 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 25 Dec 2014 11:26:52 +0800
Subject: [PATCH] rk fb: compatible when 32bit system call used on 64bit kernel

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 32323babb985..07daf387e1c9 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3404,6 +3404,7 @@ static struct fb_ops fb_ops = {
 	.fb_set_par = rk_fb_set_par,
 	.fb_blank = rk_fb_blank,
 	.fb_ioctl = rk_fb_ioctl,
+	.fb_compat_ioctl = rk_fb_ioctl,
 	.fb_pan_display = rk_fb_pan_display,
 	.fb_read = rk_fb_read,
 	.fb_write = rk_fb_write,
-- 
2.35.3

