From a0132e447e1cb125cf6db12e96edf95f22803f8d Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 13 Jul 2012 12:17:00 +0800
Subject: [PATCH] rkfb:use macro HDMI_HPD_ACTIVED to judge HDMI plug or not

---
 drivers/video/rockchip/rk_fb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 05a9c7ed19b8..5a42aa8735c3 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -217,7 +217,7 @@ static int rk_pan_display(struct fb_var_screeninfo *var, struct fb_info *info)
 
 	#if defined(CONFIG_HDMI_RK30)
 		#if defined(CONFIG_DUAL_DISP_IN_KERNEL)
-			if(hdmi_get_hotplug())
+			if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED)
 			{
 				if(inf->num_fb >= 2)
 				{
@@ -370,7 +370,7 @@ static int rk_fb_set_par(struct fb_info *info)
 	
 	#if defined(CONFIG_HDMI_RK30)
 		#if defined(CONFIG_DUAL_DISP_IN_KERNEL)
-			if(hdmi_get_hotplug())
+			if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED)
 			{
 				if(inf->num_fb >= 2)
 				{
@@ -481,7 +481,7 @@ static int rk_fb_set_par(struct fb_info *info)
 
 	#if defined(CONFIG_HDMI_RK30)
 		#if defined(CONFIG_DUAL_DISP_IN_KERNEL)
-			if(hdmi_get_hotplug())
+			if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED)
 			{
 				if(info != info2)
 				{
-- 
2.35.3

