From ea0dc9fd996806cd1d04be34d106aeb65be0c33d Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 23 Apr 2012 13:10:39 +0800
Subject: [PATCH] rk30: fix lock bug in function dvfs_get_freq_volt_table

---
 arch/arm/mach-rk30/dvfs.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index f2365da416f0..a15998c9bd12 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -169,13 +169,14 @@ void dvfs_clk_register_set_rate_callback(struct clk *clk, clk_dvfs_target_callba
 struct cpufreq_frequency_table *dvfs_get_freq_volt_table(struct clk *clk) 
 {
 	struct clk_node *info = clk_get_dvfs_info(clk);
-
+	struct cpufreq_frequency_table *table;
 	if (!info || !info->dvfs_table) {
 		return NULL;
 	}
 	mutex_lock(&mutex);
-	return info->dvfs_table;
+	table = info->dvfs_table;
 	mutex_unlock(&mutex);
+	return table;
 }
 
 int dvfs_set_freq_volt_table(struct clk *clk, struct cpufreq_frequency_table *table)
-- 
2.35.3

