From 0e4721fd00eb12b091e85ab25938eec8fc84cbce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 19 Apr 2012 15:26:30 +0800
Subject: [PATCH] rk30: fix boot mode setup

---
 arch/arm/mach-rk30/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index b78ea3195bae..866284d292a5 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -93,7 +93,7 @@ static int boot_mode;
 static void __init rk30_boot_mode_init(void)
 {
 	u32 boot_flag = readl_relaxed(RK30_PMU_BASE + PMU_SYS_REG0);
-	u32 boot_mode = readl_relaxed(RK30_PMU_BASE + PMU_SYS_REG1);
+	boot_mode = readl_relaxed(RK30_PMU_BASE + PMU_SYS_REG1);
 
 	if (boot_flag == (SYS_KERNRL_REBOOT_FLAG | BOOT_RECOVER)) {
 		boot_mode = BOOT_MODE_RECOVERY;
-- 
2.35.3

