From 5924d4c6238026e317241e57e846fa300ca5ad53 Mon Sep 17 00:00:00 2001
From: zyc <zyc@rock-chips.com>
Date: Wed, 17 Oct 2012 09:49:40 +0800
Subject: [PATCH] camera: update hi253 driver.

---
 drivers/media/video/hi253.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/media/video/hi253.c b/drivers/media/video/hi253.c
index 128faf789933..898dc1e90b0e 100755
--- a/drivers/media/video/hi253.c
+++ b/drivers/media/video/hi253.c
@@ -1244,27 +1244,35 @@ static struct reginfo sensor_vga[] =
 
 /* 352X288 CIF */
 static struct reginfo sensor_cif[] =
-{};
+{
+	{END_REG, END_REG},
+};
 
 /* 320*240 QVGA */
 static  struct reginfo sensor_qvga[] =
-{};
+{
+	{END_REG, END_REG},
+};
 
 /* 176X144 QCIF*/
 static struct reginfo sensor_qcif[] =
-{};
+{
+	{END_REG, END_REG},
+};
 
 #endif
 static  struct reginfo sensor_ClrFmt_YUYV[]=
 {
 
-    {0x00, 0x00}
+    //{0x00, 0x00}
+	{END_REG, END_REG},
 };
 
 static  struct reginfo sensor_ClrFmt_UYVY[]=
 {
 
-    {0x00, 0x00}
+    //{0x00, 0x00}
+	{END_REG, END_REG},
 };
 
 #if CONFIG_SENSOR_WhiteBalance
-- 
2.35.3

