From 50128e4fb79351675acc256e1d1571e7cc81e4b4 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 1 Feb 2013 16:57:45 +0800
Subject: [PATCH] rk3188 lcdc:wait 25ms before disable lcdc clk

---
 drivers/video/rockchip/lcdc/rk3188_lcdc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/lcdc/rk3188_lcdc.c b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
index fb56d7f82a21..412724760a67 100644
--- a/drivers/video/rockchip/lcdc/rk3188_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
@@ -844,6 +844,7 @@ static int rk3188_lcdc_early_suspend(struct rk_lcdc_device_driver *dev_drv)
 		return 0;
 	}
 
+	mdelay(25);
 	rk3188_lcdc_clk_disable(lcdc_dev);
 	return 0;
 }
-- 
2.35.3

