From 5a9f3c6cd6828c5257138b8b1eb53fa7333c894c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 6 Jun 2014 11:44:48 +0800
Subject: [PATCH] rk3288:pmic:act8846:modify shutdown func

---
 drivers/regulator/act8846.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/regulator/act8846.c b/drivers/regulator/act8846.c
index 158f491b4044..9519294a4c4a 100755
--- a/drivers/regulator/act8846.c
+++ b/drivers/regulator/act8846.c
@@ -803,6 +803,7 @@ int act8846_device_shutdown(void)
 #if 1
 	if (act8846->pmic_hold_gpio) {
 			gpio_direction_output(act8846->pmic_hold_gpio,0);
+			msleep(100);
 			arm_pm_restart('h', "charge");
 	}
 	
-- 
2.35.3

