From 2bf1c6e151f522f4e6d82fa3eb17c0c47d4ccbee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 22 Aug 2014 16:46:25 +0800
Subject: [PATCH] rk312x:clk:select 480M for clk usb480m

---
 arch/arm/mach-rockchip/rk312x.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk312x.c b/arch/arm/mach-rockchip/rk312x.c
index 0937acabe6c7..01ff0f575e0a 100755
--- a/arch/arm/mach-rockchip/rk312x.c
+++ b/arch/arm/mach-rockchip/rk312x.c
@@ -104,6 +104,8 @@ static void usb_uart_init(void)
 
 static void __init rk312x_dt_map_io(void)
 {
+	u32 val;
+	
 	iotable_init(rk312x_io_desc, ARRAY_SIZE(rk312x_io_desc));
 	debug_ll_io_init();
 	usb_uart_init();
@@ -116,6 +118,11 @@ static void __init rk312x_dt_map_io(void)
 	dsb();
 	writel_relaxed(1, RK312X_TIMER5_VIRT + 0x10);
 	dsb();
+	val = readl_relaxed(RK_CRU_VIRT + RK312X_CRU_MISC_CON);
+	val &= (~(1 << 15));
+	writel_relaxed(0x80000000 | val, RK_CRU_VIRT + RK312X_CRU_MISC_CON);
+	dsb();
+
 }
 
 static void __init rk3126_dt_map_io(void)
-- 
2.35.3

