From d96d8b069203056332219205ea39110127bbbd4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9E=97=E8=BE=89=E8=BE=89?= <lhh@rock-chips.com>
Date: Fri, 4 Jun 2010 02:16:13 +0000
Subject: [PATCH] update arch/arm/mach-rk2818/clock.c

---
 arch/arm/mach-rk2818/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index 1607ebb414f0..cf48049e5bc1 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -1376,7 +1376,7 @@ static int arm_clk_set_rate(struct clk *clk, unsigned long rate)
 			break;
 	}
 	pll_clk_slow_mode(&arm_pll_clk, 0);
-
+	mdelay(5);
 	pr_debug("clock: set %s to %ld MHz, delay count=%d\n", clk->name, rate/1000000, unit);
 
 	arm_pll_clk.rate = arm_pll_clk.recalc(&arm_pll_clk);
-- 
2.35.3

