From 317c1d30d5aa50886e3715bb9763c8f2441187fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Thu, 27 Mar 2014 14:04:36 +0800
Subject: [PATCH] RK3288: support suspend

---
 arch/arm/mach-rockchip/rk3288.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index efe01d74ae45..d2be731fc6e2 100644
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -488,7 +488,7 @@ static inline void rk_pm_soc_pd_resume(void)
 //extern bool console_suspend_enabled;
 static void __init rk3288_init_suspend(void)
 {
-    return;
+    //return;
     printk("%s\n",__FUNCTION__);
     rockchip_suspend_init();       
     //rkpm_pie_init();
-- 
2.35.3

