From 8d45acc0f060d94c691ed8c9a5e1ad14337cb3aa Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Mon, 14 Oct 2013 15:43:53 +0800
Subject: [PATCH] headset: mod log error && mic mistake

---
 drivers/headset_observe/rk_headset.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/headset_observe/rk_headset.c b/drivers/headset_observe/rk_headset.c
index 1cf717b1c6f1..b324861ca6e7 100755
--- a/drivers/headset_observe/rk_headset.c
+++ b/drivers/headset_observe/rk_headset.c
@@ -276,7 +276,7 @@ static void Hook_work(struct work_struct *work)
 		DBG("old_status == headset_info->hook_status\n");
 		goto RE_ERROR;
 	}	
-	DBG("Hook_work -- level = %d  hook status is %s\n",level,headset_info->hook_status?"key up":"key down");	
+	DBG("Hook_work -- level = %d  hook status is %s\n",level,headset_info->hook_status?"key down":"key up");	
 	if(headset_info->hook_status == HOOK_DOWN)
 	{
 		if(pdata->Hook_down_type == HOOK_DOWN_HIGH)
@@ -341,7 +341,7 @@ static void headset_timer_callback(unsigned long arg)
 		headset->isMic= 0;//No microphone
 		
 	printk("headset->isMic = %d\n",headset->isMic);	
-	headset_info->cur_headset_status = headset_info->isMic ? 1:3;//BIT_HEADSET:BIT_HEADSET_NO_MIC;//
+	headset_info->cur_headset_status = headset_info->isMic ? 1:2;//BIT_HEADSET:BIT_HEADSET_NO_MIC;//
 	#if defined(CONFIG_SND_RK_SOC_RK2928) || defined(CONFIG_SND_RK29_SOC_RK610)
 	rk2928_codec_set_spk(HEADSET_IN);
 	if(headset_info->cur_headset_status == 1)
-- 
2.35.3

