From 16d01425753659c397924529558ccdc1cf1060a0 Mon Sep 17 00:00:00 2001
From: xxh <xxh@rock-chips.com>
Date: Fri, 29 Mar 2013 18:47:18 +0800
Subject: [PATCH] remove mu509 uart gpio request

---
 drivers/misc/3g_module/mu509.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/3g_module/mu509.c b/drivers/misc/3g_module/mu509.c
index 29d341a401e0..d6b4dad93a31 100755
--- a/drivers/misc/3g_module/mu509.c
+++ b/drivers/misc/3g_module/mu509.c
@@ -203,9 +203,9 @@ static int mu509_probe(struct platform_device *pdev)
 	if(pdata->io_init)
 		pdata->io_init();
 
-   printk("******** mu509_probe_mu509_probe usb autosuspend *********\n");	
- // xxh begin
-		#if 1	
+   printk("******** mu509_probe_mu509_probe*********\n");	
+
+		#if 0	
 			rk30_mux_api_set(GPIO1A1_UART0SOUT_NAME, GPIO1A_GPIO1A4);	
 				    result = gpio_request(RK30_PIN1_PA4, "uart1_sout");			
 					if (result < 0) {	
@@ -220,7 +220,7 @@ static int mu509_probe(struct platform_device *pdev)
 			}		
    
    	  #endif	
-// xxh end
+
 
 	mu509_data = kzalloc(sizeof(struct modem_dev), GFP_KERNEL);
 	if(mu509_data == NULL)
-- 
2.35.3

