From bbccb1ea0833e18d2dbcee7e661ba2c82d38b318 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Fri, 22 May 2015 18:34:03 +0800
Subject: [PATCH] display-sys: delete unused code.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/display-sys.c | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/drivers/video/rockchip/display-sys.c b/drivers/video/rockchip/display-sys.c
index 8fd6926c67a9..c2229187bc9d 100755
--- a/drivers/video/rockchip/display-sys.c
+++ b/drivers/video/rockchip/display-sys.c
@@ -606,18 +606,10 @@ struct rk_display_device
 		mutex_unlock(&allocated_dsp_lock);
 
 		if (new_dev->idx >= 0) {
-			if (new_dev->property == DISPLAY_MAIN)
-				new_dev->dev =
-				device_create(display_class, parent,
-					      MKDEV(0, 0), new_dev,
-					      "%s", new_dev->type);
-			else
-				new_dev->dev =
-				device_create(display_class, parent,
-					      MKDEV(0, 0), new_dev,
-					      "display%d.%s",
-					      new_dev->property,
-					      new_dev->type);
+			new_dev->dev =
+			device_create(display_class, parent,
+				      MKDEV(0, 0), new_dev,
+				      "%s", new_dev->type);
 			if (!IS_ERR(new_dev->dev)) {
 				new_dev->parent = parent;
 				new_dev->driver = driver;
-- 
2.35.3

