From ef2e02705e382632b7d725fe39eea8e10371363b Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Thu, 27 Sep 2012 09:52:47 +0800
Subject: [PATCH] rk3066b m701: fix white screen when reboot loader

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index fb3b4ffc4371..17cd332f1f54 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -246,9 +246,13 @@ static int rk29_backlight_io_deinit(void)
 {
 	int ret = 0;
 #ifdef  LCD_DISP_ON_PIN
+	gpio_direction_output(BL_EN_PIN, 0);
+	gpio_set_value(BL_EN_PIN, !BL_EN_VALUE);
 	gpio_free(BL_EN_PIN);
 #endif
 	rk30_mux_api_set(PWM_MUX_NAME, PWM_MUX_MODE_GPIO);
+	gpio_request(PWM_GPIO, NULL);
+	gpio_direction_output(PWM_GPIO, GPIO_LOW);
 	return ret;
 }
 
-- 
2.35.3

