From d365bf6679aee846edef43dc7d4001fbbc329b19 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 2 Jun 2015 08:56:49 +0800
Subject: [PATCH] arm64: rockchip: efuse: support set system serial

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/mach-rockchip/efuse.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm/mach-rockchip/efuse.c b/arch/arm/mach-rockchip/efuse.c
index 439aa49bd7c1..2c1030f52018 100644
--- a/arch/arm/mach-rockchip/efuse.c
+++ b/arch/arm/mach-rockchip/efuse.c
@@ -211,7 +211,6 @@ static int rk3288_get_leakage(int ch)
 	return efuse_buf[23+ch];
 }
 
-#ifdef CONFIG_ARM
 static void __init rk3288_set_system_serial(void)
 {
 	int i;
@@ -225,9 +224,6 @@ static void __init rk3288_set_system_serial(void)
 	system_serial_low = crc32(0, buf, 8);
 	system_serial_high = crc32(system_serial_low, buf + 8, 8);
 }
-#else
-static inline void __init rk3288_set_system_serial(void) {}
-#endif
 
 int rk312x_efuse_readregs(u32 addr, u32 length, u8 *buf)
 {
-- 
2.35.3

