From f506d85379139404e776380c81441ae286c4c548 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 8 Apr 2013 11:00:31 +0800
Subject: [PATCH] rk3066: rename macro definition of RK3066

---
 arch/arm/mach-rk30/dvfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 2187dc9df0c8..06f7c6d28b3c 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -96,7 +96,7 @@ static struct cpufreq_frequency_table logic_high_arm_table[] = {
         {.frequency = 1608 * DVFS_KHZ, .index = 50 * DVFS_MV},
         {.frequency = CPUFREQ_TABLE_END},
 };
-#elif defined(CONFIG_SOC_RK3066)
+#elif defined(CONFIG_ARCH_RK30XX)
 static struct cpufreq_frequency_table arm_high_logic_table[] = {
         {.frequency = 1416 * DVFS_KHZ, .index = 50 * DVFS_MV},
         {.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
-- 
2.35.3

