From aee3c481be41b77ed3a08ad2ec68fe75aeda3310 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Sat, 27 Jul 2013 09:44:27 +0800
Subject: [PATCH] rt5616: record from mic2 because sdk default use mic2

---
 sound/soc/codecs/rt5616.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/codecs/rt5616.c b/sound/soc/codecs/rt5616.c
index 44eaa2346ed0..6472a9e93c7b 100755
--- a/sound/soc/codecs/rt5616.c
+++ b/sound/soc/codecs/rt5616.c
@@ -51,10 +51,10 @@ static struct rt5616_init_reg init_list[] = {
 	/*LOUT*/
 	{RT5616_LOUT_MIXER	, 0x3000},
 	/*Capture*/
-	{RT5616_REC_L2_MIXER	, 0x006d}, //MIC1 -> RECMIXL
-	{RT5616_REC_R2_MIXER	, 0x006d}, //MIC1 -> RECMIXR
-	//{RT5616_REC_L2_MIXER	, 0x006b}, //MIC2 -> RECMIXL
-	//{RT5616_REC_R2_MIXER	, 0x006b}, //MIC2 -> RECMIXR
+	//{RT5616_REC_L2_MIXER	, 0x006d}, //MIC1 -> RECMIXL
+	//{RT5616_REC_R2_MIXER	, 0x006d}, //MIC1 -> RECMIXR
+	{RT5616_REC_L2_MIXER	, 0x006b}, //MIC2 -> RECMIXL
+	{RT5616_REC_R2_MIXER	, 0x006b}, //MIC2 -> RECMIXR
 	{RT5616_ADC_DIG_VOL	, 0xafaf}, //mute adc for pop noise
 	/*MIC*/
 	{RT5616_STO1_ADC_MIXER	, 0x3802},
-- 
2.35.3

