From 1f6bc690c70b535de57103581ae429be46f2ef3e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 28 May 2010 15:01:35 +0000
Subject: [PATCH] change freq early

---
 arch/arm/mach-rk2818/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2818/cpufreq.c b/arch/arm/mach-rk2818/cpufreq.c
index 36b85fa26951..e35953829edc 100755
--- a/arch/arm/mach-rk2818/cpufreq.c
+++ b/arch/arm/mach-rk2818/cpufreq.c
@@ -103,5 +103,5 @@ static int __init rk2818_cpufreq_register(void)
 	return cpufreq_register_driver(&rk2818_cpufreq_driver);
 }
 
-late_initcall(rk2818_cpufreq_register);
+core_initcall_sync(rk2818_cpufreq_register);
 
-- 
2.35.3

