From 16e82ab2b6ed4a6726495ce8cdbd550cb944eab6 Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Tue, 23 Jun 2015 19:31:16 +0800
Subject: [PATCH] arm64: rockchip: rk3368: dts: add leakage adjust volt support

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index 37ac43432850..1fbf27597ae3 100644
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -1041,6 +1041,14 @@
 						/*temp    freq*/
 						100     816000
 						>;
+					lkg_adjust_volt_en = <1>;
+					channel = <0>;
+					def_table_lkg = <25>;
+					min_adjust_freq = <216000>;
+					lkg_adjust_volt_table = <
+						/*lkg(mA)  volt(uV)*/
+						0         25000
+						>;
 				};
 				clk_core_l_dvfs_table: clk_core_l {
 					operating-points = <
@@ -1065,6 +1073,14 @@
 						/*temp    freq*/
 						100     816000
 						>;
+					lkg_adjust_volt_en = <1>;
+					channel = <0>;
+					def_table_lkg = <25>;
+					min_adjust_freq = <216000>;
+					lkg_adjust_volt_table = <
+						/*lkg(mA)  volt(uV)*/
+						0         25000
+						>;
 				};
 			};
 		};
-- 
2.35.3

