From 276f88f33825119afd9a787833af283242d463b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 24 Mar 2011 09:52:18 +0800
Subject: [PATCH] pack-kernel.sh: more exclude files

---
 pack-kernel.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 54113e8b439d..c1fe6250ce7f 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -40,6 +40,15 @@ arch/arm/configs/rk29_Winaccord_defconfig
 arch/arm/mach-rk29/board-rk29-aigo.c
 arch/arm/mach-rk29/board-rk29aigo-key.c
 arch/arm/configs/rk29_aigo_defconfig
+
+arch/arm/mach-rk29/board-rk29-a22*
+arch/arm/configs/rk29_a22_defconfig
+
+arch/arm/mach-rk29/board-rk29-fih.c
+arch/arm/configs/rk29_FIH_defconfig
+
+arch/arm/mach-rk29/board-rk29-phonesdk*
+arch/arm/configs/rk29_phonesdk_defconfig
 )
 
 # ---------------------------------------------------------------------------
-- 
2.35.3

