From 8b305f246a31be12a3826a88409b28aa6a3e4ae8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=92=9F=E5=8B=87=E6=B1=AA?= <zyw@rock-chips.com>
Date: Wed, 9 Jun 2010 06:33:57 +0000
Subject: [PATCH] fix yuv4201 bug

---
 drivers/video/rk2818_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk2818_fb.c b/drivers/video/rk2818_fb.c
index 1d339e1ce6a7..2bc4d6da9e96 100644
--- a/drivers/video/rk2818_fb.c
+++ b/drivers/video/rk2818_fb.c
@@ -2326,7 +2326,7 @@ static int __init rk2818fb_probe (struct platform_device *pdev)
 		ret = -ENOENT;
 		goto unregister_win1fb;
 	}
-    inf->dclk_parent = clk_get(&pdev->dev, "codec_pll");
+    inf->dclk_parent = clk_get(&pdev->dev, "arm_pll");
     if (!inf->dclk_parent || IS_ERR(inf->dclk_parent))
     {
 		printk(KERN_ERR "failed to get lcd dclock parent source\n");
-- 
2.35.3

