From 5478bf8d59e372638be9ff764f5d2e534f426a14 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Tue, 24 Mar 2015 15:58:26 +0800
Subject: [PATCH] disabled gmac as default in rk3368.dtsi

---
 arch/arm64/boot/dts/rk3368.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index 8e361fc58371..478dfef621f5 100755
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -1101,6 +1101,7 @@
 		phy-mode = "rgmii";
 		pinctrl-names = "default";
 		pinctrl-0 = <&rgmii_pins>;
+                status = "disabled";
 	};
 
 	gpu {
-- 
2.35.3

