From f90a571757f57c07b1058ede8018f4364085064d Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Sun, 7 Oct 2012 17:07:17 +0800
Subject: [PATCH] sensor:close mma7660 debug info

---
 drivers/input/sensors/accel/mma7660.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/accel/mma7660.c b/drivers/input/sensors/accel/mma7660.c
index 6238549d1ac6..ea05e4b74355 100755
--- a/drivers/input/sensors/accel/mma7660.c
+++ b/drivers/input/sensors/accel/mma7660.c
@@ -235,7 +235,7 @@ static int sensor_report_value(struct i2c_client *client)
 		axis.y = axis_average.y_average / axis_average.count;	
 		axis.z = axis_average.z_average / axis_average.count;
 		
-		printk( "%s: axis = %d  %d  %d \n", __func__, axis.x, axis.y, axis.z);
+		DBG( "%s: axis = %d  %d  %d \n", __func__, axis.x, axis.y, axis.z);
 		
 		memset(&axis_average, 0, sizeof(struct sensor_axis_average));
 		
-- 
2.35.3

