From 2ec82dc352450e8afb89575894792f768911c2ca Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Tue, 3 Dec 2013 21:26:43 +0800
Subject: [PATCH] mask gpio irq register

---
 drivers/pinctrl/pinctrl-rockchip.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index f7f8136126eb..5091a3254dc3 100755
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -1558,8 +1558,8 @@ static int rockchip_interrupts_register(struct platform_device *pdev,
 
 		for(j=0; j<32; j++)
 		{
-			if(bank->domain->ops->map)
-			bank->domain->ops->map(bank->domain, 6*32+bank->pin_base+j, j);
+		//	if(bank->domain->ops->map)
+		//	bank->domain->ops->map(bank->domain, 6*32+bank->pin_base+j, j);
 		}
 #else
 		
-- 
2.35.3

