From f0f9e6941b640ed7061a5531efe506941d92f0c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 16 Jan 2013 16:48:51 +0800
Subject: [PATCH] rk3188: use rk30 dvfs

---
 arch/arm/mach-rk3188/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk3188/Makefile b/arch/arm/mach-rk3188/Makefile
index 783a41b85e90..270b95c7d0ca 100644
--- a/arch/arm/mach-rk3188/Makefile
+++ b/arch/arm/mach-rk3188/Makefile
@@ -15,6 +15,7 @@ obj-y += rk_timer.o
 obj-$(CONFIG_SMP) += ../mach-rk30/platsmp.o ../mach-rk30/headsmp.o
 obj-$(CONFIG_HOTPLUG_CPU) += ../mach-rk30/hotplug.o
 obj-$(CONFIG_CPU_IDLE) += ../mach-rk30/cpuidle.o
+obj-$(CONFIG_DVFS) += ../mach-rk30/dvfs.o
 
 obj-$(CONFIG_MACH_RK3188_FPGA) += board-rk3188-fpga.o
 obj-$(CONFIG_MACH_RK3188_TB) += ../mach-rk30/board-rk3168-tb.o
-- 
2.35.3

