From a7bddc36afee66e3a9cfb176949d8bc54ddf3f2c Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Mon, 5 Nov 2012 10:08:17 +0800
Subject: [PATCH] rk fb: modify  drivers/video/rockchip/Kconfig

---
 drivers/video/rockchip/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/Kconfig b/drivers/video/rockchip/Kconfig
index 27226c26a397..4bcb84ba6783 100755
--- a/drivers/video/rockchip/Kconfig
+++ b/drivers/video/rockchip/Kconfig
@@ -1,4 +1,4 @@
-config FB_ROCKCHIP
+menuconfig FB_ROCKCHIP
         tristate "Frame buffer support for Rockchip "
         depends on FB
         select FB_CFB_FILLRECT
-- 
2.35.3

