From 56b9a8b3cd3eaca13d7e32b077fa5e0ad377ad37 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Mon, 9 Jul 2012 09:55:52 +0800
Subject: [PATCH] camera:modify ov5640_for_td8801.c

---
 drivers/media/video/ov5640_for_td8801.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/video/ov5640_for_td8801.c b/drivers/media/video/ov5640_for_td8801.c
index d878085c56e9..112e3f9f80b3 100644
--- a/drivers/media/video/ov5640_for_td8801.c
+++ b/drivers/media/video/ov5640_for_td8801.c
@@ -3939,6 +3939,9 @@ static long sensor_ioctl(struct v4l2_subdev *sd, unsigned int cmd, void *arg)
 	{
 		case RK29_CAM_SUBDEV_DEACTIVATE:
 		{
+			#if CONFIG_SENSOR_Flash  //hhb
+			sensor_ioctrl(icd, Sensor_Flash, Flash_Off);
+			#endif
 			sensor_deactivate(client);
 			break;
 		}
-- 
2.35.3

