From cd779aad5de3c4a007033984ee2ebcf05d5df3b4 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 28 Aug 2012 14:13:12 +0800
Subject: [PATCH] rkfb: select current screen in lcdc driver

---
 drivers/video/rockchip/rk_fb.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 86f149ac5fb5..a93b737fa7e2 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -971,7 +971,6 @@ int rk_fb_register(struct rk_lcdc_device_driver *dev_drv,
 	}
 		
 	dev_drv->init_lcdc(dev_drv);
-	dev_drv->cur_screen = dev_drv->screen0;
 	dev_drv->load_screen(dev_drv,1);
 	/************fb set,one layer one fb ***********/
 	dev_drv->fb_index_base = fb_inf->num_fb;
-- 
2.35.3

