From 4659766d3c5efbc986257663a46872fe65abdf97 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Fri, 21 Jun 2013 11:20:23 +0800
Subject: [PATCH] sensor:dmard10 use 4G range

---
 drivers/input/sensors/accel/dmard10.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/accel/dmard10.c b/drivers/input/sensors/accel/dmard10.c
index d6610b8b2fb3..737c34957ad1 100755
--- a/drivers/input/sensors/accel/dmard10.c
+++ b/drivers/input/sensors/accel/dmard10.c
@@ -167,7 +167,7 @@
 #define DMARD10_REG_PDET        0x9
 #define DMARD10_REG_PD          0xa
 
-#define DMARD10_RANGE			2000000
+#define DMARD10_RANGE			4000000
 #define DMARD10_PRECISION       10
 #define DMARD10_BOUNDARY        (0x1 << (DMARD10_PRECISION  - 1)) 
 #define DMARD10_GRAVITY_STEP    (DMARD10_RANGE / DMARD10_BOUNDARY)
-- 
2.35.3

