From 176fe4946e2bd2e1b7ebe8b9c9bb7e1c28ad9186 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 6 Jan 2014 10:36:57 +0800
Subject: [PATCH] rk: support make kernel.img and board_kernel.img

---
 arch/arm/Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 3a2d49cd94cf..9526bdf7c166 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -327,3 +327,13 @@ define archhelp
   echo  '                  (distribution) /sbin/$(INSTALLKERNEL) or'
   echo  '                  install to $$(INSTALL_PATH) and run lilo'
 endef
+
+kernel.img: zImage
+	$(Q)$(srctree)/mkkrnlimg $(objtree)/arch/arm/boot/zImage $(objtree)/kernel.img >/dev/null
+	@echo '  Image:  kernel.img is ready'
+
+%_kernel.img: %.dtb zImage
+	$(Q)cat $(objtree)/arch/arm/boot/zImage $(objtree)/arch/arm/boot/dts/$*.dtb > $(objtree)/zImage-dtb && \
+	$(srctree)/mkkrnlimg $(objtree)/zImage-dtb $(objtree)/kernel.img >/dev/null && \
+	rm -f $(objtree)/zImage-dtb
+	@echo '  Image:  kernel.img (with $*.dtb) is ready'
-- 
2.35.3

