From 85b4270444431f6db76728aab8df7802d3871843 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Thu, 26 Aug 2010 20:21:30 +0800
Subject: [PATCH] wifi:config wifi power and reset io

---
 arch/arm/mach-rk2818/board-infosdk.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/mach-rk2818/board-infosdk.c b/arch/arm/mach-rk2818/board-infosdk.c
index 3075720c8d6a..9f482fbb3a17 100755
--- a/arch/arm/mach-rk2818/board-infosdk.c
+++ b/arch/arm/mach-rk2818/board-infosdk.c
@@ -314,6 +314,26 @@ struct rk2818_gpio_expander_info  extgpio_tca6424_settinginfo[] = {
 		.pin_type           	= GPIO_OUT,
 		.pin_value		=GPIO_HIGH,
 	 },
+         {
+                .gpio_num               = TCA6424_P25,  //wifi reg on
+                .pin_type               = GPIO_OUT,
+                .pin_value              = GPIO_HIGH,
+	},
+	{
+                .gpio_num               = TCA6424_P27,  //wifi reset
+                .pin_type               = GPIO_OUT,
+                .pin_value              = GPIO_HIGH,
+	},
+{
+                .gpio_num               = TCA6424_P25,  //wifi reg on
+                .pin_type               = GPIO_OUT,
+                .pin_value              = GPIO_HIGH,
+        },
+        {
+                .gpio_num               = TCA6424_P27,  //wifi reset
+                .pin_type               = GPIO_OUT,
+                .pin_value              = GPIO_HIGH,
+        },
 };
 
 void tca6424_reset_itr(void)
-- 
2.35.3

