From 3220121d4deebfc318b0b6403d1ba8446a36d65d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 27 Jan 2011 20:16:28 +0800
Subject: [PATCH] rk29: clock: rm unbalance mutex_unlock

---
 arch/arm/mach-rk29/clock.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 81460fb5a697..2cf337024b7d 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2247,7 +2247,6 @@ static int __init clk_disable_unused(void)
 		clk_disable_nolock(ck);
 		UNLOCK();
 	}
-	mutex_unlock(&clocks_mutex);
 
 	pmu_set_power_domain(PD_VCODEC, false);
 //	pmu_set_power_domain(PD_DISPLAY, false);
-- 
2.35.3

