From 75d0269d5deb9c6ff74e5e7f1b6c9200a826c438 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 28 Feb 2011 17:35:00 +0800
Subject: [PATCH] disable adc_test

---
 drivers/adc/plat/rk29_adc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/adc/plat/rk29_adc.c b/drivers/adc/plat/rk29_adc.c
index fec8071d6d0f..0ddbec289435 100755
--- a/drivers/adc/plat/rk29_adc.c
+++ b/drivers/adc/plat/rk29_adc.c
@@ -19,7 +19,7 @@
 
 #include "rk29_adc.h"
 
-#define ADC_TEST
+//#define ADC_TEST
 
 struct rk29_adc_device {
 	int			 		irq;
-- 
2.35.3

