From e0dd106efafaf0d253da3d4be7ce77d35606f3a9 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 24 Feb 2012 17:20:13 +0800
Subject: [PATCH] update i2c driver: for compile error

---
 drivers/i2c/busses/i2c-rk30.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rk30.c b/drivers/i2c/busses/i2c-rk30.c
index 82f3073e2e47..24e4c8a3b529 100755
--- a/drivers/i2c/busses/i2c-rk30.c
+++ b/drivers/i2c/busses/i2c-rk30.c
@@ -264,7 +264,7 @@ static int rk30_i2c_resume_noirq(struct device *dev)
 	struct rk30_i2c *i2c = platform_get_drvdata(pdev);
 
 	i2c->suspended = 0;
-    i2c->i2c_init_hw(i2c);
+    i2c->i2c_init_hw(i2c, i2c->scl_rate);
 
 	return 0;
 }
-- 
2.35.3

