From 556efcb8c7737d4916f1ab29d962d46031b37312 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Fri, 11 May 2012 15:45:32 +0800
Subject: [PATCH] rk30 phone: update aic3262 audio_map

---
 sound/soc/rk29/rk29_aic3262.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/sound/soc/rk29/rk29_aic3262.c b/sound/soc/rk29/rk29_aic3262.c
index bca93054c40b..bf28f37c3536 100644
--- a/sound/soc/rk29/rk29_aic3262.c
+++ b/sound/soc/rk29/rk29_aic3262.c
@@ -111,10 +111,10 @@ static const struct snd_soc_dapm_widget rk29_aic3262_dapm_widgets[] = {
 
 static const struct snd_soc_dapm_route audio_map[] = {
 	/* External Mics: MAINMIC, SUBMIC with bias*/
-	{"IN2L", NULL, "Mic Bias Int"},
-	{"IN2R", NULL, "Mic Bias Int"},
-	{"IN3L", NULL, "Mic Bias Int"},
-	{"IN3R", NULL, "Mic Bias Int"},
+	{"IN1L", NULL, "Mic Bias Int"},
+	{"IN1R", NULL, "Mic Bias Int"},
+	{"IN4L", NULL, "Mic Bias Int"},
+	{"IN4R", NULL, "Mic Bias Int"},
 	{"Mic Bias Int", NULL, "Ext Mic"},
 
 	/* External Speakers: HFL, HFR */
@@ -122,8 +122,8 @@ static const struct snd_soc_dapm_route audio_map[] = {
 	{"Ext Spk", NULL, "SPKR"},
 
 	/* Headset Mic: HSMIC with bias */
-	{"IN1L", NULL, "Mic Bias Ext"},
-	{"IN1R", NULL, "Mic Bias Ext"},
+	{"IN2L", NULL, "Mic Bias Ext"},
+	{"IN2R", NULL, "Mic Bias Ext"},
 	{"Mic Bias Ext", NULL, "Headset Mic"},
 
 	/* Headset Stereophone (Headphone): HPL, HPR */
@@ -135,8 +135,8 @@ static const struct snd_soc_dapm_route audio_map[] = {
 	{"Earphone Spk", NULL, "RECM"},
 
 	/* Aux/FM Stereo In: IN4L, IN4R */
-	{"IN4L", NULL, "FM Stereo In"},
-	{"IN4R", NULL, "FM Stereo In"},
+	{"IN3L", NULL, "FM Stereo In"},
+	{"IN3R", NULL, "FM Stereo In"},
 	
 	/* Aux/FM Stereo Out: LOL, LOR */
 	{"FM Stereo Out", NULL, "LOL"},
-- 
2.35.3

