From a4a2e44cb458d8627f7a93c6e6457ee25e49f15d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9E=97=E8=BE=89=E8=BE=89?= <lhh@rock-chips.com>
Date: Tue, 18 May 2010 13:11:41 +0000
Subject: [PATCH] updata adc key

---
 drivers/input/keyboard/rk2818_adckey.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/input/keyboard/rk2818_adckey.c b/drivers/input/keyboard/rk2818_adckey.c
index 3e15adb528e6..fbdd066d8d63 100644
--- a/drivers/input/keyboard/rk2818_adckey.c
+++ b/drivers/input/keyboard/rk2818_adckey.c
@@ -34,12 +34,12 @@
 
 //ROCKCHIP AD KEY CODE ,for demo board
 //      key		--->	EV	
-#define AD2KEY1			KEY_ESC
-#define AD2KEY2 		KEY_7
-#define AD2KEY3 		KEY_8
-#define AD2KEY4 		KEY_9
-#define AD2KEY5 		KEY_0
-#define AD2KEY6 		KEY_W
+#define AD2KEY1			114   ///VOLUME_DOWN 
+#define AD2KEY2 		115   ///VOLUME_UP
+#define AD2KEY3 		59    ///MENU
+#define AD2KEY4 		102   ///HOME
+#define AD2KEY5 		158   ///BACK
+#define AD2KEY6 		61    ///CALL
 
 #define KEYMENU			AD2KEY6
 #define ENDCALL			62
-- 
2.35.3

