From fdf3bf12b3776b00bac2cff62c78e2f00a9617d3 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 8 Aug 2011 17:38:23 +0800
Subject: [PATCH] hdmi drivers: bug on hdmi_unregister function

---
 drivers/video/hdmi/hdmi-core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/hdmi/hdmi-core.c b/drivers/video/hdmi/hdmi-core.c
index 31d3930e5110..441ca7a7d0d0 100755
--- a/drivers/video/hdmi/hdmi-core.c
+++ b/drivers/video/hdmi/hdmi-core.c
@@ -158,6 +158,7 @@ void hdmi_unregister(struct hdmi *hdmi)
 	if(!hdmi)
 		return;
 	id = hdmi->id;
+	del_timer(&hdmi->timer);
 	flush_scheduled_work();
 	hdmi_remove_attrs(hdmi);
 	device_unregister(hdmi->dev);
-- 
2.35.3

