From 0888f26ff083b3fbe73859c27603c1464aeceff8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 28 Feb 2011 17:26:27 +0800
Subject: [PATCH] rk29: pm: add flush_tlb_all

---
 arch/arm/mach-rk29/pm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk29/pm.c b/arch/arm/mach-rk29/pm.c
index 0702e7ddf65c..de461ecfedc9 100644
--- a/arch/arm/mach-rk29/pm.c
+++ b/arch/arm/mach-rk29/pm.c
@@ -10,6 +10,7 @@
 #include <linux/regulator/rk29-pwm-regulator.h>
 #include <linux/io.h>
 #include <linux/wakelock.h>
+#include <asm/tlbflush.h>
 
 #include <mach/rk29_iomap.h>
 #include <mach/cru.h>
@@ -172,6 +173,7 @@ static int rk29_pm_enter(suspend_state_t state)
 
 #ifdef CONFIG_RK29_PWM_REGULATOR
 	/* touch TLB */
+	flush_tlb_all();
 	readl(RK29_PWM_BASE);
 	readl(RK29_GRF_BASE);
 #endif
-- 
2.35.3

