From 874f3ff67e730e9113aa8d8de9ab7d6bb2d9a0b5 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Thu, 12 Jun 2014 08:44:26 +0800
Subject: [PATCH] rk: ion: fix build warning

---
 drivers/staging/android/ion/ion.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/staging/android/ion/ion.c b/drivers/staging/android/ion/ion.c
index 5ea1df6de325..f6b9f885d55d 100755
--- a/drivers/staging/android/ion/ion.c
+++ b/drivers/staging/android/ion/ion.c
@@ -2033,7 +2033,6 @@ static int ion_snapshot_save(struct ion_device *idev)
 {
 	static struct seq_file seqf;
 	struct ion_heap *heap;
-	struct rb_node *n;
 
 	if (!seqf.buf) {
 		seqf.buf = rockchip_ion_snapshot_get(&seqf.size);
-- 
2.35.3

