From 76068bb5d3833f633e1232175c77c5f3c3e28dcc Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Mon, 13 Feb 2012 17:43:52 +0800
Subject: [PATCH] modify dm9620 usb ethernet driver

---
 drivers/net/usb/dm9620.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/usb/dm9620.c b/drivers/net/usb/dm9620.c
index 2f5a8fa1f6c5..9c806f7c1f10 100755
--- a/drivers/net/usb/dm9620.c
+++ b/drivers/net/usb/dm9620.c
@@ -19,9 +19,9 @@
  * V1.5 - Support RK2818 (Debug the Register Function) 
  */
 
-#define DEBUG
+//#define DEBUG
 
-//#define RK2818
+#define RK2818
 
 
 #include <linux/module.h>
-- 
2.35.3

