From 86ec6e21056e517b0cc5ade6604a21b261866a6a Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Wed, 17 Oct 2012 18:13:03 +0800
Subject: [PATCH] rk_fb:fix get wrong display size when plug hdmi start system

---
 drivers/video/rockchip/rk_fb.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 0a2507c2f618..ce2fcd191be0 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -401,10 +401,18 @@ static int rk_fb_set_par(struct fb_info *info)
 	else  //ohterwise  full  screen display
 	{
 		xsize = screen->x_res;
-              	ysize = screen->y_res;
+        ysize = screen->y_res;
 	}
 
 #if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)
+#if defined(CONFIG_RK_HDMI)
+	if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED)
+	{
+		xsize = screen->x_res;
+		ysize = screen->y_res;
+	}
+#endif
+
 	if(screen->screen_id == 0) //this is for device like rk2928 ,whic have one lcdc but two display outputs
 	{			   //save parameter set by android
 		dev_drv->screen0->xsize = xsize;
-- 
2.35.3

