From 3be93856a121d7ff2af3d0315eb8b504532a64a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 17 Aug 2012 19:20:47 +0800
Subject: [PATCH] gitignore: ignore scripts/bmptologo

---
 scripts/.gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/.gitignore b/scripts/.gitignore
index 105b21f08185..78085220ba3a 100644
--- a/scripts/.gitignore
+++ b/scripts/.gitignore
@@ -3,6 +3,7 @@
 #
 conmakehash
 kallsyms
+bmptologo
 pnmtologo
 bin2c
 unifdef
-- 
2.35.3

