From 89c416b443b432a1144e055d6cc4327c71ccfbce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:20:53 +0800
Subject: [PATCH] rk_sysrq: fix for v2.6.36

---
 drivers/char/rk_sysrq.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/char/rk_sysrq.c b/drivers/char/rk_sysrq.c
index af0933555881..eb741935f57c 100755
--- a/drivers/char/rk_sysrq.c
+++ b/drivers/char/rk_sysrq.c
@@ -38,6 +38,7 @@
 #include <linux/workqueue.h>
 #include <linux/hrtimer.h>
 #include <linux/oom.h>
+#include <linux/version.h>
 
 #include <asm/ptrace.h>
 #include <asm/irq_regs.h>
@@ -744,7 +745,11 @@ void __rk_handle_sysrq(int key, struct tty_struct *tty, int check_mask)
 		if (!check_mask || rk_sysrq_on_mask(op_p->enable_mask)) {
 			printk("%s\n", op_p->action_msg);
 			console_loglevel = orig_log_level;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 36))
+			op_p->handler(key);
+#else
 			op_p->handler(key, tty);
+#endif
 		} else {
 			printk("This sysrq operation is disabled.\n");
 		}
-- 
2.35.3

