From 43572daba8a2bd7185410f0ee201408f4086a552 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Fri, 14 May 2010 14:40:14 +0000
Subject: [PATCH] rk2818_sdmmc.x --> rk2818_sdmmcx

---
 arch/arm/mach-rk2818/clock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index 37f291881305..237f36253903 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -908,7 +908,7 @@ static struct clk_lookup clks[] = {
 	CLK1(lcdc),
 	CLK1(vip),
 	CLK1(i2s),
-	CLK("rk2818_sdmmc.0", "sdmmc", &sdmmc0_clk),
+	CLK("rk2818_sdmmc0", "sdmmc", &sdmmc0_clk),
 	CLK1(ebrom),
 	CLK1(gpio0),
 	CLK1(gpio1),
@@ -925,7 +925,7 @@ static struct clk_lookup clks[] = {
 	CLK1(lsadc),
 	CLK("rk2818_serial.2", "uart", &uart2_clk),
 	CLK("rk2818_serial.3", "uart", &uart3_clk),
-	CLK("rk2818_sdmmc.1", "sdmmc", &sdmmc1_clk),
+	CLK("rk2818_sdmmc1", "sdmmc", &sdmmc1_clk),
 
 	CLK1(hsadc),
 	CLK1(sdram_common),
-- 
2.35.3

