From 95b4ccbaebc8ea0409562c6510330234b4c6965a Mon Sep 17 00:00:00 2001
From: chm <chm@rock-chips.com>
Date: Fri, 23 Nov 2012 15:44:51 +0800
Subject: [PATCH] vpu_service: remove unnecessary print

---
 arch/arm/plat-rk/vpu_service.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/plat-rk/vpu_service.c b/arch/arm/plat-rk/vpu_service.c
index bda18fe87268..1ad26aef751b 100644
--- a/arch/arm/plat-rk/vpu_service.c
+++ b/arch/arm/plat-rk/vpu_service.c
@@ -584,7 +584,6 @@ static void reg_copy_to_hw(vpu_reg *reg)
 
 		dst[VPU_REG_DEC_GATE] = src[VPU_REG_DEC_GATE] | VPU_REG_DEC_GATE_BIT;
 		dst[VPU_REG_EN_DEC]   = src[VPU_REG_EN_DEC];
-		printk("dec\n");
 	} break;
 	case VPU_PP : {
 		u32 *dst = (u32 *)dec_dev.hwregs + PP_INTERRUPT_REGISTER;
@@ -598,7 +597,6 @@ static void reg_copy_to_hw(vpu_reg *reg)
 		dsb();
 
 		dst[VPU_REG_EN_PP] = src[VPU_REG_EN_PP];
-		printk("pp\n");
 	} break;
 	case VPU_DEC_PP : {
 		u32 *dst = (u32 *)dec_dev.hwregs;
@@ -614,7 +612,6 @@ static void reg_copy_to_hw(vpu_reg *reg)
 		dst[VPU_REG_DEC_PP_GATE] = src[VPU_REG_DEC_PP_GATE] | VPU_REG_PP_GATE_BIT;
 		dst[VPU_REG_DEC_GATE]	 = src[VPU_REG_DEC_GATE]    | VPU_REG_DEC_GATE_BIT;
 		dst[VPU_REG_EN_DEC]	 = src[VPU_REG_EN_DEC];
-		printk("dec_pp\n");
 	} break;
 	default : {
 		pr_err("error: unsupport session type %d", reg->type);
-- 
2.35.3

