From 9f1512e62f5eb7a1f43c540b905d56a0ee2ad4c0 Mon Sep 17 00:00:00 2001
From: ywj <ywj@rock-chips.com>
Date: Wed, 9 Apr 2014 09:34:05 +0800
Subject: [PATCH] enable mma8452

---
 arch/arm/boot/dts/rk3288-p977.dts | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index b8c5fd1b46a9..e066e85adfa3 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -312,7 +312,6 @@
 		irq_enable = <1>;
 		poll_delay_ms = <30>;
 		layout = <8>;
-		status = "disabled";
 	};
 
 	sensor@0d {
@@ -323,7 +322,6 @@
 		irq_enable = <1>;
 		poll_delay_ms = <30>;
 		layout = <1>;
-		status = "disabled";
 	};
 
 	sensor@10 {
-- 
2.35.3

