From c3898f14d096051a0835430b69667ed727bd9717 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Thu, 11 Oct 2012 18:22:22 +0800
Subject: [PATCH] rk3066b m701: add m701 board support gpu 600MHZ freq

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index 81ea5496f783..f767d30d3805 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -1682,6 +1682,7 @@ static struct dvfs_arm_table dvfs_cpu_logic_table[] = {
 static struct cpufreq_frequency_table dvfs_gpu_table[] = {
 	{.frequency = 266 * 1000,	.index = 1000 * 1000},
 	{.frequency = 400 * 1000,	.index = 1000 * 1000},
+	{.frequency = 600 * 1000,	.index = 1200 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
 
-- 
2.35.3

