From b2a36f49c7ab480fb309063811bfaddc90325905 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 27 Sep 2011 15:08:57 +0800
Subject: [PATCH] i2c drivers: add some warning if scl_rate < 10K || scl_rate >
 400K

---
 drivers/i2c/busses/i2c-rk29.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index 5049c1acb5d9..739d7a1ef7e4 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -542,8 +542,11 @@ static int rk29_i2c_xfer(struct i2c_adapter *adap,
 	// 400k > scl_rate > 10k
 	if(msgs[0].scl_rate <= 400000 && msgs[0].scl_rate >= 10000)
 		i2c->scl_rate = msgs[0].scl_rate;
-	else if(msgs[0].scl_rate > 400000)
+	else if(msgs[0].scl_rate > 400000){
+		dev_info(i2c->dev, "Warning: msg[0].scl_rate( = %dKhz) is too high!",
+			msgs[0].scl_rate/1000);
 		i2c->scl_rate = 400000;	
+	}
 	else{
 		dev_info(i2c->dev, "Warning: msg[0].scl_rate( = %dKhz) is too low!",
 			msgs[0].scl_rate/1000);
-- 
2.35.3

