From 1cf2069b389512bcfbeafc4c58c5be98be64dbfe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 4 Jan 2013 14:05:28 +0800
Subject: [PATCH] rk30: pm: keep pwm23 when first gating

---
 arch/arm/mach-rk30/pm.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index a73e57a2f864..9ab51da4b473 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -616,8 +616,9 @@ static int rk30_pm_enter(suspend_state_t state)
 			  , clkgt_regs[5], CRU_CLKGATES_CON(5), 0xffff);
 	gate_save_soc_clk(0, clkgt_regs[6], CRU_CLKGATES_CON(6), 0xffff);
 	gate_save_soc_clk(0
-			|(1 << CLK_GATE_PCLK_PWM23%16)
-			, clkgt_regs[7], CRU_CLKGATES_CON(7), 0xffff);
+			  | (1 << CLK_GATE_PCLK_PWM01 % 16)
+			  | (1 << CLK_GATE_PCLK_PWM23 % 16)
+			  , clkgt_regs[7], CRU_CLKGATES_CON(7), 0xffff);
 	gate_save_soc_clk(0 , clkgt_regs[8], CRU_CLKGATES_CON(8), 0x01ff);
 	gate_save_soc_clk(0
 			  | (1 << CLK_GATE_CLK_L2C % 16)
-- 
2.35.3

