From 9223f55d824199d73dfef5433c796cb7d7569433 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Fri, 25 Jan 2013 08:59:34 +0800
Subject: [PATCH] rk2928 codec: reset codec when probe.

---
 sound/soc/codecs/rk2928_codec.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/codecs/rk2928_codec.c b/sound/soc/codecs/rk2928_codec.c
index 6406c4166177..a9feb0aa7eab 100755
--- a/sound/soc/codecs/rk2928_codec.c
+++ b/sound/soc/codecs/rk2928_codec.c
@@ -417,6 +417,9 @@ static int rk2928_probe(struct snd_soc_codec *codec)
 			gpio_direction_output(rk2928_data.spkctl, GPIO_LOW);
 	}
 
+	//Reset Codec
+	rk2929_codec_reset();
+
 	// Select SDI input from internal audio codec
 	writel(0x04000400, RK2928_GRF_BASE + GRF_SOC_CON0);
 	
-- 
2.35.3

