From 33dee0532c7652d6759949f562857733f13b8de6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Tue, 7 May 2013 16:42:36 +0800
Subject: [PATCH] charging-displasy: when mode_normal and mode_charge

---
 drivers/power/rk29_charger_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/rk29_charger_display.c b/drivers/power/rk29_charger_display.c
index 7d2253b29800..f384a9745dea 100755
--- a/drivers/power/rk29_charger_display.c
+++ b/drivers/power/rk29_charger_display.c
@@ -166,7 +166,7 @@ static int  __init start_charge_logo_display(void)
 
 	if(val_status.intval == POWER_SUPPLY_STATUS_CHARGING)
 	{
-		if ((board_boot_mode() != BOOT_MODE_REBOOT) || (val_capacity.intval <= pwr_on_thrsd))  //do not enter power on charge mode when soft  reset
+		if ((board_boot_mode() == BOOT_MODE_NORMAL) ||(board_boot_mode() == BOOT_MODE_CHARGE)|| (val_capacity.intval <= pwr_on_thrsd))  //do not enter power on charge mode when soft  reset
 	    {			
 			add_bootmode_charger_to_cmdline();
 			//boot_mode_init("charge");
-- 
2.35.3

