From 3f08a6de5ef7081126d8d7460d4928ffbcb1c2cd Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Thu, 30 Dec 2010 14:12:55 +0800
Subject: [PATCH] not enable win0 when fb1_set_par

---
 drivers/video/rk29_fb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index d38a7bdf09ab..f3fc1e44addd 100644
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -918,8 +918,8 @@ static int win0_set_par(struct fb_info *info)
     LcdWrReg(inf, WIN0_YRGB_MST, y_addr);
     LcdWrReg(inf, WIN0_CBR_MST, uv_addr);
 
-    LcdMskReg(inf, SYS_CONFIG, m_W0_ENABLE | m_W0_FORMAT | m_W0_AXI_OUTSTANDING2,
-                    v_W0_ENABLE(par->addr_seted) | v_W0_FORMAT(par->format) | v_W0_AXI_OUTSTANDING2(inf->video_mode==0));
+    LcdMskReg(inf, SYS_CONFIG,  m_W0_FORMAT | m_W0_AXI_OUTSTANDING2,
+           v_W0_FORMAT(par->format) | v_W0_AXI_OUTSTANDING2(inf->video_mode==0));
 
     LcdMskReg(inf, WIN0_VIR, m_WORDLO | m_WORDHI, v_VIRWIDTH(xvir) | v_VIRHEIGHT((yvir)) );
     LcdMskReg(inf, WIN0_ACT_INFO, m_WORDLO | m_WORDHI, v_WORDLO(xact) | v_WORDHI(yact));
-- 
2.35.3

