From 835766d5fe2578f6fa919afd4958f3d9ff0e6d5b Mon Sep 17 00:00:00 2001
From: Huibin Hong <huibin.hong@rock-chips.com>
Date: Thu, 19 Mar 2015 17:53:14 +0800
Subject: [PATCH] spi: add CONFIG_NEED_SG_DMA_LENGTH for dma_length

Signed-off-by: Huibin Hong <huibin.hong@rock-chips.com>
---
 drivers/spi/spi-rockchip-dma.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-rockchip-dma.c b/drivers/spi/spi-rockchip-dma.c
index 7c224da9c129..6952201406cd 100755
--- a/drivers/spi/spi-rockchip-dma.c
+++ b/drivers/spi/spi-rockchip-dma.c
@@ -265,8 +265,11 @@ static int mid_spi_dma_transfer(struct dw_spi *dws, int cs_change)
 		
 		memset(&dws->tx_sgl, 0, sizeof(dws->tx_sgl));
 		dws->tx_sgl.dma_address = dws->tx_dma;
+#ifdef CONFIG_NEED_SG_DMA_LENGTH
+		dws->tx_sgl.dma_length = dws->len;
+#else
 		dws->tx_sgl.length = dws->len;
-
+#endif
 		txdesc = dmaengine_prep_slave_sg(txchan,
 					&dws->tx_sgl,
 					1,
@@ -298,7 +301,11 @@ static int mid_spi_dma_transfer(struct dw_spi *dws, int cs_change)
 
 		memset(&dws->rx_sgl, 0, sizeof(dws->rx_sgl));
 		dws->rx_sgl.dma_address = dws->rx_dma;
-		dws->rx_sgl.length = dws->len;				
+#ifdef CONFIG_NEED_SG_DMA_LENGTH
+		dws->rx_sgl.dma_length = dws->len;				
+#else
+		dws->rx_sgl.length = dws->len;
+#endif
 
 		rxdesc = dmaengine_prep_slave_sg(rxchan,
 					&dws->rx_sgl,
-- 
2.35.3

