From 5a8c294436e4958e2ccfe75fec63088d48c0a6cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Thu, 16 Sep 2010 01:33:42 -0700
Subject: [PATCH] add delay after pll change in pm.c, fix sdk can not be resume

---
 arch/arm/mach-rk2818/pm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk2818/pm.c b/arch/arm/mach-rk2818/pm.c
index a1612fac7cff..1a39de4f905a 100755
--- a/arch/arm/mach-rk2818/pm.c
+++ b/arch/arm/mach-rk2818/pm.c
@@ -574,6 +574,8 @@ static int __tcmfunc rk2818_tcm_idle(void)
 
 	scu_writel(scu_apll, SCU_APLL_CON); // powerup
 	scu_writel(scu_clksel0 & (~3), SCU_CLKSEL0_CON);
+	
+	tcm_udelay(2, 24);
 
 	unit = 7200;  /* 24m,0.3ms , 24*300*/
 	while (unit-- > 0) {
-- 
2.35.3

