From 6ddfd32578e5d43015c6a3bd31cdcaa05f86f251 Mon Sep 17 00:00:00 2001
From: zhanghao <zhanghao@rock-chips.com>
Date: Wed, 8 Feb 2012 14:38:29 +0800
Subject: [PATCH] ili2102 driver : update for td8801_v2

---
 drivers/input/touchscreen/ili2102_ts.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/input/touchscreen/ili2102_ts.c b/drivers/input/touchscreen/ili2102_ts.c
index e043d1b5f348..fd64405c3e62 100755
--- a/drivers/input/touchscreen/ili2102_ts.c
+++ b/drivers/input/touchscreen/ili2102_ts.c
@@ -399,6 +399,11 @@ static void ili2102_ts_work_func(struct work_struct *work)
 					x = g_x[i];
 					y = g_y[i];
 				}
+				#ifdef CONFIG_MACH_RK29_TD8801_V2
+				if( y >=80 ) y-=80;
+				if( x >= 50 ) x-=50;
+				#endif
+
 				g_x[i] = x;
 				g_y[i] = y;	
 				
-- 
2.35.3

