From 19b284b24e0079bcbc5b757a0958af63c7cc18f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 22 Aug 2014 16:32:40 +0800
Subject: [PATCH] rk32:rk808:modify kconfig for rk808_mfd

---
 drivers/mfd/Kconfig | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/Kconfig b/drivers/mfd/Kconfig
index d57c5cfa7c59..9fda33029457 100755
--- a/drivers/mfd/Kconfig
+++ b/drivers/mfd/Kconfig
@@ -387,9 +387,10 @@ config MFD_MAX8998
 
 config MFD_RK808
 	bool "RK808 Power Management chip"
-	depends on I2C=y 
+	depends on I2C=y
+	select REGMAP_I2C 
 	select MFD_CORE
-	select RTC_RK808
+	select RK808_RTC
 	help
 	  if you say yes here you get support for the RK808 series of
 	  Power Management chips.
-- 
2.35.3

