From 0c8c526cce7dec880e6e66e3106237f96f18aadf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 12 Jun 2010 02:18:37 +0000
Subject: [PATCH] call enable_irq_wake on play/wakeup pin

---
 drivers/input/keyboard/rk2818_adckey.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/keyboard/rk2818_adckey.c b/drivers/input/keyboard/rk2818_adckey.c
index 45bc7ccb24eb..a921c5264f20 100644
--- a/drivers/input/keyboard/rk2818_adckey.c
+++ b/drivers/input/keyboard/rk2818_adckey.c
@@ -378,6 +378,8 @@ static int __devinit rk28_adckey_probe(struct platform_device *pdev)
 	}
 #endif
 
+	enable_irq_wake(gpio_to_irq(KEY_PLAYON_PIN)); // so play/wakeup key can wake up system
+
 #if 0
 	error = gpio_direction_input(KEY_PLAYON_PIN);
 	if (error) 
-- 
2.35.3

