From 0ba0b758e918c8a15a46db68a8e9d285f2565619 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Thu, 27 May 2010 12:09:23 +0000
Subject: [PATCH] mod sdmmc max_req_size = 1023 * 4

---
 drivers/mmc/host/rk2818-sdmmc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/rk2818-sdmmc.c b/drivers/mmc/host/rk2818-sdmmc.c
index c2c6f5f135a4..b2ed1ed97cab 100644
--- a/drivers/mmc/host/rk2818-sdmmc.c
+++ b/drivers/mmc/host/rk2818-sdmmc.c
@@ -1282,8 +1282,8 @@ static int rk2818_sdmmc_probe(struct platform_device *pdev)
 	
 	mmc->max_phys_segs = 64;
 	mmc->max_hw_segs = 64;
-	mmc->max_blk_size = 4095;
-	mmc->max_blk_count = 512;
+	mmc->max_blk_size = /*4095*/1023;
+	mmc->max_blk_count = /*512*/4;
 	mmc->max_req_size = mmc->max_blk_size * mmc->max_blk_count;
 	mmc->max_seg_size = mmc->max_req_size;
 
-- 
2.35.3

