From 40e475d997fa46cec482a632adfdac8777b58dd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Tue, 28 Aug 2012 15:17:48 +0800
Subject: [PATCH] rk3066B i2s: remove rk3066B i2s2

---
 sound/soc/rk29/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/Kconfig b/sound/soc/rk29/Kconfig
index d54108faf554..70a860662eb0 100755
--- a/sound/soc/rk29/Kconfig
+++ b/sound/soc/rk29/Kconfig
@@ -26,7 +26,7 @@ config SND_RK29_SOC_I2S_2CH
 config SND_RK_SOC_I2S2_2CH
 	bool "Soc RK29 I2S 2 Channel support(I2S2)"
 	default n
-	depends on SND_RK29_SOC_I2S && ARCH_RK30 && !ARCH_RK2928
+	depends on SND_RK29_SOC_I2S && ARCH_RK30 && !ARCH_RK2928 && !ARCH_RK3066B
 	help
 		This supports the use of the 2 Channel I2S2 interface on rk30 processors.
 
-- 
2.35.3

