From 94843abd8ddc505301176fa379da8c02e8ecbb0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Mon, 17 May 2010 11:40:08 +0000
Subject: [PATCH] mod i2c_transfer

---
 drivers/rtc/rtc-HYM8563.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-HYM8563.c b/drivers/rtc/rtc-HYM8563.c
index a9434d98e395..095127fa3607 100644
--- a/drivers/rtc/rtc-HYM8563.c
+++ b/drivers/rtc/rtc-HYM8563.c
@@ -41,7 +41,7 @@ static int hym8563_i2c_read_regs(struct i2c_client *client, u8 reg, u8 buf[], un
 	msgs[1].addr = client->addr;
 	msgs[1].flags = client->flags | I2C_M_RD;
 	msgs[1].len = len;
-	msgs[1].scl_rate = 4000 * 1000; 
+	msgs[1].scl_rate = 400 * 1000; 
 
 	ret = i2c_transfer(adap, msgs, 2);
 	/*
-- 
2.35.3

