From 717286ca99f00e2efcd8c69757d067b67cc24e0a Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 8 Apr 2013 10:24:08 +0800
Subject: [PATCH] rk3028: add dvfs voltage difference table

---
 arch/arm/mach-rk30/dvfs.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index deb38384d365..2187dc9df0c8 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -84,8 +84,7 @@ static int g_arm_high_logic = 150 * 1000;
 static int g_logic_high_arm = 100 * 1000;
 #endif
 
-
-#ifdef CONFIG_SOC_RK3168
+#if defined(CONFIG_SOC_RK3168) || defined(CONFIG_SOC_RK3028)
 static struct cpufreq_frequency_table arm_high_logic_table[] = {
         {.frequency = 1416 * DVFS_KHZ, .index = 0 * DVFS_MV},
         {.frequency = 1608 * DVFS_KHZ, .index = 0 * DVFS_MV},
@@ -97,7 +96,7 @@ static struct cpufreq_frequency_table logic_high_arm_table[] = {
         {.frequency = 1608 * DVFS_KHZ, .index = 50 * DVFS_MV},
         {.frequency = CPUFREQ_TABLE_END},
 };
-#else
+#elif defined(CONFIG_SOC_RK3066)
 static struct cpufreq_frequency_table arm_high_logic_table[] = {
         {.frequency = 1416 * DVFS_KHZ, .index = 50 * DVFS_MV},
         {.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
@@ -108,8 +107,19 @@ static struct cpufreq_frequency_table logic_high_arm_table[] = {
         {.frequency = 1416 * DVFS_KHZ, .index = 150 * DVFS_MV},
         {.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
 };
-#endif
+#else
+static struct cpufreq_frequency_table arm_high_logic_table[] = {
+        {.frequency = 1416 * DVFS_KHZ, .index = 0 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 0 * DVFS_MV},
+        {.frequency = CPUFREQ_TABLE_END},
+};
 
+static struct cpufreq_frequency_table logic_high_arm_table[] = {
+        {.frequency = 1008 * DVFS_KHZ, .index = 50 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 50 * DVFS_MV},
+        {.frequency = CPUFREQ_TABLE_END},
+};
+#endif
 
 int get_arm_logic_limit(unsigned long arm_rate, int *arm_high_logic, int *logic_high_arm)
 {
-- 
2.35.3

