From 93950592cde736188d5a62629582a5952875a8ff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 20 Mar 2014 20:24:26 +0800
Subject: [PATCH] ARM: rockchip: rk3288 disable cpu and timer temporarily

---
 arch/arm/boot/dts/rk3288.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 709923169779..d918aba09d06 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -41,6 +41,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x500>;
 		};
+/*
 		cpu@1 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a15";
@@ -56,6 +57,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x503>;
 		};
+*/
 	};
 
 	gic: interrupt-controller@ffc01000 {
@@ -174,6 +176,7 @@
 		map-exec;
 	};
 
+/*
 	timer {
 		compatible = "arm,armv7-timer";
 		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_HIGH)>,
@@ -182,6 +185,7 @@
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_HIGH)>;
 		clock-frequency = <24000000>;
 	};
+*/
 
 	timer@ff810000 {
 		compatible = "rockchip,timer";
-- 
2.35.3

