From c653a9c6da9bb8c2e95a8b5127de602f2aaccc6c Mon Sep 17 00:00:00 2001
From: wlf <wulf@rock-chips.com>
Date: Tue, 19 Mar 2013 20:51:14 +0800
Subject: [PATCH] USB: fix usb_get_configuration bug

---
 drivers/usb/core/config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/core/config.c b/drivers/usb/core/config.c
index 26678cadfb21..86167b585f84 100644
--- a/drivers/usb/core/config.c
+++ b/drivers/usb/core/config.c
@@ -704,7 +704,7 @@ int usb_get_configuration(struct usb_device *dev)
 			    "descriptor/%s: %d\n", cfgno, "start", result);
 			dev_err(ddev, "chopping to %d config(s)\n", cfgno);
 			dev->descriptor.bNumConfigurations = cfgno;
-			break;
+			goto err;
 		} else if (result < 4) {
 			dev_err(ddev, "config index %d descriptor too short "
 			    "(expected %i, got %i)\n", cfgno,
-- 
2.35.3

