From 15efdc6a78b409ac8e21061f42ef7eb45012c763 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Wed, 30 Jul 2014 13:52:29 +0800
Subject: [PATCH] rk312x dts: add hdmi in dts.

---
 arch/arm/boot/dts/rk312x.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index f4afa4422c21..ef7d1c9a7b8e 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -571,6 +571,19 @@
 		status = "disabled";
 	};
 
+	hdmi: hdmi@20034000 {
+		compatible = "rockchip,rk3036-hdmi";
+		reg = <0x20034000 0x4000>;
+		interrupts = <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>;
+		rockchip,hdmi_lcdc_source = <0>;
+		pinctrl-names = "default", "gpio";
+		pinctrl-0 = <&hdmi_cec &hdmi_sda &hdmi_scl &hdmi_hpd>;
+		pinctrl-1 = <&hdmi_gpio>;
+		clocks = <&clk_gates3 8>;
+		clock-names = "pclk_hdmi";
+		status = "disabled";
+	};
+
         vpu: vpu_service@10104000 {
 		compatible = "vpu_service";
 		reg = <0x10104000 0x800>;
-- 
2.35.3

