From ebc5e698fd5ee2664ab538770f055d804be4524e Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 21 Feb 2013 09:12:01 +0800
Subject: [PATCH] iomux: default mode: uart2&uart3: add ctsn&rtsn

---
 arch/arm/plat-rk/iomux.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/plat-rk/iomux.c b/arch/arm/plat-rk/iomux.c
index d2617ea050de..89b354754c56 100644
--- a/arch/arm/plat-rk/iomux.c
+++ b/arch/arm/plat-rk/iomux.c
@@ -133,10 +133,16 @@ static unsigned int default_mode[] = {
 
 	#if defined(CONFIG_UART2_RK29) || (CONFIG_RK_DEBUG_UART == 2)
         UART2_SIN, UART2_SOUT,
+	#ifdef CONFIG_UART2_CTS_RTS_RK29
+	UART2_CTSN, UART2_RTSN,
+	#endif
 	#endif
 
 	#if defined(CONFIG_UART3_RK29) || (CONFIG_RK_DEBUG_UART == 3)
         UART3_SIN, UART3_SOUT,
+	#ifdef CONFIG_UART3_CTS_RTS_RK29
+	UART3_CTSN, UART3_RTSN,
+	#endif
 	#endif
 
 	#ifdef CONFIG_SPIM0_RK29
-- 
2.35.3

