From 662295b615b074fc3a888d89cd0d647f9fe855ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:26:03 +0800
Subject: [PATCH] bcm4329: add WEXT_PRIV, fix for v2.6.36

---
 drivers/net/wireless/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/Kconfig b/drivers/net/wireless/Kconfig
index 3dd812b165e7..2b513b52e8a5 100644
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -36,6 +36,7 @@ choice
 	config BCM4329
 	  depends on WLAN_80211 && MMC
 	  select WIRELESS_EXT
+	  select WEXT_PRIV
 	  select IEEE80211
 	  select FW_LOADER
 	  bool "Broadcom BCM4329 WiFi/BT Combo SDIO"
-- 
2.35.3

