From 0db5b3e04a6a6e8b18fe8258b20d9bb9824b6e72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Jan 2013 11:51:08 +0800
Subject: [PATCH] rk30_camera: fix gcc warning

---
 drivers/media/video/rk30_camera.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/video/rk30_camera.c b/drivers/media/video/rk30_camera.c
index 909c7c396a92..d33b78e32766 100755
--- a/drivers/media/video/rk30_camera.c
+++ b/drivers/media/video/rk30_camera.c
@@ -1506,7 +1506,7 @@ static void rk30_camera_request_reserve_mem(void)
 {
 #ifdef CONFIG_VIDEO_RK29_WORK_IPP
         rk_camera_platform_data.meminfo.vbase = rk_camera_platform_data.meminfo_cif1.vbase = NULL;
-    #if defined(CONFIG_VIDEO_RKCIF_WORK_SIMUL_OFF) || ((RK_SUPPORT_CIF0 && RK_SUPPORT_CIF1) == false)
+    #if defined(CONFIG_VIDEO_RKCIF_WORK_SIMUL_OFF) || ((RK_SUPPORT_CIF0 && RK_SUPPORT_CIF1) == 0)
         rk_camera_platform_data.meminfo.name = "camera_ipp_mem";
         rk_camera_platform_data.meminfo.start = board_mem_reserve_add("camera_ipp_mem",PMEM_CAMIPP_NECESSARY);
         rk_camera_platform_data.meminfo.size= PMEM_CAMIPP_NECESSARY;
-- 
2.35.3

