From 05f87d6627e8be4931e1118ed421ac179aa4f7b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=98=8A?= <zhanghao@rock-chips.com>
Date: Sat, 19 Nov 2011 16:36:29 +0800
Subject: [PATCH] wm831x_bl : fix param error

---
 drivers/video/backlight/wm831x_bl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/backlight/wm831x_bl.c b/drivers/video/backlight/wm831x_bl.c
index d94845185fee..ca6d6b9cd2fc 100755
--- a/drivers/video/backlight/wm831x_bl.c
+++ b/drivers/video/backlight/wm831x_bl.c
@@ -270,7 +270,7 @@ static int wm831x_backlight_probe(struct platform_device *pdev)
 
 	props.max_brightness = max_isel;
 	bl = backlight_device_register("wm831x", &pdev->dev, data,
-				       &wm831x_backlight_ops,&props);
+				       &wm831x_backlight_ops,NULL);
 	if (IS_ERR(bl)) {
 		dev_err(&pdev->dev, "failed to register backlight\n");
 		kfree(data);
-- 
2.35.3

