From 968caf254a41658ff42fdcb75ab12e30985d32ef Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Tue, 1 Apr 2014 17:47:53 +0800
Subject: [PATCH] i2s: add 3288 gate control

---
 sound/soc/rockchip/rk30_i2s.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk30_i2s.c b/sound/soc/rockchip/rk30_i2s.c
index f2d466cd2231..dd4a6fa89f5f 100755
--- a/sound/soc/rockchip/rk30_i2s.c
+++ b/sound/soc/rockchip/rk30_i2s.c
@@ -59,7 +59,7 @@ struct rk30_i2s_info {
 
 	struct clk *i2s_clk;// i2s clk ,is bclk lrck
 	struct clk *i2s_mclk;//i2s mclk,rk32xx can different i2s clk.
-
+	struct clk *i2s_hclk;
 	struct snd_dmaengine_dai_dma_data capture_dma_data;
 	struct snd_dmaengine_dai_dma_data playback_dma_data;
 
@@ -549,6 +549,14 @@ static int rockchip_i2s_probe(struct platform_device *pdev)
 	}
 
 	rk30_i2s = i2s;
+
+	i2s->i2s_hclk = clk_get(&pdev->dev, "i2s_hclk");
+	if(IS_ERR(i2s->i2s_hclk) ) {
+                dev_err(&pdev->dev, "get i2s_hclk failed.\n");
+        } else{
+		clk_prepare_enable(i2s->i2s_hclk);
+	}
+
 	i2s->i2s_clk= clk_get(&pdev->dev, "i2s_clk");
 	if (IS_ERR(i2s->i2s_clk)) {
 		dev_err(&pdev->dev, "Can't retrieve i2s clock\n");
-- 
2.35.3

