From 1c742ff5def931307b3d356ad7c413375f69c103 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 23 Oct 2012 10:12:01 +0800
Subject: [PATCH] rk2928: enable ARM ERRATA 775420 workaround

---
 arch/arm/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 7dd6b179c0e1..6c2522812c15 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -879,6 +879,7 @@ config ARCH_RK2928
 	select RK_PL330_DMA
 	select MIGHT_HAVE_CACHE_L2X0
 	select ARM_ERRATA_754322
+	select ARM_ERRATA_775420
 	help
 	  Support for Rockchip's RK2928 SoCs.
 
-- 
2.35.3

