From d93c580979f1a0147c718ada1fa1ce4028716ed8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 2 Apr 2014 14:21:53 +0800
Subject: [PATCH] bq24296_charger: fix probe failed bug, which cause resume
 panic

---
 drivers/power/bq24296_charger.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/power/bq24296_charger.c b/drivers/power/bq24296_charger.c
index 7312c87c06f9..9bdc35e8877c 100755
--- a/drivers/power/bq24296_charger.c
+++ b/drivers/power/bq24296_charger.c
@@ -523,7 +523,7 @@ static int bq24296_battery_probe(struct i2c_client *client,const struct i2c_devi
 	u8 retval = 0;
 	struct bq24296_board *pdev;
 	struct device_node *bq24296_node;
-	int ret=0;
+	int ret = -EINVAL;
 	
 	 DBG("%s,line=%d\n", __func__,__LINE__);
 	 
@@ -535,7 +535,7 @@ static int bq24296_battery_probe(struct i2c_client *client,const struct i2c_devi
 	di = devm_kzalloc(&client->dev,sizeof(*di), GFP_KERNEL);
 	if (!di) {
 		dev_err(&client->dev, "failed to allocate device info data\n");
-		retval = -ENOMEM;
+		ret = -ENOMEM;
 		goto batt_failed_2;
 	}
 	i2c_set_clientdata(client, di);
@@ -595,7 +595,7 @@ static int bq24296_battery_probe(struct i2c_client *client,const struct i2c_devi
 err_chgirq_failed:
 	free_irq(gpio_to_irq(pdev->chg_irq_pin), NULL);
 batt_failed_2:
-	return retval;
+	return ret;
 }
 
 static void bq24296_battery_shutdown(struct i2c_client *client)
-- 
2.35.3

