From 61acb3872d66235123fed0486dbaf60dd1460a8b Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 28 Feb 2013 17:04:08 +0800
Subject: [PATCH] rk3168_86v: modify the resolutation of tp

---
 drivers/input/touchscreen/86v_gt811_ts.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/touchscreen/86v_gt811_ts.c b/drivers/input/touchscreen/86v_gt811_ts.c
index 92e94a280b7b..cd72af55546d 100755
--- a/drivers/input/touchscreen/86v_gt811_ts.c
+++ b/drivers/input/touchscreen/86v_gt811_ts.c
@@ -839,7 +839,11 @@ static int goodix_ts_probe(struct i2c_client *client, const struct i2c_device_id
 	input_mt_init_slots(ts->input_dev, ts->max_touch_num);
 	input_set_abs_params(ts->input_dev, ABS_MT_TOUCH_MAJOR, 0, 255, 0, 0);
 	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_X, 0, ts->abs_x_max, 0, 0);
+#ifdef CONFIG_MACH_RK3168_86V_OLD
 	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_Y, 0, 415, 0, 0);	
+#else
+	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_Y, 0, ts->abs_y_max, 0, 0);
+#endif
 	//printk("\n\nhjc:%s,x_max=%d,y_max=%d\n",__func__,ts->abs_x_max,ts->abs_y_max);
 	
 	
-- 
2.35.3

