From e562cfaebbb43868b278e35d8d37a6b69a0e2d8e Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 14 Aug 2012 11:12:33 +0800
Subject: [PATCH] rkfbsys:fix bug in set_dsp_lut

---
 drivers/video/rockchip/rkfb_sysfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index b0588c6e7b5e..97401d2a49d8 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -263,6 +263,7 @@ static ssize_t set_dsp_lut(struct device *dev,struct device_attribute *attr,
 	for(i=0;i<256;i++)
 	{
 		temp = simple_strtoul(start,NULL,10);
+		dsp_lut[i] = temp;
 		do
 		{
 			start++;
-- 
2.35.3

