From d935e3b46cb3756f96c1a79175dd9e39bf118dc8 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Tue, 30 Jul 2013 10:08:41 +0800
Subject: [PATCH] fix ak8963 building error

---
 drivers/input/sensors/compass/ak8963.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/compass/ak8963.c b/drivers/input/sensors/compass/ak8963.c
index 8c9530c6baa8..0f78192682fa 100755
--- a/drivers/input/sensors/compass/ak8963.c
+++ b/drivers/input/sensors/compass/ak8963.c
@@ -641,7 +641,7 @@ static long compass_dev_ioctl(struct file *file,
 		break;
 	case ECS_IOCTL_GET_OUTBIT:
 		DBG("%s:ECS_IOCTL_GET_OUTBIT start\n",__func__);
-		outbit = sensor->pdata->outbit;
+		outbit = 1; //sensor->pdata->outbit;
 		break;
 	case ECS_IOCTL_RESET:
 		DBG("%s:ECS_IOCTL_RESET start\n",__func__);
-- 
2.35.3

