From 609161946706962e14e46f307cac917e2b902015 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Thu, 11 Sep 2014 18:18:34 -0700
Subject: [PATCH] rk312x: enable jtag clk by rockchip_jtag_enabled
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 陈亮 <cl@rock-chips.com>
---
 arch/arm/mach-rockchip/rk312x.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk312x.c b/arch/arm/mach-rockchip/rk312x.c
index 5308c528c2b7..d9e08033a7b0 100755
--- a/arch/arm/mach-rockchip/rk312x.c
+++ b/arch/arm/mach-rockchip/rk312x.c
@@ -358,6 +358,8 @@ static void __init rk312x_init_late(void)
 #ifdef CONFIG_PM
 	rk321x_init_suspend();
 #endif
+	if (rockchip_jtag_enabled)
+		clk_prepare_enable(clk_get_sys(NULL, "clk_jtag"));
 }
 
 static void rk312x_restart(char mode, const char *cmd)
-- 
2.35.3

