From 203dd9c2b1fcb2eef1a7c84caef7e243ad2c7418 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Thu, 15 Sep 2011 14:26:25 +0800
Subject: [PATCH] update fb fix UI scaling up when 1080p HDMI video playing
 next

---
 drivers/video/rk29_fb.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index a646813b1cb1..7eff5a019a7e 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -2220,7 +2220,8 @@ int fb1_release(struct fb_info *info, int user)
         par->par_seted = 0;
         par->addr_seted = 0;
         win1_blank(FB_BLANK_NORMAL, info);
-        fb0_set_par(inf->fb0);
+        if(inf->cur_screen->type != SCREEN_HDMI)
+            fb0_set_par(inf->fb0);
 
         // unmap memory
         info->screen_base = 0;
@@ -2303,10 +2304,10 @@ static int fb1_ioctl(struct fb_info *info, unsigned int cmd, unsigned long arg)
                 ipp_req.src0.fmt = 3;
                 ipp_req.src0.YrgbMst = yuv_phy[0];
                 ipp_req.src0.CbrMst = yuv_phy[1];
-          	  ipp_req.src0.w = var->xres ;
+          	    ipp_req.src0.w = var->xres ;
                 ipp_req.src0.h = var->yres ;
-  		 ipp_req.src_vir_w= (var->xres + 15) & (~15);
-		 ipp_req.dst_vir_w=screen->x_res;
+  		        ipp_req.src_vir_w= (var->xres + 15) & (~15);
+		        ipp_req.dst_vir_w=screen->x_res;
 
                 ipp_req.dst0.fmt = 3;
                 ipp_req.dst0.YrgbMst = inf->fb0->fix.mmio_start + screen->x_res*screen->y_res*2*dstoffset;
-- 
2.35.3

