From eb81634aa54df8f3a51eefeda7665c884d31baac Mon Sep 17 00:00:00 2001
From: chenhui <chenhui@rock-chips.com>
Date: Wed, 27 Mar 2013 12:04:30 +0800
Subject: [PATCH] ARCH_RK3066B default: disable DVFS_WITH_UOC

---
 arch/arm/plat-rk/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 10dd7e30f7e5..8d2f4dbfef76 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -112,7 +112,7 @@ config DVFS
 config DVFS_WITH_UOC
 	bool "Use dvfs with uoc, no voltage difference"
 	depends on DVFS && ARCH_RK3066B
-	default y
+	default n
 
 config RK_CLOCK_PROC
 	bool "/proc/clocks support"
-- 
2.35.3

