From 286811adddebad1b405b2fb89788232b49207f0b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 29 Aug 2011 10:24:26 +0800
Subject: [PATCH] pack-kernel.sh: exclude ddr_reconfig.c

---
 pack-kernel.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index b6cc914406e5..6bb24bf0bd16 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -25,6 +25,8 @@ sound/*rk28*.c
 sound/*rk28*.h
 include/*rk28*
 
+arch/arm/mach-rk29/ddr_reconfig.c
+
 drivers/staging/rk29/vivante/*.c
 drivers/staging/rk29/vivante/*.h
 
-- 
2.35.3

