From 9f9adde57ac6a7d49c556ba03de04416d5680b4c Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 7 May 2014 13:38:05 +0800
Subject: [PATCH] rk fb: do not print timing info when prase_timing

---
 drivers/video/rockchip/rk_fb.c | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 0b92c5539e7e..9074f2e18724 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -386,30 +386,6 @@ int rk_fb_prase_timing_dt(struct device_node *np, struct rk_screen *screen)
 	}
 	dt = display_timings_get(disp_timing, 0);
 	rk_fb_video_mode_from_timing(dt, screen);
-	printk(KERN_DEBUG "dclk:%d\n"
-			 "hactive:%d\n"
-			 "hback_porch:%d\n"
-			 "hfront_porch:%d\n"
-			 "hsync_len:%d\n"
-			 "vactive:%d\n"
-			 "vback_porch:%d\n"
-			 "vfront_porch:%d\n"
-			 "vsync_len:%d\n"
-			 "screen_type:%d\n"
-			 "lvds_format:%d\n"
-			 "face:%d\n",
-			dt->pixelclock.typ,
-			dt->hactive.typ,
-			dt->hback_porch.typ,
-			dt->hfront_porch.typ,
-			dt->hsync_len.typ,
-			dt->vactive.typ,
-			dt->vback_porch.typ,
-			dt->vfront_porch.typ,
-			dt->vsync_len.typ,
-			dt->screen_type,
-			dt->lvds_format,
-			dt->face);
 	return 0;
 
 }
-- 
2.35.3

