From b4db2c3cb7942383307e96376f1ffe510ac40593 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Mon, 21 Jul 2014 15:18:55 +0800
Subject: [PATCH] 312x: adc is 4 channels

---
 drivers/iio/adc/rockchip_adc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/iio/adc/rockchip_adc.c b/drivers/iio/adc/rockchip_adc.c
index c25aa5937f22..256f036da4a9 100755
--- a/drivers/iio/adc/rockchip_adc.c
+++ b/drivers/iio/adc/rockchip_adc.c
@@ -190,6 +190,7 @@ static const struct iio_chan_spec rk_adc_iio_channels[] = {
 	ADC_CHANNEL(0, "adc0"),
 	ADC_CHANNEL(1, "adc1"),
 	ADC_CHANNEL(2, "adc2"),
+	ADC_CHANNEL(3, "adc3"),
 };
 
 static int rk_adc_remove_devices(struct device *dev, void *c)
-- 
2.35.3

