From 0e6ec11f61e1788159fb59d941aeb4c6438e46ff Mon Sep 17 00:00:00 2001
From: xbw <xbw@rock-chips.com>
Date: Sat, 11 Aug 2012 12:51:47 +0800
Subject: [PATCH] sdmmc: mask the SD Ver3.0 supported

---
 drivers/mmc/core/sd.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/sd.c b/drivers/mmc/core/sd.c
index dd1bfdd97782..dcbd5b0e6dea 100755
--- a/drivers/mmc/core/sd.c
+++ b/drivers/mmc/core/sd.c
@@ -306,7 +306,9 @@ static int mmc_read_switch(struct mmc_card *card)
 		goto out;
 	}
 
-	if (card->scr.sda_spec3) {
+    //mask the the SD Ver3.0 support,modifyed by xbw at 2012-8-09
+	//if (card->scr.sda_spec3) {
+	if(0) {
 		card->sw_caps.sd3_bus_mode = status[13];
 
 		/* Find out Driver Strengths supported by the card */
-- 
2.35.3

