From 9344f9d5994122f2d7fe17a01758d064cac23315 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Wed, 11 Jul 2012 18:53:07 +0800
Subject: [PATCH] rk2928: i2c: SZ_4K -> RK2928_XXI2CX_SIZE

---
 arch/arm/mach-rk2928/devices.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-rk2928/devices.c b/arch/arm/mach-rk2928/devices.c
index 112f302cd6a0..6b72f1019092 100755
--- a/arch/arm/mach-rk2928/devices.c
+++ b/arch/arm/mach-rk2928/devices.c
@@ -185,45 +185,45 @@ static void __init rk2928_init_uart(void)
 #ifdef CONFIG_I2C0_CONTROLLER_RK29
 #define I2C0_ADAP_TYPE  I2C_RK29_ADAP
 #define I2C0_START      RK2928_I2C0_PHYS
-#define I2C0_END        RK2928_I2C0_PHYS + SZ_4K - 1
+#define I2C0_END        RK2928_I2C0_PHYS + RK2928_I2C0_SIZE - 1
 #endif
 #ifdef CONFIG_I2C0_CONTROLLER_RK30
 #define I2C0_ADAP_TYPE   I2C_RK30_ADAP
 #define I2C0_START      RK2928_RKI2C0_PHYS
-#define I2C0_END        RK2928_RKI2C0_PHYS + SZ_4K - 1
+#define I2C0_END        RK2928_RKI2C0_PHYS + RK2928_RKI2C0_SIZE - 1
 #endif
 
 #ifdef CONFIG_I2C1_CONTROLLER_RK29
 #define I2C1_ADAP_TYPE  I2C_RK29_ADAP
 #define I2C1_START      RK2928_I2C1_PHYS
-#define I2C1_END        RK2928_I2C1_PHYS + SZ_4K - 1
+#define I2C1_END        RK2928_I2C1_PHYS + RK2928_I2C1_SIZE - 1
 #endif
 #ifdef CONFIG_I2C1_CONTROLLER_RK30
 #define I2C1_ADAP_TYPE   I2C_RK30_ADAP
 #define I2C1_START      RK2928_RKI2C1_PHYS 
-#define I2C1_END        RK2928_RKI2C1_PHYS + SZ_4K - 1
+#define I2C1_END        RK2928_RKI2C1_PHYS + RK2928_RKI2C1_SIZE - 1
 #endif
 
 #ifdef CONFIG_I2C2_CONTROLLER_RK29
 #define I2C2_ADAP_TYPE  I2C_RK29_ADAP
 #define I2C2_START      RK2928_I2C2_PHYS
-#define I2C2_END        RK2928_I2C2_PHYS + SZ_4K - 1
+#define I2C2_END        RK2928_I2C2_PHYS + RK2928_I2C2_SIZE - 1
 #endif
 #ifdef CONFIG_I2C2_CONTROLLER_RK30
 #define I2C2_ADAP_TYPE   I2C_RK30_ADAP
 #define I2C2_START      RK2928_RKI2C2_PHYS
-#define I2C2_END        RK2928_RKI2C2_PHYS + SZ_4K - 1
+#define I2C2_END        RK2928_RKI2C2_PHYS + RK2928_RKI2C2_SIZE - 1
 #endif
 
 #ifdef CONFIG_I2C3_CONTROLLER_RK29
 #define I2C3_ADAP_TYPE  I2C_RK29_ADAP
 #define I2C3_START      RK2928_I2C3_PHYS
-#define I2C3_END        RK2928_I2C3_PHYS + SZ_4K - 1
+#define I2C3_END        RK2928_I2C3_PHYS + RK2928_I2C3_SIZE - 1
 #endif
 #ifdef CONFIG_I2C3_CONTROLLER_RK30
 #define I2C3_ADAP_TYPE   I2C_RK30_ADAP
 #define I2C3_START      RK2928_RKI2C3_PHYS
-#define I2C3_END        RK2928_RKI2C3_PHYS + SZ_4K - 1
+#define I2C3_END        RK2928_RKI2C3_PHYS + RK2928_RKI2C3_SIZE - 1
 #endif
 
 #ifdef CONFIG_I2C0_RK30
-- 
2.35.3

