From e343017db47b4e201297e445db3e13b7af41f876 Mon Sep 17 00:00:00 2001
From: eddie <cf@rock-chips.com>
Date: Thu, 1 Sep 2011 11:23:09 +0800
Subject: [PATCH] fix bu92747 cir compile error

---
 drivers/cir/bu92747guw_cir.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/cir/bu92747guw_cir.c b/drivers/cir/bu92747guw_cir.c
index 66048d443c2b..c63ea295a027 100644
--- a/drivers/cir/bu92747guw_cir.c
+++ b/drivers/cir/bu92747guw_cir.c
@@ -64,10 +64,10 @@ void bu92747_lock(void)
 {
 	mutex_lock(&bu92747_mutex);
 }
-void bu92747_unlock(void)
-{
-	mutex_unlock(&bu92747_mutex);
-}
+//void bu92747_unlock(void)
+//{
+//	mutex_unlock(&bu92747_mutex);
+//}
 
 
 
-- 
2.35.3

