From 84a54e9455c43cfcdb3ff1ddaf8850de50e0fda3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Oct 2014 18:39:14 +0800
Subject: [PATCH] ARM: rockchip: cpu.c support export cpu version

---
 arch/arm/mach-rockchip/cpu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rockchip/cpu.c b/arch/arm/mach-rockchip/cpu.c
index d9aaa12a07d1..a8b4c226a52d 100644
--- a/arch/arm/mach-rockchip/cpu.c
+++ b/arch/arm/mach-rockchip/cpu.c
@@ -28,6 +28,10 @@ static ssize_t type_show(struct device *dev, struct device_attribute *attr, char
 	else
 		type = "";
 
+	if (rockchip_get_cpu_version())
+		return sprintf(buf, "%sv%lu\n", type,
+			       rockchip_get_cpu_version());
+
 	return sprintf(buf, "%s\n", type);
 }
 
-- 
2.35.3

