From 0a6d05ddd6c2936b9e30af9d7390252c9fcec084 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 23 Dec 2014 22:30:58 +0800
Subject: [PATCH] arm64: rockchip: rk3368 enable 8 CPUs support

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368.dtsi | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index c63537f65232..77dd88ae5347 100755
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -61,7 +61,6 @@
 			reg = <0x0 0x103>;
 			enable-method = "psci";
 		};
-/*
 		little0: cpu@0 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53", "arm,armv8";
@@ -86,7 +85,6 @@
 			reg = <0x0 0x3>;
 			enable-method = "psci";
 		};
-*/
 
 		cpu-map {
 			cluster0 {
@@ -103,7 +101,6 @@
 					cpu = <&big3>;
 				};
 			};
-/*
 			cluster1 {
 				core0 {
 					cpu = <&little0>;
@@ -118,7 +115,6 @@
 					cpu = <&little3>;
 				};
 			};
-*/
 		};
 	};
 
-- 
2.35.3

