From d6f60860594d5b765ae25482216308ba12beb346 Mon Sep 17 00:00:00 2001
From: Tang Yun ping <typ@rock-chips.com>
Date: Wed, 10 Jun 2015 15:32:09 +0800
Subject: [PATCH] RK3368 Mailbox: add mailbox version information

Signed-off-by: Tang Yun ping <typ@rock-chips.com>
---
 drivers/mailbox/rockchip_mailbox.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mailbox/rockchip_mailbox.c b/drivers/mailbox/rockchip_mailbox.c
index ca88773a48ed..02b41415fd8d 100644
--- a/drivers/mailbox/rockchip_mailbox.c
+++ b/drivers/mailbox/rockchip_mailbox.c
@@ -25,6 +25,8 @@
 #include <linux/rockchip-mailbox.h>
 #include <linux/scpi_protocol.h>
 
+#define MAILBOX_VERSION			"V1.00"
+
 #define MAILBOX_A2B_INTEN		0x00
 #define MAILBOX_A2B_STATUS		0x04
 #define MAILBOX_A2B_CMD(x)		(0x08 + (x) * 8)
@@ -193,6 +195,9 @@ static int rockchip_mbox_probe(struct platform_device *pdev)
 	struct resource *res;
 	int ret, irq, i;
 
+	dev_info(&pdev->dev,
+		 "Rockchip mailbox initialize, version: "MAILBOX_VERSION"\n");
+
 	if (!pdev->dev.of_node)
 		return -ENODEV;
 
-- 
2.35.3

