From d2d8872a0a738993e9a0dd7f637725eb97b7297b Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 14 Aug 2012 17:44:08 +0800
Subject: [PATCH] rkfbsys:mask printk int  set_dsp_lut

---
 drivers/video/rockchip/rkfb_sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index 97401d2a49d8..240d48364e54 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -259,7 +259,7 @@ static ssize_t set_dsp_lut(struct device *dev,struct device_attribute *attr,
 		temp = i;
 		dsp_lut[i] = temp + (temp<<8) + (temp<<16);  //init by default value
 	}
-	printk("%s\n",start);
+	//printk("%s\n",start);
 	for(i=0;i<256;i++)
 	{
 		temp = simple_strtoul(start,NULL,10);
-- 
2.35.3

