From 2db8c0fab7590c5ecca7d0852affecc109a4129c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Thu, 26 Apr 2012 15:39:07 +0800
Subject: [PATCH] rt5631 : disable rt5631 ALC function

---
 sound/soc/codecs/rt5631.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rt5631.c b/sound/soc/codecs/rt5631.c
index 9948c2bb72ba..db1b1397c90d 100755
--- a/sound/soc/codecs/rt5631.c
+++ b/sound/soc/codecs/rt5631.c
@@ -37,7 +37,7 @@
 #endif
 #define RT5631_VERSION "0.01 alsa 1.0.24"
 
-#define RT5631_ALC_DAC_FUNC_ENA 1	//ALC functio for DAC
+#define RT5631_ALC_DAC_FUNC_ENA 0	//ALC functio for DAC
 #define RT5631_ALC_ADC_FUNC_ENA 1	//ALC function for ADC
 #define RT5631_SPK_TIMER	0	//if enable this, MUST enable RT5631_EQ_FUNC_ENA first!
 
-- 
2.35.3

