From 55a4cbf7c8084bcbd6f2e5eed7d0715c1843fe8c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Tue, 21 Aug 2012 11:23:00 +0800
Subject: [PATCH] rk30: rm the sram_printch(8)

---
 arch/arm/mach-rk30/pm.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 03b08f8c6383..99fb2ca7a280 100644
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -387,9 +387,6 @@ static void __sramfunc rk30_sram_suspend(void)
 				  | (1 << CLK_GATE_PERIPH_SRC % 16)
 				  | (1 << CLK_GATE_PCLK_PERIPH % 16)
 				, clkgt_regs[2], CRU_CLKGATES_CON(2), 0xffff);
-			sram_printch('8');
-
-
 	}
 	gate_save_soc_clk(0
 			  | (1 << CLK_GATE_ACLK_STRC_SYS % 16)
-- 
2.35.3

