From af68afbbaeba1734ec4cf31d14a6ec95cb8d3762 Mon Sep 17 00:00:00 2001
From: ddl <ddl@rock-chips.com>
Date: Thu, 15 Nov 2012 11:51:45 +0800
Subject: [PATCH] camera: fix struct v4l2_mbus_framefmt have not initiation

---
 drivers/media/video/rk30_camera_oneframe.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/video/rk30_camera_oneframe.c b/drivers/media/video/rk30_camera_oneframe.c
index d8ad2d80d497..cd9f1b503e66 100755
--- a/drivers/media/video/rk30_camera_oneframe.c
+++ b/drivers/media/video/rk30_camera_oneframe.c
@@ -2202,6 +2202,7 @@ static int rk_camera_try_fmt(struct soc_camera_device *icd,
 		return pix->bytesperline;
 
     /* limit to sensor capabilities */
+    memset(&mf,0x00,sizeof(struct v4l2_mbus_framefmt));
 	mf.width	= pix->width;
 	mf.height	= pix->height;
 	mf.field	= pix->field;
-- 
2.35.3

