From d6a211dbc9038853c79eab45a5ea43b93daa9ea3 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 15 Apr 2014 19:08:07 +0800
Subject: [PATCH] rk fb: fix first ddr freq NULL point

---
 drivers/video/rockchip/rk_fb.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 2299e2c2bc7e..e718ab4a4d7a 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -596,8 +596,10 @@ bool rk_fb_poll_wait_frame_complete(void)
 
        
 	if (rk_fb_poll_prmry_screen_vblank() == RK_LF_STATUS_NC){
-		if(dev_drv->ops->set_irq_to_cpu)
-                        dev_drv->ops->set_irq_to_cpu(dev_drv,1);		
+		if (likely(dev_drv)) {
+			if(dev_drv->ops->set_irq_to_cpu)
+	                        dev_drv->ops->set_irq_to_cpu(dev_drv,1);
+		}
 		return false;
 	}	
 
-- 
2.35.3

