From fb90364541893c71d2911d221f8f2066f1983be2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 26 Apr 2012 17:51:41 +0800
Subject: [PATCH] rk30: smp: call fiq_glue_resume on platform_secondary_init

---
 arch/arm/mach-rk30/platsmp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/platsmp.c b/arch/arm/mach-rk30/platsmp.c
index 8e26efa25b00..b36ecd42ed2d 100755
--- a/arch/arm/mach-rk30/platsmp.c
+++ b/arch/arm/mach-rk30/platsmp.c
@@ -12,6 +12,7 @@
 #include <linux/version.h>
 
 #include <asm/cacheflush.h>
+#include <asm/fiq_glue.h>
 #include <asm/hardware/gic.h>
 #include <asm/smp_scu.h>
 
@@ -64,6 +65,7 @@ void __cpuinit platform_secondary_init(unsigned int cpu)
 
 #ifdef CONFIG_FIQ
 	gic_secondary_init_non_secure();
+	fiq_glue_resume();
 #endif
 }
 
-- 
2.35.3

