From 83f21e697868e374709cfd8430b584339cd2cb84 Mon Sep 17 00:00:00 2001
From: xjh <xujh@rock-chips.com>
Date: Sun, 11 May 2014 10:42:19 +0800
Subject: [PATCH] MIPI: fixed mipi merge error

---
 drivers/video/rockchip/screen/lcd_mipi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/screen/lcd_mipi.c b/drivers/video/rockchip/screen/lcd_mipi.c
index 023a873ea508..9689f5af9d02 100755
--- a/drivers/video/rockchip/screen/lcd_mipi.c
+++ b/drivers/video/rockchip/screen/lcd_mipi.c
@@ -114,6 +114,7 @@ static void rk_mipi_screen_cmd_init(struct mipi_screen *screen)
 			else if (dcs_cmd->dcs_cmd.dsi_id == 2){
 				MIPI_SCREEN_DBG("dcs_cmd.dsi_id == 2 line=%d\n",__LINE__); 
 				dsi_send_packet(0, cmds, len);
+				dsi_send_packet(1, cmds, len);
 			}
 			else{
 				MIPI_SCREEN_DBG("dsi is err.");
-- 
2.35.3

