From a6b813e804c9468183d5d80a5720c20e48b31ba8 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Fri, 14 Oct 2011 12:41:07 +0800
Subject: [PATCH] rk29fb:close win0 win1 when suspend to fix display old frame
 when resume

---
 drivers/video/rk29_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 250a4ed8af52..6d1666484658 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -2814,6 +2814,7 @@ static void rk29fb_early_suspend(struct early_suspend *h)
 		fbprintk(">>>>>> power down the screen! \n");
 		inf->cur_screen->standby(1);
 	}
+	LcdMskReg(inf, SYS_CONFIG, m_W0_ENABLE | m_W1_ENABLE, v_W0_ENABLE(0) | v_W1_ENABLE(0));
 
     LcdMskReg(inf, DSP_CTRL0, m_HSYNC_POLARITY | m_VSYNC_POLARITY | m_DEN_POLARITY ,
        v_HSYNC_POLARITY(1) | v_VSYNC_POLARITY(1) | v_DEN_POLARITY(1) );
-- 
2.35.3

