From 3ed00a8562848a557239e84aac3ddf5bbecd7073 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Nov 2011 22:09:46 +0800
Subject: [PATCH] pack-kernel.sh: fix local version

---
 pack-kernel.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index c8e6c1fc0f00..5c78d466a6b9 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -117,6 +117,9 @@ make distclean >/dev/null 2>&1
 
 popd >/dev/null
 
+# fix local version
+echo "+" > $kerndir/.scmversion
+
 # tar kernel
 pushd $kerndir/../ >/dev/null
 package=$(basename $kerndir).tar
@@ -136,6 +139,8 @@ gzip -9 -c $package > $package.gz
 rm $ex
 popd >/dev/null
 
+rm -f $kerndir/.scmversion
+
 echo done
 
 
-- 
2.35.3

