From d4d0f587f9c9376b71dc0e494763bf6103b3c236 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Mar 2014 18:14:49 +0800
Subject: [PATCH] ARM: rockchip: iomap.h add RK3288_IMEM_PHYS define

---
 include/linux/rockchip/iomap.h | 22 ++++++++++------------
 1 file changed, 10 insertions(+), 12 deletions(-)

diff --git a/include/linux/rockchip/iomap.h b/include/linux/rockchip/iomap.h
index 7a3d046e7df4..ae4e1fd6446d 100755
--- a/include/linux/rockchip/iomap.h
+++ b/include/linux/rockchip/iomap.h
@@ -17,11 +17,10 @@
 #define RK_DEBUG_UART_VIRT              RK_IO_ADDRESS(0x00060000)
 #define RK_CPU_AXI_BUS_VIRT             RK_IO_ADDRESS(0x00070000)
 #define RK_TIMER_VIRT                   RK_IO_ADDRESS(0x00080000)
-#define RK_GIC_VIRT                      RK_IO_ADDRESS(0x00090000)
-#define RK_BOOTRAM_VIRT           RK_IO_ADDRESS(0x000a0000)
+#define RK_GIC_VIRT                     RK_IO_ADDRESS(0x00090000)
+#define RK_BOOTRAM_VIRT                 RK_IO_ADDRESS(0x000a0000)
 #define RK_DDR_VIRT                     RK_IO_ADDRESS(0x000d0000)
 
-
 #define RK3188_CRU_PHYS                 0x20000000
 #define RK3188_CRU_SIZE                 SZ_4K
 #define RK3188_GRF_PHYS                 0x20008000
@@ -105,14 +104,13 @@
 #define RK3288_UART_GPS_PHYS            0xFF1B0000
 #define RK3288_UART_EXP_PHYS            0xFF1C0000
 #define RK3288_UART_SIZE                SZ_4K
-
-#define RK3288_GIC_DIST_PHYS            0xffc01000
-#define RK3288_GIC_DIST_SIZE                SZ_4K
-#define RK3288_GIC_CPU_PHYS            0xffc02000
-#define RK3288_GIC_CPU_SIZE                SZ_4K
-
-#define RK3288_BOOTRAM_PHYS            0xff720000
-#define RK3288_BOOTRAM_SIZE            SZ_4K
-
+#define RK3288_GIC_DIST_PHYS            0xFFC01000
+#define RK3288_GIC_DIST_SIZE            SZ_4K
+#define RK3288_GIC_CPU_PHYS             0xFFC02000
+#define RK3288_GIC_CPU_SIZE             SZ_4K
+#define RK3288_BOOTRAM_PHYS             0xFF720000
+#define RK3288_BOOTRAM_SIZE             SZ_4K
+#define RK3288_IMEM_PHYS                0xFF700000
+#define RK3288_IMEM_SZIE                0x00018000
 
 #endif
-- 
2.35.3

