From a3e500dca7327d679608de6f08e092d78b10d801 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Mar 2014 19:01:00 +0800
Subject: [PATCH] ARM: rockchip: rename rk3288 machine name

---
 arch/arm/mach-rockchip/rk3288.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index f493c5718cb8..9496432d8f9d 100755
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -345,9 +345,10 @@ static void rk3288_restart(char mode, const char *cmd)
 	writel_relaxed(0xeca8, RK_CRU_VIRT + RK3288_CRU_GLB_SRST_SND_VALUE);
 	dsb();
 }
+
 static void __init rk3288_init_suspend(void);
 
-DT_MACHINE_START(RK3288_DT, "RK30board")
+DT_MACHINE_START(RK3288_DT, "Rockchip RK3288 (Flattened Device Tree)")
 	.smp		= smp_ops(rockchip_smp_ops),
 	.map_io		= rk3288_dt_map_io,
 	.init_time	= rk3288_dt_init_timer,
-- 
2.35.3

