From 24da4f2bb4e76ea63dc993a6057666568b550c27 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9E=97=E8=BE=89=E8=BE=89?= <lhh@rock-chips.com>
Date: Sat, 24 Apr 2010 06:33:04 +0000
Subject: [PATCH] update rk2818_serial

---
 drivers/serial/rk2818_serial.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/serial/rk2818_serial.c b/drivers/serial/rk2818_serial.c
index 839c637b7dd3..1bdd67749698 100644
--- a/drivers/serial/rk2818_serial.c
+++ b/drivers/serial/rk2818_serial.c
@@ -280,6 +280,8 @@ static int rk2818_serial_startup(struct uart_port *port)
 		rk2818_serial_shutdown(port);
 		return 	retval;
 	}	
+	rk2818_uart_write(port,0xf1,UART_FCR);
+	rk2818_uart_write(port,0x01,UART_SFE);///enable fifo
     rk2818_uart_write(port,UART_IER_RECV_DATA_AVAIL_INT_ENABLE,UART_IER);  //enable uart recevice IRQ
 	return 0;
 }
-- 
2.35.3

