From 368fd1a457db7c31159190479aeeae93a7b248e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 19 Apr 2012 19:37:16 +0800
Subject: [PATCH] rk30: cpufreq: fix rk30_cpu_exit for cpu 1

---
 arch/arm/mach-rk30/cpufreq.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index 600a8e19b3fa..748796df4194 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -272,6 +272,8 @@ static int rk30_cpu_init(struct cpufreq_policy *policy)
 
 static int rk30_cpu_exit(struct cpufreq_policy *policy)
 {
+	if (policy->cpu != 0)
+		return 0;
 	cpufreq_frequency_table_cpuinfo(policy, freq_table);
 	clk_put(cpu_clk);
 #ifdef CONFIG_RK30_CPU_FREQ_LIMIT_BY_TEMP
-- 
2.35.3

