From dde191432cdb5dc3c8c4169f5448a2229707a2cb Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 14 Mar 2014 17:09:54 +0800
Subject: [PATCH] rk fb:fix pwr_ctr_parse_dt

---
 drivers/video/rockchip/rk_fb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index ce7ebe16a329..c48781212fe3 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -189,8 +189,8 @@ static int rk_fb_data_fmt(int data_format,int bits_per_pixel)
 */
 int rk_disp_pwr_ctr_parse_dt(struct rk_lcdc_driver *dev_drv)
 {
-	struct device_node *root  = of_parse_phandle(dev_drv->dev->of_node,
-				"power_ctr", 0);
+	struct device_node *root  = of_get_child_by_name(dev_drv->dev->of_node,
+				"power_ctr");
 	struct device_node *child;
 	struct rk_disp_pwr_ctr_list *pwr_ctr;
 	struct list_head *pos;
-- 
2.35.3

