From 09a960744d931d3f20f3339d22b5447b8226d7c0 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 14 Mar 2013 16:47:00 +0800
Subject: [PATCH] lcdc Kconfig:RK30 lcdc depends on ARCH_RK30XX

---
 drivers/video/rockchip/lcdc/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/lcdc/Kconfig b/drivers/video/rockchip/lcdc/Kconfig
index 44d7182b6080..ee1a0d2eb155 100644
--- a/drivers/video/rockchip/lcdc/Kconfig
+++ b/drivers/video/rockchip/lcdc/Kconfig
@@ -6,7 +6,7 @@ config LCDC_RK2928
 
 config LCDC_RK30
         tristate "rk30 lcdc support"
-        depends on FB_ROCKCHIP && SOC_RK3066
+        depends on FB_ROCKCHIP && ARCH_RK30XX
         help
           Driver for rk30 lcdc .There are two lcd controllers on rk30
 
-- 
2.35.3

