From b8fa73ae68f15ebf60427e42d0f1dd40cf3e0a23 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 21 Jan 2013 14:11:29 +0800
Subject: [PATCH] rk3188: support RK_USB_UART config

---
 arch/arm/plat-rk/Kconfig | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 3624aa91109f..33bbb511ffcd 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -165,8 +165,7 @@ config RK_DEBUG_UART
 
 config RK_USB_UART
 	bool "Support USB UART Bypass Function"
-	depends on ARCH_RK2928 && (RK_DEBUG_UART = 2)
-	default n
+	depends on (ARCH_RK2928 || ARCH_RK3188) && (RK_DEBUG_UART = 2)
 
 config RK_CONSOLE_THREAD
 	bool "Console write by thread"
-- 
2.35.3

