From baddb94fea7e5599ccb43f8b9320e433de8446d7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Wed, 9 May 2012 16:55:33 +0800
Subject: [PATCH] rk30:phone:report key for suspend and resume

---
 drivers/input/misc/twl6030-pwrbutton.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/misc/twl6030-pwrbutton.c b/drivers/input/misc/twl6030-pwrbutton.c
index 3b5b76191a7c..051a056276df 100644
--- a/drivers/input/misc/twl6030-pwrbutton.c
+++ b/drivers/input/misc/twl6030-pwrbutton.c
@@ -123,7 +123,7 @@ static int __devinit twl6030_pwrbutton_probe(struct platform_device *pdev)
 
 	__set_bit(EV_KEY, pwr_button->input_dev->evbit);
 
-	pwr_button->report_key = KEY_END;
+	pwr_button->report_key = KEY_POWER;
 	pwr_button->dev = &pdev->dev;
 	pwr_button->input_dev->evbit[0] = BIT_MASK(EV_KEY);
 	pwr_button->input_dev->keybit[BIT_WORD(pwr_button->report_key)] =
-- 
2.35.3

