From 34a8da9eb28cb38420539087ae55a94c84f357f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 10 Feb 2012 16:10:05 +0800
Subject: [PATCH] rk30: clock: add spi clock for fpga

---
 arch/arm/mach-rk30/clock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/clock.c b/arch/arm/mach-rk30/clock.c
index 18a1966a61a6..8982ed56f9bd 100644
--- a/arch/arm/mach-rk30/clock.c
+++ b/arch/arm/mach-rk30/clock.c
@@ -62,6 +62,8 @@ static struct clk_lookup clks[] = {
 	CLK("rk30_i2c.2", "i2c", &xin24m),
 	CLK("rk30_i2c.3", "i2c", &xin24m),
 	CLK("rk30_i2c.4", "i2c", &xin24m),
+	CLK("rk29xx_spim.0", "spi", &xin24m),
+	CLK("rk29xx_spim.1", "spi", &xin24m),
 };
 
 void __init rk30_clock_init(void)
-- 
2.35.3

