From 0a708a171b910afe23137f729e32bc57650fbb7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 18 May 2012 09:37:43 +0800
Subject: [PATCH] rk30: lcdc: remove set aclk rate code

---
 drivers/video/rockchip/chips/rk30_lcdc.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/video/rockchip/chips/rk30_lcdc.c b/drivers/video/rockchip/chips/rk30_lcdc.c
index 9ce62be385bf..bfec242709c1 100644
--- a/drivers/video/rockchip/chips/rk30_lcdc.c
+++ b/drivers/video/rockchip/chips/rk30_lcdc.c
@@ -213,21 +213,6 @@ static int rk30_load_screen(struct rk_lcdc_device_driver *dev_drv, bool initscre
 	fps = div64_u64(1000000000000llu,ft);
 	screen->ft = 1000/fps;
     	printk("%s: dclk:%lu>>fps:%d ",lcdc_dev->driver.name,clk_get_rate(lcdc_dev->dclk),fps);
-    	if(initscreen)
-    	{
-        	if(screen->lcdc_aclk)
-		{
-			ret = clk_set_rate(lcdc_dev->aclk, screen->lcdc_aclk);
-			if(ret)
-			{
-           	 		printk(KERN_ERR ">>>>>> set lcdc%d aclk  rate failed\n",lcdc_dev->id);
-        		}
-			
-        		clk_enable(lcdc_dev->aclk);
-			printk("aclk:%lu\n",clk_get_rate(lcdc_dev->aclk));
-        	}
-        	
-	}
 
     	if(screen->init)
     	{
-- 
2.35.3

