From 3101bcb39bd0ad4cf89c707edb6fe5cabb17dc7e Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 23 Aug 2012 17:43:41 +0800
Subject: [PATCH] struct rk_screen:add dsp_lut

---
 include/linux/rk_screen.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index 2409ddc0938e..02aa5d38a343 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -120,6 +120,7 @@ typedef struct rk29fb_screen {
 	u16 lower_margin;
 	u16 vsync_len;
 	u8  ft;	//the time need to display one frame,in ms
+	int *dsp_lut; //display lut 
 #ifdef CONFIG_HDMI_DUAL_DISP
     /* Scaler mode Timing */
 	u32 s_pixclock;
-- 
2.35.3

