From 02cdac8bf8b1e746dc48ddb485dd76d0ecec4590 Mon Sep 17 00:00:00 2001
From: ljf <ljf@rock-chips.com>
Date: Fri, 4 Jul 2014 15:28:15 +0800
Subject: [PATCH] rk3036: add dts node for vpu and hevc

---
 arch/arm/boot/dts/rk3036.dtsi | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/rk3036.dtsi b/arch/arm/boot/dts/rk3036.dtsi
index 836becc4e820..c8616f020bbc 100755
--- a/arch/arm/boot/dts/rk3036.dtsi
+++ b/arch/arm/boot/dts/rk3036.dtsi
@@ -400,4 +400,25 @@
 		clock-names = "clk_usbphy1", "hclk_usb1";
 	};
 
+        vpu: vpu_service@10108000 {
+		compatible = "vpu_service";
+		reg = <0x10108000 0x800>;
+		interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_enc", "irq_dec";
+		clocks = <&clk_vdpu>, <&hclk_vdpu>;
+		clock-names = "aclk_vcodec", "hclk_vcodec";
+		name = "vpu_service";
+		status = "disabled";
+	};
+
+	hevc: hevc_service@1010c000 {
+		compatible = "rockchip,hevc_service";
+		reg = <0x1010c000 0x800>;
+		interrupts = <GIC_SPI 57 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_dec";
+		clocks = <&aclk_hevc>, <&hclk_hevc>, <&clk_hevc_core>, <&clk_hevc_cabac>;
+		clock-names = "aclk_vcodec", "hclk_vcodec", "clk_core", "clk_cabac";
+		name = "hevc_service";
+		status = "disabled";
+	};
 };
-- 
2.35.3

