From 27dc823ee959bb59f37f4a7bb639ac7bc0088df2 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Thu, 9 Oct 2014 16:56:35 +0800
Subject: [PATCH] mmc: only ctrl_all_reset if slot state really changed

---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 553d780e0e24..cc1272daf03d 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -2957,7 +2957,6 @@ static void dw_mci_work_routine_card(struct work_struct *work)
 		int present;
 
 		present = dw_mci_get_cd(mmc);
-		dw_mci_ctrl_all_reset(host);
 
                 /* Stop edma when rountine card triggered */
                 if(cpu_is_rk3036() || cpu_is_rk312x())
@@ -2970,6 +2969,7 @@ static void dw_mci_work_routine_card(struct work_struct *work)
 			MMC_DBG_BOOT_FUNC(mmc, "  The card is %s.  ===!!!!!!==[%s]\n",
 				present ? "inserted" : "removed.", mmc_hostname(mmc));
 	
+			dw_mci_ctrl_all_reset(host);
 			rk_send_wakeup_key();//wake up system
 			spin_lock_bh(&host->lock);
 
-- 
2.35.3

