From 0d615e8cdad79e5e2a37efe6717fb292c5158fc0 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 9 Mar 2015 18:09:37 +0800
Subject: [PATCH] rk fb: if not want the config effect,set reserved[3] bit[0] 1

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9492ebec7650..7a30703fcc4f 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1374,7 +1374,9 @@ static int rk_fb_pan_display(struct fb_var_screeninfo *var,
 	if (video_data_to_mirroring)
 		video_data_to_mirroring(info, NULL);
 #endif
-	dev_drv->ops->cfg_done(dev_drv);
+        /*if not want the config effect,set reserved[3] bit[0] 1*/
+        if (likely((var->reserved[3] & 0x1) == 0))
+	        dev_drv->ops->cfg_done(dev_drv);
 
 	return 0;
 }
-- 
2.35.3

