From add7c61b1903421e62678a923d521922f445bb59 Mon Sep 17 00:00:00 2001
From: yangkai <yk@rock-chips.com>
Date: Thu, 13 Sep 2012 10:35:19 +0800
Subject: [PATCH] suspend usb phy before bypass to uart

---
 arch/arm/mach-rk2928/common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/common.c b/arch/arm/mach-rk2928/common.c
index d26d028cdcfb..8c93626a649a 100755
--- a/arch/arm/mach-rk2928/common.c
+++ b/arch/arm/mach-rk2928/common.c
@@ -119,6 +119,7 @@ void __init rk2928_map_io(void)
 	rk2928_map_common_io();
 #ifdef DEBUG_UART_BASE
         #ifdef CONFIG_RK_USB_UART
+        writel_relaxed(0x007f0051, RK2928_GRF_BASE + GRF_UOC0_CON5);
         writel_relaxed(0x30003000, RK2928_GRF_BASE + GRF_UOC1_CON4);
         #endif
         writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
-- 
2.35.3

