From a8dceb6bb97c21e4bd55f11ed4ac9cb4160ec30f Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Wed, 15 Apr 2015 15:53:39 +0800
Subject: [PATCH] change rga.h define to match rk_fb config

---
 drivers/video/rockchip/rga/rga.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rga/rga.h b/drivers/video/rockchip/rga/rga.h
index 31ab69401b4a..0a66dbe5996d 100755
--- a/drivers/video/rockchip/rga/rga.h
+++ b/drivers/video/rockchip/rga/rga.h
@@ -120,7 +120,7 @@ typedef struct rga_img_info_t
     unsigned long yrgb_addr;      /* yrgb    mem addr         */
     unsigned long uv_addr;        /* cb/cr   mem addr         */
     unsigned long v_addr;         /* cr      mem addr         */
-    unsigned long format;         //definition by RK_FORMAT
+    unsigned int format;         //definition by RK_FORMAT
 
     unsigned short act_w;
     unsigned short act_h;
-- 
2.35.3

