From eb97899693779b85a1f22402190dbf3cbce1a28f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:30:13 +0800
Subject: [PATCH] gpio.h: fix for v2.6.36

---
 include/asm-generic/gpio.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/include/asm-generic/gpio.h b/include/asm-generic/gpio.h
index f744e93d1943..d2d5ede00546 100644
--- a/include/asm-generic/gpio.h
+++ b/include/asm-generic/gpio.h
@@ -108,12 +108,9 @@ struct gpio_chip {
 
 	void			(*set)(struct gpio_chip *chip,
 						unsigned offset, int value);
-	
-    	int        	      (*set_debounce)(struct gpio_chip *chip,
-					unsigned offset, unsigned debounce);     
-		
-   	int        	      (*pull_updown)(struct gpio_chip *chip,
-					unsigned offset, unsigned value);     
+
+	int			(*pull_updown)(struct gpio_chip *chip,
+						unsigned offset, unsigned value);
 
 	int			(*to_irq)(struct gpio_chip *chip,
 						unsigned offset);
-- 
2.35.3

