From cf1a1774b3518e5b99eb36526405093109cda7a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 10 Aug 2012 15:47:37 +0800
Subject: [PATCH] rk2928: pm: build pm.c use Thumb instruction set

---
 arch/arm/mach-rk2928/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/Makefile b/arch/arm/mach-rk2928/Makefile
index 68053947e830..a2169a7bc77c 100644
--- a/arch/arm/mach-rk2928/Makefile
+++ b/arch/arm/mach-rk2928/Makefile
@@ -7,6 +7,7 @@ obj-y += iomux.o
 obj-y += clock.o
 obj-y += clock_data.o
 obj-$(CONFIG_PM) += pm.o
+CFLAGS_pm.o += -Os -mthumb
 
 obj-$(CONFIG_MACH_RK2928_FPGA) += board-rk2928-fpga.o board-rk2928-fpga-key.o
 obj-$(CONFIG_MACH_RK2928_SDK) += board-rk2928-sdk.o board-rk2928-sdk-key.o
-- 
2.35.3

