From 6a12d3d1d380c987d209d121208d2fbbcbb8c1ff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Thu, 17 Jan 2013 11:40:45 +0800
Subject: [PATCH] rk610 codec : cloes debug

---
 sound/soc/codecs/rk610_codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rk610_codec.c b/sound/soc/codecs/rk610_codec.c
index 02c44f288d78..a6b89a04fa58 100755
--- a/sound/soc/codecs/rk610_codec.c
+++ b/sound/soc/codecs/rk610_codec.c
@@ -63,7 +63,7 @@
 /*
  * Debug
  */
-#if 1
+#if 0
 #define	DBG(x...)	printk(KERN_INFO x)
 #else
 #define	DBG(x...)
-- 
2.35.3

