From 78b87074ab196638315bd33b3247f34692c7909d Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 19 Jul 2013 18:54:53 +0800
Subject: [PATCH] mtdpart: put_partition: start from p1

---
 fs/partitions/mtdpart.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/partitions/mtdpart.c b/fs/partitions/mtdpart.c
index ea433583859c..7e93ddd7026b 100644
--- a/fs/partitions/mtdpart.c
+++ b/fs/partitions/mtdpart.c
@@ -327,8 +327,8 @@ int mtdpart_partition(struct parsed_partitions *state)
 		return num_parts;
 
 	for(i = 0; i < num_parts; i++){
-		put_partition(state, i, parts[i].from + FROM_OFFSET, parts[i].size);
-		strcpy(state->parts[i].info.volname, parts[i].name);
+		put_partition(state, i+1, parts[i].from + FROM_OFFSET, parts[i].size);
+		strcpy(state->parts[i+1].info.volname, parts[i].name);
 		printk(KERN_INFO "%10s: 0x%09llx -- 0x%09llx (%llu MB)\n", 
 				parts[i].name,
 				parts[i].from * 512,
-- 
2.35.3

