From baacfc8d145da6195d704982d6f2ba73e905304b Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Mon, 12 Jan 2015 14:01:48 +0800
Subject: [PATCH] arm64: rockchip: add watchdog support

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 drivers/watchdog/rk29_wdt.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/watchdog/rk29_wdt.c b/drivers/watchdog/rk29_wdt.c
index 5ed94e5c020d..4065372a3e94 100644
--- a/drivers/watchdog/rk29_wdt.c
+++ b/drivers/watchdog/rk29_wdt.c
@@ -32,7 +32,9 @@
 #include <linux/clk.h>
 #include <linux/uaccess.h>
 #include <linux/io.h>
+#ifndef CONFIG_ARM64
 #include <asm/mach/map.h>
+#endif
 #ifdef CONFIG_OF
 #include <linux/of.h>
 #endif
@@ -124,7 +126,17 @@ static char		 expect_close;
 		printk(KERN_INFO msg); \
 	} while (0)
 
-#define wdt_writel(v, offset) do { writel_relaxed(v, wdt_base + offset); dsb(); } while (0)
+#ifdef CONFIG_ARM64
+#define wdt_writel(v, offset) do { \
+	writel_relaxed(v, wdt_base + offset);\
+	dsb(sy);\
+	} while (0)
+#else
+#define wdt_writel(v, offset) do { \
+	writel_relaxed(v, wdt_base + offset);\
+	dsb();\
+	} while (0)
+#endif
 
 /* functions */
 void rk29_wdt_keepalive(void)
-- 
2.35.3

