From 5854697912d5590da8a32c98c38483033af6dd45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Feb 2011 18:16:26 +0800
Subject: [PATCH] rk29: L2 Data RAM latency set to 9 cycles

---
 arch/arm/mm/proc-v7.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mm/proc-v7.S b/arch/arm/mm/proc-v7.S
index b388331c25ac..7a246d5c268c 100644
--- a/arch/arm/mm/proc-v7.S
+++ b/arch/arm/mm/proc-v7.S
@@ -272,7 +272,7 @@ __v7_setup:
 	bic	r5, r5, #7 << 6
 	bic	r5, r5, #15
 	orr	r5, r5, #3 << 6			@ Tag RAM latency: b011 = 4 cycles
-	orr	r5, r5, #12			@ Data RAM latency: b0101 = 6 cycles
+	orr	r5, r5, #8			@ Data RAM latency: b1000 = 9 cycles
 	mcr	p15, 1, r5, c9, c0, 2
 #endif
 
-- 
2.35.3

