From 94dfc90f19ca9de4b7b1a252e16108ebc0a7a841 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Mon, 1 Jun 2015 15:41:27 +0800
Subject: [PATCH] rfkill-wlan, solve rockchip_wifi_mac_addr error

---
 net/rfkill/rfkill-wlan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/rfkill/rfkill-wlan.c b/net/rfkill/rfkill-wlan.c
index ade0577d42e2..bdcc8aa56c9c 100755
--- a/net/rfkill/rfkill-wlan.c
+++ b/net/rfkill/rfkill-wlan.c
@@ -618,10 +618,10 @@ int rockchip_wifi_mac_addr(unsigned char *buf)
 
     if (is_valid_ether_addr(wifi_custom_mac_addr) && !strncmp(wifi_chip_type_string, "rtl", 3)) {
         wifi_custom_mac_addr[0] &= ~0x2; // for p2p
-    } else {
+    }/* else {
         LOG("This mac address is not valid, ignored...\n");
         return -1;
-    }
+    }*/
 
     memcpy(buf, wifi_custom_mac_addr, 6);
 
-- 
2.35.3

