From 33246c59d03ae67b6d1dfa2f12a59f4812ae2a32 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 13 Jul 2012 11:44:33 +0800
Subject: [PATCH] rk30: ddr.c: always compile with Os, else may hung on boot

---
 arch/arm/mach-rk30/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/Makefile b/arch/arm/mach-rk30/Makefile
index e240e14532d7..73f0cf156036 100755
--- a/arch/arm/mach-rk30/Makefile
+++ b/arch/arm/mach-rk30/Makefile
@@ -3,6 +3,7 @@ obj-y += clock_data.o
 obj-y += common.o
 CFLAGS_common.o += -DTEXT_OFFSET=$(TEXT_OFFSET)
 obj-y += ddr.o
+CFLAGS_ddr.o += -Os
 obj-y += devices.o
 obj-y += io.o
 obj-y += iomux.o
-- 
2.35.3

