From d12c9020e9c9648e6d7c8d487857cdff72da4884 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 28 May 2012 19:57:55 +0800
Subject: [PATCH] rk30:sdk: clock_data add cpu 1.1G and 1.3G support

---
 arch/arm/mach-rk30/clock_data.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk30/clock_data.c b/arch/arm/mach-rk30/clock_data.c
index 3fb733b0ef69..b69c9251f1b0 100644
--- a/arch/arm/mach-rk30/clock_data.c
+++ b/arch/arm/mach-rk30/clock_data.c
@@ -985,9 +985,13 @@ static const struct apll_clk_set apll_clks[] = {
 	_APLL_SET_CLKS(1464, 1, 61, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1416, 1, 59, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1368, 1, 57, 1, 8, 41, 21, 41, 21),
+	_APLL_SET_CLKS(1320, 1, 55, 1, 8, 41, 21, 41, 21),
+	_APLL_SET_CLKS(1296, 1, 54, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1272, 1, 53, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1200, 1, 50, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1176, 1, 49, 1, 8, 41, 21, 41, 21),
+	_APLL_SET_CLKS(1128, 1, 47, 1, 8, 41, 21, 41, 21),
+	_APLL_SET_CLKS(1104, 1, 46, 1, 8, 41, 21, 41, 21),
 	_APLL_SET_CLKS(1008, 1, 42, 1, 8, 31, 21, 41, 21),
 	_APLL_SET_CLKS(888,  1, 37, 1, 8, 31, 21, 41, 21),
 	_APLL_SET_CLKS(816 , 1, 34, 1, 8, 31, 21, 41, 21),
-- 
2.35.3

