From a00b8903eb8c402e7d6b4da7599c23b36f48c988 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Sat, 29 Sep 2012 14:56:23 +0800
Subject: [PATCH] rk30:set i2c clk 200k in i2c_sram.c

---
 arch/arm/mach-rk30/i2c_sram.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/i2c_sram.c b/arch/arm/mach-rk30/i2c_sram.c
index 21537c85d301..e91ed63b66bf 100755
--- a/arch/arm/mach-rk30/i2c_sram.c
+++ b/arch/arm/mach-rk30/i2c_sram.c
@@ -143,8 +143,8 @@ void __sramfunc sram_i2c_init()
     data[2] = readl_relaxed(RK30_GRF_BASE + GRF_GPIO_IOMUX);
     writel_relaxed(data[2]| I2C_GRF_GPIO_IOMUX, RK30_GRF_BASE + GRF_GPIO_IOMUX);
 	
-	div = rk30_ceil(24*1000*1000, I2C_SPEED*1000 * 8);
-	divh = divl = rk30_ceil(div, 2);	
+	div = 0x1e;
+	divh = divl = 0xf;	
 	writel_relaxed(I2C_CLKDIV_VAL(divl, divh), SRAM_I2C_ADDRBASE + I2C_CLKDIV);
 	data[3]  = readl_relaxed(SRAM_I2C_ADDRBASE + I2C_CLKDIV);
 	
-- 
2.35.3

