From 5eb17849a0bcf5650543c90893a219e775ba064a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:22:01 +0800
Subject: [PATCH] mma8452: fix for v2.6.36

---
 drivers/input/gsensor/mma8452.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/gsensor/mma8452.c b/drivers/input/gsensor/mma8452.c
index b842d8a8e9cd..8aade9457583 100755
--- a/drivers/input/gsensor/mma8452.c
+++ b/drivers/input/gsensor/mma8452.c
@@ -578,7 +578,7 @@ static struct file_operations mma8452_fops = {
 	.owner = THIS_MODULE,
 	.open = mma8452_open,
 	.release = mma8452_release,
-	.ioctl = mma8452_ioctl,
+	.unlocked_ioctl = mma8452_ioctl,
 };
 
 static struct miscdevice mma8452_device = {
-- 
2.35.3

