From 66b0c972f4abe276695553b341204bc04e220c0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 16 Mar 2011 17:07:05 +0800
Subject: [PATCH] rk29: pm: remove unused variable "delay"

---
 arch/arm/mach-rk29/pm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk29/pm.c b/arch/arm/mach-rk29/pm.c
index db1e8cffbfd9..9fbf1bfd5c2c 100644
--- a/arch/arm/mach-rk29/pm.c
+++ b/arch/arm/mach-rk29/pm.c
@@ -49,7 +49,6 @@ extern void ddr_resume(void);
 static void inline printch(char byte)
 {
 	unsigned long flags;
-	unsigned int delay;
 	u32 gate1, gate2;
 
 	local_irq_save(flags);
-- 
2.35.3

