From 194b5f525d50de9e7f56233e44cdbaf09a4ab013 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 7 Dec 2012 19:37:21 +0800
Subject: [PATCH] rk29_keys: fix compile error caused by commit d9f2bfa1

---
 drivers/input/keyboard/rk29_keys.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/input/keyboard/rk29_keys.c b/drivers/input/keyboard/rk29_keys.c
index 47141b284c65..c476b2468efe 100755
--- a/drivers/input/keyboard/rk29_keys.c
+++ b/drivers/input/keyboard/rk29_keys.c
@@ -100,10 +100,7 @@ static ssize_t rk29key_set(struct device *dev,
 		
 		p = strstr(buf,Arrary[i].keyArrary);
 		if(p==0)
-              {
-                   dev_dbg(" rk29key_set p == 0 error ...............\n");
-                   continue;
-              }
+			continue;
 		
 		start = strcspn(p,":");
 		
-- 
2.35.3

