From 9261177cdee772fde2e795274a2a8c900fc3e02a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 28 Jun 2011 17:00:04 +0800
Subject: [PATCH] pack-kernel.sh: exclude newton board, add phone sdk

---
 pack-kernel.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 7d46be14068e..9efa97fd2628 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -43,8 +43,8 @@ arch/arm/configs/rk29_a22_defconfig
 arch/arm/mach-rk29/board-rk29-fih*
 arch/arm/configs/rk29_FIH_defconfig
 
-arch/arm/mach-rk29/board-rk29-phonesdk*
-arch/arm/configs/rk29_phonesdk_defconfig
+arch/arm/mach-rk29/board-rk29-newton*
+arch/arm/configs/rk29_newton_defconfig
 )
 
 # ---------------------------------------------------------------------------
-- 
2.35.3

