From ce551444a18d74e65e1e8b770cac657de250003d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 13 Dec 2013 19:44:36 +0800
Subject: [PATCH] i2c: rockchip: i2c_msg add scl_rate for compat support old
 driver

---
 include/uapi/linux/i2c.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/uapi/linux/i2c.h b/include/uapi/linux/i2c.h
index 0e949cbfd333..740002ed9b33 100644
--- a/include/uapi/linux/i2c.h
+++ b/include/uapi/linux/i2c.h
@@ -78,6 +78,9 @@ struct i2c_msg {
 #define I2C_M_RECV_LEN		0x0400	/* length will be first received byte */
 	__u16 len;		/* msg length				*/
 	__u8 *buf;		/* pointer to msg data			*/
+#ifdef CONFIG_I2C_ROCKCHIP_COMPAT
+	__u32 scl_rate;		/* add by kfx */
+#endif
 };
 
 /* To determine what functionality is present */
-- 
2.35.3

