From 6deed4556679bb18bf86ac600b492cf5a2b9dda2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 28 Dec 2012 20:11:08 +0800
Subject: [PATCH] rk: fiq_debugger: fix add last_kmsg command commit

---
 arch/arm/common/fiq_debugger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/common/fiq_debugger.c b/arch/arm/common/fiq_debugger.c
index bd0e4575fa61..d23809165927 100644
--- a/arch/arm/common/fiq_debugger.c
+++ b/arch/arm/common/fiq_debugger.c
@@ -623,10 +623,10 @@ static void debug_help(struct fiq_debugger_state *state)
 				" reboot        Reboot\n"
 				" irqs          Interupt status\n"
 				" kmsg          Kernel log\n"
+				" version       Kernel version\n");
 #ifdef CONFIG_RK29_LAST_LOG
-				" last_kmsg     Last kernel log\n"
+	debug_printf(state,	" last_kmsg     Last kernel log\n");
 #endif
-				" version       Kernel version\n");
 	debug_printf(state,	" sleep         Allow sleep while in FIQ\n"
 				" nosleep       Disable sleep while in FIQ\n"
 				" console       Switch terminal to console\n"
-- 
2.35.3

