From 4e449bdf66d45d3df73b91528c34f6974830f1c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 25 Jul 2011 16:57:50 +0800
Subject: [PATCH] rk29: clock: more increase aclk_cpu

---
 arch/arm/mach-rk29/clock.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 524882ff2999..105ad5d0761f 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -388,14 +388,15 @@ static unsigned long lpj_gpll;
 static const struct arm_pll_set arm_pll[] = {
 	// rate = 24 * NF / (NR * NO)
 	//      rate NR  NF NO adiv hdiv pdiv
-	ARM_PLL(1200, 1, 50, 1, 31, 21, 81),
-	ARM_PLL(1104, 1, 46, 1, 31, 21, 81),
+	ARM_PLL(1200, 1, 50, 1, 21, 21, 81),
+	ARM_PLL(1176, 2, 98, 1, 21, 21, 81),
+	ARM_PLL(1104, 1, 46, 1, 21, 21, 81),
 	ARM_PLL(1008, 1, 42, 1, 21, 21, 81),
-	ARM_PLL( 912, 1, 38, 1, 31, 21, 81),
-	ARM_PLL( 888, 2, 74, 1, 31, 21, 81),
-	ARM_PLL( 816, 1, 34, 1, 31, 21, 81),
-	ARM_PLL( 696, 1, 58, 2, 31, 21, 81),
-	ARM_PLL( 624, 1, 52, 2, 31, 21, 81),
+	ARM_PLL( 912, 1, 38, 1, 21, 21, 81),
+	ARM_PLL( 888, 2, 74, 1, 21, 21, 81),
+	ARM_PLL( 816, 1, 34, 1, 21, 21, 81),
+	ARM_PLL( 696, 1, 58, 2, 21, 21, 81),
+	ARM_PLL( 624, 1, 52, 2, 21, 21, 81),
 	ARM_PLL( 600, 1, 50, 2, 21, 21, 81),
 	ARM_PLL( 504, 1, 42, 2, 21, 21, 81),
 	ARM_PLL( 408, 1, 34, 2, 21, 21, 81),
-- 
2.35.3

