From c0ac86ad944b0ba97b43d292d67b388bf74314a5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 29 Mar 2013 10:53:37 +0800
Subject: [PATCH] ts: rk29_i2c_goodix: unregister early suspend when probe
 failed

---
 drivers/input/touchscreen/rk29_i2c_goodix.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/touchscreen/rk29_i2c_goodix.c b/drivers/input/touchscreen/rk29_i2c_goodix.c
index 2a2fde8585cd..4ad739b43d32 100755
--- a/drivers/input/touchscreen/rk29_i2c_goodix.c
+++ b/drivers/input/touchscreen/rk29_i2c_goodix.c
@@ -855,6 +855,9 @@ static int rk_ts_probe(struct i2c_client *client, const struct i2c_device_id *id
 
 	
 err_input_register_device_failed:
+#ifdef CONFIG_HAS_EARLYSUSPEND
+	unregister_early_suspend(&ts->early_suspend);
+#endif
 	input_free_device(ts->input_dev);
 	i2c_set_clientdata(client, NULL);	
 	kfree(ts);
-- 
2.35.3

