From 068c5feda77239a493da584e0f4c36d7e9bc8c03 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 28 May 2010 14:21:52 +0000
Subject: [PATCH] fix wrong usecount bug

---
 arch/arm/mach-rk2818/clock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index bad081b6bac2..9298ec18d1f5 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -1014,9 +1014,9 @@ static void __clk_disable(struct clk *clk)
 		if (clk->mode)
 			clk->mode(clk, 0);
 		pr_debug("clock: %s disabled\n", clk->name);
+		if (clk->parent)
+			__clk_disable(clk->parent);
 	}
-	if (clk->parent)
-		__clk_disable(clk->parent);
 }
 
 void clk_disable(struct clk *clk)
-- 
2.35.3

