From afbb4b080ea00996937f985b3393e8501a9f8a3f Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 28 Jan 2013 15:59:00 +0800
Subject: [PATCH] rk3188: init sdio rate=24.75M to ensure even div

---
 arch/arm/mach-rk3188/clock_data.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk3188/clock_data.c b/arch/arm/mach-rk3188/clock_data.c
index 11bc67008b73..cd8adee1fb41 100755
--- a/arch/arm/mach-rk3188/clock_data.c
+++ b/arch/arm/mach-rk3188/clock_data.c
@@ -3141,6 +3141,7 @@ static void __init rk30_clock_common_init(unsigned long gpll_rate, unsigned long
 	
 	clk_set_rate_nolock(&clk_uart0, 49500000);
 	clk_set_rate_nolock(&clk_sdmmc, 24750000);
+	clk_set_rate_nolock(&clk_sdio, 24750000);
 
 }
 
-- 
2.35.3

