From 0fc71500f8845b8beabc457b792113a5ccc93fc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 4 Jun 2014 20:30:43 +0800
Subject: [PATCH] rk: ion: fix ion.h build warnings

---
 drivers/staging/android/ion/ion.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/android/ion/ion.h b/drivers/staging/android/ion/ion.h
index 3c1a838ca3fa..05f6382ccb15 100755
--- a/drivers/staging/android/ion/ion.h
+++ b/drivers/staging/android/ion/ion.h
@@ -201,10 +201,14 @@ int ion_share_dma_buf_fd(struct ion_client *client, struct ion_handle *handle);
  */
 struct ion_handle *ion_import_dma_buf(struct ion_client *client, int fd);
 
+#ifdef CONFIG_ARCH_ROCKCHIP
+struct device;
+
 int ion_map_iommu(struct device *iommu_dev, struct ion_client *client,
-		struct ion_handle *handle, unsigned long *iova, unsigned long *size);
+		  struct ion_handle *handle, unsigned long *iova,
+		  unsigned long *size);
 
 void ion_unmap_iommu(struct device *iommu_dev, struct ion_client *client,
-			struct ion_handle *handle);
-
+		     struct ion_handle *handle);
+#endif
 #endif /* _LINUX_ION_H */
-- 
2.35.3

