From 214649ba0fea638b0c11788764f5cacc748a01af Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Sat, 20 Jul 2013 14:51:00 +0800
Subject: [PATCH] emmc: do NOT alloc boot partition now

---
 drivers/mmc/core/mmc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc.c b/drivers/mmc/core/mmc.c
index e7762ad0dde3..403905501f5f 100755
--- a/drivers/mmc/core/mmc.c
+++ b/drivers/mmc/core/mmc.c
@@ -341,7 +341,10 @@ static int mmc_read_ext_csd(struct mmc_card *card, u8 *ext_csd)
 		 * There are two boot regions of equal size, defined in
 		 * multiples of 128K.
 		 */
-		card->ext_csd.boot_size = ext_csd[EXT_CSD_BOOT_MULT] << 17;
+		if(HOST_IS_EMMC(card->host)) //emmc: We do NOT alloc boot partition now (kfx)
+			card->ext_csd.boot_size = 0;
+		else
+			card->ext_csd.boot_size = ext_csd[EXT_CSD_BOOT_MULT] << 17;
 	}
 
 	card->ext_csd.raw_hc_erase_gap_size =
-- 
2.35.3

