From 4f9e05b983031a8505d91d00b001c02dc183fbf6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 2 Dec 2013 10:38:19 +0800
Subject: [PATCH] clocksource: rockchip_timer: clock source 32-bit mode

---
 drivers/clocksource/rockchip_timer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clocksource/rockchip_timer.c b/drivers/clocksource/rockchip_timer.c
index 8c5291fa0e47..124869d37368 100644
--- a/drivers/clocksource/rockchip_timer.c
+++ b/drivers/clocksource/rockchip_timer.c
@@ -180,14 +180,14 @@ static struct local_timer_ops rk_local_timer_ops __cpuinitdata = {
 
 static cycle_t rk_timer_read(struct clocksource *cs)
 {
-	return ~rk_timer_read_current_value64(cs_timer.base);
+	return ~rk_timer_read_current_value(cs_timer.base);
 }
 
 static struct clocksource rk_timer_clocksource = {
 	.name	= TIMER_NAME,
 	.rating	= 200,
 	.read	= rk_timer_read,
-	.mask	= CLOCKSOURCE_MASK(64),
+	.mask	= CLOCKSOURCE_MASK(32),
 	.flags	= CLOCK_SOURCE_IS_CONTINUOUS,
 };
 
-- 
2.35.3

