From 9d91cc184699f2eb69df8906091f236b9c14cfaa Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Thu, 21 Jun 2012 10:59:25 +0800
Subject: [PATCH] fix rga prescale mode mem leak

---
 drivers/video/rockchip/rga/rga_drv.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index a6bbd9882ba1..0474d2da293f 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -796,6 +796,11 @@ static int rga_blit(rga_session *session, struct rga_req *req)
                 break;
             }
             num = 2;
+
+            if(NULL != req2)
+            {
+                kfree(req2);
+            }
         }
         else
         {
-- 
2.35.3

