From 3bc178e8bcf63e55852cca0baf8fdd1a3b178217 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Wed, 30 Jul 2014 17:28:33 +0800
Subject: [PATCH] rockchip:iommu:fix build error when iommu disabled

---
 include/linux/rockchip/iovmm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/rockchip/iovmm.h b/include/linux/rockchip/iovmm.h
index 48e212d4dadc..d67454fb74da 100755
--- a/include/linux/rockchip/iovmm.h
+++ b/include/linux/rockchip/iovmm.h
@@ -68,7 +68,7 @@ struct device *rockchip_get_sysmmu_device_by_compatible(const char *compt);
 #define iovmm_unmap(dev, iova)		do { } while (0)
 #define iovmm_map_oto(dev, phys, size)	(-ENOSYS)
 #define iovmm_unmap_oto(dev, phys)	do { } while (0)
-#define rockchip_get_sysmmu_device_by_compatible(compt) do { } while (0)
+#define rockchip_get_sysmmu_device_by_compatible(compt) (NULL)
 
 #endif /* CONFIG_ROCKCHIP_IOVMM */
 
-- 
2.35.3

