From 2bc313b36cf498d8a61513930aec8ac149234217 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 18 Feb 2013 10:46:05 +0800
Subject: [PATCH] rk3188: Kconfig: enable RK_SRAM_DMA

---
 arch/arm/plat-rk/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 725bfb6b8cb8..8d7aba6a812c 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -184,7 +184,7 @@ config RK_CONSOLE_THREAD
 
 config RK_SRAM_DMA
 	bool "Sound DMA buffer in internal SRAM"
-	depends on ARCH_RK30
+	depends on ARCH_RK30 || ARCH_RK3188
 
 config RK_PL330_DMA
 	bool
-- 
2.35.3

