From dd70d9c0612b9655e29b4581237200e84a4d81ea Mon Sep 17 00:00:00 2001
From: gwl <gwl@rock-chips.com>
Date: Tue, 11 Mar 2014 17:30:13 +0800
Subject: [PATCH] wifi: fix rfkill-wlan.c error.

---
 net/rfkill/rfkill-wlan.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/rfkill/rfkill-wlan.c b/net/rfkill/rfkill-wlan.c
index 8f475149962e..2e47fe9d6deb 100755
--- a/net/rfkill/rfkill-wlan.c
+++ b/net/rfkill/rfkill-wlan.c
@@ -23,7 +23,6 @@
 #include <asm/gpio.h>
 #include <linux/regulator/consumer.h>
 #include <linux/delay.h>
-#include <linux/rfkill-rk.h>
 #include <linux/rfkill-wlan.h>
 #include <linux/wakelock.h>
 #include <linux/interrupt.h>
-- 
2.35.3

