From 70ed8b38ee6a083c4e46bef492d6e14a0c3e76ed Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Dec 2010 13:48:34 +0800
Subject: [PATCH] rk29_fb: fix setup DMC AXI host priority

---
 drivers/video/rk29_fb.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 38a50cb9c714..3d3634d54b55 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1917,6 +1917,21 @@ static int __init rk29fb_probe (struct platform_device *pdev)
 
     fbprintk(">>>>>> %s : %s\n", __FILE__, __FUNCTION__);
 
+    /* Fix DMC AXI host priority. FIXME: Move to bootloader in the future? */
+    __raw_writel(0x7, RK29_DDRC_BASE+0x200);
+    __raw_writel(0x7, RK29_DDRC_BASE+0x204);
+    __raw_writel(0x7, RK29_DDRC_BASE+0x208);
+    __raw_writel(0x7, RK29_DDRC_BASE+0x20c);
+    __raw_writel(0x7, RK29_DDRC_BASE+0x210);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x280);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x284);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x288);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x28c);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x290);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x294);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x298);
+    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x29c);
+    __raw_writel(0x2, RK29_DDRC_BASE+0x2a0);
     *(volatile u32 *)(RK29_GRF_BASE+0xac) = ((*(volatile u32 *)(RK29_GRF_BASE+0xac)) & ~0x3FF) | 0x246;
 
     /* Malloc rk29fb_inf and set it to pdev for drvdata */
-- 
2.35.3

