From 5d698a48e06dda3a2bce40fe13e2b025e7ae38a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Tue, 22 May 2012 17:22:24 +0800
Subject: [PATCH] rk30 phone: aic3262 set codec phone call slave

---
 sound/soc/rk29/rk29_aic3262.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk29_aic3262.c b/sound/soc/rk29/rk29_aic3262.c
index 09b6455459b5..e43bed7d2f73 100644
--- a/sound/soc/rk29/rk29_aic3262.c
+++ b/sound/soc/rk29/rk29_aic3262.c
@@ -367,7 +367,7 @@ static int rk29_aif3_hw_params(struct snd_pcm_substream *substream,
 
 	/* set codec DAI configuration */
 	ret = snd_soc_dai_set_fmt(codec_dai, SND_SOC_DAIFMT_DSP_A |
-			SND_SOC_DAIFMT_IB_NF | SND_SOC_DAIFMT_CBM_CFM);
+			SND_SOC_DAIFMT_IB_NF | SND_SOC_DAIFMT_CBS_CFS);
 	if (ret < 0)
 	{
 		printk("%s: snd_soc_dai_set_fmt err =%d\n",__FUNCTION__,ret);
-- 
2.35.3

