From 7d80392f6c0338239b7d1e9b7c80a2343e23f118 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 18 Jul 2012 11:18:43 +0800
Subject: [PATCH] rkfb:save data format when this format changed

---
 drivers/video/rockchip/rk_fb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index a07940b493a2..ecd893dbdd7e 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -375,7 +375,6 @@ static int rk_fb_set_par(struct fb_info *info)
 	u32 yvir = var->yres_virtual;
 	u8 data_format = var->nonstd&0xff;
 	var->pixclock = dev_drv->pixclock;
-	
 	#if defined(CONFIG_HDMI_RK30)
 		#if defined(CONFIG_DUAL_DISP_IN_KERNEL)
 			if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED)
@@ -496,6 +495,8 @@ static int rk_fb_set_par(struct fb_info *info)
 					par2->xact = par->xact;
 					par2->yact = par->yact;
 					par2->format = par->format;
+					info2->var.nonstd &= 0xffffff00;
+					info2->var.nonstd |= par->format;
 					dev_drv1->set_par(dev_drv1,layer_id);
 				}
 			}
-- 
2.35.3

