From 1cc1a50561f4467a86d60b715ffbdc355d3f41a1 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 10 May 2013 17:51:10 +0800
Subject: [PATCH] rk3188 lcdc: do not call sscreen_set in shutdown

---
 drivers/video/rockchip/lcdc/rk3188_lcdc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/video/rockchip/lcdc/rk3188_lcdc.c b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
index 8ac6fa3ad2e2..4d103899e6c0 100644
--- a/drivers/video/rockchip/lcdc/rk3188_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
@@ -1520,8 +1520,6 @@ static void rk3188_lcdc_shutdown(struct platform_device *pdev)
 		lcdc_dev->driver.cur_screen->standby(1);
 	if(lcdc_dev->driver.screen_ctr_info->io_disable) //power off the screen if necessary
 		lcdc_dev->driver.screen_ctr_info->io_disable();
-	if(lcdc_dev->driver.cur_screen->sscreen_set) //turn off  lvds if necessary
-		lcdc_dev->driver.cur_screen->sscreen_set(lcdc_dev->driver.cur_screen , 0);
 	rk3188_lcdc_deint(lcdc_dev);
 	rk_fb_unregister(&(lcdc_dev->driver));	
 }
-- 
2.35.3

