From 401817083f1f9b610e7a2f6d659414282323959b Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 22 Mar 2011 11:09:38 +0800
Subject: [PATCH] update i2c drivers about 'set scl_rate'

---
 drivers/i2c/busses/i2c-rk29.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index 1d0be8d550dd..efc129f0e0fa 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -184,6 +184,7 @@ static void  rk29_i2c_clockrate(struct rk29_i2c_data *i2c)
 	rk29_i2c_calcdivisor(i2c->i2c_rate, scl_rate, &real_rate, &rem, &exp);
 
 	tmp = readl(i2c->regs + I2C_OPR);
+	tmp &= ~0x3f;
 	tmp |= exp;
 	tmp |= rem<<I2CCDVR_EXP_BITS;	
 	writel(tmp, i2c->regs + I2C_OPR);
-- 
2.35.3

