From e9390e5ce87577cce77c9771b2c26da520545958 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Wed, 29 Oct 2014 08:25:44 +0800
Subject: [PATCH] mmc: add more dbg info for dto

---
 drivers/mmc/host/rk_sdmmc.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 0733a1d87514..f7b90d0deca6 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3766,8 +3766,9 @@ static struct dw_mci_board *dw_mci_parse_dt(struct dw_mci *host)
 static void dw_mci_dealwith_timeout(struct dw_mci *host)
 {
         u32 regs;
-	u32 sdio_int;
+        u32 sdio_int;
 
+        dev_err(host->dev, "host->state = 0x%x\n", host->state);
         switch(host->state){
                 case STATE_IDLE:
                         break;
@@ -3777,8 +3778,9 @@ static void dw_mci_dealwith_timeout(struct dw_mci *host)
 		        mci_writel(host, RINTSTS, SDMMC_INT_DRTO);  // clear interrupt
                         set_bit(EVENT_DATA_COMPLETE, &host->pending_events);
                         host->state = STATE_DATA_BUSY;
-	                if (!dw_mci_ctrl_all_reset(host)) {
-                                return ;
+                        if (!dw_mci_ctrl_all_reset(host)) {
+	                        dev_err(host->dev, "dto: ctrl_all_reset failed!\n");
+	                        return ;
                         }
 
                         /* NO requirement to reclaim slave chn using external dmac */
-- 
2.35.3

