From 69b55eda28377c2fa22435e9ce8e038ac090a654 Mon Sep 17 00:00:00 2001
From: wlq <wlq@rock-chips.com>
Date: Mon, 26 Mar 2012 15:20:28 +0800
Subject: [PATCH] td8801:add reset modem

---
 drivers/misc/tdsc8800.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/misc/tdsc8800.c b/drivers/misc/tdsc8800.c
index d30d82ccb17a..fed23a14a257 100755
--- a/drivers/misc/tdsc8800.c
+++ b/drivers/misc/tdsc8800.c
@@ -33,6 +33,7 @@ MODULE_LICENSE("GPL");
 
 #define SLEEP 1
 #define READY 0
+#define RESET 1
 struct rk2818_23d_data *gpdata = NULL;
 
 
@@ -69,6 +70,16 @@ static int tdsc8800_release(struct inode *inode, struct file *file)
 }
 static long  tdsc8800_ioctl(struct file *file, unsigned int a, unsigned long b)
 {
+	switch(a){
+		case RESET:
+			modem_poweron_off(0);
+			msleep(1000);
+			modem_poweron_off(1);
+			break;
+		default:
+			MODEMDBG("cmd error !!!\n");
+			break;
+	}
 	return 0;
 }
 
-- 
2.35.3

