From e75b1bce6d4635ee41b87e507e8aedcb15baa56e Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Mon, 23 Apr 2012 20:29:16 +0800
Subject: [PATCH] rk30:hdmi: fix hdmi crash when enable DEBUG_LOCKDEP.

---
 drivers/video/display/display-sys.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/display/display-sys.c b/drivers/video/display/display-sys.c
index 7a8c6466727b..ef8d665cb646 100755
--- a/drivers/video/display/display-sys.c
+++ b/drivers/video/display/display-sys.c
@@ -154,7 +154,8 @@ static struct device_attribute display_attrs[] = {
 	__ATTR(enable, 0664, display_show_enable, display_store_enable),
 	__ATTR(connect, S_IRUGO, display_show_connect, NULL),
 	__ATTR(modes, S_IRUGO, display_show_modes, NULL),
-	__ATTR(mode, 0664, display_show_mode, display_store_mode)
+	__ATTR(mode, 0664, display_show_mode, display_store_mode),
+	__ATTR_NULL
 };
 
 static int display_suspend(struct device *dev, pm_message_t state)
-- 
2.35.3

