From ead9010b042bfcd2fca7a78a7f843f6bca0c4df9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 20 Mar 2013 14:55:13 +0800
Subject: [PATCH] pack-kernel.sh: exclude rk_pm_tests

---
 pack-kernel.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 6aa1729d2c8d..2416d7c8821a 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -9,6 +9,8 @@ pack-kernel*
 defconfig
 kernel.img
 arch/arm/plat-rk/vpu*.c
+arch/arm/plat-rk/rk_pm_tests/*.c
+arch/arm/plat-rk/rk_pm_tests/*.h
 drivers/staging/rk29/vivante
 drivers/staging/rk29/ipp/rk29-ipp.c
 drivers/*rk28*.c
-- 
2.35.3

