From 866f2f04cc7eb09c2c562cecd92a774d5ef97ad8 Mon Sep 17 00:00:00 2001
From: ddl <ddl@rock-chips.com>
Date: Wed, 26 Dec 2012 17:44:00 +0800
Subject: [PATCH] camera: ov2659 complie warning

---
 drivers/media/video/ov2659.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/video/ov2659.c b/drivers/media/video/ov2659.c
index 0bd0f8647e5b..6b3ce6441db5 100755
--- a/drivers/media/video/ov2659.c
+++ b/drivers/media/video/ov2659.c
@@ -1362,8 +1362,8 @@ struct sensor_parameter
 	unsigned short int preview_line_width;
 	unsigned short int preview_gain;
 
-	unsigned short int PreviewPclk;
-	unsigned short int CapturePclk;
+	unsigned int PreviewPclk;
+	unsigned int CapturePclk;
 	char awb[6];
 };
 
-- 
2.35.3

