From 278d2a1ab231c04241452583bf3f55bc45202aac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 16 Aug 2012 11:51:17 +0800
Subject: [PATCH] gpio: rk2928: enable support rk30_gpiolib_pull_updown

---
 drivers/gpio/gpio-rk30.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpio/gpio-rk30.c b/drivers/gpio/gpio-rk30.c
index 14417a33a4a9..911636cf49c1 100755
--- a/drivers/gpio/gpio-rk30.c
+++ b/drivers/gpio/gpio-rk30.c
@@ -318,10 +318,9 @@ static void rk30_gpiolib_set(struct gpio_chip *chip, unsigned offset, int val)
 	spin_unlock_irqrestore(&bank->lock, flags);
 }
 
-
 static int rk30_gpiolib_pull_updown(struct gpio_chip *chip, unsigned offset, unsigned enable)
 {
-#if defined(CONFIG_ARCH_RK30)
+#if defined(CONFIG_ARCH_RK30) || defined(CONFIG_ARCH_RK2928)
 	struct rk30_gpio_bank *bank = to_rk30_gpio_bank(chip);
 	unsigned long flags;
 
@@ -335,7 +334,6 @@ static int rk30_gpiolib_pull_updown(struct gpio_chip *chip, unsigned offset, uns
 	return 0;
 }
 
-
 static int rk30_gpiolib_to_irq(struct gpio_chip *chip, unsigned offset)
 {
 	return chip->base + offset;
-- 
2.35.3

