From 144eee39098bf9db628c3ac09eea78db4aa80ee3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Tue, 25 May 2010 02:49:28 +0000
Subject: [PATCH] change sdmmc

---
 arch/arm/mach-rk2818/clock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index d43d43763434..bad081b6bac2 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -912,7 +912,7 @@ static struct clk_lookup clks[] = {
 	CLK1(lcdc),
 	CLK1(vip),
 	CLK1(i2s),
-	CLK("rk2818_sdmmc0", "sdmmc", &sdmmc0_clk),
+	CLK("rk2818_sdmmc.0", "sdmmc", &sdmmc0_clk),
 	CLK1(ebrom),
 	CLK1(gpio0),
 	CLK1(gpio1),
@@ -929,7 +929,7 @@ static struct clk_lookup clks[] = {
 	CLK1(lsadc),
 	CLK("rk2818_serial.2", "uart", &uart2_clk),
 	CLK("rk2818_serial.3", "uart", &uart3_clk),
-	CLK("rk2818_sdmmc1", "sdmmc", &sdmmc1_clk),
+	CLK("rk2818_sdmmc.1", "sdmmc", &sdmmc1_clk),
 
 	CLK1(hsadc),
 	CLK1(sdram_common),
-- 
2.35.3

