From ddcdb8e5f8aeea778d622c990c584178281115bb Mon Sep 17 00:00:00 2001
From: xxx <xxx@rock-chips.com>
Date: Mon, 17 Aug 2015 17:34:38 +0800
Subject: [PATCH] rk3368-p9_818.dts: add suspend config

---
 arch/arm64/boot/dts/rk3368-p9_818.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368-p9_818.dts b/arch/arm64/boot/dts/rk3368-p9_818.dts
index 158922f204ba..c0f3b79bfca8 100755
--- a/arch/arm64/boot/dts/rk3368-p9_818.dts
+++ b/arch/arm64/boot/dts/rk3368-p9_818.dts
@@ -765,3 +765,15 @@
 		<&clk_hevc_cabac 300000000>;
 };
 
+&rockchip_suspend {
+	rockchip,ctrbits = <
+		(0
+		| RKPM_SLP_ARMOFF
+		| RKPM_SLP_PMU_PLLS_PWRDN
+		| RKPM_SLP_PMU_PMUALIVE_32K
+		| RKPM_SLP_SFT_PLLS_DEEP
+		| RKPM_SLP_PMU_DIS_OSC
+		| RKPM_SLP_SFT_PD_NBSCUS
+		)
+		>;
+};
-- 
2.35.3

