From c10a1e0ad5ebbfe3b7fecec4ebd694a29fdade4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 16 Apr 2012 11:56:47 +0800
Subject: [PATCH] rk30: cpufreq: suspend freq set to 816MHz

---
 arch/arm/mach-rk30/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index a2500ff5f052..600a8e19b3fa 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -44,7 +44,7 @@ static struct cpufreq_frequency_table *freq_table = default_freq_table;
 #define MASK_FURTHER_CPUFREQ            0x30
 /* With 0x00(NOCHANGE), it depends on the previous "further" status */
 static int no_cpufreq_access;
-static unsigned int suspend_freq = 1008 * 1000;
+static unsigned int suspend_freq = 816 * 1000;
 
 #define NUM_CPUS	2
 static struct workqueue_struct *freq_wq;
-- 
2.35.3

