From b72c7672e9b04c27fa7a508ec23681f44866931d Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Sun, 17 Jul 2011 23:45:50 -0700
Subject: [PATCH] newton: modify irda serial device suspend&resume function

---
 drivers/net/irda/ir_serial.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/irda/ir_serial.c b/drivers/net/irda/ir_serial.c
index c51a3520ca51..76dfdf688e52 100755
--- a/drivers/net/irda/ir_serial.c
+++ b/drivers/net/irda/ir_serial.c
@@ -614,7 +614,7 @@ static int bu92747_irda_suspend(struct platform_device *pdev, pm_message_t state
 	s->suspending = 1;
 	uart_suspend_port(&bu92747_irda_uart_driver, &s->port);
 
-	irda_hw_shutdown();
+	//irda_hw_shutdown();
 	if (s->pdata->irda_pwr_ctl)
 		s->pdata->irda_pwr_ctl(0);
 
@@ -631,7 +631,8 @@ static int bu92747_irda_resume(struct platform_device *pdev)
 	if (s->pdata->irda_pwr_ctl)
 		s->pdata->irda_pwr_ctl(1);
 	
-	irda_hw_startup();
+	//irda_hw_startup();
+	irda_hw_set_speed(s->baud);
 
 	uart_resume_port(&bu92747_irda_uart_driver, &s->port);
 	s->suspending = 0;
-- 
2.35.3

