From 1c9bff3029b6ad72eaa1c824ad09851853cd1d2b Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Sat, 4 Feb 2012 10:25:26 +0800
Subject: [PATCH] add some 3G support

---
 drivers/usb/serial/option.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 8dc95fdfa2fe..2fbdf8f68ce5 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -1108,6 +1108,12 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(ZTE_VENDOR_ID, 0xFFFF) },
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x5700) },
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x6341) },
+	{ USB_DEVICE(0x230d, 0x000d) },
+	{ USB_DEVICE(0x0E8D, 0x00A2) },
+	{ USB_DEVICE(0x1E89, 0x1A20) },
+	{ USB_DEVICE(0x12D1, 0x1C05) },
+	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x0007, 0xff, 0xff, 0xff) },
+	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x0167, 0xff, 0xff, 0xff) },
 
 	{ USB_DEVICE(OLIVETTI_VENDOR_ID, OLIVETTI_PRODUCT_OLICARD100) },
 	{ USB_DEVICE(CELOT_VENDOR_ID, CELOT_PRODUCT_CT680M) }, /* CT-650 CDMA 450 1xEVDO modem */
-- 
2.35.3

