From fca072d64bf891f3bedb41dbe12a2051b491aa2a Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Thu, 25 Jun 2015 20:05:35 +0800
Subject: [PATCH] dvfs: rockchip: fix min_temp_limit

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm/mach-rockchip/dvfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rockchip/dvfs.c b/arch/arm/mach-rockchip/dvfs.c
index d9d060dd011e..6f65f9f4e6f5 100644
--- a/arch/arm/mach-rockchip/dvfs.c
+++ b/arch/arm/mach-rockchip/dvfs.c
@@ -2111,6 +2111,7 @@ static int dvfs_node_parse_dt(struct device_node *np,
 	if (dvfs_node->temp_limit_enable) {
 		of_property_read_u32_index(np, "min_temp_limit",
 					   0, &dvfs_node->min_temp_limit);
+		dvfs_node->min_temp_limit *= 1000;
 		of_property_read_u32_index(np, "target-temp",
 					   0, &dvfs_node->target_temp);
 		pr_info("target-temp:%d\n", dvfs_node->target_temp);
-- 
2.35.3

