From 7535fcd3dd71a32c0003bf616b435c084abae68f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 21 Mar 2014 11:00:32 +0800
Subject: [PATCH] rk3288:pmic:rk808:modify int_mask_reg2

---
 include/linux/mfd/rk808.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/mfd/rk808.h b/include/linux/mfd/rk808.h
index 9038aa88c930..f69a64e05e77 100755
--- a/include/linux/mfd/rk808.h
+++ b/include/linux/mfd/rk808.h
@@ -90,7 +90,7 @@
 #define RK808_INT_STS_REG1 0X4c
 #define RK808_INT_STS_MSK_REG1 0X4d
 #define RK808_INT_STS_REG2 0X4e
-#define RK808_INT_STS_MSK_REG2 0X4d
+#define RK808_INT_STS_MSK_REG2 0X4f
 #define RK808_IO_POL_REG 0X50
 
 /* IRQ Definitions */
-- 
2.35.3

