From 3033f0d12aaef3212f0aa998bfc55e8e878148b7 Mon Sep 17 00:00:00 2001
From: qjb <qjb@rock-chips.com>
Date: Mon, 24 Feb 2014 18:00:14 +0800
Subject: [PATCH] audio pcm kernel 3.10 : support

---
 sound/soc/rockchip/rk_pcm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk_pcm.c b/sound/soc/rockchip/rk_pcm.c
index 7b90199466b1..f758aa2a4ef6 100755
--- a/sound/soc/rockchip/rk_pcm.c
+++ b/sound/soc/rockchip/rk_pcm.c
@@ -52,7 +52,8 @@ static const struct snd_dmaengine_pcm_config rockchip_dmaengine_pcm_config = {
 int rockchip_pcm_platform_register(struct device *dev)
 {
 	return snd_dmaengine_pcm_register(dev, &rockchip_dmaengine_pcm_config,
-			SND_DMAENGINE_PCM_FLAG_COMPAT);
+			SND_DMAENGINE_PCM_FLAG_COMPAT|
+			SND_DMAENGINE_PCM_FLAG_NO_RESIDUE);
 }
 EXPORT_SYMBOL_GPL(rockchip_pcm_platform_register);
 
-- 
2.35.3

