From 4a8a2ed2a7be54f89ec6b12fc8909858c1cf1292 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Thu, 16 Jul 2015 04:49:42 -0700
Subject: [PATCH] rk3288: clk: slove make warning for clk-ops.c

Signed-off-by: zhangqing <zhangqing@rock-chips.com>
---
 drivers/clk/rockchip/clk-ops.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/clk/rockchip/clk-ops.c b/drivers/clk/rockchip/clk-ops.c
index a10c62ae924c..7930a8cd9e3b 100644
--- a/drivers/clk/rockchip/clk-ops.c
+++ b/drivers/clk/rockchip/clk-ops.c
@@ -644,9 +644,11 @@ static int clk_3288_dclk_lcdc0_set_rate(struct clk_hw *hw, unsigned long rate,
 
 	/* set aclk_vio */
 	if (parent_rate	== __clk_get_rate(gpll)) {
+		parent = clk_get(NULL, "clk_gpll");
 		clk_set_parent(aclk_vio0, gpll);
 		clk_set_rate(aclk_vio0, 300*MHZ);
 	} else {
+		parent = clk_get(NULL, "clk_cpll");
 		clk_set_parent(aclk_vio0, cpll);
 		clk_set_rate(aclk_vio0, __clk_get_rate(cpll));
 	}
@@ -714,9 +716,11 @@ static int clk_3288_dclk_lcdc1_set_rate(struct clk_hw *hw, unsigned long rate,
 
 	/* set aclk_vio */
 	if (parent_rate	== __clk_get_rate(gpll)) {
+		parent = clk_get(NULL, "clk_gpll");
 		clk_set_parent(aclk_vio1, gpll);
 		clk_set_rate(aclk_vio1, 300*MHZ);
 	} else {
+		parent = clk_get(NULL, "clk_cpll");
 		clk_set_parent(aclk_vio1, cpll);
 		clk_set_rate(aclk_vio1, __clk_get_rate(cpll));
 	}
-- 
2.35.3

