From 04f0e18fdd8fd60b79f609530e9967376bc08f65 Mon Sep 17 00:00:00 2001
From: xxx <xxx@rock-chips.com>
Date: Wed, 15 May 2013 17:05:23 +0800
Subject: [PATCH] pm.c remove redefine clkgt_regs_sram

---
 arch/arm/mach-rk30/pm.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 72b482413697..c09d3a1af0c8 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -528,8 +528,6 @@ static u32 __sramdata sram_grf_uoc0_con0_status;
 
 static void __sramfunc rk_pm_soc_sram_clk_gating(void)
 {
-
-	u32 clkgt_regs_sram[CRU_CLKGATES_CON_CNT];
 	int i;
 
 	#if defined(CONFIG_ARCH_RK3188) && (CONFIG_RK_DEBUG_UART == 2)
-- 
2.35.3

