From 4f4343d438c97e98239434158b49c6db27ed6ac4 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 May 2015 20:22:22 +0800
Subject: [PATCH] iommu: rockchip-iovmm: fix compilation warning

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/iommu/rockchip-iovmm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/rockchip-iovmm.c b/drivers/iommu/rockchip-iovmm.c
index 193e0b5e8657..bb68d16575dd 100644
--- a/drivers/iommu/rockchip-iovmm.c
+++ b/drivers/iommu/rockchip-iovmm.c
@@ -149,7 +149,7 @@ dma_addr_t rockchip_iovmm_map(struct device *dev,
 	if (ret)
 		goto err_map_map;
 
-	dev_dbg(dev->archdata.iommu, "IOVMM: Allocated VM region @ %pad%#zx bytes.\n",
+	dev_dbg(dev->archdata.iommu, "IOVMM: Allocated VM region @ %p/%#X bytes.\n",
 	&region->start, region->size);
 	
 	return region->start;
-- 
2.35.3

