From 9cbd68512af8891607cb1e71f157393dd7988c4d Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 29 Jan 2013 00:10:03 +0800
Subject: [PATCH] rk_screen.h:modify rk_disp_prop

---
 include/linux/rk_screen.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index 7a9dfd7867e2..ad278d086859 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -78,7 +78,7 @@ typedef enum _MCU_STATUS {
 } MCU_STATUS;
 
 enum rk_disp_prop{       //display device property
-    PRMRY = 0,                     //primary display device ,like LCD screen
+    PRMRY = 1,                     //primary display device ,like LCD screen
     EXTEND,                        //extend display device ,like hdmi ,tv out
 };
 
-- 
2.35.3

