From 229d5011408642429d8d66aa7e3762ea59c825b6 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Wed, 1 Jun 2011 18:41:46 -0700
Subject: [PATCH] improve spi lock for lcd

---
 drivers/spi/rk29_spim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/rk29_spim.c b/drivers/spi/rk29_spim.c
index 7438d6193b84..08fd217d8311 100755
--- a/drivers/spi/rk29_spim.c
+++ b/drivers/spi/rk29_spim.c
@@ -100,9 +100,9 @@ void rk29_lcd_spim_spin_lock(void)
 
 void rk29_lcd_spim_spin_unlock(void)
 {
-	enable_irq(IRQ_SPI0);
-	enable_irq(IRQ_SPI1);
 	preempt_enable();
+	enable_irq(IRQ_SPI1);
+	enable_irq(IRQ_SPI0);
 }
 #else
 void rk29_lcd_spim_spin_lock(void)
-- 
2.35.3

