From 49b81878952d6634de173f61b381418d42c2a9a8 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 12 Sep 2013 11:28:51 +0800
Subject: [PATCH] rk fb:bank lcdc when change jetta scaler timing

---
 drivers/video/rockchip/rk_fb.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index aecca925d42f..190d72e475f6 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1235,7 +1235,10 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 	{
 		if(dev_drv->screen0->sscreen_set)
 		{
+			dev_drv->blank(dev_drv,0,FB_BLANK_NORMAL);
+			msleep(100);
 			dev_drv->screen0->sscreen_set(dev_drv->screen0,enable);
+			dev_drv->blank(dev_drv,0,FB_BLANK_UNBLANK);
 		}
 	}
 #if defined(CONFIG_NO_DUAL_DISP)  //close backlight for device whic do not support dual display
-- 
2.35.3

