From aac38a12173fa02b3f8588f009740974f270efa6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 26 Mar 2014 16:53:16 +0800
Subject: [PATCH] ARM: rockchip: rk3288 set vop qos priority 2,2

---
 arch/arm/boot/dts/rk3288.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index a44d231bccdf..46e95ce7cbbf 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -117,6 +117,7 @@
 			/* service vio */
 			vio1_vop {
 				reg = <0xffad0000 0x20>;
+				rockchip,priority = <2 2>;
 			};
 			vio1_isp_w0 {
 				reg = <0xffad0100 0x20>;
@@ -126,6 +127,7 @@
 			};
 			vio0_vop {
 				reg = <0xffad0400 0x20>;
+				rockchip,priority = <2 2>;
 			};
 			vio0_vip {
 				reg = <0xffad0480 0x20>;
-- 
2.35.3

