From 899286fb6a7a20f5b5c5d0a9a22e4fc39b2d9e70 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 26 May 2015 14:57:22 +0800
Subject: [PATCH] display-sys: function mode_string check input parameter.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/display-sys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/display-sys.c b/drivers/video/rockchip/display-sys.c
index c2229187bc9d..a4b0d982ca28 100755
--- a/drivers/video/rockchip/display-sys.c
+++ b/drivers/video/rockchip/display-sys.c
@@ -77,6 +77,10 @@ static int mode_string(char *buf, unsigned int offset,
 {
 	char v = 'p';
 
+	if (!buf || !mode) {
+		pr_err("%s parameter error %x %x\n", __func__, buf, mode);
+		return 0;
+	}
 	if (mode->xres == 0 && mode->yres == 0)
 		return snprintf(&buf[offset], PAGE_SIZE - offset, "auto\n");
 /*
-- 
2.35.3

