From 5de58df8a6428ec1a140264bba3f2905371d16c2 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 13 Apr 2015 09:39:50 +0800
Subject: [PATCH] rk fb: set retire fence fd to -1 when ignore this frame

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9818c3a7367e..b663a6ea5a96 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -2189,6 +2189,7 @@ static int rk_fb_set_win_config(struct fb_info *info,
 		        pr_err("error config ,ignore\n");
 		for (j = 0; j < RK_MAX_BUF_NUM; j++)
 		        win_data->rel_fence_fd[j] = -1;
+		win_data->ret_fence_fd = -1;
 		return 0;
 	}
 
-- 
2.35.3

