From 8c58cb2ba04264690b84f00f5043cc7e2a5539a5 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 26 Sep 2014 12:38:20 +0800
Subject: [PATCH] mmc: remove useless rescan for io function

---
 drivers/mmc/host/rk_sdmmc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 310b591080ea..cd78befd129a 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -4202,7 +4202,10 @@ int dw_mci_resume(struct dw_mci *host)
 			dw_mci_setup_bus(slot, true);
 		}
 	}
-	schedule_delayed_work(&host->resume_rescan, msecs_to_jiffies(2000));
+
+	if((host->mmc->restrict_caps & RESTRICT_CARD_TYPE_SD))
+		schedule_delayed_work(&host->resume_rescan, msecs_to_jiffies(2000));
+
 	return 0;
 }
 EXPORT_SYMBOL(dw_mci_resume);
-- 
2.35.3

