From bbb4a5e9331d94b8091f24336d07d39f83b4ec0e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:30:29 +0800
Subject: [PATCH] wakelock.h: fix for v2.6.36

---
 include/linux/wakelock.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/wakelock.h b/include/linux/wakelock.h
index a096d24ada1d..492aa649629a 100755
--- a/include/linux/wakelock.h
+++ b/include/linux/wakelock.h
@@ -72,6 +72,7 @@ int wake_lock_active(struct wake_lock *lock);
  * number of jiffies until all active wake locks time out.
  */
 long has_wake_lock(int type);
+void print_active_wake_locks(int type);
 
 #else
 
@@ -84,6 +85,7 @@ static inline void wake_unlock(struct wake_lock *lock) {}
 
 static inline int wake_lock_active(struct wake_lock *lock) { return 0; }
 static inline long has_wake_lock(int type) { return 0; }
+static inline void print_active_wake_locks(int type) {}
 
 #endif
 
-- 
2.35.3

