From fd879d14c3777a917c851bb34012bc61560fa4d9 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Fri, 14 Jun 2013 09:36:27 +0800
Subject: [PATCH] rk3188: for version -t: ddr gpll path always on

---
 arch/arm/mach-rk3188/clock_data.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk3188/clock_data.c b/arch/arm/mach-rk3188/clock_data.c
index 11e10c995898..c04177a7b4b9 100755
--- a/arch/arm/mach-rk3188/clock_data.c
+++ b/arch/arm/mach-rk3188/clock_data.c
@@ -3022,6 +3022,7 @@ static void __init rk30_init_enable_clocks(void)
 	//clk_enable_nolock(&clk_core);
 	clk_enable_nolock(&clk_cpu_div);
 	clk_enable_nolock(&clk_core_gpll_path);
+	clk_enable_nolock(&clk_ddr_gpll_path);
 	clk_enable_nolock(&clk_l2c);
 	clk_enable_nolock(&clk_core_dbg);
 	clk_enable_nolock(&core_periph);
-- 
2.35.3

