From 4474be14d6bf1cf1fa08b6bcce6500760f93544c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 19 Apr 2011 12:43:47 +0800
Subject: [PATCH] pack-kernel.sh: exclude fih board

---
 pack-kernel.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index c8244a830259..7d46be14068e 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -40,7 +40,7 @@ arch/arm/configs/rk29_Winaccord_defconfig
 arch/arm/mach-rk29/board-rk29-a22*
 arch/arm/configs/rk29_a22_defconfig
 
-arch/arm/mach-rk29/board-rk29-fih.c
+arch/arm/mach-rk29/board-rk29-fih*
 arch/arm/configs/rk29_FIH_defconfig
 
 arch/arm/mach-rk29/board-rk29-phonesdk*
@@ -109,7 +109,7 @@ for d in ${DIRS[@]}; do
 	done
 done
 
-make clean >/dev/null 2>&1
+make distclean >/dev/null 2>&1
 
 popd >/dev/null
 
-- 
2.35.3

