From 419e6b0f5beb289d6575acdd1b039cdf5628e1ff Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Tue, 3 Sep 2013 18:45:43 +0800
Subject: [PATCH] add vendor and idb ioremap

---
 drivers/mtd/rknand/rknand_base_ko.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/mtd/rknand/rknand_base_ko.c b/drivers/mtd/rknand/rknand_base_ko.c
index ea49e277ea8a..8a34c2aea003 100755
--- a/drivers/mtd/rknand/rknand_base_ko.c
+++ b/drivers/mtd/rknand/rknand_base_ko.c
@@ -278,6 +278,25 @@ char GetSNSectorInfo(char * pbuf)
     return 0;
 }
 
+
+char GetSNSectorInfoBeforeNandInit(char * pbuf)
+{
+    char * sn_addr = ioremap(0x10501600,0x200);
+    memcpy(pbuf,sn_addr,0x200);
+    iounmap(sn_addr);
+	//print_hex_dump(KERN_WARNING, "sn:", DUMP_PREFIX_NONE, 16,1, sn_addr, 0x200, 0);
+    return 0;
+} 
+
+char GetVendor0InfoBeforeNandInit(char * pbuf)
+{
+    char * sn_addr = ioremap(0x10501400,0x200);
+    memcpy(pbuf,sn_addr + 8,504);
+    iounmap(sn_addr);
+	//print_hex_dump(KERN_WARNING, "sn:", DUMP_PREFIX_NONE, 16,1, sn_addr, 0x200, 0);
+    return 0;
+} 
+
 char GetChipSectorInfo(char * pbuf)
 {
     if(gpNandInfo->GetChipSectorInfo)
-- 
2.35.3

