From 51316e7b6fbeff807ecc42b6e05ad78deeab7f46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 23 Jan 2013 23:12:20 +0800
Subject: [PATCH] rk30_i2s: fix for rk3188, add i2s1 support

---
 sound/soc/rk29/rk30_i2s.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index a10c548c9d1e..62bb21bd6f98 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -678,6 +678,14 @@ static int __devinit rockchip_i2s_probe(struct platform_device *pdev)
 		i2s->dma_playback->dma_addr = RK30_I2S2_2CH_PHYS + I2S_TXR_BUFF;	
 		break;
 #endif
+#if defined(CONFIG_ARCH_RK3188)
+	case 1:
+		i2s->dma_capture->channel = DMACH_I2S1_2CH_RX;
+		i2s->dma_capture->dma_addr = RK30_I2S1_2CH_PHYS + I2S_RXR_BUFF;
+		i2s->dma_playback->channel = DMACH_I2S1_2CH_TX;
+		i2s->dma_playback->dma_addr = RK30_I2S1_2CH_PHYS + I2S_TXR_BUFF;
+		break;
+#endif
 #ifdef CONFIG_ARCH_RK2928
 	case 0:
 		i2s->dma_capture->channel = DMACH_I2S0_8CH_RX;
-- 
2.35.3

