From b514287dc91664abc047bc0b864856d6c9199d35 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 15 Aug 2012 18:39:11 +0800
Subject: [PATCH] rk30:sdk: dvfs add err info when current voltage out of bound

---
 arch/arm/mach-rk30/dvfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index f1fc2eafe714..a2838794d264 100644
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -929,7 +929,7 @@ int dvfs_scale_volt_bystep(struct vd_node *vd_clk, struct vd_node *vd_dep, int v
 
 	if (check_volt_correct(volt_old, &volt_new, volt_dep_old, &volt_dep_new, 
 				clk_biger_than_dep, dep_biger_than_clk) < 0) {
-		DVFS_ERR("CURRENT VOLT INCORRECT\n");
+		DVFS_ERR("CURRENT VOLT INCORRECT, VD=%s, VD_DEP=%s\n", vd_clk->name, vd_dep->name);
 		return -1;
 	}
 	DVFS_DBG("ENTER %s, volt=%d(old=%d), volt_dep=%d(dep_old=%d)\n", __func__, 
-- 
2.35.3

