From 4f787ded317e8e9979985ddb91436665fd60895e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:28:00 +0800
Subject: [PATCH] gadget_chips.h: fix for v2.6.36

---
 drivers/usb/gadget/gadget_chips.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/gadget_chips.h b/drivers/usb/gadget/gadget_chips.h
index edc9bc52c620..410016034619 100644
--- a/drivers/usb/gadget/gadget_chips.h
+++ b/drivers/usb/gadget/gadget_chips.h
@@ -140,6 +140,7 @@
 #define gadget_is_dwc_otg(g)    !strcmp("dwc_otg_pcd", (g)->name)
 #else
 #define gadget_is_dwc_otg(g)    0
+#endif
 
 #ifdef CONFIG_USB_S3C_HSOTG
 #define gadget_is_s3c_hsotg(g)    (!strcmp("s3c-hsotg", (g)->name))
-- 
2.35.3

