From 74d4c03dbd8db92ff3d74012a291dff28c10869d Mon Sep 17 00:00:00 2001
From: heyun <heyun@rock-chips.com>
Date: Fri, 24 Oct 2014 08:43:22 +0800
Subject: [PATCH] rk312x-dts: disable gmac

---
 arch/arm/boot/dts/rk3126-86v.dts | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/arch/arm/boot/dts/rk3126-86v.dts b/arch/arm/boot/dts/rk3126-86v.dts
index 573355a02592..f01125c3b6ce 100644
--- a/arch/arm/boot/dts/rk3126-86v.dts
+++ b/arch/arm/boot/dts/rk3126-86v.dts
@@ -359,13 +359,5 @@
 };
 
 &gmac {
-//      power_ctl_by = "gpio";  //"gpio" "pmu"
-        power-gpio = <&gpio0 GPIO_A6 GPIO_ACTIVE_HIGH>;
-//      power-pmu = "act_ldo"
-//      reset-gpio = <&gpio4 GPIO_A7 GPIO_ACTIVE_LOW>;
-        phy-mode = "rgmii";
-        clock_in_out = "input";
-        tx_delay = <0x28>;
-        rx_delay = <0x10>;
         status = "disabled";
 };
-- 
2.35.3

