From 64974bbd644f8956ebd80e1b3dad5c93ee145cb0 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 9 Jan 2015 17:38:55 +0800
Subject: [PATCH] rk fb: fix compile warning

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 7113efa5b9f6..b9ba562a51d1 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -4234,7 +4234,7 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 
 			return 0;
 		} else if (dev_drv->uboot_logo && uboot_logo_base) {
-			phys_addr_t start = uboot_logo_base;
+			u32 start = uboot_logo_base;
 			int logo_len, i=0;
 			unsigned int nr_pages;
 			struct page **pages;
@@ -4261,8 +4261,8 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 			vaddr = vmap(pages, nr_pages, VM_MAP,
 					pgprot_writecombine(PAGE_KERNEL));
 			if (!vaddr) {
-				pr_err("failed to vmap phy addr %x\n",
-					uboot_logo_base);
+				pr_err("failed to vmap phy addr 0x%llx\n",
+				       uboot_logo_base);
 				return -1;
 			}
 
-- 
2.35.3

