From fc4c39bc30ebca57738fad47880fdb35c5ca1926 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 29 Apr 2014 10:56:55 +0800
Subject: [PATCH] rk fb: fix dts name for vop big and vop little mmu name

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index d0cc4582400d..5d4d7ebf58f1 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -2889,7 +2889,7 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 		fbi->pseudo_palette = dev_drv->win[i]->pseudo_pal;
 		if (i == 0){ /* only alloc memory for main fb*/
 		#ifdef CONFIG_ROCKCHIP_IOMMU
-			mmu_dev = rockchip_get_sysmmu_device_by_compatible("iommu,vopl_mmu");
+			mmu_dev = rockchip_get_sysmmu_device_by_compatible(dev_drv->mmu_dts_name);
 			if (mmu_dev) {
 				platform_set_sysmmu(mmu_dev, dev_drv->dev);
 				rockchip_sysmmu_set_fault_handler(dev_drv->dev, rk_fb_sysmmu_fault_handler);
-- 
2.35.3

