From ecbe57960bce3c2245dd7ae3f8fd4f2ab0d22f2e Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Fri, 14 Sep 2012 17:34:33 +0800
Subject: [PATCH] rk3066b:switch between spk and ear in M701

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index 9f41b9cdd379..94ba8c63dff2 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -1572,6 +1572,10 @@ static void __init machine_rk30_board_init(void)
 #ifdef CONFIG_WIFI_CONTROL_FUNC
 	rk29sdk_wifi_bt_gpio_control_init();
 #endif
+
+	// spk and ear switch
+	gpio_request(RK30_PIN2_PB1, NULL);
+        gpio_direction_output(RK30_PIN2_PB1, GPIO_HIGH);
 }
 
 static void __init rk30_reserve(void)
-- 
2.35.3

