From 5605fd2209999baf68e4014705a91eb6ed4d9342 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 4 Sep 2012 16:38:19 +0800
Subject: [PATCH] adc: export symbol

---
 drivers/adc/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/adc/core.c b/drivers/adc/core.c
index 3e92c3a04c5b..d423782cc60d 100755
--- a/drivers/adc/core.c
+++ b/drivers/adc/core.c
@@ -65,6 +65,7 @@ struct adc_client *adc_register(int chn,
         dev_err(adc->dev, "chn(%d) is not support\n", chn);
         return NULL;
 }
+EXPORT_SYMBOL(adc_register);
 
 void adc_unregister(struct adc_client *client)
 {
@@ -75,6 +76,7 @@ void adc_unregister(struct adc_client *client)
 	client = NULL;
 	return;
 }
+EXPORT_SYMBOL(adc_unregister);
 
 static inline void trigger_next_adc_job_if_any(struct adc_host *adc)
 {
@@ -227,12 +229,14 @@ int adc_sync_read(struct adc_client *client)
 {
         return adc_host_read(client, ADC_SYNC_READ);
 }
+EXPORT_SYMBOL(adc_sync_read);
 
 int adc_async_read(struct adc_client *client)
 {
         return adc_host_read(client, ADC_ASYNC_READ);
 }
 
+EXPORT_SYMBOL(adc_async_read);
 
 static int __init adc_core_init(void)
 {
-- 
2.35.3

