From fdc84d558cc7f848b7c45154ffbb73e1583790c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Mon, 18 Apr 2011 21:58:29 -0700
Subject: [PATCH] modify Kconfig for bq27510

---
 drivers/power/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/Kconfig b/drivers/power/Kconfig
index a5d37157e344..3fc01f9028a7 100755
--- a/drivers/power/Kconfig
+++ b/drivers/power/Kconfig
@@ -118,13 +118,13 @@ config CHARGER_PCF50633
 
 config BATTERY_STC3100
 	tristate "STC3100 battery driver"
-	select I2C_DEV_RK29
 	depends on I2C && ARCH_RK29
 	help
 	  Say Y here to enable support for batteries with STC3100(I2C) chip.
 
 config BATTERY_BQ27510
 	tristate "BQ27510 battery driver"
+	select I2C_DEV_RK29
 	depends on I2C && ARCH_RK29
 	help
 	  Say Y here to enable support for batteries with BQ27510(I2C) chip.
-- 
2.35.3

