From 40481bbdcc46af46936efcdff61cea741541f566 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 15 Aug 2014 14:00:55 +0800
Subject: [PATCH] ARM: rockchip: rk312x: core qos use default value

---
 arch/arm/boot/dts/rk312x.dtsi | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index e0901fbe4a6d..8483401d8b43 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -84,7 +84,6 @@
 			};
 			core {
 				reg = <0x1012a000 0x20>;
-				rockchip,priority = <3 2>;
 			};
 			peri {
 				reg = <0x1012c000 0x20>;
-- 
2.35.3

