From 9ae7ce1331460a570f043eab03f7c1ed2e6434e6 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 4 Jul 2014 10:29:20 +0800
Subject: [PATCH] 3036: default jtag_clk is enable

---
 arch/arm/mach-rockchip/rk3036.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 767b2be24aa4..3dcd886fdf74 100644
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -135,8 +135,6 @@ static void __init rk3036_dt_init_timer(void)
 
 static void __init rk3036_init_late(void)
 {
-    if (rockchip_jtag_enabled)
-    	clk_prepare_enable(clk_get_sys(NULL, "clk_jtag"));
     return;
 }
 
-- 
2.35.3

