From 6d8c70793a7c8010d763f664a2fa70f7054a4f1c Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Thu, 26 Dec 2013 10:32:04 +0800
Subject: [PATCH] fiq debugger: modify switch_cpu()

---
 arch/arm/common/fiq_debugger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/common/fiq_debugger.c b/arch/arm/common/fiq_debugger.c
index f1771c00398c..5f87cb07385e 100644
--- a/arch/arm/common/fiq_debugger.c
+++ b/arch/arm/common/fiq_debugger.c
@@ -739,7 +739,7 @@ static void switch_cpu(struct fiq_debugger_state *state, int cpu)
 {
 	if (!debug_have_fiq(state))
 		smp_call_function_single(cpu, take_affinity, state, false);
-#ifdef CONFIG_PLAT_RK
+#ifdef CONFIG_ARCH_ROCKCHIP
 	else {
 		struct cpumask cpumask;
 
-- 
2.35.3

