From ce04c161114678f5df13e37f660ddaa4590632bb Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Fri, 8 May 2015 15:12:13 +0800
Subject: [PATCH] rk3368: ddrfreq: add compat_ioctl support

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm/mach-rockchip/ddr_freq.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rockchip/ddr_freq.c b/arch/arm/mach-rockchip/ddr_freq.c
index 09a1d80413b7..a6c3ab8b8c59 100644
--- a/arch/arm/mach-rockchip/ddr_freq.c
+++ b/arch/arm/mach-rockchip/ddr_freq.c
@@ -652,6 +652,9 @@ static long ddr_freq_ioctl(struct file *file, unsigned int cmd, unsigned long ar
 static const struct file_operations ddr_freq_fops = {
 	.owner	= THIS_MODULE,
 	.unlocked_ioctl = ddr_freq_ioctl,
+#ifdef CONFIG_COMPAT
+	.compat_ioctl	= ddr_freq_ioctl,
+#endif
 };
 
 static struct miscdevice ddr_freq_dev = {
-- 
2.35.3

