From febe90b54be8e858d412aa92f4d03194b881b361 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 17:01:49 +0800
Subject: [PATCH] input: sensor-dev: fix compilation error on arm64

---
 drivers/input/sensors/sensor-dev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/sensor-dev.c b/drivers/input/sensors/sensor-dev.c
index 960d44f2ee03..cd05bca2fec5 100755
--- a/drivers/input/sensors/sensor-dev.c
+++ b/drivers/input/sensors/sensor-dev.c
@@ -1656,7 +1656,7 @@ int sensor_probe(struct i2c_client *client, const struct i2c_device_id *devid)
 	int result = 0;
 	int type = 0;
 	
-	dev_info(&client->adapter->dev, "%s: %s,0x%x\n", __func__, devid->name,(unsigned int)client);
+	dev_info(&client->adapter->dev, "%s: %s,%p\n", __func__, devid->name, client);
 
 	if (!i2c_check_functionality(client->adapter, I2C_FUNC_I2C)) {
 		result = -ENODEV;
-- 
2.35.3

