From 2b368afa99c169f74f84fbb97c8ebabd2944e1c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Jan 2011 17:15:49 +0800
Subject: [PATCH] rk29: clock: enable debug/jtag

---
 arch/arm/mach-rk29/clock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index ad98a7808c98..34123cb6f383 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2267,6 +2267,8 @@ static void __init clk_enable_init_clocks(void)
 	clk_enable_nolock(&clk_emem);
 	clk_enable_nolock(&clk_intmem);
 	clk_enable_nolock(&clk_ddr);
+	clk_enable_nolock(&clk_debug);
+	clk_enable_nolock(&clk_jtag);
 #ifdef CONFIG_DEBUG_LL
 	clk_enable_nolock(&clk_uart1);
 #endif
-- 
2.35.3

