From da924c4816c290ba8804a7142b03d36d61d00fb6 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Tue, 16 Aug 2011 10:11:12 +0800
Subject: [PATCH] uart:fix couldn't use F5+t function bug

---
 drivers/serial/rk_serial.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/serial/rk_serial.c b/drivers/serial/rk_serial.c
index c12e9db9bb4d..91f35d1669e7 100644
--- a/drivers/serial/rk_serial.c
+++ b/drivers/serial/rk_serial.c
@@ -944,8 +944,13 @@ static void serial_rk_handle_port(struct uart_rk_port *up)
 		 */
 
 		if (status & UART_LSR_RFE) {
-			status = serial_in(up, UART_LSR);
-			dev_info(up->port.dev, "error:lsr=0x%x\n", status);
+			
+			if(up->port.line != DBG_PORT){
+				status = serial_in(up, UART_LSR);
+				dev_info(up->port.dev, "error:lsr=0x%x\n", status);
+			}
+			
+			
 		//	rk29_uart_dump_register(up);
 		}
 
-- 
2.35.3

