From b41ba644446fe9d2a699a2412e2f9b2093106471 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 23 Mar 2012 19:19:32 +0800
Subject: [PATCH] rk30: timer: revert fa827cb change

---
 arch/arm/mach-rk30/timer.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rk30/timer.c b/arch/arm/mach-rk30/timer.c
index 422eb8c466c7..23ab6ef39a35 100755
--- a/arch/arm/mach-rk30/timer.c
+++ b/arch/arm/mach-rk30/timer.c
@@ -203,8 +203,6 @@ static void __init rk30_sched_clock_init(void)
 
 static void __init rk30_timer_init(void)
 {
-
-printk("rk30_timer_init\n");
 #ifdef CONFIG_HAVE_ARM_TWD
 	twd_base = RK30_PTIMER_BASE;
 #endif
-- 
2.35.3

