From 374350cea9d2dca6f8a8558c5cb55b100957288f Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Tue, 26 Apr 2011 17:51:06 +0800
Subject: [PATCH] update for fix a bug in switch win0 win1

---
 drivers/video/rk29_fb.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index e78c36fa7da2..b90ae41b8d82 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1938,6 +1938,8 @@ int fb1_open(struct fb_info *info, int user)
         return -EACCES;
     } else {
         par->refcount++;
+        fb0_set_par(inf->fb0);
+        fb0_pan_display(&inf->fb0->var, inf->fb0);
         win0_blank(FB_BLANK_POWERDOWN, info);
         return 0;
     }
@@ -1956,7 +1958,9 @@ int fb1_release(struct fb_info *info, int user)
         inf->video_mode = 0;
         par->par_seted = 0;
         par->addr_seted = 0;
-        win0_blank(FB_BLANK_POWERDOWN, info);
+        //win0_blank(FB_BLANK_POWERDOWN, info);
+        fb0_set_par(inf->fb0);
+        fb0_pan_display(&inf->fb0->var, inf->fb0);
         win1_blank(FB_BLANK_POWERDOWN, info);
         // wait for lcdc stop access memory
         //msleep(50);
-- 
2.35.3

