From afdaaa54c293de44b6df6a43743b347c2536afe8 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Wed, 5 Sep 2012 18:16:08 +0800
Subject: [PATCH] rk3066b: update rk610 core

---
 drivers/mfd/rk610-core.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/rk610-core.c b/drivers/mfd/rk610-core.c
index 7b1fa0e73b4e..e83f2de8823e 100755
--- a/drivers/mfd/rk610-core.c
+++ b/drivers/mfd/rk610-core.c
@@ -221,8 +221,13 @@ static int rk610_control_probe(struct i2c_client *client,
         return -ENOMEM;
     }
     memset(core_info, 0, sizeof(struct rk610_core_info));
-    
-		iis_clk = clk_get_sys("rk29_i2s.0", "i2s");
+		#if defined(CONFIG_SND_RK29_SOC_I2S_8CH)        
+        	iis_clk = clk_get_sys("rk29_i2s.0", "i2s");
+		#elif defined(CONFIG_SND_RK29_SOC_I2S_2CH)
+		iis_clk = clk_get_sys("rk29_i2s.1", "i2s");
+		#else
+        	iis_clk = clk_get_sys("rk29_i2s.2", "i2s");
+		#endif
 		if (IS_ERR(iis_clk)) {
 			printk("failed to get i2s clk\n");
 			ret = PTR_ERR(iis_clk);
-- 
2.35.3

