From 85243104836bcc58335166b9a03e2f0425da7d83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 5 May 2011 11:27:25 +0800
Subject: [PATCH] backlight: Kconfig: FIH_TOUCHKEY_LED and AW9364 default n

---
 drivers/video/backlight/Kconfig | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/backlight/Kconfig b/drivers/video/backlight/Kconfig
index 6d7fc6783dce..07f29590de62 100644
--- a/drivers/video/backlight/Kconfig
+++ b/drivers/video/backlight/Kconfig
@@ -275,16 +275,15 @@ config BACKLIGHT_RK29_BL
 	default y
 	help
 	  rk29 backlight support.	  
+
 config FIH_TOUCHKEY_LED
         bool "fih touch key led driver"
 	depends on BACKLIGHT_CLASS_DEVICE && ARCH_RK29
-	default y
 	help
 	  fih touch key led support.
 
 config BACKLIGHT_AW9364
         bool "aw9364 backlight driver"
 	depends on BACKLIGHT_CLASS_DEVICE && ARCH_RK29
-	default y
 	help
 	  aw9364 backlight support.	  
-- 
2.35.3

