From 7f78b843e24298f23696ed71b90ac6d992bd0e1b Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Thu, 13 Sep 2012 11:15:37 +0800
Subject: [PATCH] mtp: solve open_excl lock problem

---
 drivers/usb/gadget/f_mtp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/gadget/f_mtp.c b/drivers/usb/gadget/f_mtp.c
index 034cdf68f9f6..363e7c0d0add 100755
--- a/drivers/usb/gadget/f_mtp.c
+++ b/drivers/usb/gadget/f_mtp.c
@@ -1165,6 +1165,8 @@ static int mtp_function_set_alt(struct usb_function *f,
 		return ret;
 	}
 	dev->state = STATE_READY;
+	
+	atomic_set(&dev->open_excl, 0); // solve open_excl lock problem, add by Huweiguo
 
 	/* readers may be blocked waiting for us to go online */
 	wake_up(&dev->read_wq);
-- 
2.35.3

