From 13b95ebabe72c165e82c06ab54b40158937a185d Mon Sep 17 00:00:00 2001
From: Alpha Lin <alpha.lin@rock-chips.com>
Date: Fri, 10 Jul 2015 11:45:39 +0800
Subject: [PATCH] RK3288 VPU: add cru reset resource.

Signed-off-by: Alpha Lin <alpha.lin@rock-chips.com>
---
 arch/arm/boot/dts/rk3288.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index bd78759977ba..a83dac9bcfea 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1051,6 +1051,8 @@
 		interrupt-names = "irq_enc", "irq_dec";
 		clocks = <&clk_vdpu>, <&hclk_vdpu>;
 		clock-names = "aclk_vcodec", "hclk_vcodec";
+		resets = <&reset RK3288_SOFT_RST_VCODEC_H>, <&reset RK3288_SOFT_RST_VCODEC_A>;
+		reset-names = "video_h", "video_a";
 		name = "vpu_service";
 		dev_mode = <0>;
 		//status = "disabled";
@@ -1064,6 +1066,9 @@
 		interrupt-names = "irq_dec";
 		clocks = <&aclk_hevc>, <&hclk_hevc>, <&clk_hevc_core>, <&clk_hevc_cabac>;
 		clock-names = "aclk_vcodec", "hclk_vcodec", "clk_core", "clk_cabac";
+		resets = <&reset RK3288_SOFT_RST_VCODEC_H>, <&reset RK3288_SOFT_RST_VCODEC_A>,
+			<&reset RK3288_SOFT_RST_HEVC>;
+		reset-names = "video_h", "video_a", "video";
 		dev_mode = <1>;
 		name = "hevc_service";
 		//status = "disabled";
-- 
2.35.3

