From 1ce15dac9d2ea09a97764bc4b82de64d0add8a8a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 15 Jan 2013 16:07:45 +0800
Subject: [PATCH] rk3188: add pwm support

---
 arch/arm/plat-rk/pwm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-rk/pwm.c b/arch/arm/plat-rk/pwm.c
index c4d01cb46504..40d0d06c5269 100644
--- a/arch/arm/plat-rk/pwm.c
+++ b/arch/arm/plat-rk/pwm.c
@@ -19,7 +19,7 @@ struct clk *rk_pwm_get_clk(unsigned id)
 #if defined(CONFIG_ARCH_RK29)
 	if (id < 4)
 		return clk_get(NULL, "pwm");
-#elif defined(CONFIG_ARCH_RK30)
+#elif defined(CONFIG_ARCH_RK30) || defined(CONFIG_ARCH_RK3188)
 	if (id == 0 || id == 1)
 		return clk_get(NULL, "pwm01");
 	else if (id== 2 || id == 3)
@@ -39,7 +39,7 @@ void __iomem *rk_pwm_get_base(unsigned id)
 #if defined(CONFIG_ARCH_RK29)
 	if (id < 4)
 		return RK29_PWM_BASE + id * 0x10;
-#elif defined(CONFIG_ARCH_RK30)
+#elif defined(CONFIG_ARCH_RK30) || defined(CONFIG_ARCH_RK3188)
 	if (id == 0 || id == 1)
 		return RK30_PWM01_BASE + id * 0x10;
 	else if (id== 2 || id == 3)
-- 
2.35.3

