From 0fa2fd8d71c865453b2e632fe840c730d3d5a91b Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 1 Feb 2013 17:52:27 +0800
Subject: [PATCH] rk fb:stop  vsync thread when rk fb shut down

---
 drivers/video/rockchip/rk_fb.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9ff5f6f13f48..d632d8b5cefe 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1352,7 +1352,16 @@ static int __devexit rk_fb_remove(struct platform_device *pdev)
 
 static void rk_fb_shutdown(struct platform_device *pdev)
 {
-//	struct rk_fb_inf *fb_inf = platform_get_drvdata(pdev);
+	struct rk_fb_inf *inf = platform_get_drvdata(pdev);
+	int i;
+	for(i = 0; i < inf->num_lcdc; i++)
+	{
+		if (!inf->lcdc_dev_drv[i])
+			continue;
+
+		if(inf->lcdc_dev_drv[i]->vsync_info.thread)
+			kthread_stop(inf->lcdc_dev_drv[i]->vsync_info.thread);
+	}
 //	kfree(fb_inf);
 //	platform_set_drvdata(pdev, NULL);
 #ifdef CONFIG_HAS_EARLYSUSPEND
-- 
2.35.3

