From fc3bb66391188134a5b11f2a8a60c812c36e8875 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 21 Mar 2014 17:46:38 +0800
Subject: [PATCH] rk3288:pwm_regulator:set pwm_vol 1.0v when pwm_regulator_init

---
 drivers/regulator/rockchip-pwm-regulator.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/regulator/rockchip-pwm-regulator.c b/drivers/regulator/rockchip-pwm-regulator.c
index 78451a795e21..15d3acfcfc95 100755
--- a/drivers/regulator/rockchip-pwm-regulator.c
+++ b/drivers/regulator/rockchip-pwm-regulator.c
@@ -35,7 +35,7 @@
 #include <linux/regulator/machine.h>
 
 
-#if 1
+#if 0
 #define DBG(x...)	printk(KERN_INFO x)
 #else
 #define DBG(x...)
@@ -369,12 +369,14 @@ static int __init pwm_regulator_probe(struct platform_device *pdev)
 		goto err;
 		}
 		pdata->rdev[i] = pwm_rdev;
+		
+		/*********set pwm vol by defult***********/
+		dc = regulator_get(NULL,rail_name);
+		regulator_set_voltage(dc,pdata->pwm_voltage,pdata->pwm_voltage);
+		regulator_put(dc);
+		/**************************************/
 		}
 	}
-	/*********set pwm vol by defult***********/
-	//dc = regulator_get(NULL,"vdd_gpu");
-	//regulator_set_voltage(dc,pdata->pwm_voltage,pdata->pwm_voltage);
-	/**************************************/
 	return 0;
 err:
 	printk("%s:error\n",__func__);
-- 
2.35.3

