From 9ba1b1bcc5e74874f49f6a76791128f6c6b4b6c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 29 Aug 2013 09:48:59 +0800
Subject: [PATCH]  rk: cpufreq: interactive: adjust default value

---
 drivers/cpufreq/cpufreq_interactive.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/cpufreq/cpufreq_interactive.c b/drivers/cpufreq/cpufreq_interactive.c
index 07c2f3b94422..fa55603a0a95 100755
--- a/drivers/cpufreq/cpufreq_interactive.c
+++ b/drivers/cpufreq/cpufreq_interactive.c
@@ -931,6 +931,13 @@ static int cpufreq_governor_interactive(struct cpufreq_policy *policy,
 				hispeed_freq = 816000;
 			else if (cpufreq_frequency_table_target(policy, freq_table, policy->min + 1, CPUFREQ_RELATION_L, &index) == 0)
 				hispeed_freq = freq_table[index].frequency;
+			if (policy->max > 1416000) {
+				timer_slack_val = 20000;
+				min_sample_time = 40000;
+				above_hispeed_delay_val = 80000;
+				store_target_loads(NULL, NULL, "70 1200000:80 1416000:99", 0);
+			}
+			boostpulse_duration_val = 500000;
 		}
 #else
 			hispeed_freq = policy->max;
-- 
2.35.3

