From 111efae5e70e19d15e845df3343a129f6efabc38 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 29 Oct 2013 15:45:23 +0800
Subject: [PATCH] rkemmc: if card is emmc, msdos_partition return false

---
 fs/partitions/msdos.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/fs/partitions/msdos.c b/fs/partitions/msdos.c
index a138e5e8ab09..bb7f006f20f8 100644
--- a/fs/partitions/msdos.c
+++ b/fs/partitions/msdos.c
@@ -440,6 +440,11 @@ int msdos_partition(struct parsed_partitions *state)
 	struct partition *p;
 	struct fat_boot_sector *fb;
 	int slot;
+#ifdef CONFIG_EMMC_RK
+	//if card is emmc, msdos_partition return false
+	if(state->bdev->bd_disk->first_minor == 0)
+		return 0;
+#endif
 
 	data = read_part_sector(state, 0, &sect);
 	if (!data)
-- 
2.35.3

