From b9079b1855537e7fcd6af6e69a0fd5cff4a6ad2a Mon Sep 17 00:00:00 2001
From: ddl <ddl@rock-chips.com>
Date: Wed, 27 Mar 2013 15:05:24 +0800
Subject: [PATCH] camera: ov5642 wait af finish when capture

---
 drivers/media/video/ov5642.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/video/ov5642.c b/drivers/media/video/ov5642.c
index e64f3b8cf998..bcdde053b4d7 100755
--- a/drivers/media/video/ov5642.c
+++ b/drivers/media/video/ov5642.c
@@ -4812,8 +4812,8 @@ static int sensor_s_fmt(struct v4l2_subdev *sd, struct v4l2_mbus_framefmt *mf)
                 if (sensor_af_workqueue_set(icd, WqCmd_af_update_zone,0,true) == 0) {
                     if ((sensor->info_priv.auto_focus == SENSOR_AF_MODE_AUTO) || 
             		    (sensor->info_priv.auto_focus == SENSOR_AF_MODE_CONTINUOUS)) {
-            		    msleep(80);
-                        sensor_af_workqueue_set(icd, WqCmd_af_single,0,false);
+            		    //msleep(80);
+                        sensor_af_workqueue_set(icd, WqCmd_af_single,0,true);
             		}
         		}
             }
-- 
2.35.3

