From 13c0df69eac90a92d347e843429050157d3dc428 Mon Sep 17 00:00:00 2001
From: wuhao <wuhao@wuhao@rock-chips.com>
Date: Fri, 25 Jan 2013 15:46:15 +0800
Subject: [PATCH] rk3188 ds1006h: hym8563 shutdown do not disable clkout

---
 drivers/rtc/rtc-HYM8563.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-HYM8563.c b/drivers/rtc/rtc-HYM8563.c
index 7d6aa26c3c3a..01399e8794dd 100755
--- a/drivers/rtc/rtc-HYM8563.c
+++ b/drivers/rtc/rtc-HYM8563.c
@@ -538,7 +538,7 @@ static struct i2c_driver hym8563_driver = {
 	},
 	.probe		= hym8563_probe,
 	.remove		= __devexit_p(hym8563_remove),
-#if defined(CONFIG_ARCH_RK3066B)
+#if defined(CONFIG_ARCH_RK3066B) || defined(CONFIG_ARCH_RK3188)
 	//.shutdown=hym8563_shutdown,
 #else
 	.shutdown=hym8563_shutdown,
-- 
2.35.3

