From a98a934641a28d26c75b8b7effe2d7a0fac67900 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 20 Feb 2012 14:24:32 +0800
Subject: [PATCH] arm: gic: port to rk platform

---
 arch/arm/common/gic.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/common/gic.c b/arch/arm/common/gic.c
index 9e74d12057f1..be20b306d94a 100755
--- a/arch/arm/common/gic.c
+++ b/arch/arm/common/gic.c
@@ -92,7 +92,7 @@ static void gic_mask_irq(struct irq_data *d)
 	writel_relaxed(mask, gic_dist_base(d) + GIC_DIST_ENABLE_CLEAR + (gic_irq(d) / 32) * 4);
 	if (gic_arch_extn.irq_mask)
 		gic_arch_extn.irq_mask(d);
-#ifdef CONFIG_ARCH_RK29
+#if defined(CONFIG_PLAT_RK) && !defined(CONFIG_SMP)
 	dsb();
 #endif
 	spin_unlock(&irq_controller_lock);
@@ -106,7 +106,7 @@ static void gic_unmask_irq(struct irq_data *d)
 	if (gic_arch_extn.irq_unmask)
 		gic_arch_extn.irq_unmask(d);
 	writel_relaxed(mask, gic_dist_base(d) + GIC_DIST_ENABLE_SET + (gic_irq(d) / 32) * 4);
-#ifdef CONFIG_ARCH_RK29
+#if defined(CONFIG_PLAT_RK) && !defined(CONFIG_SMP)
 	dsb();
 #endif
 	spin_unlock(&irq_controller_lock);
@@ -121,7 +121,7 @@ static void gic_eoi_irq(struct irq_data *d)
 	}
 
 	writel_relaxed(gic_irq(d), gic_cpu_base(d) + GIC_CPU_EOI);
-#ifdef CONFIG_ARCH_RK29
+#ifdef CONFIG_PLAT_RK
 	dsb();
 #endif
 }
@@ -212,7 +212,7 @@ static int gic_set_wake(struct irq_data *d, unsigned int on)
 {
 	int ret = -ENXIO;
 
-#ifdef CONFIG_ARCH_RK29
+#ifdef CONFIG_PLAT_RK
 	return 0;
 #endif
 	if (gic_arch_extn.irq_set_wake)
-- 
2.35.3

