From e79fdf73299fb00731b3fe8ca2684eeb3e648e2d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:21:06 +0800
Subject: [PATCH] dbg: fix for v2.6.36

---
 drivers/dbg/debug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dbg/debug.c b/drivers/dbg/debug.c
index 91f05c4dc9ee..7c60d2a8b11f 100644
--- a/drivers/dbg/debug.c
+++ b/drivers/dbg/debug.c
@@ -22,7 +22,7 @@
 #include <linux/kallsyms.h>
 #include <linux/delay.h>
 #include <linux/version.h>
-
+#include <linux/slab.h>
 
 /*
  * 20090725,hsl,some debug function start from here
-- 
2.35.3

