From ff1c6cb48714a1a0b19240282264fe1f028fb4be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 12 Dec 2011 22:53:01 +0800
Subject: [PATCH] inv_mpu: do not select the driver by default

---
 drivers/misc/inv_mpu/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/misc/inv_mpu/Kconfig b/drivers/misc/inv_mpu/Kconfig
index 53c7c200d453..43f8881192bf 100644
--- a/drivers/misc/inv_mpu/Kconfig
+++ b/drivers/misc/inv_mpu/Kconfig
@@ -9,7 +9,6 @@ config MPU_SENSORS_TIMERIRQ
 menuconfig: INV_SENSORS
 	tristate "Motion Processing Unit"
 	depends on I2C
-	default y
 
 if INV_SENSORS
 
-- 
2.35.3

