From c768774b04ce994ed0b7ab1722ec7a80ec211ad6 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 12 Dec 2014 10:41:03 +0800
Subject: [PATCH] rk fb: delete unused message

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 457394d9374d..be3e5cf80c4b 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -841,8 +841,6 @@ static int get_extend_fb_id(struct fb_info *info)
 		fb_id = 3;
 	else if (!strcmp(id, "fb4") && (dev_drv->lcdc_win_num > 4))
 		fb_id = 4;
-	else
-		dev_err(dev_drv->dev, "get_extend_fb_id info error\n");
 	return fb_id;
 }
 
-- 
2.35.3

