From 0ee1cefb2f26426e2d97d040bfd5e86e7e976413 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 7 Jun 2013 09:55:00 +0800
Subject: [PATCH] rk30/rk31: set gpu qos from 0xd to 9

---
 arch/arm/mach-rk30/common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 39930f456a10..68a0c5f8f666 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -36,6 +36,7 @@ static void __init rk30_cpu_axi_init(void)
 #endif
 	CPU_AXI_SET_QOS_PRIORITY(3, 3, LCDC0);
 	CPU_AXI_SET_QOS_PRIORITY(3, 3, LCDC1);
+	CPU_AXI_SET_QOS_PRIORITY(2, 1, GPU);
 
 	writel_relaxed(0x3f, RK30_CPU_AXI_BUS_BASE + 0x0014);	// memory scheduler read latency
 	dsb();
-- 
2.35.3

