From 3149d13f5adf14009c83fac1858e718b508741db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Wed, 17 Oct 2012 17:55:43 +0800
Subject: [PATCH] rk2926:m713:slove io conflict for act8931 and tps65910

---
 drivers/mfd/tps65910.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mfd/tps65910.c b/drivers/mfd/tps65910.c
index 6b16534548ad..181d68aa5902 100755
--- a/drivers/mfd/tps65910.c
+++ b/drivers/mfd/tps65910.c
@@ -268,6 +268,12 @@ static int tps65910_i2c_probe(struct i2c_client *i2c,
 			      NULL, 0);
 	if (ret < 0)
 		goto err;
+
+	ret = tps65910_reg_read(tps65910,0x22);
+	if ((ret < 0) || (ret == 0xff)){
+		printk("The device is not tps65910\n");
+		goto err;
+	}
 	
 	g_tps65910 = tps65910;
 	
-- 
2.35.3

