From c901d947ae7c33568a9a082106637390b64e3506 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Sat, 20 Sep 2014 15:37:49 +0800
Subject: [PATCH] mmc: cd-gpio: wakeup system wheter debounce or not

Signed-off-by: lintao <lintao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index a20fec90a7c7..8ea7e838c8e8 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3173,6 +3173,8 @@ static irqreturn_t dw_mci_gpio_cd_irqt(int irq, void *dev_id)
         mmc_detect_change(mmc, msecs_to_jiffies(200));
         #endif
 
+        /* wakeup system whether gpio debounce or not */
+        rk_send_wakeup_key();
         queue_work(host->card_workqueue, &host->card_work);
         return IRQ_HANDLED;
 }
-- 
2.35.3

