From 2b4b8a080ab355778353ea55314b1d903a4cd843 Mon Sep 17 00:00:00 2001
From: dkl <dkl@rock-chips.com>
Date: Tue, 1 Apr 2014 11:51:35 +0800
Subject: [PATCH] clk: rockchip: rk3288: fix APLL 48M\126M setting

---
 drivers/clk/rockchip/clk-pll.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/rockchip/clk-pll.c b/drivers/clk/rockchip/clk-pll.c
index df21324825ee..cceafebaa888 100644
--- a/drivers/clk/rockchip/clk-pll.c
+++ b/drivers/clk/rockchip/clk-pll.c
@@ -145,8 +145,8 @@ static const struct apll_clk_set rk3288_apll_table[] = {
 	_RK3288_APLL_SET_CLKS(312,	1,	52,	4,	2,      2,      4,      4,      4),
 	_RK3288_APLL_SET_CLKS(252,	1,	84,	8,	2,      2,      4,      4,      4),
 	_RK3288_APLL_SET_CLKS(216,	1,	72,	8,	2,      2,      4,      4,      4),
-	_RK3288_APLL_SET_CLKS(126,	1,	84,	16,	2,      2,      4,      4,      4),
-	_RK3288_APLL_SET_CLKS(48,  	1,	32,	16,	2,      2,      4,      4,      4),
+	_RK3288_APLL_SET_CLKS(126,	2,	84,	8,	2,      2,      4,      4,      4),
+	_RK3288_APLL_SET_CLKS(48,  	2,	32,	8,	2,      2,      4,      4,      4),
 	_RK3288_APLL_SET_CLKS(0,	1,	32,	16,	2,      2,      4,      4,      4),
 };
 
-- 
2.35.3

