From 1143c5e145eb78cb3e6262e8f5f55d7b2aa5adee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Tue, 21 Sep 2010 04:56:38 -0700
Subject: [PATCH]  fix the Response of PWR key in resume

---
 drivers/input/keyboard/rk2818_adckey.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/rk2818_adckey.c b/drivers/input/keyboard/rk2818_adckey.c
index 8c35be83040c..686c892695b0 100755
--- a/drivers/input/keyboard/rk2818_adckey.c
+++ b/drivers/input/keyboard/rk2818_adckey.c
@@ -136,7 +136,7 @@ static void rk28_adkeyscan_timer(unsigned long data)
 	{
 		if(++gPlayCount > 20000)
 			gPlayCount = 101;
-		if((2 == gPlayCount) && (0 == gFlagShortPlay))
+		if((1 == gPlayCount) && (0 == gFlagShortPlay))
 		{
 			gFlagShortPlay = 1;			
 		}
-- 
2.35.3

