From 649662820c197eed870d80d54a6ae6689d964de7 Mon Sep 17 00:00:00 2001
From: qjb <qjb@rock-chips.com>
Date: Tue, 10 Jun 2014 09:42:08 +0800
Subject: [PATCH] pl330 : fix ifdef error

---
 drivers/dma/pl330.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/pl330.c b/drivers/dma/pl330.c
index 3eba20adcccc..2ea17d0f2970 100755
--- a/drivers/dma/pl330.c
+++ b/drivers/dma/pl330.c
@@ -3188,7 +3188,7 @@ pl330_probe(struct amba_device *adev, const struct amba_id *id)
 	pd->device_prep_slave_sg = pl330_prep_slave_sg;
 	pd->device_control = pl330_control;
 	pd->device_issue_pending = pl330_issue_pending;
-#ifdef drivers/dma/pl330.c
+#ifdef CONFIG_ARCH_ROCKCHIP
 	pd->dma_getposition = pl330_dma_getposition;
 #endif
 
-- 
2.35.3

