From a61f27e2ba0a4b3c08fa8976a0a3d66da73261ad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 25 Apr 2014 18:32:15 +0800
Subject: [PATCH] ARM: rockchip: rk3288: pll enter slow mode before reboot

---
 arch/arm/mach-rockchip/rk3288.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index 4299020808ec..c74793df4cf9 100644
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -426,6 +426,9 @@ static void rk3288_restart(char mode, const char *cmd)
 	writel_relaxed(boot_mode, RK_PMU_VIRT + RK3288_PMU_SYS_REG1);	// for linux
 	dsb();
 
+	/* pll enter slow mode */
+	writel_relaxed(0xf3030000, RK_CRU_VIRT + RK3288_CRU_MODE_CON);
+	dsb();
 	writel_relaxed(0xeca8, RK_CRU_VIRT + RK3288_CRU_GLB_SRST_SND_VALUE);
 	dsb();
 }
-- 
2.35.3

