From c187c7656096d17bcf1c79c536957fd617234551 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 27 Jun 2012 16:23:31 +0800
Subject: [PATCH] rk30: record arm voltage value before dvfs start

---
 arch/arm/mach-rk30/dvfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 180966847590..c4dc325eddc4 100644
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -337,6 +337,7 @@ int clk_enable_dvfs(struct clk *clk)
 			if (!IS_ERR(regulator)) {
 				// DVFS_DBG("dvfs_regulator_get(%s)\n",dvfs_clk->vd->regulator_name);
 				dvfs_clk->vd->regulator = regulator;
+				dvfs_clk->vd->cur_volt = dvfs_regulator_get_voltage(dvfs_clk->vd->regulator);
 			} else {
 				dvfs_clk->vd->regulator = NULL;
 				dvfs_clk->enable_dvfs = 0;
-- 
2.35.3

