From 46f85025633fe28c3d80b41bbdbbbfab59b358b7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 28 Oct 2010 09:08:11 +0800
Subject: [PATCH] rk29: clock: disable debug

---
 arch/arm/mach-rk29/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 2452d0f035a3..bad616833a95 100644
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -13,7 +13,7 @@
  *
  */
 
-#define DEBUG
+//#define DEBUG
 #define pr_fmt(fmt) "clock: %s: " fmt, __func__
 
 #include <linux/clk.h>
-- 
2.35.3

