From 7646793a44989c163380f105e0594feb024003b8 Mon Sep 17 00:00:00 2001
From: David Wu <wdc@rock-chips.com>
Date: Tue, 16 Jun 2015 15:16:49 +0800
Subject: [PATCH] rk3368: pinctrl: fix bank:3 gpio irq wakeup not worked

Signed-off-by: David Wu <wdc@rock-chips.com>
---
 drivers/pinctrl/pinctrl-rk3368.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-rk3368.c b/drivers/pinctrl/pinctrl-rk3368.c
index 2ed7c1d3e45f..5dfb37d1d7f4 100755
--- a/drivers/pinctrl/pinctrl-rk3368.c
+++ b/drivers/pinctrl/pinctrl-rk3368.c
@@ -2083,7 +2083,7 @@ static int rockchip_pinctrl_suspend(void)
 	int n;
 	//int value = 0;
 		
-	for(n=0; n<ctrl->nr_banks-1; n++)
+	for(n=0; n<ctrl->nr_banks; n++)
 	{
 #if 0
 		int i;
@@ -2117,7 +2117,7 @@ static void rockchip_pinctrl_resume(void)
 	int n;
 	u32 isr;
 
-	for(n=0; n<ctrl->nr_banks-1; n++)
+	for(n=0; n<ctrl->nr_banks; n++)
 	{
 #if 0
 		int i;
-- 
2.35.3

