From c1b7a0d9463104ec8acac7e43d5144db21484889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Fri, 28 Sep 2012 12:03:43 +0800
Subject: [PATCH] add I2C_ADDRESS pre-read,slove the bug of suspend

---
 arch/arm/mach-rk30/pm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index cccf0e5cb0f1..ddb7b66792f3 100644
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -229,6 +229,12 @@ static void pm_pll_wait_lock(int pll_idx)
 	u32 bit = 0x10u << pll_state[pll_idx];
 #endif
 	u32 delay = 2400000U;
+	dsb();
+	dsb();
+	dsb();
+	dsb();
+	dsb();
+	dsb();
 	while (delay > 0) {
 		if (grf_readl(GRF_SOC_STATUS0) & bit)
 			break;
@@ -271,7 +277,7 @@ static noinline void interface_ctr_reg_pread(void)
 	readl_relaxed(RK30_GRF_BASE);
 	readl_relaxed(RK30_DDR_PCTL_BASE);
 	readl_relaxed(RK30_DDR_PUBL_BASE);
-	readl_relaxed(RK30_I2C1_BASE);
+	readl_relaxed(RK30_I2C1_BASE+SZ_4K);
 }
 
 static inline bool pm_pmu_power_domain_is_on(enum pmu_power_domain pd, u32 pmu_pwrdn_st)
-- 
2.35.3

