From fd8103d56ad8b98f28b9a665dd4f3eefd9d90be2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 7 May 2012 20:11:36 +0800
Subject: [PATCH] rk30: clock: clk_get_parent return error when clk invalid

---
 arch/arm/mach-rk30/clock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rk30/clock.c b/arch/arm/mach-rk30/clock.c
index 572e59e2f9ed..6950186b5cf3 100755
--- a/arch/arm/mach-rk30/clock.c
+++ b/arch/arm/mach-rk30/clock.c
@@ -445,6 +445,9 @@ struct clk *rk30_clk_get_parent(struct clk *clk)
 struct clk *clk_get_parent(struct clk *clk)
 #endif
 {
+	if (clk == NULL || IS_ERR(clk)) {
+		return ERR_PTR(-EINVAL);
+	}
 	return clk->parent;
 }
 
-- 
2.35.3

