From e6537363e7b3ce79029a98762758f87eb49fb408 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 21 Feb 2011 09:46:25 +0800
Subject: [PATCH] i2c timeout: 10ms ==> 100ms

---
 drivers/i2c/busses/i2c-rk29.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index a8cba27ef59b..1d0be8d550dd 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -36,7 +36,7 @@
 									
 #define RK29_UDELAY_TIME(scl_rate)			((400*1000)/(scl_rate))
 /*max ACK delay time = RK29_I2C_ACK_TIMEOUT_COUNT * RK29_UDELAY_TIME(scl_rate)   us */
-#define RK29_I2C_ACK_TIMEOUT_COUNT			(10 * 1000)
+#define RK29_I2C_ACK_TIMEOUT_COUNT			(100 * 1000)
 /*max STOP delay time = RK29_I2C_STOP_TIMEOUT_COUNT * RK29_UDELAY_TIME(scl_rate)   us */
 #define RK29_I2C_STOP_TIMEOUT_COUNT			1000
 /*max START delay time = RK29_I2C_START_TIMEOUT_COUNT * RK29_UDELAY_TIME(scl_rate)   us */
-- 
2.35.3

