From 16ebedae0e4314ec34242649bc32baba5bb24df4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 26 Jul 2010 17:59:04 +0800
Subject: [PATCH] =?UTF-8?q?.config=E4=B8=8D=E5=BA=94=E8=AF=A5=E8=A2=AB?=
 =?UTF-8?q?=E5=BF=BD=E7=95=A5?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 28d040bd9366..5aae20242d80 100644
--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,7 @@
 # Normal rules
 #
 .*
+!.config
 *.o
 *.o.*
 *.a
-- 
2.35.3

