From c666de7e1fdf0dc1f239fa45688ba214fd9ebf78 Mon Sep 17 00:00:00 2001
From: typ <typ@rock-chips.com>
Date: Sat, 24 Aug 2013 12:08:40 +0800
Subject: [PATCH] RK3028A ddr:save sram capacity

---
 arch/arm/mach-rk2928/ddr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk2928/ddr.c b/arch/arm/mach-rk2928/ddr.c
index 6872f7bf761d..25418c10cb06 100755
--- a/arch/arm/mach-rk2928/ddr.c
+++ b/arch/arm/mach-rk2928/ddr.c
@@ -771,7 +771,7 @@ typedef struct BACKUP_REG_Tag
 __sramdata BACKUP_REG_T ddr_reg;
 
 
-uint32_t __sramdata ddr3_cl_cwl[22][4]={
+uint32_t  ddr3_cl_cwl[22][4]={
 /*   0~330           330~400         400~533        speed
 * tCK  >3             2.5~3          1.875~2.5     1.875~1.5
 *    cl<<16, cwl    cl<<16, cwl     cl<<16, cwl              */
@@ -805,7 +805,7 @@ uint32_t __sramdata ddr3_cl_cwl[22][4]={
     {((6<<16)|5),   ((6<<16)|5),    ((8<<16)|6),   ((10<<16)|7)} //DDR3_DEFAULT
 
 };
-uint32_t __sramdata ddr3_tRC_tFAW[22]={
+uint32_t  ddr3_tRC_tFAW[22]={
 /**    tRC    tFAW   */
     ((50<<16)|50), //DDR3_800D
     ((53<<16)|50), //DDR3_800E
-- 
2.35.3

