From ed0041b98a96ba7879ef12a4a792883698ac3c90 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 4 Jul 2014 09:22:44 +0800
Subject: [PATCH] 3036:for cru slow mode

---
 arch/arm/mach-rockchip/rk3036.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index c9d21e7b8be1..97f5b7657ab1 100644
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -159,10 +159,10 @@ static void rk3036_restart(char mode, const char *cmd)
 	dsb();
 
 	/* pll enter slow mode */
-	//writel_relaxed(0xf3030000, RK_CRU_VIRT + RK3288_CRU_MODE_CON);
-	//dsb();
-	//writel_relaxed(0xeca8, RK_CRU_VIRT + RK3288_CRU_GLB_SRST_SND_VALUE);
-	//dsb();
+	writel_relaxed(0x30010000, RK_CRU_VIRT + RK3036_CRU_MODE_CON);
+	dsb();
+	writel_relaxed(0xeca8, RK_CRU_VIRT + RK3036_CRU_GLB_SRST_SND_VALUE);
+	dsb();
 }
 
 static const char * const rk3036_dt_compat[] __initconst = {
-- 
2.35.3

