From 809bc041c8a6b9f89fde7ee2905c08dd8451ca55 Mon Sep 17 00:00:00 2001
From: "dalon.zhang" <dalon.zhang@rock-chips.com>
Date: Fri, 17 Jul 2015 15:12:06 +0800
Subject: [PATCH] uvc: support the query of iommu_enabled

---
 drivers/media/usb/uvc/uvc_v4l2.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/media/usb/uvc/uvc_v4l2.c b/drivers/media/usb/uvc/uvc_v4l2.c
index b2dc32623a71..196198fe873f 100644
--- a/drivers/media/usb/uvc/uvc_v4l2.c
+++ b/drivers/media/usb/uvc/uvc_v4l2.c
@@ -28,7 +28,7 @@
 #include <media/v4l2-ctrls.h>
 #include <media/v4l2-event.h>
 #include <media/v4l2-ioctl.h>
-
+#include <linux/of.h>
 #include "uvcvideo.h"
 
 /* ------------------------------------------------------------------------
@@ -558,6 +558,8 @@ static long uvc_v4l2_do_ioctl(struct file *file, unsigned int cmd, void *arg)
 	case VIDIOC_QUERYCAP:
 	{
 		struct v4l2_capability *cap = arg;
+		struct device_node *vpu_node;
+		int vpu_iommu_enabled = 0;
 
 		memset(cap, 0, sizeof *cap);
 		strlcpy(cap->driver, "uvcvideo", sizeof cap->driver);
@@ -573,6 +575,13 @@ static long uvc_v4l2_do_ioctl(struct file *file, unsigned int cmd, void *arg)
 		else
 			cap->device_caps = V4L2_CAP_VIDEO_OUTPUT
 					 | V4L2_CAP_STREAMING;
+        vpu_node = of_find_node_by_name(NULL, "vpu_service");
+    	if(vpu_node){
+        	ret = of_property_read_u32(vpu_node, "iommu_enabled", &vpu_iommu_enabled);
+    	}else{
+        	printk("get vpu_node failed,vpu_iommu_enabled == 0 !!!!!!\n");
+    	}
+        cap->reserved[0] = vpu_iommu_enabled;
 		break;
 	}
 
-- 
2.35.3

