From 5a23f1756143119477828a5a3212b4189efb835c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 9 Sep 2013 15:28:04 +0800
Subject: [PATCH] rk3026: cpu1 local timer support hrtimer

---
 arch/arm/mach-rk3026/rk_timer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk3026/rk_timer.c b/arch/arm/mach-rk3026/rk_timer.c
index 3ffe68fb99ff..589c7b4b07a8 100644
--- a/arch/arm/mach-rk3026/rk_timer.c
+++ b/arch/arm/mach-rk3026/rk_timer.c
@@ -77,6 +77,7 @@ int __cpuinit local_timer_setup(struct clock_event_device *evt)
 {
 	evt->irq = IRQ_LOCALTIMER;
 	twd_timer_setup(evt);
+	evt->features &= ~CLOCK_EVT_FEAT_C3STOP;
 	return 0;
 }
 #endif
-- 
2.35.3

