From 53639230dd32e033e3f66463b7ea0e8ab614dba0 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 29 Aug 2012 14:29:21 +0800
Subject: [PATCH] rkfb:set fix.line_lenght in rk_fb_register

---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e3f6f17dbcc7..e11b731aa26f 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1035,6 +1035,7 @@ int rk_fb_register(struct rk_lcdc_device_driver *dev_drv,
 	#else
 		fbi->var.bits_per_pixel = 16; 
 	#endif
+	fbi->fix.line_length  = (fbi->var.xres)*(fbi->var.bits_per_pixel>>3);
         fbi->var.xres_virtual = fbi->var.xres;
         fbi->var.yres_virtual = fbi->var.yres;
         fbi->var.width =  fb_inf->lcdc_dev_drv[lcdc_id]->cur_screen->width;
-- 
2.35.3

