From 686f12e57a5ac63d0e351da6fe106eb46d0d2d78 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Fri, 21 Jun 2013 14:17:32 +0800
Subject: [PATCH] rk_screen.c add mipi config

---
 drivers/video/rockchip/screen/rk_screen.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/video/rockchip/screen/rk_screen.c b/drivers/video/rockchip/screen/rk_screen.c
index 06a0d00ad4df..d0c3975d2e53 100644
--- a/drivers/video/rockchip/screen/rk_screen.c
+++ b/drivers/video/rockchip/screen/rk_screen.c
@@ -230,10 +230,21 @@ void set_lcd_info(struct rk29fb_screen *screen, struct rk29lcd_info *lcd_info )
 	
 #if defined(CONFIG_MIPI_DSI)
        /* MIPI DSI */
+#if defined(MIPI_DSI_LANE)       
     screen->dsi_lane = MIPI_DSI_LANE;
+#else
+	screen->dsi_lane = 4;
+#endif
     //screen->dsi_video_mode = MIPI_DSI_VIDEO_MODE;
+#if defined(MIPI_DSI_HS_CLK)    
     screen->hs_tx_clk = MIPI_DSI_HS_CLK;
+#else    
+    screen->hs_tx_clk = 1000000000;        //1GHz
+#endif
+
 #endif
+
+
 	/* Operation function*/
 #if defined(RK_SCREEN_INIT)  //some screen need to init by spi or i2c
 	screen->init = rk_lcd_init;
-- 
2.35.3

