From ea219a87a136dcc189779a0183ab33b45498b3f7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 25 Jan 2013 19:14:53 +0800
Subject: [PATCH] screen.h: no deleted by make distclean

---
 drivers/video/display/screen/screen.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/display/screen/screen.h b/drivers/video/display/screen/screen.h
index e69de29bb2d1..40a8c178f10d 100755
--- a/drivers/video/display/screen/screen.h
+++ b/drivers/video/display/screen/screen.h
@@ -0,0 +1 @@
+/* empty */
-- 
2.35.3

