From 550a29f0cf9b12f2fadca04f631d2ef771310808 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Thu, 18 Sep 2014 19:59:31 +0800
Subject: [PATCH] mmc: rk3126: add wakeup event for SD insert/remove in deep
 sleep

Signed-off-by: lintao <lintao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 41e3c20573af..fd72bcac5e76 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3198,6 +3198,9 @@ static void dw_mci_of_set_cd_gpio_irq(struct device *dev, u32 gpio,
 		if (ret < 0) {
 			irq = ret;
 			dev_err(host->dev, "Request cd-gpio %d interrupt error!\n", gpio);
+		} else{
+		        /* enable wakeup event for gpio-cd in idle or deep suspend*/
+		        enable_irq_wake(irq);
 		}
 	} else {
 		dev_err(host->dev, "Cannot convert gpio %d to irq!\n", gpio);
-- 
2.35.3

