From ae315af2c6ce9e107ccb8516c19833b461c3dde1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Wed, 9 Mar 2011 11:31:33 +0800
Subject: [PATCH] disable set hardware cursor log

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index f9cb93903c94..08ed5bc01f85 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -783,7 +783,7 @@ int rk29_set_cursor_pos(struct rk29fb_inf *inf, int x, int y)
     /* set data */
     if (x >= 0x800 || y >= 0x800 )
         return -EINVAL;
-	printk("%s: %08x,%08x \n",__func__, x, y);
+	//printk("%s: %08x,%08x \n",__func__, x, y);
     x += (screen->left_margin + screen->hsync_len);
     y += (screen->upper_margin + screen->vsync_len);
     LcdWrReg(inf, HWC_DSP_ST, v_BIT11LO(x)|v_BIT11HI(y));
-- 
2.35.3

