From 73a6ed1e584115725d2fde9e4217b86ee1fde216 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 16 Jun 2014 20:26:56 +0800
Subject: [PATCH] input: touchscreen: fix tp_suspend.h compilation warning

---
 drivers/input/touchscreen/tp_suspend.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/touchscreen/tp_suspend.h b/drivers/input/touchscreen/tp_suspend.h
index 68e3d58fec17..bf23e868e770 100644
--- a/drivers/input/touchscreen/tp_suspend.h
+++ b/drivers/input/touchscreen/tp_suspend.h
@@ -20,7 +20,7 @@ struct  tp_device{
 	struct mutex ops_lock;
 };
 
-static int fb_notifier_callback(struct notifier_block *self,
+static inline int fb_notifier_callback(struct notifier_block *self,
 				unsigned long action, void *data)
 {
 	struct tp_device *tp;
@@ -63,7 +63,7 @@ static int fb_notifier_callback(struct notifier_block *self,
 	return NOTIFY_OK;
 }
 
-static int tp_register_fb(struct tp_device *tp)
+static inline int tp_register_fb(struct tp_device *tp)
 {
 	memset(&tp->fb_notif, 0, sizeof(tp->fb_notif));
 	tp->fb_notif.notifier_call = fb_notifier_callback;
@@ -72,7 +72,7 @@ static int tp_register_fb(struct tp_device *tp)
 	return fb_register_client(&tp->fb_notif);
 }
 
-static void tp_unregister_fb(struct tp_device *tp)
+static inline void tp_unregister_fb(struct tp_device *tp)
 {
 	fb_unregister_client(&tp->fb_notif);
 }
-- 
2.35.3

