From e774f49a4e01e95a2e1aba8c888ed0bef0d404f8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Tue, 17 Sep 2013 10:06:51 +0800
Subject: [PATCH] i2s: early initialization spin_lock for regiter read/write

---
 sound/soc/rk29/rk30_i2s.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index 801464e94b76..b38de18b7695 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -636,6 +636,7 @@ static int __devinit rockchip_i2s_probe(struct platform_device *pdev)
 	struct snd_soc_dai_driver *dai;
 	int    ret;
 
+	spin_lock_init(&i2s->spinlock_wr);
 #if defined(CONFIG_SND_I2S_USE_18V)	
 	writel_relaxed(0x2000200,RK30_GRF_BASE + GRF_IO_CON4);//bit9: 1,1.8v;0,3.3v
 #elif defined(CONFIG_SND_I2S_USE_33V)
@@ -764,7 +765,6 @@ static int __devinit rockchip_i2s_probe(struct platform_device *pdev)
 	if (ret != 0)
 		goto err_i2sv2;
 
-	spin_lock_init(&i2s->spinlock_wr);
 	return 0;
 
 err_i2sv2:
-- 
2.35.3

