From 05b0bee3ba8d382651bc7d65e8519fc82ac160d7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 7 Sep 2012 14:33:06 +0800
Subject: [PATCH] rk30: tsadc: remove msleep after stop

---
 arch/arm/mach-rk30/tsadc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk30/tsadc.c b/arch/arm/mach-rk30/tsadc.c
index 702d816bf09c..0415f8f67d99 100644
--- a/arch/arm/mach-rk30/tsadc.c
+++ b/arch/arm/mach-rk30/tsadc.c
@@ -134,7 +134,6 @@ static void rk30_tsadc_get(unsigned int chn, int *temp, int *code)
 		}
 	}
 	tsadc_writel(0, TSADC_CTRL);
-	msleep(10);
 
 	clk_disable(g_dev->clk);
 	clk_disable(g_dev->pclk);
-- 
2.35.3

