From f573dc23a404e38c91aa776a106833d71684fb1a Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 10 Sep 2013 15:11:49 +0800
Subject: [PATCH] rk_emmc: fixed drmboot wr bug

---
 drivers/mmc/card/block.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 38b0c374f7ca..d2dfeaa9fe2c 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2009,7 +2009,8 @@ static int mmc_blk_probe(struct mmc_card *card)
 			goto out;
 	}
 #ifdef CONFIG_EMMC_RK
-	this_card = card;
+	if(HOST_IS_EMMC(card->host))
+		this_card = card;
 #endif
 	return 0;
 
@@ -2024,7 +2025,8 @@ static void mmc_blk_remove(struct mmc_card *card)
 	struct mmc_blk_data *md = mmc_get_drvdata(card);
 
 #ifdef CONFIG_EMMC_RK
-	this_card = NULL;
+	if(HOST_IS_EMMC(card->host))
+		this_card = NULL;
 #endif
 	mmc_blk_remove_parts(card, md);
 	mmc_claim_host(card->host);
-- 
2.35.3

