From 5aa311af0399ac7feed2c3df77b1d1645b4dbcd1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 28 Jul 2011 15:20:41 +0800
Subject: [PATCH] rk29: adc: fix error: incompatible type for argument 1 of
 spin_lock

---
 drivers/adc/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/adc/core.c b/drivers/adc/core.c
index 0756c7ea4f80..eb7ddef884ac 100755
--- a/drivers/adc/core.c
+++ b/drivers/adc/core.c
@@ -203,7 +203,7 @@ void adc_core_irq_handle(struct adc_host *adc)
 {
 	struct adc_request *req;
 	int head, res;
-	spin_lock(adc->lock);
+	spin_lock(&adc->lock);
 	head = adc->queue_head;
 
 	req = adc->queue[head];
-- 
2.35.3

