From c8bad46591c5a226be3e2706f480a4ccd08f9e85 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Mon, 6 Sep 2010 13:06:39 +0800
Subject: [PATCH] change spi_gpio interrupt trigger type name

---
 drivers/fpga/spi_gpio.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/fpga/spi_gpio.c b/drivers/fpga/spi_gpio.c
index bf3df3938cee..cd5879757f21 100755
--- a/drivers/fpga/spi_gpio.c
+++ b/drivers/fpga/spi_gpio.c
@@ -670,8 +670,9 @@ int spi_gpio_init_first(void)
 	spi_gpio_set_pindirection(SPI_GPIO_P2_15, SPI_GPIO_OUT);
 
 	//spi_gpio_set_pindirection(SPI_GPIO_P4_06, SPI_GPIO_IN);		//CHARGER_INT_END input
-	spi_gpio_set_pindirection(SPI_GPIO_P4_06, SPI_GPIO_OUT);		//CHARGER_INT_END output
 	spi_gpio_set_pinlevel(SPI_GPIO_P4_06, SPI_GPIO_LOW);		//CM3605_PWD low
+	spi_gpio_set_pindirection(SPI_GPIO_P4_06, SPI_GPIO_OUT);		//CHARGER_INT_END output
+	
 
 	spi_gpio_set_pinlevel(SPI_GPIO_P4_07, SPI_GPIO_LOW);		//CM3605_PWD output
 	spi_gpio_set_pindirection(SPI_GPIO_P4_07, SPI_GPIO_OUT);
@@ -880,10 +881,10 @@ static int _spi_gpio_irq_set_type(unsigned int irq, unsigned int type)
 	}
 	switch(type)
 	{
-		case GPIOEdgelFalling:
+		case IRQF_TRIGGER_FALLING:
 			int_type = SPI_GPIO_EDGE_FALLING;
 			break;
-		case GPIOEdgelRising:
+		case IRQF_TRIGGER_RISING:
 			int_type = SPI_GPIO_EDGE_RISING;
 			break;
 		default:
-- 
2.35.3

