From 16c1ff51e2ca4ded45ce7623bbd1bc591330da54 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 29 Aug 2012 18:12:15 +0800
Subject: [PATCH] rk30: l2: never set way size and associativity

---
 arch/arm/mach-rk30/common.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 1a3050139400..0663ec8cae77 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -68,24 +68,18 @@ static void __init rk30_l2_cache_init(void)
 	/* L2X0 Power Control */
 	writel_relaxed(L2X0_DYNAMIC_CLK_GATING_EN | L2X0_STNDBY_MODE_EN, RK30_L2C_BASE + L2X0_POWER_CTRL);
 
-	/*
-         * 16-way associativity, parity disabled
-         * Way size - 32KB
-         */
-	aux_ctrl = ((1 << L2X0_AUX_CTRL_ASSOCIATIVITY_SHIFT) | // 16-way
+	aux_ctrl = (
 			(0x1 << 25) | 	// round-robin
 			(0x1 << 0) |		// Full Line of Zero Enable
 			(0x1 << L2X0_AUX_CTRL_NS_LOCKDOWN_SHIFT) |
-			(0x2 << L2X0_AUX_CTRL_WAY_SIZE_SHIFT) | // 32KB way-size
 			(0x1 << L2X0_AUX_CTRL_DATA_PREFETCH_SHIFT) |
 			(0x1 << L2X0_AUX_CTRL_INSTR_PREFETCH_SHIFT) |
 			(0x1 << L2X0_AUX_CTRL_EARLY_BRESP_SHIFT) );
 
-	aux_ctrl_mask = ~((1 << L2X0_AUX_CTRL_ASSOCIATIVITY_SHIFT) | // 16-way
+	aux_ctrl_mask = ~(
 			(0x1 << 25) | 	// round-robin
 			(0x1 << 0) |		// Full Line of Zero Enable
 			(0x1 << L2X0_AUX_CTRL_NS_LOCKDOWN_SHIFT) |
-			(0x7 << L2X0_AUX_CTRL_WAY_SIZE_SHIFT) | // 32KB way-size
 			(0x1 << L2X0_AUX_CTRL_DATA_PREFETCH_SHIFT) |
 			(0x1 << L2X0_AUX_CTRL_INSTR_PREFETCH_SHIFT) |
 			(0x1 << L2X0_AUX_CTRL_EARLY_BRESP_SHIFT) );
-- 
2.35.3

