From 24e11d13783a71b3e8ba0c2f1ffb652554a40240 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 13 Sep 2011 19:33:49 +0800
Subject: [PATCH] rk29: mmu.c: set vmalloc_reserve to 512MB for better support
 1G memory

---
 arch/arm/mm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mm/mmu.c b/arch/arm/mm/mmu.c
index 9c9a47a2f9ba..6cfc032ae80c 100644
--- a/arch/arm/mm/mmu.c
+++ b/arch/arm/mm/mmu.c
@@ -672,7 +672,7 @@ void __init iotable_init(struct map_desc *io_desc, int nr)
 }
 
 #if defined(CONFIG_RK29_MEM_SIZE_M) && CONFIG_RK29_MEM_SIZE_M >= 1024
-static unsigned long __initdata vmalloc_reserve = SZ_256M;
+static unsigned long __initdata vmalloc_reserve = SZ_512M;
 #else
 static unsigned long __initdata vmalloc_reserve = SZ_128M;
 #endif
-- 
2.35.3

