From 37f550b37659009fa6b367311a81262dd5b42e01 Mon Sep 17 00:00:00 2001
From: "wuhao@rock-chips.com" <wuhao@rock-chips.com>
Date: Fri, 28 Sep 2012 17:27:45 +0800
Subject: [PATCH] ACT8931 reboot

---
 arch/arm/mach-rk2928/board-rk2928-a720.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/mach-rk2928/board-rk2928-a720.c b/arch/arm/mach-rk2928/board-rk2928-a720.c
index 48cdf18a3a67..550da6c0d50a 100755
--- a/arch/arm/mach-rk2928/board-rk2928-a720.c
+++ b/arch/arm/mach-rk2928/board-rk2928-a720.c
@@ -816,9 +816,20 @@ static void __init rk30_i2c_register_board_info(void)
 //end of i2c
 
 #define POWER_ON_PIN RK2928_PIN1_PA2   //power_hold
+#if defined(CONFIG_REGULATOR_ACT8931)
+extern  int act8931_charge_det ;
+#endif
 static void rk2928_pm_power_off(void)
 {
 	printk(KERN_ERR "rk2928_pm_power_off start...\n");
+        
+        #if defined(CONFIG_REGULATOR_ACT8931)
+        if(g_pmic_type == PMIC_TYPE_ACT8931)
+        {
+              if(act8931_charge_det)
+                   arm_pm_restart(0, NULL);
+        }
+        #endif
 	
 	#if defined(CONFIG_MFD_TPS65910)	
 	if(g_pmic_type == PMIC_TYPE_TPS65910)
-- 
2.35.3

