From 9c2730125ecfabcd327416d544e9d9bdb26449bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:30:43 +0800
Subject: [PATCH] v4l2-chip-ident.h: fix for v2.6.36

---
 include/media/v4l2-chip-ident.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/media/v4l2-chip-ident.h b/include/media/v4l2-chip-ident.h
index 2a3b3371806c..b61d1c937534 100755
--- a/include/media/v4l2-chip-ident.h
+++ b/include/media/v4l2-chip-ident.h
@@ -299,7 +299,6 @@ enum {
 	V4L2_IDENT_MT9P111		= 45033,     /* ddl@rock-chips.com : MT9P111 support */
 	V4L2_IDENT_MT9D113      = 45034,     /* ddl@rock-chips.com : MT9D113 support */
 
-	V4L2_IDENT_MT9T111      = 45035, 	/*xbl@rock-chips.com: mt9t111 support*/
 	/* HV7131R CMOS sensor: just ident 46000 */
 	V4L2_IDENT_HV7131R		= 46000,
 
-- 
2.35.3

