From b1056ddddf63876b23d068c03a5f433bd3b9b94d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9F=AF=E9=A3=9E=E9=9B=84?= <kfx@rock-chips.com>
Date: Tue, 1 Jun 2010 03:48:50 +0000
Subject: [PATCH] add wifi power control test

---
 arch/arm/mach-rk2818/board-midsdk.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-rk2818/board-midsdk.c b/arch/arm/mach-rk2818/board-midsdk.c
index b0555584bdc4..375e4cdcec0d 100644
--- a/arch/arm/mach-rk2818/board-midsdk.c
+++ b/arch/arm/mach-rk2818/board-midsdk.c
@@ -162,6 +162,14 @@ void rk2818_sdmmc1_cfg_gpio(struct platform_device *dev)
 {
 	rk2818_mux_api_set(GPIOG_MMC1_SEL_NAME, IOMUXA_SDMMC1_CMD_DATA0_CLKOUT);
 	rk2818_mux_api_set(GPIOG_MMC1D_SEL_NAME, IOMUXA_SDMMC1_DATA123);
+#if 1
+	/* wifi power up (gpio control) */
+	rk2818_mux_api_set(GPIOH7_HSADCCLK_SEL_NAME,IOMUXB_GPIO1_D7);
+	rk2818_mux_api_set(GPIOF5_APWM3_DPWM3_NAME,IOMUXB_GPIO1_B5);
+	gpio_request(RK2818_PIN_PH7, "sdio");
+	gpio_direction_output(RK2818_PIN_PH7,GPIO_HIGH);
+#endif
+
 }
 #define CONFIG_SDMMC0_USE_DMA
 #define CONFIG_SDMMC1_USE_DMA
-- 
2.35.3

