From 346d56fd7b34577d828c39419247c8a764327d71 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 25 Feb 2013 16:51:33 +0800
Subject: [PATCH] rk3168_86v: delete arm:1.4G and 1.6G; gpu:600M

---
 arch/arm/mach-rk30/board-rk3168-86v.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/mach-rk30/board-rk3168-86v.c b/arch/arm/mach-rk30/board-rk3168-86v.c
index 772f5c5788ae..1ed9eb93c17a 100755
--- a/arch/arm/mach-rk30/board-rk3168-86v.c
+++ b/arch/arm/mach-rk30/board-rk3168-86v.c
@@ -1886,8 +1886,6 @@ static struct cpufreq_frequency_table dvfs_arm_table[] = {
 	{.frequency = 816 * 1000,       .index = 1000 * 1000},
 	{.frequency = 1008 * 1000,      .index = 1075 * 1000},
 	{.frequency = 1200 * 1000,      .index = 1150 * 1000},
-	//{.frequency = 1416 * 1000,      .index = 1250 * 1000},
-	//{.frequency = 1608 * 1000,      .index = 1300 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
 
@@ -1897,7 +1895,6 @@ static struct cpufreq_frequency_table dvfs_gpu_table[] = {
 	{.frequency = 266 * 1000,	.index = 1050 * 1000},
 	{.frequency = 300 * 1000,	.index = 1050 * 1000},
 	{.frequency = 400 * 1000,	.index = 1100 * 1000},
-	//{.frequency = 600 * 1000,	.index = 1200 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
 
-- 
2.35.3

