From 27af8bfff7a6c9d2e14d6555ea1024bdb9f441bb Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Sat, 25 Aug 2012 11:59:53 +0800
Subject: [PATCH] rk2928: a720: set min_brightness 80

---
 arch/arm/mach-rk2928/board-rk2928-a720.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/board-rk2928-a720.c b/arch/arm/mach-rk2928/board-rk2928-a720.c
index 9ee25c281577..8f60b7af22ba 100755
--- a/arch/arm/mach-rk2928/board-rk2928-a720.c
+++ b/arch/arm/mach-rk2928/board-rk2928-a720.c
@@ -143,6 +143,7 @@ static int rk29_backlight_pwm_resume(void)
 
 static struct rk29_bl_info rk29_bl_info = {
 	.pwm_id = PWM_ID,
+        .min_brightness = 80,
 	.bl_ref = PWM_EFFECT_VALUE,
 	.io_init = rk29_backlight_io_init,
 	.io_deinit = rk29_backlight_io_deinit,
-- 
2.35.3

