From 71f77e9722a156121229e500c822da9cbc5399f5 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Mon, 14 Jul 2014 21:43:34 +0800
Subject: [PATCH] 3036: fix boot mode register

---
 arch/arm/mach-rockchip/rk3036.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index b9e01522251c..a992fe03f6ca 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -72,8 +72,8 @@ static struct map_desc rk3036_io_desc[] __initdata = {
 
 static void __init rk3036_boot_mode_init(void)
 {
-	u32 flag = readl_relaxed(RK_GRF_VIRT + RK3036_GRF_OS_REG0);
-	u32 mode = readl_relaxed(RK_GRF_VIRT + RK3036_GRF_OS_REG1);
+	u32 flag = readl_relaxed(RK_GRF_VIRT + RK3036_GRF_OS_REG4);
+	u32 mode = readl_relaxed(RK_GRF_VIRT + RK3036_GRF_OS_REG5);
 	u32 rst_st = readl_relaxed(RK_CRU_VIRT + RK3036_CRU_RST_ST);
 
 	if (flag == (SYS_KERNRL_REBOOT_FLAG | BOOT_RECOVER))
@@ -189,8 +189,8 @@ static void rk3036_restart(char mode, const char *cmd)
 
 	rockchip_restart_get_boot_mode(cmd, &boot_flag, &boot_mode);
 
-	writel_relaxed(boot_flag, RK_GRF_VIRT + RK3036_GRF_OS_REG0);	// for loader
-	writel_relaxed(boot_mode, RK_GRF_VIRT + RK3036_GRF_OS_REG1);	// for linux
+	writel_relaxed(boot_flag, RK_GRF_VIRT + RK3036_GRF_OS_REG4);	// for loader
+	writel_relaxed(boot_mode, RK_GRF_VIRT + RK3036_GRF_OS_REG5);	// for linux
 	dsb();
 
 	/* pll enter slow mode */
-- 
2.35.3

