From 100b7257b5b3fa8c2195b4322270d9c99e8557d7 Mon Sep 17 00:00:00 2001
From: xieyi <xieyi@rock-chips.com>
Date: Tue, 9 Oct 2012 10:00:10 +0800
Subject: [PATCH] sensor: correct mma7660 range to 1.5g

---
 drivers/input/sensors/accel/mma7660.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/accel/mma7660.c b/drivers/input/sensors/accel/mma7660.c
index ea05e4b74355..232d3e670d26 100755
--- a/drivers/input/sensors/accel/mma7660.c
+++ b/drivers/input/sensors/accel/mma7660.c
@@ -54,7 +54,7 @@
 #define MMA7660_REG_PD          0xa
 
 
-#define MMA7660_RANGE			2000000
+#define MMA7660_RANGE			1500000
 
 /* LIS3DH */
 #define MMA7660_PRECISION       6
-- 
2.35.3

