From 8529a7930033bf170d907eb49b831c07f49e3a11 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 7 Jun 2010 07:21:59 +0000
Subject: [PATCH] fix rk2818_spim_resume

---
 drivers/spi/rk2818_spim.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/rk2818_spim.c b/drivers/spi/rk2818_spim.c
index bc586df782a2..7cb074fea0e0 100644
--- a/drivers/spi/rk2818_spim.c
+++ b/drivers/spi/rk2818_spim.c
@@ -958,7 +958,8 @@ static int __init rk2818_spim_probe(struct platform_device *pdev)
 
 static void __exit rk2818_spim_remove(struct platform_device *pdev)
 {
-	struct rk2818_spi *dws = platform_get_drvdata(pdev);
+	struct spi_master *master = platform_get_drvdata(pdev);
+	struct rk2818_spi *dws = spi_master_get_devdata(master);
 	int status = 0;
 
 	if (!dws)
@@ -995,7 +996,8 @@ static int rk2818_spim_suspend(struct platform_device *pdev, pm_message_t mesg)
 
 static int rk2818_spim_resume(struct platform_device *pdev)
 {
-	struct rk2818_spi *dws = platform_get_drvdata(pdev);
+	struct spi_master *master = platform_get_drvdata(pdev);
+	struct rk2818_spi *dws = spi_master_get_devdata(master);
 	int ret;
 	
 	clk_enable(dws->clock_spim);	
-- 
2.35.3

