From 1c3778cecede2d8d1f58ab8e501f604253a80557 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 16 Mar 2015 09:58:27 +0800
Subject: [PATCH] rk3368 lcdc: add yuv420sp nv21 format at lcdc scale fac

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index ba13a770bbb5..151ee9457678 100755
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -2273,6 +2273,7 @@ static int rk3368_lcdc_cal_scl_fac(struct rk_lcdc_win *win)
 		break;
 	case YUV420:
 	case YUV420_A:
+	case YUV420_NV21:
 		cbcr_srcW = srcW / 2;
 		cbcr_dstW = dstW;
 		cbcr_srcH = srcH / 2;
@@ -2329,6 +2330,7 @@ static int rk3368_lcdc_cal_scl_fac(struct rk_lcdc_win *win)
 	/*line buffer mode */
 	if ((win->area[0].format == YUV422) ||
 	    (win->area[0].format == YUV420) ||
+	    (win->area[0].format == YUV420_NV21) ||
 	    (win->area[0].format == YUV422_A) ||
 	    (win->area[0].format == YUV420_A)) {
 		if (win->cbr_hor_scl_mode == SCALE_DOWN) {
-- 
2.35.3

