From 5968267e7ae485dc02b2f0e3f6cd72ef4147f3e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Sep 2010 10:53:01 +0800
Subject: [PATCH] devices: fix gcc warning: resources_uart* defined but not
 used

---
 arch/arm/mach-rk2818/devices.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-rk2818/devices.c b/arch/arm/mach-rk2818/devices.c
index b10c62d6dd54..97dfd87e64b1 100755
--- a/arch/arm/mach-rk2818/devices.c
+++ b/arch/arm/mach-rk2818/devices.c
@@ -90,6 +90,7 @@ static struct resource resources_i2c1[] = {
 /*
  * rk2818 4 uarts device
  */
+#ifdef CONFIG_UART0_RK2818
 static struct resource resources_uart0[] = {
 	{
 		.start	= IRQ_NR_UART0,
@@ -102,6 +103,8 @@ static struct resource resources_uart0[] = {
 		.flags	= IORESOURCE_MEM,
 	},
 };
+#endif
+#ifdef CONFIG_UART1_RK2818
 static struct resource resources_uart1[] = {
 	{
 		.start	= IRQ_NR_UART1,
@@ -114,6 +117,8 @@ static struct resource resources_uart1[] = {
 		.flags	= IORESOURCE_MEM,
 	},
 };
+#endif
+#ifdef CONFIG_UART2_RK2818
 static struct resource resources_uart2[] = {
 	{
 		.start	= IRQ_NR_UART2,
@@ -126,6 +131,8 @@ static struct resource resources_uart2[] = {
 		.flags	= IORESOURCE_MEM,
 	},
 };
+#endif
+#ifdef CONFIG_UART3_RK2818
 static struct resource resources_uart3[] = {
 	{
 		.start	= IRQ_NR_UART3,
@@ -138,6 +145,7 @@ static struct resource resources_uart3[] = {
 		.flags	= IORESOURCE_MEM,
 	},
 };
+#endif
 /* sdmmc */
 struct platform_device rk2818_device_sdmmc0 = {
 	.name			= "rk2818_sdmmc",
-- 
2.35.3

