From 2baeb11265d0b10d630fbf4d2ca1518f08bf8438 Mon Sep 17 00:00:00 2001
From: zwl <zwl@rock-chips.com>
Date: Fri, 9 May 2014 11:51:31 +0800
Subject: [PATCH] rk drm fb: add init overscan value

---
 drivers/video/rockchip/rk_drm_fb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/rk_drm_fb.c b/drivers/video/rockchip/rk_drm_fb.c
index 41e7bfa00bbb..6381615b41ce 100755
--- a/drivers/video/rockchip/rk_drm_fb.c
+++ b/drivers/video/rockchip/rk_drm_fb.c
@@ -335,6 +335,10 @@ static int init_lcdc_device_driver(struct rk_drm_screen_private *screen_priv,
 	
 	screen->screen_id = 0;
 	screen->lcdc_id = dev_drv->id;
+	screen->overscan.left = 100;
+	screen->overscan.top = 100;
+	screen->overscan.right = 100;
+	screen->overscan.bottom = 100;
 	dev_drv->screen0 = screen;
 	dev_drv->cur_screen = screen;
 	/* devie use one lcdc + rk61x scaler for dual display*/
-- 
2.35.3

