From 2374ea662404571acf3a4364b2e0931edecd8be2 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 2 May 2013 11:43:41 +0800
Subject: [PATCH] mfd:rk616:core:power ctrl after mclk in

---
 drivers/mfd/rk616-core.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/mfd/rk616-core.c b/drivers/mfd/rk616-core.c
index accfca416292..f8c0ec82e144 100644
--- a/drivers/mfd/rk616-core.c
+++ b/drivers/mfd/rk616-core.c
@@ -364,9 +364,6 @@ static int rk616_i2c_probe(struct i2c_client *client,const struct i2c_device_id
 	i2c_set_clientdata(client, rk616);
 	dev_set_drvdata(rk616->dev,rk616);
 	
-	if(rk616->pdata->power_init)
-		rk616->pdata->power_init();
-
 #if defined(CONFIG_SND_RK29_SOC_I2S_8CH)        
 	iis_clk = clk_get_sys("rk29_i2s.0", "i2s");
 #elif defined(CONFIG_SND_RK29_SOC_I2S_2CH)
@@ -390,6 +387,11 @@ static int rk616_i2c_probe(struct i2c_client *client,const struct i2c_device_id
 		clk_set_rate(iis_clk, 11289600);
 		clk_put(iis_clk);
 	}
+
+	
+	if(rk616->pdata->power_init)
+		rk616->pdata->power_init();
+	
 	rk616->read_dev = rk616_i2c_read_reg;
 	rk616->write_dev = rk616_i2c_write_reg;
 	
-- 
2.35.3

