From 027eabd98bd69420d89de4c285fed8d23e3f9b9d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 1 Aug 2013 19:40:22 +0800
Subject: [PATCH] rk3188: allow compile vpu_service on rk3188

---
 arch/arm/plat-rk/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 07d0400de902..da2d8915950d 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -157,7 +157,7 @@ endchoice
 
 config RK29_VPU
 	tristate "VPU (Video Processing Unit) service driver in kernel"
-	depends on ARCH_RK29 || ARCH_RK30 || ARCH_RK2928 || ARCH_RK3026
+	depends on ARCH_RK29 || ARCH_RK30 || ARCH_RK2928 || ARCH_RK3026 || ARCH_RK3188
 	default m
 
 config RK29_LAST_LOG
-- 
2.35.3

