From 4700b767e80eb19d7e108427bc4c68f8c89ea7d9 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 10 Mar 2014 12:35:09 +0800
Subject: [PATCH] mmc: sdio: fix sdio_reset_comm

---
 drivers/mmc/core/sdio.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index 413989e9d533..adfabcef3437 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -1275,7 +1275,7 @@ int mmc_attach_sdio(struct mmc_host *host)
 	return err;
 }
 
-/*
+
 int sdio_reset_comm(struct mmc_card *card)
 {
 	struct mmc_host *host = card->host;
@@ -1293,13 +1293,13 @@ int sdio_reset_comm(struct mmc_card *card)
 	if (err)
 		goto err;
 
-	host->ocr = mmc_select_voltage(host, ocr);
-	if (!host->ocr) {
+	host->ocr_avail_sdio = mmc_select_voltage(host, ocr);
+	if (!host->ocr_avail_sdio) {
 		err = -EINVAL;
 		goto err;
 	}
 
-	err = mmc_sdio_init_card(host, host->ocr, card, 0);
+	err = mmc_sdio_init_card(host, host->ocr_avail_sdio, card, 0);
 	if (err)
 		goto err;
 
@@ -1312,4 +1312,4 @@ int sdio_reset_comm(struct mmc_card *card)
 	return err;
 }
 EXPORT_SYMBOL(sdio_reset_comm);
-*/
+
-- 
2.35.3

