From 4c626a0afa2bf8a6f5bef49b5bd6bc1d78559d82 Mon Sep 17 00:00:00 2001
From: lhh <lhh@rock-chips.com>
Date: Sun, 23 Jan 2011 18:49:06 +0800
Subject: [PATCH] add err printk rk29 mmcdrivers/mmc/host/rk29_sdmmc.c

---
 drivers/mmc/host/rk29_sdmmc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/mmc/host/rk29_sdmmc.c b/drivers/mmc/host/rk29_sdmmc.c
index 812ee390866c..bf82ca536478 100755
--- a/drivers/mmc/host/rk29_sdmmc.c
+++ b/drivers/mmc/host/rk29_sdmmc.c
@@ -993,6 +993,8 @@ static void rk29_sdmmc_read_data_pio(struct rk29_sdmmc *host)
 			host->data_status = status;
 			if(data)
 				data->bytes_xfered += nbytes;
+			else
+				printk("%s %d host data NULL err",__FUNCTION__,__LINE__);
 			smp_wmb();
 			rk29_sdmmc_set_pending(host, EVENT_DATA_ERROR);
 			tasklet_schedule(&host->tasklet);
@@ -1004,11 +1006,15 @@ static void rk29_sdmmc_read_data_pio(struct rk29_sdmmc *host)
 	host->pio_offset = offset;
 	if(data)
 		data->bytes_xfered += nbytes;
+	else
+		printk("%s %d host data NULL err",__FUNCTION__,__LINE__);	
 	return;
 
 done:
 	if(data)
 		data->bytes_xfered += nbytes;
+	else
+		printk("%s %d host data NULL err",__FUNCTION__,__LINE__);	
 	smp_wmb();
 	rk29_sdmmc_set_pending(host, EVENT_XFER_COMPLETE);
 }
@@ -1060,6 +1066,8 @@ static void rk29_sdmmc_write_data_pio(struct rk29_sdmmc *host)
 			host->data_status = status;
 			if(data)
 				data->bytes_xfered += nbytes;
+			else
+				printk("%s %d host data NULL err",__FUNCTION__,__LINE__);	
 			smp_wmb();
 			rk29_sdmmc_set_pending(host, EVENT_DATA_ERROR);
 			tasklet_schedule(&host->tasklet);
@@ -1070,11 +1078,15 @@ static void rk29_sdmmc_write_data_pio(struct rk29_sdmmc *host)
 	host->pio_offset = offset;
 	if(data)
 		data->bytes_xfered += nbytes;
+	else
+		printk("%s %d host data NULL err",__FUNCTION__,__LINE__);	
 	return;
 
 done:
 	if(data)
 		data->bytes_xfered += nbytes;
+	else
+		printk("%s %d host data NULL err",__FUNCTION__,__LINE__);	
 	smp_wmb();
 	rk29_sdmmc_set_pending(host, EVENT_XFER_COMPLETE);
 }
-- 
2.35.3

