From 5e44a64a6d6cd2e511359558dd5ec785b9a7ceeb Mon Sep 17 00:00:00 2001
From: ljf <ljf@rock-chips.com>
Date: Thu, 17 Jul 2014 16:15:50 +0800
Subject: [PATCH] rk3036: resolve problem in previous commit

---
 arch/arm/mach-rockchip/vcodec_service.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/vcodec_service.c b/arch/arm/mach-rockchip/vcodec_service.c
index 73c48ee6317c..b849b1acf13c 100755
--- a/arch/arm/mach-rockchip/vcodec_service.c
+++ b/arch/arm/mach-rockchip/vcodec_service.c
@@ -2064,7 +2064,6 @@ static irqreturn_t vdpu_isr(int irq, void *dev_id)
 
 static irqreturn_t vepu_irq(int irq, void *dev_id)
 {
-	struct vpu_device *dev = (struct vpu_device *) dev_id;
 	struct vpu_service_info *pservice = (struct vpu_service_info*)dev_id;
 	vpu_device *dev = &pservice->enc_dev;
 	u32 irq_status;
-- 
2.35.3

