From a3423fed6a76e295a901c5dcd9ad224f9846fe0a Mon Sep 17 00:00:00 2001
From: zwp <zwp@rock-chips.com>
Date: Mon, 15 Aug 2011 01:13:32 -0700
Subject: [PATCH] phone-pad:modify wm8994 I2S Channel

---
 sound/soc/rk29/rk29_wm8994.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk29_wm8994.c b/sound/soc/rk29/rk29_wm8994.c
index 5c82142fb15b..b666b0c20906 100755
--- a/sound/soc/rk29/rk29_wm8994.c
+++ b/sound/soc/rk29/rk29_wm8994.c
@@ -183,7 +183,7 @@ static struct snd_soc_dai_link rk29_dai = {
 #ifdef CONFIG_MACH_RK29_PHONEPADSDK
 	  .cpu_dai = &rk29_i2s_dai[1],
 #else
-          .cpu_dai = &rk29_i2s_dai[1],
+          .cpu_dai = &rk29_i2s_dai[0],
 #endif
 	  .codec_dai = &wm8994_dai,
 	  .init = rk29_wm8994_init,
-- 
2.35.3

