From 868ead9f12b8a89ab72c922e74e367f1f4236b9b Mon Sep 17 00:00:00 2001
From: Jianhong Chen <chenjh@rock-chips.com>
Date: Tue, 16 Jun 2015 21:26:39 +0800
Subject: [PATCH] power: rk81x-battery: count all avaraible rsoc into calc

Signed-off-by: Jianhong Chen <chenjh@rock-chips.com>
---
 drivers/power/rk818_battery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index 704a91f55822..256eb44a7b39 100755
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -2452,7 +2452,7 @@ static void rk81x_bat_zero_calc_linek(struct rk81x_battery *di)
 	rsoc = ocv_soc - dead_soc;
 	if ((di->dsoc == 1) && (rsoc > 0)) {/*discharge*/
 		di->line_k = 1000;
-	} else if (rsoc > 1) {
+	} else if (rsoc > 0) {
 		di->line_k = (di->display_soc + rsoc / 2) / div(rsoc);
 	} else {
 		di->dsoc--;
-- 
2.35.3

