From ebb79b3e0bfca6df4a4e987b59a92381acdfc92c Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 12 Mar 2015 16:16:56 +0800
Subject: [PATCH] rk312x lcdc: add to get hwc layer status

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk312x_lcdc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk312x_lcdc.c b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
index 86039500cd49..26c4ca84eec2 100755
--- a/drivers/video/rockchip/lcdc/rk312x_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
@@ -1744,6 +1744,8 @@ static int rk312x_lcdc_get_win_state(struct rk_lcdc_driver *dev_drv, int win_id)
 	        win_status = lcdc_read_bit(lcdc_dev, SYS_CTRL, m_WIN0_EN);
 	else if (win_id == 1)
 	        win_status = lcdc_read_bit(lcdc_dev, SYS_CTRL, m_WIN1_EN);
+	else if (win_id == 2)
+                win_status = lcdc_read_bit(lcdc_dev, SYS_CTRL, m_HWC_EN);
 	else
 	        pr_err("!!!%s,win_id :%d,unsupport!!!\n",__func__,win_id);
 
-- 
2.35.3

