From 27437642f25a1894998a53dc1fa73f8ea0a1f096 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Sat, 1 Sep 2012 17:07:53 +0800
Subject: [PATCH] rk29_key: del adc timer if probe failed

---
 drivers/input/keyboard/rk29_keys.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/keyboard/rk29_keys.c b/drivers/input/keyboard/rk29_keys.c
index 711388cd3b18..16b56198709f 100755
--- a/drivers/input/keyboard/rk29_keys.c
+++ b/drivers/input/keyboard/rk29_keys.c
@@ -425,6 +425,8 @@ static int __devinit keys_probe(struct platform_device *pdev)
 	}
 	if(pdata->chn >= 0 && ddata->client);
 		adc_unregister(ddata->client);
+	if(pdata->chn >= 0)
+	        del_timer_sync(&ddata->timer);
  fail1:
  	platform_set_drvdata(pdev, NULL);
  fail0:
-- 
2.35.3

