From a64cf6365d0d313c3d5927e6f0480e407f395338 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 25 Apr 2012 22:13:54 +0800
Subject: [PATCH] rk30:sdk: delete useless code

---
 arch/arm/mach-rk30/dvfs.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 54d2b8cbcbaa..fd9369bb5949 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -327,10 +327,6 @@ int clk_enable_dvfs(struct clk *clk)
 			}
 		}
 
-		if (dvfs_clk_get_ref_volt(dvfs_clk, dvfs_clk->set_freq, &clk_fv)) {
-			dvfs_clk->enable_dvfs = 0;
-			return -1;
-		}
 		dvfs_clk->set_volt = clk_fv.index;
 		// DVFS_DBG("%s,%s,freq%u(ref vol %u)\n",__func__,dvfs_clk->name,
 		//	 dvfs_clk->set_freq,dvfs_clk->set_volt);
-- 
2.35.3

