From 594bf2dd958cf0fac1363d6cac4fb827c40828f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Mon, 28 Apr 2014 10:26:01 +0800
Subject: [PATCH] rk3288:syb827:set voltage by 10mv/2.4us

---
 drivers/regulator/syb827.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/regulator/syb827.c b/drivers/regulator/syb827.c
index ec1ada5e11b3..5deee81604c8 100755
--- a/drivers/regulator/syb827.c
+++ b/drivers/regulator/syb827.c
@@ -233,7 +233,7 @@ static int syb827_dcdc_set_voltage_time_sel(struct regulator_dev *dev,   unsigne
 	if (new_volt < 0)
 		return new_volt;
 
-	return DIV_ROUND_UP(abs(old_volt - new_volt), 10000);
+	return DIV_ROUND_UP(abs(old_volt - new_volt)*4, 10000);
 }
 static int syb827_dcdc_suspend_enable(struct regulator_dev *dev)
 {
@@ -498,6 +498,8 @@ static int syb827_i2c_probe(struct i2c_client *i2c, const struct i2c_device_id *
 		printk("The device is not syb827 %x \n",ret);
 		goto err;
 	}
+	
+	ret = syb827_set_bits(syb827,SYB827_CONTR_REG1,(1 << 6),(1<<6));  //10mv/2.4us
 
 	if (syb827->dev->of_node)
 		pdev = syb827_parse_dt(syb827);
-- 
2.35.3

