From 7be1faace72940e9dca3d03773fde60ac3ee1807 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Thu, 20 Aug 2015 16:03:42 +0800
Subject: [PATCH] mmc: rk_sdmmc: reset cmd_status before request

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 7c2b6fe855fe..68dbc6491a44 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -1242,6 +1242,7 @@ static void __dw_mci_start_request(struct dw_mci *host,
 	host->pending_events = 0;
 	host->completed_events = 0;
 	host->data_status = 0;
+	host->cmd_status = 0;
 
 	data = cmd->data;
 	if (data) {
-- 
2.35.3

