From 4bebf13a245a152a3668978c03c80a5f8300d27c Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Tue, 1 Apr 2014 16:32:41 +0800
Subject: [PATCH] keys: fix keys dev name not matching

---
 drivers/input/keyboard/rk_keys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/rk_keys.c b/drivers/input/keyboard/rk_keys.c
index 8dc4d84e1511..066777e6f5ef 100755
--- a/drivers/input/keyboard/rk_keys.c
+++ b/drivers/input/keyboard/rk_keys.c
@@ -320,7 +320,7 @@ static int  keys_probe(struct platform_device *pdev)
 	}
 	platform_set_drvdata(pdev, ddata);
 
-	input->name = pdev->name;
+	input->name = "rk29-keypad";//pdev->name;
 	input->phys = "gpio-keys/input0";
 	input->dev.parent = dev;
 
-- 
2.35.3

