From 2fb0e2cd0d8b77c61c4cfebe11950d62f237e956 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Mon, 6 Jun 2011 18:56:52 -0700
Subject: [PATCH] a22: modify buttonlight name

---
 drivers/video/backlight/rk29_buttonlight.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/backlight/rk29_buttonlight.c b/drivers/video/backlight/rk29_buttonlight.c
index bb72a6643411..342c6eae7c08 100755
--- a/drivers/video/backlight/rk29_buttonlight.c
+++ b/drivers/video/backlight/rk29_buttonlight.c
@@ -76,7 +76,7 @@ static int rk29_button_light_probe(struct platform_device *pdev)
 {		 
     struct rk29_button_light_info *button_light_info = pdev->dev.platform_data;
     
-	rk29_button_light_device = backlight_device_register("rk29_button_light", &pdev->dev, NULL, &rk29_button_light_ops);
+	rk29_button_light_device = backlight_device_register("rk28_button_light", &pdev->dev, NULL, &rk29_button_light_ops);
 	if (!rk29_button_light_device) {
         DBG("rk29_button_light_probe error\n"); 
 		return -ENODEV;		
-- 
2.35.3

