From b490bce9e083ea9a5fd0a3dd2a6dcb9ce35558b3 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 16 Dec 2014 10:00:55 +0800
Subject: [PATCH] rk fb: fix open hdmi open backlight again in no dual mode

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 83a078244a1f..32323babb985 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3588,7 +3588,8 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 			return 0;
 
 		/* if used one lcdc to dual disp, no need to close win */
-		if (rk_fb->disp_mode == ONE_DUAL) {
+		if ((rk_fb->disp_mode == ONE_DUAL) ||
+		    (rk_fb->disp_mode == NO_DUAL)) {
 			dev_drv->cur_screen = dev_drv->screen0;
 			dev_drv->ops->load_screen(dev_drv, 1);
 
@@ -3677,7 +3678,8 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 		if (dev_drv->ops->dsp_black)
 			dev_drv->ops->dsp_black(dev_drv, 0);
 		if ((dev_drv->ops->backlight_close) &&
-		    (rk_fb->disp_policy != DISPLAY_POLICY_BOX))
+		    (rk_fb->disp_policy != DISPLAY_POLICY_BOX) &&
+		    (rk_fb->disp_mode == ONE_DUAL))
 			dev_drv->ops->backlight_close(dev_drv, 0);
 	}
 	return 0;
@@ -4198,8 +4200,9 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 			vaddr = vmap(pages, nr_pages, VM_MAP,
 					pgprot_writecombine(PAGE_KERNEL));
 			if (!vaddr) {
-				pr_err("failed to vmap phy addr %x\n",
-					uboot_logo_base + uboot_logo_offset);
+				pr_err("failed to vmap phy addr 0x%lx\n",
+				       (long)(uboot_logo_base +
+				       uboot_logo_offset));
 				return -1;
 			}
 
-- 
2.35.3

