From 8e44e7d4fd5bcea403b88f3597a128f4f757660c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 9 Nov 2010 10:32:35 +0800
Subject: [PATCH] rk29: vpu: fix gcc error

---
 arch/arm/mach-rk29/vpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/vpu.c b/arch/arm/mach-rk29/vpu.c
index b703fa1b94da..1749374dddd0 100644
--- a/arch/arm/mach-rk29/vpu.c
+++ b/arch/arm/mach-rk29/vpu.c
@@ -80,7 +80,7 @@ static struct vpu_client client;
 
 static void vpu_release_io(void);
 #ifdef CONFIG_RK29_VPU_DEBUG
-static void dump_regs(unsigned long data);
+static void dump_regs(struct vpu_device *);
 #endif
 
 static long vpu_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
-- 
2.35.3

