From 6823e0329f4d6a67c827c7290c57e725c6badc39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 23 Jun 2014 14:25:05 +0800
Subject: [PATCH] ARM: rockchip: support reboot fastboot

---
 arch/arm/mach-rockchip/common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rockchip/common.c b/arch/arm/mach-rockchip/common.c
index b9893d362a13..87aba158e9e4 100755
--- a/arch/arm/mach-rockchip/common.c
+++ b/arch/arm/mach-rockchip/common.c
@@ -254,6 +254,8 @@ void rockchip_restart_get_boot_mode(const char *cmd, u32 *flag, u32 *mode)
 			*flag = SYS_LOADER_REBOOT_FLAG + BOOT_LOADER;
 		else if(!strcmp(cmd, "recovery"))
 			*flag = SYS_LOADER_REBOOT_FLAG + BOOT_RECOVER;
+		else if (!strcmp(cmd, "fastboot"))
+			*flag = SYS_LOADER_REBOOT_FLAG + BOOT_FASTBOOT;
 		else if (!strcmp(cmd, "charge")) {
 			*flag = SYS_LOADER_REBOOT_FLAG + BOOT_CHARGING;
 			*mode = BOOT_MODE_CHARGE;
-- 
2.35.3

