From 02873058c5cdeb369680e2b7eacf1440ab34bff1 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 12 Sep 2012 20:24:33 +0800
Subject: [PATCH] rkfb:modify power on sequence for lcd and lcdc

---
 drivers/video/rockchip/rk_fb.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 2620761b3e42..d078e38b4f74 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1171,12 +1171,15 @@ static void rkfb_early_resume(struct early_suspend *h)
 	{
 		if (!inf->lcdc_dev_drv[i])
 			continue;
-		if(inf->lcdc_dev_drv[i]->screen_ctr_info->io_enable)
+		if(inf->lcdc_dev_drv[i]->screen_ctr_info->io_enable) 		//power on
 			inf->lcdc_dev_drv[i]->screen_ctr_info->io_enable();
+		
+		inf->lcdc_dev_drv[i]->resume(inf->lcdc_dev_drv[i]);	       // data out
+		
 		if(inf->lcdc_dev_drv[i]->screen0->standby)
-			inf->lcdc_dev_drv[i]->screen0->standby(0);
+			inf->lcdc_dev_drv[i]->screen0->standby(0);	      //screen wake up
+		
 		
-		inf->lcdc_dev_drv[i]->resume(inf->lcdc_dev_drv[i]);
 	}
 
 }
-- 
2.35.3

