From 77ab6aeb9929942b1f3320d13d5bc8847735ace4 Mon Sep 17 00:00:00 2001
From: xxx <xxx@rock-chips.com>
Date: Thu, 21 Feb 2013 17:25:46 +0800
Subject: [PATCH] remove printk in last updateing

---
 arch/arm/plat-rk/dvfs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/plat-rk/dvfs.c b/arch/arm/plat-rk/dvfs.c
index a9901daef16f..f645f6a09b94 100755
--- a/arch/arm/plat-rk/dvfs.c
+++ b/arch/arm/plat-rk/dvfs.c
@@ -90,7 +90,7 @@ static void dvfs_get_vd_regulator_volt_list(struct vd_node *vd)
 			break;
 		}
 		vd->volt_list[i]=sel_volt;	
-		printk("%s:selector=%u,volt %d\n",vd->name,i,sel_volt);
+		DVFS_DBG("%s:selector=%u,volt %d\n",vd->name,i,sel_volt);
 	}
 	vd->n_voltages=selector;
 	mutex_unlock(&mutex);
@@ -166,7 +166,7 @@ static void dvfs_table_round_volt(struct clk_node  *dvfs_clk)
 			DVFS_WARNING("clk %s:round_volt : is %d,but list <=0\n",dvfs_clk->name,dvfs_clk->dvfs_table[i].index);
 			break;
 		}
-		printk("clk %s:round_volt %d to %d\n",dvfs_clk->name,dvfs_clk->dvfs_table[i].index,test_volt);
+		DVFS_DBG("clk %s:round_volt %d to %d\n",dvfs_clk->name,dvfs_clk->dvfs_table[i].index,test_volt);
 		dvfs_clk->dvfs_table[i].index=test_volt;		
 	}
 	mutex_unlock(&mutex);
@@ -259,7 +259,7 @@ static void dvfs_table_round_clk_rate(struct clk_node  *dvfs_clk)
 		}
 		temp_rate=(temp_rate/1000)+flags;
 		
-		printk("clk %s round_clk_rate %d to %d\n",
+		DVFS_DBG("clk %s round_clk_rate %d to %d\n",
 			dvfs_clk->name,dvfs_clk->dvfs_table[i].frequency,(int)(temp_rate));
 		
 		dvfs_clk->dvfs_table[i].frequency=temp_rate;		
-- 
2.35.3

