From c0e429f602979d364652c81084b621418c2c5216 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 25 Jan 2011 17:52:10 +0800
Subject: [PATCH] fb: rk29: rename suspend/resume

---
 drivers/video/rk29_fb.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 258d79140181..507adf179dfd 100644
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1879,7 +1879,7 @@ struct suspend_info {
 	struct rk29fb_inf *inf;
 };
 
-void suspend(struct early_suspend *h)
+static void rk29fb_suspend(struct early_suspend *h)
 {
 	struct suspend_info *info = container_of(h, struct suspend_info,
 						early_suspend);
@@ -1925,7 +1925,7 @@ void suspend(struct early_suspend *h)
 
 }
 
-void resume(struct early_suspend *h)
+static void rk29fb_resume(struct early_suspend *h)
 {
 	struct suspend_info *info = container_of(h, struct suspend_info,
 					early_suspend);
@@ -1969,9 +1969,9 @@ void resume(struct early_suspend *h)
 	memcpy((u8*)inf->preg, (u8*)&inf->regbak, 0xa4);  //resume reg
 }
 
-struct suspend_info suspend_info = {
-	.early_suspend.suspend = suspend,
-	.early_suspend.resume = resume,
+static struct suspend_info suspend_info = {
+	.early_suspend.suspend = rk29fb_suspend,
+	.early_suspend.resume = rk29fb_resume,
 	.early_suspend.level = EARLY_SUSPEND_LEVEL_DISABLE_FB,
 };
 #endif
-- 
2.35.3

