From ef3256073988a71dff2888df2ac37077a05ce443 Mon Sep 17 00:00:00 2001
From: huang zhibao <hzb@rock-chips.com>
Date: Tue, 5 May 2015 17:44:13 +0800
Subject: [PATCH] rk1000 codec: fixed rk1000 av startup pop

---
 sound/soc/codecs/rk1000_codec.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/codecs/rk1000_codec.c b/sound/soc/codecs/rk1000_codec.c
index f97b797d3628..8237ec0b8630 100755
--- a/sound/soc/codecs/rk1000_codec.c
+++ b/sound/soc/codecs/rk1000_codec.c
@@ -391,9 +391,9 @@ static int rk1000_codec_set_bias_level(struct snd_soc_codec *codec,
 
 	case SND_SOC_BIAS_STANDBY:
 		DBG("rk1000 standby\n");
-		rk1000_codec_write(codec, ACCELCODEC_R1D, 0xFF);
-		rk1000_codec_write(codec, ACCELCODEC_R1E, 0xFF);
-		rk1000_codec_write(codec, ACCELCODEC_R1F, 0xFF);
+		rk1000_codec_write(codec, ACCELCODEC_R1D, 0x2a);
+		rk1000_codec_write(codec, ACCELCODEC_R1E, 0x40);
+		rk1000_codec_write(codec, ACCELCODEC_R1F, 0x49);
 		break;
 
 	case SND_SOC_BIAS_OFF:
@@ -738,7 +738,7 @@ static int rk1000_codec_suspend(struct snd_soc_codec *codec)
 {
 	DBG("Enter::%s----%d\n", __func__, __LINE__);
 	spk_ctrl_fun(GPIO_LOW);
-	rk1000_codec_set_bias_level(codec, SND_SOC_BIAS_STANDBY);
+	rk1000_codec_set_bias_level(codec, SND_SOC_BIAS_OFF);
 	return 0;
 }
 
-- 
2.35.3

