From 823584064d8bfc3cfbaf94af5024234ebb46b5ac Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Wed, 30 Oct 2013 17:39:42 +0800
Subject: [PATCH] rkemmc: msdos: check_partition: fixed usb check error

---
 fs/partitions/msdos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/partitions/msdos.c b/fs/partitions/msdos.c
index bb7f006f20f8..8b098c1b1d2b 100644
--- a/fs/partitions/msdos.c
+++ b/fs/partitions/msdos.c
@@ -442,7 +442,7 @@ int msdos_partition(struct parsed_partitions *state)
 	int slot;
 #ifdef CONFIG_EMMC_RK
 	//if card is emmc, msdos_partition return false
-	if(state->bdev->bd_disk->first_minor == 0)
+	if(state->bdev->bd_disk->major == MMC_BLOCK_MAJOR &&state->bdev->bd_disk->first_minor == 0)
 		return 0;
 #endif
 
-- 
2.35.3

