From 9530ed2dfd2c774fa8aa671ebbf4691c2f82e4f1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 16 Jan 2013 14:40:13 +0800
Subject: [PATCH] rfkill-rk: setup rts gpio with IOMUX_FMUX

---
 net/rfkill/rfkill-rk.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/net/rfkill/rfkill-rk.c b/net/rfkill/rfkill-rk.c
index 835345d70542..ccfb6a88f882 100755
--- a/net/rfkill/rfkill-rk.c
+++ b/net/rfkill/rfkill-rk.c
@@ -460,11 +460,7 @@ static int rfkill_rk_probe(struct platform_device *pdev)
     ret = rfkill_rk_setup_wake_irq(rfkill);
     if (ret) goto fail_wake;
 
-#ifdef CONFIG_ARCH_RK2928
     ret = rfkill_rk_setup_gpio(&pdata->rts_gpio, IOMUX_FMUX, rfkill->pdata->name, "rts"); 
-#else
-    ret = rfkill_rk_setup_gpio(&(pdata->rts_gpio), IOMUX_FNORMAL, rfkill->pdata->name, "rts");
-#endif
     if (ret) goto fail_wake_host_irq;
 
     // 创建并注册RFKILL设备
-- 
2.35.3

