From a39be434979f73f65d21d345b9a0b96cb737b84e Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Fri, 15 Aug 2014 17:47:16 +0800
Subject: [PATCH] gmac: delete /sys/class/vmac/exist

---
 .../ethernet/rockchip/gmac/stmmac_platform.c    | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/drivers/net/ethernet/rockchip/gmac/stmmac_platform.c b/drivers/net/ethernet/rockchip/gmac/stmmac_platform.c
index 46b662ab23da..fbdd386deba1 100755
--- a/drivers/net/ethernet/rockchip/gmac/stmmac_platform.c
+++ b/drivers/net/ethernet/rockchip/gmac/stmmac_platform.c
@@ -607,21 +607,6 @@ static int stmmac_probe_config_dt(struct platform_device *pdev,
 }
 #endif /* CONFIG_OF */
 
-static struct class *vmac_class = NULL;
-static CLASS_ATTR(exist, 0664, NULL, NULL);
-int rockchip_gmac_sysif_init(void)
-{
-       int ret;
-
-       vmac_class = class_create(THIS_MODULE, "vmac");
-       ret = class_create_file(vmac_class, &class_attr_exist);
-       if(ret) {
-           printk("%s: Fail to creat class\n",__func__);
-           return ret;
-       }
-       return 0;
-}
-
 /**
  * stmmac_pltfr_probe
  * @pdev: platform device pointer
@@ -709,8 +694,6 @@ static int stmmac_pltfr_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, priv->dev);
 
-	rockchip_gmac_sysif_init();
-
 	pr_debug("STMMAC platform driver registration completed");
 
 	return 0;
-- 
2.35.3

