From 8135895ffe21194de78d1670406a745292681e91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 02:13:29 +0800
Subject: [PATCH] PM: wakelock: dump active wake lock name when unable to
 suspend

---
 kernel/power/process.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/power/process.c b/kernel/power/process.c
index cdac5f955b85..1ac4d4582bf2 100644
--- a/kernel/power/process.c
+++ b/kernel/power/process.c
@@ -106,6 +106,7 @@ static int try_to_freeze_tasks(bool sig_only)
 			printk("\n");
 			printk(KERN_ERR "Freezing of %s aborted\n",
 					sig_only ? "user space " : "tasks ");
+			print_active_wake_locks(WAKE_LOCK_SUSPEND);
 		}
 		else {
 			printk("\n");
-- 
2.35.3

