From b9bff5d706921e9da8ac016fd35497826026cc0b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 21 Feb 2011 11:12:34 +0800
Subject: [PATCH] keyboad: Synaptics So340010 TouchPad KEY default n

---
 drivers/input/keyboard/Kconfig | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/input/keyboard/Kconfig b/drivers/input/keyboard/Kconfig
index fa189912158d..0ce985320e5d 100755
--- a/drivers/input/keyboard/Kconfig
+++ b/drivers/input/keyboard/Kconfig
@@ -17,12 +17,13 @@ config KEYS_RK29
 	default y
 	help
 		rk29 keyboard drivers(gpio and adc)
+
 config SYNAPTICS_SO340010
 	tristate "Synaptics So340010 TouchPad KEY"
-	depends on ARCH_RK29
-	default y
+	depends on I2C
 	help
 	   "Synaptics So340010 Touch Key (I2C) driver"
+
 config KEYBOARD_AAED2000
 	tristate "AAED-2000 keyboard"
 	depends on MACH_AAED2000
-- 
2.35.3

