From 5b18ea09e60d49108650483b03177ee31d7170fe Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Mon, 18 Mar 2013 17:21:48 +0800
Subject: [PATCH] fix compile err when no select CONFIG_DISPLAY_SUPPORT

---
 include/linux/rk_screen.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index ad278d086859..84f538aa1b20 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -194,6 +194,7 @@ struct rk29fb_info {
 
 #ifndef CONFIG_DISPLAY_SUPPORT
 static inline void set_lcd_info(struct rk29fb_screen *screen, struct rk29lcd_info *lcd_info) {}
+static inline size_t get_fb_size(void) { return 0;}
 #else
 extern void set_lcd_info(struct rk29fb_screen *screen, struct rk29lcd_info *lcd_info);
 extern size_t get_fb_size(void);
-- 
2.35.3

