From 97b97440a4278a415296e9a4cd663a0fce161f6c Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 25 May 2015 16:01:12 +0800
Subject: [PATCH] rk3368 lcdc: fix iommu crash when resume

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index cc9c2b7475f4..dd469c27a81a 100755
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -3315,6 +3315,10 @@ static int rk3368_lcdc_early_resume(struct rk_lcdc_driver *dev_drv)
 		lcdc_cfg_done(lcdc_dev);
 
 		if (dev_drv->iommu_enabled) {
+			/* win address maybe effect after next frame start,
+			 * but mmu maybe effect right now, so we delay 50ms
+			 */
+			mdelay(50);
 			if (dev_drv->mmu_dev)
 				rockchip_iovmm_activate(dev_drv->dev);
 		}
-- 
2.35.3

