From d7e25391ed360ed0c0041610b22676fce72b8386 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Mon, 18 Apr 2011 15:50:36 +0800
Subject: [PATCH] remove hdmi auto display hardware cursor

---
 drivers/video/rk29_fb.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 7f41873da733..bcb756e6011c 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -2066,11 +2066,6 @@ int FB_Switch_Screen( struct rk29fb_screen *screen, u32 enable )
     fb0_set_par(inf->fb0);
     LcdMskReg(inf, DSP_CTRL1, m_BLACK_MODE,  v_BLACK_MODE(0));
     LcdWrReg(inf, REG_CFG_DONE, 0x01);
-	if(enable){
-		rk29_set_cursor_en(inf, enable);
-	}else{
-		rk29_set_cursor_en(inf, enable);
-	}
     return 0;
 }
 
-- 
2.35.3

