From a736f82e0eb33bdf98195e8ab67f6f4709d1bc53 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Sat, 12 Jul 2014 17:07:27 +0800
Subject: [PATCH] 3036: fix build error

---
 arch/arm/mach-rockchip/rk3036.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 68405cc54e9a..416a441b240c 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -110,7 +110,7 @@ static void __init rk3036_dt_map_io(void)
 
     iotable_init(rk3036_io_desc, ARRAY_SIZE(rk3036_io_desc));
     debug_ll_io_init();
-    //usb_uart_init();
+    usb_uart_init();
 
     /* enable timer5 for core */
     writel_relaxed(0, RK3036_TIMER5_VIRT + 0x10);
-- 
2.35.3

