From fa2b29afd52174ec4719138da87d029bd104c106 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 10 Aug 2012 10:12:36 +0800
Subject: [PATCH] rkfb:fix rk30lcdc win2 crash

---
 drivers/video/rockchip/rk_fb.c | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 88d51b26435a..66dd7845a16a 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1019,21 +1019,13 @@ int rk_fb_register(struct rk_lcdc_device_driver *dev_drv,
 
     if(id == 0)
     {
-	    fb_inf->fb[fb_inf->num_fb-2]->fbops->fb_open(fb_inf->fb[fb_inf->num_fb-2],1);
-	    fb_inf->fb[fb_inf->num_fb-2]->fbops->fb_set_par(fb_inf->fb[fb_inf->num_fb-2]);
-		#ifdef  CONFIG_LOGO_LINUX_BMP
-	   	if(fb_prepare_bmp_logo(fb_inf->fb[fb_inf->num_fb-2], FB_ROTATE_UR)) {
+	    fb_inf->fb[fb_inf->num_fb-3]->fbops->fb_open(fb_inf->fb[fb_inf->num_fb-3],1);
+	    fb_inf->fb[fb_inf->num_fb-3]->fbops->fb_set_par(fb_inf->fb[fb_inf->num_fb-3]);
+	    if(fb_prepare_logo(fb_inf->fb[fb_inf->num_fb-3], FB_ROTATE_UR)) {
 	        /* Start display and show logo on boot */
-	        fb_set_cmap(&fb_inf->fb[fb_inf->num_fb-2]->cmap, fb_inf->fb[fb_inf->num_fb-2]);
-	        fb_show_bmp_logo(fb_inf->fb[fb_inf->num_fb-2], FB_ROTATE_UR);
-			fb_inf->fb[fb_inf->num_fb-2]->fbops->fb_pan_display(&(fb_inf->fb[fb_inf->num_fb-2]->var), fb_inf->fb[fb_inf->num_fb-2]);
-	    }
-		#else
-		if(fb_prepare_logo(fb_inf->fb[fb_inf->num_fb-2], FB_ROTATE_UR)) {
-	        /* Start display and show logo on boot */
-	        fb_set_cmap(&fb_inf->fb[fb_inf->num_fb-2]->cmap, fb_inf->fb[fb_inf->num_fb-2]);
-	        fb_show_logo(fb_inf->fb[fb_inf->num_fb-2], FB_ROTATE_UR);
-			fb_inf->fb[fb_inf->num_fb-2]->fbops->fb_pan_display(&(fb_inf->fb[fb_inf->num_fb-2]->var), fb_inf->fb[fb_inf->num_fb-2]);
+	        fb_set_cmap(&fb_inf->fb[fb_inf->num_fb-3]->cmap, fb_inf->fb[fb_inf->num_fb-3]);
+	        fb_show_logo(fb_inf->fb[fb_inf->num_fb-3], FB_ROTATE_UR);
+		fb_inf->fb[fb_inf->num_fb-3]->fbops->fb_pan_display(&(fb_inf->fb[fb_inf->num_fb-3]->var), fb_inf->fb[fb_inf->num_fb-3]);
 	    }
 		#endif
     }
-- 
2.35.3

