From f452ca767cd6d64f374a91dcaeb98b6014fd3a8d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Thu, 19 Apr 2012 19:07:20 +0800
Subject: [PATCH] rk30:sdk:close wm8326 i2c shutdown

---
 drivers/mfd/wm831x-i2c.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/wm831x-i2c.c b/drivers/mfd/wm831x-i2c.c
index f618a4af127f..717f629336f3 100755
--- a/drivers/mfd/wm831x-i2c.c
+++ b/drivers/mfd/wm831x-i2c.c
@@ -207,8 +207,8 @@ static int wm831x_i2c_resume(struct device *dev)
 void wm831x_i2c_shutdown(struct i2c_client *i2c)
 {
 	struct wm831x *wm831x = i2c_get_clientdata(i2c);
-	printk("%s\n", __FUNCTION__);
-	wm831x_device_shutdown(wm831x);
+//	printk("%s\n", __FUNCTION__);
+//	wm831x_device_shutdown(wm831x);
 }
 
 static const struct i2c_device_id wm831x_i2c_id[] = {
-- 
2.35.3

