From 1bedeca093dd3d402080c6e94d98932aa6d64641 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 17 Jul 2013 12:56:43 +0800
Subject: [PATCH] rk: Kconfig add RK3026 support

---
 arch/arm/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 442ff24c412f..00b8b80a80a7 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1031,6 +1031,7 @@ source "arch/arm/mach-realview/Kconfig"
 source "arch/arm/plat-rk/Kconfig"
 source "arch/arm/mach-rk29/Kconfig"
 source "arch/arm/mach-rk2928/Kconfig"
+source "arch/arm/mach-rk3026/Kconfig"
 source "arch/arm/mach-rk30/Kconfig"
 source "arch/arm/mach-rk3188/Kconfig"
 
-- 
2.35.3

