From 60dad30fe9d315b74cb01df10fc24eeb7dc4b7f3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 31 May 2014 10:37:12 +0800
Subject: [PATCH] ARM: rockchip: set default reboot flag as BOOT_NORMAL

---
 arch/arm/mach-rockchip/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/common.c b/arch/arm/mach-rockchip/common.c
index a41565adbacc..b9893d362a13 100755
--- a/arch/arm/mach-rockchip/common.c
+++ b/arch/arm/mach-rockchip/common.c
@@ -246,7 +246,7 @@ void __init rockchip_boot_mode_init(u32 flag, u32 mode)
 
 void rockchip_restart_get_boot_mode(const char *cmd, u32 *flag, u32 *mode)
 {
-	*flag = 0;
+	*flag = SYS_LOADER_REBOOT_FLAG + BOOT_NORMAL;
 	*mode = BOOT_MODE_REBOOT;
 
 	if (cmd) {
-- 
2.35.3

