From 3e6d9ab94abd3b1a0ad589cece1fe5717de17d9c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Mon, 18 Jul 2011 17:19:57 +0800
Subject: [PATCH] newton:fix touch screen compile error

---
 drivers/input/touchscreen/gt819.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/gt819.c b/drivers/input/touchscreen/gt819.c
index 1b9f32421ef8..4c56425da840 100755
--- a/drivers/input/touchscreen/gt819.c
+++ b/drivers/input/touchscreen/gt819.c
@@ -99,7 +99,7 @@ unsigned int ulPolynomial = 0x04c11db7;
 struct i2c_client * i2c_connect_client = NULL;
 static u8 gt819_fw[]=
 {
-#include "gt819_fw.bin"
+#include "gt819_fw.i"
 };
 #if 0
 uint8_t config_info[] = {
-- 
2.35.3

