From a7b38129ad8b613580a8ec6630c4f4395b8a4d39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Feb 2011 13:09:11 +0800
Subject: [PATCH] pack-kernel.sh: pack toolchain

---
 pack-kernel.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index b3339d6a03fc..54113e8b439d 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -110,7 +110,7 @@ popd >/dev/null
 
 # tar kernel
 pushd $kerndir/../ >/dev/null
-package=$(basename $kerndir).tar.gz
+package=$(basename $kerndir).tar
 ex=$package.ex
 > $ex
 for file in ${FILES[@]}; do
@@ -120,7 +120,10 @@ for file in ${EXCLUDES[@]}; do
 	echo "$file" >> $ex
 done
 echo TAR $(pwd)/$package
-tar czf $package --numeric-owner --exclude-from $ex --exclude=.git --exclude=`basename $0` $(basename $kerndir)
+tar cf $package --numeric-owner --exclude-from $ex --exclude=.git --exclude=`basename $0` $(basename $kerndir)
+tar rf $package --numeric-owner --exclude=.git toolchain/arm-eabi-4.4.0
+echo GZIP $(pwd)/$package.gz
+gzip -9 -c $package > $package.gz
 rm $ex
 popd >/dev/null
 
-- 
2.35.3

