From ab77b75794035070f6eeebf066d5183487f1dc83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 24 Jan 2013 14:31:02 +0800
Subject: [PATCH] rk3188: enable rk30_io_drive_strength_init

---
 arch/arm/mach-rk30/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 6a0bc268cacc..8c3f436d279e 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -54,7 +54,7 @@ static void __init rk30_cpu_axi_init(void)
 
 static void __init rk30_io_drive_strength_init(void)
 {
-#if defined(CONFIG_ARCH_RK3066B)
+#if defined(CONFIG_ARCH_RK3066B) || defined(CONFIG_ARCH_RK3188)
 	writel_relaxed(0x11001100, RK30_GRF_BASE + GRF_IO_CON4);
 #endif
 }
-- 
2.35.3

