From 2cf99697de0dc7f940679e907f834c0b11b82f08 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Wed, 10 Dec 2014 16:44:05 +0800
Subject: [PATCH] rk312x lcdc: fix no used function call.

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk312x_lcdc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/rockchip/lcdc/rk312x_lcdc.c b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
index b617eb255606..107f7ad231f8 100755
--- a/drivers/video/rockchip/lcdc/rk312x_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
@@ -615,7 +615,6 @@ static int rk312x_lcdc_mmu_en(struct rk_lcdc_driver *dev_drv)
 		if (!lcdc_dev->iommu_status && dev_drv->mmu_dev) {
 			lcdc_dev->iommu_status = 1;
 			rockchip_iovmm_activate(dev_drv->dev);
-			rk312x_lcdc_mmu_en(dev_drv);
 		}
 	}
 	#endif
-- 
2.35.3

