From 1dd81790e671fd5c09258fba2a47fb013cf07ad3 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Wed, 19 Jun 2013 16:59:03 +0800
Subject: [PATCH] sensor:add layout control direction for ak8963

---
 drivers/input/sensors/compass/ak8963.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/sensors/compass/ak8963.c b/drivers/input/sensors/compass/ak8963.c
index 6b454dcbde05..8c9530c6baa8 100755
--- a/drivers/input/sensors/compass/ak8963.c
+++ b/drivers/input/sensors/compass/ak8963.c
@@ -637,11 +637,11 @@ static long compass_dev_ioctl(struct file *file,
 		break;
 	case ECS_IOCTL_GET_LAYOUT:
 		DBG("%s:ECS_IOCTL_GET_LAYOUT start\n",__func__);
-		layout = 1;	//sensor->pdata->layout;
+		layout = sensor->pdata->layout;
 		break;
 	case ECS_IOCTL_GET_OUTBIT:
 		DBG("%s:ECS_IOCTL_GET_OUTBIT start\n",__func__);
-		outbit = 1;	//sensor->pdata->outbit;
+		outbit = sensor->pdata->outbit;
 		break;
 	case ECS_IOCTL_RESET:
 		DBG("%s:ECS_IOCTL_RESET start\n",__func__);
-- 
2.35.3

