From f66b133848468585a2fac0cb2b8f237aa874ba55 Mon Sep 17 00:00:00 2001
From: qiuen <qiuen@rock-chips.com>
Date: Thu, 16 Feb 2012 17:52:28 +0800
Subject: [PATCH] phonepad:modify mu509 shutdown time

---
 drivers/misc/mu509.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/misc/mu509.c b/drivers/misc/mu509.c
index d05e95337338..c56e098eaec0 100755
--- a/drivers/misc/mu509.c
+++ b/drivers/misc/mu509.c
@@ -93,7 +93,6 @@ int modem_poweron_off(int on_off)
   {
 		MODEMDBG("------------modem_poweroff\n");
 		gpio_set_value(pdata->bp_power, GPIO_LOW);
-		msleep(2500);
 		gpio_set_value(pdata->bp_power, GPIO_HIGH);
 		msleep(2500);
 		gpio_set_value(pdata->bp_power, GPIO_LOW);
-- 
2.35.3

