From 86e76f233f849fda3bf3c708074051a2e4d399b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Thu, 8 May 2014 17:15:58 +0800
Subject: [PATCH] rk3288:pmic:rk808:modify warning for shutdown

---
 drivers/mfd/rk808.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index 7132dd37aeeb..d143ce34a4f9 100755
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -1121,7 +1121,7 @@ static struct rk808_board *rk808_parse_dt(struct i2c_client *i2c)
 	return NULL;
 }
 #endif
-static int rk808_shutdown(void)
+static int rk808_shutdown(struct i2c_client *i2c)
 {
 	int ret,i,val;
 	u16 reg = 0;
-- 
2.35.3

