From fd69fd6325eb9179c6679d458998b83d210fe52c Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 24 Jan 2013 09:56:31 +0800
Subject: [PATCH] rk3188 lcdc:add rk610 jetta support

---
 drivers/video/rockchip/lcdc/rk3188_lcdc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3188_lcdc.c b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
index aae9b1a92579..44a678ad43b5 100644
--- a/drivers/video/rockchip/lcdc/rk3188_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
@@ -381,6 +381,11 @@ static int rk3188_load_screen(struct rk_lcdc_device_driver *dev_drv, bool initsc
     	{
     		screen->init();
     	}
+
+	if(screen->sscreen_set)
+	{
+		screen->sscreen_set(screen,!initscreen);
+	}
 	
 	dev_info(dev_drv->dev,"%s for lcdc%d ok!\n",__func__,lcdc_dev->id);
 	return 0;
-- 
2.35.3

