From 7ff013247a301feca5c1d8d87741a16c621b1cd9 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 19 Feb 2013 10:06:13 +0800
Subject: [PATCH] rk3168_86v: front camera gc0308 turn upside down

---
 drivers/media/video/gc0308.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/video/gc0308.c b/drivers/media/video/gc0308.c
index 5193eab94794..7dc2544eacc6 100755
--- a/drivers/media/video/gc0308.c
+++ b/drivers/media/video/gc0308.c
@@ -566,7 +566,7 @@ static struct reginfo sensor_init_data[] =
 
 
 	//-------------H_V_Switch(4)---------------//
-			{0x14 , 0x12},  //0x10
+			{0x14 , 0x10},  //0x10
 
 	 /*GC0308_H_V_Switch,
 
-- 
2.35.3

