From 567b81071dbc3f3c1597d0511d14e98546e31bd9 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Thu, 17 Feb 2011 11:40:41 +0800
Subject: [PATCH] add support ddr3

---
 arch/arm/mach-rk29/ddr.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/ddr.c b/arch/arm/mach-rk29/ddr.c
index 657b59e9acea..42991ca2151e 100644
--- a/arch/arm/mach-rk29/ddr.c
+++ b/arch/arm/mach-rk29/ddr.c
@@ -972,7 +972,7 @@ void DDR_Init(void)
     }
     pDDR_Reg->DTAR = value;
 
-    if(memType == DDRII)
+    if((memType == DDRII) || (memType == DDR3))
     {
         pDDR_Reg->ALPMR = LPPERIOD_POWER_DOWN(0xFF)|AUTOPD;
     }
@@ -987,10 +987,13 @@ void DDR_Init(void)
     Hz = clk_get_rate(clk_get(NULL,"ddr"));
     MHz = Hz/1000000;   //PLLGetDDRFreq()/1000;
     printk("DDR_Init: freq=%dMHz\n", MHz);
+    if((memType == DDRII) || (memType == Mobile_DDR))
+    {
     DDRPreUpdateRef(MHz);
     DDRPreUpdateTiming(MHz);
     DDRUpdateRef();
     DDRUpdateTiming();
+    }
 	
 }
 
-- 
2.35.3

