From ceaf071f7955c5c632c29bc7c5927957e0546375 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 20 May 2015 15:34:59 +0800
Subject: [PATCH] rk: gcc-wrapper.py ignore sysctl_net_core.c:24

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 scripts/gcc-wrapper.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/gcc-wrapper.py b/scripts/gcc-wrapper.py
index 8753754536a8..147ae09af62b 100755
--- a/scripts/gcc-wrapper.py
+++ b/scripts/gcc-wrapper.py
@@ -51,6 +51,7 @@ allowed_warnings = set([
     "compat_binfmt_elf.c:58",
     "psci.c:299",
     "psci.c:304",
+    "sysctl_net_core.c:24",
  ])
 
 # Capture the name of the object file, can find it.
-- 
2.35.3

