From a6d038ae6c1ebd931be75327816cd70289868cbd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Fri, 9 Mar 2012 21:32:11 +0800
Subject: [PATCH] update rt5625 driver for soundrecorder

---
 sound/soc/codecs/rt5625.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/rt5625.c b/sound/soc/codecs/rt5625.c
index c79ca0fbd303..7ef6437e47e5 100644
--- a/sound/soc/codecs/rt5625.c
+++ b/sound/soc/codecs/rt5625.c
@@ -2005,6 +2005,7 @@ static const struct snd_soc_dapm_widget rt5625_dapm_widgets[] = {
 					RT5625_P_ADCL_BIT, 0),
 	SND_SOC_DAPM_ADC("Right ADC", NULL, RT5625_PWR_ADD2,
 					RT5625_P_ADCR_BIT, 0),
+	SND_SOC_DAPM_PGA("ADC", SND_SOC_NOPM, 0, 0, NULL, 0),//johnny-3-9
 	SND_SOC_DAPM_MUX("PCM src select Mux", SND_SOC_NOPM, 0, 0,
 				&pcm_src_sel_mux),
 	SND_SOC_DAPM_MUX("IIS src select Mux", SND_SOC_NOPM, 0, 0,
@@ -2158,8 +2159,13 @@ static const struct snd_soc_dapm_route rt5625_dapm_routes[] = {
 	{"PCM src select Mux", "TxDP", "TxDP"},
 	{"PCM src select Mux", "ADCR", "Right ADC"},
 
-	{"IIS src select Mux", "ADC", "Left ADC"},
-	{"IIS src select Mux", "ADC", "Right ADC"},
+	//johnny-3-9 [s]
+	{"ADC", NULL, "Left ADC"},
+	{"ADC", NULL, "Right ADC"},
+	{"IIS src select Mux", "ADC", "ADC"},
+	//{"IIS src select Mux", "ADC", "Left ADC"},
+	//{"IIS src select Mux", "ADC", "Right ADC"},
+	//johnny-3-9 [e]
 	{"IIS src select Mux", "Voice to Stereo", "AIF2RX"},
 	{"IIS src select Mux", "SRC2 Output", "SRC2 Enable"},
 
-- 
2.35.3

