From 860361510bc45b3c4ed02487a7e2eedbccd56d72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 14 Mar 2014 09:11:01 +0800
Subject: [PATCH] rk: support make resource.img

---
 arch/arm/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 9526bdf7c166..c27cdf4c7a94 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -337,3 +337,7 @@ kernel.img: zImage
 	$(srctree)/mkkrnlimg $(objtree)/zImage-dtb $(objtree)/kernel.img >/dev/null && \
 	rm -f $(objtree)/zImage-dtb
 	@echo '  Image:  kernel.img (with $*.dtb) is ready'
+
+%.img: %.dtb kernel.img
+	$(Q)$(srctree)/resource_tool $(objtree)/arch/arm/boot/dts/$*.dtb
+	@echo '  Image:  resource.img (with $*.dtb) is ready'
-- 
2.35.3

