From 13f3700bed424f5883b7e7fcf3299a2c9e19a232 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 15 Jan 2013 20:31:50 +0800
Subject: [PATCH] rk3188 lcdc:fix rk3188 lcdc probe

---
 drivers/video/rockchip/lcdc/rk3188_lcdc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3188_lcdc.c b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
index 6e4cc5dc09ac..7524e654ae37 100644
--- a/drivers/video/rockchip/lcdc/rk3188_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3188_lcdc.c
@@ -883,6 +883,8 @@ static int __devinit rk3188_lcdc_probe(struct platform_device *pdev)
 		goto err4;
 	}
 	printk("rk3188 lcdc%d probe ok!\n",lcdc_dev->id);
+	
+	return 0;
 
 err4:
 	free_irq(lcdc_dev->irq,lcdc_dev);
-- 
2.35.3

