From af54fe0e76176c0f40026bad0a3efdf09371691c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 3 Feb 2012 16:18:51 +0800
Subject: [PATCH] driver: Makefile: build dbg subdirectories depend on config
 option

---
 drivers/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/Makefile b/drivers/Makefile
index 8e593d120cab..12345dd77085 100755
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -129,4 +129,4 @@ obj-$(CONFIG_CMMB)		+= cmmb/
 obj-$(CONFIG_TEST_CODE)		+= testcode/
 obj-y				+= smc/
 obj-y				+= cir/
-obj-y				+= dbg/
+obj-$(CONFIG_ARCH_RK29)		+= dbg/
-- 
2.35.3

