From 9e6960debbee06e1982bfdc077be1d6a845b60e6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 3 Sep 2012 18:59:27 +0800
Subject: [PATCH] mma8452: remove sensor_init printk

---
 drivers/input/sensors/accel/mma8452.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/input/sensors/accel/mma8452.c b/drivers/input/sensors/accel/mma8452.c
index e25e0022690e..9b3ca773d2ad 100755
--- a/drivers/input/sensors/accel/mma8452.c
+++ b/drivers/input/sensors/accel/mma8452.c
@@ -125,8 +125,6 @@ static int sensor_init(struct i2c_client *client)
 
 	DBG("mma8452 MMA8452_REG_SYSMOD:%x\n",sensor_read_reg(client,MMA8452_REG_SYSMOD));
 
-	printk("------sensor_chip_init\n");
-	
 	return ret;
 }
 
-- 
2.35.3

