From f98ebb4056a193d2f22160939134af54be8b44c3 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 1 Sep 2014 18:28:12 +0800
Subject: [PATCH] rk hdmi: add display_show_debug to fix warning

---
 drivers/video/rockchip/display-sys.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/display-sys.c b/drivers/video/rockchip/display-sys.c
index 373cba98ac05..3f8e6b85bb6b 100755
--- a/drivers/video/rockchip/display-sys.c
+++ b/drivers/video/rockchip/display-sys.c
@@ -189,6 +189,12 @@ static ssize_t display_store_scale(struct device *dev,
 	return -EINVAL;
 }
 
+static ssize_t display_show_debug(struct device *dev,
+				struct device_attribute *attr, char *buf)
+{
+	return -EINVAL;
+}
+
 static ssize_t display_store_debug(struct device *dev,
 					struct device_attribute *attr,
 					const char *buf, size_t count)
@@ -212,7 +218,7 @@ static struct device_attribute display_attrs[] = {
 	__ATTR(modes, S_IRUGO, display_show_modes, NULL),
 	__ATTR(mode, 0664, display_show_mode, display_store_mode),
 	__ATTR(scale, 0664, display_show_scale, display_store_scale),
-	__ATTR(debug, 0664, NULL, display_store_debug),
+	__ATTR(debug, 0664, display_show_debug, display_store_debug),
 	__ATTR_NULL
 };
 
-- 
2.35.3

