From 61e905928de0854b4674581417450af03615827b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 14 Mar 2013 18:31:11 +0800
Subject: [PATCH] rk: alarm: disable rtc alarm irq when shutdown

---
 drivers/rtc/alarm.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/rtc/alarm.c b/drivers/rtc/alarm.c
index 2c9492ad38cf..3ad91d9e4299 100644
--- a/drivers/rtc/alarm.c
+++ b/drivers/rtc/alarm.c
@@ -465,14 +465,9 @@ static int alarm_resume(struct platform_device *pdev)
 
 static void alarm_shutdown(struct platform_device *pdev)
 {
-	struct rtc_wkalrm alarm;
-
 	pr_alarm(FLOW, "alarm_shutdown(%p)\n", pdev);
 
-	memset(&alarm, 0, sizeof(alarm));
-	rtc_time_to_tm(0, &alarm.time);
-	alarm.enabled = 0;
-	rtc_set_alarm(alarm_rtc_dev, &alarm);
+	rtc_alarm_irq_enable(alarm_rtc_dev, false);
 }
 
 static struct rtc_task alarm_rtc_task = {
-- 
2.35.3

