From ca5addfee896d8feeece49a0a84cbdd09f2657f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Mon, 24 Dec 2012 17:28:55 +0800
Subject: [PATCH] rk292x:move pwm set suspend voltage

---
 arch/arm/mach-rk2928/pm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk2928/pm.c b/arch/arm/mach-rk2928/pm.c
index a50dccba0827..2c95c023ef9b 100644
--- a/arch/arm/mach-rk2928/pm.c
+++ b/arch/arm/mach-rk2928/pm.c
@@ -459,7 +459,7 @@ static int rk2928_pm_enter(suspend_state_t state)
 	cru_writel(CRU_W_MSK_SETBIT(0x01, PLL_PWR_DN_SHIFT), PLL_CONS(GPLL_ID, 1));//power down gpll
 
 	sram_printch('3');
-	rk30_pwm_suspend_voltage_set();
+//	rk30_pwm_suspend_voltage_set();
 
 	board_gpio_suspend();
 
@@ -470,7 +470,7 @@ static int rk2928_pm_enter(suspend_state_t state)
 	sram_printch('4');
 
 	board_gpio_resume();
-	rk30_pwm_resume_voltage_set();
+//	rk30_pwm_resume_voltage_set();
 	sram_printch('3');
 
 	//gpll
-- 
2.35.3

