From f2fb554178d19cef9b4302b8f4e12fc599286789 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Thu, 13 Jun 2013 14:40:34 +0800
Subject: [PATCH] i2s: add support samplerate up to 192K and support S32_LE

---
 sound/soc/rk29/rk30_i2s.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index 0f8f219b8a4d..0395fe345557 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -597,8 +597,9 @@ static int __devinit rockchip_i2s_probe(struct platform_device *pdev)
 		break;
 	}	
 
-	dai->playback.rates = ROCKCHIP_I2S_RATES;
-	dai->playback.formats = SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | SNDRV_PCM_FMTBIT_S24_LE;
+	dai->playback.rates = SNDRV_PCM_RATE_8000_192000;
+	dai->playback.formats = SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE |
+		SNDRV_PCM_FMTBIT_S24_LE| SNDRV_PCM_FMTBIT_S32_LE;
 	dai->capture.channels_min = 2;
 	dai->capture.channels_max = 2;
 	dai->capture.rates = ROCKCHIP_I2S_RATES;
-- 
2.35.3

