From 8bbedaea0eabc01743b95ddcf0f1860416c6cf5a Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Thu, 3 Jul 2014 17:53:25 +0800
Subject: [PATCH] fix gpu build error

---
 drivers/video/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/Kconfig b/drivers/video/Kconfig
index a45746bebf8b..323b16d75ed3 100644
--- a/drivers/video/Kconfig
+++ b/drivers/video/Kconfig
@@ -25,9 +25,9 @@ source "drivers/gpu/host1x/Kconfig"
 
 source "drivers/gpu/arm/midgard/Kconfig"
 
-source "drivers/gpu/arm/mali400/mali/Kconfig"
+#source "drivers/gpu/arm/mali400/mali/Kconfig"
 
-source "drivers/gpu/arm/mali400/ump/Kconfig"
+#source "drivers/gpu/arm/mali400/ump/Kconfig"
 
 config VGASTATE
        tristate
-- 
2.35.3

