From fdbdd54a8240b78a9b968e6dc87e8a317ab52363 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 29 Dec 2010 15:33:51 +0800
Subject: [PATCH] rk29_fb: fix fix setup DMC AXI host priority

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 4f811e972686..f8201598bfce 100644
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1941,7 +1941,7 @@ static int __init rk29fb_probe (struct platform_device *pdev)
     __raw_writel(0x7, RK29_DDRC_BASE+0x208);
     __raw_writel(0x7, RK29_DDRC_BASE+0x20c);
     __raw_writel(0x7, RK29_DDRC_BASE+0x210);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x280);
+    __raw_writel(0x10007eff, RK29_DDRC_BASE+0x280);
     __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x284);
     __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x288);
     __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x28c);
-- 
2.35.3

