From 72a572703f5dbeba21bfdca71bbcde4bd1e63b26 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 28 Mar 2014 13:53:53 +0800
Subject: [PATCH] mmc: host: rockchip: pm_caps with MMC_PM_KEEP_POWER for eMMC

---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 9c101eaeef0e..df3604b47919 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -2539,7 +2539,7 @@ static int dw_mci_init_slot(struct dw_mci *host, unsigned int id)
 
     /* eMMC should not sched into pm mgmt framework*/
     if(mmc->restrict_caps & RESTRICT_CARD_TYPE_EMMC)
-        mmc->pm_flags |= MMC_PM_IGNORE_PM_NOTIFY;
+        mmc->pm_flags |= (MMC_PM_IGNORE_PM_NOTIFY | MMC_PM_KEEP_POWER);
 
 	if (host->pdata->get_ocr)
 		mmc->ocr_avail = host->pdata->get_ocr(id);
-- 
2.35.3

