From e81b40dae0a9401e926597bf1da0bd51c00edc75 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 3 Jul 2012 10:03:23 +0800
Subject: [PATCH] adc drivers: add comments

---
 include/linux/adc.h | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/include/linux/adc.h b/include/linux/adc.h
index 90e44913dae7..be4547b038c4 100755
--- a/include/linux/adc.h
+++ b/include/linux/adc.h
@@ -41,8 +41,19 @@ struct adc_client *adc_register(int chn,
 				void (*callback)(struct adc_client *, void *, int), 
 				void *callback_param);
 void adc_unregister(struct adc_client *client);
-
+/*
+ * function: adc_sync_read
+ * 1)return value:
+ *     if correct, return adc sample value;
+ *     if error, return negative;
+ */
 int adc_sync_read(struct adc_client *client);
+/*
+ * function: adc_async_read
+ * 1)return value: if error, return negative; else return 0;
+ * 2)adc sample value: the third parameter of callback.
+ *     if timeout, sample value is -1; else sample value is non-negative
+ */
 int adc_async_read(struct adc_client *client);
 #else
 static inline struct adc_client *adc_register(int chn,
-- 
2.35.3

