From 8de4e34970db073d02bc335e12ef4eccf504ebda Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 4 Jan 2011 10:58:41 +0800
Subject: [PATCH] update i2c driver

---
 drivers/i2c/busses/i2c-rk29.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index eafef9acc4da..3a48ea7f3f57 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -382,10 +382,8 @@ static int rk29_send_address(struct rk29_i2c_data *i2c,
 			i2c_err(i2c->dev, "i2c is busy, when send address\n");
 			return ret;
 		}
-		writel(I2C_LCMR_START, i2c->regs + I2C_LCMR);
 	}
-	else
-		writel(I2C_LCMR_START|I2C_LCMR_RESUME, i2c->regs + I2C_LCMR);
+	writel(I2C_LCMR_START|I2C_LCMR_RESUME, i2c->regs + I2C_LCMR);
 
 	if((ret = rk29_wait_event(i2c, RK2818_EVENT_MTX_RCVD_ACK)) != 0)
 	{
-- 
2.35.3

