From 4c84046d331a0324e065c5c8c526463eb5709b67 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 9 Sep 2013 10:41:22 +0800
Subject: [PATCH] wifi: fix NL80211_TESTMODE depends on diff modules

---
 net/wireless/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/Kconfig b/net/wireless/Kconfig
index eee1ef2409ed..e7f54e5df1ec 100755
--- a/net/wireless/Kconfig
+++ b/net/wireless/Kconfig
@@ -31,7 +31,7 @@ config CFG80211
 
 config NL80211_TESTMODE
 	bool "nl80211 testmode command"
-	depends on CFG80211 && (MT5931 || MT5931_MT6622 || MTK_COMBO)
+	depends on CFG80211 && (MT5931 || MT5931_MT6622 || MTK_COMBO || MT7601 ||ESP8089)	
 	help
 	  The nl80211 testmode command helps implementing things like
 	  factory calibration or validation tools for wireless chips.
-- 
2.35.3

