From a3cebb509a577dace3e5ff8c933901a06949d3e5 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 20 May 2015 20:31:57 +0800
Subject: [PATCH] arm: rockchip: efuse: fix compilation error on arm

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/mach-rockchip/efuse.c | 28 +++++++---------------------
 1 file changed, 7 insertions(+), 21 deletions(-)

diff --git a/arch/arm/mach-rockchip/efuse.c b/arch/arm/mach-rockchip/efuse.c
index 7311d5918334..12e17982c430 100644
--- a/arch/arm/mach-rockchip/efuse.c
+++ b/arch/arm/mach-rockchip/efuse.c
@@ -10,32 +10,13 @@
 #include <linux/delay.h>
 #include <linux/rockchip/cpu.h>
 #include <linux/rockchip/iomap.h>
-
-#include <linux/cpuidle.h>
-#include <linux/init.h>
-#include <linux/of.h>
-#include <linux/smp.h>
-#include <linux/slab.h>
-#include <linux/reboot.h>
-#include <linux/pm.h>
-#include <linux/delay.h>
-#include <linux/slab.h>
-#include <uapi/linux/psci.h>
-
 #include <asm/compiler.h>
-#include <asm/cpu_ops.h>
-#include <asm/errno.h>
 #include <asm/psci.h>
-#include <asm/smp_plat.h>
-#include <asm/suspend.h>
-#include <asm/system_misc.h>
-
 #ifdef CONFIG_ARM
 #include <asm/system_info.h>
 #endif
 #include "efuse.h"
 
-
 #define efuse_readl(offset) readl_relaxed(RK_EFUSE_VIRT + offset)
 #define efuse_writel(val, offset) writel_relaxed(val, RK_EFUSE_VIRT + offset)
 
@@ -50,8 +31,7 @@ struct rockchip_efuse {
 
 static struct rockchip_efuse efuse;
 
-
-
+#ifdef CONFIG_ARM64
 /****************************secure reg access****************************/
 
 #define SEC_REG_RW_SHT (0x0)
@@ -180,6 +160,12 @@ static int rk3368_efuse_readregs(u32 addr, u32 length, u8 *buf)
 
 	return ret;
 }
+#else
+static int rk3368_efuse_readregs(u32 addr, u32 length, u8 *buf)
+{
+	return 0;
+}
+#endif
 
 static int __init rk3288_efuse_readregs(u32 addr, u32 length, u8 *buf)
 {
-- 
2.35.3

