From 61ef8d731194774a23a36a9bb1fe23034351d54d Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 May 2015 21:03:48 +0800
Subject: [PATCH] rfkill-bt: fix compilation warning

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 net/rfkill/rfkill-bt.c | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/net/rfkill/rfkill-bt.c b/net/rfkill/rfkill-bt.c
index 2386716f7a27..a1a156c22efd 100755
--- a/net/rfkill/rfkill-bt.c
+++ b/net/rfkill/rfkill-bt.c
@@ -422,26 +422,29 @@ static const struct rfkill_ops rfkill_rk_ops = {
 
 static struct proc_dir_entry *bluetooth_dir, *sleep_dir;
 
-static int bluesleep_read_proc_lpm(struct file *file, char __user *buffer, 
-                    size_t count, loff_t *data)
+static ssize_t bluesleep_read_proc_lpm(struct file *file, char __user *buffer,
+				       size_t count, loff_t *data)
 {
     return sprintf(buffer, "unsupported to read\n");
 }
 
-static int bluesleep_write_proc_lpm(struct file *file, const char __user *buffer, 
-                    size_t count, loff_t *data)
+static ssize_t bluesleep_write_proc_lpm(struct file *file,
+					const char __user *buffer,
+					size_t count, loff_t *data)
 {
     return count;
 }
 
-static int bluesleep_read_proc_btwrite(struct file *file, char __user *buffer, 
-                    size_t count, loff_t *data)
+static ssize_t bluesleep_read_proc_btwrite(struct file *file,
+					   char __user *buffer,
+					   size_t count, loff_t *data)
 {
     return sprintf(buffer, "unsupported to read\n");
 }
 
-static int bluesleep_write_proc_btwrite(struct file *file, const char __user *buffer, 
-                    size_t count, loff_t *data)
+static ssize_t bluesleep_write_proc_btwrite(struct file *file,
+					    const char __user *buffer,
+					    size_t count, loff_t *data)
 {
     char b;
 
-- 
2.35.3

