From e33970c7456b80e82153498b7a335caba740d73a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 15 Jul 2014 17:19:35 +0800
Subject: [PATCH] ARM: rockchip: rk3036.dtsi make gpu clk always on

---
 arch/arm/boot/dts/rk3036.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/rk3036.dtsi b/arch/arm/boot/dts/rk3036.dtsi
index 6bbc3e1ce295..11cb30d3b8eb 100755
--- a/arch/arm/boot/dts/rk3036.dtsi
+++ b/arch/arm/boot/dts/rk3036.dtsi
@@ -267,6 +267,8 @@
 				<&clk_gates1 13>,
 				<&clk_gates8 2>,/*pclk_uart2*/
 
+				<&clk_gpu_pre>,
+
 				/*jtag*/
 				<&clk_gates1 3>;/*clk_jtag*/
 	};
-- 
2.35.3

