From 2a346d552227dbfad6f3730fd41c0ccdab206fbf Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Fri, 1 Jun 2012 18:19:10 +0800
Subject: [PATCH] fix rga power off flag init error

---
 drivers/video/rockchip/rga/rga_drv.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index 5c3fcef3b141..4b4def59b8c1 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -1189,7 +1189,6 @@ static int __devinit rga_drv_probe(struct platform_device *pdev)
 	}
 
 	mutex_init(&rga_service.mutex);
-	rga_service.enable = false;
 	INIT_DELAYED_WORK(&data->power_off_work, rga_power_off);
 	data->rga_irq_callback = NULL;
 	
-- 
2.35.3

