From be3d562d8c79809b131e947d70961e7c84ee49a7 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 May 2015 19:00:03 +0800
Subject: [PATCH] rk: gcc-wrapper.py ignore compat_binfmt_elf.c:58

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 scripts/gcc-wrapper.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/gcc-wrapper.py b/scripts/gcc-wrapper.py
index 1f4ee7b5fc68..e3452e225692 100755
--- a/scripts/gcc-wrapper.py
+++ b/scripts/gcc-wrapper.py
@@ -48,6 +48,7 @@ allowed_warnings = set([
     "menu.c:129",
     "bus.c:318", # FIXME
     "atags_to_fdt.c:96",
+    "compat_binfmt_elf.c:58",
  ])
 
 # Capture the name of the object file, can find it.
-- 
2.35.3

