From c8d6f4f188c2ac0a92341f31e606a067626be69e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 25 Jan 2013 19:17:03 +0800
Subject: [PATCH] pack-kernel.sh: support rk3188

---
 pack-kernel.sh | 29 +++++++++++++++++++----------
 1 file changed, 19 insertions(+), 10 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index c6d52741330b..a586e517a055 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -3,7 +3,7 @@ set -e
 
 # ---------------------------------------------------------------------------
 usage() {
-	echo "usage: $0 [rk2928|rk30|rk3066b]"
+	echo "usage: $0 [rk2928|rk30|rk3066b|rk3188]"
 	echo "example: $0 rk3066b"
 	exit
 }
@@ -30,6 +30,12 @@ EXCLUDES=(
 arch/arm/mach-rk30/*rk3066b*
 arch/arm/mach-rk30/include/mach/*rk3066b*
 
+arch/arm/mach-rk3188/*.c
+arch/arm/mach-rk3188/*.h
+arch/arm/mach-rk3188/Makefile*
+arch/arm/mach-rk3188/include
+arch/arm/configs/rk3188_*
+
 arch/arm/mach-rk29*/*.c
 arch/arm/mach-rk29*/*.h
 arch/arm/mach-rk29*/*.S
@@ -56,6 +62,12 @@ arch/arm/mach-rk30/Makefile*
 arch/arm/mach-rk30/include
 arch/arm/configs/rk30*
 
+arch/arm/mach-rk3188/*.c
+arch/arm/mach-rk3188/*.h
+arch/arm/mach-rk3188/Makefile*
+arch/arm/mach-rk3188/include
+arch/arm/configs/rk3188_*
+
 drivers/video/rockchip/lcdc/rk30*
 drivers/video/rockchip/hdmi/chips/rk30/rk30*
 drivers/video/rockchip/hdmi/chips/rk30/hdcp/rk30*
@@ -72,7 +84,7 @@ arch/arm/configs/rk2928_phonepad_defconfig
 
 defconfig=rk2928_sdk_defconfig
 
-elif [ "$arch" == "rk3066b" ]; then
+elif [ "$arch" == "rk3066b" -o "$arch" == "rk3188" ]; then
 
 EXCLUDES=(
 arch/arm/mach-rk30/clock_data.c
@@ -95,7 +107,11 @@ arch/arm/mach-rk29*/include
 arch/arm/configs/rk29*
 )
 
-defconfig=rk3168_tb_defconfig
+if [ "$arch" == "rk3066b" ]; then
+	defconfig=rk3168_tb_defconfig
+else
+	defconfig=rk3188_tb_defconfig
+fi
 
 else
 	echo "unknown arch" && usage
@@ -110,21 +126,14 @@ drivers/staging/rk29/vivante
 drivers/staging/rk29/ipp/rk29-ipp.c
 drivers/*rk28*.c
 include/*rk28*
-arch/arm/mach-rk3188/*.c
-arch/arm/mach-rk3188/*.h
-arch/arm/mach-rk3188/Makefile*
-arch/arm/mach-rk3188/include
 arch/arm/mach-rk29/*.c
 arch/arm/mach-rk29/*.h
 arch/arm/mach-rk29/*.S
 arch/arm/mach-rk29/Makefile*
 arch/arm/mach-rk29/include
 arch/arm/mach-rk*/*-fpga*
-arch/arm/mach-rk*/*-tb*
-arch/arm/configs/rk3188_*
 arch/arm/configs/rk29_*
 arch/arm/configs/*_fpga_*
-arch/arm/configs/*_tb_*
 )
 # ---------------------------------------------------------------------------
 
-- 
2.35.3

