From 797954d65bc0fe015c5108048714c7c258a26c1b Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Fri, 12 Jun 2015 18:20:57 +0800
Subject: [PATCH] ASoC: hdmi_spdif: add 192k samplerate support.

Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 sound/soc/codecs/hdmi_spdif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/hdmi_spdif.c b/sound/soc/codecs/hdmi_spdif.c
index f97de191c62c..cc4c5429126b 100644
--- a/sound/soc/codecs/hdmi_spdif.c
+++ b/sound/soc/codecs/hdmi_spdif.c
@@ -35,7 +35,7 @@
 
 #define DRV_NAME "spdif-dit"
 
-#define STUB_RATES	SNDRV_PCM_RATE_8000_96000
+#define STUB_RATES	SNDRV_PCM_RATE_8000_192000
 #define STUB_FORMATS	SNDRV_PCM_FMTBIT_S16_LE
 
 
-- 
2.35.3

