From 79ef987e89915791fa5c2acc834639d2aa738b7a Mon Sep 17 00:00:00 2001
From: dkl <dkl@rock-chips.com>
Date: Thu, 12 Jun 2014 17:59:55 +0800
Subject: [PATCH] iclk: rk3288: set hclk_vio when clk_3288_dclk_lcdc0_set_rate

---
 drivers/clk/rockchip/clk-ops.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/rockchip/clk-ops.c b/drivers/clk/rockchip/clk-ops.c
index d80bff0c10d8..3417e27769b0 100644
--- a/drivers/clk/rockchip/clk-ops.c
+++ b/drivers/clk/rockchip/clk-ops.c
@@ -594,6 +594,7 @@ static int clk_3288_dclk_lcdc0_set_rate(struct clk_hw *hw, unsigned long rate,
 		unsigned long parent_rate)
 {
 	struct clk* aclk_vio0 = clk_get(NULL, "aclk_vio0");
+	struct clk* hclk_vio = clk_get(NULL, "hclk_vio");
 	struct clk* parent;
 
 	clk_divider_ops.set_rate(hw, rate, parent_rate);
@@ -606,6 +607,7 @@ static int clk_3288_dclk_lcdc0_set_rate(struct clk_hw *hw, unsigned long rate,
 
 	clk_set_parent(aclk_vio0, parent);
 	clk_set_rate(aclk_vio0, __clk_get_rate(parent));
+	clk_set_rate(hclk_vio, 100*MHZ);
 
 	return 0;
 }
-- 
2.35.3

