From 998c96018fe4c06b4e598377096d455dab81cfe6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Fri, 17 May 2013 17:04:31 +0800
Subject: [PATCH] rk30 i2s : clear format bit when update format

---
 sound/soc/rk29/rk30_i2s.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk30_i2s.c b/sound/soc/rk29/rk30_i2s.c
index 0ac35544e415..96cbcf0f197d 100755
--- a/sound/soc/rk29/rk30_i2s.c
+++ b/sound/soc/rk29/rk30_i2s.c
@@ -259,7 +259,7 @@ static int rockchip_i2s_hw_params(struct snd_pcm_substream *substream,
 	/* Working copies of register */
 	iismod = readl(&(pheadi2s->I2S_TXCR));
 	
-//	iismod &= (~((1<<5)-1));
+	iismod &= (~((1<<5)-1));
 	switch (params_format(params)) {
         case SNDRV_PCM_FORMAT_S8:
         	iismod |= SAMPLE_DATA_8bit;
-- 
2.35.3

