From 3bbdce336792ae04734b399c8ed66c5cfc1773d2 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 7 Mar 2014 17:26:15 +0800
Subject: [PATCH] mmc: core: remove mmc_resume_bus

---
 drivers/mmc/core/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index ad2f92bb35c7..4b6c0cae12c7 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -1645,6 +1645,7 @@ static inline void mmc_bus_put(struct mmc_host *host)
 	spin_unlock_irqrestore(&host->lock, flags);
 }
 
+/*
 int mmc_resume_bus(struct mmc_host *host)
 {
 	unsigned long flags;
@@ -1674,6 +1675,7 @@ int mmc_resume_bus(struct mmc_host *host)
 }
 
 EXPORT_SYMBOL(mmc_resume_bus);
+*/
 
 /*
  * Assign a mmc bus handler to a host. Only one bus handler may control a
-- 
2.35.3

