From b42061c4fa3f72c2ccc515eb095609009e328777 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 11 Jul 2014 18:52:28 +0800
Subject: [PATCH] 3036: fix iomap base addr error

---
 include/linux/rockchip/iomap.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/rockchip/iomap.h b/include/linux/rockchip/iomap.h
index ac555eef33bd..364e2e0914cc 100755
--- a/include/linux/rockchip/iomap.h
+++ b/include/linux/rockchip/iomap.h
@@ -125,11 +125,11 @@
 #define RK3036_GIC_CPU_SIZE		SZ_4K
 #define RK3036_CRU_PHYS			0x20000000
 #define RK3036_CRU_SIZE			SZ_4K
-#define RK3036_DDR_PCTL_PHYS		0x20040000
+#define RK3036_DDR_PCTL_PHYS		0x20004000
 #define RK3036_DDR_PCTL_SIZE		SZ_4K
-#define RK3036_GRF_PHYS         0x20080000
-#define RK3036_GRF_SIZE          SZ_4K
-#define RK3036_DDR_PHY_PHYS		0x200a0000
+#define RK3036_GRF_PHYS			0x20008000
+#define RK3036_GRF_SIZE			SZ_4K
+#define RK3036_DDR_PHY_PHYS		0x2000a000
 #define RK3036_DDR_PHY_SIZE		SZ_4K
 #define RK3036_TIMER_PHYS		0x20044000
 #define RK3036_TIMER_SIZE		SZ_4K
-- 
2.35.3

