From 519dd69f1cf11c9cd229b55fd5fe1ffffa1c0ef0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Tue, 8 Apr 2014 09:24:48 +0800
Subject: [PATCH] rk3288:rk808:disable dcdc uv function

---
 drivers/mfd/rk808.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index f5f2c2e45d6e..b3547f8f7a5c 100755
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -1183,6 +1183,13 @@ static int rk808_pre_init(struct rk808 *rk808)
 {
 	int ret,val;
 	 printk("%s,line=%d\n", __func__,__LINE__);
+	 /**********disable dcdc uv func****************/
+        ret = rk808_reg_write(rk808,RK808_DCDC_UV_ACT_REG,0x10);
+         if (ret <0) {
+                printk(KERN_ERR "Unable to write RK808_DCDC_UV_ACT_REG reg\n");
+                return ret;
+        }
+	/**********************************/
 	 /***********set ILIM ************/
 	val = rk808_reg_read(rk808,RK808_BUCK3_CONFIG_REG);
 	val &= (~(0x7 <<0));
-- 
2.35.3

