From 6b4c9944d9c4ba4334061d90fb400fe5811851b7 Mon Sep 17 00:00:00 2001
From: libing <libing@rock-chips.com>
Date: Sun, 30 Mar 2014 22:51:15 +0800
Subject: [PATCH] rk3288-mipi-dsi: add support mipi dsi.

---
 arch/arm/boot/dts/rk3288.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 824aba921a44..93467e9bef79 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1,6 +1,7 @@
 #include <dt-bindings/clock/ddr.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/rkfb/rk_fb.h>
+#include <dt-bindings/rkmipi/mipi_dsi.h>
 #include <dt-bindings/suspend/rockchip-pm.h>
 #include <dt-bindings/sensor-dev.h>
 
@@ -569,6 +570,22 @@
 	rk_screen: rk_screen{
 			compatible = "rockchip,screen";
 	};
+		
+	dsihost0: mipi@ff960000{
+		compatible = "rockchip,rk32-dsi";
+		rockchip,prop = <0>;
+		reg = <0xff960000 0x4000>;
+		interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
+		status = "okay";
+	};
+
+	dsihost1: mipi@ff964000{
+		compatible = "rockchip,rk32-dsi";
+		rockchip,prop = <1>;
+		reg = <0xff964000 0x4000>;
+		interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
+		status = "disabled"; 
+	};
 	
 	lvds: lvds@ff96c000 {
 		compatible = "rockchip,rk32-lvds";
-- 
2.35.3

