From ea6cdc44a464e61ab367b6669bcd0bed4ec1450c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=B5=B5=E5=AD=90=E5=88=9D?= <zzc@rock-chips.com>
Date: Mon, 23 Sep 2013 15:39:27 +0800
Subject: [PATCH] bp id: change m50 to m51

---
 include/linux/bp-auto.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/bp-auto.h b/include/linux/bp-auto.h
index e70b59ca5fa7..602592ff7d19 100755
--- a/include/linux/bp-auto.h
+++ b/include/linux/bp-auto.h
@@ -32,7 +32,7 @@ enum bp_id{
 	BP_ID_MW100,	//4 thinkwill MW100 WCDMA
 	BP_ID_TD8801,	//5 spreadtrum SC8803 TD-SCDMA
 	BP_ID_SC6610,	//6 spreadtrum SC6610 GSM
-	BP_ID_M50,	//7 spreadtrum RDA GSM
+	BP_ID_M51,	//7 RDA GSM
 	BP_ID_MT6250,   //8 ZINN M50  EDGE
 	BP_ID_C66A,     //9 ZHIGUAN C66A GSM
 	BP_ID_SEW290,   //10 SCV SEW290 WCDMA
-- 
2.35.3

