From dfeca01a38d17344e0f8f078c611f5f6dd2d404a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 16 May 2011 19:54:02 +0800
Subject: [PATCH] rk29: clock: always enable tpiu clock for jtag debug

---
 arch/arm/mach-rk29/clock.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 12705ac78ce4..47bd254caa58 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2476,6 +2476,7 @@ static void __init clk_enable_init_clocks(void)
 	clk_enable_nolock(&clk_intmem);
 	clk_enable_nolock(&clk_ddr);
 	clk_enable_nolock(&clk_debug);
+	clk_enable_nolock(&clk_tpiu);
 	clk_enable_nolock(&clk_jtag);
 	clk_enable_nolock(&clk_uart1);
 }
-- 
2.35.3

