From 821c3958fb705a57adf3b574e38a354882e16d6f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 02:16:09 +0800
Subject: [PATCH] pmem: quiet while request for physical address of pmem region
 from process

---
 drivers/misc/pmem.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/pmem.c b/drivers/misc/pmem.c
index 890831e2deb7..547afd2ebdda 100644
--- a/drivers/misc/pmem.c
+++ b/drivers/misc/pmem.c
@@ -1086,8 +1086,8 @@ static long pmem_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 				region.offset = pmem_start_addr(id, data);
 				region.len = pmem_len(id, data);
 			}
-			printk(KERN_INFO "pmem: request for physical address of pmem region "
-					"from process %d.\n", current->pid);
+			//printk(KERN_INFO "pmem: request for physical address of pmem region "
+			//		"from process %d.\n", current->pid);
 			if (copy_to_user((void __user *)arg, &region,
 						sizeof(struct pmem_region)))
 				return -EFAULT;
-- 
2.35.3

