From c9842f0488964fbe48cf30cc627c9585b97a6950 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 13 Sep 2012 09:28:12 +0800
Subject: [PATCH] rk2928: timer: fix timer clk parent name

---
 arch/arm/mach-rk2928/timer.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk2928/timer.c b/arch/arm/mach-rk2928/timer.c
index 1969681c8285..d0a8241e7faf 100644
--- a/arch/arm/mach-rk2928/timer.c
+++ b/arch/arm/mach-rk2928/timer.c
@@ -125,9 +125,9 @@ static __init int rk2928_timer_init_clockevent(void)
 	struct clock_event_device *ce = &rk2928_timer_clockevent;
 	struct clk *clk = clk_get(NULL, TIMER_CLKEVT_NAME);
 	struct clk *pclk = clk_get(NULL, "pclk_" TIMER_CLKEVT_NAME);
-	struct clk *pclk_periph = clk_get(NULL, "pclk_periph");
+	struct clk *peri_pclk = clk_get(NULL, "peri_pclk");
 
-	clk_set_parent(clk, pclk_periph);
+	clk_set_parent(clk, peri_pclk);
 	clk_enable(pclk);
 	clk_enable(clk);
 
@@ -163,9 +163,9 @@ static void __init rk2928_timer_init_clocksource(void)
 	struct clocksource *cs = &rk2928_timer_clocksource;
 	struct clk *clk = clk_get(NULL, TIMER_CLKSRC_NAME);
 	struct clk *pclk = clk_get(NULL, "pclk_" TIMER_CLKSRC_NAME);
-	struct clk *pclk_periph = clk_get(NULL, "pclk_periph");
+	struct clk *peri_pclk = clk_get(NULL, "peri_pclk");
 
-	clk_set_parent(clk, pclk_periph);
+	clk_set_parent(clk, peri_pclk);
 	clk_enable(pclk);
 	clk_enable(clk);
 
-- 
2.35.3

