From bba149e876d814b5555939d18ac48555ff01d649 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Sun, 7 Apr 2013 17:26:13 +0800
Subject: [PATCH] rk610 codec : modfiy rk610_codec_write_incall funtion

---
 sound/soc/codecs/rk610_codec.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/sound/soc/codecs/rk610_codec.c b/sound/soc/codecs/rk610_codec.c
index d735aba1cc75..d850f7b69ec8 100755
--- a/sound/soc/codecs/rk610_codec.c
+++ b/sound/soc/codecs/rk610_codec.c
@@ -249,25 +249,25 @@ void call_set_spk(int on)
 	{
 	case 0:
 		//modem exit call,codec disable loopback
-		printk("%s modem exit call \n", __FUNCTION__);
-		rk610_codec_write(rk610_codec_codec,ACCELCODEC_R0E, 0x80);
+		printk("%s modem exit call \n", __FUNCTION__);		
+		rk610_codec_write_incall(rk610_codec_codec,ACCELCODEC_R0E, 0x80);
 		rk610_codec->call_enable = 0;	
 		break;
 	case 1:
 		//modem calling,codec enable loopback,spk hp different volume
 		printk("%s spk incalling\n", __FUNCTION__);
-		rk610_codec_write(rk610_codec_codec,ACCELCODEC_R0E, 0x00);
 		rk610_codec->call_enable = 1;	
+		rk610_codec_write_incall(rk610_codec_codec,ACCELCODEC_R0E, 0x00);
 		return;
 	case 2:
 		printk("%s hp incalling\n", __FUNCTION__);
-		rk610_codec_write(rk610_codec_codec,ACCELCODEC_R0E, 0x00);
-		rk610_codec->call_enable = 1;	
+		rk610_codec->call_enable = 1;
+		rk610_codec_write_incall(rk610_codec_codec,ACCELCODEC_R0E, 0x00);	
 		break;
 	case 3:
 		printk("%s bt incalling\n", __FUNCTION__);    
-		rk610_codec_write(rk610_codec_codec,ACCELCODEC_R0E, 0x00);	
-		rk610_codec->call_enable = 1;
+		rk610_codec->call_enable = 1;	
+		rk610_codec_write_incall(rk610_codec_codec,ACCELCODEC_R0E, 0x00);
 		break;
 	}
 
-- 
2.35.3

