From ebc62c46086fd1c20aea668378b6e026c8736345 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Mon, 24 Mar 2014 10:06:53 +0800
Subject: [PATCH] rk3288:rk808:support clkout2 enable when pmic power on

---
 drivers/mfd/rk808.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index 5c31ba7cf9de..0ba09d08553f 100755
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -1233,6 +1233,13 @@ static int rk808_pre_init(struct rk808 *rk808)
                 return ret;
         }
 	/**********************************/
+	/**********enable clkout2****************/
+        ret = rk808_reg_write(rk808,RK808_CLK32OUT_REG,0x01);
+         if (ret <0) {
+                printk(KERN_ERR "Unable to write RK808_CLK32OUT_REG reg\n");
+                return ret;
+        }
+	/**********************************/
 	return 0;
 }
 
-- 
2.35.3

