From 75bdc645cad8ae838796995f8cb69491869a5b6c Mon Sep 17 00:00:00 2001
From: simon <xxm@rock-chips.com>
Date: Fri, 22 Aug 2014 09:30:26 +0800
Subject: [PATCH] rockchip: iommu: use devm_ioremap instead of ioremap when
 probe

---
 drivers/iommu/rockchip-iommu.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 3c1ca9d7c32b..ca87d01f95f8 100755
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -1218,7 +1218,7 @@ for (i = 0; i < pdev->num_resources; i++, res++) {
 			ret = -ENOENT;
 			goto err_res;
 		}
-		data->res_bases[i] = ioremap(res->start, resource_size(res));
+		data->res_bases[i] = devm_ioremap(dev,res->start, resource_size(res));
 		dev_dbg(dev,"res->start = 0x%08x  ioremap to  data->res_bases[%d] = 0x%08x\n",
 			res->start, i, (unsigned int)data->res_bases[i]);
 		if (!data->res_bases[i]) {
@@ -1270,8 +1270,6 @@ for (i = 0; i < pdev->num_resources; i++, res++) {
 
 err_irq:
 err_res:
-	while (data->num_res_mem-- > 0)
-		devm_iounmap(dev,data->res_bases[data->num_res_mem]);
 err_init:
 err_alloc:
 	dev_err(dev, "Failed to initialize\n");
-- 
2.35.3

