From a91d269c8e4ffae5e6f5ac26871d13b7241b1bff Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Mon, 15 Aug 2011 18:30:36 -0700
Subject: [PATCH] rk29 vmac: remove msleep(1000) when vmac resume

---
 drivers/net/rk29_vmac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/rk29_vmac.c b/drivers/net/rk29_vmac.c
index d83f6f4daafc..0f04f92fc076 100755
--- a/drivers/net/rk29_vmac.c
+++ b/drivers/net/rk29_vmac.c
@@ -1207,7 +1207,7 @@ static void rk29_init_vmac(struct net_device *dev)
 	if (pdata && pdata->rmii_power_control)
 		pdata->rmii_power_control(1);
 
-	msleep(1000);
+	//msleep(1000);
 
 	/* IRQ mask */
 	temp = RXINT_MASK | ERR_MASK | TXCH_MASK | MDIO_MASK;
-- 
2.35.3

