From 69b558b528bf73f1bd3041d2a7635c440d229785 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Jun 2011 13:06:52 +0800
Subject: [PATCH] rk29: mmu.c: enlarge vmalloc_reserve to support 1G memory

---
 arch/arm/mm/mmu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mm/mmu.c b/arch/arm/mm/mmu.c
index c2f22eaa388e..0a24e5e04cb3 100644
--- a/arch/arm/mm/mmu.c
+++ b/arch/arm/mm/mmu.c
@@ -664,7 +664,11 @@ void __init iotable_init(struct map_desc *io_desc, int nr)
 		create_mapping(io_desc + i);
 }
 
+#if defined(CONFIG_RK29_MEM_SIZE_M) && CONFIG_RK29_MEM_SIZE_M >= 1024
+static unsigned long __initdata vmalloc_reserve = SZ_128M + (0xFF000000 - VMALLOC_END);
+#else
 static unsigned long __initdata vmalloc_reserve = SZ_128M;
+#endif
 
 /*
  * vmalloc=size forces the vmalloc area to be exactly 'size'
-- 
2.35.3

