From adf525716c46a4e778264cac5b29ad247e589845 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 17 Dec 2014 13:41:37 +0800
Subject: [PATCH] ARM: rk: kernel.img depends on logo_kernel.bmp

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 3d88c39670fd..c2660c67399d 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -341,6 +341,6 @@ kernel.img: zImage
 
 LOGO := $(notdir $(wildcard $(srctree)/logo.bmp))
 LOGO_KERNEL := $(notdir $(wildcard $(srctree)/logo_kernel.bmp))
-%.img: %.dtb kernel.img $(LOGO)
+%.img: %.dtb kernel.img $(LOGO) $(LOGO_KERNEL)
 	$(Q)$(srctree)/resource_tool $(objtree)/arch/arm/boot/dts/$*.dtb $(LOGO) $(LOGO_KERNEL)
 	@echo '  Image:  resource.img (with $*.dtb $(LOGO) $(LOGO_KERNEL)) is ready'
-- 
2.35.3

