From 445f59633d1f0b7bcbcb8c106c9a5aa3b9ba6009 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 23 Mar 2012 09:53:51 +0800
Subject: [PATCH] rk30: default memory size 1G

---
 arch/arm/mach-rk30/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index ae11da0df2cd..33d5bb406212 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -75,6 +75,6 @@ void __init rk30_fixup(struct machine_desc *desc, struct tag *tags,
 {
 	mi->nr_banks = 1;
 	mi->bank[0].start = PLAT_PHYS_OFFSET;
-	mi->bank[0].size = SZ_128M;
+	mi->bank[0].size = SZ_1G;
 }
 
-- 
2.35.3

