From 32adcaf1d9211806292288ce62213047fc7b3e64 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 20 May 2015 22:07:02 +0800
Subject: [PATCH] arm: rockchip: select MFD_SYSCON

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/mach-rockchip/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rockchip/Kconfig b/arch/arm/mach-rockchip/Kconfig
index ec4c8b46283a..9212184b4f56 100755
--- a/arch/arm/mach-rockchip/Kconfig
+++ b/arch/arm/mach-rockchip/Kconfig
@@ -22,5 +22,6 @@ config ARCH_ROCKCHIP
 	select FIQ_GLUE
 	select ARM_ERRATA_818325 if !ARM_TRUSTZONE
 	select ARM_ERRATA_821420 if !ARM_TRUSTZONE
+	select MFD_SYSCON
 
 source "arch/arm/mach-rockchip/Kconfig.common"
-- 
2.35.3

