From 07ac21bb0504812b029bf59318fd2a28ee439188 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 10 Dec 2014 19:40:53 +0800
Subject: [PATCH] ARM: rockchip: common: covert dsb() to dsb(sy)

---
 arch/arm/mach-rockchip/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/common.c b/arch/arm/mach-rockchip/common.c
index 9ee976af9b7e..bd944c53fa3b 100755
--- a/arch/arm/mach-rockchip/common.c
+++ b/arch/arm/mach-rockchip/common.c
@@ -102,7 +102,7 @@ static int __init rockchip_cpu_axi_init(void)
 				iounmap(base);
 		}
 	}
-	dsb();
+	dsb(sy);
 
 #undef MAP
 
-- 
2.35.3

