From e6f34ca54b78f4e7bb7e80df29411ca887910c35 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 20 May 2015 22:09:32 +0800
Subject: [PATCH] usb: rockchip: fix compilation error on arm

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/usb/host/ohci-rockchip.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/host/ohci-rockchip.c b/drivers/usb/host/ohci-rockchip.c
index 4ea543284463..d63a1bbad2bc 100755
--- a/drivers/usb/host/ohci-rockchip.c
+++ b/drivers/usb/host/ohci-rockchip.c
@@ -89,14 +89,18 @@ static const struct hc_driver ohci_rk_hc_driver = {
 };
 
 static struct of_device_id rk_ohci_of_match[] = {
+#ifdef CONFIG_ARM
 	{
 	 .compatible = "rockchip,rk3126_ohci",
 	 .data = &usb20ohci_pdata_rk3126,
 	 },
+#endif
+#ifdef CONFIG_ARM64
 	{
 	 .compatible = "rockchip,rk3368_ohci",
 	 .data = &usb20ohci_pdata_rk3368,
 	 },
+#endif
 	{},
 };
 
-- 
2.35.3

