From c077367b5c1f0b98ab3ae09358dfe619ee8125db Mon Sep 17 00:00:00 2001
From: zhanghao <zhanghao@rock-chips.com>
Date: Mon, 26 Mar 2012 15:56:57 +0800
Subject: [PATCH] CTS : fix cmmb permission

---
 drivers/cmmb/cmmb_class.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cmmb/cmmb_class.c b/drivers/cmmb/cmmb_class.c
index 794db94004d4..349acd9ec911 100755
--- a/drivers/cmmb/cmmb_class.c
+++ b/drivers/cmmb/cmmb_class.c
@@ -186,7 +186,7 @@ ssize_t cmmb_class_show_name(struct class * class, char * buf, size_t count, lof
 	
 }  
 
-static CLASS_ATTR(name, 0777, cmmb_class_show_name, NULL);
+static CLASS_ATTR(name, 0660, cmmb_class_show_name, NULL);
 
 static int __init init_cmmbclass(void)
 {
-- 
2.35.3

