From 6c5a84de53226ec229b9607802e327cf65062607 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 17 Dec 2010 22:03:17 +0800
Subject: [PATCH] support build uu -> o

---
 Makefile               | 4 ++++
 scripts/Makefile.build | 9 +++++++++
 2 files changed, 13 insertions(+)

diff --git a/Makefile b/Makefile
index 240d6a0743a2..7be29d6c48cd 100644
--- a/Makefile
+++ b/Makefile
@@ -1596,3 +1596,7 @@ FORCE:
 # Declare the contents of the .PHONY variable as phony.  We keep that
 # information in a variable so we can use it in if_changed and friends.
 .PHONY: $(PHONY)
+
+
+%.o: %.uu prepare scripts FORCE
+	$(Q)$(MAKE) $(build)=$(build-dir) $(target-dir)$(notdir $@)
diff --git a/scripts/Makefile.build b/scripts/Makefile.build
index 341b58902ffc..17e8638eabba 100644
--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -388,3 +388,12 @@ endif
 # information in a variable se we can use it in if_changed and friends.
 
 .PHONY: $(PHONY)
+
+
+# .uu -> .o
+# ---------------------------------------------------------------------------
+quiet_cmd_uudecode_o_uu = DECODE  $@
+      cmd_uudecode_o_uu = uudecode -o $@ $<
+
+$(obj)/%.o: $(src)/%.uu FORCE
+	$(call if_changed,uudecode_o_uu)
-- 
2.35.3

