From 3502a1e8c0177c9af6a844be6af234f50f3b1fa7 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 16 Jun 2011 16:16:21 +0800
Subject: [PATCH] update adc driver

---
 drivers/adc/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/adc/core.c b/drivers/adc/core.c
index ce00e01c1c97..dea19cc744d9 100755
--- a/drivers/adc/core.c
+++ b/drivers/adc/core.c
@@ -135,7 +135,7 @@ int adc_sync_read(struct adc_client *client)
 		dev_dbg(client->adc->dev, "system enter sleep\n");
 		return -1;
 	}
-	req = kzalloc(sizeof(*req), GFP_KERNEL);
+	req = kzalloc(sizeof(*req), GFP_ATOMIC);
 	if (!req){
 		dev_err(client->adc->dev, "no memory for adc request\n");
 		return -ENOMEM;
@@ -179,7 +179,7 @@ int adc_async_read(struct adc_client *client)
 		dev_dbg(client->adc->dev, "system enter sleep\n");
 		return -1;
 	}
-	req = kzalloc(sizeof(*req), GFP_KERNEL);
+	req = kzalloc(sizeof(*req), GFP_ATOMIC);
 	if (!req) {
 		dev_err(client->adc->dev, "no memory for adc request\n");
 		return -ENOMEM;
-- 
2.35.3

