From 916124aeea15fb72315b44a09cdf759dc8345031 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Mon, 20 May 2013 18:30:43 +0800
Subject: [PATCH] rk29_pcm : temp disable infiniteloop

---
 sound/soc/rk29/rk29_pcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rk29/rk29_pcm.c b/sound/soc/rk29/rk29_pcm.c
index 69b871a3bd98..20e43275ac9f 100755
--- a/sound/soc/rk29/rk29_pcm.c
+++ b/sound/soc/rk29/rk29_pcm.c
@@ -36,7 +36,7 @@
 #define DBG(x...) do { } while (0)
 #endif
 
-#define INFIN_LOOP
+//#define INFIN_LOOP
 #ifdef INFIN_LOOP
 #define DMA_INFIN_LOOP() rk29_dma_has_infiniteloop()
 #else
-- 
2.35.3

