From 0513cdce77f1469b025cfa3ac8be987ef6149af3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 16:57:50 +0800
Subject: [PATCH] block/partitions/rk.c: fix compilation error on arm64

---
 block/partitions/rk.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/block/partitions/rk.c b/block/partitions/rk.c
index 93cb0d6d4b60..0a20ababa44b 100755
--- a/block/partitions/rk.c
+++ b/block/partitions/rk.c
@@ -45,7 +45,7 @@ static struct rk_partition *newpart(char *s,
 		/* No sense support partition less than 8B */
 		if (size < ((PAGE_SIZE) >> 9))
 		{
-			printk(KERN_ERR ERRP "partition size too small (%llx)\n", size);
+			printk(KERN_ERR ERRP "partition size too small (%llx)\n", (u64)size);
 			return NULL;
 		}
 	}
@@ -320,9 +320,9 @@ int rkpart_partition(struct parsed_partitions *state)
 		strcpy(state->parts[i+1].info.volname, parts[i].name);
                 printk(KERN_INFO "%10s: 0x%09llx -- 0x%09llx (%llu MB)\n", 
 				parts[i].name,
-				parts[i].from * 512,
-				(parts[i].from + parts[i].size) * 512,
-				parts[i].size / 2048);
+				(u64)parts[i].from * 512,
+				(u64)(parts[i].from + parts[i].size) * 512,
+				(u64)parts[i].size / 2048);
 	}
 
 	rkpart_bootmode_fixup();
-- 
2.35.3

