From 7f9abe9bcfe9b1a96f1f05b76723ad4c14ea0689 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Thu, 12 Mar 2015 09:32:00 -0700
Subject: [PATCH] rk3288:tsadc:enable tsadc chn3

---
 drivers/hwmon/rockchip_tsadc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/rockchip_tsadc.c b/drivers/hwmon/rockchip_tsadc.c
index 4290f1a6ae51..22691294087c 100755
--- a/drivers/hwmon/rockchip_tsadc.c
+++ b/drivers/hwmon/rockchip_tsadc.c
@@ -277,7 +277,7 @@ static void rockchip_tsadc_auto_mode_set(int chn, int int_temp,
 	
 	msleep(10);
 	tsadc_writel(0, TSADC_AUTO_CON);
-	tsadc_writel(1 << (4+chn), TSADC_AUTO_CON);
+	tsadc_writel(3 << (4+chn), TSADC_AUTO_CON);
 	msleep(10);
 	if ((tsadc_readl(TSADC_AUTO_CON) & TSADC_AUTO_STAS_BUSY_MASK) != TSADC_AUTO_STAS_BUSY) {
 		rockchip_tsadc_set_cmpn_int_vale(chn,int_temp);
-- 
2.35.3

