From 709bdba41b712815897a3d3465fdeac8ab96853b Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Wed, 8 Oct 2014 08:20:33 +0800
Subject: [PATCH] fix pio complie warning

---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 4d3cf7c67862..537ae42067bc 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -2961,7 +2961,7 @@ static void dw_mci_work_routine_card(struct work_struct *work)
 
                 /* Stop edma when rountine card triggered */
                 if(cpu_is_rk3036() || cpu_is_rk312x())
-		        if(host->dma_ops->stop)
+		        if(host->dma_ops && host->dma_ops->stop)
 		                host->dma_ops->stop(host);
 
 		while (present != slot->last_detect_state) {
-- 
2.35.3

