From d7b65047d4b1eb10736e15662b1549f88677a4f7 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 12 Sep 2014 18:06:18 +0800
Subject: [PATCH] mmc: move cap with hs200 to caps

Signed-off-by: lintao <lintao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index b09e0333e91b..41e3c20573af 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -1112,7 +1112,7 @@ static void dw_mci_setup_bus(struct dw_mci_slot *slot, bool force_clkinit)
                          */
                         if ((div == 0) &&
                                 (host->mmc->caps & (MMC_CAP_1_8V_DDR | MMC_CAP_1_2V_DDR)) &&
-                                !(host->mmc->caps & MMC_CAP2_HS200)) {
+                                !(host->mmc->caps2 & MMC_CAP2_HS200)) {
                                 /*  Fixup DDR MMC */
                                 div = 1;
                                 host->set_div = div;
-- 
2.35.3

