From 169fda383f2e64fea53f47e30bf32ab5b42f003a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 1 Apr 2014 20:07:18 +0800
Subject: [PATCH] rk: cpufreq: adjust for rk3288

---
 drivers/cpufreq/cpufreq_interactive.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/drivers/cpufreq/cpufreq_interactive.c b/drivers/cpufreq/cpufreq_interactive.c
index 4f61467f2421..6170da2a6830 100644
--- a/drivers/cpufreq/cpufreq_interactive.c
+++ b/drivers/cpufreq/cpufreq_interactive.c
@@ -1167,16 +1167,14 @@ static int cpufreq_governor_interactive(struct cpufreq_policy *policy,
 			unsigned int index;
 			freq_table = cpufreq_frequency_get_table(policy->cpu);
 			tunables->hispeed_freq = policy->max;
-			if (policy->min < 816000)
-				tunables->hispeed_freq = 816000;
+			if (policy->min < 600000)
+				tunables->hispeed_freq = 600000;
 			else if (cpufreq_frequency_table_target(policy, freq_table, policy->min + 1, CPUFREQ_RELATION_L, &index) == 0)
 				tunables->hispeed_freq = freq_table[index].frequency;
-			if (policy->max > 1416000) {
-				tunables->timer_slack_val = 20 * USEC_PER_MSEC;
-				tunables->min_sample_time = 40 * USEC_PER_MSEC;
-				tunables->above_hispeed_delay[0] = 80 * USEC_PER_MSEC;
-				store_target_loads(tunables, "70 1200000:80 1416000:99", 0);
-			}
+			tunables->timer_slack_val = 20 * USEC_PER_MSEC;
+			tunables->min_sample_time = 40 * USEC_PER_MSEC;
+			tunables->above_hispeed_delay[0] = 80 * USEC_PER_MSEC;
+			store_target_loads(tunables, "70 1008000:80 1296000:99", 0);
 			tunables->boostpulse_duration_val = 500 * USEC_PER_MSEC;
 		}
 #endif
-- 
2.35.3

