From 895f21d5a7351de9eaf72eb8c2def3be2966e45d Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Wed, 29 Apr 2015 14:39:18 +0800
Subject: [PATCH] rk fb: kfree envp[1], if we not do real switch screen

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index d76c8876b5a1..cefd38046ca9 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3250,6 +3250,7 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 		if (dev_drv->cur_screen->type != screen->type) {
 			dev_drv->hdmi_switch = 0;
 			mutex_unlock(&dev_drv->switch_screen);
+			kfree(envp[1]);
 			return 0;
 		}
 
-- 
2.35.3

