From da94e41f433daedea0ff711d6a08440b1aaa0a11 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Mon, 21 Apr 2014 19:14:37 +0800
Subject: [PATCH] backlight: power-off the backlight,when reboot

---
 drivers/video/backlight/pwm_bl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index 209d5a085f9d..ec14d6793e1a 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -393,6 +393,7 @@ static struct platform_driver pwm_backlight_driver = {
 	},
 	.probe		= pwm_backlight_probe,
 	.remove		= pwm_backlight_remove,
+	.shutdown       = pwm_backlight_remove,
 };
 
 module_platform_driver(pwm_backlight_driver);
-- 
2.35.3

