From 197d36cdb7c284bb5e0c86dd51ab3bac12db740d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Nov 2011 19:53:00 +0800
Subject: [PATCH] pack-kernel.sh: merge from stable, remove 5625

---
 pack-kernel.sh | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 6bb24bf0bd16..c8e6c1fc0f00 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -8,7 +8,6 @@ arch/arm/mach-rk29/ddr.c
 arch/arm/mach-rk29/vpu*.c
 
 drivers/staging/rk29/ipp/rk29-ipp.c
-sound/soc/codecs/rt5625.c
 )
 
 DIRS=(
@@ -93,11 +92,6 @@ for d in ${DIRS[@]}; do
 	[ -d $d ] && find $d -type f -name '*.uu' -print0 | xargs -0 rm -f
 done
 
-#zwp
-mv sound/soc/codecs/rt5625.c sound/soc/codecs/rt5625.c.bak
-mv sound/soc/codecs/rt5625.c.enc sound/soc/codecs/rt5625.c
-####
-
 echo build kernel on $kerndir with $defconfig
 make clean >/dev/null 2>&1
 make $defconfig >/dev/null 2>&1
@@ -119,11 +113,6 @@ for d in ${DIRS[@]}; do
 	done
 done
 
-#zwp
-mv sound/soc/codecs/rt5625.c sound/soc/codecs/rt5625.c.enc
-mv sound/soc/codecs/rt5625.c.bak sound/soc/codecs/rt5625.c
-####
-
 make distclean >/dev/null 2>&1
 
 popd >/dev/null
@@ -139,8 +128,6 @@ done
 for file in ${EXCLUDES[@]}; do
 	echo "$file" >> $ex
 done
-echo sound/soc/codecs/rt5625.c >>$ex
-echo sound/soc/codecs/rt5625.c.enc >>$ex
 echo TAR $(pwd)/$package
 tar cf $package --numeric-owner --exclude-from $ex --exclude=.git --exclude=`basename $0` $(basename $kerndir)
 tar rf $package --numeric-owner --exclude=.git toolchain/arm-eabi-4.4.0
-- 
2.35.3

