From 2dee5b520113ad3f1675068117382efbcbb2836b Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 31 Jul 2014 18:06:34 +0800
Subject: [PATCH] rk321x.dtsi: add tve node.

---
 arch/arm/boot/dts/rk312x.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 630506037939..9bbb421f3088 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -642,6 +642,12 @@
 		status = "disabled";
 	};
 
+	tve: tve{
+		compatible = "rockchip,rk312x-tve";
+		reg = <0x1011e200 0x100>;
+		status = "disabled";
+	};
+
         vpu: vpu_service@10104000 {
 		compatible = "vpu_service";
 		reg = <0x10104000 0x800>;
-- 
2.35.3

