From 6946c14ef02e78f81e52e6fb7e05e92227bcc6d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 17:03:52 +0800
Subject: [PATCH] fiq_debugger: rockchip: fix compilation error on arm64

---
 .../android/fiq_debugger/fiq_debugger.c       | 21 +++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/android/fiq_debugger/fiq_debugger.c b/drivers/staging/android/fiq_debugger/fiq_debugger.c
index b0204f398357..b02cfe89eb0c 100755
--- a/drivers/staging/android/fiq_debugger/fiq_debugger.c
+++ b/drivers/staging/android/fiq_debugger/fiq_debugger.c
@@ -361,6 +361,8 @@ static void fiq_debugger_dump_irqs(struct fiq_debugger_state *state)
 		}
 	}
 	for (n = 0; n < NR_IPI; n++) {
+#define S(x,s)	[x] = s
+#ifdef CONFIG_ARM
 		enum ipi_msg_type {
 			IPI_WAKEUP,
 			IPI_TIMER,
@@ -372,7 +374,6 @@ static void fiq_debugger_dump_irqs(struct fiq_debugger_state *state)
 			IPI_CPU_BACKTRACE,
 		};
 		static const char *ipi_types[NR_IPI] = {
-#define S(x,s)	[x] = s
 			S(IPI_WAKEUP, "CPU wakeup"),
 			S(IPI_TIMER, "Timer broadcast"),
 			S(IPI_RESCHEDULE, "Rescheduling"),
@@ -381,8 +382,24 @@ static void fiq_debugger_dump_irqs(struct fiq_debugger_state *state)
 			S(IPI_CPU_STOP, "CPU stop"),
 			S(IPI_COMPLETION, "Completion"),
 			S(IPI_CPU_BACKTRACE, "CPU backtrace"),
-#undef S
 		};
+#elif defined(CONFIG_ARM64)
+		enum ipi_msg_type {
+			IPI_RESCHEDULE,
+			IPI_CALL_FUNC,
+			IPI_CALL_FUNC_SINGLE,
+			IPI_CPU_STOP,
+			IPI_TIMER,
+		};
+		static const char *ipi_types[NR_IPI] = {
+			S(IPI_RESCHEDULE, "Rescheduling"),
+			S(IPI_CALL_FUNC, "Function call"),
+			S(IPI_CALL_FUNC_SINGLE, "Single function call"),
+			S(IPI_CPU_STOP, "CPU stop"),
+			S(IPI_TIMER, "Timer broadcast"),
+		};
+#endif
+#undef S
 		for (cpu = 0; cpu < NR_CPUS; cpu++) {
 			unsigned int irqs = __get_irq_stat(cpu, ipi_irqs[n]);
 			if (irqs == 0)
-- 
2.35.3

