From 4cd9825cd81ec37a054bb8bc3e7412e82166ca75 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:29:00 +0800
Subject: [PATCH] rk29_fb: fix for v2.6.36

---
 drivers/video/rk29_fb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 3c20aa3d2987..d7a94fc0672d 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -2531,7 +2531,7 @@ static struct suspend_info suspend_info = {
 };
 #endif
 struct fb_info *g_fb0_inf = NULL;  //add cym@rk 20101027 for charger logo
-static int __init rk29fb_probe (struct platform_device *pdev)
+static int __devinit rk29fb_probe (struct platform_device *pdev)
 {
     struct rk29fb_inf *inf = NULL;
     struct resource *res = NULL;
@@ -2887,7 +2887,7 @@ static int __init rk29fb_probe (struct platform_device *pdev)
 	return ret;
 }
 
-static int rk29fb_remove(struct platform_device *pdev)
+static int __devexit rk29fb_remove(struct platform_device *pdev)
 {
     struct rk29fb_inf *inf = platform_get_drvdata(pdev);
     struct fb_info *info = NULL;
@@ -3012,7 +3012,7 @@ static void rk29fb_shutdown(struct platform_device *pdev)
 
 static struct platform_driver rk29fb_driver = {
 	.probe		= rk29fb_probe,
-	.remove		= rk29fb_remove,
+	.remove		= __devexit_p(rk29fb_remove),
 	.driver		= {
 		.name	= "rk29-fb",
 		.owner	= THIS_MODULE,
-- 
2.35.3

