From 40681a7238decff47de0414ecbf92215cf44d632 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Sat, 11 Feb 2012 16:19:45 +0800
Subject: [PATCH] incalling cannot close I2S mclk

---
 arch/arm/mach-rk29/pm.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-rk29/pm.c b/arch/arm/mach-rk29/pm.c
index dd05338146d1..0deac233de86 100755
--- a/arch/arm/mach-rk29/pm.c
+++ b/arch/arm/mach-rk29/pm.c
@@ -460,6 +460,15 @@ static int rk29_pm_enter(suspend_state_t state)
 		   | (1 << CLK_GATE_TPIU)
 #endif
 		   ) | clkgate[0], CRU_CLKGATE0_CON);
+
+#ifdef CONFIG_PHONE_INCALL_IS_SUSPEND
+#if defined(CONFIG_SND_RK29_SOC_I2S_8CH)
+	cru_writel(clkgate[0]&(~(1<<CLK_GATE_I2S0)),CRU_CLKGATE0_CON);
+#elif defined(CONFIG_SND_RK29_SOC_I2S_2CH)
+	cru_writel(clkgate[0]&(~(1<<CLK_GATE_I2S1)),CRU_CLKGATE0_CON);
+#endif
+#endif	
+
 	cru_writel(~0, CRU_CLKGATE1_CON);
 	cru_writel(~((1 << CLK_GATE_GPIO1 % 32)
 		   | (1 << CLK_GATE_GPIO2 % 32)
-- 
2.35.3

