From 8b5624865bc5f9c9d2bd19712a3f185f5fae5c6d Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 8 May 2014 10:19:28 +0800
Subject: [PATCH] rk fb: print last timeout value when pagefault

---
 drivers/video/rockchip/rk_fb.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9074f2e18724..e22684a7e822 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -977,6 +977,7 @@ void rk_fd_fence_wait(struct rk_lcdc_driver *dev_drv,
 
 #ifdef CONFIG_ROCKCHIP_IOMMU
 static int g_last_addr[4];
+int g_last_timeout;
 u32 freed_addr[10];
 u32 freed_index;
 
@@ -1003,6 +1004,7 @@ int rk_fb_sysmmu_fault_handler(struct device *dev,
 	printk("last freed buffer:\n");
 	for (i = 0; freed_addr[i] != 0xfefefefe; i++)
 		printk("%d:0x%08x\n",i, freed_addr[i]);
+	printk("last timeout:%d\n", g_last_timeout);
 	dev_drv->ops->get_disp_info(dev_drv, buf,0) ;
 	for (i = 0; i < PAGE_SIZE; i += DUMP_CHUNK) {
 		if ((PAGE_SIZE - i) > DUMP_CHUNK) {
@@ -1252,6 +1254,7 @@ static void rk_fb_update_reg(struct rk_lcdc_driver * dev_drv,struct rk_fb_reg_da
 #if defined(CONFIG_ROCKCHIP_IOMMU)
 		if (dev_drv->iommu_enabled) {
 			freed_index = 0;
+			g_last_timeout = timeout;
 			if (timeout >= 3)
 				msleep(15);
 		}
-- 
2.35.3

