From b518aad2a6fe47f5825e2e0a13cf2af691449aa6 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Mon, 15 Oct 2012 17:47:11 +0800
Subject: [PATCH] rk3066b m701:set backlight delay_ms zero to make resume fast

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index 4c21be42be83..1c4c3e53bf9f 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -286,6 +286,7 @@ static struct rk29_bl_info rk29_bl_info = {
 	.io_init = rk29_backlight_io_init,
 	.io_deinit = rk29_backlight_io_deinit,
 	.min_brightness = 48,
+	.delay_ms = 0,
 	.pwm_suspend = rk29_backlight_pwm_suspend,
 	.pwm_resume = rk29_backlight_pwm_resume,
 };
-- 
2.35.3

