From 8d0ba95c947aa5838e6381639038dae61a699835 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Fri, 17 Oct 2014 09:17:41 +0800
Subject: [PATCH] DMA:pl330: add burst length config

---
 drivers/dma/pl330.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/dma/pl330.c b/drivers/dma/pl330.c
index 2ea17d0f2970..5a5614d83f34 100755
--- a/drivers/dma/pl330.c
+++ b/drivers/dma/pl330.c
@@ -2918,7 +2918,11 @@ static struct dma_async_tx_descriptor *pl330_prep_dma_cyclic(
 		}
 
 		desc->rqcfg.brst_size = pch->burst_sz;
+#ifdef CONFIG_ARCH_ROCKCHIP
+		desc->rqcfg.brst_len = pch->burst_len;
+#else
 		desc->rqcfg.brst_len = 1;
+#endif
 		desc->req.infiniteloop = *infinite;
 		fill_px(&desc->px, dst, src, period_len);
 
@@ -3047,7 +3051,11 @@ pl330_prep_slave_sg(struct dma_chan *chan, struct scatterlist *sgl,
 		}
 
 		desc->rqcfg.brst_size = pch->burst_sz;
+#ifdef CONFIG_ARCH_ROCKCHIP
+		desc->rqcfg.brst_len = pch->burst_len;
+#else
 		desc->rqcfg.brst_len = 1;
+#endif
 	}
 
 	/* Return the last desc in the chain */
-- 
2.35.3

