From 859a2839869c0e1d7484dae361e7c284744d5ef6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Tue, 19 Apr 2011 11:39:19 +0800
Subject: [PATCH] use foreground colour to set hardware cursor

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index bcb756e6011c..e634ffe29733 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -825,7 +825,7 @@ int rk29_set_cursor_img_transform(char *data)
 			} else {
 				dmsk <<= 1;
 			}
-			op = (dmsk & 0x80) ? 1 : 3;
+			op = (dmsk & 0x80) ? 2 : 3;
 			shift = (x*2)%8;
 			offset = x/4;
 			RK29_CURSOR_WRITE_BIT((dst+offset),(3<<shift),(op<<shift));
-- 
2.35.3

