From aff184b0e3f6153738831e5b737cc396a3fd59d0 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Mon, 19 Dec 2011 17:19:49 +0800
Subject: [PATCH] rtl8192c not available if RFKILL is selected

---
 drivers/net/wireless/rtl8192c/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/rtl8192c/Kconfig b/drivers/net/wireless/rtl8192c/Kconfig
index a0f326be561e..88d1a6acd326 100755
--- a/drivers/net/wireless/rtl8192c/Kconfig
+++ b/drivers/net/wireless/rtl8192c/Kconfig
@@ -1,6 +1,7 @@
 config RTL8192CU
 	tristate "Realtek 8192C USB WiFi"
 	depends on USB
+	depends on !RFKILL
 	select WIRELESS_EXT
 	select WEXT_PRIV
 	select IEEE80211
-- 
2.35.3

