From 1f53f27f83a879ec3f05b3aec0422cb775eefc13 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Thu, 24 Apr 2014 17:47:51 +0800
Subject: [PATCH] iommu: delete log when map and unmap

---
 drivers/iommu/rockchip-iovmm.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/iommu/rockchip-iovmm.c b/drivers/iommu/rockchip-iovmm.c
index 90fcefc08bae..6ba3d084cef8 100755
--- a/drivers/iommu/rockchip-iovmm.c
+++ b/drivers/iommu/rockchip-iovmm.c
@@ -136,9 +136,9 @@ dma_addr_t iovmm_map(struct device *dev,struct scatterlist *sg, off_t offset,siz
 	spin_unlock(&vmm->lock);
 
 	rockchip_sysmmu_tlb_invalidate(dev);
-
+	/*
 	pr_err("IOVMM: Allocated VM region @ %#x/%#X bytes.\n",region->start, region->size);
-
+	*/
 	return region->start;
 
 err_map_map:
@@ -182,8 +182,9 @@ void iovmm_unmap(struct device *dev, dma_addr_t iova)
 	gen_pool_free(vmm->vmm_pool, region->start, region->size);
 
 	WARN_ON(unmapped_size != region->size);
+	/*
 	pr_err("IOVMM: Unmapped %#x bytes from %#x.\n",unmapped_size, region->start);
-
+	*/
 	kfree(region);
 }
 
-- 
2.35.3

