From 8deea5364256e55648afce5028f72228cf3fd464 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Tue, 8 Jan 2013 17:18:46 +0800
Subject: [PATCH] i2c: check_idle: gpio_free

---
 drivers/i2c/busses/i2c-rk30-adapter.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/i2c/busses/i2c-rk30-adapter.c b/drivers/i2c/busses/i2c-rk30-adapter.c
index 88e1ea62aa1c..4f71bf80ee8f 100755
--- a/drivers/i2c/busses/i2c-rk30-adapter.c
+++ b/drivers/i2c/busses/i2c-rk30-adapter.c
@@ -134,6 +134,9 @@ static int rk30_i2c_check_idle(struct rk30_i2c *i2c)
 	sda_lev = gpio_get_value(sda_io);
 	scl_lev = gpio_get_value(scl_io);
 
+	gpio_free(sda_io);
+	gpio_free(scl_io);
+
 	iomux_set(i2c->sda_mode);
 	iomux_set(i2c->scl_mode);
 
-- 
2.35.3

