From 6033c470b8401610117f9ea844f8431e40e9e001 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Jan 2013 11:49:13 +0800
Subject: [PATCH] adc: support rk3188

---
 drivers/adc/plat/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/adc/plat/Kconfig b/drivers/adc/plat/Kconfig
index 7666ff7acc0a..928b58911949 100644
--- a/drivers/adc/plat/Kconfig
+++ b/drivers/adc/plat/Kconfig
@@ -5,7 +5,7 @@
 choice
 	prompt "ADC hardware drivers"
 	default ADC_RK29 if ARCH_RK29
-	default ADC_RK30 if ARCH_RK30
+	default ADC_RK30 if ARCH_RK30 || ARCH_RK2928 || ARCH_RK3188
 
 config ADC_NULL
 	bool "NULL"
-- 
2.35.3

