From 4590ad36d4030b876565a0873a395aad7a35129a Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 21 May 2013 10:23:45 +0800
Subject: [PATCH] rk: fix compile warning of plat-rk/clock.c

---
 arch/arm/plat-rk/clock.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-rk/clock.c b/arch/arm/plat-rk/clock.c
index 473d161354a8..4d674327c29a 100644
--- a/arch/arm/plat-rk/clock.c
+++ b/arch/arm/plat-rk/clock.c
@@ -492,8 +492,10 @@ void clk_disable(struct clk *clk)
 {
 	if (clk == NULL || IS_ERR(clk))
 		return;
-	if (dvfs_support_clk_disable(clk->dvfs_info)==true)
-		return dvfs_vd_clk_disable(clk, 0);
+	if (dvfs_support_clk_disable(clk->dvfs_info) == true) {
+		dvfs_vd_clk_disable(clk, 0);
+		return;
+	}
 
 	LOCK();
 	clk_disable_nolock(clk);
-- 
2.35.3

