From c9bbb3032c50e9b9ecaca58a4aa06ef4192bdbba Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Mon, 2 Mar 2015 11:33:35 +0800
Subject: [PATCH] rk: uboot_mem_reserve: fix print format

Signed-off-by: CMY <cmy@rock-chips.com>
---
 arch/arm/mach-rockchip/common.c | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git a/arch/arm/mach-rockchip/common.c b/arch/arm/mach-rockchip/common.c
index a3d129a0a272..96cf6c760eef 100755
--- a/arch/arm/mach-rockchip/common.c
+++ b/arch/arm/mach-rockchip/common.c
@@ -306,15 +306,16 @@ phys_addr_t uboot_logo_offset=0;
 
 void __init rockchip_uboot_mem_reserve(void)
 {
-	if (uboot_logo_size) {
-		if (!memblock_is_region_reserved(uboot_logo_base, uboot_logo_size)
-			&& !memblock_reserve(uboot_logo_base, uboot_logo_size)){
-			pr_info("%s: reserve %zx@%zx for uboot logo\n", __func__,
-				uboot_logo_size, uboot_logo_base);
-		} else {
-			pr_err("%s: reserve of %zx@%zx failed\n", __func__,
-			       uboot_logo_size, uboot_logo_base);
-		}
+	if (uboot_logo_size=0)
+		return;
+
+	if (!memblock_is_region_reserved(uboot_logo_base, uboot_logo_size)
+	    && !memblock_reserve(uboot_logo_base, uboot_logo_size)){
+		pr_info("%s: reserve %pa@%pa for uboot logo\n", __func__,
+			&uboot_logo_size, &uboot_logo_base);
+	} else {
+		pr_err("%s: reserve of %pa@%pa failed\n", __func__,
+		       &uboot_logo_size, &uboot_logo_base);
 	}
 }
 
@@ -330,8 +331,8 @@ static int __init rockchip_uboot_logo_setup(char *p)
 		}
 	}
 
-	pr_info("%s: mem: %zx@%zx, offset:%zx\n", __func__,
-		uboot_logo_size, uboot_logo_base, uboot_logo_offset);
+	pr_info("%s: mem: %pa@%pa, offset:%pa\n", __func__,
+		&uboot_logo_size, &uboot_logo_base, &uboot_logo_offset);
 
 	return 0;
 }
@@ -346,8 +347,8 @@ static int __init rockchip_uboot_mem_late_init(void)
 		addr = PAGE_ALIGN(uboot_logo_base);
 		end = (uboot_logo_base+uboot_logo_size)&PAGE_MASK;
 
-		pr_info("%s: Freeing uboot logo memory: %zx@%zx\n", __func__,
-			uboot_logo_size, uboot_logo_base);
+		pr_info("%s: Freeing uboot logo memory: %pa@%pa\n", __func__,
+			&uboot_logo_size, &uboot_logo_base);
 
 		memblock_free(uboot_logo_base, uboot_logo_size);
 
-- 
2.35.3

