From 21c1d0e049cbdb38167aaadf7d94dc4be6a750af Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Tue, 17 Mar 2015 14:23:19 +0800
Subject: [PATCH] fix conpile error

---
 drivers/mmc/host/rk_sdmmc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index b4b2ee921c1e..1276d727e680 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -4149,7 +4149,7 @@ EXPORT_SYMBOL(dw_mci_remove);
 extern int get_wifi_chip_type(void);
 int dw_mci_suspend(struct dw_mci *host)
 {
-	int present = dw_mci_get_cd(mmc);
+	int present = dw_mci_get_cd(host->mmc);
 
 	if((host->mmc->restrict_caps & RESTRICT_CARD_TYPE_SDIO) &&
 		(get_wifi_chip_type() == WIFI_ESP8089 || get_wifi_chip_type() > WIFI_AP6XXX_SERIES))
@@ -4187,7 +4187,7 @@ int dw_mci_resume(struct dw_mci *host)
 	int i, ret;
 	u32 regs;
 	struct dw_mci_slot *slot;
-	int present = dw_mci_get_cd(mmc);
+	int present = dw_mci_get_cd(host->mmc);
 
 	if ((host->mmc->restrict_caps & RESTRICT_CARD_TYPE_SDIO) &&
 		(get_wifi_chip_type() == WIFI_ESP8089 ||
-- 
2.35.3

