From 3292ba3d84aa359f889326aedc777c865261f0c0 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Tue, 17 Mar 2015 11:43:32 +0800
Subject: [PATCH] wifi: use buildin instand of ko for compatibility

---
 drivers/net/wireless/Kconfig | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/Kconfig b/drivers/net/wireless/Kconfig
index a8b40006b7d7..94c07fe4a96e 100755
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -25,16 +25,30 @@ if WLAN
 #	  bool "rda 5990p"
 #	  ---help---
 #	    rda5990P fm bt wifi
+	    
+menuconfig RTL_WIRELESS_SOLUTION
+    bool "Realtek Wireless Device Driver Support"
+    default y
+
+if RTL_WIRELESS_SOLUTION
+choice
+    prompt "Realtek WiFi Device Driver Support"
+    default RTL8188EU
+
+    config RTL_WIFI_NONE
+    bool "No Realtek WiFi"
 
 source "drivers/net/wireless/rockchip_wlan/rtl8188eu/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8189es/Kconfig"
-#source "drivers/net/wireless/rockchip_wlan/rtl8192cu/Kconfig"
+source "drivers/net/wireless/rockchip_wlan/rtl8192cu/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8192du/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8723au/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8723bu/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8723bs/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8723bs-vq0/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/rtl8812au/Kconfig"
+endchoice
+endif
 
 #source "drivers/net/wireless/rockchip_wlan/mt5931/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/esp8089/Kconfig"
-- 
2.35.3

