From 5ee3dd9a454e2c52fb65c1bd3e09ac5255ba005d Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 20 May 2014 12:13:58 +0800
Subject: [PATCH] rk fb: sync with hwc when wait at fb

---
 drivers/video/rockchip/rk_fb.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 45c685990ac3..c71126089d79 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1279,6 +1279,9 @@ void rk_fb_free_dma_buf(struct rk_lcdc_driver *dev_drv,struct rk_fb_reg_win_data
 #endif
 		if(area_data->ion_handle != NULL)
 			ion_free(rk_fb->ion_client, area_data->ion_handle);
+		if(area_data->acq_fence) {
+			sync_fence_put(area_data->acq_fence);
+		}
 	}
 	memset(reg_win_data, 0, sizeof(struct rk_fb_reg_win_data));
 }
-- 
2.35.3

