From 7f889b469708387dfd61dc8d5627b4e7555f02bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Thu, 3 Jul 2014 20:58:01 -0700
Subject: [PATCH] change ddrfreq directly in ddr_init() since lcdc is not init
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 陈亮 <cl@rock-chips.com>
---
 arch/arm/mach-rockchip/ddr_rk32.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rockchip/ddr_rk32.c b/arch/arm/mach-rockchip/ddr_rk32.c
index 9381089806a3..27367cec5b72 100755
--- a/arch/arm/mach-rockchip/ddr_rk32.c
+++ b/arch/arm/mach-rockchip/ddr_rk32.c
@@ -4025,6 +4025,9 @@ static int _ddr_change_freq(uint32 nMHz)
 	{
 		ddr_freq_t.screen_ft_us = rk_fb_get_prmry_screen_ft();
 		ddr_freq_t.t0 = rk_fb_get_prmry_screen_framedone_t();
+		if (!ddr_freq_t.screen_ft_us)
+			return __ddr_change_freq(nMHz, ddr_freq_t);
+
 		tmp = cpu_clock(0) - ddr_freq_t.t0;
 		do_div(tmp, 1000);
 		pass_t = tmp;
-- 
2.35.3

