From f4b9b8d34c0c967d7d6bf614d06df25ced1a5cde Mon Sep 17 00:00:00 2001
From: zsq <zsq@rock-chips.com>
Date: Fri, 14 Mar 2014 09:43:25 +0800
Subject: [PATCH] fix rga DMA buf ref error

---
 drivers/video/rockchip/rga/rga_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index b8e38f5328dd..e34920832d19 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -802,6 +802,7 @@ static int rga_convert_dma_buf(struct rga_req *req)
 	    ion_phys(drvdata->ion_client, hdl, &phy_addr, &len);
         req->src.yrgb_addr = phy_addr;
         req->src.uv_addr = req->src.yrgb_addr + (req->src.vir_w * req->src.vir_h);
+        ion_free(drvdata->ion_client, hdl);
     }
     else {
         req->src.yrgb_addr = req->src.uv_addr;
@@ -813,6 +814,7 @@ static int rga_convert_dma_buf(struct rga_req *req)
 	    ion_phys(drvdata->ion_client, hdl, &phy_addr, &len);
         req->dst.yrgb_addr = phy_addr;
         req->dst.uv_addr = req->dst.yrgb_addr + (req->dst.vir_w * req->dst.vir_h);
+        ion_free(drvdata->ion_client, hdl);
     }
     else {
         req->dst.yrgb_addr = req->dst.uv_addr;
-- 
2.35.3

