From 1126eaeb1a909d8aec18da75d6f5be3410f141b7 Mon Sep 17 00:00:00 2001
From: zyk <“zyk@rock-chips.com”>
Date: Thu, 20 Mar 2014 17:22:39 +0800
Subject: [PATCH] TP goodix: change init time

---
 drivers/input/touchscreen/rk29_i2c_goodix.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/rk29_i2c_goodix.c b/drivers/input/touchscreen/rk29_i2c_goodix.c
index 87f765e4331e..f2408940bbe3 100755
--- a/drivers/input/touchscreen/rk29_i2c_goodix.c
+++ b/drivers/input/touchscreen/rk29_i2c_goodix.c
@@ -752,7 +752,7 @@ static int rk_ts_probe(struct i2c_client *client, const struct i2c_device_id *id
 			dev_err(&client->dev, "goodix power error\n");
 			return -EIO;
 		}
-		msleep(100);
+		//msleep(100);
 	} else {
 		dev_info(&client->dev, "power pin invalid\n");
 	}
@@ -764,7 +764,7 @@ static int rk_ts_probe(struct i2c_client *client, const struct i2c_device_id *id
 			dev_err(&client->dev, "goodix gpio_request error\n");
 			return -EIO;
 		}
-                msleep(100);
+             //  msleep(100);
 	} else {
 		dev_info(&client->dev, "reset pin invalid\n");
 	}
-- 
2.35.3

