From 1c209c256f686d8e5d806f48ebbe029f4545df8a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 26 Mar 2011 12:08:37 +0800
Subject: [PATCH] rk29: call resume_console while dpm_drv_timeout, make debug
 more easy

---
 drivers/base/power/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/base/power/main.c b/drivers/base/power/main.c
index 0855a10337ee..079024a40a9d 100644
--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -453,6 +453,11 @@ static void dpm_drv_timeout(unsigned long data)
 	struct device *dev = dpm_drv_wd_data.dev;
 	struct task_struct *tsk = dpm_drv_wd_data.tsk;
 
+#ifdef CONFIG_ARCH_RK29
+#include <linux/console.h>
+	resume_console();
+#endif
+
 	printk(KERN_EMERG "**** DPM device timeout: %s (%s)\n", dev_name(dev),
 	       (dev->driver ? dev->driver->name : "no driver"));
 
-- 
2.35.3

