From 6572226b230c72016a159ffacae0c548a5ee20d3 Mon Sep 17 00:00:00 2001
From: ddl <ddl@rock-chips.com>
Date: Fri, 6 Aug 2010 18:09:49 +0800
Subject: [PATCH] Resume Makefile

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 806fd6a7af09..d7cf5448ee7b 100644
--- a/Makefile
+++ b/Makefile
@@ -184,7 +184,7 @@ export KBUILD_BUILDHOST := $(SUBARCH)
 #CROSS_COMPILE	?=
 ARCH		?= arm
 #CROSS_COMPILE	:=/opt/android0320/mydroid/prebuilt/linux-x86/toolchain/arm-eabi-4.2.1/bin/arm-eabi-
-CROSS_COMPILE	?=/root/toolchain/arm-eabi-4.4.0/bin/arm-eabi-
+CROSS_COMPILE	?=../toolchain/arm-eabi-4.4.0/bin/arm-eabi-
 
 # Architecture as present in compile.h
 UTS_MACHINE 	:= $(ARCH)
-- 
2.35.3

