From f6f149454378841a5f55579ab7ba1137c6b1b4bf Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Sat, 29 Jun 2013 17:07:08 +0800
Subject: [PATCH] rk3188: do nothing with usb clks when system deep suspend

---
 arch/arm/mach-rk30/pm.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 6ff5a64d901e..3e3a381e8052 100755
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -514,13 +514,13 @@ static void __sramfunc rk_pm_soc_sram_volt_resume(void)
 }
 
 #define CLK_GATE_W_MSK0 (0xffff)
-#define CLK_GATE_W_MSK1 (0xffff)
+#define CLK_GATE_W_MSK1 (0xff9f)	//defult:(0xffff); ignore usb:(0xff9f) G1_[6:5]
 #define CLK_GATE_W_MSK2 (0xffff)
-#define CLK_GATE_W_MSK3 (0xff9f)
+#define CLK_GATE_W_MSK3 (0xff9f)	//defult:(0xff9f); ignore use:(0xff9f) G3_[6]
 #define CLK_GATE_W_MSK4 (0xffff)
-#define CLK_GATE_W_MSK5 (0xffff)
+#define CLK_GATE_W_MSK5 (0xdfff)	//defult:(0xffff); ignore usb:(0xdfff) G5_[13]
 #define CLK_GATE_W_MSK6 (0xffff)
-#define CLK_GATE_W_MSK7 (0xffff)
+#define CLK_GATE_W_MSK7 (0xffe7)	//defult:(0xffff); ignore usb:(0xffe7) G7_[4:3]
 #define CLK_GATE_W_MSK8 (0x01ff)
 #define CLK_GATE_W_MSK9 (0x07ff)
 static u32 __sramdata clkgt_regs_sram[CRU_CLKGATES_CON_CNT];
-- 
2.35.3

