From ab7a54fe9d8c91127cc39cc1afafd03215370a29 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Tue, 11 Sep 2012 09:34:07 +0800
Subject: [PATCH] rk610 lcd:add scale pll config

---
 drivers/video/display/transmitter/rk610_lcd.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/display/transmitter/rk610_lcd.h b/drivers/video/display/transmitter/rk610_lcd.h
index 8b46e07fedd7..f62d79160ffd 100644
--- a/drivers/video/display/transmitter/rk610_lcd.h
+++ b/drivers/video/display/transmitter/rk610_lcd.h
@@ -192,6 +192,7 @@ enum{
     SCALE_PLL(148500000,    30375000,   18, 11, 8),
     SCALE_PLL(148500000,    29700000,   16, 10, 8),
     SCALE_PLL(148500000,    25312500,   15, 11, 8),
+    SCALE_PLL(148500000,    74250000,   12, 6, 4),
 
     SCALE_PLL(74250000,     66000000,   32, 9,  4),
     SCALE_PLL(74250000,     57375000,   34, 11, 4),
@@ -200,6 +201,7 @@ enum{
     SCALE_PLL(74250000,     30375000,   36, 11, 8),
     SCALE_PLL(74250000,     25312500,   30, 11, 8),
     SCALE_PLL(74250000,     74250000,   12, 3, 4),
+    SCALE_PLL(74250000,    67500000,   40, 11, 4),
 
     SCALE_PLL(27000000,     75000000,   100, 9,  4),
     SCALE_PLL(27000000,     72000000,   32, 3,  4),
@@ -207,6 +209,7 @@ enum{
     SCALE_PLL(27000000,     54375000,   145, 9,  8),
     SCALE_PLL(27000000,     31500000,   28, 3,  8),
     SCALE_PLL(27000000,     30000000,   80, 9,  8),
+    SCALE_PLL(27000000,     70312500,   125, 6,  8)
 };
 
 enum {
-- 
2.35.3

