From 8e0c251378526491ac392026c493678f1d2dadbd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Fri, 9 Nov 2012 18:29:40 +0800
Subject: [PATCH] rk:modify tps65910 i2c write speed from 100k to 200k

---
 drivers/mfd/tps65910.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/tps65910.c b/drivers/mfd/tps65910.c
index de3d2ace237c..af4e442dcfb4 100755
--- a/drivers/mfd/tps65910.c
+++ b/drivers/mfd/tps65910.c
@@ -96,7 +96,7 @@ static int tps65910_i2c_write(struct tps65910 *tps65910, u8 reg,
 	//for(i=0;i<bytes;i++)
 	//printk("%s:reg=0x%x,value=0x%x\n",__func__,reg+i,msg[i+1]);
 	
-	ret = i2c_master_send(i2c, msg, bytes + 1);
+	ret = i2c_master_normal_send(i2c, msg, bytes + 1,TPS65910_SPEED);
 	if (ret < 0)
 		return ret;
 	if (ret != bytes + 1)
-- 
2.35.3

