From 42fb8708c5c5811ae6a2299f3f149c0e728d33d0 Mon Sep 17 00:00:00 2001
From: heyun <heyun@rock-chips.com>
Date: Thu, 23 Oct 2014 10:07:40 +0800
Subject: [PATCH] rk312x dts: disable ethernet

---
 arch/arm/boot/dts/rk3126-86v.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/rk3126-86v.dts b/arch/arm/boot/dts/rk3126-86v.dts
index d42823efb1b0..573355a02592 100644
--- a/arch/arm/boot/dts/rk3126-86v.dts
+++ b/arch/arm/boot/dts/rk3126-86v.dts
@@ -357,3 +357,15 @@
 &rk3126_cif_sensor{
 	status = "okay";
 };
+
+&gmac {
+//      power_ctl_by = "gpio";  //"gpio" "pmu"
+        power-gpio = <&gpio0 GPIO_A6 GPIO_ACTIVE_HIGH>;
+//      power-pmu = "act_ldo"
+//      reset-gpio = <&gpio4 GPIO_A7 GPIO_ACTIVE_LOW>;
+        phy-mode = "rgmii";
+        clock_in_out = "input";
+        tx_delay = <0x28>;
+        rx_delay = <0x10>;
+        status = "disabled";
+};
-- 
2.35.3

