From c8356f0cd530036f4ffb4f88a0e99a7dc7d0df47 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Tue, 25 Mar 2014 11:02:51 +0800
Subject: [PATCH] cw2015: decrese report battery change

---
 drivers/power/cw2015_battery.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/cw2015_battery.c b/drivers/power/cw2015_battery.c
index 5e13849b11a8..65e53266c140 100755
--- a/drivers/power/cw2015_battery.c
+++ b/drivers/power/cw2015_battery.c
@@ -705,7 +705,7 @@ static void rk_bat_update_vol(struct cw_battery *cw_bat)
         ret = cw_get_vol(cw_bat);
         if ((ret >= 0) && (cw_bat->voltage != ret)) {
                 cw_bat->voltage = ret;
-                cw_bat->bat_change = 1;
+             //   cw_bat->bat_change = 1;
         }
 }
 
@@ -736,7 +736,7 @@ static void rk_bat_update_time_to_empty(struct cw_battery *cw_bat)
         ret = cw_get_time_to_empty(cw_bat);
         if ((ret >= 0) && (cw_bat->time_to_empty != ret)) {
                 cw_bat->time_to_empty = ret;
-                cw_bat->bat_change = 1;
+              //  cw_bat->bat_change = 1;
         }
         
 }
-- 
2.35.3

