From 712887404e2703da3b3f83ff8804d20ab7763440 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Tue, 28 Aug 2012 10:41:56 +0800
Subject: [PATCH] rk2928: the bug of suspen , RK2928_GPIO1_BASE preread

---
 arch/arm/mach-rk2928/pm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/pm.c b/arch/arm/mach-rk2928/pm.c
index 13f32dc1b60b..0f2ebb327d7e 100644
--- a/arch/arm/mach-rk2928/pm.c
+++ b/arch/arm/mach-rk2928/pm.c
@@ -236,6 +236,7 @@ static noinline void interface_ctr_reg_pread(void)
 	readl_relaxed(RK2928_GRF_BASE);
 	readl_relaxed(RK2928_DDR_PCTL_BASE);
 	readl_relaxed(RK2928_DDR_PHY_BASE);
+	readl_relaxed(RK2928_GPIO1_BASE);
 	readl_relaxed(RK2928_GPIO3_BASE);
 //	readl_relaxed(RK2928_I2C1_BASE);
 }
-- 
2.35.3

