From cdeca016f0e40375e89f6e73dd5a6091e268af8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 22 May 2014 08:29:50 +0800
Subject: [PATCH] clocksource: rockchip_timer: rk_timer_read_current_value64
 return u64

---
 drivers/clocksource/rockchip_timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clocksource/rockchip_timer.c b/drivers/clocksource/rockchip_timer.c
index 1f74db6a8210..319ef5049f8b 100644
--- a/drivers/clocksource/rockchip_timer.c
+++ b/drivers/clocksource/rockchip_timer.c
@@ -81,7 +81,7 @@ static inline u32 rk_timer_read_current_value(void __iomem *base)
 	return readl_relaxed(base + TIMER_CURRENT_VALUE0);
 }
 
-static inline u32 rk_timer_read_current_value64(void __iomem *base)
+static inline u64 rk_timer_read_current_value64(void __iomem *base)
 {
 	u32 upper, lower;
 
-- 
2.35.3

