From 993a886e9c8a1f35ebd13f7175fbb03c1b7b9f5c Mon Sep 17 00:00:00 2001
From: "lw@rock-chips.com" <lw@rock-chips.com>
Date: Thu, 5 Apr 2012 16:21:59 +0800
Subject: [PATCH] rk30:close spim dma default

---
 drivers/spi/rk29_spim.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/rk29_spim.c b/drivers/spi/rk29_spim.c
index 606c53fd4c26..aceaa99e13e6 100755
--- a/drivers/spi/rk29_spim.c
+++ b/drivers/spi/rk29_spim.c
@@ -1654,7 +1654,7 @@ static int rk29xx_spi_setup(struct spi_device *spi)
 			return -ENOMEM;
 
 		chip->cs_control = spi_cs_control;
-		chip->enable_dma = 1;  //0;
+		chip->enable_dma = 0;  //0;
 	}
 
 	/*
-- 
2.35.3

