From 026195c1e35597d5540ee3448d8b6c2eca25a1ca Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Fri, 7 Feb 2014 13:42:17 +0800
Subject: [PATCH] close pinctrl debug log

---
 drivers/pinctrl/pinctrl-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index f15b23e9ca7f..34850f6d688f 100755
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -2194,7 +2194,7 @@ static int rockchip_pinctrl_probe(struct platform_device *pdev)
 	info->ctrl = ctrl;
 	info->dev = dev;
 	
-	atomic_set(&info->debug_flag, 1);
+	atomic_set(&info->debug_flag, 0);
 
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	info->reg_base = devm_ioremap_resource(&pdev->dev, res);
-- 
2.35.3

