From 9678b87aee2b22f0a4f33ab123325af356cf4456 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E6=99=BA?= <cz@rock-chips.com>
Date: Thu, 31 Mar 2011 09:25:02 +0800
Subject: [PATCH] BT: change macro CONFIG_BT_HCIBCM432x to CONFIG_BT_HCIBCM4325

---
 drivers/bluetooth/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bluetooth/Kconfig b/drivers/bluetooth/Kconfig
index c328aa12a71e..fd35fb05d8c7 100755
--- a/drivers/bluetooth/Kconfig
+++ b/drivers/bluetooth/Kconfig
@@ -206,7 +206,7 @@ config BT_HCIBCM4325
 	  Say Y here to compile support for HCI BCM4325 devices into the
 	  kernel or say M to compile it as module (bcm4325).
 
-if BT_HCIBCM432x
+if BT_HCIBCM4325
 choice
         prompt "BD_ADDR read from"
 config IDBLOCK
-- 
2.35.3

