From b39dd2b30c232bc3953022d3b985a5fadeb0b41a Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 21 Jul 2011 09:44:14 +0800
Subject: [PATCH] close hdmi debug info

---
 drivers/video/hdmi/hdmi-sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/hdmi/hdmi-sysfs.c b/drivers/video/hdmi/hdmi-sysfs.c
index a8013270a261..afe20d9da28e 100755
--- a/drivers/video/hdmi/hdmi-sysfs.c
+++ b/drivers/video/hdmi/hdmi-sysfs.c
@@ -2,7 +2,7 @@
 #include <linux/hdmi.h>
 #include <linux/string.h>
 
-int debug_en = 1;
+int debug_en = 0;
 static ssize_t hdmi_show_state_attrs(struct device *dev,
 					      struct device_attribute *attr,
 					      char *buf) 
-- 
2.35.3

