From 2cadf9992b51f578c1b75bc0df6c1e9c0747ed23 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 20 Mar 2015 10:53:50 +0800
Subject: [PATCH] rk fb: add support phy address config

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 9e624b08a2eb..e6adcbbaf202 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1495,7 +1495,8 @@ void rk_fb_free_dma_buf(struct rk_lcdc_driver *dev_drv,
 		index_buf = area_data->index_buf;
 #if defined(CONFIG_ROCKCHIP_IOMMU)
 		if (dev_drv->iommu_enabled) {
-			if (rk_fb->disp_policy != DISPLAY_POLICY_BOX)
+			if ((rk_fb->disp_policy != DISPLAY_POLICY_BOX) &&
+			     (area_data->ion_handle != NULL))
 				ion_unmap_iommu(dev_drv->dev, rk_fb->ion_client,
 						area_data->ion_handle);
 			freed_addr[freed_index++] = area_data->smem_start;
@@ -1906,6 +1907,7 @@ static int rk_fb_set_win_buffer(struct fb_info *info,
 		reg_win_data->reg_area_data[0].smem_start =
 		    win_par->area_par[0].phy_addr;
 		reg_win_data->area_num = 1;
+		reg_win_data->area_buf_num++;
 		fbi->screen_base = phys_to_virt(win_par->area_par[0].phy_addr);
 	}
 
-- 
2.35.3

