From b64d2bee8c94c8791e7d5c8043234474c136c062 Mon Sep 17 00:00:00 2001
From: zyc <zyc@rock-chips.com>
Date: Thu, 15 Nov 2012 16:37:32 +0800
Subject: [PATCH] camera: add focus zone ioctrl id.

---
 include/linux/videodev2.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/videodev2.h b/include/linux/videodev2.h
index a1a3296a720a..f8b5600bb88f 100644
--- a/include/linux/videodev2.h
+++ b/include/linux/videodev2.h
@@ -1011,6 +1011,7 @@ struct v4l2_ext_control {
 		__s64 value64;
 		char *string;
 	};
+	__s32 rect[4];
 } __attribute__ ((packed));
 
 struct v4l2_ext_controls {
@@ -1401,6 +1402,7 @@ enum  v4l2_exposure_auto_type {
 #define V4L2_CID_EFFECT				(V4L2_CID_CAMERA_CLASS_BASE_ROCK+2)
 #define V4L2_CID_FLASH				(V4L2_CID_CAMERA_CLASS_BASE_ROCK+3)
 #define V4L2_CID_FOCUS_CONTINUOUS		(V4L2_CID_CAMERA_CLASS_BASE_ROCK+4)
+#define V4L2_CID_FOCUSZONE       (V4L2_CID_CAMERA_CLASS_BASE_ROCK+5)
 
 /* FM Modulator class control IDs */
 #define V4L2_CID_FM_TX_CLASS_BASE		(V4L2_CTRL_CLASS_FM_TX | 0x900)
-- 
2.35.3

