From 433b7b6a39d08af4cccf096d51b0964aeba36d9e Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 5 May 2015 17:36:46 +0800
Subject: [PATCH] USB: gadget: rndis: fix compilation warning

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/usb/gadget/rndis.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/rndis.c b/drivers/usb/gadget/rndis.c
index cb2767df3fba..6a59290abb12 100644
--- a/drivers/usb/gadget/rndis.c
+++ b/drivers/usb/gadget/rndis.c
@@ -1050,7 +1050,7 @@ int rndis_rm_hdr(struct gether *port,
 		}
 
 		if (skb->len < sizeof *hdr) {
-			pr_err("invalid rndis pkt: skblen:%u hdr_len:%u",
+			pr_err("invalid rndis pkt: skblen:%u hdr_len:%zu",
 					skb->len, sizeof *hdr);
 			dev_kfree_skb_any(skb);
 			return -EINVAL;
-- 
2.35.3

