From 2a1d66500addc0658830db005eadb827bbaf0d71 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9E=97=E8=BE=89=E8=BE=89?= <lhh@rock-chips.com>
Date: Tue, 8 Jun 2010 07:10:05 +0000
Subject: [PATCH] pmem cache off

---
 arch/arm/mach-rk2818/devices.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2818/devices.c b/arch/arm/mach-rk2818/devices.c
index 981e6c564c42..d6436ae6c729 100644
--- a/arch/arm/mach-rk2818/devices.c
+++ b/arch/arm/mach-rk2818/devices.c
@@ -394,7 +394,7 @@ struct platform_device rk2818_device_dsp = {
 static struct android_pmem_platform_data pmem_pdata = {
 	.name = "pmem",
 	.no_allocator = 1,
-	.cached = 1,
+	.cached = 0,
 	.start = 0x6f000000,
 	.size =  0x1000000,
 };
-- 
2.35.3

