From 5adf4d05e63eeb146a8b53c562637586dd39838a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sun, 7 Oct 2012 18:17:33 +0800
Subject: [PATCH] rk30: scu enable speculative linefills

---
 arch/arm/mach-rk30/platsmp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/platsmp.c b/arch/arm/mach-rk30/platsmp.c
index 6c3ff7a051df..d44c80d4eeda 100755
--- a/arch/arm/mach-rk30/platsmp.c
+++ b/arch/arm/mach-rk30/platsmp.c
@@ -20,6 +20,7 @@
 
 #define SCU_CTRL             0x00
 #define   SCU_STANDBY_EN     (1 << 5)
+#define   SCU_SPECULATIVE_LINEFILLS_EN (1 << 3)
 
 #ifdef CONFIG_FIQ
 static void gic_raise_softirq_non_secure(const struct cpumask *mask, unsigned int irq)
@@ -130,7 +131,7 @@ void __init platform_smp_prepare_cpus(unsigned int max_cpus)
 		set_cpu_present(i, true);
 #endif
 
-	writel_relaxed(readl_relaxed(RK30_SCU_BASE + SCU_CTRL) | SCU_STANDBY_EN, RK30_SCU_BASE + SCU_CTRL);
+	writel_relaxed(readl_relaxed(RK30_SCU_BASE + SCU_CTRL) | SCU_STANDBY_EN | SCU_SPECULATIVE_LINEFILLS_EN, RK30_SCU_BASE + SCU_CTRL);
 
 	scu_enable(RK30_SCU_BASE);
 }
-- 
2.35.3

