From cab3775e0f6565184b459fa3f20f655f036968be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Thu, 11 Aug 2011 14:39:03 +0800
Subject: [PATCH] avoid initialize backlight twice

---
 drivers/video/backlight/rk29_backlight.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/backlight/rk29_backlight.c b/drivers/video/backlight/rk29_backlight.c
index 21d0d8782bf9..b7d664235b58 100755
--- a/drivers/video/backlight/rk29_backlight.c
+++ b/drivers/video/backlight/rk29_backlight.c
@@ -208,7 +208,7 @@ static int rk29_backlight_probe(struct platform_device *pdev)
 	clk_enable(pwm_clk);
 	write_pwm_reg(id, PWM_REG_CTRL, PWM_DIV|PWM_RESET);
 	write_pwm_reg(id, PWM_REG_LRC, div_total);
-	write_pwm_reg(id, PWM_REG_HRC, divh);
+	//write_pwm_reg(id, PWM_REG_HRC, divh);
 	write_pwm_reg(id, PWM_REG_CNTR, 0x0);
 	write_pwm_reg(id, PWM_REG_CTRL, PWM_DIV|PWM_ENABLE|PWM_TIME_EN);
 
-- 
2.35.3

