From ab09f9d7e431fccaac0b7b0e2361dd79eaf03902 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Tue, 12 Mar 2013 10:16:06 +0800
Subject: [PATCH] USB: support RK2928B for usb host and usb otg

---
 drivers/usb/dwc_otg/usbdev_rk30.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/dwc_otg/usbdev_rk30.c b/drivers/usb/dwc_otg/usbdev_rk30.c
index 9c4c20093e45..c00e790efb8d 100755
--- a/drivers/usb/dwc_otg/usbdev_rk30.c
+++ b/drivers/usb/dwc_otg/usbdev_rk30.c
@@ -30,6 +30,9 @@
 #elif defined(CONFIG_SOC_RK3168) || defined(CONFIG_SOC_RK3188) || defined(CONFIG_SOC_RK3168M)
 #define RK3066B_HOST_DRV_VBUS RK30_PIN0_PC0
 #define RK3066B_OTG_DRV_VBUS  RK30_PIN3_PD5
+#elif defined(CONFIG_SOC_RK2928B)
+#define RK3066B_HOST_DRV_VBUS RK30_PIN1_PA4
+#define RK3066B_OTG_DRV_VBUS  RK30_PIN3_PD7
 #endif
 
 #else
-- 
2.35.3

