From 4958c33109825ebec3f9b0368c9722225eb87075 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Thu, 17 Mar 2011 20:59:14 +0800
Subject: [PATCH] update fb for hdmi crash

---
 drivers/video/rk29_fb.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index bf8390e546a4..752c489e4eef 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -94,7 +94,7 @@ char cursor_img[] = {
 0x00,0x00,0x00,0x00,
 0x00,0x00,0x00,0x00,
 0x00,0x00,0x00,0x00,
-0x00,0x00,0x00,0x00 
+0x00,0x00,0x00,0x00
 };
 
 #if 0
@@ -591,7 +591,7 @@ void load_screen(struct fb_info *info, bool initscreen)
 	    right_margin = x_res/6;
 	}
 
-    printk("screen->hsync_len =%d,  screen->left_margin =%d, x_res =%d,  right_margin = %d \n",
+    fbprintk("screen->hsync_len =%d,  screen->left_margin =%d, x_res =%d,  right_margin = %d \n",
         screen->hsync_len , screen->left_margin , x_res , right_margin );
     LcdMskReg(inf, DSP_HTOTAL_HS_END, m_BIT12LO | m_BIT12HI, v_BIT12LO(screen->hsync_len) |
              v_BIT12HI(screen->hsync_len + screen->left_margin + x_res + right_margin));
@@ -682,20 +682,19 @@ void load_screen(struct fb_info *info, bool initscreen)
         printk(KERN_ERR ">>>>>> set lcdc dclk_divider faild \n ");
     }
 
-    clk_enable(inf->dclk);
-    clk_enable(inf->aclk);
-    clk_enable(inf->clk);
-
     clk_enable(inf->aclk_ddr_lcdc);
     clk_enable(inf->aclk_disp_matrix);
     clk_enable(inf->hclk_cpu_display);
 
+    clk_enable(inf->aclk);
+    clk_enable(inf->clk);
+    clk_enable(inf->dclk);
+
     // init screen panel
     if(screen->init)
     {
     	screen->init();
     }
-
 }
 #if 0 //def  CONFIG_CPU_FREQ
 /*
-- 
2.35.3

