From 98576991cb605114e47f8b063f8bfd60c256b418 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:30:58 +0800
Subject: [PATCH] wm8900: fix for v2.6.36

---
 sound/soc/codecs/wm8900.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/sound/soc/codecs/wm8900.c b/sound/soc/codecs/wm8900.c
index 94717434b4b9..10b84f9c7bb5 100755
--- a/sound/soc/codecs/wm8900.c
+++ b/sound/soc/codecs/wm8900.c
@@ -1004,6 +1004,7 @@ static int wm8900_resume(struct platform_device *pdev)
 
 	/* Restart the FLL? */
 	if (wm8900->fll_out) {
+		int ret;
 		int fll_out = wm8900->fll_out;
 		int fll_in  = wm8900->fll_in;
 
@@ -1173,7 +1174,6 @@ static int wm8900_probe(struct platform_device *pdev)
 
 	wm8900_workq = create_freezeable_workqueue("wm8900");
 	if (wm8900_workq == NULL) {
-		kfree(codec->private_data);
 		kfree(codec);
 		return -ENOMEM;
 	}
@@ -1184,9 +1184,6 @@ static int wm8900_probe(struct platform_device *pdev)
 
 	return ret;
 
-card_err:
-	snd_soc_free_pcms(socdev);
-	snd_soc_dapm_free(socdev);
 pcm_err:
 	return ret;
 }
-- 
2.35.3

