From 43e3beb4e90fd65abae47b1ff7f465adc7239848 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 29 Dec 2010 16:16:33 +0800
Subject: [PATCH] rk29: cpufreq adjust

---
 arch/arm/mach-rk29/cpufreq.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-rk29/cpufreq.c b/arch/arm/mach-rk29/cpufreq.c
index b1d437bdf4ff..fb8a0b410cf5 100755
--- a/arch/arm/mach-rk29/cpufreq.c
+++ b/arch/arm/mach-rk29/cpufreq.c
@@ -20,20 +20,13 @@
 #include <linux/regulator/consumer.h>
 
 static struct cpufreq_frequency_table freq_table[] = {
-#if 0
 //	{ .index =  950000, .frequency =  204000 },
 //	{ .index = 1050000, .frequency =  300000 },
 	{ .index = 1050000, .frequency =  408000 },
 //	{ .index = 1125000, .frequency =  600000 },
-	{ .index = 1225000, .frequency =  816000 },
-	{ .index = 1250000, .frequency = 1008000 },
+	{ .index = 1200000, .frequency =  816000 },
+//	{ .index = 1250000, .frequency = 1008000 },
 //	{ .index = 1300000, .frequency = 1200000 },
-#else
-	{ .index = 1100000, .frequency =  408000 },
-//	{ .index = 1200000, .frequency =  600000 },
-	{ .index = 1300000, .frequency =  816000 },
-//	{ .index = 1350000, .frequency = 1008000 },
-#endif
 	{ .frequency = CPUFREQ_TABLE_END },
 };
 static struct clk *arm_clk;
-- 
2.35.3

