From c8c21ee0a08d6edad0e1dd9a0a127f392762e1eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Fri, 21 Mar 2014 04:10:22 -0700
Subject: [PATCH] rk3288: change clk_vepu rate to 297M as default rate

---
 drivers/cpufreq/rockchip-cpufreq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/cpufreq/rockchip-cpufreq.c b/drivers/cpufreq/rockchip-cpufreq.c
index 89788efb91c8..0f1eac489881 100644
--- a/drivers/cpufreq/rockchip-cpufreq.c
+++ b/drivers/cpufreq/rockchip-cpufreq.c
@@ -383,6 +383,7 @@ static int cpufreq_init_cpu0(struct cpufreq_policy *policy)
 	if (clk_vepu_dvfs_node){
 		clk_enable_dvfs(clk_vepu_dvfs_node);
 		dvfs_clk_set_rate(clk_vepu_dvfs_node, 198000000);
+		dvfs_clk_set_rate(clk_vepu_dvfs_node, 297000000);
 	}
 
 	clk_cpu_dvfs_node = clk_get_dvfs_node("clk_core");
-- 
2.35.3

