From 247368a8b0056978c253bb909720975234db1323 Mon Sep 17 00:00:00 2001
From: xuhuicong <xhc@rock-chips.com>
Date: Fri, 12 Jul 2013 15:16:28 +0800
Subject: [PATCH] rk616 hdmi: default vif 12M rebuild

---
 drivers/mfd/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mfd/Kconfig b/drivers/mfd/Kconfig
index a8c429cb5b2d..cb39666a85b3 100644
--- a/drivers/mfd/Kconfig
+++ b/drivers/mfd/Kconfig
@@ -885,6 +885,8 @@ config MFD_RK616
 config RK616_USE_MCLK_12M	
 	bool "Enable 12M clock for timing reconstruction"
 	depends on MFD_RK616
+	select RK_HDMI_CTL_CODEC
+	default y
 	help
 	   if you say y here ,it will enable 12M clock for timing reconstruction
 
-- 
2.35.3

