From e0fb27faf512c2dcf7444dcf27775e14e81a4a03 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 May 2015 19:43:13 +0800
Subject: [PATCH] arm64: cpuinfo: compat task get hwcap from compat_hwcap_str

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/kernel/setup.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/setup.c b/arch/arm64/kernel/setup.c
index fedd56b36cf0..1ef1cd30793d 100644
--- a/arch/arm64/kernel/setup.c
+++ b/arch/arm64/kernel/setup.c
@@ -552,7 +552,8 @@ static int c_show(struct seq_file *m, void *v)
 		 * software which does already (at least for 32-bit).
 		 */
 		seq_puts(m, "Features\t:");
-		if (personality(current->personality) == PER_LINUX32) {
+		if (personality(current->personality) == PER_LINUX32 ||
+		    is_compat_task()) {
 #ifdef CONFIG_COMPAT
 			for (j = 0; compat_hwcap_str[j]; j++)
 				if (COMPAT_ELF_HWCAP & (1 << j))
-- 
2.35.3

