From bc25e32c153cd7d274b0755827db14392232ef2b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Fri, 30 Sep 2011 10:23:49 +0800
Subject: [PATCH] headset : headset interrupt to take the name

---
 drivers/headset_observe/rk_headset.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/headset_observe/rk_headset.c b/drivers/headset_observe/rk_headset.c
index 22f2222a451b..64a3c495508b 100755
--- a/drivers/headset_observe/rk_headset.c
+++ b/drivers/headset_observe/rk_headset.c
@@ -441,7 +441,7 @@ static int rockchip_headsetobserve_probe(struct platform_device *pdev)
 		headset->irq_type[HEADSET] = IRQF_TRIGGER_RISING;
 	else
 		headset->irq_type[HEADSET] = IRQF_TRIGGER_FALLING;
-	ret = request_irq(headset->irq[HEADSET], headset_interrupt, headset->irq_type[HEADSET], NULL, NULL);
+	ret = request_irq(headset->irq[HEADSET], headset_interrupt, headset->irq_type[HEADSET], "headset_input", NULL);
 	if (ret) 
 		goto failed_free;
 	enable_irq_wake(headset->irq[HEADSET]);
@@ -454,7 +454,7 @@ static int rockchip_headsetobserve_probe(struct platform_device *pdev)
 	headset->irq[HOOK] = gpio_to_irq(pdata->Hook_gpio);
 	headset->irq_type[HOOK] = IRQF_TRIGGER_FALLING;
 	
-	ret = request_irq(headset->irq[HOOK], Hook_interrupt, headset->irq_type[HOOK] , NULL, NULL);
+	ret = request_irq(headset->irq[HOOK], Hook_interrupt, headset->irq_type[HOOK] , "headset_hook", NULL);
 	if (ret) 
 		goto failed_free;
 	disable_irq(headset->irq[HOOK]);
-- 
2.35.3

