From 28b00ae9feb8305a90fb622bff8c71cb778ca35f Mon Sep 17 00:00:00 2001
From: Simon <xxm@rock-chips.com>
Date: Fri, 3 Apr 2015 15:31:08 +0800
Subject: [PATCH] rk3368: iommu: delete vop/hevc attach/detach log

Signed-off-by: Simon <xxm@rock-chips.com>
---
 drivers/iommu/rockchip-iommu.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 005e91c1271b..be089f10ecd8 100644
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -935,8 +935,9 @@ static void rockchip_iommu_detach_device(struct iommu_domain *domain, struct dev
 	}
 
 	if (rockchip_iommu_disable(data)) {
-		dev_dbg(dev->archdata.iommu,"%s: Detached IOMMU with pgtable %08lx\n",
-			__func__, (unsigned long)virt_to_phys(priv->pgtable));
+		if (!(strstr(data->dbgname, "vpu") || strstr(data->dbgname, "hevc")))
+			dev_dbg(dev->archdata.iommu,"%s: Detached IOMMU with pgtable %08lx\n",
+				__func__, (unsigned long)virt_to_phys(priv->pgtable));
 		data->domain = NULL;
 		list_del_init(&data->node);
 
@@ -974,8 +975,9 @@ static int rockchip_iommu_attach_device(struct iommu_domain *domain, struct devi
 		dev_dbg(dev->archdata.iommu,"%s: IOMMU with pgtable 0x%x already attached\n",
 			__func__, (unsigned int)virt_to_phys(priv->pgtable));
 	} else {
-		dev_info(dev->archdata.iommu,"%s: Attached new IOMMU with pgtable 0x%x\n",
-			__func__, (unsigned int)virt_to_phys(priv->pgtable));
+		if (!(strstr(data->dbgname, "vpu") || strstr(data->dbgname, "hevc")))
+			dev_info(dev->archdata.iommu,"%s: Attached new IOMMU with pgtable 0x%x\n",
+				__func__, (unsigned int)virt_to_phys(priv->pgtable));
 	}
 
 	return ret;
-- 
2.35.3

