From 815cfdcd4b0abf71ae13958223653715df009b68 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Thu, 6 Jan 2011 19:30:57 +0800
Subject: [PATCH] modified mma8452 for aigo board

---
 drivers/input/gsensor/mma8452.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/gsensor/mma8452.c b/drivers/input/gsensor/mma8452.c
index dcc834b5bd71..ec4838906e63 100644
--- a/drivers/input/gsensor/mma8452.c
+++ b/drivers/input/gsensor/mma8452.c
@@ -337,6 +337,9 @@ static int mma8452_get_data(struct i2c_client *client)
 		axis.y = -axis.y;
 		swap(axis.x,axis.y);		
 	}
+	#if defined(CONFIG_MACH_RK29_AIGO)
+	   axis.x = -axis.x;
+	#endif
 	
    // mmaprintk( "%s: ------------------mma8452_GetData axis = %d  %d  %d--------------\n",
     //       __func__, axis.x, axis.y, axis.z); 
-- 
2.35.3

