From e74543f6e1882870641a3805161af6ebbde81686 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=88=98=E7=9B=8A=E6=98=9F?= <lyx@rock-chips.com>
Date: Fri, 28 May 2010 08:58:14 +0000
Subject: [PATCH] modify net Kconfig

---
 drivers/net/Kconfig | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index 949a8348817d..41a3405c6fe3 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -947,6 +947,17 @@ config DM9000_DEBUGLEVEL
 	  The maximum level of debugging code compiled into the DM9000
 	  driver.
 
+if DM9000
+choice
+  prompt "choose control"
+	config DM9000_USE_NAND_CONTROL
+		tristate "DM9000 with NANDC Interface"
+
+	config DM9000_USE_NOR_CONTROL
+		tristate  "DM9000 with NOR Interface"
+endchoice
+endif
+	  
 config DM9000_FORCE_SIMPLE_PHY_POLL
 	bool "Force simple NSR based PHY polling"
 	depends on DM9000
-- 
2.35.3

