From 46f1476d2d9dcfe4054108a1fae935e3ddadabf8 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 2 Jun 2015 14:26:58 +0800
Subject: [PATCH] rk: gcc-wrapper.py fix psci ignore

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 scripts/gcc-wrapper.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/gcc-wrapper.py b/scripts/gcc-wrapper.py
index 147ae09af62b..73ced6707c60 100755
--- a/scripts/gcc-wrapper.py
+++ b/scripts/gcc-wrapper.py
@@ -49,8 +49,8 @@ allowed_warnings = set([
     "bus.c:318", # FIXME
     "atags_to_fdt.c:96",
     "compat_binfmt_elf.c:58",
-    "psci.c:299",
-    "psci.c:304",
+    "psci.c:268",
+    "psci.c:273",
     "sysctl_net_core.c:24",
  ])
 
-- 
2.35.3

