From 3bbeb2731b8490887366d463cb8d63b4753a5a55 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 15 Nov 2011 10:36:32 +0800
Subject: [PATCH] apanic: do not depend on android ram console

---
 drivers/misc/apanic.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/misc/apanic.c b/drivers/misc/apanic.c
index 4a9f48ac7480..58cc36e25ada 100644
--- a/drivers/misc/apanic.c
+++ b/drivers/misc/apanic.c
@@ -548,7 +548,9 @@ static int apanic(struct notifier_block *this, unsigned long event,
 	if (!threads_offset)
 		threads_offset = ctx->mtd->writesize;
 
+#ifdef CONFIG_ANDROID_RAM_CONSOLE
 	ram_console_enable_console(0);
+#endif
 
 	log_buf_clear();
 	show_state_filter(0);
-- 
2.35.3

