From cd6b90ed7f6f185ac3fd1b89fa5589ee294bb9df Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Fri, 24 Jun 2011 10:27:56 +0800
Subject: [PATCH] rk29:add rk29_dma_config(dws->rx_dmach, 1, 1) one more
 argument

---
 drivers/spi/rk29_spim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/rk29_spim.c b/drivers/spi/rk29_spim.c
index 08fd217d8311..4cd50405e527 100755
--- a/drivers/spi/rk29_spim.c
+++ b/drivers/spi/rk29_spim.c
@@ -1049,7 +1049,7 @@ static void dma_transfer(struct rk29xx_spi *dws) //int cs_change)
 			burst = 4;
 		}
 		if (rk29_dma_config(dws->tx_dmach, burst)) {*/
-		if (rk29_dma_config(dws->tx_dmach, 1)) {//there is not dma burst but bitwide, set it 1 alwayss
+		if (rk29_dma_config(dws->tx_dmach, 1, 1)) {//there is not dma burst but bitwide, set it 1 alwayss
 			dev_err(&dws->master->dev, "function: %s, line: %d\n", __FUNCTION__, __LINE__);
 			goto err_out;
 		}
@@ -1074,7 +1074,7 @@ static void dma_transfer(struct rk29xx_spi *dws) //int cs_change)
 
 	if (transfer->rx_buf != NULL) {
 		dws->state |= RXBUSY;
-		if (rk29_dma_config(dws->rx_dmach, 1)) {
+		if (rk29_dma_config(dws->rx_dmach, 1, 1)) {
 			dev_err(&dws->master->dev, "function: %s, line: %d\n", __FUNCTION__, __LINE__);
 			goto err_out;
 		}
-- 
2.35.3

