From 70ae3114c89ea6ce29d8e5349a038d99ee3c96d6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 9 Nov 2010 10:20:14 +0800
Subject: [PATCH] rk29: clock: add i2c/gpio clk

---
 arch/arm/mach-rk29/clock.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index bad616833a95..53c46c707452 100644
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -1126,6 +1126,19 @@ static struct clk clk_##NAME = { \
 	.gate_idx	= CLK_GATE_##ID, \
 }
 
+GATE_CLK(i2c0, pclk_cpu, I2C0);
+GATE_CLK(i2c1, pclk_periph, I2C1);
+GATE_CLK(i2c2, pclk_periph, I2C2);
+GATE_CLK(i2c3, pclk_periph, I2C3);
+
+GATE_CLK(gpio0, pclk_cpu, GPIO0);
+GATE_CLK(gpio1, pclk_periph, GPIO1);
+GATE_CLK(gpio2, pclk_periph, GPIO2);
+GATE_CLK(gpio3, pclk_periph, GPIO3);
+GATE_CLK(gpio4, pclk_cpu, GPIO4);
+GATE_CLK(gpio5, pclk_periph, GPIO5);
+GATE_CLK(gpio6, pclk_cpu, GPIO6);
+
 #define CLK(dev, con, ck) \
 	{ \
 		.dev_id = dev, \
@@ -1231,6 +1244,19 @@ static struct clk_lookup clks[] = {
 	CLK(NULL, "hclk_vdpu", &hclk_vdpu),
 	CLK1(gpu),
 	CLK(NULL, "aclk_gpu", &aclk_gpu),
+
+	CLK("rk29_i2c.0", "i2c", &clk_i2c0),
+	CLK("rk29_i2c.1", "i2c", &clk_i2c1),
+	CLK("rk29_i2c.2", "i2c", &clk_i2c2),
+	CLK("rk29_i2c.3", "i2c", &clk_i2c3),
+
+	CLK1(gpio0),
+	CLK1(gpio1),
+	CLK1(gpio2),
+	CLK1(gpio3),
+	CLK1(gpio4),
+	CLK1(gpio5),
+	CLK1(gpio6),
 };
 
 static LIST_HEAD(clocks);
-- 
2.35.3

