From ee62d49a9d8c4b40531281203c88ad725e63f59f Mon Sep 17 00:00:00 2001
From: gwl <gwl@rock-chips.com>
Date: Mon, 3 Mar 2014 09:27:24 +0800
Subject: [PATCH] wifi: modify rfkill_wlan_driver match table's name.

---
 net/rfkill/rfkill-wlan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/rfkill/rfkill-wlan.c b/net/rfkill/rfkill-wlan.c
index 2253ad87d7ff..8f2694b46804 100755
--- a/net/rfkill/rfkill-wlan.c
+++ b/net/rfkill/rfkill-wlan.c
@@ -620,7 +620,7 @@ static int rfkill_wlan_resume(struct platform_device *pdev)
 }
 
 #ifdef CONFIG_OF
-static struct of_device_id pwm_backlight_of_match[] = {
+static struct of_device_id wlan_platdata_of_match[] = {
     { .compatible = "wlan-platdata" },
     { }
 };
@@ -635,7 +635,7 @@ static struct platform_driver rfkill_wlan_driver = {
 	.driver = {
 		.name = "wlan-platdata",
 		.owner = THIS_MODULE,
-        .of_match_table = of_match_ptr(pwm_backlight_of_match),
+        .of_match_table = of_match_ptr(wlan_platdata_of_match),
 	},
 };
 
-- 
2.35.3

