From 4e31fe5041ef095e8fae0cee04be26f7fc13b588 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Wed, 15 Jul 2015 11:05:22 -0700
Subject: [PATCH] rk3288: clk: support aclk_vio 594M.

Signed-off-by: zhangqing <zhangqing@rock-chips.com>
---
 drivers/clk/rockchip/clk-ops.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/rockchip/clk-ops.c b/drivers/clk/rockchip/clk-ops.c
index 98957d323b54..b4d143c174bc 100644
--- a/drivers/clk/rockchip/clk-ops.c
+++ b/drivers/clk/rockchip/clk-ops.c
@@ -597,7 +597,7 @@ const struct clk_ops clkops_rate_3288_usb480m = {
 	.recalc_rate	= clk_3288_usb480m_recalc_rate,
 };
 
-#define RK3288_LIMIT_PLL_VIO0 (410*MHZ)
+#define RK3288_LIMIT_PLL_VIO0 (600*MHZ)
 
 static long clk_3288_dclk_lcdc0_determine_rate(struct clk_hw *hw, unsigned long rate,
 		unsigned long *best_parent_rate,
-- 
2.35.3

