From 517021c00de1a1027299dc4eaa10518c7aeb2905 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 31 May 2012 09:42:00 +0800
Subject: [PATCH] lightsensor: cm3217: unregister earlysuspend when shutdown

---
 drivers/input/lightsensor/cm3217.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/input/lightsensor/cm3217.c b/drivers/input/lightsensor/cm3217.c
index e50ed235de2c..36fea403c6dc 100755
--- a/drivers/input/lightsensor/cm3217.c
+++ b/drivers/input/lightsensor/cm3217.c
@@ -345,10 +345,17 @@ static const struct i2c_device_id cm3217_id[] = {
 	{ "lightsensor", 0 },
 };
 
+static void cm3217_shutdown(struct i2c_client *client)
+{
+#ifdef CONFIG_HAS_EARLYSUSPEND
+	unregister_early_suspend(&cm3217_early_suspend);
+#endif
+}
 
 static struct i2c_driver cm3217_driver = {
 	.probe = cm3217_probe,
 	.remove = __devexit_p(cm3217_remove),
+	.shutdown = cm3217_shutdown,
 	.id_table = cm3217_id,
 	.driver = {
 		.owner = THIS_MODULE,
-- 
2.35.3

