From 392faa8e4bf0877c8aabc206af38a6cf76584152 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Thu, 27 Jun 2013 19:16:19 +0800
Subject: [PATCH] vmac: reset phy to solve sometimes phy do not work when boot

---
 drivers/net/rk29_vmac.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/rk29_vmac.c b/drivers/net/rk29_vmac.c
index 7c445739d757..d32da064eaad 100755
--- a/drivers/net/rk29_vmac.c
+++ b/drivers/net/rk29_vmac.c
@@ -1070,8 +1070,11 @@ int vmac_open(struct net_device *dev)
 	clk_enable(clk_get(NULL,"mac_ref"));
 
 	//phy power on
-	if (pdata && pdata->rmii_power_control)
+	if (pdata && pdata->rmii_power_control) {
+        pdata->rmii_power_control(0);
+        msleep(100);
 		pdata->rmii_power_control(1);
+    }
 
 	msleep(1000);
 
-- 
2.35.3

