From 9a0212d3fe7552f0aea146ee34eb6095c15c3bc8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 10 Feb 2012 15:48:17 +0800
Subject: [PATCH] rk30: indent devices.c

---
 arch/arm/mach-rk30/devices.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index 9daef9b8e39b..f597d442b503 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -652,6 +652,7 @@ static struct platform_device device_nand = {
 	.num_resources	= ARRAY_SIZE(resources_nand),
 };
 #endif
+
 #ifdef CONFIG_KEYS_RK29
 extern struct rk29_keys_platform_data rk29_keys_pdata;
 static struct platform_device device_keys = {
@@ -662,6 +663,7 @@ static struct platform_device device_keys = {
 	},
 };
 #endif
+
 static int __init rk30_init_devices(void)
 {
 	rk30_init_dma();
-- 
2.35.3

