From b8f53ed1615b22fc2a1ed86853863628958c4731 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 10 Oct 2012 18:50:30 +0800
Subject: [PATCH] rk: config: ignore unknown parameter when insmod

---
 kernel/params.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/kernel/params.c b/kernel/params.c
index ce0b0a80bc69..c29d12381213 100644
--- a/kernel/params.c
+++ b/kernel/params.c
@@ -218,6 +218,11 @@ int parse_args(const char *name,
 }
 
 #ifdef CONFIG_RK_CONFIG
+static int ignore_unknown(char *param, char *val)
+{
+	return 0;
+}
+
 int module_parse_kernel_cmdline(const char *name, const struct kernel_param *params, unsigned num)
 {
 	int ret;
@@ -244,7 +249,7 @@ int module_parse_kernel_cmdline(const char *name, const struct kernel_param *par
 		new_params[i].name = new_name;
 	}
 
-	ret = parse_args(name, args, new_params, num, NULL);
+	ret = parse_args(name, args, new_params, num, ignore_unknown);
 
 out:
 	for (i = 0; i < num; i++)
-- 
2.35.3

