From db0c3f9ea7d311de9b89e22f55d3ff4342826c56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:21:46 +0800
Subject: [PATCH] i2c: rk29: fix for v2.6.36

---
 drivers/i2c/busses/i2c-rk29.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index 03715e2744c4..96bb9c93fa52 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -27,6 +27,7 @@
 #include <linux/delay.h>
 #include <linux/i2c.h>
 #include <linux/wakelock.h>
+#include <linux/slab.h>
 #include <mach/board.h>
 #include <asm/io.h>
 
-- 
2.35.3

