From f496df60b3f0bcc8d3f55cca5e6473ad59be1a9c Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 20 Aug 2010 11:09:19 +0800
Subject: [PATCH] modify: i2c bug about 'I2C_MODE_POLL'

---
 drivers/i2c/busses/i2c-rk2818.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rk2818.c b/drivers/i2c/busses/i2c-rk2818.c
index 54dacbb14f93..4cd29a2667c0 100755
--- a/drivers/i2c/busses/i2c-rk2818.c
+++ b/drivers/i2c/busses/i2c-rk2818.c
@@ -214,7 +214,7 @@ static int wait_for_completion_poll_timeout(struct rk2818_i2c_data *i2c, unsigne
 	unsigned int time = RK2818_DELAY_TIME;
 	int res;
 
-	while(!time_after(jiffies, timeout))
+	while(!time_after(jiffies, jiffies + timeout))
 	{
 		res = rk2818_event_occurred(i2c);
 		if(res)
-- 
2.35.3

