From 5ca10b99dfeffdc0fc71ced399466c5f33dc0fbc Mon Sep 17 00:00:00 2001
From: Jianhong Chen <chenjh@rock-chips.com>
Date: Sun, 26 Jul 2015 19:24:31 +0800
Subject: [PATCH] power: rk81x-battery: fix power on sec base error

Signed-off-by: Jianhong Chen <chenjh@rock-chips.com>
Signed-off-by: xushengfei <xsf@rock-chips.com>
---
 drivers/power/rk818_battery.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index 2cb49439d329..22d7c190d208 100644
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -135,6 +135,7 @@ module_param_named(dbg_level, dbg_enable, int, 0644);
 #define	TREND_STAT_UP		1
 #define	TREND_CAP_DIFF		5
 
+#define	POWER_ON_SEC_BASE	1
 #define MINUTE			60
 
 #define SLP_CURR_MAX		40
@@ -3909,7 +3910,7 @@ static void rk81x_bat_info_init(struct rk81x_battery *di,
 				struct rk818 *chip)
 {
 	u8 val;
-	unsigned long time_base = get_runtime_sec();
+	unsigned long time_base = POWER_ON_SEC_BASE;
 
 	rk81x_bat_read(di, RK818_VB_MON_REG, &val, 1);
 	if (val & PLUG_IN_STS)
-- 
2.35.3

