From 324e9548d7c06ca167b1f3b749f29f6cd9e7a695 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 11 Jul 2014 21:02:13 +0800
Subject: [PATCH] 3036: remove usb bypass func temporarily

---
 arch/arm/mach-rockchip/rk3036.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 72317f89e002..4ff1ecefda72 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -82,6 +82,7 @@ static void __init rk3036_boot_mode_init(void)
     rockchip_boot_mode_init(flag, mode);
 }
 
+#if 0
 static void usb_uart_init(void)
 {
 #ifdef CONFIG_RK_USB_UART
@@ -105,6 +106,7 @@ static void usb_uart_init(void)
     writel_relaxed(0x03, RK_DEBUG_UART_VIRT + 0x0c);
 #endif //end of DEBUG_UART_BASE
 }
+#endif
 
 static void __init rk3036_dt_map_io(void)
 {
@@ -112,7 +114,7 @@ static void __init rk3036_dt_map_io(void)
 
     iotable_init(rk3036_io_desc, ARRAY_SIZE(rk3036_io_desc));
     debug_ll_io_init();
-    usb_uart_init();
+    //usb_uart_init();
 
     /* enable timer5 for core */
     writel_relaxed(0, RK3036_TIMER5_VIRT + 0x10);
-- 
2.35.3

