From 3408c4921dd2a62dfaf247878d1e3988b5487435 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Mon, 15 Aug 2011 18:40:33 -0700
Subject: [PATCH] mpu3050: remove accel irq warning

---
 drivers/misc/mpu3050/mpu-dev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/mpu3050/mpu-dev.c b/drivers/misc/mpu3050/mpu-dev.c
index e84b63f97faa..34f6ba44ea31 100755
--- a/drivers/misc/mpu3050/mpu-dev.c
+++ b/drivers/misc/mpu3050/mpu-dev.c
@@ -1069,7 +1069,7 @@ int mpu3050_probe(struct i2c_client *client,
 				if (res)
 					goto out_accelirq_failed;
 			} else {
-				dev_WARN(&this_client->adapter->dev,
+				dev_warn(&this_client->adapter->dev,
 					"WARNING: Accel irq not assigned\n");
 			}
 		} else {
-- 
2.35.3

