From 782c10d4cc592e6c43c4d323c48b6b6e3bdb167a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 24 May 2013 16:33:29 +0800
Subject: [PATCH] rk808: fix compile error when build without earlysuspend
 support

---
 include/linux/mfd/rk808.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/linux/mfd/rk808.h b/include/linux/mfd/rk808.h
index aa5a3b87ca96..7ede72c9e456 100755
--- a/include/linux/mfd/rk808.h
+++ b/include/linux/mfd/rk808.h
@@ -15,9 +15,7 @@
 
 #include <linux/regulator/machine.h>
 #include <linux/wakelock.h>
-#ifdef CONFIG_HAS_EARLYSUSPEND
 #include <linux/earlysuspend.h>
-#endif
 
 //#define RK808_START 30
 
-- 
2.35.3

