From 9f7b31ec9f1cdd0cec6a6c676b935991aea7bce8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Mar 2011 18:31:23 +0800
Subject: [PATCH] rk29: clock: default armclk set to 300MHz, fix reboot problem

---
 arch/arm/mach-rk29/clock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 512cff763510..4032b4c0c2da 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -1702,7 +1702,7 @@ GATE_CLK(hclk_emmc, hclk_periph, HCLK_EMMC);
 static int pd_vcodec_mode(struct clk *clk, int on)
 {
 	if (on) {
-		u32 gate, gate2;
+		u32 gate;
 
 		gate = cru_clkgate3_con_mirror;
 		gate |= (1 << CLK_GATE_ACLK_DDR_VEPU % 32);
@@ -2295,7 +2295,7 @@ static int clk_register(struct clk *clk)
 	return 0;
 }
 
-static unsigned int __initdata armclk = 600 * MHZ;
+static unsigned int __initdata armclk = 300 * MHZ;
 
 /*
  * You can override arm_clk rate with armclk= cmdline option.
-- 
2.35.3

