From 5ac33d7c5c0f48d2cd21a3cdc469910146222d59 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E8=BE=89?= <chenhui@rock-chips.com>
Date: Thu, 15 Nov 2012 23:49:19 +0800
Subject: [PATCH] to support 4.2_r1

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index d2b9cb2841b8..b22d80e944c8 100644
--- a/Makefile
+++ b/Makefile
@@ -200,6 +200,9 @@ endif
 ifneq ($(wildcard ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.3),)
 CROSS_COMPILE	?= ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.3/bin/arm-eabi-
 endif
+ifneq ($(wildcard ../prebuilts/gcc/linux-x86/arm/arm-eabi-4.6),)
+CROSS_COMPILE   ?= ../prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
+endif
 CROSS_COMPILE	?= $(CONFIG_CROSS_COMPILE:"%"=%)
 
 # Architecture as present in compile.h
-- 
2.35.3

