From 7cd6c5c2116b1f786205637a04a8adf9181c7268 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Sat, 19 Oct 2013 14:16:32 +0800
Subject: [PATCH] fiq debugger: fix bug debug uart interrupt might be masked in
 IRQ exception

---
 kernel/irq/chip.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/kernel/irq/chip.c b/kernel/irq/chip.c
index 990965ec5d0b..b1c616e81b8e 100644
--- a/kernel/irq/chip.c
+++ b/kernel/irq/chip.c
@@ -409,7 +409,10 @@ handle_fasteoi_irq(unsigned int irq, struct irq_desc *desc)
 	 */
 	if (unlikely(!desc->action || irqd_irq_disabled(&desc->irq_data))) {
 		desc->istate |= IRQS_PENDING;
-		mask_irq(desc);
+#ifdef CONFIG_FIQ_DEBUGGER
+		if(irq != (CONFIG_RK_DEBUG_UART + 66))
+#endif
+			mask_irq(desc);
 		goto out;
 	}
 
-- 
2.35.3

