From db270573b263d0338821fdb71b46bfccfbf29d6d Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Thu, 21 May 2015 15:10:25 +0800
Subject: [PATCH] dtsi: ddr_freq: add vop bandwidth freq table

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index 7950af8912b4..eb2a3e526189 100644
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -1072,6 +1072,15 @@
 						300000 1200000
 						400000 1200000
 						>;
+					bd-freq-table = <
+						/* bandwidth   freq */
+						2700           792000
+						2600           600000
+						2280           456000
+						1560           396000
+						1020           324000
+						720            240000
+						>;
 					channel = <2>;
 					status = "disabled";
 				};
-- 
2.35.3

