From d109ea3728f8f21166a5a4ad75926481bbb99aab Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 29 Nov 2012 19:34:23 +0800
Subject: [PATCH] rkfb:fix rk_fb_check_var err

---
 drivers/video/rockchip/rk_fb.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 6ef5bd13cf40..533a57e29256 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -374,12 +374,12 @@ static int rk_fb_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 		 return -EINVAL;
 	 }
  
-	 if( ((var->xoffset+var->xres) > info->var.xres_virtual) ||
-	     ((var->yoffset+var->yres) > (info->var.yres_virtual)) )
+	 if( ((var->xoffset+var->xres) > var->xres_virtual) ||
+	     ((var->yoffset+var->yres) > (var->yres_virtual)) )
 	 {
 		 printk("%s check_var fail 2!!! \n",info->fix.id);
-		 printk("xoffset:%d>>xres:%d>>xres_vir:%d\n",var->xoffset,var->xres,info->var.xres_virtual);
-		 printk("yoffset:%d>>yres:%d>>yres_vir:%d\n",var->yoffset,var->yres,info->var.yres_virtual);
+		 printk("xoffset:%d>>xres:%d>>xres_vir:%d\n",var->xoffset,var->xres,var->xres_virtual);
+		 printk("yoffset:%d>>yres:%d>>yres_vir:%d\n",var->yoffset,var->yres,var->yres_virtual);
 		 return -EINVAL;
 	 }
 
-- 
2.35.3

