From 2ed910d1edad7b8e40da9799e4eb9d6b8cfbc29a Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 24 Jan 2013 11:04:28 +0800
Subject: [PATCH] rk_fb.h:add struct rk_fb_vysnc

---
 include/linux/rk_fb.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/linux/rk_fb.h b/include/linux/rk_fb.h
index eeeff2f54bdc..6f19857ca97f 100644
--- a/include/linux/rk_fb.h
+++ b/include/linux/rk_fb.h
@@ -162,6 +162,15 @@ struct rk_fb_rgb {
 	struct fb_bitfield	transp;
 };
 
+struct rk_fb_vsync {
+	wait_queue_head_t	wait;
+	ktime_t			timestamp;
+	bool			active;
+	int			irq_refcount;
+	struct mutex		irq_lock;
+	struct task_struct	*thread;
+};
+
 typedef enum _TRSP_MODE
 {
     TRSP_CLOSE = 0,
-- 
2.35.3

