From c02c7deb9586175aa51c6157fa73720f42c6bcd0 Mon Sep 17 00:00:00 2001
From: Sun Mingjun <smj@rock-chips.com>
Date: Wed, 20 Aug 2014 09:48:14 +0800
Subject: [PATCH] codec : modified the codec debug level

---
 sound/soc/codecs/rk312x_codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rk312x_codec.c b/sound/soc/codecs/rk312x_codec.c
index 1cd363bfa939..8f59f29029dc 100755
--- a/sound/soc/codecs/rk312x_codec.c
+++ b/sound/soc/codecs/rk312x_codec.c
@@ -41,7 +41,7 @@
 #include <sound/tlv.h>
 #include "rk312x_codec.h"
 
-static int debug = 7;
+static int debug = 0;
 module_param(debug, int, S_IRUGO|S_IWUSR);
 
 #define dbg_codec(level, fmt, arg...)		\
-- 
2.35.3

