From 5f05b2fda5ca6e2962b19d8e227d3d6b06e5222e Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 30 Nov 2012 12:12:19 +0800
Subject: [PATCH] rkfb_sysfs:modify print style

---
 drivers/video/rockchip/rkfb_sysfs.c | 33 +++++++++++++++++++++++------
 1 file changed, 26 insertions(+), 7 deletions(-)

diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index be9b6f72f1b9..67ba6c9f8db7 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -234,13 +234,32 @@ static ssize_t set_fb_win_map(struct device *dev,struct device_attribute *attr,
 	   (order != FB0_WIN2_FB1_WIN0_FB2_WIN1) && (order != FB0_WIN0_FB1_WIN2_FB2_WIN1 ) &&
 	   (order != FB0_WIN0_FB1_WIN1_FB2_WIN2) && (order != FB0_WIN1_FB1_WIN0_FB2_WIN2 ))
 	{
-		printk(KERN_ERR "un support map\nyou can use the following order: \
-			\n201:\nfb0-win1\nfb1-win0\nfb2-win2\n			   \
-			\n210:\nfb0-win0\nfb1-win1\nfb2-win2\n			  \
-			\n120:\nfb0-win0\nfb1-win2\nfb2-win1\n			  \
-			\n102:\nfb0-win2\nfb1-win0\nfb2-win1\n			   \
-			\n021:\nfb0-win1\nfb1-win2\nfb2-win0\n			   \
-			\n012:\nfb0-win2\nfb1-win1\nfb2-win0\n");
+		printk(KERN_ERR "un supported map\n"
+				"you can use the following order:\n"
+			        "201:\n"
+			        "fb0-win1\n"
+			        "fb1-win0\n"
+			        "fb2-win2\n"
+			        "210:\n"
+			        "fb0-win0\n"
+			        "fb1-win1\n"
+			        "fb2-win2\n"			  
+				"120:\n"
+				"fb0-win0\n"
+				"fb1-win2\n"
+				"fb2-win1\n"			  
+				"102:\n"
+				"fb0-win2\n"
+				"fb1-win0\n"
+				"fb2-win1\n"			   
+				"021:\n"
+				"fb0-win1\n"
+				"fb1-win2\n"
+				"fb2-win0\n"			   
+				"012:\n"
+				"fb0-win2\n"
+				"fb1-win1\n"
+				"fb2-win0\n");
 		return count;
 	}
 	else
-- 
2.35.3

