From 83878559d0c6d34af31c779c79ecee76852cc139 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Wed, 3 Aug 2011 01:19:50 -0700
Subject: [PATCH] compass ak8975: add suspend and resume return value

---
 drivers/input/magnetometer/ak8975.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/input/magnetometer/ak8975.c b/drivers/input/magnetometer/ak8975.c
index 5b424add2b31..9380c9ac26d6 100755
--- a/drivers/input/magnetometer/ak8975.c
+++ b/drivers/input/magnetometer/ak8975.c
@@ -701,7 +701,7 @@ static void akm8975_early_resume(struct early_suspend *handler)
 	AKMDBG("resumed with flag=%d", 
 	       atomic_read(&reserve_open_flag));
 }
-static void akm8975_suspend(struct i2c_client *client, pm_message_t mesg)
+static int akm8975_suspend(struct i2c_client *client, pm_message_t mesg)
 {
 	AKMFUNC("akm8975_early_suspend");
 	atomic_set(&suspend_flag, 1);
@@ -711,9 +711,10 @@ static void akm8975_suspend(struct i2c_client *client, pm_message_t mesg)
 	disable_irq(this_client->irq);
 	AKMDBG("suspended with flag=%d", 
 	       atomic_read(&reserve_open_flag));
+	return 0;
 }
 
-static void akm8975_resume(struct i2c_client *client)
+static int akm8975_resume(struct i2c_client *client)
 {
 	AKMFUNC("akm8975_early_resume");
 	enable_irq(this_client->irq);
@@ -722,6 +723,7 @@ static void akm8975_resume(struct i2c_client *client)
 	wake_up(&open_wq);
 	AKMDBG("resumed with flag=%d", 
 	       atomic_read(&reserve_open_flag));
+	return 0;
 }
 /*********************************************/
 static struct file_operations akmd_fops = {
-- 
2.35.3

