From 74b63344fd8c7cd5a2fef85a28ccc8fd8bca8365 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Sat, 12 Jul 2014 14:57:51 +0800
Subject: [PATCH] 3036: fix hym gpio judgement

---
 drivers/rtc/rtc-HYM8563.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-HYM8563.c b/drivers/rtc/rtc-HYM8563.c
index d30da29f39ee..c20f4b867bfe 100755
--- a/drivers/rtc/rtc-HYM8563.c
+++ b/drivers/rtc/rtc-HYM8563.c
@@ -592,7 +592,7 @@ static int  hym8563_probe(struct i2c_client *client, const struct i2c_device_id
 	}	
 	
 	client->irq = of_get_named_gpio_flags(np, "irq_gpio", 0,(enum of_gpio_flags *)&irq_flags);
-        if(client->irq < 0)
+	if(client->irq >= 0)
         {
 	        hym8563->irq = gpio_to_irq(client->irq);
 	        result = devm_request_threaded_irq(&client->dev, hym8563->irq, NULL, hym8563_wakeup_irq, irq_flags | IRQF_ONESHOT, client->dev.driver->name,hym8563 );
-- 
2.35.3

