From 216db9ff010466023c66e040051c7ad6092886b7 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Wed, 17 Oct 2012 16:59:45 +0800
Subject: [PATCH] rk_fb: one lcdc dual screen mode,if plug hdmi,not blank lcdc.

---
 drivers/video/rockchip/rk_fb.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 5bfe1540011c..0a2507c2f618 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -304,10 +304,17 @@ static int rk_fb_blank(int blank_mode, struct fb_info *info)
 	{
 		return  -ENODEV;
 	}
-	
-    	dev_drv->blank(dev_drv,layer_id,blank_mode);
-
-    	return 0;
+#if defined(CONFIG_RK_HDMI)
+#if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)
+	if(hdmi_get_hotplug() == HDMI_HPD_ACTIVED){
+		printk("hdmi is connect , not blank lcdc\n");
+	}else
+#endif
+#endif
+	{
+		dev_drv->blank(dev_drv,layer_id,blank_mode);
+	}
+	return 0;
 }
 
 static int rk_fb_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
-- 
2.35.3

