From 46152e11da1813df2d79880b5eae1730edbb081b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 12 Apr 2012 17:05:44 +0800
Subject: [PATCH] rk: mmu: default cachepolicy set to writeback

---
 arch/arm/mm/mmu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mm/mmu.c b/arch/arm/mm/mmu.c
index 2b28c080a529..85ecb83bea0c 100644
--- a/arch/arm/mm/mmu.c
+++ b/arch/arm/mm/mmu.c
@@ -312,8 +312,10 @@ static void __init build_mem_type_table(void)
 			cachepolicy = CPOLICY_WRITEBACK;
 		ecc_mask = 0;
 	}
+#ifndef CONFIG_PLAT_RK
 	if (is_smp())
 		cachepolicy = CPOLICY_WRITEALLOC;
+#endif
 
 	/*
 	 * Strip out features not present on earlier architectures.
-- 
2.35.3

