From aa9086b62af51526467d14e7adbfa54b5054989e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Mon, 24 Mar 2014 18:15:14 +0800
Subject: [PATCH] rk3288:pwm_regultor:close pwm_regulator dbg

---
 drivers/regulator/rockchip-pwm-regulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/rockchip-pwm-regulator.c b/drivers/regulator/rockchip-pwm-regulator.c
index ad72c3b7916c..3e5d1acf2a4f 100755
--- a/drivers/regulator/rockchip-pwm-regulator.c
+++ b/drivers/regulator/rockchip-pwm-regulator.c
@@ -35,7 +35,7 @@
 #include <linux/regulator/machine.h>
 
 
-#if 1
+#if 0
 #define DBG(x...)	printk(KERN_INFO x)
 #else
 #define DBG(x...)
-- 
2.35.3

