From b797d2a1bf6b1b742b7f55c18b6d4c02eb099bf4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=B5=B5=E5=AD=90=E5=88=9D?= <zzc@rock-chips.com>
Date: Thu, 8 Aug 2013 11:30:02 +0800
Subject: [PATCH] update drivers/misc/bp/chips/u7501.c

---
 drivers/misc/bp/chips/u7501.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/bp/chips/u7501.c b/drivers/misc/bp/chips/u7501.c
index 87fd8b96a11a..2e0e5c21b93c 100644
--- a/drivers/misc/bp/chips/u7501.c
+++ b/drivers/misc/bp/chips/u7501.c
@@ -203,7 +203,7 @@ struct bp_operate bp_U7501_ops = {
 	.bp_wakeup_ap		= RK2928_PIN3_PC3,	//
 	.bp_uart_en		= BP_UNKNOW_DATA, 	//EINT9
 	.bp_usb_en		= BP_UNKNOW_DATA, 	//W_disable
-	.trig			= IRQF_TRIGGER_RISING,
+	.trig			= IRQF_TRIGGER_FALLING,//IRQF_TRIGGER_RISING,
 
 	.active			= bp_active,
 	.init			= bp_init,
@@ -232,7 +232,7 @@ struct bp_operate bp_U7501_ops = {
 	.bp_wakeup_ap		= RK30_PIN0_PC5,	//
 	.bp_uart_en		= BP_UNKNOW_DATA, 	//EINT9
 	.bp_usb_en		= BP_UNKNOW_DATA, 	//W_disable
-	.trig			= IRQF_TRIGGER_RISING,
+	.trig			= IRQF_TRIGGER_FALLING,//IRQF_TRIGGER_RISING,
 
 	.active			= bp_active,
 	.init			= bp_init,
@@ -261,7 +261,7 @@ struct bp_operate bp_U7501_ops = {
 	.bp_wakeup_ap		= BP_UNKNOW_DATA,//RK2928_PIN3_PC3,	//
 	.bp_uart_en		= BP_UNKNOW_DATA, 	//EINT9
 	.bp_usb_en		= BP_UNKNOW_DATA, 	//W_disable
-	.trig			= IRQF_TRIGGER_RISING,
+	.trig			= IRQF_TRIGGER_FALLING,//IRQF_TRIGGER_RISING,
 
 	.active			= bp_active,
 	.init			= bp_init,
-- 
2.35.3

