From 199e4a8fe6ceb626663fd6d165c5c576c5718d90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 14 Sep 2011 13:13:26 +0800
Subject: [PATCH] rk29: enable DDR frequency scaling by default

---
 arch/arm/mach-rk29/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk29/Kconfig b/arch/arm/mach-rk29/Kconfig
index 414c875a8a24..a738886e2c6b 100644
--- a/arch/arm/mach-rk29/Kconfig
+++ b/arch/arm/mach-rk29/Kconfig
@@ -164,7 +164,8 @@ config DDR_SDRAM_FREQ
 	default 400
 
 config DDR_FREQ
-	bool "Enable DDR frequency scaling (EXPERIMENTAL)"
+	bool "Enable DDR frequency scaling"
+	default y
 
 config DDR_RECONFIG
 	bool "Enable dynamic DDR reconfiguration (EXPERIMENTAL)"
-- 
2.35.3

