From 9cadf3891e4a0d3fa004dd475ee1e55127cbeeb0 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Tue, 27 Mar 2012 15:37:05 +0800
Subject: [PATCH] rk30: fix bug fiq debugger doesn't work when irq disable

---
 arch/arm/mach-rk30/fiq.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/fiq.c b/arch/arm/mach-rk30/fiq.c
index 2c8d176f9781..4883d4b7a6bb 100644
--- a/arch/arm/mach-rk30/fiq.c
+++ b/arch/arm/mach-rk30/fiq.c
@@ -101,8 +101,9 @@ void rk30_fiq_init(void)
 		writel_relaxed(0xffffffff, RK30_GICD_BASE + GIC_DIST_SECURITY + (i<<2));
 	}
 	dsb();
-
 	writel_relaxed(0x3, RK30_GICD_BASE + GIC_DIST_CTRL);
-	writel_relaxed(0xf, RK30_GICC_BASE + GIC_CPU_CTRL);
+	writel_relaxed(0x0f, RK30_GICC_BASE + GIC_CPU_CTRL);
+	//set the uart 2(the debug port) priority a little higher than other interrupts
+	writel_relaxed(0xa0a0a090, RK30_GICD_BASE + GIC_DIST_PRI + (IRQ_UART2/4)*4);
 	dsb();
 }
-- 
2.35.3

