From dbff91698c8472c98ec783a9cd9b86740237498a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Thu, 11 Aug 2011 18:16:29 +0800
Subject: [PATCH] newton:fix gt819 drop point bug

---
 drivers/input/touchscreen/gt819.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/gt819.c b/drivers/input/touchscreen/gt819.c
index 8c0188b00827..91baacda5a4d 100755
--- a/drivers/input/touchscreen/gt819.c
+++ b/drivers/input/touchscreen/gt819.c
@@ -471,8 +471,12 @@ static void gt819_queue_work(struct work_struct *work)
 		check_sum += point_data[count];
 	if(check_sum  != 0)			//checksum verify error
 		{
-			printk("coor checksum error!\n");
+			input_report_abs(ts->input_dev, ABS_MT_TOUCH_MAJOR, 0);
+			input_report_abs(ts->input_dev, ABS_MT_WIDTH_MAJOR, 0);
+			//input_mt_sync(data->input_dev);
+			input_sync(ts->input_dev);
 			enable_irq(ts->irq);
+			dev_info(&ts->client->dev, "coor checksum error!touch release\n");
 			return;
 		}
 		
-- 
2.35.3

