From d9cbf1941d1c6940a9b635efd59e2f1a28e05544 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Thu, 21 May 2015 11:56:16 -0700
Subject: [PATCH] rk3368: pmic: rk818:adjust dcdc voltage step

Signed-off-by: zhangqing <zhangqing@rock-chips.com>
---
 drivers/mfd/rk818.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/mfd/rk818.c b/drivers/mfd/rk818.c
index 71e4575fd828..c72a04a555c2 100755
--- a/drivers/mfd/rk818.c
+++ b/drivers/mfd/rk818.c
@@ -119,7 +119,7 @@ const static int ldo_set_slp_vol_base_addr[] = {
 #define rk818_LDO_SET_SLP_VOL_REG(x) (ldo_set_slp_vol_base_addr[x])
 
 const static int buck_voltage_map[] = {
-	  700,  712,  725,  737,  750, 762,  775,  787,  800, 
+	  712,  725,  737,  750, 762,  775,  787,  800,
 	  812,  825,  837,  850,862,  875,  887,  900,  912,
 	  925,  937,  950, 962,  975,  987, 1000, 1012, 1025, 
 	  1037, 1050,1062, 1075, 1087, 1100, 1112, 1125, 1137, 
@@ -383,7 +383,7 @@ static int rk818_dcdc_list_voltage(struct regulator_dev *dev, unsigned selector)
 	switch (buck) {
 	case 0:
 	case 1:
-		volt = 700000 + selector * 12500;
+		volt = 712500 + selector * 12500;
 		break;
 	case 3:
 		volt = 1800000 + selector * 100000;
@@ -446,10 +446,10 @@ static int rk818_dcdc_select_min_voltage(struct regulator_dev *dev,
 	u16 vsel =0;
 	
 	if (buck == 0 || buck ==  1){
-		if (min_uV < 700000)
+		if (min_uV < 712500)
 		vsel = 0;
 		else if (min_uV <= 1500000)
-		vsel = ((min_uV - 700000) / 12500) ;
+		vsel = ((min_uV - 712500) / 12500) ;
 		else
 		return -EINVAL;
 	}
-- 
2.35.3

