From 4ae0521064d4475beba864d0a21e37ace27dc14e Mon Sep 17 00:00:00 2001
From: dkl <dkl@rock-chips.com>
Date: Tue, 12 Aug 2014 11:57:29 +0800
Subject: [PATCH] rk818: disable otg and boost when in sleep mode

---
 drivers/mfd/rk818.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mfd/rk818.c b/drivers/mfd/rk818.c
index 591ff2f9016e..19f4e9df6f11 100755
--- a/drivers/mfd/rk818.c
+++ b/drivers/mfd/rk818.c
@@ -1146,6 +1146,16 @@ static int rk818_pre_init(struct rk818 *rk818)
 	/**********************************/
 	ret = rk818_clear_bits(rk818, RK818_INT_STS_MSK_REG1,(0x3<<5)); //open rtc int when power on
  	ret = rk818_set_bits(rk818, RK818_RTC_INT_REG,(0x1<<3),(0x1<<3)); //open rtc int when power on
+
+	/*****disable otg and boost when in sleep mode****/
+	val = rk818_reg_read(rk818, RK818_SLEEP_SET_OFF_REG1);
+	val |= ((0x1 << 7) | (0x1 << 4));
+	ret =  rk818_reg_write(rk818, RK818_SLEEP_SET_OFF_REG1, val);
+	if (ret < 0) {
+		pr_err("Unable to write RK818_SLEEP_SET_OFF_REG1 reg\n");
+		return ret;
+	}
+
 	return 0;
 }
  
-- 
2.35.3

