From 0f80424a2ab9b022facbf1d4422e72154352c397 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 1 Sep 2014 12:07:01 +0800
Subject: [PATCH] mmc: update rto for cmd to CMD_FUNC

---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 57fe7ab93862..0238bb94e86f 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -2071,7 +2071,7 @@ static void dw_mci_command_complete(struct dw_mci *host, struct mmc_command *cmd
                 del_timer_sync(&host->dto_timer);
 	    if(MMC_SEND_STATUS != cmd->opcode)
 	        if(host->cmd_rto >= SDMMC_CMD_RTO_MAX_HOLD){
-	                MMC_DBG_ERR_FUNC(host->mmc, " command complete, cmd=%d,cmdError=%d [%s]",\
+	                MMC_DBG_CMD_FUNC(host->mmc, " command complete, cmd=%d,cmdError=%d [%s]",\
                                 cmd->opcode, cmd->error,mmc_hostname(host->mmc));
                         host->cmd_rto = 0;
     	        }
-- 
2.35.3

