From 4fee3c868843bea2d0fe2d5e1ac8c30d15c8b4a2 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 11 Jul 2014 21:18:24 +0800
Subject: [PATCH] 3036: add ion at dtsi

---
 arch/arm/boot/dts/rk3036.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/rk3036.dtsi b/arch/arm/boot/dts/rk3036.dtsi
index fc1f7cf10725..ab609658bd3b 100755
--- a/arch/arm/boot/dts/rk3036.dtsi
+++ b/arch/arm/boot/dts/rk3036.dtsi
@@ -539,6 +539,21 @@
 		reg = <0x10118200 0x100>;
 		status = "disabled";
 	};
+
+	ion {
+		compatible = "rockchip,ion";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ion_cma: rockchip,ion-heap@1 { /* CMA HEAP */
+			compatible = "rockchip,ion-reserve";
+			rockchip,ion_heap = <1>;
+			reg = <0x00000000 0x10000000>; /* 256MB */
+		};
+		rockchip,ion-heap@3 { /* VMALLOC HEAP */
+			rockchip,ion_heap = <3>;
+		};
+	};
 	
         vpu: vpu_service@10108000 {
 		compatible = "vpu_service";
-- 
2.35.3

