From 3ee07d81aaa372d30f1dc38ac7b9658b952dc2c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Aug 2012 19:27:54 +0800
Subject: [PATCH] rk30: refactor wdt init code

---
 arch/arm/mach-rk30/devices.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index fddd9a09ea4f..b528a9498b9a 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -1304,12 +1304,7 @@ static struct platform_device device_vmac = {
 };
 #endif
 
-/*
- * rk29 wdt device  ADDED BY HHB@ROCK-CHIPS.COM
- */
-
 #ifdef CONFIG_RK29_WATCHDOG
-
 static struct resource resources_wdt[] = {
 	{
 		.start	= IRQ_WDT,
@@ -1323,13 +1318,12 @@ static struct resource resources_wdt[] = {
 	},
 };
 
-struct platform_device rk29_device_wdt = {
+static struct platform_device device_wdt = {
 	.name	= "rk29-wdt",
 	.id	= 0,
 	.num_resources	= ARRAY_SIZE(resources_wdt),
 	.resource	= resources_wdt,
 };
-
 #endif
 
 static int __init rk30_init_devices(void)
@@ -1378,7 +1372,7 @@ static int __init rk30_init_devices(void)
 	platform_device_register(&device_vmac);
 #endif
 #ifdef CONFIG_RK29_WATCHDOG
-	platform_device_register(&rk29_device_wdt);
+	platform_device_register(&device_wdt);
 #endif
 	return 0;
 }
-- 
2.35.3

