From 23993e84e0766114f83d31e6175897dd2b722671 Mon Sep 17 00:00:00 2001
From: "lw@rock-chips.com" <lw@rock-chips.com>
Date: Fri, 27 Apr 2012 10:50:13 +0800
Subject: [PATCH] rk30:async init touch screen

---
 drivers/input/touchscreen/rk29_i2c_goodix.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/rk29_i2c_goodix.c b/drivers/input/touchscreen/rk29_i2c_goodix.c
index 6bd9bde71d7b..0dfa571d55d1 100755
--- a/drivers/input/touchscreen/rk29_i2c_goodix.c
+++ b/drivers/input/touchscreen/rk29_i2c_goodix.c
@@ -33,6 +33,7 @@
 #include <linux/syscalls.h>
 #include <linux/reboot.h>
 #include <linux/proc_fs.h>
+#include <linux/async.h>
 #include "rk29_i2c_goodix.h"
 
 #include <linux/vmalloc.h>
@@ -1664,12 +1665,17 @@ static int dbg_sys_init(void)
 return:
 	Executive Outcomes. 0---succeed.
 ********************************************************/
-static int __devinit rk_ts_init(void)
+
+static void __init rk_ts_init_async(void *unused, async_cookie_t cookie)
 {
-	int ret ;
-	ret=i2c_add_driver(&rk_ts_driver);
+	i2c_add_driver(&rk_ts_driver);
 	dbg_sys_init();  //for debug
-	return ret; 
+}
+
+static int __init rk_ts_init(void)
+{
+	async_schedule(rk_ts_init_async, NULL);
+	return 0;
 }
 
 /*******************************************************	
-- 
2.35.3

