From a873cdc7b5d30785e941ee55ba72673b96d3d438 Mon Sep 17 00:00:00 2001
From: lxt <lxt@rock-chips.com>
Date: Tue, 2 Sep 2014 11:02:40 +0800
Subject: [PATCH] rk312x : close i2s0 which not used

---
 arch/arm/boot/dts/rk312x.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index cb94027dca9a..f75c674f031a 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -483,6 +483,7 @@
 		//pinctrl-names = "default", "sleep";
 		//pinctrl-0 = <&i2s0_mclk &i2s0_sclk &i2s0_lrckrx &i2s0_lrcktx &i2s0_sdi &i2s0_sdo>;
 		//pinctrl-1 = <&i2s0_gpio>;
+		status = "disabled";
 	};
 
 	i2s1: i2s@10200000 {
-- 
2.35.3

