From f1c44b7322a0e3069230c8c1051ad90b20dc68e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 31 Dec 2010 15:55:13 +0800
Subject: [PATCH] rk29_fb: rm setup DMC AXI host priority

---
 drivers/video/rk29_fb.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index de9416318a5f..4153fd8e411c 100644
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1941,23 +1941,6 @@ static int __init rk29fb_probe (struct platform_device *pdev)
 
     fbprintk(">>>>>> %s : %s\n", __FILE__, __FUNCTION__);
 
-    /* Fix DMC AXI host priority. FIXME: Move to bootloader in the future? */
-    __raw_writel(0x7, RK29_DDRC_BASE+0x200);
-    __raw_writel(0x7, RK29_DDRC_BASE+0x204);
-    __raw_writel(0x7, RK29_DDRC_BASE+0x208);
-    __raw_writel(0x7, RK29_DDRC_BASE+0x20c);
-    __raw_writel(0x7, RK29_DDRC_BASE+0x210);
-    __raw_writel(0x10007eff, RK29_DDRC_BASE+0x280);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x284);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x288);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x28c);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x290);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x294);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x298);
-    __raw_writel(0x1e007eff, RK29_DDRC_BASE+0x29c);
-    __raw_writel(0x2, RK29_DDRC_BASE+0x2a0);
-    *(volatile u32 *)(RK29_GRF_BASE+0xac) = ((*(volatile u32 *)(RK29_GRF_BASE+0xac)) & ~0x3FF) | 0x246;
-
     /* Malloc rk29fb_inf and set it to pdev for drvdata */
     fbprintk(">> Malloc rk29fb_inf and set it to pdev for drvdata \n");
     inf = kmalloc(sizeof(struct rk29fb_inf), GFP_KERNEL);
-- 
2.35.3

