From 9e9f60e0bccc74c2a9b7a4c674032efab77ca7b2 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Wed, 29 Jun 2011 09:57:29 +0800
Subject: [PATCH] disable cursor when init

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 8322db410557..b3bfb945d5cf 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -805,7 +805,7 @@ int rk29_set_cursor_img(struct rk29fb_inf *inf, char *data)
     }
     LcdWrReg(inf, HWC_MST, __pa(rk29_cursor_buf));
     //LcdSetBit(inf, SYS_CONFIG,m_HWC_RELOAD_EN);
-    LcdSetBit(inf, SYS_CONFIG, m_HWC_RELOAD_EN |m_HWC_ENABLE);
+    LcdSetBit(inf, SYS_CONFIG, m_HWC_RELOAD_EN);
     flush_cache_all();
     LcdWrReg(inf, REG_CFG_DONE, 0x01);
     return 0;
-- 
2.35.3

