From 301057966e89be905bdf6689c4b20a929b353689 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 14 Aug 2014 11:31:22 +0800
Subject: [PATCH] rk312x lcdc: add alpha config

---
 drivers/video/rockchip/lcdc/rk312x_lcdc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/lcdc/rk312x_lcdc.c b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
index d016f1e05569..401de7869511 100755
--- a/drivers/video/rockchip/lcdc/rk312x_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
@@ -363,6 +363,7 @@ static void lcdc_layer_update_regs(struct lcdc_device *lcdc_dev,
 		else if (win->id == 2)
 			lcdc_msk_reg(lcdc_dev, SYS_CTRL, m_HWC_EN, v_HWC_EN(0));
 	}
+	rk312x_lcdc_alpha_cfg(lcdc_dev);
 }
 
 static void lcdc_layer_enable(struct lcdc_device *lcdc_dev, unsigned int win_id,
-- 
2.35.3

