From 6b2c0a765f65ac9b038d907da5849ab2b47010a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 10 Dec 2013 14:21:38 +0800
Subject: [PATCH] rk: Makefile set SUBLEVEL to 0 and add empty .scmversion

---
 .scmversion | 0
 Makefile    | 2 ++
 2 files changed, 2 insertions(+)
 create mode 100644 .scmversion

diff --git a/.scmversion b/.scmversion
new file mode 100644
index 000000000000..e69de29bb2d1
diff --git a/Makefile b/Makefile
index b46dca925f4f..d22b9f2da984 100644
--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,8 @@ NAME = TOSSUG Baby Fish
 # Comments in this file are targeted only to the developer, do not
 # expect to learn how to build the kernel reading this file.
 
+SUBLEVEL = 0
+
 # Do not:
 # o  use make's built-in rules and variables
 #    (this increases performance and avoids hard-to-debug behaviour);
-- 
2.35.3

