From b5e0c04a4b23cbad97899b83d5b768d7742dfa51 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Wed, 1 Jun 2011 18:47:09 -0700
Subject: [PATCH] a22 backlight: fix bug----bl shine when suspend

---
 drivers/video/backlight/aw9364_bl.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/drivers/video/backlight/aw9364_bl.c b/drivers/video/backlight/aw9364_bl.c
index 3884a16b2d24..35b05142020e 100755
--- a/drivers/video/backlight/aw9364_bl.c
+++ b/drivers/video/backlight/aw9364_bl.c
@@ -81,18 +81,18 @@ static int aw9364_backlight_set(struct backlight_device *bl, int brightness)
 		gpio_direction_output(data->pin_en, GPIO_LOW);
 		mdelay(3);
 	}
-	
-	for(i=0; i<num_clk; i++)
-	{
-		gpio_direction_output(data->pin_en, GPIO_LOW);
-		udelay(5);	
-		gpio_direction_output(data->pin_en, GPIO_HIGH);
-		if(i==0)
-		udelay(50);
-		else 
-		udelay(2);		
+	else {
+		for(i=0; i<num_clk; i++)
+		{
+			gpio_direction_output(data->pin_en, GPIO_LOW);
+			udelay(5);	
+			gpio_direction_output(data->pin_en, GPIO_HIGH);
+			if(i==0)
+			udelay(50);
+			else 
+			udelay(2);		
+		}
 	}
-			
 	DBG("%s:current_bl=%d,bl=%d,num_clk_to=%d,num_clk_from=%d,num_clk=%d\n",__FUNCTION__,
 		data->current_brightness,brightness,num_clk_to,num_clk_from,num_clk);
 
-- 
2.35.3

