From b2e6b2e1b0ea9ba31f3ca3cf5e1532813f7d4aea Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Thu, 17 Apr 2014 17:46:05 +0800
Subject: [PATCH] gpu midgard : avoid hard reset isp mmu when initiate

---
 drivers/iommu/rockchip-iommu.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 4d96f1e61a6e..2ad9ca152155 100755
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -1044,11 +1044,14 @@ static int rockchip_sysmmu_probe(struct platform_device *pdev)
 			ret = -ENOENT;
 			goto err_res;
 		}
-		/*reset sysmmu*/
-		if(!sysmmu_reset(data->res_bases[i],data->dbgname))
+		if(!strstr(data->dbgname,"isp"))
 		{
-			ret = -ENOENT;
-			goto err_res;
+			/*reset sysmmu*/
+			if(!sysmmu_reset(data->res_bases[i],data->dbgname))
+			{
+				ret = -ENOENT;
+				goto err_res;
+			}
 		}
 	}
 
-- 
2.35.3

