From 2cdc1deac0d12e83d2d75a15b441fc47a0a41493 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Thu, 12 Mar 2015 12:21:54 +0800
Subject: [PATCH] battery: enable dc charge
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 许盛飞 <xsf@rock-chips.com>
---
 drivers/power/rk818_battery.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index 1aa7c5a645a3..ff4016326999 100644
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -1505,7 +1505,7 @@ static void set_charge_current(struct battery_info *di, int charge_current)
 
 	battery_read(di->rk818, USB_CTRL_REG, &usb_ctrl_reg, 1);
 	usb_ctrl_reg &= (~0x0f);/* (VLIM_4400MV | ILIM_1200MA) |(0x01 << 7); */
-	usb_ctrl_reg |= (charge_current | CHRG_CT_EN);
+	usb_ctrl_reg |= (charge_current);
 	battery_write(di->rk818, USB_CTRL_REG, &usb_ctrl_reg, 1);
 }
 
@@ -1524,9 +1524,9 @@ static void rk_battery_charger_init(struct  battery_info *di)
 	DBG("old usb_ctrl_reg = 0x%2x, CHRG_CTRL_REG1 = 0x%2x\n ", usb_ctrl_reg, chrg_ctrl_reg1);
 	usb_ctrl_reg &= (~0x0f);
 #ifdef SUPPORT_USB_CHARGE
-	usb_ctrl_reg |= (ILIM_450MA | CHRG_CT_EN);
+	usb_ctrl_reg |= (ILIM_450MA);
 #else
-	usb_ctrl_reg |= (ILIM_3000MA | CHRG_CT_EN);
+	usb_ctrl_reg |= (ILIM_3000MA);
 #endif
 	chrg_ctrl_reg1 &= (0x00);
 	chrg_ctrl_reg1 |= (CHRG_EN) | (CHRG_VOL4200 | CHRG_CUR1400mA);
@@ -2286,7 +2286,7 @@ static int  get_charging_status_type(struct battery_info *di)
 
 	if (0 == otg_status) {
 		di->usb_online = 0;
-		di->ac_online = 0;
+		di->ac_online = 1;
 		di->check_count = 0;
 
 	} else if (1 == otg_status) {
@@ -2446,8 +2446,6 @@ static void update_battery_info(struct battery_info *di)
 	di->remain_capacity = _get_realtime_capacity(di);
 	if (di->remain_capacity > di->fcc)
 		_capacity_init(di, di->fcc);
-	else if (di->remain_capacity < 0)
-		_capacity_init(di, 0);
 
 	if (di->real_soc > 100)
 		di->real_soc = 100;
-- 
2.35.3

