From 271f5aedb18cf6d093f64a8a6f6560b487060bff Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Fri, 27 Mar 2015 18:57:38 +0800
Subject: [PATCH] rk3368 lcdc: fix deadlock when close lcdc and update reg

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index 3007d42cc978..1a524ffd088f 100755
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -1392,7 +1392,6 @@ static int rk3368_lcdc_layer_update_regs(struct lcdc_device *lcdc_dev,
 	int timeout;
 	unsigned long flags;
 
-	spin_lock(&lcdc_dev->reg_lock);
 	if (likely(lcdc_dev->clk_on)) {
 		lcdc_msk_reg(lcdc_dev, SYS_CTRL, m_STANDBY_EN,
 			     v_STANDBY_EN(lcdc_dev->standby));
@@ -1405,7 +1404,6 @@ static int rk3368_lcdc_layer_update_regs(struct lcdc_device *lcdc_dev,
 		/*rk3368_lcdc_post_cfg(dev_drv); */
 		lcdc_cfg_done(lcdc_dev);
 	}
-	spin_unlock(&lcdc_dev->reg_lock);
 
 	/*if (dev_drv->wait_fs) { */
 	if (0) {
-- 
2.35.3

