From e46fa33f60ef54b8bbb28b95cfe9d482c5e4390e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 11 Apr 2011 14:50:51 +0800
Subject: [PATCH] rk29: clock: mac clock from general pll

---
 arch/arm/mach-rk29/clock.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index ed908ef1914e..5fd83f1fda91 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2476,7 +2476,6 @@ static void __init rk29_clock_common_init(unsigned long ppll_rate)
 		clk_set_parent_nolock(&clk_gpu, &codec_pll_clk);
 		/* arm pll */
 		clk_set_rate_nolock(&arm_pll_clk, armclk);
-		clk_set_parent_nolock(&clk_mac_ref_div, &arm_pll_clk);
 		/*you can choose clk parent form codec pll or periph pll for following logic*/
 		clk_set_parent_nolock(&aclk_vepu, aclk_vepu_parent);
 		clk_set_rate_nolock(&aclk_vepu, aclk_vepu_rate);
-- 
2.35.3

