From 60913765680100392c6626856c59e6d89b9bf772 Mon Sep 17 00:00:00 2001
From: yangyun <yangyun@rock-chips.com>
Date: Thu, 23 Feb 2012 11:36:09 +0800
Subject: [PATCH] add ums by xzj

---
 drivers/usb/gadget/android.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 6b40a29fe9cc..1c3eb953cfb6 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -538,8 +538,9 @@ static int mass_storage_function_init(struct android_usb_function *f,
 	if (!config)
 		return -ENOMEM;
 
-	config->fsg.nluns = 1;
+	config->fsg.nluns = 2;
 	config->fsg.luns[0].removable = 1;
+	config->fsg.luns[1].removable = 1;
 
 	common = fsg_common_init(NULL, cdev, &config->fsg);
 	if (IS_ERR(common)) {
@@ -549,7 +550,8 @@ static int mass_storage_function_init(struct android_usb_function *f,
 
 	err = sysfs_create_link(&f->dev->kobj,
 				&common->luns[0].dev.kobj,
-				"lun");
+				"lun0");
+        err = sysfs_create_link(&f->dev->kobj, &common->luns[1].dev.kobj,"lun1");
 	if (err) {
 		kfree(config);
 		return err;
-- 
2.35.3

