From f4047b849226bafea8df72cde3609891f0b9528b Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Mon, 11 Apr 2011 01:14:26 -0700
Subject: [PATCH] backlight: reconfig brightness

---
 drivers/video/backlight/wm831x_bl.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/video/backlight/wm831x_bl.c b/drivers/video/backlight/wm831x_bl.c
index 643d07f913ef..d0c1d6f38461 100755
--- a/drivers/video/backlight/wm831x_bl.c
+++ b/drivers/video/backlight/wm831x_bl.c
@@ -24,9 +24,7 @@
 #include <linux/delay.h>
 #include <linux/ktime.h>
 #define BL_SET   255
-#define BL_DIFF_VALUE 100
 #define BL_INIT_VALUE 102
-#define BL_MIN_VALUE 0
 struct wm831x_backlight_data {
 	struct wm831x *wm831x;
 	int isink_reg;
@@ -119,13 +117,11 @@ static int wm831x_backlight_set(struct backlight_device *bl, int brightness)
 static int wm831x_backlight_update_status(struct backlight_device *bl)
 {
 	int brightness = bl->props.brightness;
-	if ((brightness<=BL_INIT_VALUE) && (brightness>0)) {
-		brightness = brightness + BL_DIFF_VALUE;
+	if (brightness<=BL_INIT_VALUE) {
+		brightness = 111*brightness/51;
 	}
-	else if (brightness > BL_INIT_VALUE) {
-		brightness = 
-			((BL_SET-BL_INIT_VALUE-BL_DIFF_VALUE)*brightness+BL_SET*BL_DIFF_VALUE)
-				/(BL_SET-BL_INIT_VALUE);
+	else {
+		brightness = 33*brightness/153 + 200;
 	}
 
 	if(gwm831x_data->suspend_flag == 1)
-- 
2.35.3

