From 029f5a92bc2527a9454f843e88401bba962f0d7e Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Fri, 20 Oct 2017 19:57:09 +0800
Subject: [PATCH] mfd: rk808: add rk816-battery child cell of rk816s

Change-Id: I6a6a1c05be947f9084f80e7357b32b1708b27c57
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 drivers/mfd/rk808.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index 69cd488b0002..3bfcd9ef16e8 100644
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -288,6 +288,7 @@ static const struct mfd_cell rk816s[] = {
 	{ .name = "rk808-clkout", },
 	{ .name = "rk808-regulator", },
 	{ .name = "rk8xx-gpio", },
+	{ .name = "rk816-battery", .of_compatible = "rk816-battery", },
 	{
 		.name = "rk8xx-pwrkey",
 		.num_resources = ARRAY_SIZE(rk816_pwrkey_resources),
-- 
2.35.3

