From 0947d63edff01c372754766ab34ecce40b323ee0 Mon Sep 17 00:00:00 2001
From: cym <cym@rock-chips.com>
Date: Fri, 16 Aug 2013 09:35:04 +0800
Subject: [PATCH] RK30/RK31:remove "__sramlocalfunc" for ddr_get_bw().

---
 arch/arm/mach-rk30/ddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/ddr.c b/arch/arm/mach-rk30/ddr.c
index f327f0d1e517..c03d15043b7a 100755
--- a/arch/arm/mach-rk30/ddr.c
+++ b/arch/arm/mach-rk30/ddr.c
@@ -1152,7 +1152,7 @@ uint32 ddr_get_col(void)
     return col;
 }
 
-__sramlocalfunc uint32 ddr_get_bw(void)
+uint32 ddr_get_bw(void)
 {
     uint32 bw;
 
-- 
2.35.3

