From 051d437e3215cb3e1a892212be5a3e5262313f78 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Fri, 27 Mar 2015 19:42:02 +0800
Subject: [PATCH] clk: rockchip: dump more cru register when panic

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/clk/rockchip/clk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/rockchip/clk.c b/drivers/clk/rockchip/clk.c
index 233defda7c80..dcc80e542a4b 100755
--- a/drivers/clk/rockchip/clk.c
+++ b/drivers/clk/rockchip/clk.c
@@ -1569,8 +1569,8 @@ static void rkclk_cache_parents(struct rkclk *rkclk)
 void rk_dump_cru(void)
 {
 	printk(KERN_WARNING "CRU:\n");
-	print_hex_dump(KERN_WARNING, "", DUMP_PREFIX_OFFSET, 16, 4, rk_cru_base,
-		       0x220, false);
+	print_hex_dump(KERN_WARNING, "", DUMP_PREFIX_OFFSET, 32, 4, rk_cru_base,
+		       0x420, false);
 }
 EXPORT_SYMBOL_GPL(rk_dump_cru);
 
-- 
2.35.3

