From 703ab7b30a3a784d78e4164289231e5d6e7d25eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 18 Sep 2012 16:37:51 +0800
Subject: [PATCH] rk30: cpufreq: fix gcc error when build without earlysuspend

---
 arch/arm/mach-rk30/cpufreq.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index 27ae5450786d..c5d44d3324ac 100644
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -510,9 +510,11 @@ static void ff_early_resume_func(struct early_suspend *h)
 static int __init ff_init(void)
 {
 	FF_DEBUG("enter %s\n", __func__);
+#ifdef CONFIG_HAS_EARLYSUSPEND
 	ff_early_suspend.suspend = ff_early_suspend_func;
 	ff_early_suspend.resume = ff_early_resume_func;
 	ff_early_suspend.level = EARLY_SUSPEND_LEVEL_DISABLE_FB + 100;
+#endif
 	register_early_suspend(&ff_early_suspend);
 	return 0;
 }
-- 
2.35.3

