From 3a354c544c3a0ee71ee910d490f51b7d96b4b564 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 28 May 2015 10:13:44 +0800
Subject: [PATCH] rk fb: act_height should be 2 pix align for interlace output

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index cba2cb3133a4..ce3a90bce7ba 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -2185,6 +2185,11 @@ static int rk_fb_set_win_buffer(struct fb_info *info,
 		reg_win_data->reg_area_data[i].xact = win_par->area_par[i].xact;
 		reg_win_data->reg_area_data[i].yact = win_par->area_par[i].yact;
 
+		/*act_height should be 2 pix align for interlace output*/
+		if ((screen->mode.vmode == FB_VMODE_INTERLACED) &&
+		    (reg_win_data->reg_area_data[i].yact % 2 == 1))
+			reg_win_data->reg_area_data[i].yact -= 1;
+
 		xoffset = win_par->area_par[i].x_offset;	/* buf offset */
 		yoffset = win_par->area_par[i].y_offset;
 		reg_win_data->reg_area_data[i].xoff = xoffset;
-- 
2.35.3

