From 1900714c6021621f4a0de5a9d31cdf4fad8540fd Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 9 Jul 2013 09:40:02 +0800
Subject: [PATCH] rk3188: plus: add adjust dvfs table support

---
 arch/arm/mach-rk3188/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk3188/cpufreq.c b/arch/arm/mach-rk3188/cpufreq.c
index ba6c2ae85a3c..c8c59423fac0 100644
--- a/arch/arm/mach-rk3188/cpufreq.c
+++ b/arch/arm/mach-rk3188/cpufreq.c
@@ -425,7 +425,7 @@ static int rk3188_cpufreq_init_cpu0(struct cpufreq_policy *policy)
 		return PTR_ERR(cpu_clk);
 
 #if defined(CONFIG_ARCH_RK3188)
-	if (soc_is_rk3188()) {
+	if (soc_is_rk3188() || soc_is_rk3188plus()) {
 		struct cpufreq_frequency_table *table_adjust;
 		/* Adjust dvfs table avoid overheat */
 		table_adjust = dvfs_get_freq_volt_table(cpu_clk);
-- 
2.35.3

