From d495b998e28dd0ccc119f566cba1c347c2ae66ae Mon Sep 17 00:00:00 2001
From: guoyi <gy@rock-chips.com>
Date: Mon, 30 Mar 2015 17:49:16 +0800
Subject: [PATCH] rk3368-p9 : dts: enable dsihost0

---
 arch/arm64/boot/dts/rk3368-p9_818.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368-p9_818.dts b/arch/arm64/boot/dts/rk3368-p9_818.dts
index 4c9548222b4a..9ce3eef90ceb 100755
--- a/arch/arm64/boot/dts/rk3368-p9_818.dts
+++ b/arch/arm64/boot/dts/rk3368-p9_818.dts
@@ -237,6 +237,10 @@
 	status = "okay";
 };
 
+&dsihost0{
+	status = "okay";
+};
+
 &spi0 {
 	status = "disabled";
 };
-- 
2.35.3

