From 25ec3f1f2838a092c60561b4b01ad42e680d60ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 30 Mar 2012 20:57:17 +0800
Subject: [PATCH] i2c: rk30: fix idle lock name alloced on stack

---
 drivers/i2c/busses/i2c-rk30.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-rk30.c b/drivers/i2c/busses/i2c-rk30.c
index aa817e2c9a0a..63dd362af514 100755
--- a/drivers/i2c/busses/i2c-rk30.c
+++ b/drivers/i2c/busses/i2c-rk30.c
@@ -86,7 +86,6 @@ static int rk30_i2c_probe(struct platform_device *pdev)
 	struct rk30_i2c *i2c = NULL;
 	struct rk30_i2c_platform_data *pdata = NULL;
 	struct resource *res;
-        char name[5];
 	int ret;
 
 	pdata = pdev->dev.platform_data;
@@ -201,10 +200,9 @@ static int rk30_i2c_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, i2c);
 
-        sprintf(name, "%s%d", "i2c", i2c->adap.nr);
         i2c->is_div_from_arm[i2c->adap.nr] = pdata->is_div_from_arm;
         if(i2c->is_div_from_arm[i2c->adap.nr])
-                wake_lock_init(&i2c->idlelock[i2c->adap.nr], WAKE_LOCK_IDLE, name);
+                wake_lock_init(&i2c->idlelock[i2c->adap.nr], WAKE_LOCK_IDLE, dev_name(&pdev->dev));
 
         i2c->i2c_init_hw(i2c, 100 * 1000);
 	dev_info(&pdev->dev, "%s: RK30 I2C adapter\n", dev_name(&i2c->adap.dev));
-- 
2.35.3

