From c80ab65510a1d9c3c72ade6649bcc1b5faef681d Mon Sep 17 00:00:00 2001
From: cym <cym@rock-chips.com>
Date: Thu, 24 Jan 2013 15:47:14 +0800
Subject: [PATCH] RK30/RK31:fix ddr.c code error

---
 arch/arm/mach-rk30/ddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/ddr.c b/arch/arm/mach-rk30/ddr.c
index cbce94b257b1..0b903e79cacb 100755
--- a/arch/arm/mach-rk30/ddr.c
+++ b/arch/arm/mach-rk30/ddr.c
@@ -2990,7 +2990,7 @@ void __sramlocalfunc idle_port(void)
         dsb();
         while( (pPMU_Reg->PMU_PWRDN_ST & idle_dma) == 0 );
 #else
-        pPMU_Reg->PMU_MISC_CON1 |= idle_req_cup_cfg;
+        pPMU_Reg->PMU_MISC_CON1 |= idle_req_cpu_cfg;
         dsb();
         while( (pPMU_Reg->PMU_PWRDN_ST & idle_cpu) == 0 );
 #endif
-- 
2.35.3

