From 429fc9474d460309b601316aff5f4c99f16924f8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 24 Sep 2012 11:36:03 +0800
Subject: [PATCH] rk: add RK_CONFIG config

---
 arch/arm/plat-rk/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index a1a1ec1e5483..68960de88248 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -192,4 +192,7 @@ config RK_PL330_DMA_TEST
 config RK_FPGA
 	bool
 
+config RK_CONFIG
+	bool
+
 endif
-- 
2.35.3

