From 88f3f5d1046f0115bda52628f593aaa1065d5c35 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Fri, 1 Oct 2010 07:26:15 -0700
Subject: [PATCH] open touchscreen it7250 suspend and resume

---
 drivers/input/touchscreen/ctp_it7250.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/ctp_it7250.c b/drivers/input/touchscreen/ctp_it7250.c
index 26706afcfd4b..5c7f3833227e 100755
--- a/drivers/input/touchscreen/ctp_it7250.c
+++ b/drivers/input/touchscreen/ctp_it7250.c
@@ -1086,7 +1086,9 @@ static struct i2c_driver Ctp_it7250_driver = {
 	},
 	.id_table 	= Ctp_it7250_id,
 	.probe	= Ctp_it7250_probe,
-	 .remove     =  Ctp_it7250_remove,
+	.remove     =  Ctp_it7250_remove,
+	.suspend = Ctp_it7250_suspend,
+	.resume = Ctp_it7250_resume,
 };
 
 static int __init Ctp_it7250_init(void)
-- 
2.35.3

