From d032591169761d2cac4faa2f427250da171312a6 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Fri, 1 Jul 2011 20:15:44 -0700
Subject: [PATCH] gsensor: improve gsensor

---
 drivers/input/gsensor/mma8452.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/gsensor/mma8452.c b/drivers/input/gsensor/mma8452.c
index cbcf6f5f4ecf..b842d8a8e9cd 100755
--- a/drivers/input/gsensor/mma8452.c
+++ b/drivers/input/gsensor/mma8452.c
@@ -365,7 +365,7 @@ static int mma8452_get_data(struct i2c_client *client)
     do {
         memset(buffer, 0, 3);
         buffer[0] = MMA8452_REG_X_OUT_MSB;
-		ret = mma8452_tx_data(client, &buffer[0], 1);
+		//ret = mma8452_tx_data(client, &buffer[0], 1);
         ret = mma8452_rx_data(client, &buffer[0], 3);
         if (ret < 0)
             return ret;
-- 
2.35.3

