From 5dc0c3f593a97ccdeb66c0082c9f9a22c981f0f7 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Wed, 26 Mar 2014 12:18:22 +0800
Subject: [PATCH] rk fb: close fence

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index c5bb65a4d934..162a3d0cb2ac 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -51,7 +51,7 @@
 #include <linux/dma-buf.h>
 #endif
 #endif
-#define H_USE_FENCE 1
+//#define H_USE_FENCE 1
 static int hdmi_switch_complete;
 static struct platform_device *fb_pdev;
 struct list_head saved_list;
-- 
2.35.3

