From e3a3ed9e6a8066020d4b97c6826ddbce160fdc79 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:27:13 +0800
Subject: [PATCH] f_mass_storage: fix for v2.6.36

---
 drivers/usb/gadget/f_mass_storage.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/gadget/f_mass_storage.c b/drivers/usb/gadget/f_mass_storage.c
index 236863e64747..1a5119c1af4c 100644
--- a/drivers/usb/gadget/f_mass_storage.c
+++ b/drivers/usb/gadget/f_mass_storage.c
@@ -2226,9 +2226,9 @@ static int do_scsi_command(struct fsg_common *common)
 		break;
 #ifdef CONFIG_ARCH_RK29
 	case 0xff:
-		if (fsg->cmnd_size >= 6 && fsg->cmnd[1] == 0xe0 &&
-		    fsg->cmnd[2] == 0xff && fsg->cmnd[3] == 0xff &&
-		    fsg->cmnd[4] == 0xff && fsg->cmnd[5] == 0xfe) {
+		if (common->cmnd_size >= 6 && common->cmnd[1] == 0xe0 &&
+		    common->cmnd[2] == 0xff && common->cmnd[3] == 0xff &&
+		    common->cmnd[4] == 0xff && common->cmnd[5] == 0xfe) {
 			schedule_work(&restart_work);
 		}
 		break;
-- 
2.35.3

