From d27062163c3b36421f4cb6dd2a8cafc1ce8dab18 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 27 Jul 2012 10:12:54 +0800
Subject: [PATCH] rk2928: adc support

---
 drivers/adc/plat/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/adc/plat/Kconfig b/drivers/adc/plat/Kconfig
index 390639868fd7..c70b12a5a349 100644
--- a/drivers/adc/plat/Kconfig
+++ b/drivers/adc/plat/Kconfig
@@ -24,7 +24,7 @@ config ADC_RK29
 
 config ADC_RK30
 	bool "RK30 adc interface"
-        depends on ARCH_RK30
+        depends on ARCH_RK30 || ARCH_RK2928
 	help
 		This supports the use of the ADC interface on rk30 processors.
 endchoice
-- 
2.35.3

