From f9fc02f2166433abed8da64b62314ba1b594163a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 5 Jul 2012 18:47:03 +0800
Subject: [PATCH] rk2928: mach-types add rk2928

---
 arch/arm/tools/mach-types | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/tools/mach-types b/arch/arm/tools/mach-types
index 1fad7ed44a6d..5bb762af6c1d 100644
--- a/arch/arm/tools/mach-types
+++ b/arch/arm/tools/mach-types
@@ -568,8 +568,9 @@ oratisaes		MACH_ORATISAES		ORATISAES		2924
 smdkv310		MACH_SMDKV310		SMDKV310		2925
 siemens_l0		MACH_SIEMENS_L0		SIEMENS_L0		2926
 ventana			MACH_VENTANA		VENTANA			2927
-wm8505_7in_netbook	MACH_WM8505_7IN_NETBOOK	WM8505_7IN_NETBOOK	2928
+#wm8505_7in_netbook	MACH_WM8505_7IN_NETBOOK	WM8505_7IN_NETBOOK	2928
 rk29			MACH_RK29		RK29			2929
+rk2928			ARCH_RK2928		RK2928			2928
 rk30			ARCH_RK30		RK30			3066
 #ec4350sdb		MACH_EC4350SDB		EC4350SDB		2929
 mimas			MACH_MIMAS		MIMAS			2930
-- 
2.35.3

