From 84277541296d1757937b26c8174db63017513fb1 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 8 Jul 2013 17:12:23 +0800
Subject: [PATCH] mmc: emmc: revert thread name

---
 drivers/mmc/card/queue.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/card/queue.c b/drivers/mmc/card/queue.c
index 9088d26b7d51..9678bf579f01 100644
--- a/drivers/mmc/card/queue.c
+++ b/drivers/mmc/card/queue.c
@@ -295,7 +295,8 @@ int mmc_init_queue(struct mmc_queue *mq, struct mmc_card *card,
 	sema_init(&mq->thread_sem, 1);
 
 	if(HOST_IS_EMMC(card->host))
-		mq->thread = kthread_run(emmc_queue_thread, mq, "emmcqd");
+		mq->thread = kthread_run(emmc_queue_thread, mq, "mmcqd/%d%s",
+			host->index, subname ? subname : "");
 	else
 		mq->thread = kthread_run(sdmmc_queue_thread, mq, "mmcqd/%d%s",
 			host->index, subname ? subname : "");
-- 
2.35.3

