From 96ea1afcf8c49fc3f001d078988ca006b4dd10be Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 30 Jan 2013 12:39:07 +0800
Subject: [PATCH] rk_fb.h:add ioctl RK_FBIOSET_W0W1_ZORDER_SWAP

---
 include/linux/rk_fb.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/rk_fb.h b/include/linux/rk_fb.h
index 8a220269a78c..3bd7dd52ef7c 100644
--- a/include/linux/rk_fb.h
+++ b/include/linux/rk_fb.h
@@ -53,6 +53,7 @@
 #define FBIOGET_ENABLE			0x5020
 #define RK_FBIOSET_CONFIG_DONE		0x4628
 #define RK_FBIOSET_VSYNC_ENABLE		0x4629
+#define RK_FBIOSET_W0W1_ZORDER_SWAP	0x4630
 
 
 /********************************************************************
-- 
2.35.3

