From 559c32a3cda0844927d7f83cc3864acf02bc66ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 17 Mar 2014 19:39:24 +0800
Subject: [PATCH] rk: FIQ_DEBUGGER_CONSOLE depends on TTY

---
 arch/arm/common/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/common/Kconfig b/arch/arm/common/Kconfig
index 992d4046bb8a..787ed3393c3b 100644
--- a/arch/arm/common/Kconfig
+++ b/arch/arm/common/Kconfig
@@ -55,6 +55,7 @@ config FIQ_DEBUGGER_WAKEUP_IRQ_ALWAYS_ON
 config FIQ_DEBUGGER_CONSOLE
 	bool "Console on FIQ Serial Debugger port"
 	depends on FIQ_DEBUGGER
+	depends on TTY
 	default n
 	help
 	  Enables a console so that printk messages are displayed on
-- 
2.35.3

