From b70209020396247d18206096fa1cf7158b735504 Mon Sep 17 00:00:00 2001
From: ddl <ddl@rock-chips.com>
Date: Tue, 15 Jan 2013 11:23:06 +0800
Subject: [PATCH] camera ov5640: add some delay from capture to preview

---
 drivers/media/video/ov5640.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/video/ov5640.c b/drivers/media/video/ov5640.c
index 62cc47e12f20..16658e553b05 100755
--- a/drivers/media/video/ov5640.c
+++ b/drivers/media/video/ov5640.c
@@ -2946,6 +2946,8 @@ static int sensor_s_fmt(struct v4l2_subdev *sd,struct v4l2_mbus_framefmt *mf)
                 if (sensor->info_priv.auto_focus == SENSOR_AF_MODE_CONTINUOUS) {
                     sensor_af_workqueue_set(icd,WqCmd_af_continues,0,false,NULL); 
                 }
+
+                msleep(300);
             }
             #else   
             msleep(500);
-- 
2.35.3

