From 80e3280652c992387501cc2d43720758cd12725e Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Wed, 9 Jan 2013 10:45:09 +0800
Subject: [PATCH] rk3168: enable rtc 32K clock out in shut down

---
 drivers/rtc/rtc-HYM8563.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-HYM8563.c b/drivers/rtc/rtc-HYM8563.c
index 2c292af643dc..ff766d44df41 100755
--- a/drivers/rtc/rtc-HYM8563.c
+++ b/drivers/rtc/rtc-HYM8563.c
@@ -538,7 +538,9 @@ static struct i2c_driver hym8563_driver = {
 	},
 	.probe		= hym8563_probe,
 	.remove		= __devexit_p(hym8563_remove),
-	.shutdown=hym8563_shutdown,
+#if defined(CONFIG_ARCH_RK3066B)
+	//.shutdown=hym8563_shutdown,
+#endif
 	.id_table	= hym8563_id,
 };
 
-- 
2.35.3

