From 4ec8369de294714eb4d06773c42edf269fb59242 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 2 Apr 2012 18:29:59 +0800
Subject: [PATCH] rk30: cpufreq: disable debug print

---
 arch/arm/mach-rk30/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index 90e2469926cd..44ab6b2b6cbc 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -28,7 +28,7 @@
 
 #define FREQ_PRINTK_DBG(fmt, args...) {while(0);}//pr_debug(fmt, ## args)
 #define FREQ_PRINTK_ERR(fmt, args...) pr_err(fmt, ## args)
-#define FREQ_PRINTK_LOG(fmt, args...) printk(fmt, ## args)
+#define FREQ_PRINTK_LOG(fmt, args...) pr_debug(fmt, ## args)
 /* Frequency table index must be sequential starting at 0 */
 static struct cpufreq_frequency_table default_freq_table[] = {
 	{.frequency = 816*1000, .index	= 1080*1000},
-- 
2.35.3

