From cdff984abff056096bd45a6e39f08a62aa4308f0 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Thu, 21 May 2015 18:28:22 +0800
Subject: [PATCH] arm: rockchip: vcodec_service: allow no rockchip,grf in dts

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/mach-rockchip/vcodec_service.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rockchip/vcodec_service.c b/arch/arm/mach-rockchip/vcodec_service.c
index 3fa90d33f10e..548f9556c5a9 100644
--- a/arch/arm/mach-rockchip/vcodec_service.c
+++ b/arch/arm/mach-rockchip/vcodec_service.c
@@ -2137,8 +2137,12 @@ static void vcodec_read_property(struct device_node *np,
 	pservice->grf_base = (u32*)RK_GRF_VIRT;
 #endif
 	if (IS_ERR(pservice->grf_base)) {
+#ifdef CONFIG_ARM
+		pservice->grf_base = RK_GRF_VIRT;
+#else
 		vpu_err("can't find vpu grf property\n");
 		return;
+#endif
 	}
 	of_property_read_string(np, "name", (const char**)&pservice->name);
 }
-- 
2.35.3

