From 67d7c692601a9133d87f0fe66edd95c2e20d4069 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 21 Mar 2013 18:17:43 +0800
Subject: [PATCH] rk3188: arm-pmu: support qual core

---
 arch/arm/mach-rk30/devices.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index f751bb130835..32ddb4b9f1a4 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -1173,6 +1173,18 @@ static struct resource resource_arm_pmu[] = {
 		.end	= IRQ_ARM_PMU + 1,
 		.flags	= IORESOURCE_IRQ,
 	},
+#if defined(CONFIG_ARCH_RK3188)
+	{
+		.start	= IRQ_ARM_PMU + 2,
+		.end	= IRQ_ARM_PMU + 2,
+		.flags	= IORESOURCE_IRQ,
+	},
+	{
+		.start	= IRQ_ARM_PMU + 3,
+		.end	= IRQ_ARM_PMU + 3,
+		.flags	= IORESOURCE_IRQ,
+	},
+#endif
 };
 
 struct platform_device device_arm_pmu = {
-- 
2.35.3

