From 2465250f7e9c350fd0390fc910db235e95161424 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 17 Aug 2012 22:37:19 +0800
Subject: [PATCH] rk2928: devices.c: add dummy i2c check idle

---
 arch/arm/mach-rk2928/devices.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/mach-rk2928/devices.c b/arch/arm/mach-rk2928/devices.c
index 0e2c3290cd30..9be98c671c93 100755
--- a/arch/arm/mach-rk2928/devices.c
+++ b/arch/arm/mach-rk2928/devices.c
@@ -280,10 +280,16 @@ static struct platform_device device_lcdc = {
 #endif
 
 #ifdef CONFIG_I2C0_RK30
+static int i2c0_check_idle(void)
+{
+	return I2C_IDLE;
+}
+
 static struct rk30_i2c_platform_data default_i2c0_data = {
 	.bus_num = 0,
 	.is_div_from_arm = 1,
 	.adap_type = I2C0_ADAP_TYPE,
+	.check_idle = &i2c0_check_idle,
 };
 
 static struct resource resources_i2c0[] = {
@@ -311,10 +317,16 @@ static struct platform_device device_i2c0 = {
 #endif
 
 #ifdef CONFIG_I2C1_RK30
+static int i2c1_check_idle(void)
+{
+	return I2C_IDLE;
+}
+
 static struct rk30_i2c_platform_data default_i2c1_data = {
 	.bus_num = 1,
 	.is_div_from_arm = 1,
 	.adap_type = I2C1_ADAP_TYPE,
+	.check_idle = &i2c1_check_idle,
 };
 
 static struct resource resources_i2c1[] = {
@@ -342,10 +354,16 @@ static struct platform_device device_i2c1 = {
 #endif
 
 #ifdef CONFIG_I2C2_RK30
+static int i2c2_check_idle(void)
+{
+	return I2C_IDLE;
+}
+
 static struct rk30_i2c_platform_data default_i2c2_data = {
 	.bus_num = 2,
 	.is_div_from_arm = 0,
 	.adap_type = I2C2_ADAP_TYPE,
+	.check_idle = &i2c2_check_idle,
 };
 
 static struct resource resources_i2c2[] = {
@@ -373,10 +391,16 @@ static struct platform_device device_i2c2 = {
 #endif
 
 #ifdef CONFIG_I2C3_RK30
+static int i2c3_check_idle(void)
+{
+	return I2C_IDLE;
+}
+
 static struct rk30_i2c_platform_data default_i2c3_data = {
 	.bus_num = 3,
 	.is_div_from_arm = 0,
 	.adap_type = I2C3_ADAP_TYPE,
+	.check_idle = &i2c3_check_idle,
 };
 
 static struct resource resources_i2c3[] = {
-- 
2.35.3

