From 314bfb2db73f622544e9147aba3304674aae1bc2 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Thu, 20 Jun 2013 16:52:48 +0800
Subject: [PATCH] Wi-Fi:MT7601U select NL80211_TESTMODE

---
 drivers/net/wireless/rkusbwifi/mt7601/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/rkusbwifi/mt7601/Kconfig b/drivers/net/wireless/rkusbwifi/mt7601/Kconfig
index 23a906b8bb4f..40f154af96a1 100755
--- a/drivers/net/wireless/rkusbwifi/mt7601/Kconfig
+++ b/drivers/net/wireless/rkusbwifi/mt7601/Kconfig
@@ -3,6 +3,8 @@ config MT7601
    	select WIRELESS_EXT
 	select WEXT_PRIV
 	select IEEE80211
+	select CFG80211
+	select NL80211_TESTMODE
 	---help---
 	  Help message of MT7601
 
-- 
2.35.3

