From 0cea378785425159e66fe8f28f72f501328f4362 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E6=81=92=E6=98=8E?= <chm@rock-chips.com>
Date: Wed, 29 Dec 2010 18:53:37 -0800
Subject: [PATCH] rk29: vpu: remove printk

---
 arch/arm/mach-rk29/vpu.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk29/vpu.c b/arch/arm/mach-rk29/vpu.c
index fb343c4c28e9..2bc9aa7320ac 100644
--- a/arch/arm/mach-rk29/vpu.c
+++ b/arch/arm/mach-rk29/vpu.c
@@ -102,10 +102,10 @@ static void vpu_put_clk(void)
 
 static void vpu_power_on(void)
 {
-	printk("power on\n");
+	pr_debug("power on\n");
 	if (client.enabled)
 		return;
-	printk("power domain on\n");
+	pr_debug("power domain on\n");
 	pmu_set_power_domain(PD_VCODEC, true);
     udelay(10);
 	clk_enable(aclk_vepu);
@@ -125,14 +125,14 @@ static void vpu_power_on(void)
 
 static void vpu_power_off(void)
 {
-	printk("power off\n");
+	pr_debug("power off\n");
 	if (!client.enabled)
 		return;
 	clk_disable(hclk_cpu_vcodec);
 	clk_disable(aclk_ddr_vepu);
 	clk_disable(hclk_vepu);
 	clk_disable(aclk_vepu);
-	printk("power domain off\n");
+	pr_debug("power domain off\n");
 	pmu_set_power_domain(PD_VCODEC, false);
 	client.enabled = false;
 }
-- 
2.35.3

