From b4e1e85c1428f65112163fb6abe96a897a050495 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Feb 2012 16:58:16 +0800
Subject: [PATCH] rk29: fix vivante Kconfig, only for rk29

---
 drivers/staging/rk29/vivante/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/staging/rk29/vivante/Kconfig b/drivers/staging/rk29/vivante/Kconfig
index 7e0bfe8f7273..e8e7de59ef0a 100644
--- a/drivers/staging/rk29/vivante/Kconfig
+++ b/drivers/staging/rk29/vivante/Kconfig
@@ -1,4 +1,6 @@
 menu "GPU Vivante"
+	depends on ARCH_RK29
+
 config VIVANTE
 	tristate "ROCKCHIP Vivante GPU"
 	default y
-- 
2.35.3

