From 468a5f734f22e88c893a9f17e1d10f3c7cf4189f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 26 Feb 2013 16:23:51 +0800
Subject: [PATCH] rk3188: set l2c data latency 231

---
 arch/arm/mach-rk30/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index ec38ceaca46f..c28e801aa2ab 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -59,6 +59,9 @@ static void __init rk30_l2_cache_init(void)
 {
 #ifdef CONFIG_CACHE_L2X0
 	u32 aux_ctrl, aux_ctrl_mask, data_latency_ctrl;
+#if defined(CONFIG_ARCH_RK3188)
+	data_latency_ctrl = L2_LATENCY(2, 3, 1);
+#else
 	unsigned int max_cpu_freq = 1608000; // kHz
 	struct cpufreq_frequency_table *table = NULL;
 	struct clk *clk_cpu;
@@ -84,6 +87,7 @@ static void __init rk30_l2_cache_init(void)
 		data_latency_ctrl = L2_LATENCY(5, 8, 1);
 	else
 		data_latency_ctrl = L2_LATENCY(6, 8, 1);
+#endif
 
 	writel_relaxed(L2_LATENCY(1, 1, 1), RK30_L2C_BASE + L2X0_TAG_LATENCY_CTRL);
 	writel_relaxed(data_latency_ctrl, RK30_L2C_BASE + L2X0_DATA_LATENCY_CTRL);
-- 
2.35.3

