From 3388a670bd597689bf24ff187a9a1e915974df46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 17 Jun 2014 12:53:45 +0800
Subject: [PATCH] rk: gcc-wrapper.py ignore atags_to_fdt.c:96

---
 scripts/gcc-wrapper.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/gcc-wrapper.py b/scripts/gcc-wrapper.py
index c43deedfd799..1f4ee7b5fc68 100755
--- a/scripts/gcc-wrapper.py
+++ b/scripts/gcc-wrapper.py
@@ -47,6 +47,7 @@ allowed_warnings = set([
     "core.c:1334",
     "menu.c:129",
     "bus.c:318", # FIXME
+    "atags_to_fdt.c:96",
  ])
 
 # Capture the name of the object file, can find it.
-- 
2.35.3

