From f83bb2e8d2f99577cdb04f9ae50c59b8653dd09e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 20 Mar 2014 21:57:04 +0800
Subject: [PATCH] ARM: rockchip: rk3288 disable suspend init temporarily

---
 arch/arm/mach-rockchip/rk3288.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index fac75f596a3d..1ee73bb90e51 100755
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -390,6 +390,7 @@ arch_initcall(rk3288_pie_init);
 #include "pm-rk3288.c"
 static void __init rk3288_init_suspend(void)
 {
+	return;
         rockchip_suspend_init();       
         rkpm_pie_init();
         rk3288_suspend_init();
-- 
2.35.3

