From a882ac960e745ff46b65623e4713493a6b7e1ea1 Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Mon, 30 Dec 2013 09:39:15 +0800
Subject: [PATCH] pinctrl:fix irq operation bug

---
 drivers/pinctrl/pinctrl-rockchip.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index fff102b4eb50..7ce7a95a6198 100755
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -1689,7 +1689,7 @@ static void rockchip_irq_demux(unsigned int irq, struct irq_desc *desc)
 	u32 pend;
 	bool edge_changed = false;
 
-	dev_dbg(bank->drvdata->dev, "got irq for bank %s\n", bank->name);
+	DBG_PINCTRL("%s:got irq for bank %s\n", __func__, bank->name);
 
 	chained_irq_enter(chip, desc);
 
@@ -1830,9 +1830,11 @@ static inline void rockchip_gpio_bit_op(void __iomem *reg_base, unsigned int off
 {
 	u32 val = __raw_readl(reg_base + offset);
 	if (flag)
-		val |= bit;
+		val |= BIT(bit);
 	else
-		val &= ~bit;
+		val &= ~BIT(bit);
+
+	
 	__raw_writel(val, reg_base + offset);
 }
 
-- 
2.35.3

