From d16df4062df26a95e580dd8779152ce902d8cf4b Mon Sep 17 00:00:00 2001
From: ChenJQ <chenjq@rock-chips.com>
Date: Mon, 11 Apr 2011 02:37:11 -0700
Subject: [PATCH] change wm8900, delete useless code

---
 sound/soc/codecs/wm8900.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/sound/soc/codecs/wm8900.c b/sound/soc/codecs/wm8900.c
index 2551e7e09618..f970e30cef7b 100755
--- a/sound/soc/codecs/wm8900.c
+++ b/sound/soc/codecs/wm8900.c
@@ -1095,16 +1095,9 @@ static void wm8900_shutdown(struct snd_pcm_substream *substream,
 	struct snd_soc_dai_link *machine = rtd->dai;
 	struct snd_soc_dai *codec_dai = machine->codec_dai;
 	struct snd_soc_codec *codec = socdev->card->codec;
-	
-	WM8900_DBG("Enter::%s----%d substream->stream:%s \n",__FUNCTION__,__LINE__,
-		   substream->stream == SNDRV_PCM_STREAM_PLAYBACK ? "PLAYBACK":"CAPTURE");	
-
-	/* If  */
-	if (!codec_dai->capture.active && !codec_dai->playback.active &&
-	    (substream->stream == SNDRV_PCM_STREAM_CAPTURE)) {
 
-		msleep(200);
-	}
+	WM8900_DBG("Enter::%s----%d substream->stream:%s \n",__FUNCTION__,__LINE__,
+		   substream->stream == SNDRV_PCM_STREAM_PLAYBACK ? "PLAYBACK":"CAPTURE");
 
 	if (!codec_dai->capture.active && !codec_dai->playback.active) {
 
-- 
2.35.3

