From 295211b721e6797ff665e83a5186bf629ec42e8a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E6=81=92=E6=98=8E?= <chm@rock-chips.com>
Date: Fri, 23 Mar 2012 15:37:07 +0800
Subject: [PATCH] rk30: vpu_service: add rk30 enc hw id checklist

---
 arch/arm/plat-rk/vpu_service.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/plat-rk/vpu_service.c b/arch/arm/plat-rk/vpu_service.c
index c2c68821ffac..ab3559fd376e 100644
--- a/arch/arm/plat-rk/vpu_service.c
+++ b/arch/arm/plat-rk/vpu_service.c
@@ -64,7 +64,11 @@
 #define DEC_IO_SIZE 				((100 + 1) * 4) /* bytes */
 #define ENC_IO_SIZE 				(96 * 4)	/* bytes */
 static const u16 dec_hw_ids[] = { 0x8190, 0x8170, 0x9170, 0x9190, 0x6731 };
+#if defined(CONFIG_ARCH_RK29)
 static const u16 enc_hw_ids[] = { 0x6280, 0x7280, 0x8270 };
+#elif defined(CONFIG_ARCH_RK30)
+static const u16 enc_hw_ids[] = { 0x6280, 0x7280, 0x8270, 0x8290, 0x4831 };
+#endif
 
 #define VPU_REG_EN_ENC				14
 #define VPU_REG_ENC_GATE			2
-- 
2.35.3

