From 095540f090bb5fcf4b02c5bae00a753d43d804a5 Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Tue, 13 Nov 2012 16:06:34 +0800
Subject: [PATCH] rk fb : fix flash screen when switch hdmi

---
 drivers/video/rockchip/rk_fb.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index eea704bc2b7b..9016edaae177 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -629,7 +629,6 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 	int ret;
 	int i;
 	int layer_id;
-	bool backlight_status =	rk29_get_backlight_status();
 
 #if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)
 	rk29_backlight_set(0);
@@ -764,10 +763,10 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 	#endif 
 
 #if defined(CONFIG_NO_DUAL_DISP)  //close backlight for device whic do not support dual display
-	rk29_backlight_set(!enable);
-#elif defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)  //close backlight for device whic do not support dual display
-	if(backlight_status == true)
+	if(!enable)
 		rk29_backlight_set(1);
+#elif defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)  //close backlight for device whic do not support dual display
+	rk29_backlight_set(1);
 #endif
 	return 0;
 
-- 
2.35.3

