From bc4a13cf3fd6f1e0c4b4ad7fe37a55fb4b594129 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 16 Jan 2013 14:29:13 +0800
Subject: [PATCH] rk30: update dvfs volt diff table

---
 arch/arm/mach-rk30/dvfs.c | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index b9066b23e3a0..bafda02d6f67 100644
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -1032,18 +1032,31 @@ int dvfs_get_depend_volt(struct clk_node *dvfs_clk, struct vd_node *dvfs_vd_dep,
 	return -1;
 }
 
+#ifdef CONFIG_SOC_RK3168
 static struct cpufreq_frequency_table arm_high_logic_table[] = {
-	{.frequency = 1416 * DVFS_KHZ, .index = 50 * DVFS_MV},
-	{.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
-	{.frequency = CPUFREQ_TABLE_END},
+        {.frequency = 1416 * DVFS_KHZ, .index = 50 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 75 * DVFS_MV},
+        {.frequency = CPUFREQ_TABLE_END},
 };
 
 static struct cpufreq_frequency_table logic_high_arm_table[] = {
-	{.frequency = 816 * DVFS_KHZ,  .index = 200 * DVFS_MV},
-	{.frequency = 1416 * DVFS_KHZ, .index = 150 * DVFS_MV},
-	{.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
-	{.frequency = CPUFREQ_TABLE_END},
+        {.frequency = 1008 * DVFS_KHZ, .index = 275 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 150 * DVFS_MV},
+        {.frequency = CPUFREQ_TABLE_END},
+};
+#else
+static struct cpufreq_frequency_table arm_high_logic_table[] = {
+        {.frequency = 1416 * DVFS_KHZ, .index = 50 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
+        {.frequency = CPUFREQ_TABLE_END},
 };
+static struct cpufreq_frequency_table logic_high_arm_table[] = {
+        {.frequency = 816 * DVFS_KHZ,  .index = 200 * DVFS_MV},
+        {.frequency = 1416 * DVFS_KHZ, .index = 150 * DVFS_MV},
+        {.frequency = 1608 * DVFS_KHZ, .index = 100 * DVFS_MV},
+};
+#endif
+
 
 int get_arm_logic_limit(unsigned long arm_rate, int *arm_high_logic, int *logic_high_arm)
 {
-- 
2.35.3

