From b94cf54a8161d004ad648c6ae54c19c53b90e71d Mon Sep 17 00:00:00 2001
From: chenli <chenli@rock-chips.com>
Date: Sat, 31 Mar 2012 11:07:15 +0800
Subject: [PATCH] RK30 IPP:modify to add ko support

---
 arch/arm/mach-rk30/devices.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk30/devices.c b/arch/arm/mach-rk30/devices.c
index 184f80d94000..2dd1c4ef3406 100755
--- a/arch/arm/mach-rk30/devices.c
+++ b/arch/arm/mach-rk30/devices.c
@@ -804,7 +804,7 @@ static struct platform_device device_rga = {
 };
 #endif
 
-#ifdef CONFIG_RK29_IPP
+//#ifdef CONFIG_RK29_IPP
 static struct resource resource_ipp[] = {
 	[0] = {
 		.start = RK30_IPP_PHYS,
@@ -824,7 +824,7 @@ static struct platform_device device_ipp = {
 	.num_resources	= ARRAY_SIZE(resource_ipp),
 	.resource	= resource_ipp,
 };
-#endif
+//#endif
 
 #ifdef CONFIG_SND_RK29_SOC_I2S
 #ifdef CONFIG_SND_RK29_SOC_I2S_8CH
@@ -1080,9 +1080,9 @@ static int __init rk30_init_devices(void)
 #ifdef CONFIG_RGA_RK30
 	platform_device_register(&device_rga);
 #endif
-#ifdef CONFIG_RK29_IPP
+//#ifdef CONFIG_RK29_IPP
 	platform_device_register(&device_ipp);
-#endif
+//#endif
 #ifdef CONFIG_LCDC0_RK30
 	platform_device_register(&device_lcdc0);
 #endif
-- 
2.35.3

