From 599a16a283e9025befa5b77ac76d11ce47d4e6c8 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 2 Sep 2014 17:48:46 +0800
Subject: [PATCH] audio: rk3036 codec: add init codec sample clock

---
 sound/soc/rockchip/rk_rk3036.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/rockchip/rk_rk3036.c b/sound/soc/rockchip/rk_rk3036.c
index b08134dfcfd2..7547c0cc22f0 100755
--- a/sound/soc/rockchip/rk_rk3036.c
+++ b/sound/soc/rockchip/rk_rk3036.c
@@ -85,6 +85,7 @@ static int rk30_hw_params(struct snd_pcm_substream *
 		return ret;
 	}
 
+	snd_soc_dai_set_sysclk(cpu_dai, 0, pll_out, 0);
 	snd_soc_dai_set_clkdiv(cpu_dai, ROCKCHIP_DIV_BCLK,
 			       (pll_out/4)/params_rate(params)-1);
 	snd_soc_dai_set_clkdiv(cpu_dai, ROCKCHIP_DIV_MCLK, 3);
-- 
2.35.3

