From a507fb5df3e74d8981a1774ba388f8e288937c00 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 7 Feb 2012 15:33:13 +0800
Subject: [PATCH] rk30: timer: fix for smp

---
 arch/arm/mach-rk30/timer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/timer.c b/arch/arm/mach-rk30/timer.c
index c4d35965d1bb..1ee3310eb3d6 100644
--- a/arch/arm/mach-rk30/timer.c
+++ b/arch/arm/mach-rk30/timer.c
@@ -21,6 +21,7 @@
 #include <linux/clockchips.h>
 #include <linux/io.h>
 
+#include <asm/localtimer.h>
 #include <asm/sched_clock.h>
 #include <asm/mach/time.h>
 
-- 
2.35.3

