From f4a75bd4aa164910b7a9b01a72369f601e8df383 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=93=9D=E6=96=8C=E5=85=83?= <lby@rock-chips.com>
Date: Fri, 4 May 2012 10:37:08 +0800
Subject: [PATCH] sc8803 : modify sc8803 for rk30

---
 drivers/misc/sc8800.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/misc/sc8800.c b/drivers/misc/sc8800.c
index 0a13bb432c6a..88199b8bf6f7 100755
--- a/drivers/misc/sc8800.c
+++ b/drivers/misc/sc8800.c
@@ -316,11 +316,19 @@ static int sc8800_tx(struct sc8800_data *sc8800)
 		schedule();
 	}
 	mutex_lock(&sc8800s_lock);
+	#if defined(CONFIG_ARCH_RK30)
+	ap_rts(sc8800,1);
+	#else
 	ap_rts(sc8800,0);
+	#endif
 #if 1
 	while(bp_rdy(sc8800)){
 		if(sc8800->write_tmo){
+			#if defined(CONFIG_ARCH_RK30)
+			ap_rts(sc8800,0);
+			#else
 			ap_rts(sc8800,1);
+			#endif
 			sc8800_dbg(sc8800->dev, "ERR: %s write timeout ->bp not ready (bp_rdy = 1)\n", __func__);
 			kfree(buf);
 			mutex_unlock(&sc8800s_lock);
@@ -334,7 +342,11 @@ static int sc8800_tx(struct sc8800_data *sc8800)
 
 	if(ret < 0)
 		dev_err(sc8800->dev, "ERR: %s spi in err = %d\n", __func__, ret);
+	#if defined(CONFIG_ARCH_RK30)
+	ap_rts(sc8800,0);
+	#else
 	ap_rts(sc8800,1);
+	#endif
 	if(buf){
 		kfree(buf);
 		buf = NULL;
-- 
2.35.3

