From e5802a4d63392b83dbccf77e5dc62b594a2bb29e Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Mon, 31 Mar 2014 16:35:49 +0800
Subject: [PATCH] pinctrl:fix _rockchip_pinconf_set bug

---
 drivers/pinctrl/pinctrl-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index b875374722c0..2a8db2972ee5 100755
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -1630,7 +1630,7 @@ static int _rockchip_pinconf_set(struct rockchip_pin_bank *bank,
 		for(i=0; i < j; i++)
 		{
 			reg = info->reg_base + reg_offset[i].reg_offset;
-			data |= ((arg & reg_offset[i].bit_mask) << (16 + reg_offset[i].bit_offset));
+			data |= ((reg_offset[i].bit_mask) << (16 + reg_offset[i].bit_offset));
 			data |= ((arg & reg_offset[i].bit_mask) << reg_offset[i].bit_offset);
 			spin_lock_irqsave(&bank->slock, flags);
 			writel_relaxed(data, reg);	
-- 
2.35.3

