From c56c6a38c6187cbe11a1522bcb4edea13b37275e Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Tue, 15 Oct 2013 20:02:31 +0800
Subject: [PATCH] bluetooth: add ap6210 24m support

---
 net/rfkill/rfkill-rk.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/rfkill/rfkill-rk.c b/net/rfkill/rfkill-rk.c
index 52d6bf2f695b..cb5c717cd494 100755
--- a/net/rfkill/rfkill-rk.c
+++ b/net/rfkill/rfkill-rk.c
@@ -122,7 +122,11 @@ static const char bt_name[] =
 #elif defined(CONFIG_MV8787)
         "mv8787"
 #elif defined(CONFIG_AP6210)
+    #if defined(CONFIG_RKWIFI_26M)
         "ap6210"
+    #else
+        "ap6210_24M"
+    #endif
 #elif defined(CONFIG_AP6330)
 		"ap6330"
 #elif defined(CONFIG_AP6476)
-- 
2.35.3

