From 28ef1d8c93c668c535145d9e94385335138247e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 16 Aug 2012 12:57:25 +0800
Subject: [PATCH] Input: rk29_keys: close debug print, open by 9ea1329

---
 drivers/input/keyboard/rk29_keys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/rk29_keys.c b/drivers/input/keyboard/rk29_keys.c
index d184b51a1e0a..b5881d5b5e93 100755
--- a/drivers/input/keyboard/rk29_keys.c
+++ b/drivers/input/keyboard/rk29_keys.c
@@ -32,7 +32,7 @@
 #define INVALID_ADVALUE 				10
 
 
-#if 1
+#if 0
 #define key_dbg(bdata, format, arg...)		\
 	dev_printk(KERN_INFO , &bdata->input->dev , format , ## arg)
 #else
-- 
2.35.3

