From 7b1cd94ff4114dcdcefd4f540fa7e0861b806758 Mon Sep 17 00:00:00 2001
From: zhaoyifeng <zyf@rock-chips.com>
Date: Wed, 8 Jun 2011 15:01:09 +0800
Subject: [PATCH] support ext3 and ext4 file system for 'system' mtd part

---
 drivers/mtd/rknand/rknand_base.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/rknand/rknand_base.c b/drivers/mtd/rknand/rknand_base.c
index 3f4977ae96fa..ef7ad1eee7a8 100755
--- a/drivers/mtd/rknand/rknand_base.c
+++ b/drivers/mtd/rknand/rknand_base.c
@@ -659,10 +659,10 @@ static int rknand_probe(struct platform_device *pdev)
     parts = nand_info->parts;
     for(i=0;i<g_num_partitions;i++)
     {
-        printk(">>> part[%d]: name=%s offset=0x%012llx\n", i, parts[i].name, parts[i].offset);
-        if(strcmp(parts[i].name,"cache") == 0)
+        //printk(">>> part[%d]: name=%s offset=0x%012llx\n", i, parts[i].name, parts[i].offset);
+        if(strcmp(parts[i].name,"backup") == 0)
         {
-            SysImageWriteEndAdd = (unsigned long)parts[i].offset>>9;//sector
+            SysImageWriteEndAdd = (unsigned long)(parts[i].offset + parts[i].size)>>9;//sector
 	        printk(">>> SysImageWriteEndAdd=0x%lx\n", SysImageWriteEndAdd);
             break;
         }
-- 
2.35.3

