From eb4f115fee8e015dbb3369c027f7066bf2ae33eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 22 Feb 2013 18:03:43 +0800
Subject: [PATCH] rk: enable synchronization framework, for support android 4.2
 with mali gpu

---
 arch/arm/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 07be42f84d3e..d6a248a2e4d9 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1091,6 +1091,9 @@ config PLAT_RK
 	select ARCH_HAS_CPUFREQ
 	select GENERIC_CLOCKEVENTS
 	select ARCH_REQUIRE_GPIOLIB
+	select SYNC
+	select SW_SYNC
+	select SW_SYNC_USER
 
 config PLAT_VERSATILE
 	bool
-- 
2.35.3

