From b956cbf9135ca890de335664a4b723b0abea3b6a Mon Sep 17 00:00:00 2001
From: qjb <qjb@rock-chips.com>
Date: Tue, 20 Jan 2015 10:28:07 +0800
Subject: [PATCH] ddr_freq: fix when enable armtrustzone ,then ddr_freq error.

---
 arch/arm/mach-rockchip/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/Makefile b/arch/arm/mach-rockchip/Makefile
index 8dfc9e53f23f..d3e49d113acb 100755
--- a/arch/arm/mach-rockchip/Makefile
+++ b/arch/arm/mach-rockchip/Makefile
@@ -5,7 +5,7 @@ obj-y += rk312x.o
 obj-y += rk3126b.o
 obj-y += rk3188.o
 obj-y += rk3288.o
-ifeq ($(CONFIG_ARM_TRUSTZONE),n)
+ifneq ($(CONFIG_ARM_TRUSTZONE),y)
 obj-y += ddr_freq.o
 endif
 obj-y += efuse.o
-- 
2.35.3

