From 0e980e465de44e8601c20708df49c944eae9326c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9E=97=E8=BE=89=E8=BE=89?= <lhh@rock-chips.com>
Date: Mon, 17 May 2010 03:43:35 +0000
Subject: [PATCH] update i2c

---
 drivers/i2c/busses/i2c-rk2818.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rk2818.c b/drivers/i2c/busses/i2c-rk2818.c
index eb2c3638ddd8..f7f3f65e9282 100755
--- a/drivers/i2c/busses/i2c-rk2818.c
+++ b/drivers/i2c/busses/i2c-rk2818.c
@@ -587,7 +587,7 @@ static int rk2818_i2c_probe(struct platform_device *pdev)
 
 	i2c->dev = &pdev->dev;
 	
-	i2c->clk = clk_get(&pdev->dev, DRV_NAME);
+	i2c->clk = clk_get(&pdev->dev, "i2c");
 	if (IS_ERR(i2c->clk)) {
 		dev_err(&pdev->dev, "cannot get clock\n");
 		ret = -ENOENT;
-- 
2.35.3

