From 82c4a8857af6ffbc2b1d529d64abee2c02da8b24 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 10 Dec 2014 19:24:50 +0800
Subject: [PATCH] irqchip: gic: covert dsb() to dsb(sy)

---
 drivers/irqchip/irq-gic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic.c b/drivers/irqchip/irq-gic.c
index c58923eee0c5..ce24a7e86349 100644
--- a/drivers/irqchip/irq-gic.c
+++ b/drivers/irqchip/irq-gic.c
@@ -406,7 +406,7 @@ static void __init gic_dist_init(struct gic_chip_data *gic)
 	for (i = 0; i < gic_irqs; i += 32) {
 		writel_relaxed(0xffffffff, base + GIC_DIST_IGROUP + i * 4 / 32);
 	}
-	dsb();
+	dsb(sy);
 	writel_relaxed(3, base + GIC_DIST_CTRL);
 #else
 	writel_relaxed(1, base + GIC_DIST_CTRL);
-- 
2.35.3

