From 4b8332ed3d45db6aab16b8b3c4189cb4b0627fcf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 20 Feb 2012 15:17:28 +0800
Subject: [PATCH] arm: head.S: port to rk platform

---
 arch/arm/kernel/head.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/kernel/head.S b/arch/arm/kernel/head.S
index 81356ebd993b..5a9c6cace9a6 100644
--- a/arch/arm/kernel/head.S
+++ b/arch/arm/kernel/head.S
@@ -42,7 +42,7 @@
 	.equ	swapper_pg_dir, KERNEL_RAM_VADDR - 0x4000
 
 	.macro	pgtbl, rd, phys
-#if defined(CONFIG_ARCH_RK29) || defined(CONFIG_ARCH_RK30)
+#ifdef CONFIG_PLAT_RK
 	add	\rd, \phys, #((TEXT_OFFSET - 0x4000) & 0xffff0000)
 	add	\rd, \rd,   #((TEXT_OFFSET - 0x4000) & 0x0000ffff)
 #else
-- 
2.35.3

