From fff138fbe8e1f58324e2342ed4143bc495c15fea Mon Sep 17 00:00:00 2001
From: zwl <zwl@rock-chips.com>
Date: Tue, 25 Mar 2014 22:16:24 +0800
Subject: [PATCH] rk fb: fix HDMI display err when system boot

---
 drivers/video/rockchip/rk_fb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 6a8626676b4c..766fe1ab51da 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -2402,6 +2402,8 @@ int rk_fb_switch_screen(struct rk_screen *screen , int enable, int lcdc_id)
 	if (dev_drv->ops->lcdc_hdmi_process)
 		dev_drv->ops->lcdc_hdmi_process(dev_drv, enable);
 
+	hdmi_switch_complete = enable;
+
 	if (rk_fb->disp_mode == DUAL) {
 		if (likely(rk_fb->num_lcdc == 2)) {
 			pmy_info = rk_fb->fb[0];
@@ -2424,7 +2426,6 @@ int rk_fb_switch_screen(struct rk_screen *screen , int enable, int lcdc_id)
 
 	//if (rk_fb->disp_mode != DUAL)
 	//	rk29_backlight_set(1);
-	hdmi_switch_complete = enable;
 	return 0;
 
 }
-- 
2.35.3

