From 3ba2657b65fb3c17efdbd23a86862d37d7c75a0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 23 Mar 2012 10:34:31 +0800
Subject: [PATCH] rk30: fix smp startup

---
 arch/arm/mach-rk30/headsmp.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/headsmp.S b/arch/arm/mach-rk30/headsmp.S
index 5599b86dfc4d..48c3e1454a54 100644
--- a/arch/arm/mach-rk30/headsmp.S
+++ b/arch/arm/mach-rk30/headsmp.S
@@ -55,7 +55,7 @@ ENDPROC(rk30_secondary_startup)
 
 ENTRY(rk30_sram_secondary_startup)
         ldr     pc, 1f
-        .word   0xdeadbeaf
+        ldr     pc, 1f
         ldr     pc, 1f
         ldr     pc, 1f
         ldr     pc, 1f
-- 
2.35.3

