From c56c87ef0d605e5b46c669cbe1e341966d34b292 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=92=9F=E5=8B=87=E6=B1=AA?= <zyw@rock-chips.com>
Date: Tue, 8 Jun 2010 09:05:46 +0000
Subject: [PATCH] update fb power control 2

---
 drivers/video/rk2818_fb.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/video/rk2818_fb.c b/drivers/video/rk2818_fb.c
index 3a45f4964671..1d339e1ce6a7 100644
--- a/drivers/video/rk2818_fb.c
+++ b/drivers/video/rk2818_fb.c
@@ -203,11 +203,9 @@ void set_lcd_pin(struct platform_device *pdev, int enable)
     int ret =0;
 	struct rk2818_fb_mach_info *mach_info = pdev->dev.platform_data;
 
-	unsigned lcd_cs = mach_info->gpio->lcd_cs&0xffff;
 	unsigned display_on = mach_info->gpio->display_on&0xffff;
 	unsigned lcd_standby = mach_info->gpio->lcd_standby&0xffff;
-
-	int lcd_cs_pol = (mach_info->gpio->lcd_cs>>16)&0xffff;
+    
 	int display_on_pol = (mach_info->gpio->display_on>>16)&0xffff;
 	int lcd_standby_pol = (mach_info->gpio->lcd_standby>>16)&0xffff;
 
-- 
2.35.3

