From e24fe4d32a1017cf20af107b82de3ba414927ab7 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Thu, 18 Aug 2011 18:28:53 +0800
Subject: [PATCH] remove 1920x1080p scale up

---
 drivers/video/rk29_fb.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 7fcfbd4a9cb8..9bb8a6164f55 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1223,8 +1223,8 @@ static int win1_set_par(struct fb_info *info)
     //fbprintk(">>>>>> %s : %s\n", __FILE__, __FUNCTION__);
 
    #ifdef CONFIG_FB_SCALING_OSD
-    if(((screen->x_res != var->xres) || (screen->y_res != var->yres))
-        && !((screen->x_res>1280) && (var->bits_per_pixel == 32)))
+    if(((screen->x_res != var->xres) || (screen->y_res != var->yres)) 
+            && (screen->x_res<=1280))
     {
         addr = fix->mmio_start + par->y_offset* hdmi_get_fbscale()/100;
         xres_virtual = screen->x_res* hdmi_get_fbscale()/100;      //virtual screen size
@@ -1281,8 +1281,8 @@ static int win1_pan( struct fb_info *info )
     #ifdef CONFIG_FB_SCALING_OSD
     struct rk29fb_screen *screen = inf->cur_screen;
     struct fb_var_screeninfo *var = &info->var;
-    if(((screen->x_res != var->xres) || (screen->y_res != var->yres))
-        && !((screen->x_res>1280) && (var->bits_per_pixel == 32)))
+    if(((screen->x_res != var->xres) || (screen->y_res != var->yres)) 
+            && (screen->x_res<=1280))
     {
         addr = fix1->mmio_start + par->y_offset* hdmi_get_fbscale()/100;
     }
@@ -1445,8 +1445,8 @@ static int fb0_set_par(struct fb_info *info)
     if(inf->video_mode == 1)
     {
         #ifdef CONFIG_FB_SCALING_OSD
-        if(((screen->x_res != var->xres) || (screen->y_res != var->yres))
-        && !((screen->x_res>1280) && (var->bits_per_pixel == 32)))
+        if(((screen->x_res != var->xres) || (screen->y_res != var->yres)) 
+            && (screen->x_res<=1280))
         {
             par->xpos = 0;
             par->ypos = 0;
@@ -1543,8 +1543,8 @@ static int fb0_pan_display(struct fb_var_screeninfo *var, struct fb_info *info)
     if(inf->video_mode == 1)
     {
         #ifdef CONFIG_FB_SCALING_OSD
-        if(((screen->x_res != var->xres) || (screen->y_res != var->yres))
-        && !((screen->x_res>1280) && (var->bits_per_pixel == 32)))
+        if(((screen->x_res != var->xres) || (screen->y_res != var->yres)) 
+            && (screen->x_res<=1280))
         {
             par->y_offset = dstoffset;
 
-- 
2.35.3

