From dfc290d419d7044f1b7dc10cc6ac23260293e1cf Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 13 Sep 2012 14:50:57 +0800
Subject: [PATCH] fix bug in fb check var

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index d078e38b4f74..a1722de805c0 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -327,7 +327,7 @@ static int rk_fb_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 	 }
  
 	 if( ((var->xoffset+var->xres) > info->var.xres_virtual) ||
-	     ((var->yoffset+var->yres) > (info->var.yres_virtual*2)) )
+	     ((var->yoffset+var->yres) > (info->var.yres_virtual)) )
 	 {
 		 printk("%s check_var fail 2!!! \n",info->fix.id);
 		 printk("xoffset:%d>>xres:%d>>xres_vir:%d\n",var->xoffset,var->xres,info->var.xres_virtual);
-- 
2.35.3

