From d11238ad51fd172858986b3c264176efb23295c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 27 Jun 2011 13:57:32 +0800
Subject: [PATCH] rk29: Kconfig: support 1G memory

---
 arch/arm/mach-rk29/Kconfig | 23 +++++++++++++++++++++--
 1 file changed, 21 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk29/Kconfig b/arch/arm/mach-rk29/Kconfig
index 3ebe36311d86..eac7f56550a5 100755
--- a/arch/arm/mach-rk29/Kconfig
+++ b/arch/arm/mach-rk29/Kconfig
@@ -142,9 +142,28 @@ config DDR_TYPE_DDR3_DEFAULT
 
 endchoice
 
+choice RK29_MEM_SIZE
+	prompt "DDR Memory Size"
+	default RK29_MEM_SIZE_512M
+	help
+	  Select High Memory Support (from Kernel Features) for 1G memory.
+	  Deselect High Memory Support for 512MB memory.
+
+config RK29_MEM_SIZE_512M
+	bool "512MB"
+	depends on !HIGHMEM
+
+config RK29_MEM_SIZE_1G
+	bool "1GB"
+	depends on HIGHMEM
+	select ZONE_DMA
+
+endchoice
+
 config RK29_MEM_SIZE_M
-	int "DDR Memory size (in MB)"
-	default 512
+	int
+	default 512 if RK29_MEM_SIZE_512M
+	default 1024 if RK29_MEM_SIZE_1G
 
 config DDR_SDRAM_FREQ
 	int "DDR SDRAM frequence (in MHz)"
-- 
2.35.3

