From 3414cafedaf9aca394f9ba0a101c0a5f25e0b7c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 22 Dec 2010 11:40:16 +0800
Subject: [PATCH] rk29: clock: rm printk

---
 arch/arm/mach-rk29/clock.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 5a3eaa9fd21c..b049f345fe5d 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2330,7 +2330,6 @@ static int __init clk_disable_unused(void)
 		if (ck->usecount > 0 || ck->mode == NULL)
 			continue;
 
-printk("disable %s\n", ck->name);
 		LOCK();
 		clk_enable_nolock(ck);
 		clk_disable_nolock(ck);
-- 
2.35.3

