From d2beb3e459be1db600245267c671d263a6c071b2 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Fri, 4 Jul 2014 10:10:43 +0800
Subject: [PATCH] 3036: enble dpll slow mode at restart

---
 arch/arm/mach-rockchip/rk3036.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 97f5b7657ab1..767b2be24aa4 100644
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -159,7 +159,7 @@ static void rk3036_restart(char mode, const char *cmd)
 	dsb();
 
 	/* pll enter slow mode */
-	writel_relaxed(0x30010000, RK_CRU_VIRT + RK3036_CRU_MODE_CON);
+	writel_relaxed(0x30110000, RK_CRU_VIRT + RK3036_CRU_MODE_CON);
 	dsb();
 	writel_relaxed(0xeca8, RK_CRU_VIRT + RK3036_CRU_GLB_SRST_SND_VALUE);
 	dsb();
-- 
2.35.3

