From 9153f4506331688b94ece974a692a0f385e93ea9 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Wed, 11 Sep 2013 10:04:08 +0800
Subject: [PATCH] support ap6441 bt

---
 net/rfkill/rfkill-rk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/rfkill/rfkill-rk.c b/net/rfkill/rfkill-rk.c
index 5ede7421fecd..91d7e0182d24 100755
--- a/net/rfkill/rfkill-rk.c
+++ b/net/rfkill/rfkill-rk.c
@@ -129,6 +129,8 @@ static const char bt_name[] =
 		"ap6476"
 #elif defined(CONFIG_AP6493)
 		"ap6493"
+#elif defined(CONFIG_AP6441)
+        "ap6441"
 #else
         "bt_default"
 #endif
-- 
2.35.3

