From 8db1e9f2c392dd75b215f672fef516ee8421bfc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 23 Feb 2013 13:28:40 +0800
Subject: [PATCH] rk3066b: use rk3188 cpufreq driver

---
 arch/arm/mach-rk30/Makefile | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/Makefile b/arch/arm/mach-rk30/Makefile
index 53d7fba54459..17a2f1d17d32 100755
--- a/arch/arm/mach-rk30/Makefile
+++ b/arch/arm/mach-rk30/Makefile
@@ -9,7 +9,12 @@ CFLAGS_common.o += -DTEXT_OFFSET=$(TEXT_OFFSET)
 obj-y += ddr.o
 obj-y += devices.o
 obj-y += io.o
-obj-$(CONFIG_ARCH_RK30XX) += iomux.o
+ifeq ($(CONFIG_ARCH_RK30XX),y)
+obj-y += iomux.o
+obj-$(CONFIG_CPU_FREQ) += cpufreq.o
+else
+obj-$(CONFIG_CPU_FREQ) += ../mach-rk3188/cpufreq.o
+endif
 obj-y += pmu.o
 obj-y += reset.o
 obj-y += timer.o
@@ -19,7 +24,6 @@ obj-$(CONFIG_HOTPLUG_CPU) += hotplug.o
 obj-$(CONFIG_LOCAL_TIMERS) += localtimer.o
 obj-$(CONFIG_PM) += pm.o
 obj-$(CONFIG_CPU_IDLE) += cpuidle.o
-obj-$(CONFIG_CPU_FREQ) += cpufreq.o
 obj-$(CONFIG_DVFS) += dvfs.o
 obj-$(CONFIG_RK30_I2C_INSRAM) += i2c_sram.o
 
-- 
2.35.3

