From 474610f48c6e44f25cc297cc20f557c5298a09f7 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Thu, 9 Apr 2015 18:06:02 +0800
Subject: [PATCH] rk fb: when unplug hdmi,post fb to display

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index dc225b198e4a..9818c3a7367e 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3216,8 +3216,9 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 				(dev_drv->cur_screen->mode.xres << 8) +
 				(dev_drv->cur_screen->mode.yres << 20);
 			mutex_lock(&dev_drv->win_config);
-			/*info->fbops->fb_set_par(info);
-			info->fbops->fb_pan_display(&info->var, info);*/
+			info->var.yoffset = 0;
+			info->fbops->fb_set_par(info);
+			info->fbops->fb_pan_display(&info->var, info);
 			mutex_unlock(&dev_drv->win_config);
 
 			/*if (dev_drv->ops->dsp_black)
-- 
2.35.3

