From 1e836b3284ec564264c95f654132074400951c39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 12 Mar 2014 10:44:45 +0800
Subject: [PATCH] ARM: rockchip: fix debug/rockchip.S compile error

---
 arch/arm/include/debug/rockchip.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/debug/rockchip.S b/arch/arm/include/debug/rockchip.S
index ea312fa8a5a4..91a662f1d49b 100644
--- a/arch/arm/include/debug/rockchip.S
+++ b/arch/arm/include/debug/rockchip.S
@@ -12,7 +12,7 @@
  *
  */
 
-#include <../mach-rockchip/iomap.h>
+#include <linux/rockchip/iomap.h>
 #define UART_SHIFT 2
 
 	.macro addruart, rp, rv, tmp
-- 
2.35.3

