From a3a803e57cde1c05d08d0ab9e7d20639750474e7 Mon Sep 17 00:00:00 2001
From: xuhuicong <xhc@rock-chips.com>
Date: Thu, 8 Aug 2013 09:12:05 +0800
Subject: [PATCH] rk616: set mclk 11.2896m when unplug hdmi because mipi splash
 screen

---
 drivers/mfd/rk616-vif.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mfd/rk616-vif.c b/drivers/mfd/rk616-vif.c
index efbefdbc7b72..d6e0ed199584 100755
--- a/drivers/mfd/rk616-vif.c
+++ b/drivers/mfd/rk616-vif.c
@@ -782,6 +782,7 @@ int rk616_set_vif(struct mfd_rk616 *rk616,rk_screen *screen,bool connect)
 	{
 		rk616_vif_disable(rk616,0);
 		rk616_vif_disable(rk616,1);
+                rk616_mclk_set_rate(rk616->mclk, 11289600);
 		return 0;
 	}
 #if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)
-- 
2.35.3

