From 1bac773778e5fa6ea893c8fb30900a2c203de9b9 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 11 Apr 2013 09:38:56 +0800
Subject: [PATCH] rk screen:add SCREEN_TYPE SCREEN_MIPI

---
 include/linux/rk_screen.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index 84f538aa1b20..8e45f9284f44 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -44,12 +44,13 @@ Y	TX11   	G7		G1		GND		GND
 */
 
 typedef enum _SCREEN_TYPE {
-    SCREEN_NULL = 0,
-    SCREEN_RGB,
-    SCREEN_LVDS,
+	SCREEN_NULL = 0,
+	SCREEN_RGB,
+	SCREEN_LVDS,
 	SCREEN_MCU,
-    SCREEN_TVOUT,
-    SCREEN_HDMI,
+	SCREEN_TVOUT,
+	SCREEN_HDMI,
+	SCREEN_MIPI,
 } SCREEN_TYPE;
 
 typedef enum _REFRESH_STAGE {
-- 
2.35.3

