From 36e89e1637649e440a9b26d157cc3f50af969ff7 Mon Sep 17 00:00:00 2001
From: gwl <gwl@rock-chips.com>
Date: Tue, 1 Apr 2014 11:52:55 +0800
Subject: [PATCH] sdio:Wifi: tmp modify for wifi abnormal after suspend.

---
 drivers/mmc/core/sdio.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index adfabcef3437..dc2178c6615b 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -1022,7 +1022,12 @@ static int mmc_sdio_resume(struct mmc_host *host)
 	}
 
 	/* No need to reinitialize powered-resumed nonremovable cards */
-	if (mmc_card_is_removable(host) || !mmc_card_keep_power(host)) {
+    // tmp modify for wifi abnormal after suspend (gwl)
+    // mmc2: error -110 during resume (card was removed?)
+    // dpm_run_callback(): mmc_bus_resume+0x0/0x78 returns -110
+    // PM: Device mmc2:0001 failed to resume: error -110
+	if (!(host->restrict_caps & RESTRICT_CARD_TYPE_SDIO) &&
+           (mmc_card_is_removable(host) || !mmc_card_keep_power(host))) {
 		sdio_reset(host);
 		mmc_go_idle(host);
 		err = mmc_sdio_init_card(host, host->card->ocr, host->card,
-- 
2.35.3

