From a5201d21675683bb8b144781c57d6c6a0c2aaea2 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 30 Aug 2013 09:18:52 +0800
Subject: [PATCH] wifi:esp8089 use sdmmc suspend

---
 drivers/mmc/host/rk29_sdmmc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk29_sdmmc.h b/drivers/mmc/host/rk29_sdmmc.h
index 53853b95ecca..d7998672831d 100755
--- a/drivers/mmc/host/rk29_sdmmc.h
+++ b/drivers/mmc/host/rk29_sdmmc.h
@@ -378,7 +378,7 @@ static struct sdmmc_reg rk_sdmmc_regs[] =
 ** As the following example.
 ** added by xbw at 2013-05-08
 */
-#if defined(CONFIG_MTK_COMBO_DRIVER_VERSION_JB2)
+#if defined(CONFIG_MTK_COMBO_DRIVER_VERSION_JB2) || defined(CONFIG_ESP8089)
 #define RK_SDMMC_USE_SDIO_SUSPEND_RESUME    1
 #else
 #define RK_SDMMC_USE_SDIO_SUSPEND_RESUME    0
-- 
2.35.3

