From 24fcc8339e760cacd63dcd73edbe15d074e73c8f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 23 Oct 2012 10:08:59 +0800
Subject: [PATCH] rk30: enable ARM ERRATA 775420 workaround

---
 arch/arm/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 526c18dd37f3..7dd6b179c0e1 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -892,6 +892,7 @@ config ARCH_RK30
 	select MIGHT_HAVE_CACHE_L2X0
 	select ARM_ERRATA_764369
 	select ARM_ERRATA_754322
+	select ARM_ERRATA_775420
 	help
 	  Support for Rockchip's RK30xx/RK31xx SoCs.
 
-- 
2.35.3

