From ac12f21803ec04dd78061eccbb54b40d50ee4a9a Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Tue, 3 Mar 2015 21:09:54 +0800
Subject: [PATCH] mmc: rk_sdmmc: doesn't enforce hpclk to be used

Signed-off-by: lintao <lintao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 09d0e323f57e..6d5564734f91 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3878,10 +3878,9 @@ int dw_mci_probe(struct dw_mci *host)
 	host->hpclk_mmc= devm_clk_get(host->dev, "hpclk_mmc");
 	if (IS_ERR(host->hpclk_mmc)) {
 		dev_err(host->dev, "failed to get hpclk_mmc\n");
-		ret = PTR_ERR(host->hpclk_mmc);
-		goto err_hpclk_mmc;
+	} else {
+		clk_prepare_enable(host->hpclk_mmc);
 	}
-	clk_prepare_enable(host->hpclk_mmc);
 
         //hclk enable
         host->hclk_mmc= devm_clk_get(host->dev, "hclk_mmc");
@@ -4083,9 +4082,6 @@ int dw_mci_probe(struct dw_mci *host)
 err_hclk_mmc:
 	if (!IS_ERR(host->hclk_mmc))
 		clk_disable_unprepare(host->hclk_mmc);
-err_hpclk_mmc:
-	if (!IS_ERR(host->hpclk_mmc))
-		clk_disable_unprepare(host->hpclk_mmc);
 	return ret;
 }
 EXPORT_SYMBOL(dw_mci_probe);
-- 
2.35.3

