From bb9ffb80f07af3f3283e11c8b867981a1f8820df Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Thu, 31 Jul 2014 11:49:02 +0800
Subject: [PATCH] rk: ion: dts: add ion configure to rk312x dts file

---
 arch/arm/boot/dts/rk312x.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 0d685eb55f84..77ccb7189687 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -766,5 +766,18 @@
 			};
 		};
 	};
+	ion {
+		compatible = "rockchip,ion";
+		#address-cells = <1>;
+		#size-cells = <0>;
 
+		ion_cma: rockchip,ion-heap@1 { /* CMA HEAP */
+			compatible = "rockchip,ion-reserve";
+			rockchip,ion_heap = <1>;
+			reg = <0x00000000 0x10000000>; /* 256MB */
+		};
+		rockchip,ion-heap@3 { /* VMALLOC HEAP */
+			rockchip,ion_heap = <3>;
+		};
+	};
 };
-- 
2.35.3

