From 84115ec4579eda7b7702fedd7675fd07800db6de Mon Sep 17 00:00:00 2001
From: ZHW <zhw@rock-chips.com>
Date: Tue, 9 Jul 2013 14:25:14 +0800
Subject: [PATCH] rk_fb:fix CONFIG_DONE in rk_fb_disp_scaler

---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index b3538c1c557e..2c61788b04aa 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1093,6 +1093,7 @@ int rk_fb_disp_scale(u8 scale_x, u8 scale_y,u8 lcdc_id)
 	}
 
 	info->fbops->fb_set_par(info);
+	info->fbops->fb_ioctl(info,RK_FBIOSET_CONFIG_DONE,0);
 	return 0;
 	
 	
-- 
2.35.3

