From 6001d8c20351ec10c972944976a5dc1e08d43dab Mon Sep 17 00:00:00 2001
From: lhh <lhh@rock-chips.com>
Date: Fri, 19 Nov 2010 14:19:28 +0800
Subject: [PATCH] update stc3100 battery

---
 drivers/power/stc3100_battery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/stc3100_battery.c b/drivers/power/stc3100_battery.c
index 5abec0ef7404..fbb5c6ecaa26 100644
--- a/drivers/power/stc3100_battery.c
+++ b/drivers/power/stc3100_battery.c
@@ -146,7 +146,7 @@ static int stc3100_battery_rsoc(struct stc3100_device_info *di)
 		return ret;
 	}
 
-	return (((((int)regs[1])<<8) & (int)regs[0])*67/(10*RSENSE_RESISTANCE));    ////charge data (mA.h) = 6.70 * charge_code / Rsense (m¦¸).
+	return (((((int)regs[1])<<8) & (int)regs[0])*67/(10*RSENSE_RESISTANCE))*100/BATTAERY_CAPACITY_MAH;    ////charge data (mA.h) = 6.70 * charge_code / Rsense (m¦¸).
 }
 
 static int dc_charge_status(void)
-- 
2.35.3

