From 310a780b162c786f811a480e5daae8fabb28eb5a Mon Sep 17 00:00:00 2001
From: lhh <lhh@rock-chips.com>
Date: Fri, 3 Sep 2010 09:22:36 +0800
Subject: [PATCH] update rk2818_headset for cjq

---
 drivers/headset_observe/rk2818_headset.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/headset_observe/rk2818_headset.c b/drivers/headset_observe/rk2818_headset.c
index 8bbf1e0567e3..65d56512d848 100644
--- a/drivers/headset_observe/rk2818_headset.c
+++ b/drivers/headset_observe/rk2818_headset.c
@@ -65,7 +65,7 @@ static void headsetobserve_work(void)
 {
 	if(gpio_get_value(prk2818_headset_info->irq)){
 		if(prk2818_headset_info->headset_in_type)
-			Headset_dev.cur_headset_status = BIT_HEADSET;
+			{DBG("headset in-----cjq------/n");Headset_dev.cur_headset_status = BIT_HEADSET;}
 		else
 			Headset_dev.cur_headset_status = ~(BIT_HEADSET|BIT_HEADSET_NO_MIC);
 
@@ -74,7 +74,7 @@ static void headsetobserve_work(void)
 	}
 	else{
 		if(prk2818_headset_info->headset_in_type)
-			Headset_dev.cur_headset_status = ~(BIT_HEADSET|BIT_HEADSET_NO_MIC);
+			{DBG("headset out-----cjq------/n");Headset_dev.cur_headset_status = ~(BIT_HEADSET|BIT_HEADSET_NO_MIC);}
 		else
 			Headset_dev.cur_headset_status = BIT_HEADSET;
 
-- 
2.35.3

