From bace137c33c7a42b14087132507f69be1fcfc903 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Fri, 25 Mar 2011 17:56:45 +0800
Subject: [PATCH] update fb for cursor

---
 drivers/video/rk29_fb.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index c4a1a8e7d2c1..b29e887221a6 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -803,7 +803,6 @@ int rk29_set_cursor_colour_map(struct rk29fb_inf *inf, int bg_col, int fg_col)
 	printk("%s: bg = %08x,fg = %08x \n",__func__, bg_col, fg_col);
 	LcdMskReg(inf, HWC_COLOR_LUT0, m_HWC_R|m_HWC_G|m_HWC_B,bg_col);
 	LcdMskReg(inf, HWC_COLOR_LUT2, m_HWC_R|m_HWC_G|m_HWC_B,fg_col);
-	LcdSetBit(inf, SYS_CONFIG,m_HWC_RELOAD_EN);
 	LcdWrReg(inf, REG_CFG_DONE, 0x01);
     return 0;
 }
-- 
2.35.3

