From e6a80df673b8666b77ab612a00bd69a5dd14b036 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Sat, 1 Dec 2012 15:12:28 +0800
Subject: [PATCH] add comment for some key fucntion

---
 drivers/video/rockchip/rk_fb.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 1f0068a678e3..ddf628113c0b 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -657,6 +657,14 @@ void rk_direct_fb_show(struct fb_info * fbi)
 EXPORT_SYMBOL(rk_direct_fb_show);
 
 
+
+/******************************************
+function:this function will be called by hdmi,when 
+              hdmi plug in/out
+screen: the screen attached to hdmi
+enable: 1,hdmi plug in,0,hdmi plug out
+lcdc_id: the lcdc id the hdmi attached ,0 or 1
+******************************************/
 int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 {
 	struct rk_fb_inf *inf =  platform_get_drvdata(g_fb_pdev);
@@ -700,7 +708,7 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 	}
 	else if((lcdc_id == 1)&&(inf->num_lcdc == 2))
 	{
-		info = inf->fb[dev_drv->num_layer]; //the main fb of lcdc2
+		info = inf->fb[dev_drv->num_layer]; //the main fb of lcdc1
 	}
 
 	if(dev_drv->screen1) //device like rk2928 ,have only one lcdc but two outputs
@@ -814,6 +822,17 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 
 }
 
+
+
+
+/******************************************
+function:this function current only called by hdmi for 
+	scale the display
+scale_x: scale rate of x resolution
+scale_y: scale rate of y resolution
+lcdc_id: the lcdc id the hdmi attached ,0 or 1
+******************************************/
+
 int rk_fb_disp_scale(u8 scale_x, u8 scale_y,u8 lcdc_id)
 {
 	struct rk_fb_inf *inf =  platform_get_drvdata(g_fb_pdev);
-- 
2.35.3

