From 3503e5c80534a18b6e2c5d1f7b2d6dae68226cd4 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Fri, 8 Aug 2014 15:14:56 +0800
Subject: [PATCH] rk: fb: fix build error

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 22f9134ca121..8490f03ea4fb 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1351,7 +1351,7 @@ u32 freed_index;
 char buf[PAGE_SIZE];
 
 int rk_fb_sysmmu_fault_handler(struct device *dev,
-			       enum rk_sysmmu_inttype itype,
+			       enum rk_iommu_inttype itype,
 			       unsigned long pgtable_base,
 			       unsigned long fault_addr, unsigned int status)
 {
-- 
2.35.3

