From 7e47f8bd4bc11d7fe56111c2501a2fc9829e4228 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 18 Apr 2013 15:33:06 +0800
Subject: [PATCH] fix bug of rk_fb_set_par in hdmi mode

---
 drivers/video/rockchip/rk_fb.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 6a2cd20dd428..4fd861dd15e3 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -607,12 +607,18 @@ static int rk_fb_set_par(struct fb_info *info)
 						par2->xact = par->yact;
 						par2->yact = par->xact;
 						par2->xvir = par->yact;
+						info2->var.xres = var->yres;
+						info2->var.yres = var->xres;
+						info2->var.xres_virtual = var->yres;
 					}
 					else
 					{
 						par2->xact = par->xact;
 						par2->yact = par->yact;
 						par2->xvir = par->xvir;
+						info2->var.xres = var->xres;
+						info2->var.yres = var->yres;
+						info2->var.xres_virtual = var->xres_virtual;
 					}
 					par2->format = par->format;
 					info2->var.nonstd &= 0xffffff00;
-- 
2.35.3

