From 4b4be9d79abfd1b5150577fe1cc05ff9ea48f59f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 5 Apr 2012 20:46:24 +0800
Subject: [PATCH] arm: smp: skip calibrate_delay on rk platform when boot
 secondary cpu

---
 arch/arm/kernel/smp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/kernel/smp.c b/arch/arm/kernel/smp.c
index 9739bb8a2d28..2d9fb0a794c0 100644
--- a/arch/arm/kernel/smp.c
+++ b/arch/arm/kernel/smp.c
@@ -304,7 +304,9 @@ asmlinkage void __cpuinit secondary_start_kernel(void)
 
 	notify_cpu_starting(cpu);
 
+#ifndef CONFIG_PLAT_RK
 	calibrate_delay();
+#endif
 
 	smp_store_cpu_info(cpu);
 
-- 
2.35.3

