From 886605cffc8d9ac9bc25930391c9c74b013884f7 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Tue, 4 Mar 2014 16:03:55 +0800
Subject: [PATCH] rk: ion: support rk3288 fpga

---
 arch/arm/boot/dts/rk3288.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 8ec0495e4f5e..b1c0cafa027f 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -220,4 +220,15 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&spdif_tx>;
 	};
+	ion: ion{
+		compatible = "rockchip,ion";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		rockchip,ion-heap@1 { /* CMA HEAP */
+			reg = <1>;
+		};
+		rockchip,ion-heap@3 { /* SYSTEM HEAP */
+			reg = <3>;
+		};
+	};
 };
-- 
2.35.3

