From 55d7b213e03bcd4c330e4a0ba94a17005bdf5594 Mon Sep 17 00:00:00 2001
From: zwp <zwp@rock-chips.com>
Date: Tue, 26 Jun 2012 09:05:38 +0800
Subject: [PATCH] modify tdsc8800 bug.

---
 drivers/misc/tdsc8800.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/tdsc8800.c b/drivers/misc/tdsc8800.c
index fed23a14a257..6db48f071b76 100755
--- a/drivers/misc/tdsc8800.c
+++ b/drivers/misc/tdsc8800.c
@@ -65,6 +65,7 @@ static int tdsc8800_open(struct inode *inode, struct file *file)
 static int tdsc8800_release(struct inode *inode, struct file *file)
 {
 	MODEMDBG("tdsc8800_release\n");
+	modem_poweron_off(0);
 
 	return 0;
 }
@@ -124,7 +125,7 @@ static int tdsc8800_probe(struct platform_device *pdev)
 	
         gpio_direction_output(pdata->bp_power, GPIO_LOW);
 
-	gpio_set_value(pdata->bp_power, pdata->bp_reset_active_low? GPIO_LOW:GPIO_HIGH);
+	gpio_set_value(pdata->bp_power, pdata->bp_power_active_low? GPIO_LOW:GPIO_HIGH);
 	result = misc_register(&tdsc8800_misc);
 	if(result)
 	{
-- 
2.35.3

