From b70f43f91f70cd6fcbaf98a3c6a9f0a9ce383ea3 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Wed, 4 Mar 2015 10:37:27 +0800
Subject: [PATCH] solve make clean error causeed by wifi driver

---
 drivers/net/wireless/Makefile | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index ce75498e6466..fb61bb0007c9 100755
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -10,11 +10,6 @@ obj-$(CONFIG_RTL8723AU)	       += rockchip_wlan/rtl8723au/
 obj-$(CONFIG_RTL8723BU)	       += rockchip_wlan/rtl8723bu/
 obj-$(CONFIG_RTL8812AU)	       += rockchip_wlan/rtl8812au/
 obj-$(CONFIG_RKWIFI)           += rockchip_wlan/rkwifi/
-obj-$(CONFIG_RDA5990)          += rockchip_wlan/rda5990/
-obj-$(CONFIG_MT5931)           += rockchip_wlan/mt5931/
-obj-$(CONFIG_MT5931_MT6622)    += rockchip_wlan/mt5931/
 obj-$(CONFIG_RTL8723BS)        += rockchip_wlan/rtl8723bs/
 obj-$(CONFIG_RTL8723BS_VQ0)    += rockchip_wlan/rtl8723bs-vq0/
-obj-$(CONFIG_MTK_COMBO_WIFI)   += rockchip_wlan/combo_mt66xx/
-obj-$(CONFIG_MTK_MT5931)       += rockchip_wlan/mt5931_kk/
 obj-$(CONFIG_ESP8089)   += rockchip_wlan/esp8089/
-- 
2.35.3

