From ae0099f8000e11a19294ec92c97e945eea064f79 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Thu, 9 Sep 2010 18:22:48 -0700
Subject: [PATCH] spi:modify spi drivers, add wait_busy after write

---
 drivers/spi/rk2818_spim.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/rk2818_spim.c b/drivers/spi/rk2818_spim.c
index 3edc86a9ded7..8c47834b1784 100755
--- a/drivers/spi/rk2818_spim.c
+++ b/drivers/spi/rk2818_spim.c
@@ -799,14 +799,15 @@ static int do_transfer(struct rk2818_spi *dws)
 		if (dws->tx) {
 			do_write(dws);
 		}
+		wait_till_tf_empty(dws);
+		wait_till_not_busy(dws);
 		do_read(dws);
 	}
 	else {
 		do_write(dws);
+		wait_till_tf_empty(dws);
+		wait_till_not_busy(dws);
 	}
-
-	wait_till_tf_empty(dws);
-	wait_till_not_busy(dws);
 	
 	dws->cur_msg->actual_length += dws->len;
 	
-- 
2.35.3

