From 023280c1e40d02fc2248fc1d0ccb473b9d3db0ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 31 Jan 2013 11:23:57 +0800
Subject: [PATCH] rk30: no use Thumb instruction set

---
 arch/arm/mach-rk30/Makefile | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/mach-rk30/Makefile b/arch/arm/mach-rk30/Makefile
index 9835756a1a1a..10364157106e 100755
--- a/arch/arm/mach-rk30/Makefile
+++ b/arch/arm/mach-rk30/Makefile
@@ -7,7 +7,6 @@ endif
 obj-y += common.o
 CFLAGS_common.o += -DTEXT_OFFSET=$(TEXT_OFFSET)
 obj-y += ddr.o
-CFLAGS_ddr.o += -mthumb
 obj-y += devices.o
 obj-y += io.o
 obj-$(CONFIG_ARCH_RK30XX) += iomux.o
@@ -19,12 +18,10 @@ obj-$(CONFIG_SMP) += platsmp.o headsmp.o
 obj-$(CONFIG_HOTPLUG_CPU) += hotplug.o
 obj-$(CONFIG_LOCAL_TIMERS) += localtimer.o
 obj-$(CONFIG_PM) += pm.o
-CFLAGS_pm.o += -mthumb
 obj-$(CONFIG_CPU_IDLE) += cpuidle.o
 obj-$(CONFIG_CPU_FREQ) += cpufreq.o
 obj-$(CONFIG_DVFS) += dvfs.o
 obj-$(CONFIG_RK30_I2C_INSRAM) += i2c_sram.o
-CFLAGS_i2c_sram.o += -mthumb
 
 obj-$(CONFIG_MACH_RK3066_SDK) += board-rk30-sdk.o
 obj-$(CONFIG_MACH_RK30_SDK) += board-rk30-sdk.o
-- 
2.35.3

