From a6fd7f7fb5fb6a502caae27c4837b0403b97de3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Mon, 9 Jul 2012 10:01:02 +0800
Subject: [PATCH] rk30:sdk:change load order order

---
 drivers/power/rk29_charger_display.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/power/rk29_charger_display.c b/drivers/power/rk29_charger_display.c
index f4d5412d7929..c9e596aef524 100755
--- a/drivers/power/rk29_charger_display.c
+++ b/drivers/power/rk29_charger_display.c
@@ -106,6 +106,7 @@ static int  __init start_charge_logo_display(void)
 
 
 	//low power and charging
+#if 0
 	if((val_capacity.intval < pwr_on_thrsd )&&(val_status.intval == POWER_SUPPLY_STATUS_CHARGING))
 	{
 		while((val_capacity.intval < pwr_on_thrsd ))
@@ -119,7 +120,7 @@ static int  __init start_charge_logo_display(void)
 		}
 	}
 
-
+#endif
 
 	if(val_status.intval == POWER_SUPPLY_STATUS_CHARGING)
 	{
@@ -134,5 +135,5 @@ static int  __init start_charge_logo_display(void)
 	return 0;
 } 
 
-subsys_initcall_sync(start_charge_logo_display);
-
+//subsys_initcall_sync(start_charge_logo_display);
+module_init(start_charge_logo_display);
-- 
2.35.3

