From 87ec53df0f94a51afc2ff09dda0e629da2f1fe2b Mon Sep 17 00:00:00 2001
From: xbw <xbw@rock-chips.com>
Date: Tue, 9 Aug 2011 14:50:12 +0800
Subject: [PATCH] modify a small flaw in calling iomap function

---
 drivers/mmc/host/rk29_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk29_sdmmc.c b/drivers/mmc/host/rk29_sdmmc.c
index a4635fd95afa..1c70acd04d9a 100755
--- a/drivers/mmc/host/rk29_sdmmc.c
+++ b/drivers/mmc/host/rk29_sdmmc.c
@@ -2785,7 +2785,7 @@ static int rk29_sdmmc_probe(struct platform_device *pdev)
 	clk_enable(clk_get(&pdev->dev, "hclk_mmc"));
 
 	ret = -ENOMEM;
-	host->regs = ioremap(regs->start, regs->end - regs->start);
+	host->regs = ioremap(regs->start, regs->end - regs->start + 1);
 	if (!host->regs)
 	{
 	    host->errorstep = 0x8A;
-- 
2.35.3

