From ea01aa531e356fe0168e59706070fadf14e748fd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 9 Mar 2012 20:37:05 +0800
Subject: [PATCH] rk30: smp: fix boot_secondary with l2 cache

---
 arch/arm/mach-rk30/platsmp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/platsmp.c b/arch/arm/mach-rk30/platsmp.c
index ebcedb01f8b8..41752d902300 100644
--- a/arch/arm/mach-rk30/platsmp.c
+++ b/arch/arm/mach-rk30/platsmp.c
@@ -36,9 +36,10 @@ int __cpuinit boot_secondary(unsigned int cpu, struct task_struct *idle)
 	if (!copied) {
 		unsigned long sz = 0x100;
 
-		memcpy(RK30_SCU_BASE + sz - 4, (void *)rk30_sram_secondary_startup + sz - 4, 4);
+		memcpy(RK30_IMEM_BASE + sz - 4, (void *)rk30_sram_secondary_startup + sz - 4, 4);
 		memcpy(RK30_IMEM_BASE, rk30_sram_secondary_startup, sz);
 		flush_icache_range((unsigned long)RK30_IMEM_BASE, (unsigned long)RK30_IMEM_BASE + sz);
+		outer_clean_range(0, sz);
 		copied = true;
 	}
 
-- 
2.35.3

