From b93d89f3dd79ad415ac15dfde9f4f8ddf1d2ea7f Mon Sep 17 00:00:00 2001
From: ZHW <zhw@rock-chips.com>
Date: Tue, 8 Oct 2013 10:24:33 +0800
Subject: [PATCH] rk616: rgb screen no invert d_clk

---
 drivers/mfd/rk616-vif.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/rk616-vif.c b/drivers/mfd/rk616-vif.c
index ebc719066da1..848606c6d467 100755
--- a/drivers/mfd/rk616-vif.c
+++ b/drivers/mfd/rk616-vif.c
@@ -737,7 +737,10 @@ static int rk616_dither_cfg(struct mfd_rk616 *rk616,rk_screen *screen,bool enabl
 {
 	u32 val = 0;
 	int ret = 0;
-	val = FRC_DCLK_INV | (FRC_DCLK_INV << 16);
+
+	if(screen->type != SCREEN_RGB) //if RGB screen , not invert D_CLK
+		val = FRC_DCLK_INV | (FRC_DCLK_INV << 16);
+	
 	if((screen->face != OUT_P888) && enable)  //enable frc dither if the screen is not 24bit
 		val |= FRC_DITHER_EN | (FRC_DITHER_EN << 16);
 		//val |= (FRC_DITHER_EN << 16);
-- 
2.35.3

