From 206a8497a6e6275fe981a11e4330d35c9204fb17 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Wed, 7 May 2014 17:53:00 +0800
Subject: [PATCH] codec: mod err for soc-core .

---
 sound/soc/soc-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 015fe0b470ea..48d8c6e3a6db 100755
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -2120,7 +2120,7 @@ EXPORT_SYMBOL_GPL(snd_soc_read);
 unsigned int snd_soc_write(struct snd_soc_codec *codec,
 			   unsigned int reg, unsigned int val)
 {
-	dev_dbg(codec->dev, "write%x => %x\n", reg, ret);
+	dev_dbg(codec->dev, "write %x => %x\n", reg, val);
 	trace_snd_soc_reg_write(codec, reg, val);
 	return codec->write(codec, reg, val);
 }
-- 
2.35.3

