From 7b57aa1ef9207d78732a53556ad7d7633453e5f9 Mon Sep 17 00:00:00 2001
From: "Shengqin.Zhang" <zsq@rock-chips.com>
Date: Tue, 21 Jul 2015 14:44:46 +0800
Subject: [PATCH] fix rga_ioctl return error

Signed-off-by: Shengqin.Zhang <zsq@rock-chips.com>
---
 drivers/video/rockchip/rga2/rga2_drv.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/video/rockchip/rga2/rga2_drv.c b/drivers/video/rockchip/rga2/rga2_drv.c
index 041847d68c15..a3de7916655f 100644
--- a/drivers/video/rockchip/rga2/rga2_drv.c
+++ b/drivers/video/rockchip/rga2/rga2_drv.c
@@ -828,8 +828,6 @@ static long rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 	int ret = 0;
     rga2_session *session;
 
-    return ret;
-
     memset(&req, 0x0, sizeof(req));
 
     mutex_lock(&rga2_service.mutex);
-- 
2.35.3

