From 323e1a575efcc223329695767c30fa2a25c677d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 17 Jul 2012 14:30:05 +0800
Subject: [PATCH] pl330: fix support RK2928/RK31

---
 arch/arm/common/pl330.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/common/pl330.c b/arch/arm/common/pl330.c
index ecfc14c11215..7dc18b42d510 100644
--- a/arch/arm/common/pl330.c
+++ b/arch/arm/common/pl330.c
@@ -379,11 +379,11 @@ static inline u32 get_id(struct pl330_info *pi, u32 off)
 	void __iomem *regs = pi->base;
 	u32 id = 0;
 
-#ifdef CONFIG_ARCH_RK30
+#ifdef CONFIG_PLAT_RK
 	id |= ((readl(regs + off + 0x0) & 0xff) << 0);
-	id |= ((readl(regs + off + 0x4) & 0xff)<< 8);
+	id |= ((readl(regs + off + 0x4) & 0xff) << 8);
 	id |= ((readl(regs + off + 0x8) & 0xff) << 16);
-	id |= ((readl(regs + off + 0xc) & 0xff)<< 24);
+	id |= ((readl(regs + off + 0xc) & 0xff) << 24);
 #else
 	id |= (readb(regs + off + 0x0) << 0);
 	id |= (readb(regs + off + 0x4) << 8);
-- 
2.35.3

