From fb36c32ecb8613489ceb0302f49d08fd8c48c997 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Wed, 24 Jun 2015 09:59:25 +0800
Subject: [PATCH] usb: bc: fix function name in hader file

---
 include/linux/power/rk_usbbc.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/power/rk_usbbc.h b/include/linux/power/rk_usbbc.h
index e1d2b7221812..c2113d50d0cb 100644
--- a/include/linux/power/rk_usbbc.h
+++ b/include/linux/power/rk_usbbc.h
@@ -21,8 +21,8 @@ enum bc_port_type{
  * 3 : CDP - pc with big currect charge
  ************************************/
 extern int dwc_otg_check_dpdm(bool wait);
-extern int rk_bc_detect_register_notifier(struct notifier_block *nb,
+extern int rk_bc_detect_notifier_register(struct notifier_block *nb,
 					  enum bc_port_type *type);
-extern int rk_bc_detect_unregister_notifier(struct notifier_block *nb);
+extern int rk_bc_detect_notifier_unregister(struct notifier_block *nb);
 
 #endif
-- 
2.35.3

