From 7264875085d799932541daa194d0e9e0e291bde9 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Wed, 26 Sep 2012 16:52:00 +0800
Subject: [PATCH] rk2928: loader v1.08: usb mode enable(default usb debug mode)

---
 arch/arm/mach-rk2928/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/common.c b/arch/arm/mach-rk2928/common.c
index 46c68f8c2540..a7ece64b1c4e 100755
--- a/arch/arm/mach-rk2928/common.c
+++ b/arch/arm/mach-rk2928/common.c
@@ -128,7 +128,7 @@ void __init rk2928_map_io(void)
 #ifdef DEBUG_UART_BASE
         #ifdef CONFIG_RK_USB_UART
         writel_relaxed(0x007f0051, RK2928_GRF_BASE + GRF_UOC0_CON5);
-        writel_relaxed(0x30003000, RK2928_GRF_BASE + GRF_UOC1_CON4);
+        writel_relaxed(0x34003000, RK2928_GRF_BASE + GRF_UOC1_CON4);
         #endif
         writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
         writel_relaxed(0x07, DEBUG_UART_BASE + 0x88);
-- 
2.35.3

