From 9f3be04b2cad15a4e29f259cb80d42b316108598 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=83=AD=E6=AF=85?= <gy@rock-chips.com>
Date: Sat, 22 Mar 2014 11:53:31 +0800
Subject: [PATCH] sensor: fix free memery bug

---
 drivers/input/sensors/sensor-dev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/sensors/sensor-dev.c b/drivers/input/sensors/sensor-dev.c
index 5f93d9d6c89f..a02fa6b1c31a 100755
--- a/drivers/input/sensors/sensor-dev.c
+++ b/drivers/input/sensors/sensor-dev.c
@@ -2047,7 +2047,7 @@ int sensor_probe(struct i2c_client *client, const struct i2c_device_id *devid)
 out_input_register_device_failed:
 	input_free_device(sensor->input_dev);	
 out_free_memory:
-	kfree(sensor);
+	//kfree(sensor);
 out_no_free:
 	dev_err(&client->adapter->dev, "%s failed %d\n\n", __func__, result);
 	return result;
-- 
2.35.3

