From 24386413a2b0c30cb46069a16da5423a101334c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 16 Apr 2012 17:48:41 +0800
Subject: [PATCH] drivers/base/power/main.c: port rk29 change to plat rk

---
 drivers/base/power/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/base/power/main.c b/drivers/base/power/main.c
index 0a02634fb0f1..7687c5df8ff3 100644
--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -28,7 +28,7 @@
 #include <linux/async.h>
 #include <linux/suspend.h>
 #include <linux/timer.h>
-#ifdef CONFIG_ARCH_RK29
+#ifdef CONFIG_PLAT_RK
 #include <linux/console.h>
 #endif
 
@@ -611,7 +611,7 @@ static void dpm_drv_timeout(unsigned long data)
 	printk(KERN_EMERG "**** DPM device timeout: %s (%s)\n", dev_name(dev),
 	       (dev->driver ? dev->driver->name : "no driver"));
 
-#ifdef CONFIG_ARCH_RK29
+#ifdef CONFIG_PLAT_RK
 	resume_console();
 #endif
 	printk(KERN_EMERG "dpm suspend stack:\n");
-- 
2.35.3

