From 14cec1c89c09825c8b72f4f41bbb1ebc87c7f434 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 6 Jan 2014 10:35:20 +0800
Subject: [PATCH] rk: fix PIE support, libpie_stage2.o depends on
 pie_rename.syms

---
 pie/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pie/Makefile b/pie/Makefile
index bdd64a6dd470..bd90261704ce 100644
--- a/pie/Makefile
+++ b/pie/Makefile
@@ -41,7 +41,7 @@ $(obj)/libpie_stage1.o: $(KBUILD_LIBPIE)
 OBJCOPYFLAGS_libpie_stage2.o += --redefine-syms=$(obj)/pie_rename.syms
 OBJCOPYFLAGS_libpie_stage2.o += --rename-section .text=.pie.text
 
-$(obj)/libpie_stage2.o: $(obj)/libpie_stage1.o
+$(obj)/libpie_stage2.o: $(obj)/libpie_stage1.o $(obj)/pie_rename.syms
 	$(call if_changed,objcopy)
 
 # Generate a version of vmlinux.o with weakened and rename references to gcc
-- 
2.35.3

