From bdf9f95e32a449a44f1f1fd0e7a855a3750c4c2a Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Mon, 1 Jul 2013 16:03:39 +0800
Subject: [PATCH] wifi: rtl8723as add sdio_reset

---
 drivers/mmc/core/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index cc973ce3d993..193ed714012d 100755
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -1690,8 +1690,12 @@ static int mmc_rescan_try_freq(struct mmc_host *host, unsigned freq)
     {
         mmc_go_idle(host);
     }
+#else
+#if defined(CONFIG_RTL8723AS)
+    sdio_reset(host); //make no sense; noteed by xbw at 2011-12-14
 #else
     //sdio_reset(host); //make no sense; noteed by xbw at 2011-12-14
+#endif
 	mmc_go_idle(host);
 
 	if (!(init_ret=mmc_attach_sdio(host)))
-- 
2.35.3

