From 28c4147c01e4cad01b3142fb03f95a00a67668be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 25 Jan 2011 17:50:22 +0800
Subject: [PATCH] regulator: rk29-pwm: sleep 10ms instead of 5ms

---
 drivers/regulator/rk29-pwm-regulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/rk29-pwm-regulator.c b/drivers/regulator/rk29-pwm-regulator.c
index 8b998eb5e9b6..4091dc3f92a7 100644
--- a/drivers/regulator/rk29-pwm-regulator.c
+++ b/drivers/regulator/rk29-pwm-regulator.c
@@ -107,7 +107,7 @@ static int pwm_set_rate(struct pwm_platform_data *pdata,int nHz,u32 rate)
 	return -1;
 	}
 
-	msleep(5);
+	msleep(10);
 	
 	
     return (0);
-- 
2.35.3

