From 76511ba28026499f469708e14eb1d65960e9852b Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Sat, 13 Dec 2014 22:12:26 +0800
Subject: [PATCH] arm64: rockchip: rk3368 disable qos init temporarily

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368.dtsi b/arch/arm64/boot/dts/rk3368.dtsi
index 1f0b127121bf..a902a21205d0 100644
--- a/arch/arm64/boot/dts/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rk3368.dtsi
@@ -81,6 +81,7 @@
 			     <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;
 	};
 
+#if 0
 	cpu_axi_bus: cpu_axi_bus {
 		compatible = "rockchip,cpu_axi_bus";
 		#address-cells = <2>;
@@ -180,6 +181,7 @@
 			};
 		};
 	};
+#endif
 
 	timer {
 		compatible = "arm,armv8-timer";
-- 
2.35.3

