From 26e15446ab1ef81b23ffd1905a5865c372c5819a Mon Sep 17 00:00:00 2001
From: typ <typ@rock-chips.com>
Date: Thu, 1 Aug 2013 18:51:04 +0800
Subject: [PATCH] RK3028A/RK3026:change driving and odt strength setting

---
 arch/arm/mach-rk2928/ddr.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rk2928/ddr.c b/arch/arm/mach-rk2928/ddr.c
index 5e5b7311f569..36dd58e8a0d3 100755
--- a/arch/arm/mach-rk2928/ddr.c
+++ b/arch/arm/mach-rk2928/ddr.c
@@ -1876,12 +1876,12 @@ void __sramlocalfunc ddr_update_odt(void)
 #if defined (CONFIG_ARCH_RK3026)    //RK3028A   RK3026
     else
     {
-        pPHY_Reg->PHY_REG27 = ((PHY_RTT_215ohm<<3) | PHY_RTT_215ohm);       
-        pPHY_Reg->PHY_REG28 = ((PHY_RTT_215ohm<<3) | PHY_RTT_215ohm);    
+        pPHY_Reg->PHY_REG27 = ((PHY_RTT_215ohm<<4) | PHY_RTT_215ohm);       
+        pPHY_Reg->PHY_REG28 = ((PHY_RTT_215ohm<<4) | PHY_RTT_215ohm);    
         pPHY_Reg->PHY_REG0e4 = 0x0E;           //on DQS ODT default:0x0E
         pPHY_Reg->PHY_REG124 = 0x0E;           //on DQS ODT default:0x0E
     }
-    tmp = ((PHY_RON_45ohm<<3) | PHY_RON_45ohm);    
+    tmp = ((PHY_RON_45ohm<<4) | PHY_RON_45ohm);    
 #else                   //RK2928  R2926
     else
     {
-- 
2.35.3

