From 93f7f04203b1fa71f571a30a9990823f47e265f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 12 Mar 2013 22:27:07 +0800
Subject: [PATCH] rk: meminfo: show real totalram

---
 fs/proc/meminfo.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/fs/proc/meminfo.c b/fs/proc/meminfo.c
index a96282781f94..022ccc6ad4e1 100644
--- a/fs/proc/meminfo.c
+++ b/fs/proc/meminfo.c
@@ -14,6 +14,9 @@
 #include <asm/page.h>
 #include <asm/pgtable.h>
 #include "internal.h"
+#ifdef CONFIG_PLAT_RK
+#include <mach/ddr.h>
+#endif
 
 void __attribute__((weak)) arch_report_meminfo(struct seq_file *m)
 {
@@ -105,7 +108,15 @@ static int meminfo_proc_show(struct seq_file *m, void *v)
 		"AnonHugePages:  %8lu kB\n"
 #endif
 		,
+#ifdef CONFIG_PLAT_RK
+#ifdef CONFIG_RK29_MEM_SIZE_M
+		(unsigned long)CONFIG_RK29_MEM_SIZE_M * 1024,
+#else
+		(unsigned long)ddr_get_cap() >> 10,
+#endif
+#else
 		K(i.totalram),
+#endif
 		K(i.freeram),
 		K(i.bufferram),
 		K(cached),
-- 
2.35.3

