From ad7468eca8ca7e1cc247bcdf3232615cea89df0d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E8=BE=89?= <chenhui@rock-chips.com>
Date: Tue, 21 Dec 2010 09:59:04 +0800
Subject: [PATCH] modify usb lun number to 2, and vendor id to be HTC

---
 arch/arm/mach-rk29/devices.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rk29/devices.c b/arch/arm/mach-rk29/devices.c
index b4011611657b..6f5cb47f118b 100755
--- a/arch/arm/mach-rk29/devices.c
+++ b/arch/arm/mach-rk29/devices.c
@@ -636,8 +636,8 @@ static struct android_usb_product usb_products[] = {
  * please change vendor_id to 0x0bb4 and product_id to 0x0c02.
  */
 static struct android_usb_platform_data android_usb_pdata = {
-	.vendor_id	= 0x2207,//0x0bb4,//0x18d1,
-	.product_id	= 0x2810,//0x4e11,
+	.vendor_id	= 0x0bb4,//0x2207,//0x0bb4,//0x18d1,
+	.product_id	= 0x4e11,//0x2810,//0x4e11,
 	.version	= 0x0100,
 	.product_name		= "rk2818 sdk",
 	.manufacturer_name	= "RockChip",
@@ -658,7 +658,7 @@ struct platform_device android_usb_device = {
 
 /********************usb*********************/
 struct usb_mass_storage_platform_data mass_storage_pdata = {
-	.nluns		= 1,
+	.nluns		= 2,
 	.vendor		= "RockChip",
 	.product	= "rk29 sdk",
 	.release	= 0x0100,
-- 
2.35.3

