From f4a9049a56d137e99c924dac71c1814bd44e329c Mon Sep 17 00:00:00 2001
From: "Jony.Qiu" <jony.qiu@rock-chips.com>
Date: Mon, 15 Jun 2015 16:04:58 +0800
Subject: [PATCH] optee : optee linux build to .ko depend on
 function(sched_setaffinity)

Signed-off-by: Jony.Qiu <jony.qiu@rock-chips.com>
---
 kernel/sched/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index fbd6ccd99be3..33454d9756b8 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4289,6 +4289,7 @@ long sched_setaffinity(pid_t pid, const struct cpumask *in_mask)
 	put_online_cpus();
 	return retval;
 }
+EXPORT_SYMBOL(sched_setaffinity);
 
 static int get_user_cpu_mask(unsigned long __user *user_mask_ptr, unsigned len,
 			     struct cpumask *new_mask)
-- 
2.35.3

