From a448a9de9f8a59a21843afee15dafc8577679aa5 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 6 Jan 2011 14:52:48 +0800
Subject: [PATCH] i2c 'Sr' do not send 'STOP'

---
 drivers/i2c/busses/i2c-rk29.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-rk29.c b/drivers/i2c/busses/i2c-rk29.c
index d020ddda31d1..fdda20294820 100755
--- a/drivers/i2c/busses/i2c-rk29.c
+++ b/drivers/i2c/busses/i2c-rk29.c
@@ -469,10 +469,10 @@ static int rk29_xfer_msg(struct i2c_adapter *adap,
 	}
 	
 exit:	
-	//if(stop || ret < 0)
-	//{
+	if(stop || ret < 0)
+	{
 		rk29_i2c_stop(i2c);			
-	//}
+	}
 	return ret;
 
 }
-- 
2.35.3

