From e56e1214dcdd089699a45f0d27760a7fd0fc3c45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 27 May 2014 18:36:21 +0800
Subject: [PATCH] ARM: PIE: empty.S add __gnu_mcount_nc, fix for function
 tracer

---
 arch/arm/libpie/empty.S | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/libpie/empty.S b/arch/arm/libpie/empty.S
index 241686206d3a..f73e3e7b2861 100644
--- a/arch/arm/libpie/empty.S
+++ b/arch/arm/libpie/empty.S
@@ -1,11 +1,13 @@
 #include <linux/linkage.h>
 
 ENTRY(__div0)
+ENTRY(__gnu_mcount_nc)
 ENTRY(__aeabi_unwind_cpp_pr0)
 ENTRY(__aeabi_unwind_cpp_pr1)
 ENTRY(__aeabi_unwind_cpp_pr2)
 	mov	pc, lr
 ENDPROC(__div0)
+ENDPROC(__gnu_mcount_nc)
 ENDPROC(__aeabi_unwind_cpp_pr0)
 ENDPROC(__aeabi_unwind_cpp_pr1)
 ENDPROC(__aeabi_unwind_cpp_pr2)
-- 
2.35.3

