From 6f36aed2274ec360dba7ec351acdf20a63325fc6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=83=AD=E6=AF=85?= <gy@rock-chips.com>
Date: Fri, 18 Jul 2014 09:53:15 +0800
Subject: [PATCH] cw201x: shielding unnecessary dev_info

---
 drivers/power/cw2015_battery.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/power/cw2015_battery.c b/drivers/power/cw2015_battery.c
index 6d284121a163..47cd9bd6024b 100755
--- a/drivers/power/cw2015_battery.c
+++ b/drivers/power/cw2015_battery.c
@@ -198,8 +198,7 @@ static int cw_update_config_info(struct cw_battery *cw_bat)
 
         /* update new battery info */
         for (i = 0; i < SIZE_BATINFO; i++) {
-                dev_info(&cw_bat->client->dev, "cw_bat->plat_data.cw_bat_config_info[%d] = 0x%x\n", i, \
-                                cw_bat->plat_data.cw_bat_config_info[i]);
+           //     dev_info(&cw_bat->client->dev, "cw_bat->plat_data.cw_bat_config_info[%d] = 0x%x\n", i, cw_bat->plat_data.cw_bat_config_info[i]);
                 ret = cw_write(cw_bat->client, REG_BATINFO + i, (u8 *)&cw_bat->plat_data.cw_bat_config_info[i]);
 
                 if (ret < 0) 
-- 
2.35.3

