From a197e87b572539516b14adb68467b8ed6e15447e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 9 Oct 2012 18:27:13 +0800
Subject: [PATCH] rk: config: enlarge COMMAND_LINE_SIZE from 1024 to 65536

---
 arch/arm/include/asm/setup.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/include/asm/setup.h b/arch/arm/include/asm/setup.h
index ee2ad8ae07af..3f7ff7972e3c 100644
--- a/arch/arm/include/asm/setup.h
+++ b/arch/arm/include/asm/setup.h
@@ -16,7 +16,11 @@
 
 #include <linux/types.h>
 
+#ifdef CONFIG_RK_CONFIG
+#define COMMAND_LINE_SIZE 65536
+#else
 #define COMMAND_LINE_SIZE 1024
+#endif
 
 /* The list ends with an ATAG_NONE node. */
 #define ATAG_NONE	0x00000000
-- 
2.35.3

