From 2efcc1342f98890c91f7858fe8e1f32a09110b61 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Thu, 6 Nov 2014 15:46:04 +0800
Subject: [PATCH] rk3036: fix system halt when suspend

---
 arch/arm/mach-rockchip/rk3036.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index 99fec2de39fd..e0f0a25bd1b5 100755
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -583,13 +583,10 @@ void PIE_FUNC(pwm_regulator_suspend)(void)
 	}
 
 	if (rkpm_chk_sram_ctrbit(RKPM_CTR_VOL_PWM2)) {
-		FUNC(sram_printch)('p');
-		FUNC(sram_printch)('o');
-		FUNC(sram_printch)('l');
-	rkpm_pwm_duty2 = readl_relaxed(RK_PWM_VIRT + 0x28);
-	writel_relaxed(PWM_VOLTAGE, RK_PWM_VIRT + 0x28);
+		rkpm_pwm_duty2 = readl_relaxed(RK_PWM_VIRT + 0x28);
+		writel_relaxed(PWM_VOLTAGE, RK_PWM_VIRT + 0x28);
 	}
-	rkpm_udelay(30);
+	//rkpm_udelay(30);
 }
 
 void PIE_FUNC(pwm_regulator_resume)(void)
-- 
2.35.3

