From 4f56d18f832b175c6cb09d04e7f053aa70eb4083 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 11 Jun 2010 04:45:07 +0000
Subject: [PATCH] add KERN_DEBUG

---
 arch/arm/mach-rk2818/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2818/cpufreq.c b/arch/arm/mach-rk2818/cpufreq.c
index 60caf3dff345..dbed9f9e52cf 100755
--- a/arch/arm/mach-rk2818/cpufreq.c
+++ b/arch/arm/mach-rk2818/cpufreq.c
@@ -42,7 +42,7 @@ static int rk2818_cpufreq_target(struct cpufreq_policy *policy, unsigned int tar
 		return 0;
 
 #ifdef CONFIG_CPU_FREQ_DEBUG
-	printk("%s %d r %d (%d-%d) selected %d\n", __func__, target_freq, relation, policy->min, policy->max, freq_table[index].frequency);
+	printk(KERN_DEBUG "%s %d r %d (%d-%d) selected %d\n", __func__, target_freq, relation, policy->min, policy->max, freq_table[index].frequency);
 #endif
 	freqs.old = policy->cur;
 	freqs.new = freq_table[index].frequency;
-- 
2.35.3

