From 2bb4bce1846948ae471d45570fae5805e3ae6c58 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Wed, 28 Jul 2010 02:39:17 -0700
Subject: [PATCH] modify dm9000 interrupt flags

---
 drivers/net/dm9000.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/dm9000.c b/drivers/net/dm9000.c
index 5044b51fac67..509493d3c41d 100755
--- a/drivers/net/dm9000.c
+++ b/drivers/net/dm9000.c
@@ -1228,8 +1228,13 @@ dm9000_open(struct net_device *dev)
 
 	irqflags |= IRQF_SHARED;
 
-	if (request_irq(dev->irq, &dm9000_interrupt, irqflags, dev->name, dev))
+	#ifndef CONFIG_MACH_RK2818MID
+	if (request_irq(dev->irq, dm9000_interrupt, IRQF_TRIGGER_HIGH, dev->name, dev))
 		return -EAGAIN;
+	#else
+	if (request_irq(dev->irq, dm9000_interrupt, irqflags, dev->name, dev))
+		return -EAGAIN;
+	#endif
 
 	/* Initialize DM9000 board */
 	dm9000_reset(db);
-- 
2.35.3

