From 101aadaefedc91d1a793b21328ab2c8a77a9e129 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Thu, 16 Aug 2012 19:20:03 +0800
Subject: [PATCH] 3G: Support for more 3g devices

---
 drivers/usb/serial/option.c | 30 +++++++++++++++++++++++++++++-
 1 file changed, 29 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 8e8430342230..ab31c05b731a 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -1108,13 +1108,41 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(ZTE_VENDOR_ID, 0xFFFF) },
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x5700) },
 	{ USB_DEVICE(LEADCORE_VENDOR_ID, 0x6341) },
-	{ USB_DEVICE(0x230d, 0x000d) },
+	{ USB_DEVICE(0x230D, 0x000D) },
 	{ USB_DEVICE(0x0E8D, 0x00A2) },
 	{ USB_DEVICE(0x1E89, 0x1A20) },
 	{ USB_DEVICE(0x12D1, 0x1C05) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x0007, 0xff, 0xff, 0xff) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x0167, 0xff, 0xff, 0xff) },
+	{ USB_DEVICE(HUAWEI_VENDOR_ID, 0x14A8) },
 	{ USB_DEVICE(0x20A6, 0x1105) },
+	{ USB_DEVICE(0x1EE8, 0x005F) }, // Onda-MSA14.4
+	{ USB_DEVICE(0x0421, 0x061E) }, // Nokia CS-11
+	{ USB_DEVICE(0x0BDB, 0x190A) },
+	{ USB_DEVICE(0x1d09, 0x1010) },
+    { USB_DEVICE(0x19D2, 0x1181) },
+	{ USB_DEVICE(0x2020, 0x1005)},//S830 3G Dongle
+// cmy end
+//xxh
+   { USB_DEVICE_AND_INTERFACE_INFO(0x05c6, 0x1000, 0xff, 0xff, 0xff) },
+   { USB_DEVICE_AND_INTERFACE_INFO(0x05c6, 0x6000, 0xff, 0xff, 0xff) },
+   { USB_DEVICE_AND_INTERFACE_INFO(0x20A6, 0xF00E, 0xff, 0xff, 0xff) },
+   { USB_DEVICE_AND_INTERFACE_INFO(0x20A6, 0x1105, 0xff, 0xff, 0xff) },
+   { USB_DEVICE(0x21f5, 0x1101)},
+   { USB_DEVICE(0x230d,0x0101)},
+   { USB_DEVICE_AND_INTERFACE_INFO(0x230d, 0x1101, 0xff, 0xff, 0xff) },
+   { USB_DEVICE(0x2077,0xa000)},
+   { USB_DEVICE(0x1c9e,0x6061)},
+   { USB_DEVICE(0x1c9e,0x9605)},
+   { USB_DEVICE(0x19d2, 0x0170) },
+   { USB_DEVICE(0x19d2, 0xffe8) },
+   { USB_DEVICE(0x19D2,0x1177) },//K3770-Z
+   { USB_DEVICE(0x230D, 0x000c) },
+   { USB_DEVICE(0x21F5, 0x2009) },
+   { USB_DEVICE(HUAWEI_VENDOR_ID, HUAWEI_PRODUCT_E353) },
+   { USB_DEVICE(0x15eb,0x0001)},
+//xxh end
+
 
 	{ USB_DEVICE(OLIVETTI_VENDOR_ID, OLIVETTI_PRODUCT_OLICARD100) },
 	{ USB_DEVICE(CELOT_VENDOR_ID, CELOT_PRODUCT_CT680M) }, /* CT-650 CDMA 450 1xEVDO modem */
-- 
2.35.3

