From b9eb3fa8bb0980e3a736bfebcd5715e83de6487c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 10 Jul 2012 13:13:32 +0800
Subject: [PATCH] rk2928: enable RK_PL330_DMA config

---
 arch/arm/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 873f0174c837..fe16128b67b5 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -876,7 +876,7 @@ config ARCH_RK2928
 	select PLAT_RK
 	select CPU_V7
 	select ARM_GIC
-	select PL330
+	select RK_PL330_DMA
 	select MIGHT_HAVE_CACHE_L2X0
 	select ARM_ERRATA_754322
 	help
-- 
2.35.3

