From d632c24a4222f46255c4d235f2b74a3170432db6 Mon Sep 17 00:00:00 2001
From: eddie <cf@rock-chips.com>
Date: Wed, 24 Aug 2011 17:48:26 +0800
Subject: [PATCH] newton: remove backlight default

---
 drivers/video/backlight/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/backlight/Kconfig b/drivers/video/backlight/Kconfig
index 28fda05ef926..f284fd52fc90 100644
--- a/drivers/video/backlight/Kconfig
+++ b/drivers/video/backlight/Kconfig
@@ -279,7 +279,6 @@ config BACKLIGHT_RK29_BL
 config BACKLIGHT_RK29_NEWTON_BL
         bool "rk29 newton backlight driver"
 	depends on BACKLIGHT_CLASS_DEVICE && ARCH_RK29
-	default y
 	help
 	  rk29 newton backlight support.	
 
-- 
2.35.3

