From d735b9be474a35051b18bd32a498b12acc62fd02 Mon Sep 17 00:00:00 2001
From: yangyun <yangyun@rock-chips.com>
Date: Fri, 9 Mar 2012 16:27:07 +0800
Subject: [PATCH] alter pack-kernel.sh

---
 pack-kernel.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 9df543470c8b..36f0e9d8c8cf 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -57,9 +57,6 @@ arch/arm/configs/rk29_newton_defconfig
 arch/arm/mach-rk29/board-rk29-p91*
 arch/arm/configs/rk29_p91_defconfig
 
-arch/arm/mach-rk29/board-rk29phonepadsdk*
-arch/arm/configs/rk29_phonepadsdk_defconfig
-
 arch/arm/mach-rk29/board-rk29-phonesdk*
 arch/arm/configs/rk29_phonesdk_defconfig
 
-- 
2.35.3

