From b5805fd10aea82e4974ca173099062a9ac14b225 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 2 Jul 2011 12:05:05 +0800
Subject: [PATCH] rk29: disable print_modules

---
 kernel/module.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/module.c b/kernel/module.c
index 4b270e666212..eb1b9a77ebfb 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -3093,6 +3093,7 @@ EXPORT_SYMBOL_GPL(__module_text_address);
 /* Don't grab lock, we're oopsing. */
 void print_modules(void)
 {
+#ifndef CONFIG_ARCH_RK29
 	struct module *mod;
 	char buf[8];
 
@@ -3105,6 +3106,7 @@ void print_modules(void)
 	if (last_unloaded_module[0])
 		printk(" [last unloaded: %s]", last_unloaded_module);
 	printk("\n");
+#endif
 }
 
 #ifdef CONFIG_MODVERSIONS
-- 
2.35.3

