From cd75fbf90112e7feaeadcf1c1df938d2f25baa1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Fri, 5 Dec 2014 19:05:59 +0800
Subject: [PATCH] RK3126B: RK3126B delete the EBC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 许盛飞 <xsf@rock-chips.com>
---
 arch/arm/mach-rockchip/rk312x.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk312x.c b/arch/arm/mach-rockchip/rk312x.c
index 452bdc0dbd31..6ff7f52c9548 100755
--- a/arch/arm/mach-rockchip/rk312x.c
+++ b/arch/arm/mach-rockchip/rk312x.c
@@ -39,6 +39,7 @@
 #include "sram.h"
 #include "pm.h"
 #include "pm-rk312x.c"
+#include <linux/rockchip/cpu.h>
 #define RK312X_DEVICE(name) \
 	{ \
 		.virtual	= (unsigned long) RK_##name##_VIRT, \
@@ -256,7 +257,8 @@ static int rk312x_pmu_set_power_domain(enum pmu_power_domain pd, bool on)
 			rk312x_pmu_set_idle_request(IDLE_REQ_GPU, true);
 		} else if (pd == PD_VIO) {
 			SAVE_QOS(rga_qos, VIO_RGA);
-			SAVE_QOS(ebc_qos, VIO_EBC);
+			if (!soc_is_rk3126b())
+				SAVE_QOS(ebc_qos, VIO_EBC);
 			SAVE_QOS(iep_qos, VIO_IEP);
 			SAVE_QOS(lcdc0_qos, VIO_LCDC0);
 			SAVE_QOS(vip0_qos, VIO_VIP0);
@@ -276,7 +278,8 @@ static int rk312x_pmu_set_power_domain(enum pmu_power_domain pd, bool on)
 		} else if (pd == PD_VIO) {
 			rk312x_pmu_set_idle_request(IDLE_REQ_VIO, false);
 			RESTORE_QOS(rga_qos, VIO_RGA);
-			RESTORE_QOS(ebc_qos, VIO_EBC);
+			if (!soc_is_rk3126b())
+				RESTORE_QOS(ebc_qos, VIO_EBC);
 			RESTORE_QOS(iep_qos, VIO_IEP);
 			RESTORE_QOS(lcdc0_qos, VIO_LCDC0);
 			RESTORE_QOS(vip0_qos, VIO_VIP0);
-- 
2.35.3

