From c791a3498d3d1dbe57b9068f1ad836728c1bab59 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 14 May 2010 01:31:26 +0000
Subject: [PATCH] add rk2818_sdmmc.* and rk2818_i2c.*

---
 arch/arm/mach-rk2818/clock.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index 4fa5b845a153..37f291881305 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -908,14 +908,14 @@ static struct clk_lookup clks[] = {
 	CLK1(lcdc),
 	CLK1(vip),
 	CLK1(i2s),
-	CLK1(sdmmc0),
+	CLK("rk2818_sdmmc.0", "sdmmc", &sdmmc0_clk),
 	CLK1(ebrom),
 	CLK1(gpio0),
 	CLK1(gpio1),
 	CLK("rk2818_serial.0", "uart", &uart0_clk),
 	CLK("rk2818_serial.1", "uart", &uart1_clk),
-	CLK1(i2c0),
-	CLK1(i2c1),
+	CLK("rk2818_i2c.0", "i2c", &i2c0_clk),
+	CLK("rk2818_i2c.1", "i2c", &i2c1_clk),
 	CLK("rk2818_spim.0", "spi", &spi0_clk),
 	CLK1(spi1),
 	CLK1(pwm),
@@ -925,7 +925,7 @@ static struct clk_lookup clks[] = {
 	CLK1(lsadc),
 	CLK("rk2818_serial.2", "uart", &uart2_clk),
 	CLK("rk2818_serial.3", "uart", &uart3_clk),
-	CLK1(sdmmc1),
+	CLK("rk2818_sdmmc.1", "sdmmc", &sdmmc1_clk),
 
 	CLK1(hsadc),
 	CLK1(sdram_common),
-- 
2.35.3

