From 7491c4b0417ab0236e23e7e007910145d741cd69 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Mon, 26 Aug 2013 10:59:08 +0800
Subject: [PATCH] RK3026: when suspend ,disable  USB UART Bypass Function

---
 arch/arm/mach-rk2928/pm.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/pm.c b/arch/arm/mach-rk2928/pm.c
index 4cf0a5adb7f3..b3447d55393b 100755
--- a/arch/arm/mach-rk2928/pm.c
+++ b/arch/arm/mach-rk2928/pm.c
@@ -605,12 +605,17 @@ static void __sramfunc rk_pm_soc_sram_volt_resume(void)
 }
 static void __sramfunc rk2928_sram_suspend(void)
 {
-
+	int grf_uoc1_con0;
 	sram_printch('5');
 	RK_SOC_PM_CTR_FUN(NO_DDR, ddr_suspend);
 	sram_printch('6');
 	RK_SOC_PM_CTR_FUN(NO_VOLT, rk_pm_soc_sram_volt_suspend);
 	sram_printch('7');
+#ifdef CONFIG_ARCH_RK3026
+       grf_uoc1_con0 = grf_readl(GRF_UOC1_CON0);
+       grf_writel(0x30000000, GRF_UOC1_CON0);
+#endif
+	
 	RK_SOC_PM_CTR_FUN(NO_CLK_GATING, rk_pm_soc_sram_clk_gating);
 	RK_SOC_PM_CTR_FUN(NO_SYS_CLK, rk_pm_soc_sram_sys_clk_suspend);
 
@@ -623,6 +628,9 @@ static void __sramfunc rk2928_sram_suspend(void)
 	//RK_SOC_PM_CTR_FUN(NO_PMIC,board_pmu_resume);
 	RK_SOC_PM_CTR_FUN(NO_SYS_CLK, rk_pm_soc_sram_sys_clk_resume);
 	RK_SOC_PM_CTR_FUN(NO_CLK_GATING, rk_pm_soc_sram_clk_ungating);
+#ifdef CONFIG_ARCH_RK3026
+       grf_writel(0x30000000 | grf_uoc1_con0, GRF_UOC1_CON0);
+#endif	
 	sram_printch('7');
 	RK_SOC_PM_CTR_FUN(NO_VOLT, rk_pm_soc_sram_volt_resume);
 	sram_printch('6');
-- 
2.35.3

