From edf82de8716f70a8ad537102d9b6f71e44d5f22f Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Fri, 6 Mar 2015 11:16:01 +0800
Subject: [PATCH] rockchip: i2s: fix i2s mclk set fail when using multi i2s.

Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 sound/soc/rockchip/rk30_i2s.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk30_i2s.c b/sound/soc/rockchip/rk30_i2s.c
index 55aeefeb31ca..bb15f42b3250 100755
--- a/sound/soc/rockchip/rk30_i2s.c
+++ b/sound/soc/rockchip/rk30_i2s.c
@@ -514,7 +514,9 @@ static int rockchip_i2s_resume_noirq(struct device *dev)
 #ifdef CLK_SET_lATER
 static void set_clk_later_work(struct work_struct *work)
 {
-	struct rk30_i2s_info *i2s = rk30_i2s;
+	struct rk30_i2s_info *i2s = container_of(work, struct rk30_i2s_info,
+						 clk_delayed_work.work);
+
 	clk_set_rate(i2s->i2s_clk, 11289600);
 	if(!IS_ERR(i2s->i2s_mclk) )
 		clk_set_rate(i2s->i2s_mclk, 11289600);
-- 
2.35.3

