From e5db25f38ea1b29ffada5bca0c78d325cd093be6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:22:16 +0800
Subject: [PATCH] eeti_egalax_i2c: fix for v2.6.36

---
 drivers/input/touchscreen/eeti_egalax_i2c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/eeti_egalax_i2c.c b/drivers/input/touchscreen/eeti_egalax_i2c.c
index 026c18bfe5df..961c245fea2b 100755
--- a/drivers/input/touchscreen/eeti_egalax_i2c.c
+++ b/drivers/input/touchscreen/eeti_egalax_i2c.c
@@ -730,7 +730,7 @@ static const struct file_operations egalax_cdev_fops = {
 	.owner	= THIS_MODULE,
 	.read	= egalax_cdev_read,
 	.write	= egalax_cdev_write,
-	.ioctl	= egalax_cdev_ioctl,
+	.unlocked_ioctl	= egalax_cdev_ioctl,
 	.poll	= egalax_cdev_poll,
 	.open	= egalax_cdev_open,
 	.release= egalax_cdev_release,
-- 
2.35.3

