From 9f5be7c2a4e38edb31c2e86387d823d6d9c4a35c Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Mon, 10 Sep 2012 14:41:35 +0800
Subject: [PATCH] rkfb: support dual display use rk610 as scaler

---
 drivers/video/rockchip/rk_fb.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 6a428b3a2866..c192a8e4d777 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -658,7 +658,15 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 			dev_drv->screen1->screen_id = 1;
 			dev_drv->screen0->lcdc_id = 1; //connect screen0 to output interface 1
 			dev_drv->cur_screen = dev_drv->screen1;
-			dev_drv->screen0->sscreen_get(dev_drv->screen0,dev_drv->cur_screen->hdmi_resolution);
+			if(dev_drv->screen0->sscreen_get)
+			{
+				dev_drv->screen0->sscreen_get(dev_drv->screen0,
+					dev_drv->cur_screen->hdmi_resolution);
+			}
+			if(dev_drv->screen0->sscreen_set)
+			{
+				dev_drv->screen0->sscreen_set(dev_drv->screen0,enable);
+			}
 			
 		}
 		else
-- 
2.35.3

