From 8f3991d3fb079b7d0ef00c7a153d8759a371192b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=98=8A?= <zhanghao@rock-chips.com>
Date: Tue, 22 Nov 2011 12:10:39 +0800
Subject: [PATCH] wm831x : fix 831x init bug for td8801

---
 drivers/mfd/wm831x-i2c.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/wm831x-i2c.c b/drivers/mfd/wm831x-i2c.c
index 3d109a52cd56..9c840dcf1f04 100755
--- a/drivers/mfd/wm831x-i2c.c
+++ b/drivers/mfd/wm831x-i2c.c
@@ -180,7 +180,8 @@ static int __init wm831x_i2c_init(void)
 	return ret;
 }
 //subsys_initcall(wm831x_i2c_init);
-fs_initcall(wm831x_i2c_init);
+//fs_initcall(wm831x_i2c_init);
+subsys_initcall_sync(wm831x_i2c_init);
 static void __exit wm831x_i2c_exit(void)
 {
 	i2c_del_driver(&wm831x_i2c_driver);
-- 
2.35.3

