From d9479fe5d6b897cf980398e9537f8f67b8b495fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 30 Aug 2010 12:14:25 +0800
Subject: [PATCH] rk2818_fb: use codec pll

---
 drivers/video/rk2818_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk2818_fb.c b/drivers/video/rk2818_fb.c
index fa16dab75d88..21d79349177c 100755
--- a/drivers/video/rk2818_fb.c
+++ b/drivers/video/rk2818_fb.c
@@ -2309,7 +2309,7 @@ static int __init rk2818fb_probe (struct platform_device *pdev)
 		ret = -ENOENT;
 		goto unregister_win1fb;
 	}
-    inf->dclk_parent = clk_get(&pdev->dev, "arm_pll");
+    inf->dclk_parent = clk_get(&pdev->dev, "codec_pll");
     if (!inf->dclk_parent || IS_ERR(inf->dclk_parent))
     {
 		printk(KERN_ERR "failed to get lcd dclock parent source\n");
-- 
2.35.3

