From 7c557888c9f4f3edecbc5e0c6f22dab52115de0a Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Fri, 16 Sep 2011 16:57:35 +0800
Subject: [PATCH] enable ipp for scale UI when support HDMI

---
 drivers/video/hdmi/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/hdmi/Kconfig b/drivers/video/hdmi/Kconfig
index 7d2cd0aefcce..2b01022ee40e 100755
--- a/drivers/video/hdmi/Kconfig
+++ b/drivers/video/hdmi/Kconfig
@@ -4,7 +4,8 @@
 menu "HDMI"
 config HDMI
 	bool "HDMI support"
-
+    select FB_WORK_IPP
+	select FB_SCALING_OSD
 if HDMI
 source "drivers/video/hdmi/chips/Kconfig"
 
-- 
2.35.3

