From 16ab5272c9d26cdbbc3b79a2bc6bffee8af715f4 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 29 May 2013 15:53:47 +0800
Subject: [PATCH] mfd:rk616:lvds:set lcd1 port as input in suspend

---
 drivers/video/display/transmitter/rk616_lvds.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/display/transmitter/rk616_lvds.c b/drivers/video/display/transmitter/rk616_lvds.c
index 78d38eadbcd9..fd1339481755 100644
--- a/drivers/video/display/transmitter/rk616_lvds.c
+++ b/drivers/video/display/transmitter/rk616_lvds.c
@@ -135,6 +135,10 @@ static void rk616_lvds_early_suspend(struct early_suspend *h)
 	val |= LVDS_PLL_PWR_DN |(LVDS_CH1_PWR_EN << 16) | (LVDS_CH0_PWR_EN << 16) |
 		(LVDS_CBG_PWR_EN << 16) | (LVDS_PLL_PWR_DN << 16);
 	ret = rk616->write_dev(rk616,CRU_LVDS_CON0,&val);
+
+	val = LCD1_INPUT_EN | (LCD1_INPUT_EN << 16);
+	ret = rk616->write_dev(rk616,CRU_IO_CON0,&val);
+	
 	
 }
 
-- 
2.35.3

