From 1e51b0ee15b9ab33137e1e942e86eb0a63b79cf5 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Thu, 19 Jun 2014 10:40:55 +0800
Subject: [PATCH] Told to update this issue to avoid midgard compile warning

---
 drivers/staging/android/sync.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/android/sync.h b/drivers/staging/android/sync.h
index 75da9e85ac69..9eac8c2cc3da 100644
--- a/drivers/staging/android/sync.h
+++ b/drivers/staging/android/sync.h
@@ -20,6 +20,7 @@
 #include <linux/spinlock.h>
 #include <linux/wait.h>
 
+#include <linux/seq_file.h>
 #include "uapi/sync.h"
 
 struct sync_timeline;
-- 
2.35.3

