From ad2abac46f4c0e5cfdd60b3615d8f2654f6c044f Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 19 Aug 2010 09:00:05 +0800
Subject: [PATCH] mod mma7660.c

---
 drivers/input/gsensor/mma7660.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/gsensor/mma7660.c b/drivers/input/gsensor/mma7660.c
index bae4fb8a8e4a..23cc299d912b 100644
--- a/drivers/input/gsensor/mma7660.c
+++ b/drivers/input/gsensor/mma7660.c
@@ -154,7 +154,7 @@ static int mma7660_start_dev(char rate)
 	ret = mma7660_tx_data(&buffer[0], 2);
 	ret = mma7660_rx_data(&buffer[0], 1);
 
-	enable_irq(56);
+	enable_irq(this_client->irq);
 	rk28printk("\n----------------------------mma7660_start------------------------\n");
 	
 	return ret;
-- 
2.35.3

