From 804c5171911401853fe0792cb2faf3e395ba6912 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 11 Mar 2014 19:54:12 +0800
Subject: [PATCH] ARM: rockchip: cpu.c support rk3288

---
 arch/arm/mach-rockchip/cpu.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rockchip/cpu.c b/arch/arm/mach-rockchip/cpu.c
index 6ac8cc02892a..998dc9a6143f 100644
--- a/arch/arm/mach-rockchip/cpu.c
+++ b/arch/arm/mach-rockchip/cpu.c
@@ -9,7 +9,9 @@ static ssize_t type_show(struct device *dev, struct device_attribute *attr, char
 {
 	const char *type;
 
-	if (cpu_is_rk319x())
+	if (cpu_is_rk3288())
+		type = "rk3288";
+	else if (cpu_is_rk319x())
 		type = "rk319x";
 	else if (cpu_is_rk3188())
 		type = "rk3188";
@@ -33,7 +35,9 @@ static ssize_t soc_show(struct device *dev, struct device_attribute *attr, char
 {
 	const char *soc;
 
-	if (soc_is_rk3190())
+	if (soc_is_rk3288())
+		soc = "rk3288";
+	else if (soc_is_rk3190())
 		soc = "rk3190";
 	else if (soc_is_rk3188plus())
 		soc = "rk3188+";
-- 
2.35.3

