From d7ec4abc8405216ee29c5b1dd5c8e9c0ba53887d Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 24 Aug 2012 09:36:55 +0800
Subject: [PATCH] rk2928: touchscreen: if A720-A define RK2928_A720_A

---
 drivers/input/touchscreen/sitronix_ts_a720.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/input/touchscreen/sitronix_ts_a720.c b/drivers/input/touchscreen/sitronix_ts_a720.c
index 7e8922dec869..3a959941b2d0 100755
--- a/drivers/input/touchscreen/sitronix_ts_a720.c
+++ b/drivers/input/touchscreen/sitronix_ts_a720.c
@@ -30,6 +30,9 @@
 #include <linux/gpio.h>
 #include <mach/board.h>
 #include <linux/input/mt.h>
+/************ for A720 A&B config **************/
+#define RK2928_A720_A    //for A 
+/***********************************************/
 
 #ifdef SITRONIX_MONITOR_THREAD
 #include <linux/kthread.h>
@@ -761,7 +764,11 @@ static void sitronix_ts_work_func(struct work_struct *work)
 				input_report_abs(ts->input_dev,  ABS_MT_TRACKING_ID, i);
 				input_report_abs(ts->input_dev,  ABS_MT_TOUCH_MAJOR, 200);
 				input_report_abs(ts->input_dev,  ABS_MT_POSITION_X, MTDStructure[i].Pixel_X);
+                                #ifdef RK2928_A720_A
 				input_report_abs(ts->input_dev,  ABS_MT_POSITION_Y, 480 - MTDStructure[i].Pixel_Y);
+                                #else
+				input_report_abs(ts->input_dev,  ABS_MT_POSITION_Y, MTDStructure[i].Pixel_Y);
+                                #endif
 				input_report_abs(ts->input_dev,  ABS_MT_WIDTH_MAJOR, 100);
 				DbgMsg("lr[%d](%d, %d)+\n", i, MTDStructure[i].Pixel_X, MTDStructure[i].Pixel_Y);
 			}else if(MTDStructure[i].Current_Pressed_area == AREA_NONE){
-- 
2.35.3

