From 1041014f366b440ea15d8a9d88b5e0c6831ede82 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Feb 2011 09:21:52 +0800
Subject: [PATCH] rk29: pm: disable console suspend for debug

---
 arch/arm/mach-rk29/pm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk29/pm.c b/arch/arm/mach-rk29/pm.c
index 09128ab22dd9..05495b52a5e7 100644
--- a/arch/arm/mach-rk29/pm.c
+++ b/arch/arm/mach-rk29/pm.c
@@ -8,6 +8,7 @@
 #include <linux/regulator/consumer.h>
 #include <linux/io.h>
 #include <linux/wakelock.h>
+#include <linux/console.h>
 
 #include <mach/rk29_iomap.h>
 #include <mach/cru.h>
@@ -234,6 +235,9 @@ static int __init rk29_pm_init(void)
 	/* set idle function */
 	pm_idle = rk29_idle;
 
+	/* for debug */
+	console_suspend_enabled = 0;
+
 	return 0;
 }
 __initcall(rk29_pm_init);
-- 
2.35.3

