From 12431148518f87db19aebf057a995a353d2bcd0e Mon Sep 17 00:00:00 2001
From: wlf <wulf@rock-chips.com>
Date: Tue, 25 Feb 2014 17:37:56 +0800
Subject: [PATCH] USB: set bcdDevice default value 3.0

---
 drivers/usb/gadget/android.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 6de842d6af97..6712beb388c6 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1347,6 +1347,7 @@ static int android_bind(struct usb_composite_dev *cdev)
 		return id;
 	strings_dev[STRING_SERIAL_IDX].id = id;
 	device_desc.iSerialNumber = id;
+	device_desc.bcdDevice = cpu_to_le16(get_default_bcdDevice());
 
 	usb_gadget_set_selfpowered(gadget);
 	dev->cdev = cdev;
-- 
2.35.3

