From 2abadea49c024ac93138cd954fec2b4b769290d0 Mon Sep 17 00:00:00 2001
From: linjh <linjh@rock-chips.com>
Date: Fri, 10 Jun 2011 15:08:57 +0800
Subject: [PATCH] phonesdk: pulldown uart3 when gps probe

---
 drivers/misc/gps/rk29_gps.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/misc/gps/rk29_gps.c b/drivers/misc/gps/rk29_gps.c
index 6c876a2c2449..2dc8f76d65cf 100644
--- a/drivers/misc/gps/rk29_gps.c
+++ b/drivers/misc/gps/rk29_gps.c
@@ -272,8 +272,11 @@ static int rk29_gps_probe(struct platform_device *pdev)
 	pdata->power_flag = 0;
 
 	//gps power down
+	rk29_gps_uart_to_gpio(pdata->uart_id);
 	if (pdata->power_down)
 		pdata->power_down();
+	if (pdata->reset)
+		pdata->reset(GPIO_LOW);
 
 	pgps = pdata;
 
-- 
2.35.3

