From 717caa67014119d936b6fa8e84b6e2459fe533f0 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Sat, 4 Dec 2010 05:14:44 +0800
Subject: [PATCH] remove debug in fb driver

---
 drivers/video/rk29_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 1f555af3db02..5b085f6ac061 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -53,7 +53,7 @@
 
 #include "./display/screen/screen.h"
 
-#if 1
+#if 0
 	#define fbprintk(msg...)	printk(msg);
 #else
 	#define fbprintk(msg...)
-- 
2.35.3

