From e76043cd80f3fbfd19aeb65662ac6389d8b9d75b Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 25 Apr 2012 20:01:19 +0800
Subject: [PATCH] rk30 fb: delete __ATTR_NULL

---
 drivers/video/rockchip/rkfb_sysfs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index 1b6e74f6ff83..17f8fafa746a 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -159,7 +159,6 @@ static struct device_attribute rkfb_attrs[] = {
 	__ATTR(screen_info, S_IRUGO, show_screen_info, NULL),
 	__ATTR(enable, S_IRUGO | S_IWUSR, show_fb_state, set_fb_state),
 	__ATTR(overlay, S_IRUGO | S_IWUSR, show_overlay, set_overlay),
-	__ATTR_NULL,
 };
 
 int rkfb_create_sysfs(struct fb_info *fbi)
-- 
2.35.3

