From 60e559cf62204e1d54b3e8528692d65b2ddc4a5e Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Mon, 22 Aug 2011 15:29:10 +0800
Subject: [PATCH] rk29-fb: fix a bug of video rotation

---
 drivers/video/rk29_fb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index 9bb8a6164f55..a254ade2f966 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -1981,8 +1981,8 @@ static int fb1_set_par(struct fb_info *info)
 				ipp_req.dst0.CbrMst = inf->fb0->fix.mmio_start + screen->x_res*screen->y_res*(2*dstoffset+1);
 				//   if(var->xres > screen->x_res)
 				//   {
-					ipp_req.dst0.w = var->xres;
-					ipp_req.dst0.h = var->yres;
+					ipp_req.dst0.w = screen->x_res;
+					ipp_req.dst0.h = screen->y_res;
 				//  }   else	{
 				//	  ipp_req.dst0.w = var->yres;
 				// 	  ipp_req.dst0.h = var->xres;
-- 
2.35.3

