From 171f2967e172c6f2a8fa589740ca18050f61fdc3 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 16 Nov 2012 10:44:25 +0800
Subject: [PATCH] adc: rk30_adc.c: remove 'while(1)'

---
 drivers/adc/plat/rk30_adc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/adc/plat/rk30_adc.c b/drivers/adc/plat/rk30_adc.c
index 601f6b6aea8a..48d9ca6155cc 100755
--- a/drivers/adc/plat/rk30_adc.c
+++ b/drivers/adc/plat/rk30_adc.c
@@ -315,7 +315,6 @@ static int __init adc_test_init(void)
 {
         printk("def_ref_volt: %dmV, curr_ref_volt: %dmV\n", 
                         get_def_ref_volt(), get_curr_ref_volt());
-        while(1);
 #ifdef ADC_TEST	
 	rk30_adc_test();
 #endif
-- 
2.35.3

