From d40a582a8cd5d73abb47b7d6c386b31799b9c406 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 2 Jun 2015 16:58:26 +0800
Subject: [PATCH] rk: RK_CONSOLE_THREAD no depends on FIQ_DEBUGGER_CONSOLE

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/mach-rockchip/Kconfig.common | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/Kconfig.common b/arch/arm/mach-rockchip/Kconfig.common
index f0a5f3a5932b..b5fc1e274364 100644
--- a/arch/arm/mach-rockchip/Kconfig.common
+++ b/arch/arm/mach-rockchip/Kconfig.common
@@ -19,7 +19,6 @@ config RK_USB_UART
 
 config RK_CONSOLE_THREAD
 	bool "Console write by thread"
-	depends on FIQ_DEBUGGER_CONSOLE
 	default y
 	help
 	  Normal kernel printk will write out to UART by "kconsole" kthread
-- 
2.35.3

