From 92a8cef30a1dce5bc0afeb7355a81be8189a478e Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Sat, 26 Jan 2013 15:24:11 +0800
Subject: [PATCH] rk3188: dvfs: fix an writen error in dvfs.c

---
 arch/arm/mach-rk3188/dvfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk3188/dvfs.c b/arch/arm/mach-rk3188/dvfs.c
index 346bbd4e8445..a228645cf56d 100755
--- a/arch/arm/mach-rk3188/dvfs.c
+++ b/arch/arm/mach-rk3188/dvfs.c
@@ -161,7 +161,7 @@ int dvfs_target(struct clk *clk, unsigned long rate_hz)
 
 	/* if down the rate */
 	if (rate_new < rate_old) {
-		ret = ret = dvfs_scale_volt_direct(dvfs_clk->vd, volt_new);
+		ret = dvfs_scale_volt_direct(dvfs_clk->vd, volt_new);
 		if (ret < 0)
 			goto out;
 	}
-- 
2.35.3

