From 6460e425e478c315e396aa4b24dd59b762d32013 Mon Sep 17 00:00:00 2001
From: David Wu <wdc@rock-chips.com>
Date: Fri, 6 Mar 2015 21:00:38 +0800
Subject: [PATCH] rk3368: add tsadc for p9

Signed-off-by: David Wu <wdc@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368-p9_818.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368-p9_818.dts b/arch/arm64/boot/dts/rk3368-p9_818.dts
index ef661fbffc98..3f7c52373208 100755
--- a/arch/arm64/boot/dts/rk3368-p9_818.dts
+++ b/arch/arm64/boot/dts/rk3368-p9_818.dts
@@ -156,6 +156,12 @@
 	};
 };
 
+&tsadc {
+       tsadc-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
+       //tsadc-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
+       status = "okay";
+};
+
 &pinctrl {
 	//used for init some gpio
 	init-gpios = <&gpio0 GPIO_A6 GPIO_ACTIVE_HIGH>;
-- 
2.35.3

