From 038069e788370c47a3650a23f285661449cf8a22 Mon Sep 17 00:00:00 2001
From: xxh <xxh@rock-chips.com>
Date: Thu, 28 Mar 2013 16:26:11 +0800
Subject: [PATCH] change Airis 3G dongle data port to ttyUSB3

---
 drivers/base/devices_filter.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/base/devices_filter.h b/drivers/base/devices_filter.h
index e488496b53c9..bfc13bcf139d 100644
--- a/drivers/base/devices_filter.h
+++ b/drivers/base/devices_filter.h
@@ -103,6 +103,7 @@ static usb_parameter skip_device_list[] = {
         {0x0af0, 0xd157, NULL, NULL, NULL},      
 	    {0x19d2, 0x1216, NULL, NULL, NULL},         
 		{0x19d2, 0x1218, NULL, NULL, NULL},
+		{0x05c6, 0x9000, NULL, NULL, NULL},
    //end xxh		 
 };
 
-- 
2.35.3

