From a76d6952813f367803c95d1cea0d372e79a3c60e Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Wed, 7 May 2014 17:28:40 +0800
Subject: [PATCH] codec: soc-core log error

---
 sound/soc/soc-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index b26cbefb918c..015fe0b470ea 100755
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -2120,7 +2120,7 @@ EXPORT_SYMBOL_GPL(snd_soc_read);
 unsigned int snd_soc_write(struct snd_soc_codec *codec,
 			   unsigned int reg, unsigned int val)
 {
-	printk("write %x = %x\n", reg, val);
+	dev_dbg(codec->dev, "write%x => %x\n", reg, ret);
 	trace_snd_soc_reg_write(codec, reg, val);
 	return codec->write(codec, reg, val);
 }
-- 
2.35.3

