From 51c9e639b689dd6b65a326c5bcd655a015421c3a Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 23 Mar 2015 09:53:48 +0800
Subject: [PATCH] mmc: rk_sdmmc: close FJTAG when drv probe

Signed-off-by: lintao <lintao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index ae0a86ccdec8..9d254bd82bb5 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3488,8 +3488,11 @@ static int dw_mci_init_slot(struct dw_mci *host, unsigned int id)
 		else
 			/* Disable force_jtag */
 			regmap_write(host->grf, 0x43c, (1<<13)<<16 | (0 << 13));
+	} else if (cpu_is_rk3288()) {
+		grf_writel(((1 << 12) << 16) | (0 << 12), RK3288_GRF_SOC_CON0);
 	}
 
+
         /* We assume only low-level chip use gpio_cd */
         if ((soc_is_rk3126() || soc_is_rk3126b() || soc_is_rk3036()) &&
                 (host->mmc->restrict_caps & RESTRICT_CARD_TYPE_SD)) {
-- 
2.35.3

