From c6d076c7c26416ea1148b47ce8e332000435919d Mon Sep 17 00:00:00 2001
From: zwl <zwl@rock-chips.com>
Date: Fri, 6 Jun 2014 17:57:21 +0800
Subject: [PATCH] rk fb: fix 10 bit yuv video play err

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index f6f7099858ea..80350c1e08e8 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -147,7 +147,7 @@ int rk_fb_pixel_width(int data_format)
 	case YUV422_A:
 	case YUV420_A:
 	case YUV444_A:
-		pixel_width = 10;
+		pixel_width = 8;
 		break;
 	default:
 		printk(KERN_WARNING "%s:un supported format:0x%x\n",
-- 
2.35.3

