From 4d6a5b21542e175d09aca9c8eff7d713e290aa00 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 24 Sep 2012 20:53:51 +0800
Subject: [PATCH] rk2928: add dvfs direct scale voltage delay

---
 arch/arm/mach-rk2928/dvfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk2928/dvfs.c b/arch/arm/mach-rk2928/dvfs.c
index ede6bb7241d6..b2f0127afd07 100644
--- a/arch/arm/mach-rk2928/dvfs.c
+++ b/arch/arm/mach-rk2928/dvfs.c
@@ -904,6 +904,7 @@ int dvfs_scale_volt_direct(struct vd_node *vd_clk, int volt_new)
 	DVFS_DBG("ENTER %s, volt=%d(old=%d)\n", __func__, volt_new, vd_clk->cur_volt);
 	if (!IS_ERR_OR_NULL(vd_clk->regulator)) {
 		ret = dvfs_regulator_set_voltage_readback(vd_clk->regulator, volt_new, volt_new);
+		udelay(get_volt_up_delay(volt_new, vd_clk->cur_volt));
 		if (ret < 0) {
 			vd_clk->volt_set_flag = DVFS_SET_VOLT_FAILURE;
 			DVFS_ERR("%s %s set voltage up err ret = %d, Vnew = %d(was %d)mV\n", 
-- 
2.35.3

