From 06ef36205484913a4cb09f165e2b10f65e8cfe1a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Sun, 26 Sep 2010 20:46:07 -0700
Subject: [PATCH] close lp8725 control in pm.c

---
 arch/arm/mach-rk2818/pm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rk2818/pm.c b/arch/arm/mach-rk2818/pm.c
index a45dde92d231..266d43c39beb 100755
--- a/arch/arm/mach-rk2818/pm.c
+++ b/arch/arm/mach-rk2818/pm.c
@@ -658,7 +658,7 @@ static void rk2818_idle(void)
 	rk2818_tcm_idle();
 	asm volatile ("mov sp, %0" :: "r" (old_sp));
 
-	printk("rk2818_idle\n");
+	//printk("rk2818_idle\n");
 
 }
 
@@ -697,7 +697,7 @@ static int rk2818_pm_enter(suspend_state_t state)
 
 	printk(KERN_DEBUG "before core halt\n");
 
-	pmu_suspend( );
+	//pmu_suspend( );
 	clk_set_rate(arm_clk, 24000000);
 	dump_register();
 
@@ -721,7 +721,7 @@ static int rk2818_pm_enter(suspend_state_t state)
 
 	rockchip_timer_clocksource_suspend_resume(0);
 #endif
-	pmu_resume( );
+	//pmu_resume( );
 	dump_register();
 	clk_set_rate(arm_clk, arm_rate);
 	//rk2818_socpm_print();
-- 
2.35.3

