From 5d8d8603d606063c40c605e5c8dc30b0fd7afb12 Mon Sep 17 00:00:00 2001
From: lyx <lyx@rock-chips.com>
Date: Tue, 13 Jul 2010 17:32:51 -0700
Subject: [PATCH] modify dm9000 for power consume and high frequence

---
 drivers/net/dm9000.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/dm9000.c b/drivers/net/dm9000.c
index 52e0bb94180e..909822a6fd39 100644
--- a/drivers/net/dm9000.c
+++ b/drivers/net/dm9000.c
@@ -959,6 +959,8 @@ dm9000_rx(struct net_device *dev)
 	do {
 		ior(db, DM9000_MRCMDX);	/* Dummy read */
 
+		udelay(1);//add by lyx@20100713,or dm9000_rx will be error in high frequence
+
 		/* Get most updated data */
 		rxbyte = readb(db->io_data);
 
@@ -1638,6 +1640,9 @@ dm9000_probe(struct platform_device *pdev)
 		       ndev->name, dm9000_type_to_char(db->type),
 		       db->io_addr, db->io_data, ndev->irq,
 		       ndev->dev_addr, mac_src);
+
+	dm9000_shutdown(ndev);//add by lyx@20100713, reduce power consume
+
 	return 0;
 
 out:
-- 
2.35.3

