From 636ebb05aa59f20cb3414963fa7413f4cbabc637 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Tue, 30 Jul 2013 09:54:20 +0800
Subject: [PATCH] rk: efuse: fix comments describe error

---
 arch/arm/plat-rk/efuse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/plat-rk/efuse.c b/arch/arm/plat-rk/efuse.c
index 930091e07d00..ff46d9eb9509 100644
--- a/arch/arm/plat-rk/efuse.c
+++ b/arch/arm/plat-rk/efuse.c
@@ -71,7 +71,7 @@ int rk_leakage_val(void)
 {
 	/*
 	 * efuse_buf[22]
-	 * bit[3]:
+	 * bit[2]:
 	 * 	0:enable leakage level auto voltage scale
 	 * 	1:disalbe leakage level avs
 	 */
-- 
2.35.3

