From 14b1c79fc6966f1cd7cd499a63ea7f253a3dfedc Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 20 Apr 2015 20:54:07 +0800
Subject: [PATCH] rk3368 lcdc: add support serial RGB888 and RGB888 DUMY output

Signed-off-by: hjc <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index 7a509cd6d5d3..3a38d3d7cdf5 100755
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -1926,6 +1926,18 @@ static int rk3368_load_screen(struct rk_lcdc_driver *dev_drv, bool initscreen)
 			val = v_DITHER_DOWN_EN(0);
 			lcdc_msk_reg(lcdc_dev, DSP_CTRL1, mask, val);
 			break;
+		case OUT_S888:
+			face = OUT_S888;
+			mask = m_DITHER_DOWN_EN;
+			val = v_DITHER_DOWN_EN(0);
+			lcdc_msk_reg(lcdc_dev, DSP_CTRL1, mask, val);
+			break;
+		case OUT_S888DUMY:
+			face = OUT_S888DUMY;
+			mask = m_DITHER_DOWN_EN;
+			val = v_DITHER_DOWN_EN(0);
+			lcdc_msk_reg(lcdc_dev, DSP_CTRL1, mask, val);
+			break;
 		default:
 			dev_err(lcdc_dev->dev, "un supported interface!\n");
 			break;
-- 
2.35.3

