From f582d25ddec7dbe1868717a16d2ef691788cfa76 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 26 Mar 2012 11:55:29 +0800
Subject: [PATCH] rk30: pm: enable l2x0 dynamic clk gating and standby mode

---
 arch/arm/mach-rk30/common.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 38ab1e3ff4b5..a248992d9827 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -20,10 +20,14 @@ static void __init rk30_l2_cache_init(void)
 #ifdef CONFIG_CACHE_L2X0
 	u32 aux_ctrl, aux_ctrl_mask;
 
-	//Tag Ram Latency All 1-cycle
+	// Tag Ram Latency All 1-cycle
 	writel_relaxed(0x0, RK30_L2C_BASE + L2X0_TAG_LATENCY_CTRL);
 	// Data Ram Latency [10:8] 1-cycle [6:4] 4-cycles [2:0] 2 cycles
 	writel_relaxed(0x031, RK30_L2C_BASE + L2X0_DATA_LATENCY_CTRL);
+
+	/* L2X0 Power Control */
+	writel_relaxed(L2X0_DYNAMIC_CLK_GATING_EN | L2X0_STNDBY_MODE_EN, RK30_L2C_BASE + L2X0_POWER_CTRL);
+
 	/*
          * 16-way associativity, parity disabled
          * Way size - 32KB
-- 
2.35.3

