From 0cd2bd1a9c445ebd40b824dccc8c0057c22b704f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 28 Feb 2013 13:56:19 +0800
Subject: [PATCH] pack-kernel.sh: exclude rk3168m

---
 pack-kernel.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index a586e517a055..0b62afa5fa40 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -131,8 +131,10 @@ arch/arm/mach-rk29/*.h
 arch/arm/mach-rk29/*.S
 arch/arm/mach-rk29/Makefile*
 arch/arm/mach-rk29/include
+arch/arm/mach-rk30/*rk3168m*
 arch/arm/mach-rk*/*-fpga*
 arch/arm/configs/rk29_*
+arch/arm/configs/rk3168m_*
 arch/arm/configs/*_fpga_*
 )
 # ---------------------------------------------------------------------------
-- 
2.35.3

