From fca83b70fe0010773165db4b3a2b9393b5ee4d7e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 19 Mar 2013 12:10:20 +0800
Subject: [PATCH] pack-kernel.sh: exclude rk3028

---
 pack-kernel.sh | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 45c90d78e065..6aa1729d2c8d 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -34,10 +34,8 @@ arch/arm/configs/rk30_phonepad*
 arch/arm/mach-rk2928/board-rk2928-a720*
 arch/arm/configs/rk2928_a720_defconfig
 
-arch/arm/mach-rk2928b/*.c
-arch/arm/mach-rk2928b/*.h
-arch/arm/mach-rk2928b/*.S
-arch/arm/configs/rk2928b_*
+arch/arm/mach-rk30/*rk3028*
+arch/arm/configs/rk3028_*
 )
 
 # ---------------------------------------------------------------------------
-- 
2.35.3

