From 0c2ef6f278a4756c88ca7f974c1fdd7708f15e2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sun, 6 Jan 2013 14:13:48 +0800
Subject: [PATCH] adc: core: better init adc_host_head

---
 drivers/adc/core.c | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/drivers/adc/core.c b/drivers/adc/core.c
index c13c15d3879e..91344386feda 100755
--- a/drivers/adc/core.c
+++ b/drivers/adc/core.c
@@ -8,7 +8,7 @@
 #include "adc_priv.h"
 
 struct adc_host *g_adc = NULL;
-struct list_head adc_host_head;
+static LIST_HEAD(adc_host_head);
 
 struct adc_host *adc_alloc_host(struct device *dev, int extra, enum host_chn_mask mask)
 {
@@ -270,18 +270,3 @@ int adc_get_curr_ref_volt(void)
         return volt * 1024 / v;
 }
 EXPORT_SYMBOL(adc_get_curr_ref_volt);
-
-static int __init adc_core_init(void)
-{
-        INIT_LIST_HEAD(&adc_host_head);
-        return 0;
-}
-subsys_initcall(adc_core_init);
-
-static void __exit adc_core_exit(void)
-{
-        return;
-}
-module_exit(adc_core_exit);  
-
-
-- 
2.35.3

