From 2c57875ccea67bc9c567e422edc712e0d0d70111 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Mon, 3 Nov 2014 17:06:23 +0800
Subject: [PATCH] sar-adc: extra channel is channel-6 via datasheet

---
 drivers/iio/adc/rockchip_adc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iio/adc/rockchip_adc.c b/drivers/iio/adc/rockchip_adc.c
index 05ca411eab49..974903cc511d 100755
--- a/drivers/iio/adc/rockchip_adc.c
+++ b/drivers/iio/adc/rockchip_adc.c
@@ -202,7 +202,7 @@ static const struct iio_chan_spec rk_adc_iio_channels[] = {
 	ADC_CHANNEL(0, "adc0"),
 	ADC_CHANNEL(1, "adc1"),
 	ADC_CHANNEL(2, "adc2"),
-	ADC_CHANNEL(3, "adc3"),
+	ADC_CHANNEL(6, "adc6"),
 };
 
 static int rk_adc_remove_devices(struct device *dev, void *c)
-- 
2.35.3

