From 479e51b5a642cb3de6e125aa27f39da48a503a05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 16 Jun 2011 16:01:46 +0800
Subject: [PATCH] arm: Makefile: add zkernel.img target

---
 arch/arm/Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index fde93af92ad3..c101382c4afe 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -271,13 +271,18 @@ archclean:
 bp:;	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $(boot)/bootpImage
 i zi:;	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $@
 
-PHONY += kernel.img
+PHONY += kernel.img zkernel.img
 kernel.img: Image FORCE
 	$(Q)$(srctree)/mkkrnlimg $(obj)/arch/arm/boot/Image $(obj)/kernel.img
 	@echo '  Image:  $@ is ready'
 
+zkernel.img: zImage FORCE
+	$(Q)$(srctree)/mkkrnlimg $(obj)/arch/arm/boot/zImage $(obj)/kernel.img
+	@echo '  Image:  kernel.img is ready'
+
 define archhelp
   echo  '  kernel.img    - Rockchip kernel image'
+  echo  '  zkernel.img   - Compressed Rockchip kernel image'
   echo  '* zImage        - Compressed kernel image (arch/$(ARCH)/boot/zImage)'
   echo  '  Image         - Uncompressed kernel image (arch/$(ARCH)/boot/Image)'
   echo  '* xipImage      - XIP kernel image, if configured (arch/$(ARCH)/boot/xipImage)'
-- 
2.35.3

