From 42c4038c46249ce61ba68f99cb0a0e0c68f5a691 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Mon, 23 Jul 2012 16:14:29 +0800
Subject: [PATCH] rk31: fix sdmmc bug after rk2928 commit sdmmc support

---
 drivers/mmc/host/rk29_sdmmc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/rk29_sdmmc.c b/drivers/mmc/host/rk29_sdmmc.c
index acab9705e8de..952b51c1dde1 100755
--- a/drivers/mmc/host/rk29_sdmmc.c
+++ b/drivers/mmc/host/rk29_sdmmc.c
@@ -55,11 +55,11 @@
 #define SD_DETECT_NAME          GPIO2A2_SDMMC0DETECTN_NAME
 #define SD_DETECT_GPIO_MODE     GPIO2L_GPIO2A2
 #define SD_DETECT_DET_MODE      GPIO2L_SDMMC0_DETECT_N
-#elif defined(CONFIG_ARCH_RK30)
+#elif defined(CONFIG_ARCH_RK30) || defined(CONFIG_ARCH_RK31)
 #define SD_DETECT_PIN           RK30_PIN3_PB6
 #define SD_DETECT_NAME          GPIO3B6_SDMMC0DETECTN_NAME
 #define SD_DETECT_GPIO_MODE     GPIO3B_GPIO3B6
-#define SD_DETECT_DET_MODE      GPIO3B_SDMMC_DETECT_N
+#define SD_DETECT_DET_MODE      GPIO3B_SDMMC0_DETECT_N
 #elif defined(CONFIG_ARCH_RK2928)
 #define SD_DETECT_PIN           RK2928_PIN1_PC7
 #define SD_DETECT_NAME          GPIO1C1_MMC0_DETN_NAME 
-- 
2.35.3

