From fd04a8bcc4a92e38b85bbf065694a372d53e4b5f Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 23 Apr 2012 09:43:45 +0800
Subject: [PATCH] rk30: i2c: Adjust the print information in the order

---
 drivers/i2c/busses/i2c-rk30-adapter.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/i2c-rk30-adapter.c b/drivers/i2c/busses/i2c-rk30-adapter.c
index c65b598b3a6b..7bdfc885425b 100755
--- a/drivers/i2c/busses/i2c-rk30-adapter.c
+++ b/drivers/i2c/busses/i2c-rk30-adapter.c
@@ -479,11 +479,11 @@ static int rk30_i2c_doxfer(struct rk30_i2c *i2c,
 	spin_unlock_irqrestore(&i2c->lock, flags);
 
 	if (timeout == 0){
+                dev_err(i2c->dev, "addr[0x%02x] wait event timeout, state: %d, is_busy: %d, error: %d\n", 
+                                msgs[0].addr, i2c->state, i2c->is_busy, i2c->error);  
                 i2c->state = STATE_IDLE;
                 if(i2c->error < 0) i2c->error = -ETIMEDOUT;
                 rk30_i2c_send_stop(i2c);
-                dev_err(i2c->dev, "addr[0x%02x] wait event timeout, state: %d, is_busy: %d, error: %d\n", 
-                                msgs[0].addr, i2c->state, i2c->is_busy, i2c->error);  
 
         }
 	rk30_i2c_disable_irq(i2c);
-- 
2.35.3

