From 26585a4e47006c7489cc93aaee3455143ec34503 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Mon, 17 Mar 2014 14:36:04 +0800
Subject: [PATCH] usb: add battery charger detect regs to rk3288.dtsi

---
 arch/arm/boot/dts/rk3288.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index a72b92b618c5..6a7898d158de 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -570,6 +570,19 @@
 		/*	<&gpio0 GPIO_B4 GPIO_ACTIVE_LOW>;*//*OTG_VBUS_DRV*/
 		/*clocks = <&clk_gates4 5>;*/
 		/*clock-names = "hclk_usb_peri";*/
+
+		usb_bc{
+			compatible = "rockchip,ctrl";
+				        /* offset bit mask */
+			rk_usb,bvalid     = <0x288 14 1>;
+			rk_usb,dcdenb     = <0x328 14 1>;
+			rk_usb,vdatsrcenb = <0x328  7 1>;
+			rk_usb,vdatdetenb = <0x328  6 1>;
+			rk_usb,chrgsel    = <0x328  5 1>;
+			rk_usb,chgdet     = <0x2cc 23 1>;
+			rk_usb,fsvminus   = <0x2cc 25 1>;
+			rk_usb,fsvplus    = <0x2cc 24 1>;
+		};
 	};
 
 	usb1: usb@ff580000 {
-- 
2.35.3

