From 5d3af616d26e0262ce894bd720404df4ca530c1e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:21:28 +0800
Subject: [PATCH] hid-core: fix for v2.6.36

---
 drivers/hid/usbhid/hid-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/usbhid/hid-core.c b/drivers/hid/usbhid/hid-core.c
index abfa9110d5e2..63c0154e9e3b 100755
--- a/drivers/hid/usbhid/hid-core.c
+++ b/drivers/hid/usbhid/hid-core.c
@@ -1356,7 +1356,7 @@ static int hid_suspend(struct usb_interface *intf, pm_message_t message)
 
 	hid_cancel_delayed_stuff(usbhid);
 	hid_cease_io(usbhid);
-    usb_control_msg(udev, usb_sndctrlpipe(udev, 0),
+    usb_control_msg(interface_to_usbdev(intf), usb_sndctrlpipe(interface_to_usbdev(intf), 0),
             USB_REQ_SET_FEATURE, USB_TYPE_STANDARD | USB_RECIP_DEVICE, USB_DEVICE_REMOTE_WAKEUP,
             0, NULL, 0, USB_CTRL_SET_TIMEOUT);
 
-- 
2.35.3

