From effbff51fae7f8ca65673af95c8b24d46a88a063 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 20 May 2015 22:08:31 +0800
Subject: [PATCH] pwm: rockchip: fix compilation error on arm

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/pwm/pwm-rockchip.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/pwm/pwm-rockchip.c b/drivers/pwm/pwm-rockchip.c
index 33b28e939493..640e7c4c8dfd 100755
--- a/drivers/pwm/pwm-rockchip.c
+++ b/drivers/pwm/pwm-rockchip.c
@@ -114,8 +114,16 @@ module_param_named(dbg_level, pwm_dbg_level, int, 0644);
 
 };
 
+#ifdef CONFIG_ARM64
 extern int rk3368_lcdc_update_pwm(int bl_pwm_period, int bl_pwm_duty);
 extern int rk3368_lcdc_cabc_status(void);
+#else
+static inline int rk3368_lcdc_update_pwm(int bl_pwm_period, int bl_pwm_duty)
+{
+	return 0;
+}
+static inline int rk3368_lcdc_cabc_status(void) { return 0; }
+#endif
 
 static inline void rk_pwm_writel(struct rk_pwm_chip *chip,
 				    unsigned int num, unsigned long offset,
-- 
2.35.3

