From 6953dcb68ce00b5101f234c3be68449ea5c37c87 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Wed, 15 Oct 2014 14:04:48 +0800
Subject: [PATCH] rk312x: remove unused usb charger detect log info

---
 drivers/usb/dwc_otg_310/usbdev_bc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/dwc_otg_310/usbdev_bc.c b/drivers/usb/dwc_otg_310/usbdev_bc.c
index a62c489d9cad..0cad8b970d01 100755
--- a/drivers/usb/dwc_otg_310/usbdev_bc.c
+++ b/drivers/usb/dwc_otg_310/usbdev_bc.c
@@ -249,8 +249,10 @@ enum bc_port_type usb_battery_charger_detect_inno(bool wait)
 
 	}
 out:
+	/*
 	printk("%s, Charger type %s, %s DCD, dcd_state = %d\n", __func__,
 	       bc_string[port_type], wait ? "wait" : "pass", dcd_state);
+	*/
 	return port_type;
 
 }
-- 
2.35.3

