From 138da94831a852fd260a14924410a6a113f5a20d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:19:59 +0800
Subject: [PATCH] rk29: vpu_service: fix for v2.6.36

---
 arch/arm/mach-rk29/vpu_service.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk29/vpu_service.c b/arch/arm/mach-rk29/vpu_service.c
index d5465cb221aa..8076490c86ea 100644
--- a/arch/arm/mach-rk29/vpu_service.c
+++ b/arch/arm/mach-rk29/vpu_service.c
@@ -35,6 +35,7 @@
 #include <linux/poll.h>
 #include <linux/platform_device.h>
 #include <linux/workqueue.h>
+#include <linux/slab.h>
 
 #include <asm/uaccess.h>
 
-- 
2.35.3

