From c3ecb7d483f8bce67dbab3bdb90898479a8c46c2 Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Fri, 6 Feb 2015 15:48:17 +0800
Subject: [PATCH] rk3368: cpufreq: fix compile warning

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 drivers/cpufreq/rockchip_big_little.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/rockchip_big_little.c b/drivers/cpufreq/rockchip_big_little.c
index 930e52318ff4..de82b1ac250c 100644
--- a/drivers/cpufreq/rockchip_big_little.c
+++ b/drivers/cpufreq/rockchip_big_little.c
@@ -507,7 +507,7 @@ static struct cpufreq_driver cpufreq_driver = {
 	.get = cpufreq_bl_get_rate,
 	.init = cpufreq_bl_init,
 	.exit = cpufreq_bl_exit,
-	.name = "rockchip-big-little",
+	.name = "rockchip-bl",
 	.have_governor_per_policy = true,
 	.attr = cpufreq_attr,
 };
-- 
2.35.3

