From ed4fba9d3435556106f7d0d903229d30ca9c628c Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 26 May 2015 16:17:05 +0800
Subject: [PATCH] Makefile: revert to 899286fb6a7a20f5b5c5d0a9a22e4fc39b2d9e70

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 0c5b19b6ca0f..fa41a5c93173 100644
--- a/Makefile
+++ b/Makefile
@@ -197,8 +197,8 @@ SUBARCH := $(shell uname -m | sed -e s/i.86/x86/ -e s/x86_64/x86/ \
 ARCH		?= arm
 ARCH		?= $(SUBARCH)
 ifeq ($(ARCH),arm64)
-ifneq ($(wildcard ../3288/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9),)
-CROSS_COMPILE	?= ../3288/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin/aarch64-linux-android-
+ifneq ($(wildcard ../prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9),)
+CROSS_COMPILE	?= ../prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin/aarch64-linux-android-
 endif
 endif
 ifeq ($(ARCH),arm)
-- 
2.35.3

