From 255d5993a01ea6f1561acb97a5a7e0904e421ac7 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 18 Mar 2014 10:09:02 +0800
Subject: [PATCH] rk fb: del some unused stagement

---
 drivers/video/rockchip/rk_fb.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index b6771e976056..591e877717e7 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1142,9 +1142,9 @@ static void rk_fb_update_reg(struct rk_lcdc_driver * dev_drv,struct rk_fb_reg_da
 
 	ret = wait_event_interruptible_timeout(dev_drv->vsync_info.wait,
 			!ktime_equal(timestamp, dev_drv->vsync_info.timestamp),msecs_to_jiffies(dev_drv->cur_screen->ft+5));
-
+#ifdef H_USE_FENCE
 	sw_sync_timeline_inc(dev_drv->timeline, 1);
-
+#endif
 	for(i=0;i<regs->win_num;i++){
 		rk_fb_free_dma_buf(dev_drv->dev,&regs->reg_win_data[i]);
 	}
@@ -1721,9 +1721,6 @@ static ssize_t rk_fb_read(struct fb_info *info, char __user *buf,
 	int win_id = 0;
 	int list_empty=0;
 
-	list_empty = rk_fb_get_list_stat(dev_drv);
-	printk("list_empty=%d\n",list_empty);
-
 	win_id = dev_drv->ops->fb_get_win_id(dev_drv, info->fix.id);
 	if (win_id < 0)
 		return  -ENODEV;
-- 
2.35.3

