From 62dfc8a780ed2a064af027cf29d7f9b053a9e263 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=B5=B5=E5=AD=90=E5=88=9D?= <zzc@rock-chips.com>
Date: Tue, 21 May 2013 11:52:47 +0800
Subject: [PATCH] add some bp id

---
 include/linux/bp-auto.h | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/include/linux/bp-auto.h b/include/linux/bp-auto.h
index 76d89c2a1390..8ade775d1913 100644
--- a/include/linux/bp-auto.h
+++ b/include/linux/bp-auto.h
@@ -26,16 +26,19 @@ struct bp_private_data;
 enum bp_id{
 	BP_ID_INVALID = 0,//no bp
 		
-	BP_ID_MT6229,	//USI MT6229 WCDMA
-	BP_ID_MU509,	//huawei MU509 WCDMA
-	BP_ID_MI700,	//thinkwill MI700 WCDMA
-	BP_ID_MW100,	//thinkwill MW100 WCDMA
-	BP_ID_TD8801,	//spreadtrum SC8803 TD-SCDMA
-	BP_ID_SC6610,	//spreadtrum SC6610 GSM
-	BP_ID_M50,		//spreadtrum RDA GSM
-	BP_ID_MT6250,  //ZINN M50  EDGE
-	BP_ID_C66A,    //zhongben
-	BP_ID_SEW290,    //SCV SEW290 WCDMA
+	BP_ID_MT6229,	//1 USI MT6229 WCDMA
+	BP_ID_MU509,	//2 huawei MU509 WCDMA
+	BP_ID_MI700,	//3 thinkwill MI700 WCDMA
+	BP_ID_MW100,	//4 thinkwill MW100 WCDMA
+	BP_ID_TD8801,	//5 spreadtrum SC8803 TD-SCDMA
+	BP_ID_SC6610,	//6 spreadtrum SC6610 GSM
+	BP_ID_M50,	//7 spreadtrum RDA GSM
+	BP_ID_MT6250,   //8 ZINN M50  EDGE
+	BP_ID_C66A,     //9 ZHIGUAN C66A GSM
+	BP_ID_SEW290,   //10 SCV SEW290 WCDMA
+	BP_ID_U5501,    //11 LONGSUNG U5501 WCDMA
+	BP_ID_U7501,    //12 LONGSUNG U7501 WCDMA/HSPA+
+	BP_ID_AW706,    //13 ANICARE AW706 EDGE
 	BP_ID_NUM,  
 };
 
-- 
2.35.3

