From 996798a6c601ba4a2ddcf3f0f73ac7a056e5c64d Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Thu, 29 Nov 2012 19:03:01 +0800
Subject: [PATCH] bt: modify for mt6622

---
 drivers/bluetooth/hci_ldisc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/bluetooth/hci_ldisc.c b/drivers/bluetooth/hci_ldisc.c
index 64bc1d9cb6c2..b79b04398a3b 100755
--- a/drivers/bluetooth/hci_ldisc.c
+++ b/drivers/bluetooth/hci_ldisc.c
@@ -144,6 +144,7 @@ int hci_uart_tx_wakeup(struct hci_uart *hu)
 
 restart:
 	clear_bit(HCI_UART_TX_WAKEUP, &hu->tx_state);
+#if !defined(CONFIG_MT5931_MT6622)
 /*added by Barry,for broadcom 4325*/
 #ifdef CONFIG_BT_AUTOSLEEP
 #ifdef CONFIG_RFKILL_RK
@@ -153,6 +154,7 @@ int hci_uart_tx_wakeup(struct hci_uart *hu)
     //extern void bcm4325_sleep(unsigned long bSleep);
     //bcm4325_sleep(0);
 #endif
+#endif
 #endif
 	while ((skb = hci_uart_dequeue(hu))) {
 		int len;
-- 
2.35.3

