From 1671db28a6db4ae09ff4c19ad08c1f1e373b78d8 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Tue, 1 Apr 2014 10:54:54 +0800
Subject: [PATCH] rt5616: disable log error

---
 sound/soc/codecs/rt5616.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/rt5616.c b/sound/soc/codecs/rt5616.c
index 9302581435c1..60a1224d7692 100755
--- a/sound/soc/codecs/rt5616.c
+++ b/sound/soc/codecs/rt5616.c
@@ -1251,10 +1251,11 @@ static int rt5616_prepare(struct snd_pcm_substream *substream,
 
 static int rt5616_set_gpio(struct rt5616_priv *rt5616, int gpio, bool level)
 {	
-	//printk("%s : set %s ctl gpio %s\n", __func__,
+#if 0
+	printk("%s : set %s ctl gpio %s\n", __func__,
 		gpio & rt5616_CODEC_SET_SPK ? "spk" : "",
 		level ? "HIGH" : "LOW");
-
+#endif
 	if ((gpio & rt5616_CODEC_SET_SPK) && gpio_is_valid(rt5616->spk_ctl_gpio)) {
 		mdelay(100);
 		gpio_set_value(rt5616->spk_ctl_gpio, level);		
-- 
2.35.3

