From 07f1e53e4cac855288729e4f8fb797d7295ed94f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Jan 2013 11:49:40 +0800
Subject: [PATCH] rga: support rk3188

---
 drivers/video/rockchip/rga/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rga/Kconfig b/drivers/video/rockchip/rga/Kconfig
index 152e2bd55ef0..6d82e86fe016 100755
--- a/drivers/video/rockchip/rga/Kconfig
+++ b/drivers/video/rockchip/rga/Kconfig
@@ -1,5 +1,5 @@
 menu "RGA"
-	depends on ARCH_RK30 || ARCH_RK2928
+	depends on ARCH_RK30 || ARCH_RK2928 || ARCH_RK3188
 
 config RGA_RK30
 	tristate "ROCKCHIP RK30 || RK2928 RGA"
-- 
2.35.3

