From 327e7a38e32c3b98bc32ff48d51b5151165d2f0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 3 Dec 2010 18:17:19 +0800
Subject: [PATCH] rk29: clock: add lcdc gate

---
 arch/arm/mach-rk29/clock.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 8c2b35dee4e8..89382ce1b676 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -1209,6 +1209,8 @@ static struct clk *dclk_lcdc_parents[2] = { &dclk_lcdc_div, &extclk };
 
 static struct clk dclk_lcdc = {
 	.name		= "dclk_lcdc",
+	.mode		= gate_mode,
+	.gate_idx	= CLK_GATE_LCDC,
 	.clksel_con	= CRU_CLKSEL16_CON,
 	.clksel_parent_mask	= 1,
 	.clksel_parent_shift	= 10,
@@ -1231,6 +1233,8 @@ static struct clk *aclk_lcdc_parents[4] = { &ddr_pll_clk, &codec_pll_clk, &perip
 
 static struct clk aclk_lcdc = {
 	.name		= "aclk_lcdc",
+	.mode		= gate_mode,
+	.gate_idx	= CLK_GATE_LCDC_AXI,
 	.recalc		= clksel_recalc_div,
 	.set_rate	= clksel_set_rate_div,
 	.clksel_con	= CRU_CLKSEL16_CON,
@@ -1243,6 +1247,8 @@ static struct clk aclk_lcdc = {
 
 static struct clk hclk_lcdc = {
 	.name		= "hclk_lcdc",
+	.mode		= gate_mode,
+	.gate_idx	= CLK_GATE_LCDC_AHB,
 	.parent		= &aclk_lcdc,
 	.clksel_con	= CRU_CLKSEL16_CON,
 	.recalc		= clksel_recalc_shift,
-- 
2.35.3

