From 79e6d3223ac12782fa0b2ebf591e2dd2e4c9896f Mon Sep 17 00:00:00 2001
From: huangdesheng <hds@rock-chips.com>
Date: Thu, 5 Jul 2012 18:17:17 +0800
Subject: [PATCH] revise rga code err

---
 drivers/video/rockchip/rga/rga_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rga/rga_drv.c b/drivers/video/rockchip/rga/rga_drv.c
index 74b810df0e54..808675c04546 100755
--- a/drivers/video/rockchip/rga/rga_drv.c
+++ b/drivers/video/rockchip/rga/rga_drv.c
@@ -956,6 +956,7 @@ static long rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 			break;
         case RGA_GET_RESULT:
             ret = rga_get_result(session, arg);
+            break;
         case RGA_GET_VERSION:
             ret = copy_to_user((void *)arg, RGA_VERSION, sizeof(RGA_VERSION));
             //ret = 0;
-- 
2.35.3

