From 3d5a644f1c8598a9c4ffbe6c8c02791708156fe5 Mon Sep 17 00:00:00 2001
From: zyw <zyw@rock-chips.com>
Date: Mon, 9 May 2011 15:31:45 +0800
Subject: [PATCH] fix open lcd pwr pin in fb_resume when hdmi insert

---
 drivers/video/rk29_fb.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rk29_fb.c b/drivers/video/rk29_fb.c
index b90ae41b8d82..162649d4969d 100755
--- a/drivers/video/rk29_fb.c
+++ b/drivers/video/rk29_fb.c
@@ -2299,7 +2299,8 @@ static void rk29fb_early_suspend(struct early_suspend *h)
         return;
     }
 
-    set_lcd_pin(g_pdev, 0);
+    if(inf->cur_screen != &inf->panel2_info)  // close lcd pwr when output screen is lcd
+        set_lcd_pin(g_pdev, 0);
 
 	if(inf->cur_screen->standby)
 	{
@@ -2381,7 +2382,8 @@ static void rk29fb_early_resume(struct early_suspend *h)
 		inf->cur_screen->standby(0);
 	}
     msleep(100);
-    set_lcd_pin(g_pdev, 1);
+    if(inf->cur_screen != &inf->panel2_info)  // open lcd pwr when output screen is lcd
+        set_lcd_pin(g_pdev, 1);
 	memcpy((u8*)inf->preg, (u8*)&inf->regbak, 0xa4);  //resume reg
 }
 
-- 
2.35.3

