From 5ded248649cc4551a908879965604f61b10416ba Mon Sep 17 00:00:00 2001
From: luowei <lw@rock-chips.com>
Date: Mon, 23 Aug 2010 20:54:52 +0800
Subject: [PATCH] change SPI_BUFSIZ for spi_dpram and spi_uart

---
 drivers/spi/spi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index b76f2468a84a..1d3ba252453d 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -774,7 +774,7 @@ int spi_sync(struct spi_device *spi, struct spi_message *message)
 EXPORT_SYMBOL_GPL(spi_sync);
 
 /* portable code must never pass more than 32 bytes */
-#define	SPI_BUFSIZ	max(32,SMP_CACHE_BYTES)
+#define	SPI_BUFSIZ	max(1028,SMP_CACHE_BYTES)
 
 static u8	*buf;
 
-- 
2.35.3

