From 9fdf501364f5411f33dce4d281f926bc624c657c Mon Sep 17 00:00:00 2001
From: gwl <gwl@rock-chips.com>
Date: Wed, 11 Sep 2013 14:24:20 +0800
Subject: [PATCH] Enable bcm wifi host wake up function as default.

---
 drivers/net/wireless/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/Kconfig b/drivers/net/wireless/Kconfig
index a6891d5d8f14..1a4b4f13d2c9 100755
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -36,7 +36,7 @@ config BCM_OOB_ENABLED
 	bool "Enable wifi host wake up function"
 	depends on RKWIFI
 	select RK29_SDIO_IRQ_FROM_GPIO
-	default n
+	default y
 	---help---
 
 choice
-- 
2.35.3

