From 7ec033f4ae504898ac70894ce3a5a5dd4b936bfb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 17:05:46 +0800
Subject: [PATCH] ARM: rockchip: iomap.h: fix compilation error on arm64

---
 include/linux/rockchip/iomap.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/rockchip/iomap.h b/include/linux/rockchip/iomap.h
index 31a58d5551ea..f8785fa69803 100755
--- a/include/linux/rockchip/iomap.h
+++ b/include/linux/rockchip/iomap.h
@@ -5,7 +5,11 @@
 #include <asm/io.h>
 #endif
 
+#ifdef IOMEM
 #define RK_IO_ADDRESS(x)                IOMEM(0xFED00000 + x)
+#else
+#define RK_IO_ADDRESS(x)                ((void __force __iomem *)(0xFED00000 + x))
+#endif
 
 #define RK_CRU_VIRT                     RK_IO_ADDRESS(0x00000000)
 #define RK_GRF_VIRT                     RK_IO_ADDRESS(0x00010000)
-- 
2.35.3

