From b9721d0365d9e88d0c16c3f0c2aff4b2c09c9dec Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Mon, 2 Mar 2015 15:42:55 +0800
Subject: [PATCH] rk: uboot_mem_reserve: lost a '=' in 'if' statement

---
 arch/arm/mach-rockchip/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/common.c b/arch/arm/mach-rockchip/common.c
index 96cf6c760eef..eded0645795d 100755
--- a/arch/arm/mach-rockchip/common.c
+++ b/arch/arm/mach-rockchip/common.c
@@ -306,7 +306,7 @@ phys_addr_t uboot_logo_offset=0;
 
 void __init rockchip_uboot_mem_reserve(void)
 {
-	if (uboot_logo_size=0)
+	if (uboot_logo_size==0)
 		return;
 
 	if (!memblock_is_region_reserved(uboot_logo_base, uboot_logo_size)
-- 
2.35.3

