From 8c3548e466a573f96f810014f01a3fa6eac27325 Mon Sep 17 00:00:00 2001
From: zwl <zwl@rock-chips.com>
Date: Thu, 19 Jun 2014 21:05:45 +0800
Subject: [PATCH] rk fb: modify get_prmry_screen_ft in case ft is zero

---
 drivers/video/rockchip/rk_fb.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index ece4d7c6111c..1713adda87b4 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -525,7 +525,10 @@ u32 rk_fb_get_prmry_screen_ft(void)
 
 	ft_us = dev_drv->frame_time.framedone_t - dev_drv->frame_time.last_framedone_t;
 	do_div(ft_us, 1000);
-	dev_drv->frame_time.ft = min(dev_drv->frame_time.ft, (u32)ft_us);
+	ft_us = min(dev_drv->frame_time.ft, (u32)ft_us);
+	if (ft_us != 0)
+		dev_drv->frame_time.ft = ft_us;
+
 	return dev_drv->frame_time.ft;
 }
 
-- 
2.35.3

