From c3769ccdc02300ba98dce6d0d15d8997ab531cd1 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Tue, 25 Mar 2014 11:28:22 +0800
Subject: [PATCH] hdmi_i2s: delete set to codec without ops

---
 sound/soc/rockchip/rk_hdmi_i2s.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rockchip/rk_hdmi_i2s.c b/sound/soc/rockchip/rk_hdmi_i2s.c
index 42d5a60e8344..2eae0335501e 100644
--- a/sound/soc/rockchip/rk_hdmi_i2s.c
+++ b/sound/soc/rockchip/rk_hdmi_i2s.c
@@ -34,14 +34,14 @@ static int hdmi_i2s_hifi_hw_params(struct snd_pcm_substream *substream,
 	int ret;
 
 	DBG("Enter::%s----%d\n", __FUNCTION__, __LINE__);
-
+#if 0
 	/* set codec DAI configuration */
 	ret = snd_soc_dai_set_fmt(codec_dai, dai_fmt);
 	if (ret < 0) {
 		printk("%s():failed to set the format for codec side\n", __FUNCTION__);
 		return ret;
 	}
-
+#endif
 	/* set cpu DAI configuration */
 	ret = snd_soc_dai_set_fmt(cpu_dai, dai_fmt);
 	if (ret < 0) {
-- 
2.35.3

