From 7991a0e366f2a7c9bb8cd14d2aa6437d0c6f9d4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 25 Jan 2011 17:55:01 +0800
Subject: [PATCH] rk29: do not call BUG when suspend/resume timeout temporarily

---
 drivers/base/power/main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/base/power/main.c b/drivers/base/power/main.c
index 0855a10337ee..82c15c393213 100644
--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -459,7 +459,9 @@ static void dpm_drv_timeout(unsigned long data)
 	printk(KERN_EMERG "dpm suspend stack:\n");
 	show_stack(tsk, NULL);
 
+#ifndef CONFIG_ARCH_RK29
 	BUG();
+#endif
 }
 
 /**
-- 
2.35.3

