From 5ceae838de2f870b51eee873aa3281dc07956a5b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Wed, 26 Sep 2012 18:49:00 +0800
Subject: [PATCH] rk2928:a713:support act8931 dc detection

---
 arch/arm/mach-rk2928/i2c_sram.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/mach-rk2928/i2c_sram.c b/arch/arm/mach-rk2928/i2c_sram.c
index f93a20cfe5ad..11c64771952d 100755
--- a/arch/arm/mach-rk2928/i2c_sram.c
+++ b/arch/arm/mach-rk2928/i2c_sram.c
@@ -349,6 +349,23 @@ void __sramfunc rk30_suspend_voltage_resume(unsigned int vol)
     sram_i2c_write(slaveaddr, slavereg, data);
     sram_i2c_deinit();  //deinit i2c device
 }
+
+int __sramfunc act8931_dc_det(unsigned int vol)
+{
+	uint8 slaveaddr;
+	uint16 slavereg;
+	uint8 ret = 0;
+	int data = 0;
+	slaveaddr = 0x5b;            //slave device addr
+       slavereg = 0x78;            // reg addr  
+       
+   	sram_i2c_init();  //init i2c device
+	ret = sram_i2c_read(slaveaddr,slavereg);
+	data = (ret & (1<<1) )? 1:0;
+	sram_i2c_deinit();  //deinit i2c device
+	return data;
+}
+
 #else
 void __sramfunc rk30_suspend_voltage_set(unsigned int vol)
 {
@@ -357,6 +374,10 @@ void __sramfunc rk30_suspend_voltage_set(unsigned int vol)
 void __sramfunc rk30_suspend_voltage_resume(unsigned int vol)
 {
    
+}
+int __sramfunc act8931_dc_det(unsigned int vol)
+{
+	return -1;
 }
 #endif
 
-- 
2.35.3

