From b69026678fe70127a3da6d65e4fcef55ae9d1617 Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Fri, 26 Oct 2012 17:39:47 +0800
Subject: [PATCH] rk2926 sdk: fix 'tp crash'

---
 drivers/input/touchscreen/sitronix_ts_a720.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/sitronix_ts_a720.c b/drivers/input/touchscreen/sitronix_ts_a720.c
index 5a56dbdd0f67..f4e4a7a9f0c3 100755
--- a/drivers/input/touchscreen/sitronix_ts_a720.c
+++ b/drivers/input/touchscreen/sitronix_ts_a720.c
@@ -926,6 +926,7 @@ static int sitronix_ts_probe(struct i2c_client *client, const struct i2c_device_
         client->irq = irq_cfg.gpio;
         tp_hw_init();
 #else
+	pdata = client->dev.platform_data;
 	if(pdata->init_platform_hw)
 		pdata->init_platform_hw();
 #endif
@@ -949,7 +950,6 @@ static int sitronix_ts_probe(struct i2c_client *client, const struct i2c_device_
         if(client->irq != INVALID_GPIO)
                 ts->irq = gpio_to_irq(client->irq);
 	i2c_set_clientdata(client, ts);
-	pdata = client->dev.platform_data;
 #if 0
 	if(pdata->reset_ic){
 		ts->reset_ic = pdata->reset_ic;
-- 
2.35.3

