From 5d7f2844c0aab91b30523f34ced9afd2054d9d75 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 11 Apr 2012 14:12:48 +0800
Subject: [PATCH] rk30: cpufreq: fix transition_latency for ondemand governor

---
 arch/arm/mach-rk30/cpufreq.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/cpufreq.c b/arch/arm/mach-rk30/cpufreq.c
index e2d8eb3a2319..a2500ff5f052 100755
--- a/arch/arm/mach-rk30/cpufreq.c
+++ b/arch/arm/mach-rk30/cpufreq.c
@@ -246,8 +246,7 @@ static int rk30_cpu_init(struct cpufreq_policy *policy)
 	
 	policy->cur = rk30_getspeed(0);
 
-	/* FIXME: what's the actual transition time? */
-	policy->cpuinfo.transition_latency = 300 * 1000;
+	policy->cpuinfo.transition_latency = 40 * NSEC_PER_USEC; // make ondemand default sampling_rate to 40000
 
 
 	/*
-- 
2.35.3

