From 143fd7744cbfaec9d8e136c7ef8e8afa375b160c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 18 Feb 2013 10:32:41 +0800
Subject: [PATCH] rk3188: Kconfig: depends on ARCH_RK3188

---
 arch/arm/mach-rk3188/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk3188/Kconfig b/arch/arm/mach-rk3188/Kconfig
index 00b85b7ed09f..7ef234b874a8 100644
--- a/arch/arm/mach-rk3188/Kconfig
+++ b/arch/arm/mach-rk3188/Kconfig
@@ -1,14 +1,15 @@
 choice
 	prompt "Rockchip SoC Type"
+	depends on ARCH_RK3188
 
 config SOC_RK3188
 	bool "RK3188"
 
 endchoice
 
-if SOC_RK3188
 choice
 	prompt "RK3188 Board Type"
+	depends on SOC_RK3188
 
 config MACH_RK3188_TB
 	bool "RK3188 Top Board"
@@ -24,4 +25,3 @@ config MACH_RK3188_DS1006H
 	bool "RK3188 ds1006h board"
 
 endchoice
-endif
-- 
2.35.3

