From ba73f493d00de9a973d807e8db965efc520e4c7d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 9 Dec 2011 14:30:42 +0800
Subject: [PATCH] rk29: Temporary disable DDR frequency scaling, which is
 broken

---
 arch/arm/mach-rk29/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk29/Kconfig b/arch/arm/mach-rk29/Kconfig
index 01d8bedbcda1..e28fb75ebd1b 100644
--- a/arch/arm/mach-rk29/Kconfig
+++ b/arch/arm/mach-rk29/Kconfig
@@ -183,7 +183,6 @@ config DDR_SDRAM_FREQ
 
 config DDR_FREQ
 	bool "Enable DDR frequency scaling"
-	default y
 
 config DDR_RECONFIG
 	bool "Enable dynamic DDR reconfiguration (EXPERIMENTAL)"
-- 
2.35.3

