From 55c23da59bcfbfa6c94c2dcd6697cf32b53acca0 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 18 May 2015 17:25:12 +0800
Subject: [PATCH] rockchip: hmp: adjust hmp_up/down_threshold

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 kernel/sched/fair.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 1baf6413a882..43f6a12564ef 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -3810,8 +3810,13 @@ static struct sched_entity *hmp_get_lightest_task(
  * hmp_packing_enabled: runtime control over pack/spread
  * hmp_full_threshold: Consider a CPU with this much unweighted load full
  */
+#ifdef CONFIG_ARCH_ROCKCHIP
+unsigned int hmp_up_threshold = 479;
+unsigned int hmp_down_threshold = 214;
+#else
 unsigned int hmp_up_threshold = 700;
 unsigned int hmp_down_threshold = 512;
+#endif
 #ifdef CONFIG_SCHED_HMP_PRIO_FILTER
 unsigned int hmp_up_prio = NICE_TO_PRIO(CONFIG_SCHED_HMP_PRIO_FILTER_VAL);
 #endif
-- 
2.35.3

