From ba3e311651201a4b85a23aaa4eab09b8894ae5a3 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Wed, 5 Mar 2014 14:07:16 +0800
Subject: [PATCH] key: add rk_send_wakeup_key function

---
 drivers/input/keyboard/rk_keys.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/input/keyboard/rk_keys.c b/drivers/input/keyboard/rk_keys.c
index 5a8b53631e44..8dc4d84e1511 100755
--- a/drivers/input/keyboard/rk_keys.c
+++ b/drivers/input/keyboard/rk_keys.c
@@ -103,6 +103,19 @@ void rk_send_power_key(int state)
 		input_sync(sinput_dev);
 	}
 }
+EXPORT_SYMBOL(rk_send_power_key);
+
+void rk_send_wakeup_key(void)
+{
+	if (!sinput_dev)
+		return;
+
+	input_report_key(sinput_dev, KEY_WAKEUP, 1);
+	input_sync(sinput_dev);
+	input_report_key(sinput_dev, KEY_WAKEUP, 0);
+	input_sync(sinput_dev);
+}
+EXPORT_SYMBOL(rk_send_wakeup_key);
 
 static void keys_timer(unsigned long _data)
 {
-- 
2.35.3

