From 938ad49aa5ecb9633449d313e15a8f41f1a2ee5c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 18 Jul 2013 19:38:28 +0800
Subject: [PATCH] rk: cpu.c support rk3026

---
 arch/arm/plat-rk/cpu.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/plat-rk/cpu.c b/arch/arm/plat-rk/cpu.c
index 139104aa64a1..b4e3fb5e0acf 100644
--- a/arch/arm/plat-rk/cpu.c
+++ b/arch/arm/plat-rk/cpu.c
@@ -10,6 +10,8 @@ static ssize_t show_type(struct sysdev_class *dev, struct sysdev_class_attribute
 		type = "rk3188";
 	else if (cpu_is_rk3066b())
 		type = "rk3066b";
+	else if (cpu_is_rk3026())
+		type = "rk3026";
 	else if (cpu_is_rk30xx())
 		type = "rk30xx";
 	else if (cpu_is_rk2928())
@@ -36,6 +38,10 @@ static ssize_t show_soc(struct sysdev_class *dev, struct sysdev_class_attribute
 		soc = "rk3028";
 	else if (soc_is_rk3066b())
 		soc = "rk3066b";
+	else if (soc_is_rk3028a())
+		soc = "rk3028a";
+	else if (soc_is_rk3026())
+		soc = "rk3026";
 	else if (soc_is_rk2928g())
 		soc = "rk2928g";
 	else if (soc_is_rk2928l())
-- 
2.35.3

