From dd0168495e97b89e62c54098d1162accf4aa4fdc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 2 Jul 2011 18:24:45 +0800
Subject: [PATCH] rk29: mmu.c: set vmalloc_reserve to 256MB for support 1G
 memory

---
 arch/arm/mm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mm/mmu.c b/arch/arm/mm/mmu.c
index 0a24e5e04cb3..36f521e710c0 100644
--- a/arch/arm/mm/mmu.c
+++ b/arch/arm/mm/mmu.c
@@ -665,7 +665,7 @@ void __init iotable_init(struct map_desc *io_desc, int nr)
 }
 
 #if defined(CONFIG_RK29_MEM_SIZE_M) && CONFIG_RK29_MEM_SIZE_M >= 1024
-static unsigned long __initdata vmalloc_reserve = SZ_128M + (0xFF000000 - VMALLOC_END);
+static unsigned long __initdata vmalloc_reserve = SZ_256M;
 #else
 static unsigned long __initdata vmalloc_reserve = SZ_128M;
 #endif
-- 
2.35.3

