From 99a2b9875639358d940726e0f26ceb961583420e Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Fri, 7 Dec 2012 17:28:20 +0800
Subject: [PATCH] rk fb: load screen when open fb

---
 drivers/video/rockchip/rk_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 3014df056770..9bbd3af95437 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -95,6 +95,7 @@ static int rk_fb_open(struct fb_info *info,int user)
     else
     {
     	dev_drv->open(dev_drv,layer_id,1);
+	dev_drv->load_screen(dev_drv,1);
     }
     
     return 0;
-- 
2.35.3

