From 95fa0ee9dcd4633ab25e5b46184cfc0f6b2688b5 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 31 Mar 2014 09:19:50 +0800
Subject: [PATCH] rk fb: close lcd pwr in suspend mode

---
 drivers/video/rockchip/rk_fb.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 7fa73861559b..ec2e916eb47b 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -315,10 +315,9 @@ int rk_disp_pwr_disable(struct rk_lcdc_driver *dev_drv)
 		pwr_ctr_list = list_entry(pos, struct rk_disp_pwr_ctr_list, list);
 		pwr_ctr = &pwr_ctr_list->pwr_ctr;
 		if (pwr_ctr->type == GPIO) {
-			gpio_set_value(pwr_ctr->gpio,pwr_ctr->atv_val);
+			gpio_set_value(pwr_ctr->gpio,!pwr_ctr->atv_val);
 		}
 	}
-
 	return 0;
 }
 
-- 
2.35.3

