From f5910b7b1a04dddc256358ca6ecd0362680fd4ae Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 23 Mar 2012 09:54:38 +0800
Subject: [PATCH] rk30: temp force enable timer clock

---
 arch/arm/mach-rk30/timer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/timer.c b/arch/arm/mach-rk30/timer.c
index 23ab6ef39a35..b9113b4d9231 100644
--- a/arch/arm/mach-rk30/timer.c
+++ b/arch/arm/mach-rk30/timer.c
@@ -206,6 +206,7 @@ static void __init rk30_timer_init(void)
 #ifdef CONFIG_HAVE_ARM_TWD
 	twd_base = RK30_PTIMER_BASE;
 #endif
+	__raw_writel(0x70000, RK30_CRU_BASE + 0xD4);
 	rk30_timer_init_clocksource();
 	rk30_timer_init_clockevent();
 	rk30_sched_clock_init();
-- 
2.35.3

