From 5e26d1f3269d1470cff6b6c92246fe93088a67bb Mon Sep 17 00:00:00 2001
From: Cody Xie <Cody.Xie@rock-chips.com>
Date: Tue, 22 Apr 2014 10:45:33 +0800
Subject: [PATCH] rk3288-p977 : ddr_dvfs set auto_freq=1

---
 arch/arm/boot/dts/rk3288-p977.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index eb67aeb63ad7..a1841194991d 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -572,6 +572,7 @@
 		SYS_STATUS_VIDEO	300000
 		SYS_STATUS_DUALVIEW	500000
 		>;
+	auto_freq=<1>;
 	status="okay";
 };
 
-- 
2.35.3

