From 1308a831a926d5359721f1a4a432cc349846dad5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Fri, 13 Jun 2014 03:20:56 -0700
Subject: [PATCH] rockchip-cpufreq.c: remove warning

---
 drivers/cpufreq/rockchip-cpufreq.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/cpufreq/rockchip-cpufreq.c b/drivers/cpufreq/rockchip-cpufreq.c
index 1c686adaf2e4..b4af52f96c1f 100644
--- a/drivers/cpufreq/rockchip-cpufreq.c
+++ b/drivers/cpufreq/rockchip-cpufreq.c
@@ -381,7 +381,7 @@ static int cpufreq_pm_notifier_event(struct notifier_block *this, unsigned long
 static struct notifier_block cpufreq_pm_notifier = {
 	.notifier_call = cpufreq_pm_notifier_event,
 };
-
+#if 0
 static int cpufreq_reboot_notifier_event(struct notifier_block *this, unsigned long event, void *ptr)
 {
 	struct cpufreq_policy *policy = cpufreq_cpu_get(0);
@@ -395,7 +395,7 @@ static int cpufreq_reboot_notifier_event(struct notifier_block *this, unsigned l
 
 	return NOTIFY_OK;
 }
-
+#endif
 int rockchip_cpufreq_reboot_limit_freq(void)
 {
 	dvfs_disable_temp_limit();
@@ -405,11 +405,11 @@ int rockchip_cpufreq_reboot_limit_freq(void)
 
 	return 0;
 }
-
+#if 0
 static struct notifier_block cpufreq_reboot_notifier = {
 	.notifier_call = cpufreq_reboot_notifier_event,
 };
-
+#endif
 static int clk_pd_vio_notifier_call(struct notifier_block *nb, unsigned long event, void *ptr)
 {
 	switch (event) {
-- 
2.35.3

