From d1afea4d43fa79edc71b5f43e3a9fc598c992223 Mon Sep 17 00:00:00 2001
From: zhanghao <zhanghao@rock-chips.com>
Date: Sat, 2 Jun 2012 16:59:21 +0800
Subject: [PATCH] WIMO : compile as default y

---
 drivers/video/rockchip/Kconfig | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/Kconfig b/drivers/video/rockchip/Kconfig
index f694ffdb3510..83ba332188c3 100644
--- a/drivers/video/rockchip/Kconfig
+++ b/drivers/video/rockchip/Kconfig
@@ -7,8 +7,12 @@ config FB_ROCKCHIP
         ---help---
           Framebuffer driver for rockchip based platform
 config FB_WIMO
-	bool "WiMo support"
+	bool "WiMO support"
 	depends on FB_ROCKCHIP
+	default y
+	help
+	    Support wimo
+
 config LCDC_RK30
         tristate "rk30 lcdc support "
         depends on FB_ROCKCHIP
-- 
2.35.3

