From 1f4f0c757070c6af7c4e0eb30a6ddd34738f17af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 9 Jun 2010 09:00:31 +0000
Subject: [PATCH] remove dsp_clk, dsp driver use dsp_pll_clk directly

---
 arch/arm/mach-rk2818/clock.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index cf48049e5bc1..f50b816982d9 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -715,7 +715,6 @@ static struct clk NAME##_clk = { \
 }
 
 GATE_CLK(arm_core, arm_clk, ARM);
-GATE_CLK(dsp, dsp_pll_clk, DSP);
 GATE_CLK(dma, arm_hclk, DMA);
 GATE_CLK(sramarm, arm_hclk, SRAMARM);
 GATE_CLK(sramdsp, arm_hclk, SRAMDSP);
@@ -899,7 +898,6 @@ static struct clk_lookup clks[] = {
 	CLK1(lcdc_divider),
 
 	CLK1(arm_core),
-	CLK1(dsp),
 	CLK1(dma),
 	CLK1(sramarm),
 	CLK1(sramdsp),
-- 
2.35.3

