From a106c8bab5fe1d236096c46809d25b3b2fd2e11f Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Thu, 3 Jul 2014 18:24:18 +0800
Subject: [PATCH] 3036: for jtag clk

---
 arch/arm/mach-rockchip/rk3036.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk3036.c b/arch/arm/mach-rockchip/rk3036.c
index f0515512eaeb..c9d21e7b8be1 100644
--- a/arch/arm/mach-rockchip/rk3036.c
+++ b/arch/arm/mach-rockchip/rk3036.c
@@ -135,6 +135,8 @@ static void __init rk3036_dt_init_timer(void)
 
 static void __init rk3036_init_late(void)
 {
+    if (rockchip_jtag_enabled)
+    	clk_prepare_enable(clk_get_sys(NULL, "clk_jtag"));
     return;
 }
 
-- 
2.35.3

