From 092493929b2fe750b9d6d394d3709f812081c219 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=AE=8B=E7=A7=80=E6=9D=B0?= <sxj@rock-chips.com>
Date: Thu, 3 Mar 2011 00:51:18 -0800
Subject: [PATCH] close log print in bq27510

---
 drivers/power/bq27510_battery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/bq27510_battery.c b/drivers/power/bq27510_battery.c
index 1e5d17017d92..399c9c43611c 100755
--- a/drivers/power/bq27510_battery.c
+++ b/drivers/power/bq27510_battery.c
@@ -48,7 +48,7 @@
 int  virtual_battery_enable = 0;
 extern int dwc_vbus_status(void);
 
-#if 1
+#if 0
 #define DBG(x...) printk(KERN_INFO x)
 #else
 #define DBG(x...) do { } while (0)
-- 
2.35.3

