From 8c254c6d02f07c0a8f69b14122a2f1d06f778f1c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 20 Feb 2012 14:30:18 +0800
Subject: [PATCH] rk: disable print_modules

---
 kernel/module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/module.c b/kernel/module.c
index eab2ea47c988..4b52684a013e 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -3395,7 +3395,7 @@ EXPORT_SYMBOL_GPL(__module_text_address);
 /* Don't grab lock, we're oopsing. */
 void print_modules(void)
 {
-#ifndef CONFIG_ARCH_RK29
+#ifndef CONFIG_PLAT_RK
 	struct module *mod;
 	char buf[8];
 
-- 
2.35.3

