From 023aa2291a028b8df8552b8a23c7ce65b7c6761f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 14 Mar 2011 21:16:15 +0800
Subject: [PATCH] rk29: support zImage for buggy RK29xxLoader

---
 arch/arm/boot/compressed/head.S | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/compressed/head.S b/arch/arm/boot/compressed/head.S
index 8d42f8ef9d4d..65f5a639c2ca 100644
--- a/arch/arm/boot/compressed/head.S
+++ b/arch/arm/boot/compressed/head.S
@@ -232,6 +232,10 @@ not_relocated:	mov	r0, #0
 		 * pointers, and start decompressing.
 		 */
 		bl	cache_on
+#ifdef CONFIG_ARCH_RK29
+		bl	cache_off
+		bl	cache_on
+#endif
 
 		mov	r1, sp			@ malloc space above stack
 		add	r2, sp, #0x10000	@ 64k max
-- 
2.35.3

