From 046d3d9157339fcc6eac05056012c45b90007ae3 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Thu, 14 Jun 2012 18:22:16 +0800
Subject: [PATCH] rk30:sdk: fix gpu round frequency error when the same rate

---
 arch/arm/mach-rk30/clock_data.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/clock_data.c b/arch/arm/mach-rk30/clock_data.c
index b69c9251f1b0..4633144d4b57 100644
--- a/arch/arm/mach-rk30/clock_data.c
+++ b/arch/arm/mach-rk30/clock_data.c
@@ -406,6 +406,8 @@ static long clk_freediv_round_autosel_parents_rate(struct clk *clk, unsigned lon
 {
 	u32 div;
 	struct clk *p_clk;
+	if(clk->rate == rate)
+		return clk->rate;
 	p_clk=get_freediv_parents_div(clk,rate,&div);
 	if(!p_clk)
 		return 0;
-- 
2.35.3

