From 2e1c681a0458b29611ea463d897955e2e329228d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 20 Apr 2012 19:52:20 +0800
Subject: [PATCH] rk30: setup lcdc1 qos

---
 arch/arm/mach-rk30/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 866284d292a5..0a5c2ca28162 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -27,7 +27,7 @@ static void __init rk30_cpu_axi_init(void)
 	writel_relaxed(0xa, RK30_CPU_AXI_BUS_BASE + 0x7008);
 	writel_relaxed(0x0, RK30_CPU_AXI_BUS_BASE + 0x7088);
 	writel_relaxed(0x0, RK30_CPU_AXI_BUS_BASE + 0x7108);
-	writel_relaxed(0x0, RK30_CPU_AXI_BUS_BASE + 0x7188);
+	writel_relaxed(0xa, RK30_CPU_AXI_BUS_BASE + 0x7188);
 	writel_relaxed(0x0, RK30_CPU_AXI_BUS_BASE + 0x7208);
 	writel_relaxed(0x0, RK30_CPU_AXI_BUS_BASE + 0x7288);
 	writel_relaxed(0x3f, RK30_CPU_AXI_BUS_BASE + 0x0014);
-- 
2.35.3

