From 488fae98bff8ca9ec84e859e01bd40a90259fcef Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Mon, 26 Mar 2012 10:50:59 +0800
Subject: [PATCH] rk30: adc drivers: read: remove udelay

---
 drivers/adc/plat/rk30_adc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/adc/plat/rk30_adc.c b/drivers/adc/plat/rk30_adc.c
index 64cdd559c75e..fbc81f3e1ad1 100755
--- a/drivers/adc/plat/rk30_adc.c
+++ b/drivers/adc/plat/rk30_adc.c
@@ -50,7 +50,6 @@ static int rk30_adc_read(struct adc_host *adc)
 {
 	struct rk30_adc_device *dev  = adc_priv(adc);
 
-	udelay(SAMPLE_RATE);
 	return readl(dev->regs + ADC_DATA) & ADC_DATA_MASK;
 }
 static irqreturn_t rk30_adc_irq(int irq, void *data)
-- 
2.35.3

