From e1be224e4cc415682892b3cf509400f2d22568c5 Mon Sep 17 00:00:00 2001
From: smj <smj@rock-chips.com>
Date: Wed, 4 Mar 2015 16:14:50 +0800
Subject: [PATCH] rk3368 codec : fix the es8316 codec capture gain

Signed-off-by: smj <smj@rock-chips.com>
---
 sound/soc/codecs/es8316.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/es8316.c b/sound/soc/codecs/es8316.c
index 3ba5edf1514a..9873e9500c87 100644
--- a/sound/soc/codecs/es8316.c
+++ b/sound/soc/codecs/es8316.c
@@ -949,7 +949,7 @@ static int es8316_init_regs(struct snd_soc_codec *codec)
 	snd_soc_write(codec, ES8316_TESTMODE_REG50, 0xA0);
 	snd_soc_write(codec, ES8316_TEST2_REG52, 0x03);
 	/*alc set*/
-	snd_soc_write(codec, ES8316_ADC_PGAGAIN_REG23, 0xa0);
+	snd_soc_write(codec, ES8316_ADC_PGAGAIN_REG23, 0x60);
 	snd_soc_write(codec, ES8316_ADC_D2SEPGA_REG24, 0x01);
 	/*adc ds mode, HPF enable*/
 	snd_soc_write(codec, ES8316_ADC_DMIC_REG25, 0x08);
-- 
2.35.3

