From 164ad8d4b67d730dafce461a6fba5638cf2b2fe9 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 12 Sep 2012 20:08:29 +0800
Subject: [PATCH] plat-rk: clock.c clk_set_parent can use default

---
 arch/arm/plat-rk/clock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-rk/clock.c b/arch/arm/plat-rk/clock.c
index 42f2c7f97350..fdd4c941bad6 100755
--- a/arch/arm/plat-rk/clock.c
+++ b/arch/arm/plat-rk/clock.c
@@ -424,7 +424,7 @@ int clk_set_parent(struct clk *clk, struct clk *parent)
 	if (clk == NULL || IS_ERR(clk) || parent == NULL || IS_ERR(parent))
 		return ret;
 
-	if (clk->set_parent==NULL||clk->parents == NULL)
+	if (clk->parents == NULL)
 		return ret;
 
 	LOCK();
@@ -443,7 +443,7 @@ int clk_set_parent_force(struct clk *clk, struct clk *parent)
 	if (clk == NULL || IS_ERR(clk) || parent == NULL || IS_ERR(parent))
 		return ret;
 
-	if (clk->set_parent==NULL||clk->parents == NULL)
+	if (clk->parents == NULL)
 		return ret;
 	LOCK();
 		ret = clk_set_parent_nolock(clk, parent);	
-- 
2.35.3

