From 997cc64cf9d31ffb329c5d6896cb28f365533ce7 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Fri, 8 Jun 2012 15:16:45 +0800
Subject: [PATCH] MTP: solve one cancel problem

---
 drivers/usb/gadget/f_mtp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/f_mtp.c b/drivers/usb/gadget/f_mtp.c
index 4ac7c8394983..034cdf68f9f6 100755
--- a/drivers/usb/gadget/f_mtp.c
+++ b/drivers/usb/gadget/f_mtp.c
@@ -1037,7 +1037,7 @@ static int mtp_ctrlrequest(struct usb_composite_dev *cdev,
 			DBG(cdev, "MTP_REQ_CANCEL %d\n", dev->state);
 
 			spin_lock_irqsave(&dev->lock, flags);
-			if (dev->state == STATE_BUSY) {
+			/*if (dev->state == STATE_BUSY)*/ {
 				dev->state = STATE_CANCELED;
 				wake_up(&dev->read_wq);
 				wake_up(&dev->write_wq);
-- 
2.35.3

