From d7badbbbe3cc216dea8259626588e74c46d14a80 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 10 Jul 2013 18:59:22 +0800
Subject: [PATCH] rk3188: add setting rate in dump clocks

---
 arch/arm/mach-rk3188/clock_data.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-rk3188/clock_data.c b/arch/arm/mach-rk3188/clock_data.c
index 10ddea3c72ea..2f010a80bc56 100755
--- a/arch/arm/mach-rk3188/clock_data.c
+++ b/arch/arm/mach-rk3188/clock_data.c
@@ -3841,6 +3841,9 @@ static void dump_clock(struct seq_file *s, struct clk *clk, int deep, const stru
 	if (clk->parent)
 		seq_printf(s, " parent = %s", clk->parent->name);
 
+	if (clk->last_set_rate != 0)
+		seq_printf(s, " [set %lu Hz]", clk->last_set_rate);
+
 	seq_printf(s, "\n");
 
 	list_for_each_entry(ck, root_clocks, node) {
-- 
2.35.3

