From ac8f55bd7c8c2f20c7ec14ca878e99ba03a765c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 19 Jul 2011 16:24:20 +0800
Subject: [PATCH] rk29: default LZO-compressed kernel for zImage

---
 init/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/Kconfig b/init/Kconfig
index 44421cd0b126..284e7cb8b9ba 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -120,6 +120,7 @@ config HAVE_KERNEL_LZO
 
 choice
 	prompt "Kernel compression mode"
+	default KERNEL_LZO if ARCH_RK29
 	default KERNEL_GZIP
 	depends on HAVE_KERNEL_GZIP || HAVE_KERNEL_BZIP2 || HAVE_KERNEL_LZMA || HAVE_KERNEL_LZO
 	help
-- 
2.35.3

