From e8259dd1892434f337098a5cd060efd439887bcb Mon Sep 17 00:00:00 2001
From: zhengxing <zhengxing@rock-chips.com>
Date: Wed, 17 Oct 2012 15:29:59 +0800
Subject: [PATCH] m701: support save_capacity policy.

---
 arch/arm/mach-rk30/board-rk3066b-m701.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk30/board-rk3066b-m701.c b/arch/arm/mach-rk30/board-rk3066b-m701.c
index 688598f2a3ba..76c04ae56f04 100755
--- a/arch/arm/mach-rk30/board-rk3066b-m701.c
+++ b/arch/arm/mach-rk30/board-rk3066b-m701.c
@@ -1117,6 +1117,7 @@ static struct rk30_adc_battery_platform_data rk30_adc_battery_platdata = {
         .dc_det_level    = GPIO_LOW,
         .charge_ok_level = GPIO_HIGH,
 		.charge_set_level = GPIO_HIGH,
+		.save_capacity   = 1,
 };
 
 static struct platform_device rk30_device_adc_battery = {
-- 
2.35.3

