From 2a3855d1298168c1c1fac303f9e65c725c46f5ba Mon Sep 17 00:00:00 2001
From: swj <swj@rock-chips.com>
Date: Thu, 9 Sep 2010 02:35:56 -0700
Subject: [PATCH] modified t50 adckey

---
 drivers/input/keyboard/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/keyboard/Kconfig b/drivers/input/keyboard/Kconfig
index b99eefca4a35..9bd7d9100378 100755
--- a/drivers/input/keyboard/Kconfig
+++ b/drivers/input/keyboard/Kconfig
@@ -425,21 +425,21 @@ config KEYBOARD_W90P910
 config KEYBOARD_RK28ADC	 
 	tristate "RK2818 ADC Keypad support"
 	depends on RK28_ADC
+	default y
 	help
 	  Say Y here to enable the adc keypad on SDK board
 	  based on RK2818.
-
+	
 	  To compile this driver as a module, choose M here: the
 	  module will be called rk2818_adckey.
 	  
 config KEYBOARD_RK28ADC_IT50	 
 	tristate "RK2818 ADC Keypad it50 support"
 	depends on RK28_ADC
-	default y
 	help
 	  Say Y here to enable the adc keypad on SDK board
 	  based on RK2818.
-
+	 
 	  To compile this driver as a module, choose M here: the
 	  module will be called rk2818_adckey.
 endif
-- 
2.35.3

