From a73af9e69bee576515150bdbc2e0aa2f331fdee0 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Fri, 1 Aug 2014 16:55:15 +0800
Subject: [PATCH] mmc: rk_sdmmc: increase max_blk_count for edmac

---
 drivers/mmc/host/rk_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 3a127c624534..0bdaf81e81f0 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3267,7 +3267,7 @@ static int dw_mci_init_slot(struct dw_mci *host, unsigned int id)
                         /* fixup for external dmac setting */
                         mmc->max_segs = 64;
 		        mmc->max_blk_size = 65536; /* BLKSIZ is 16 bits */
-		        mmc->max_blk_count = 512;
+		        mmc->max_blk_count = 65535;
 		        mmc->max_req_size = mmc->max_blk_size * mmc->max_blk_count;
 		        mmc->max_seg_size = mmc->max_req_size; 
                 }
-- 
2.35.3

