From 2752baa46bd2a6385b4b6b9552f3374be5aa46ae Mon Sep 17 00:00:00 2001
From: yzq <yzq@rock-chips.com>
Date: Thu, 15 Nov 2012 11:33:18 +0800
Subject: [PATCH] rk fb:fix NO_DUAL_DISP mode hdmi scale problem

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 33514f2408ce..3d632eb969f1 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -814,7 +814,7 @@ int rk_fb_disp_scale(u8 scale_x, u8 scale_y,u8 lcdc_id)
 	screen_x = dev_drv->cur_screen->x_res;
 	screen_y = dev_drv->cur_screen->y_res;
 	
-#if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)
+#if defined(CONFIG_ONE_LCDC_DUAL_OUTPUT_INF)||defined(CONFIG_NO_DUAL_DISP)
 	if(dev_drv->cur_screen->screen_id == 1){
 		dev_drv->cur_screen->xpos = (screen_x-screen_x*scale_x/100)>>1;
 		dev_drv->cur_screen->ypos = (screen_y-screen_y*scale_y/100)>>1;
-- 
2.35.3

