From 5ee66003f164888aa984b31ecc6238ae3f2970b9 Mon Sep 17 00:00:00 2001
From: cmc <cmc@rock-chips.com>
Date: Mon, 29 Nov 2010 15:21:12 +0800
Subject: [PATCH] adjust kconfig for rk29 spi

---
 drivers/input/touchscreen/Kconfig | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/Kconfig b/drivers/input/touchscreen/Kconfig
index 787f68400aea..7ad72bdef7cf 100755
--- a/drivers/input/touchscreen/Kconfig
+++ b/drivers/input/touchscreen/Kconfig
@@ -20,19 +20,19 @@ choice
   
   config TOUCHSCREEN_XPT2046_SPI
   bool "800X480 TOUCHSCREEN"
-	depends on SPIM_RK2818 || SPIM_RK29XX
+	depends on SPIM_RK2818 || SPIM_RK29
 
   config TOUCHSCREEN_XPT2046_CBN_SPI
   bool "800X480 CALIBRATION TOUCHSCREEN"
-  depends on SPIM_RK2818 || SPIM_RK29XX
+  depends on SPIM_RK2818 || SPIM_RK29
   
   config TOUCHSCREEN_XPT2046_320X480_SPI
   bool "320X480 TOUCHSCREEN"
-  depends on SPIM_RK2818 || SPIM_RK29XX
+  depends on SPIM_RK2818 || SPIM_RK29
   
   config TOUCHSCREEN_XPT2046_320X480_CBN_SPI
 	bool "320X480 CALIBRATION TOUCHSCREEN"
-	depends on SPIM_RK2818 || SPIM_RK29XX
+	depends on SPIM_RK2818 || SPIM_RK29
 	
 endchoice
 config TOUCHSCREEN_ADS7846
-- 
2.35.3

