From 60d69679dd89ce0cf1e9e57a3c1aadbe5422b6cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 31 Jan 2013 11:29:22 +0800
Subject: [PATCH] rk2928: no use Thumb instruction set

---
 arch/arm/mach-rk2928/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/mach-rk2928/Makefile b/arch/arm/mach-rk2928/Makefile
index 96b8d68a4d82..6be4f090ad10 100755
--- a/arch/arm/mach-rk2928/Makefile
+++ b/arch/arm/mach-rk2928/Makefile
@@ -11,7 +11,6 @@ obj-y += ddr.o
 obj-$(CONFIG_CPU_FREQ) += cpufreq.o
 obj-$(CONFIG_DVFS) += dvfs.o
 obj-$(CONFIG_PM) += pm.o
-CFLAGS_pm.o += -Os -mthumb
 obj-$(CONFIG_RK30_I2C_INSRAM) += i2c_sram.o
 
 board-y := board-rk2928.o
-- 
2.35.3

