From 9ab8fff395b4dcbaa554cb27982ea55ea393e603 Mon Sep 17 00:00:00 2001
From: Tang Yun ping <typ@rock-chips.com>
Date: Mon, 24 Aug 2015 10:55:51 +0800
Subject: [PATCH] mailbox: rk3868 change max_chan_num attribute to static

Signed-off-by: Tang Yun ping <typ@rock-chips.com>
---
 drivers/mailbox/scpi_protocol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mailbox/scpi_protocol.c b/drivers/mailbox/scpi_protocol.c
index 0404743fc617..771746396501 100644
--- a/drivers/mailbox/scpi_protocol.c
+++ b/drivers/mailbox/scpi_protocol.c
@@ -48,7 +48,7 @@
 #define DVFS_LATENCY(hdr)	((hdr) >> 16)
 #define DVFS_OPP_COUNT(hdr)	(((hdr) >> 8) & 0xff)
 
-int max_chan_num = 0;
+static int max_chan_num = 0;
 static DECLARE_BITMAP(bm_mbox_chans, 4);
 static DEFINE_MUTEX(scpi_mtx);
 
-- 
2.35.3

