From 680e2073b72b9009e0a807c8e54fd4e3263cfae6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=AE=B8=E7=9B=9B=E9=A3=9E?= <xsf@rock-chips.com>
Date: Mon, 10 Nov 2014 09:26:03 +0800
Subject: [PATCH] pwm: add min-duty = 0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: 许盛飞 <xsf@rock-chips.com>
---
 drivers/pwm/pwm-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pwm/pwm-rockchip.c b/drivers/pwm/pwm-rockchip.c
index 137b58ab230d..d44f1fa47cda 100644
--- a/drivers/pwm/pwm-rockchip.c
+++ b/drivers/pwm/pwm-rockchip.c
@@ -81,7 +81,7 @@ module_param_named(dbg_level, pwm_dbg_level, int, 0644);
 #define PWMCR_MIN_PRESCALE	0x00
 #define PWMCR_MAX_PRESCALE	0x07
 
-#define PWMDCR_MIN_DUTY		0x0001
+#define PWMDCR_MIN_DUTY		0x0000
 #define PWMDCR_MAX_DUTY		0xFFFF
 
 #define PWMPCR_MIN_PERIOD		0x0001
-- 
2.35.3

