From 9fa7b95c04ca71c0848138ab819c222ab3aacf41 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Wed, 14 Nov 2012 10:31:22 +0800
Subject: [PATCH] rk30xx: remove safety log in check_volt_correct function

---
 arch/arm/mach-rk30/dvfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 4cacf70b8095..faf977f574c0 100755
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -756,7 +756,7 @@ int check_volt_correct(int volt_old, int *volt_new, int volt_dep_old, int *volt_
 		} else if (*volt_dep_new > up_boundary) {
 			*volt_new = *volt_dep_new - dep_biger_than_clk;
 		}
-		DVFS_LOG("%s target volt out of bondary volt=%d(old=%d), volt_dep=%d(dep_old=%d), up_bnd=%d(dn=%d)\n",
+		DVFS_DBG("%s target volt out of bondary volt=%d(old=%d), volt_dep=%d(dep_old=%d), up_bnd=%d(dn=%d)\n",
 				__func__, *volt_new, volt_old, *volt_dep_new, volt_dep_old, up_boundary, low_boundary);		
 		return 0;
 	}
-- 
2.35.3

