From c9d9ae14f1f808aa13e69ac920f945501fa89016 Mon Sep 17 00:00:00 2001
From: dkl <dkl@rock-chips.com>
Date: Wed, 30 Jul 2014 19:54:01 +0800
Subject: [PATCH] rk312x dts: add reset node in dts

---
 arch/arm/boot/dts/rk312x.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 1485704d82fe..0d685eb55f84 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -109,6 +109,12 @@
 		};
 	};
 
+	reset: reset@20000110 {
+		compatible = "rockchip,reset";
+		reg = <0x20000110 0x24>;
+		rockchip,reset-flag = <ROCKCHIP_RESET_HIWORD_MASK>;
+		#reset-cells = <1>;
+	};
 
 	nandc: nandc@10500000 {
 		compatible = "rockchip,rk-nandc";
-- 
2.35.3

