From 38ec5b9c0612b0ef7d6f71b9799c900ee8108dc6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=94=A1=E6=9E=AB?= <cf@rock-chips.com>
Date: Tue, 9 Aug 2011 21:21:01 +0800
Subject: [PATCH] newton:fix gt819 drop point

---
 drivers/input/touchscreen/gt819.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/gt819.c b/drivers/input/touchscreen/gt819.c
index 76de84ba119a..8c0188b00827 100755
--- a/drivers/input/touchscreen/gt819.c
+++ b/drivers/input/touchscreen/gt819.c
@@ -447,9 +447,14 @@ static void gt819_queue_work(struct work_struct *work)
 		enable_irq(ts->irq);
 		dev_info(&ts->client->dev, "touch release\n");
 		return; 
+	}	
+	for(i=0;0!=points;)
+	{
+	if(points&0x01)
+		i++;
+	points>>=1;
 	}
-	for(i=0;0!=points;i++)
-		points>>=1;
+	
 	points = i;
 	points_chect = points;
 	ret = gt819_read_regs(ts->client,3, point_data, points*5+1);
-- 
2.35.3

