From 5c13de701aead206808943fa3a7cd6efca47b9cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 29 Aug 2012 18:16:04 +0800
Subject: [PATCH] rk30: boot mode: set recovery mode do not use command line

---
 arch/arm/mach-rk30/common.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-rk30/common.c b/arch/arm/mach-rk30/common.c
index 0663ec8cae77..21775fb0d53b 100755
--- a/arch/arm/mach-rk30/common.c
+++ b/arch/arm/mach-rk30/common.c
@@ -96,8 +96,6 @@ static void __init rk30_boot_mode_init(void)
 
 	if (boot_flag == (SYS_KERNRL_REBOOT_FLAG | BOOT_RECOVER)) {
 		boot_mode = BOOT_MODE_RECOVERY;
-	} else if (strstr(boot_command_line, "(parameter)")) {
-		boot_mode = BOOT_MODE_RECOVERY;
 	}
 	if (boot_mode || boot_flag)
 		printk("Boot mode: %d flag: 0x%08x\n", boot_mode, boot_flag);
-- 
2.35.3

