From b765844f6bd97cac7bbf3f530d352d3bd6718733 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Mon, 28 Jan 2013 17:59:55 +0800
Subject: [PATCH] rk3168_86v: add vol+ and vol- key

---
 arch/arm/mach-rk30/board-rk3168-86v.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/mach-rk30/board-rk3168-86v.c b/arch/arm/mach-rk30/board-rk3168-86v.c
index 4fe2ed589087..e1de29dde0bd 100644
--- a/arch/arm/mach-rk30/board-rk3168-86v.c
+++ b/arch/arm/mach-rk30/board-rk3168-86v.c
@@ -87,6 +87,20 @@ static struct rk29_keys_button key_button[] = {
 		.gpio = INVALID_GPIO,
 		.active_low = PRESS_LEV_LOW,
 	},
+	{
+                .desc   = "vol-",
+                .code   = KEY_VOLUMEDOWN,
+                .adc_value      = 150,
+                .gpio   = INVALID_GPIO,
+                .active_low = PRESS_LEV_LOW,
+        },
+       {
+                .desc   = "vol+",
+                .code   = KEY_VOLUMEUP,
+                .adc_value      = 1,
+                .gpio = INVALID_GPIO,
+                .active_low = PRESS_LEV_LOW,
+        },
 };
 struct rk29_keys_platform_data rk29_keys_pdata = {
 	.buttons	= key_button,
-- 
2.35.3

