From e2c1923a259c351c9764b9bbeb6053b74d03d1f2 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 20 Aug 2012 20:31:16 +0800
Subject: [PATCH] rk2928:sdk: default disable gpio clk

---
 arch/arm/mach-rk2928/clock_data.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm/mach-rk2928/clock_data.c b/arch/arm/mach-rk2928/clock_data.c
index a052019f5040..ab9105c758c7 100644
--- a/arch/arm/mach-rk2928/clock_data.c
+++ b/arch/arm/mach-rk2928/clock_data.c
@@ -2226,10 +2226,6 @@ static void __init rk30_init_enable_clocks(void)
 	clk_enable_nolock(&clk_pclk_pwm01);
 	clk_enable_nolock(&clk_hclk_otg0);
 	clk_enable_nolock(&clk_hclk_otg1);
-	clk_enable_nolock(&clk_gpio0);
-	clk_enable_nolock(&clk_gpio1);
-	clk_enable_nolock(&clk_gpio2);
-	clk_enable_nolock(&clk_gpio3);
 
 #if CONFIG_RK_DEBUG_UART == 0
 	clk_enable_nolock(&clk_uart0);
-- 
2.35.3

