From d8c1ed909ad400df8bccb2c9eee224d7559b4485 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=82=B1=E5=BB=BA=E6=96=8C?= <qjb@rock-chips.com>
Date: Thu, 25 Oct 2012 19:00:45 +0800
Subject: [PATCH] rk2928phonepad  modem_sound : 	add set volume ioctl cmd

---
 drivers/misc/modem_sound.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/modem_sound.h b/drivers/misc/modem_sound.h
index 4f03327bf11c..6de17fd20c35 100755
--- a/drivers/misc/modem_sound.h
+++ b/drivers/misc/modem_sound.h
@@ -9,7 +9,13 @@
 #define IOCTL_MODEM_SPK_PHONE      	        _IO(MODEM_SOUND, 0x02) 
 #define IOCTL_MODEM_HP_PHONE      	        _IO(MODEM_SOUND, 0x03)
 #define IOCTL_MODEM_BT_PHONE      	        _IO(MODEM_SOUND, 0x04)
-#define IOCTL_MODEM_STOP_PHONE      	        _IO(MODEM_SOUND, 0x05) 
+#define IOCTL_MODEM_STOP_PHONE      	    _IO(MODEM_SOUND, 0x05) 
+
+#define IOCTL_SET_EAR_VALUME      	    _IO(MODEM_SOUND, 0x11) 
+#define IOCTL_SET_SPK_VALUME      	    _IO(MODEM_SOUND, 0x12) 
+#define IOCTL_SET_HP_VALUME      	    _IO(MODEM_SOUND, 0x13) 
+#define IOCTL_SET_BT_VALUME      	    _IO(MODEM_SOUND, 0x14) 
+ 
 
 struct modem_sound_data {
 	int spkctl_io;
-- 
2.35.3

