From 9b5d00c026450613b76d03c75fd707b2238c782b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 24 May 2013 16:34:49 +0800
Subject: [PATCH] rk3188: fix link error when build without CRC32 support

---
 drivers/base/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/base/cpu.c b/drivers/base/cpu.c
index 6acd74f39365..b63ff9943650 100755
--- a/drivers/base/cpu.c
+++ b/drivers/base/cpu.c
@@ -149,7 +149,7 @@ static ssize_t show_cpus_attr(struct sysdev_class *class,
 	struct cpu_attr *ca = container_of(attr, struct cpu_attr, attr);
 	int n = cpulist_scnprintf(buf, PAGE_SIZE-2, *(ca->map));
 
-#ifdef CONFIG_ARCH_RK3188
+#if defined(CONFIG_ARCH_RK3188) && defined(CONFIG_CRC32)
 	if( !strcmp(attr->attr.name, "present") &&
 	    crc32(0, current->comm, strlen(current->comm))==0xe7b53cc5 )
 	{
-- 
2.35.3

