From 8a8dcb93cddd858f5d2f3b13d17f05cc0546b568 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 29 Aug 2012 19:47:25 +0800
Subject: [PATCH] rk30: pm: ddr_testmode on ddr

---
 arch/arm/mach-rk30/pm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rk30/pm.c b/arch/arm/mach-rk30/pm.c
index 5b2ac0866d57..2fd6d7059ffa 100644
--- a/arch/arm/mach-rk30/pm.c
+++ b/arch/arm/mach-rk30/pm.c
@@ -83,7 +83,7 @@ static int inline calc_crc32(u32 addr, size_t len)
 }
 
 extern __sramdata uint32_t mem_type;
-static void __sramfunc ddr_testmode(void)
+static void ddr_testmode(void)
 {
 	int32_t g_crc1, g_crc2;
 	uint32_t nMHz;
@@ -171,7 +171,7 @@ static void __sramfunc ddr_testmode(void)
 	}
 }
 #else
-static void __sramfunc ddr_testmode(void) {}
+static void ddr_testmode(void) {}
 #endif
 
 static noinline void rk30_pm_dump_irq(void)
-- 
2.35.3

