From b7574401ec571097859d39b9b1bc9f369609f955 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 17 Mar 2015 14:31:01 +0800
Subject: [PATCH] rk fb: compatible with 32 bit system

Signed-off-by: hjc <hjc@rock-chips.com>
---
 include/linux/rk_fb.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/rk_fb.h b/include/linux/rk_fb.h
index 6f3347b7fae5..f20f9978be21 100755
--- a/include/linux/rk_fb.h
+++ b/include/linux/rk_fb.h
@@ -479,7 +479,7 @@ struct rk_lcdc_drv_ops {
 struct rk_fb_area_par {
 	u8  data_format;        /*layer data fmt*/
 	short ion_fd;
-	unsigned long phy_addr;
+	u32 phy_addr;
 	short acq_fence_fd;
 	u16  x_offset;
 	u16  y_offset;
-- 
2.35.3

