From e80d6b4e95e718e7f8a301f5ad801c13878806d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Mon, 1 Aug 2011 11:16:18 +0800
Subject: [PATCH] arm: dma: more fix for v2.6.36

---
 arch/arm/kernel/dma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/kernel/dma.c b/arch/arm/kernel/dma.c
index caf60d84e08c..25e768a2c6f6 100644
--- a/arch/arm/kernel/dma.c
+++ b/arch/arm/kernel/dma.c
@@ -288,6 +288,7 @@ void get_dma_position(unsigned int chan, dma_addr_t *src_pos, dma_addr_t *dst_po
 }
 
 EXPORT_SYMBOL(dma_channel_active);
+
 #if 0
 void set_dma_page(unsigned int chan, char pagenr)
 {
@@ -317,6 +318,7 @@ int get_dma_residue(unsigned int chan)
 	return ret;
 }
 EXPORT_SYMBOL(get_dma_residue);
+#endif
 
 #ifdef CONFIG_PROC_FS
 static int proc_dma_show(struct seq_file *m, void *v)
-- 
2.35.3

