From 0957deed0ed7a77ffbac3ce53568df5fa9a3ef6d Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 23 Mar 2015 17:16:25 +0800
Subject: [PATCH] rk: arm64: disable set pm_power_off for compatible with old
 pmic driver

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/kernel/psci.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/kernel/psci.c b/arch/arm64/kernel/psci.c
index aef795214929..1cf83c2033e1 100644
--- a/arch/arm64/kernel/psci.c
+++ b/arch/arm64/kernel/psci.c
@@ -359,9 +359,11 @@ static int __init psci_0_2_init(struct device_node *np)
 		PSCI_0_2_FN_MIGRATE_INFO_TYPE;
 	psci_ops.migrate_info_type = psci_migrate_info_type;
 
+#ifndef CONFIG_ARCH_ROCKCHIP
 	arm_pm_restart = psci_sys_reset;
 
 	pm_power_off = psci_sys_poweroff;
+#endif
 
 out_put_node:
 	of_node_put(np);
-- 
2.35.3

