From 7f96362ce678f301989714e01dbd6b92c8f79b26 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 27 Mar 2013 16:36:44 +0800
Subject: [PATCH] rk2928: Makefile: fix board define

---
 arch/arm/mach-rk2928/Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-rk2928/Makefile b/arch/arm/mach-rk2928/Makefile
index a16912f8dc3a..dfc6f41314f5 100755
--- a/arch/arm/mach-rk2928/Makefile
+++ b/arch/arm/mach-rk2928/Makefile
@@ -17,11 +17,11 @@ obj-y += board.o
 
 board-$(CONFIG_MACH_RK2928) := board-rk2928.o
 
-obj-$(CONFIG_MACH_RK2928_SDK) += board-rk2928-sdk.o
-obj-$(CONFIG_MACH_RK2926_SDK) += board-rk2926-sdk.o
+board-$(CONFIG_MACH_RK2928_SDK) := board-rk2928-sdk.o
+board-$(CONFIG_MACH_RK2926_SDK) := board-rk2926-sdk.o
 board-$(CONFIG_MACH_RK2928_PHONEPAD) := board-rk2928-phonepad.o
 board-$(CONFIG_MACH_RK2928_PHONEPAD_760) := board-rk2928-phonepad-760.o
-obj-$(CONFIG_MACH_RK2928_A720) += board-rk2928-a720.o
-obj-$(CONFIG_MACH_RK2928_TB) += board-rk2928-tb.o
-obj-$(CONFIG_MACH_RK2926_TB) += board-rk2928-tb.o
-obj-$(CONFIG_MACH_RK2928_FPGA) += board-rk2928-fpga.o
+board-$(CONFIG_MACH_RK2928_A720) := board-rk2928-a720.o
+board-$(CONFIG_MACH_RK2928_TB) := board-rk2928-tb.o
+board-$(CONFIG_MACH_RK2926_TB) := board-rk2928-tb.o
+board-$(CONFIG_MACH_RK2928_FPGA) := board-rk2928-fpga.o
-- 
2.35.3

