From c99ed0e7d00d0b11f120b61d86c14b800f876cd2 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Mon, 7 Jan 2013 19:35:53 +0800
Subject: [PATCH] rk30xx: dvfs enable clk do not set voltage

---
 arch/arm/mach-rk30/dvfs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk30/dvfs.c b/arch/arm/mach-rk30/dvfs.c
index 1439e9e3504f..b9066b23e3a0 100644
--- a/arch/arm/mach-rk30/dvfs.c
+++ b/arch/arm/mach-rk30/dvfs.c
@@ -404,6 +404,7 @@ int clk_enable_dvfs(struct clk *clk)
 		}
 #endif
 		dvfs_vd_get_newvolt_byclk(dvfs_clk);
+#if 0
 		if(dvfs_clk->vd->cur_volt < dvfs_clk->set_volt) {
 			int ret;
 			mutex_lock(&rk_dvfs_mutex);
@@ -418,6 +419,7 @@ int clk_enable_dvfs(struct clk *clk)
 			dvfs_clk->vd->volt_set_flag = DVFS_SET_VOLT_SUCCESS;
 			mutex_unlock(&rk_dvfs_mutex);
 		}
+#endif
 		dvfs_clk->enable_dvfs++;
 	} else {
 		DVFS_ERR("dvfs already enable clk enable = %d!\n", dvfs_clk->enable_dvfs);
-- 
2.35.3

