From 2b6bc330c1324081a624aa73ebbde7fb434311e0 Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Thu, 27 Dec 2012 15:31:51 +0800
Subject: [PATCH] vflash: modify out of memory bug

---
 drivers/bluetooth/vflash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bluetooth/vflash.c b/drivers/bluetooth/vflash.c
index e889e823f7cb..266ea175f5c8 100755
--- a/drivers/bluetooth/vflash.c
+++ b/drivers/bluetooth/vflash.c
@@ -78,7 +78,7 @@ static long vflash_ioctl(struct file *file,
             }
 #else
             char *tempBuf = (char *)kmalloc(512, GFP_KERNEL);
-	    char bd_addr[6] = {0};
+	    char bd_addr[7] = {0};
             int i;
 
             GetSNSectorInfo(tempBuf);
-- 
2.35.3

