From 0cfeec32da0ef1dd00148e51dc327645e8a455e9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E4=BA=AE?= <cl@rock-chips.com>
Date: Tue, 29 Apr 2014 20:09:27 -0700
Subject: [PATCH] rk3288: release temp limit in time when temp fall down

---
 arch/arm/mach-rockchip/dvfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/dvfs.c b/arch/arm/mach-rockchip/dvfs.c
index 8ca36a6fbbbf..1b52d4782782 100644
--- a/arch/arm/mach-rockchip/dvfs.c
+++ b/arch/arm/mach-rockchip/dvfs.c
@@ -504,9 +504,9 @@ static void dvfs_temp_limit_work_func(struct work_struct *work)
 		for (i=0; clk_cpu_dvfs_node->per_temp_limit_table[i].frequency != CPUFREQ_TABLE_END; i++) {
 			if (temp > clk_cpu_dvfs_node->per_temp_limit_table[i].index) {
 				clk_cpu_dvfs_node->temp_limit_rate = clk_cpu_dvfs_node->per_temp_limit_table[i].frequency;
-				dvfs_clk_set_rate(clk_cpu_dvfs_node, clk_cpu_dvfs_node->last_set_rate);
 			}
 		}
+		dvfs_clk_set_rate(clk_cpu_dvfs_node, clk_cpu_dvfs_node->last_set_rate);
 	} else if (ROCKCHIP_PM_POLICY_NORMAL == rockchip_pm_get_policy()){
 		if (!clk_cpu_dvfs_node->nor_temp_limit_table) {
 			return;
-- 
2.35.3

