From dbe3cb9f74d2e47b0e716f3acd8052418a033c9b Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Sun, 19 Aug 2012 15:57:41 +0800
Subject: [PATCH] rk2928:a720: fix vdd_log dvfs error

---
 arch/arm/mach-rk2928/dvfs.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-rk2928/dvfs.c b/arch/arm/mach-rk2928/dvfs.c
index 5e1111939953..ff74f32a2fa4 100644
--- a/arch/arm/mach-rk2928/dvfs.c
+++ b/arch/arm/mach-rk2928/dvfs.c
@@ -1167,8 +1167,9 @@ int dvfs_target_core(struct clk *clk, unsigned long rate_hz)
 		if (volt_dep_new < 0) 
 			goto fail_roll_back;
 
-		ret = dvfs_scale_volt_bystep(dvfs_clk->vd, dvfs_clk_cpu->vd, volt_new, volt_dep_new, 
-					LOGIC_HIGHER_ARM, ARM_HIGHER_LOGIC); 
+		ret = dvfs_scale_volt_direct(dvfs_clk->vd, volt_new);
+		//ret = dvfs_scale_volt_bystep(dvfs_clk->vd, dvfs_clk_cpu->vd, volt_new, volt_dep_new, 
+		//			LOGIC_HIGHER_ARM, ARM_HIGHER_LOGIC); 
 		if (ret < 0) 
 			goto fail_roll_back;
 	}
@@ -1197,8 +1198,9 @@ int dvfs_target_core(struct clk *clk, unsigned long rate_hz)
 		if (volt_dep_new < 0) 
 			goto out;
 
-		ret = dvfs_scale_volt_bystep(dvfs_clk->vd, dvfs_clk_cpu->vd, volt_new, volt_dep_new, 
-					LOGIC_HIGHER_ARM, ARM_HIGHER_LOGIC); 
+		ret = dvfs_scale_volt_direct(dvfs_clk->vd, volt_new);
+		//ret = dvfs_scale_volt_bystep(dvfs_clk->vd, dvfs_clk_cpu->vd, volt_new, volt_dep_new, 
+		//			LOGIC_HIGHER_ARM, ARM_HIGHER_LOGIC); 
 		if (ret < 0) 
 			goto out;
 	}
-- 
2.35.3

