From fcbf1111baa0cea2efea18387d268471aafc294f Mon Sep 17 00:00:00 2001
From: xiaoyao <xiaoyao@rock-chips.com>
Date: Mon, 20 Apr 2015 08:53:34 +0800
Subject: [PATCH] mmc: rk_sdmmc: delete rk3126b burst_limit

tested-by: xiaoyao <xiaoyao@rock-chips.com>
Reviewed-by: lintao <lintao@rock-chips.com>
Signed-off-by: xiaoyao <xiaoyao@rock-chips.com>
---
 drivers/mmc/host/rk_sdmmc.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index e4488a8de558..6f68cd437830 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -643,13 +643,7 @@ static void dw_mci_edmac_start_dma(struct dw_mci *host, unsigned int sg_len)
 		mburst = burst_limit;
 		idx = (ilog2(mburst) > 0) ? (ilog2(mburst) - 1) : 0;
 
-		if (soc_is_rk3126b()) {
-			idx = 0;
-			rx_wmark = (host->fifo_depth) / 2 - 1;
-		} else {
-			rx_wmark = mszs[idx] - 1;
-		}
-
+		rx_wmark = mszs[idx] - 1;
 		tx_wmark = (host->fifo_depth) / 2;
 		fifoth_val = SDMMC_SET_FIFOTH(idx, rx_wmark, tx_wmark);
 
-- 
2.35.3

