From a8c420da85d0b75ac4829de2aef443f9bb824e07 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Thu, 24 Jan 2013 10:40:42 +0800
Subject: [PATCH] rk fb:all fb address can be set by user ioctl

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 77f7872007bc..2b8297d948f4 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -284,7 +284,7 @@ static int rk_fb_ioctl(struct fb_info *info, unsigned int cmd,unsigned long arg)
  		case FBIOPUT_FBPHYADD:
 			return info->fix.smem_start;
 		case FBIOSET_YUV_ADDR:   //when in video mode, buff alloc by android
-			if((!strcmp(fix->id,"fb1"))||(!strcmp(fix->id,"fb3")))
+			//if((!strcmp(fix->id,"fb1"))||(!strcmp(fix->id,"fb3")))
 			{
 				if (copy_from_user(yuv_phy, argp, 8))
 					return -EFAULT;
-- 
2.35.3

