From f601ea7f11385321672cf24f00c00b806bfc7d21 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Mon, 24 Feb 2014 11:31:49 +0800
Subject: [PATCH] rk: ion: add system heap

---
 arch/arm/boot/dts/rk3188.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/rk3188.dtsi b/arch/arm/boot/dts/rk3188.dtsi
index e2df5022e5b4..5242cf46b129 100755
--- a/arch/arm/boot/dts/rk3188.dtsi
+++ b/arch/arm/boot/dts/rk3188.dtsi
@@ -544,6 +544,9 @@
 		rockchip,ion-heap@1 { /* CMA HEAP */
 			reg = <1>;
 		};
+		rockchip,ion-heap@3 { /* SYSTEM HEAP */
+			reg = <3>;
+		};
 	};
 
 	dwc_control_usb: dwc-control-usb@0x200080ac {
-- 
2.35.3

