From a30c015d9d3b8b26960f7035ad0fc8cf26b2ce57 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 30 Jun 2014 10:03:26 +0800
Subject: [PATCH] mmc: host: rk_sdmmc: fix re-put count of dma

---
 drivers/mmc/host/rk_sdmmc.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/mmc/host/rk_sdmmc.c b/drivers/mmc/host/rk_sdmmc.c
index 1c02aa9fa3b7..d9272d28029b 100755
--- a/drivers/mmc/host/rk_sdmmc.c
+++ b/drivers/mmc/host/rk_sdmmc.c
@@ -3839,9 +3839,6 @@ void dw_mci_remove(struct dw_mci *host)
 	int i;
 	del_timer_sync(&host->dto_timer);
 
-	if(host->use_dma && host->dma_ops->exit)
-	        host->dma_ops->exit(host);
-
         mci_writel(host, RINTSTS, 0xFFFFFFFF);
         mci_writel(host, INTMASK, 0); /* disable all mmc interrupt first */
 
-- 
2.35.3

