From 081b9f69e5a88cff1dd5fe1cbcf77f8017961797 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 28 Mar 2012 09:53:21 +0800
Subject: [PATCH] rk30: reboot use the second global software reset

---
 arch/arm/mach-rk30/reset.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-rk30/reset.c b/arch/arm/mach-rk30/reset.c
index 4b677d04e971..b826ad78917a 100644
--- a/arch/arm/mach-rk30/reset.c
+++ b/arch/arm/mach-rk30/reset.c
@@ -1,7 +1,16 @@
+#include <linux/io.h>
+#include <linux/kernel.h>
 #include <mach/system.h>
+#include <mach/cru.h>
+#include <mach/iomux.h>
 
 static void rk30_arch_reset(char mode, const char *cmd)
 {
+	/* disable remap */
+	writel_relaxed(1 << (12 + 16), RK30_GRF_BASE + GRF_SOC_CON0);
+	dsb();
+	writel_relaxed(0xeca8, RK30_CRU_BASE + CRU_GLB_SRST_SND);
+	dsb();
 }
 
 void (*arch_reset)(char, const char *) = rk30_arch_reset;
-- 
2.35.3

