From 7ca158d2bb524990d3ebe874cc023ea8cb7d452c Mon Sep 17 00:00:00 2001
From: wlq <wlq@rock-chips.com>
Date: Thu, 8 Sep 2011 14:20:04 +0800
Subject: [PATCH] phone-pad:add ThinkWill MI900 product ID

---
 drivers/usb/serial/option.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index a7ab681063aa..c7bc9be950d2 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -373,6 +373,7 @@ static int  option_resume(struct usb_serial *serial);
 /* Thinkwill products */
 #define THINKWILL_VENDOR_ID 	0x19f5
 #define THINKWILL_PRODUCT_ID	0x9909
+#define	THINKWILL_MI900_PRODUCT_ID	0x9013
 
 /* Cinterion (formerly Siemens) products */
 #define SIEMENS_VENDOR_ID				0x0681
@@ -412,6 +413,9 @@ static int  option_resume(struct usb_serial *serial);
 static struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_COLT) },
 	{ USB_DEVICE(THINKWILL_VENDOR_ID,THINKWILL_PRODUCT_ID)},
+
+	{ USB_DEVICE(THINKWILL_VENDOR_ID,THINKWILL_MI900_PRODUCT_ID)},
+
 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_RICOLA) },
 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_RICOLA_LIGHT) },
 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_RICOLA_QUAD) },
-- 
2.35.3

