From 57cebea06aafb4a162139f00b302c51afa04ce0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 19 Jul 2011 16:50:47 +0800
Subject: [PATCH] arm: Makefile: boot.img use zImage

---
 arch/arm/Makefile | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 573a9b24fdff..ad743713bb5c 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -282,16 +282,11 @@ zkernel.img: zImage FORCE
 
 checkandroid: FORCE
 	$(Q)test -n "$(OUT)" || (echo "Run 'cd .. && . build/envsetup.sh && setpaths && cd -' first to build boot.img"; /bin/false)
-	$(Q)test -x $(srctree)/../out/host/linux-x86/bin/mkbootimg || (echo "No $(srctree)/../out/host/linux-x86/bin/mkbootimg"; /bin/false)
+	$(Q)test -n "which mkbootimg" || (echo "No mkbootimg"; /bin/false)
 	$(Q)test -e $(OUT)/ramdisk.img || (echo "No $(OUT)/ramdisk.img"; /bin/false)
 
-PHONY += bootimg boot.img zbootimg zboot.img
-bootimg boot.img: Image checkandroid FORCE
-	$(Q)cp -a $(obj)/arch/arm/boot/Image $(OUT)/kernel
-	$(Q)mkbootimg --kernel $(OUT)/kernel --ramdisk $(OUT)/ramdisk.img --output $(OUT)/boot.img
-	@echo '  Image:  $(OUT)/boot.img is ready'
-
-zbootimg zboot.img: zImage checkandroid FORCE
+PHONY += bootimg boot.img
+bootimg boot.img: zImage checkandroid FORCE
 	$(Q)cp -a $(obj)/arch/arm/boot/zImage $(OUT)/kernel
 	$(Q)mkbootimg --kernel $(OUT)/kernel --ramdisk $(OUT)/ramdisk.img --output $(OUT)/boot.img
 	@echo '  Image:  $(OUT)/boot.img is ready'
@@ -301,8 +296,6 @@ define archhelp
   echo  '  zkernel.img   - Compressed Rockchip kernel image'
   echo  '  boot.img'
   echo  '  bootimg       - Android boot image'
-  echo  '  zboot.img'
-  echo  '  zbootimg      - Compressed Android boot image'
   echo  '* zImage        - Compressed kernel image (arch/$(ARCH)/boot/zImage)'
   echo  '  Image         - Uncompressed kernel image (arch/$(ARCH)/boot/Image)'
   echo  '* xipImage      - XIP kernel image, if configured (arch/$(ARCH)/boot/xipImage)'
-- 
2.35.3

