From 8529d554c34f79eed1197f3620706d0341f9f20e Mon Sep 17 00:00:00 2001
From: Xiao Feng <xf@rock-chips.com>
Date: Mon, 27 Jul 2015 14:28:02 +0800
Subject: [PATCH] dvfs: rockchip: print error when clk is disable

Signed-off-by: Xiao Feng <xf@rock-chips.com>
---
 arch/arm/mach-rockchip/dvfs.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/dvfs.c b/arch/arm/mach-rockchip/dvfs.c
index 6f65f9f4e6f5..351144dfee0c 100644
--- a/arch/arm/mach-rockchip/dvfs.c
+++ b/arch/arm/mach-rockchip/dvfs.c
@@ -1671,8 +1671,11 @@ static int dvfs_target(struct dvfs_node *clk_dvfs_node, unsigned long rate)
 	if (!clk)
 		return -EINVAL;
 
-	if (!clk_dvfs_node->enable_count)
+	if (!clk_dvfs_node->enable_count) {
+		DVFS_ERR("%s: %s is disable, set rate error\n",
+			 __func__, clk_dvfs_node->name);
 		return 0;
+	}
 	
 	if (clk_dvfs_node->vd->volt_set_flag == DVFS_SET_VOLT_FAILURE) {
 		/* It means the last time set voltage error */
-- 
2.35.3

