From 47e09bd9fefd05eda228c0753d4b58117182747e Mon Sep 17 00:00:00 2001
From: xxx <xxx@rock-chips.com>
Date: Sat, 16 May 2015 17:36:14 +0800
Subject: [PATCH] efuse: change PSCI_OS_ACCESS_REG to 0xa600ffb0

---
 arch/arm/mach-rockchip/efuse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/efuse.c b/arch/arm/mach-rockchip/efuse.c
index b979e411dcd4..7311d5918334 100644
--- a/arch/arm/mach-rockchip/efuse.c
+++ b/arch/arm/mach-rockchip/efuse.c
@@ -67,7 +67,7 @@ static struct rockchip_efuse efuse;
 #define SEC_REG_WR_32 (SEC_REG_WR | SEC_REG_32)
 #define SEC_REG_WR_64 (SEC_REG_WR | SEC_REG_64)
 
-#define PSCI_OS_ACCESS_REG		(0xba00ffb0)
+#define PSCI_OS_ACCESS_REG		(0xa600ffb0)
 #define EFUSE_BASE        0xffb00000
 /*
  * arg2: rd/wr control, bit[0] 0-rd 1-rt, bit[1] 0-32bit, 1-64bit
-- 
2.35.3

