From abc9d38d73ae0573d489c004d2db6d6851293331 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 13 Jan 2011 16:48:21 +0800
Subject: [PATCH] rk29: clock: always enable hclk/pclk

---
 arch/arm/mach-rk29/clock.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 33bec46d9565..25eb8135cfbb 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2249,6 +2249,10 @@ static void __init rk29_clock_common_init(void)
 
 static void __init clk_enable_init_clocks(void)
 {
+	clk_enable_nolock(&hclk_cpu);
+	clk_enable_nolock(&pclk_cpu);
+	clk_enable_nolock(&hclk_periph);
+	clk_enable_nolock(&pclk_periph);
 	clk_enable_nolock(&clk_nandc);
 	clk_enable_nolock(&clk_aclk_cpu_peri);
 	clk_enable_nolock(&clk_aclk_ddr_peri);
-- 
2.35.3

