From 1079596946a5d93be9da8b6de2e480673020dae3 Mon Sep 17 00:00:00 2001
From: zhanghao <zhanghao@rock-chips.com>
Date: Mon, 4 Jun 2012 09:43:52 +0800
Subject: [PATCH] WIMO : Kconfig default n

---
 drivers/video/rockchip/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/Kconfig b/drivers/video/rockchip/Kconfig
index 83ba332188c3..79cbc113595c 100644
--- a/drivers/video/rockchip/Kconfig
+++ b/drivers/video/rockchip/Kconfig
@@ -6,10 +6,11 @@ config FB_ROCKCHIP
         select FB_CFB_IMAGEBLIT
         ---help---
           Framebuffer driver for rockchip based platform
+
 config FB_WIMO
 	bool "WiMO support"
 	depends on FB_ROCKCHIP
-	default y
+	default n
 	help
 	    Support wimo
 
-- 
2.35.3

