From 3f35c77fd3e1f3d82e350fec82f9f7bbaf6fc738 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 25 Jul 2012 10:51:57 +0800
Subject: [PATCH] do not call rk_release_fb_buffer in rk_fb_unregister

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index f3b03c20bc72..4ad730ad7043 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1020,7 +1020,7 @@ int rk_fb_unregister(struct rk_lcdc_device_driver *dev_drv)
 	{
 		fbi = fb_inf->fb[i];
 		unregister_framebuffer(fbi);
-		rk_release_fb_buffer(fbi);
+		//rk_release_fb_buffer(fbi);
 		framebuffer_release(fbi);	
 	}
 	fb_inf->lcdc_dev_drv[dev_drv->id]= NULL;
-- 
2.35.3

