From 1422cd135d49d988086d61dcfcb5fc4ad81ee7d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 2 Aug 2013 10:29:04 +0800
Subject: [PATCH] rk: ipp: only avaliable on arch rk29/rk30/rk3188

---
 drivers/staging/rk29/ipp/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/rk29/ipp/Kconfig b/drivers/staging/rk29/ipp/Kconfig
index 8f93652870e8..13acbc79fecd 100755
--- a/drivers/staging/rk29/ipp/Kconfig
+++ b/drivers/staging/rk29/ipp/Kconfig
@@ -1,6 +1,7 @@
 menu "IPP"
 config RK29_IPP
 	tristate "ROCKCHIP RK29 IPP"
+	depends on ARCH_RK29 || ARCH_RK30 || ARCH_RK3188
 	default m
 	help
           rk29 ipp module.	
-- 
2.35.3

