From 2c69da86146efeb05044fee9704594ecd97fe6cf Mon Sep 17 00:00:00 2001
From: zwp <zwp@rock-chips.com>
Date: Wed, 17 Aug 2011 19:33:52 -0700
Subject: [PATCH] phone-pad:modify pack script.

---
 pack-kernel.sh | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/pack-kernel.sh b/pack-kernel.sh
index 9efa97fd2628..83462cecc1f3 100755
--- a/pack-kernel.sh
+++ b/pack-kernel.sh
@@ -9,6 +9,7 @@ arch/arm/mach-rk29/vpu.c
 arch/arm/mach-rk29/vpu_mem.c
 
 drivers/staging/rk29/ipp/rk29-ipp.c
+sound/soc/codecs/rt5625.c
 )
 
 DIRS=(
@@ -88,6 +89,11 @@ for d in ${DIRS[@]}; do
 	[ -d $d ] && find $d -type f -name '*.uu' -print0 | xargs -0 rm -f
 done
 
+#zwp
+mv sound/soc/codecs/rt5625.c sound/soc/codecs/rt5625.c.bak
+mv sound/soc/codecs/rt5625.c.enc sound/soc/codecs/rt5625.c
+####
+
 echo build kernel on $kerndir with $defconfig
 make clean >/dev/null 2>&1
 make $defconfig >/dev/null 2>&1
@@ -109,6 +115,11 @@ for d in ${DIRS[@]}; do
 	done
 done
 
+#zwp
+mv sound/soc/codecs/rt5625.c sound/soc/codecs/rt5625.c.enc
+mv sound/soc/codecs/rt5625.c.bak sound/soc/codecs/rt5625.c
+####
+
 make distclean >/dev/null 2>&1
 
 popd >/dev/null
@@ -124,6 +135,8 @@ done
 for file in ${EXCLUDES[@]}; do
 	echo "$file" >> $ex
 done
+echo sound/soc/codecs/rt5625.c >>$ex
+echo sound/soc/codecs/rt5625.c.enc >>$ex
 echo TAR $(pwd)/$package
 tar cf $package --numeric-owner --exclude-from $ex --exclude=.git --exclude=`basename $0` $(basename $kerndir)
 tar rf $package --numeric-owner --exclude=.git toolchain/arm-eabi-4.4.0
-- 
2.35.3

