From 853f8972f0d8a5080e67340130862675d8e0e3a0 Mon Sep 17 00:00:00 2001
From: hxy <hxy@rock-chips.com>
Date: Fri, 24 Sep 2010 15:46:33 +0800
Subject: [PATCH] protect handle before calling regulator_set_current_limit

---
 drivers/power/rk2818_battery.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/rk2818_battery.c b/drivers/power/rk2818_battery.c
index df151d1f4c7a..fad822e27c0d 100755
--- a/drivers/power/rk2818_battery.c
+++ b/drivers/power/rk2818_battery.c
@@ -159,7 +159,7 @@ static int rk2818_get_charge_status(void)
     {
         DBG("CHARGE!\n");
         if(gVbuscharge !=1) {
-            if(rdev== pChargeregulator )
+            if(!IS_ERR(rdev))
                 regulator_set_current_limit(rdev,0,1200000);
         }
         gVbuscharge = 1;
@@ -169,7 +169,7 @@ static int rk2818_get_charge_status(void)
     {
         DBG("NOT CHARGING!\n");
         if(gVbuscharge !=0 ) {
-            if(rdev== pChargeregulator )
+            if(!IS_ERR(rdev))
             regulator_set_current_limit(rdev,0,475000);     
         }
         gVbuscharge = 0;
-- 
2.35.3

