From 541a6baf748d0c5c0e63239c31f0382273f70458 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 22 Nov 2011 17:09:57 +0800
Subject: [PATCH] earlysuspend: revert to default version

---
 kernel/power/earlysuspend.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/kernel/power/earlysuspend.c b/kernel/power/earlysuspend.c
index e073695d44f5..b15f02eba45c 100644
--- a/kernel/power/earlysuspend.c
+++ b/kernel/power/earlysuspend.c
@@ -20,7 +20,6 @@
 #include <linux/syscalls.h> /* sys_sync */
 #include <linux/wakelock.h>
 #include <linux/workqueue.h>
-#include <linux/kallsyms.h>
 
 #include "power.h"
 
@@ -99,8 +98,6 @@ static void early_suspend(struct work_struct *work)
 		if (pos->suspend != NULL) {
 			if (debug_mask & DEBUG_VERBOSE)
 				pr_info("early_suspend: calling %pf\n", pos->suspend);
-			if (debug_mask & DEBUG_VERBOSE)
-				print_symbol("early_suspend: call %s\n", (unsigned long)pos->suspend);
 			pos->suspend(pos);
 		}
 	}
@@ -142,8 +139,6 @@ static void late_resume(struct work_struct *work)
 		if (pos->resume != NULL) {
 			if (debug_mask & DEBUG_VERBOSE)
 				pr_info("late_resume: calling %pf\n", pos->resume);
-			if (debug_mask & DEBUG_VERBOSE)
-				print_symbol("late_resume: call %s\n", (unsigned long)pos->resume);
 
 			pos->resume(pos);
 		}
-- 
2.35.3

