From 4abf8b29e830d823786f47c116178b9e74473009 Mon Sep 17 00:00:00 2001
From: guoyi <gy@rock-chips.com>
Date: Tue, 31 Mar 2015 11:54:22 +0800
Subject: [PATCH] rk3368-p9: dts : add core&gpu dvfs table

---
 arch/arm64/boot/dts/rk3368-p9_818.dts | 43 +++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368-p9_818.dts b/arch/arm64/boot/dts/rk3368-p9_818.dts
index 1192c7931896..fd91bf6e72bf 100755
--- a/arch/arm64/boot/dts/rk3368-p9_818.dts
+++ b/arch/arm64/boot/dts/rk3368-p9_818.dts
@@ -480,6 +480,49 @@
 	status = "disabled";
 };
 
+&clk_core_b_dvfs_table {
+	operating-points = <
+		/* KHz    uV */
+		216000 1000000
+		312000 1000000
+		408000 1000000
+		600000 1000000
+		696000 1000000
+		816000 1025000
+		1008000 1100000
+		//1200000 1175000
+		//1296000 1200000
+		//1416000 1275000
+		//1512000 1325000
+		>;
+	status = "okay";
+};
+
+&clk_core_l_dvfs_table {
+	operating-points = <
+		/* KHz    uV */
+		216000 1000000
+		312000 1000000
+		408000 1000000
+		600000 1000000
+		696000 1000000
+		816000 1075000
+		1008000 1175000
+		//1200000 1225000
+		>;
+	status = "okay";
+};
+
+&clk_gpu_dvfs_table {
+	operating-points = <
+		/* KHz    uV */
+		200000 1100000
+		288000 1100000
+		400000 1150000
+		576000 1200000
+		>;
+};
+
 &clk_ddr_dvfs_table {
 	operating-points = <
 		/* KHz    uV */
-- 
2.35.3

