From dda555085e44a9c280d41323ff2fa8451f64e55c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=88=98=E7=9B=8A=E6=98=9F?= <lyx@rock-chips.com>
Date: Tue, 1 Jun 2010 07:07:46 +0000
Subject: [PATCH] modify dm9000

---
 drivers/net/dm9000.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/dm9000.c b/drivers/net/dm9000.c
index 701599ac4c42..52e0bb94180e 100644
--- a/drivers/net/dm9000.c
+++ b/drivers/net/dm9000.c
@@ -147,7 +147,7 @@ typedef struct board_info {
 	}						\
 } while (0)
 
-#ifdef CONFIG_DM9000_USE_NAND_CONTROL
+#if defined(CONFIG_DM9000_USE_NAND_CONTROL) && defined(CONFIG_MTD_NAND_RK2818)
 extern void rk2818_nand_status_mutex_lock(void);
 extern	int rk2818_nand_status_mutex_trylock(void);
 extern	void rk2818_nand_status_mutex_unlock(void);
-- 
2.35.3

