From 524e42b0454e39aef10994d49fdc0a0b671c9149 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E8=BE=89?= <chenhui@rock-chips.com>
Date: Fri, 27 Jul 2012 12:13:41 +0800
Subject: [PATCH] fix sh run error: toolchain is changed in jellybean

---
 pack-kernel-common.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pack-kernel-common.sh b/pack-kernel-common.sh
index 723788e69113..bdfe1a47361a 100755
--- a/pack-kernel-common.sh
+++ b/pack-kernel-common.sh
@@ -82,7 +82,7 @@ for file in ${EXCLUDES[@]}; do
 done
 echo TAR $(pwd)/$package
 tar cf $package --numeric-owner --exclude-from $ex --exclude=.git $(basename $kerndir)
-tar rf $package --numeric-owner --exclude=.git prebuilt/linux-x86/toolchain/arm-eabi-4.4.0
+#tar rf $package --numeric-owner --exclude=.git prebuilt/linux-x86/toolchain/arm-eabi-4.4.0
 echo GZIP $(pwd)/$package.gz
 gzip -9 -c $package > $package.gz
 rm $ex
-- 
2.35.3

