From effda8e01b03e137877875372bdb590c43303000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E6=99=B4?= <zhangqing@rock-chips.com>
Date: Wed, 22 May 2013 16:43:35 +0800
Subject: [PATCH] rk29:act8931:modify act8931 power on control reg

---
 drivers/regulator/act8931.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/act8931.c b/drivers/regulator/act8931.c
index e3083ad07495..d5215f1cb49a 100755
--- a/drivers/regulator/act8931.c
+++ b/drivers/regulator/act8931.c
@@ -653,7 +653,7 @@ static int __devinit act8931_i2c_probe(struct i2c_client *i2c, const struct i2c_
 	ret = act8931_reg_read(act8931,0x01);
 	if (ret < 0)		
 			goto err;
-	ret = act8931_set_bits(act8931, 0x01,(0x1<<0),(0x1<<0));
+	ret = act8931_set_bits(act8931, 0x01,(0x1<<5) | (0x1<<0),(0x1<<0));
 	if (ret < 0) {
 		printk("act8931 set 0x01 error!\n");
 		goto err;
-- 
2.35.3

