From 8e9913738f3f0c76c9e70f65f54c8bb748204ef9 Mon Sep 17 00:00:00 2001
From: hhb <hhb@rock-chips.com>
Date: Tue, 11 Nov 2014 17:31:16 +0800
Subject: [PATCH] pl330: change NR_DEFAULT_DESC from 16 to 32

---
 drivers/dma/pl330.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/pl330.c b/drivers/dma/pl330.c
index 5a5614d83f34..819328b2e3ec 100755
--- a/drivers/dma/pl330.c
+++ b/drivers/dma/pl330.c
@@ -288,7 +288,7 @@ static unsigned cmd_line;
 
 /* The number of default descriptors */
 
-#define NR_DEFAULT_DESC	16
+#define NR_DEFAULT_DESC	32
 
 /* Populated by the PL330 core driver for DMA API driver's info */
 struct pl330_config {
-- 
2.35.3

