From 03d0ce412cbda6bdfd9c1586bac81d7c9564166c Mon Sep 17 00:00:00 2001
From: hwg <hwg@rock-chips.com>
Date: Tue, 24 Mar 2015 11:42:53 +0800
Subject: [PATCH] default disable gmac in rk3368-p9_818.dts

---
 arch/arm64/boot/dts/rk3368-p9_818.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/rk3368-p9_818.dts b/arch/arm64/boot/dts/rk3368-p9_818.dts
index 8b30ac018e14..c180283a6c64 100755
--- a/arch/arm64/boot/dts/rk3368-p9_818.dts
+++ b/arch/arm64/boot/dts/rk3368-p9_818.dts
@@ -249,6 +249,10 @@
 	status = "disabled";
 };
 
+&gmac {
+        status = "disabled";
+};
+
 &uart_dbg {
 	status = "okay";
 };
-- 
2.35.3

