From b704af97ffc466afbad60cde35a500b5d9a1871b Mon Sep 17 00:00:00 2001
From: lhh <lhh@rock-chips.com>
Date: Fri, 27 Aug 2010 14:29:16 +0800
Subject: [PATCH] rk2818 info sdk run 624

---
 arch/arm/mach-rk2818/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk2818/clock.c b/arch/arm/mach-rk2818/clock.c
index 62b30881d54d..4958ad24c006 100644
--- a/arch/arm/mach-rk2818/clock.c
+++ b/arch/arm/mach-rk2818/clock.c
@@ -1301,7 +1301,7 @@ static struct rk2818_freq_info rk2818_freqs[] = {
 //X	OP(768, 350, 768, 24, 1, 16,  41,  41),
 //	OP(720, 350, 720, 24, 1, 15,  41,  21),
 //	OP(672, 350, 672, 24, 1, 14,  41,  21),
-#if defined(CONFIG_MACH_INFO)
+#if defined(CONFIG_MACH_RK2818INFO)
 	OP(624, 350, 624, 24, 1, 13,  31,  21),
 #else
 	OP(600, 350, 600, 24, 1, 12,  41,  21),
-- 
2.35.3

