From 4c4098060bf9948c1c8162c372e7badf3cdbab4e Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Sat, 3 Nov 2012 16:07:38 +0800
Subject: [PATCH] rk2928 lcdc: freee gpio before swith to RGB IO

---
 drivers/video/rockchip/lcdc/rk2928_lcdc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk2928_lcdc.c b/drivers/video/rockchip/lcdc/rk2928_lcdc.c
index 9a016300e367..fe8cc8d9c1ba 100644
--- a/drivers/video/rockchip/lcdc/rk2928_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk2928_lcdc.c
@@ -52,6 +52,14 @@ static int rk2928_lcdc_iomux(rk_screen *screen,int mode)
 	int i=0;
 	if((screen->type == SCREEN_RGB)&&(mode)) //iomux for RGB screen
 	{
+		for(i=0;i<8;i++)
+		{
+			gpio_free(RK2928_PIN2_PB0 + i); 
+		}
+		for(i=0;i<4;i++)
+		{
+			gpio_free(RK2928_PIN2_PC0 + i);
+		}
 
 		if(screen->lcdc_id == 0)
 		{
-- 
2.35.3

