From c30d0f1f2268c204b078e10e365f73b043fffd57 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 3 Feb 2012 16:17:58 +0800
Subject: [PATCH] hdmi: fix Makefile, build hdmi driver depend on config option

---
 drivers/video/hdmi/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/hdmi/Makefile b/drivers/video/hdmi/Makefile
index 89a49e5a6e7f..1b59c50ab27f 100755
--- a/drivers/video/hdmi/Makefile
+++ b/drivers/video/hdmi/Makefile
@@ -1,2 +1,2 @@
-obj-y       += hdmi-core.o hdmi-sysfs.o hdmi-fb.o hdmi-codec.o hdmi-backlight.o
-obj-y       += chips/
+obj-$(CONFIG_HDMI)	+= hdmi-core.o hdmi-sysfs.o hdmi-fb.o hdmi-codec.o hdmi-backlight.o
+obj-$(CONFIG_HDMI)	+= chips/
-- 
2.35.3

