From 2b617362c72235c5677714f36bc22212a31f1092 Mon Sep 17 00:00:00 2001
From: wdc <wdc@rock-chips.com>
Date: Thu, 20 Feb 2014 19:49:55 +0800
Subject: [PATCH] keys: depend on rockchip adc config

---
 drivers/input/keyboard/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/Kconfig b/drivers/input/keyboard/Kconfig
index 84646dcc6aa4..d554586fc184 100644
--- a/drivers/input/keyboard/Kconfig
+++ b/drivers/input/keyboard/Kconfig
@@ -14,7 +14,7 @@ if INPUT_KEYBOARD
 
 config KEYS_RK
         tristate "rk keyboard"
-        depends on RK_ADC 
+        depends on ROCKCHIP_ADC 
         default y
         help
           rk keyboard drivers(gpio and adc)
-- 
2.35.3

