From 14287817144a0c09f7a014b50c8d5b8aebaee2f0 Mon Sep 17 00:00:00 2001
From: chenxing <chenxing@rock-chips.com>
Date: Fri, 2 Aug 2013 17:49:08 +0800
Subject: [PATCH] rk3188: fix cpu_div default set error

---
 arch/arm/mach-rk3188/clock_data.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-rk3188/clock_data.c b/arch/arm/mach-rk3188/clock_data.c
index e471662c7045..ec17d0994787 100755
--- a/arch/arm/mach-rk3188/clock_data.c
+++ b/arch/arm/mach-rk3188/clock_data.c
@@ -3385,6 +3385,7 @@ static void cpu_axi_init(void)
 			break;
 
 		default:
+			cpu_div_rate = 150 * MHZ;
 			aclk_cpu_rate = 150 * MHZ;
 			hclk_cpu_rate = 150 * MHZ;
 			pclk_cpu_rate = 75 * MHZ;
-- 
2.35.3

