From c3909f03090b1c84e5d08afb387eb0ec1e1146f8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 18 Jan 2011 16:27:49 +0800
Subject: [PATCH] rk29: gic: add fake set_wake

---
 arch/arm/common/gic.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/common/gic.c b/arch/arm/common/gic.c
index 458a9ea676fc..3ff84676ace7 100755
--- a/arch/arm/common/gic.c
+++ b/arch/arm/common/gic.c
@@ -156,6 +156,13 @@ static void gic_handle_cascade_irq(unsigned int irq, struct irq_desc *desc)
 	chip->unmask(irq);
 }
 
+#if defined(CONFIG_PM) && defined(CONFIG_ARCH_RK29)
+static int gic_set_wake(unsigned int irq, unsigned int on)
+{
+	return 0;
+}
+#endif
+
 static struct irq_chip gic_chip = {
 	.name		= "GIC",
 	.ack		= gic_ack_irq,
@@ -164,6 +171,9 @@ static struct irq_chip gic_chip = {
 #ifdef CONFIG_SMP
 	.set_affinity	= gic_set_cpu,
 #endif
+#if defined(CONFIG_PM) && defined(CONFIG_ARCH_RK29)
+	.set_wake	= gic_set_wake,
+#endif
 };
 
 void __init gic_cascade_irq(unsigned int gic_nr, unsigned int irq)
-- 
2.35.3

