From 714e375ad8125485b86ac0b5d9cda58b57ae9801 Mon Sep 17 00:00:00 2001
From: zyc <zyc@rock-chips.com>
Date: Wed, 10 Sep 2014 17:27:55 +0800
Subject: [PATCH] rk312x : delay some time when sensor resolution is changed
 for gc2035.

---
 drivers/media/video/gc2035.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/video/gc2035.c b/drivers/media/video/gc2035.c
index d5d8971c5050..c39ed4e7f410 100755
--- a/drivers/media/video/gc2035.c
+++ b/drivers/media/video/gc2035.c
@@ -1259,6 +1259,8 @@ static int sensor_s_fmt_cb_th(struct i2c_client *client,struct v4l2_mbus_framefm
 */
 static int sensor_s_fmt_cb_bh (struct i2c_client *client,struct v4l2_mbus_framefmt *mf, bool capture)
 {
+	/* add delay for rk312x*/
+	msleep(300); 
 	return 0;
 }
 static int sensor_softrest_usr_cb(struct i2c_client *client,struct rk_sensor_reg *series)
-- 
2.35.3

