From 91e4ee6d052f27c6ae935dfa6fcbeac17e40ac33 Mon Sep 17 00:00:00 2001
From: yj <yangjie@rock-chips.com>
Date: Sat, 29 Mar 2014 21:12:09 +0800
Subject: [PATCH] TP_SUSPEND: mod compiler warning

---
 drivers/input/touchscreen/tp_suspend.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/tp_suspend.h b/drivers/input/touchscreen/tp_suspend.h
index 4ea2322033af..68e3d58fec17 100644
--- a/drivers/input/touchscreen/tp_suspend.h
+++ b/drivers/input/touchscreen/tp_suspend.h
@@ -56,7 +56,7 @@ static int fb_notifier_callback(struct notifier_block *self,
 
 	if (ret < 0)
 	{
-		printk("TP_notifier_callback error action=%x,blank_mode=%x\n",action,blank_mode);
+		printk("TP_notifier_callback error action=%x,blank_mode=%x\n",(int)action,blank_mode);
 		return ret;
 	}
 
@@ -76,4 +76,4 @@ static void tp_unregister_fb(struct tp_device *tp)
 {
 	fb_unregister_client(&tp->fb_notif);
 }
-#endif _RK_TP_SUSPEND_H
+#endif
-- 
2.35.3

