From 33a63cd97ff2d2f9c07f88e510a7bc6ecfa4b962 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 23 Oct 2014 18:28:06 +0800
Subject: [PATCH] rk30_factory_adc_battery: fix compilation error on arm64

---
 drivers/power/rk30_factory_adc_battery.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/power/rk30_factory_adc_battery.c b/drivers/power/rk30_factory_adc_battery.c
index 5b4d36cdb984..2e3cbde22172 100755
--- a/drivers/power/rk30_factory_adc_battery.c
+++ b/drivers/power/rk30_factory_adc_battery.c
@@ -22,9 +22,6 @@
 #include <linux/pci.h>
 #include <linux/interrupt.h>
 #include <asm/io.h>
-#include <asm/mach-types.h>
-#include <asm/mach/arch.h>
-#include <asm/mach/map.h>
 #include <linux/adc.h>
 #include <linux/delay.h>
 #include <linux/ktime.h>
-- 
2.35.3

