From e8242f4e1b7bd23441123db1ad9eceaa931cec76 Mon Sep 17 00:00:00 2001
From: zwl <zwl@rock-chips.com>
Date: Tue, 24 Jun 2014 10:58:17 +0800
Subject: [PATCH] rk fb: modify the judgement of vertical screen for hdmi

---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 3d539ca9ef5f..3c1b6c917ca9 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1524,7 +1524,7 @@ static void rk_fb_update_reg(struct rk_lcdc_driver *dev_drv,
 			ext_win->state = 0;
 		}
 
-		if (ext_win->area[0].xact < ext_win->area[0].yact) {
+		if (win->area[0].xact < win->area[0].yact) {
 			int pixel_width, vir_width_bit, stride;
 			ext_win->area[0].xact = win->area[0].yact;
 			ext_win->area[0].yact = win->area[0].xact;
-- 
2.35.3

