From 83e84ca2ed2a5b53f2febd6c5d82613dec3601fd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=99=88=E8=BE=89?= <chenhui@rock-chips.com>
Date: Wed, 11 Jul 2012 00:35:20 +0800
Subject: [PATCH] add arm-eabi-4.4.3 toolchain

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 1a0101c189e8..15702f930dd7 100644
--- a/Makefile
+++ b/Makefile
@@ -200,6 +200,9 @@ endif
 ifneq ($(wildcard ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.0),)
 CROSS_COMPILE	?= ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.0/bin/arm-eabi-
 endif
+ifneq ($(wildcard ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.3),)
+CROSS_COMPILE	?= ../prebuilt/linux-x86/toolchain/arm-eabi-4.4.3/bin/arm-eabi-
+endif
 CROSS_COMPILE	?= $(CONFIG_CROSS_COMPILE:"%"=%)
 
 # Architecture as present in compile.h
-- 
2.35.3

