From e16d1e10dde4cded817b13835a5c12fc5fafd9f7 Mon Sep 17 00:00:00 2001
From: linjh <linjh@rock-chips.com>
Date: Tue, 24 May 2011 10:35:19 +0800
Subject: [PATCH] RK29SmartPhone: power down when gps probe

---
 drivers/misc/gps/rk29_gps.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/misc/gps/rk29_gps.c b/drivers/misc/gps/rk29_gps.c
index 09eedb74eb12..6c876a2c2449 100644
--- a/drivers/misc/gps/rk29_gps.c
+++ b/drivers/misc/gps/rk29_gps.c
@@ -271,6 +271,10 @@ static int rk29_gps_probe(struct platform_device *pdev)
 	INIT_WORK(&pdata->work, rk29_gps_delay_power_downup);
 	pdata->power_flag = 0;
 
+	//gps power down
+	if (pdata->power_down)
+		pdata->power_down();
+
 	pgps = pdata;
 
 
-- 
2.35.3

