From 36253dbff8d31500fe0ed4073b860c7611c979a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 28 Dec 2010 10:25:02 +0800
Subject: [PATCH] rk29: clock: default enable intmem/emem

---
 arch/arm/mach-rk29/clock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-rk29/clock.c b/arch/arm/mach-rk29/clock.c
index 383781399f8e..b3bd131335e2 100755
--- a/arch/arm/mach-rk29/clock.c
+++ b/arch/arm/mach-rk29/clock.c
@@ -2239,6 +2239,8 @@ static void __init clk_enable_init_clocks(void)
 	clk_enable_nolock(&clk_gic);
 	clk_enable_nolock(&clk_dma2);
 	clk_enable_nolock(&clk_dma1);
+	clk_enable_nolock(&clk_emem);
+	clk_enable_nolock(&clk_intmem);
 #ifdef CONFIG_DEBUG_LL
 	clk_enable_nolock(&clk_uart1);
 #endif
-- 
2.35.3

